--- contrib/python/pygit2/ya.make	(index)
+++ contrib/python/pygit2/ya.make	(working tree)
@@ -79,4 +79,11 @@ NO_COMPILER_WARNINGS()
 )
 
+RUN_PROGRAM(
+    contrib/python/cffi/py3/gen pygit2/_run.py:ffi pygit2._libgit2.cffi.py3.c
+    IN pygit2/_run.py pygit2/_build.py pygit2/decl/attr.h pygit2/decl/blame.h pygit2/decl/buffer.h pygit2/decl/callbacks.h pygit2/decl/checkout.h pygit2/decl/clone.h pygit2/decl/commit.h pygit2/decl/common.h pygit2/decl/config.h pygit2/decl/describe.h pygit2/decl/diff.h pygit2/decl/indexer.h pygit2/decl/errors.h pygit2/decl/graph.h pygit2/decl/index.h pygit2/decl/merge.h pygit2/decl/net.h pygit2/decl/oid.h pygit2/decl/options.h pygit2/decl/pack.h pygit2/decl/proxy.h pygit2/decl/refspec.h pygit2/decl/remote.h pygit2/decl/repository.h pygit2/decl/revert.h pygit2/decl/stash.h pygit2/decl/strarray.h pygit2/decl/submodule.h pygit2/decl/transaction.h pygit2/decl/transport.h pygit2/decl/types.h
+    OUT pygit2._libgit2.cffi.py3.c
+    OUTPUT_INCLUDES git2.h git2/sys/repository.h
+)
+
 RESOURCE_FILES(
     PREFIX contrib/python/pygit2/
