--- contrib/python/zeep/py3/zeep/__main__.py	(index)
+++ contrib/python/zeep/py3/zeep/__main__.py	(working tree)
@@ -35,7 +35,7 @@ def parse_arguments(args=None):
     return parser.parse_args(args)
 
 
-def main(args):
+def _main(args):
     if args.verbose:
         logging.config.dictConfig(
             {
@@ -87,6 +87,10 @@ def main(args):
     client.wsdl.dump()
 
 
-if __name__ == "__main__":
+def main():
     args = parse_arguments()
-    main(args)
+    _main(args)
+
+
+if __name__ == "__main__":
+    main()
