commit c1f51e027d160ba84e29c335229844feec09940c (HEAD -> SHMDUTY-61-fix)
author: lexx-evd
date: 2021-06-07T11:32:59+03:00

    +

--- a/src/libsodium/crypto_secretbox/crypto_secretbox_easy.c
+++ b/src/libsodium/crypto_secretbox/crypto_secretbox_easy.c
@@ -91,6 +91,7 @@ crypto_secretbox_open_detached(unsigned char *m, const unsigned char *c,
     unsigned long long i;
     unsigned long long mlen0;
 
+    memset(block0, 0U, crypto_secretbox_ZEROBYTES);
     crypto_core_hsalsa20(subkey, n, k, NULL);
     crypto_stream_salsa20(block0, crypto_stream_salsa20_KEYBYTES,
                           n + 16, subkey);
