--- contrib/python/trafilatura/trafilatura/settings.py	(index)
+++ contrib/python/trafilatura/trafilatura/settings.py	(working tree)
@@ -21,6 +21,7 @@ from lxml.etree import _Element, Element, XPath

 from .utils import line_processing

+import importlib.resources

 SUPPORTED_FMT_CLI = ["csv", "json", "html", "markdown", "txt", "xml", "xmltei"]
 SUPPORTED_FORMATS = set(SUPPORTED_FMT_CLI) | {"python"}  # for bare_extraction() only
@@ -36,7 +37,10 @@ def use_config(
         return config

     if filename is None:
-        filename = str(Path(__file__).parent / "settings.cfg")
+        config_str = (importlib.resources.files(__package__) / "settings.cfg").read_text()
+        config = ConfigParser()
+        config.read_string(config_str)
+        return config
     elif not Path(filename).is_file():
         raise FileNotFoundError("The given config file does not exist")

--- contrib/python/trafilatura/trafilatura/xml.py	(index)
+++ contrib/python/trafilatura/trafilatura/xml.py	(working tree)
@@ -19,12 +19,13 @@ from lxml.etree import (_Element, Element, SubElement, XMLParser,
 from .settings import Document, Extractor
 from .utils import sanitize, sanitize_tree, text_chars_test

+import importlib.resources

 LOGGER = logging.getLogger(__name__)
 PKG_VERSION = version("trafilatura")

 # validation
-TEI_SCHEMA = str(Path(__file__).parent / "data" / "tei_corpus.dtd")
+TEI_SCHEMA = importlib.resources.files(__package__) / "data" / "tei_corpus.dtd"
 TEI_VALID_TAGS = {'ab', 'body', 'cell', 'code', 'del', 'div', 'graphic', 'head', 'hi', \
                   'item', 'lb', 'list', 'p', 'quote', 'ref', 'row', 'table'}
 TEI_VALID_ATTRS = {'rend', 'rendition', 'role', 'target', 'type'}
@@ -241,7 +242,7 @@ def validate_tei(xmldoc: _Element) -> bool:

     if TEI_DTD is None:
         # https://tei-c.org/release/xml/tei/custom/schema/dtd/tei_corpus.dtd
-        TEI_DTD = DTD(TEI_SCHEMA)
+        TEI_DTD = DTD(TEI_SCHEMA.open())

     result = TEI_DTD.validate(xmldoc)
     if result is False:
