--- contrib/python/trafilatura/tests/cli_tests.py	(index)
+++ contrib/python/trafilatura/tests/cli_tests.py	(working tree)
@@ -23,8 +23,13 @@ from trafilatura import cli, cli_utils, spider, settings
 from trafilatura.downloads import add_to_compressed_dict, fetch_url
 from trafilatura.utils import LANGID_FLAG

+from .utils import IS_INTERNET_AVAILABLE
+
 logging.basicConfig(stream=sys.stdout, level=logging.DEBUG)
-RESOURCES_DIR = path.join(path.abspath(path.dirname(__file__)), "resources")
+
+import yatest.common as yc
+TEST_DIR = path.abspath(path.dirname(yc.source_path(__file__)))
+RESOURCES_DIR = path.join(TEST_DIR, 'resources')

 settings.MAX_FILES_PER_DIRECTORY = 1

@@ -118,6 +123,7 @@ def test_parser():
     )


+@pytest.mark.skip()
 def test_climain(capfd):
     """test arguments and main CLI entrypoint"""
     # exit status required: 0
@@ -154,6 +160,7 @@ def test_climain(capfd):
     assert captured.out.strip().endswith("in deep-red West Virginia.")


+@pytest.mark.skip("No folder docs")
 def test_input_type():
     """test input type errors"""
     testfile = "docs/trafilatura-demo.gif"
@@ -246,6 +253,7 @@ def test_sysoutput():
     assert filepath == "test/uOHdo6wKo4IK0pkL.txt"


+@pytest.mark.skipif(not IS_INTERNET_AVAILABLE, reason="No internet in CI")
 def test_download():
     """test page download and command-line interface"""
     assert cli_utils._define_exit_code([], 0) == 0
@@ -280,6 +288,7 @@ def test_download():


 # @patch('trafilatura.settings.MAX_FILES_PER_DIRECTORY', 1)
+@pytest.mark.skipif(not IS_INTERNET_AVAILABLE, reason="No internet in CI")
 def test_cli_pipeline():
     """test command-line processing pipeline"""
     # Force encoding to utf-8 for Windows in future processes spawned by multiprocessing.Pool
@@ -504,6 +513,7 @@ def test_input_filtering():
     assert url_store.get_known_domains() == ["https://test.info"]


+@pytest.mark.skipif(not IS_INTERNET_AVAILABLE, reason="No internet in CI")
 def test_crawling():
     "Test crawling and exploration functions."

@@ -565,6 +575,7 @@ def test_crawling():
     assert f.getvalue().strip() == "https://httpbun.com/html"


+@pytest.mark.skipif(not IS_INTERNET_AVAILABLE, reason="No internet in CI")
 def test_probing():
     "Test webpage probing functions."
     url = "https://example.org/"
--- contrib/python/trafilatura/tests/downloads_tests.py	(index)
+++ contrib/python/trafilatura/tests/downloads_tests.py	(working tree)
@@ -43,6 +43,7 @@ from trafilatura.downloads import (DEFAULT_HEADERS, HAS_PYCURL, USER_AGENT, Resp
 from trafilatura.settings import DEFAULT_CONFIG, args_to_extractor, use_config
 from trafilatura.utils import decode_file, handle_compressed_file, load_html

+from .utils import IS_INTERNET_AVAILABLE

 logging.basicConfig(stream=sys.stdout, level=logging.DEBUG)

@@ -50,7 +51,9 @@ ZERO_CONFIG = DEFAULT_CONFIG
 ZERO_CONFIG['DEFAULT']['MIN_OUTPUT_SIZE'] = '0'
 ZERO_CONFIG['DEFAULT']['MIN_EXTRACTED_SIZE'] = '0'

-RESOURCES_DIR = os.path.join(os.path.abspath(os.path.dirname(__file__)), 'resources')
+import yatest.common as yc
+TEST_DIR = os.path.abspath(os.path.dirname(yc.source_path(__file__)))
+RESOURCES_DIR = os.path.join(TEST_DIR, 'resources')
 UA_CONFIG = use_config(filename=os.path.join(RESOURCES_DIR, 'newsettings.cfg'))

 DEFAULT_OPTS = Extractor(config=DEFAULT_CONFIG)
@@ -92,6 +95,7 @@ def test_response_object():
     assert extract(response, url=response.url, config=ZERO_CONFIG) is None


+@pytest.mark.skipif(not IS_INTERNET_AVAILABLE, reason="No internet in CI")
 def test_is_live_page():
     '''Test if pages are available on the network.'''
     # is_live general tests
@@ -103,6 +107,7 @@ def test_is_live_page():
         assert _pycurl_is_live_page('https://httpbun.com/status/301') is True


+@pytest.mark.skipif(not IS_INTERNET_AVAILABLE, reason="No internet in CI")
 def test_fetch():
     '''Test URL fetching.'''
     # sanity check
@@ -241,6 +246,7 @@ def test_decode():
         assert handle_compressed_file(bad_file) == bad_file


+@pytest.mark.skipif(not IS_INTERNET_AVAILABLE, reason="No internet in CI")
 def test_queue():
     'Test creation, modification and download of URL queues.'
     # test conversion and storage
--- contrib/python/trafilatura/tests/feeds_tests.py	(index)
+++ contrib/python/trafilatura/tests/feeds_tests.py	(working tree)
@@ -19,10 +19,15 @@ from trafilatura.feeds import (
     probe_gnews,
 )

+import pytest
+
 logging.basicConfig(stream=sys.stdout, level=logging.DEBUG)

-TEST_DIR = os.path.abspath(os.path.dirname(__file__))
-RESOURCES_DIR = os.path.join(TEST_DIR, "resources")
+from .utils import IS_INTERNET_AVAILABLE
+
+import yatest.common as yc
+TEST_DIR = os.path.abspath(os.path.dirname(yc.source_path(__file__)))
+RESOURCES_DIR = os.path.join(TEST_DIR, 'resources')

 XMLDECL = '<?xml version="1.0" encoding="utf-8"?>\n'

@@ -159,6 +164,7 @@ def test_json_extraction():
     assert len(links) == 1


+@pytest.mark.skipif(not IS_INTERNET_AVAILABLE, reason="No internet in CI")
 def test_feeds_helpers():
     """Test helper functions for feed extraction"""
     params = FeedParameters("https://example.org", "example.org", "https://example.org")
@@ -308,6 +314,7 @@ def test_feeds_helpers():
     assert probe_gnews(params, None) is not None


+@pytest.mark.skipif(not IS_INTERNET_AVAILABLE, reason="No internet in CI")
 def test_cli_behavior():
     """Test command-line interface with respect to feeds"""
     testargs = ["", "--list", "--feed", "https://httpbun.com/xml"]
--- contrib/python/trafilatura/tests/sitemaps_tests.py	(index)
+++ contrib/python/trafilatura/tests/sitemaps_tests.py	(working tree)
@@ -13,12 +13,18 @@ from trafilatura import sitemaps
 from trafilatura.deduplication import is_similar_domain
 from trafilatura.utils import decode_file

+import pytest
+
 logging.basicConfig(stream=sys.stdout, level=logging.DEBUG)

-TEST_DIR = os.path.abspath(os.path.dirname(__file__))
+from .utils import IS_INTERNET_AVAILABLE
+
+import yatest.common as yc
+TEST_DIR = os.path.abspath(os.path.dirname(yc.source_path(__file__)))
 RESOURCES_DIR = os.path.join(TEST_DIR, 'resources')


+@pytest.mark.skipif(not IS_INTERNET_AVAILABLE, reason="No internet in CI")
 def test_search():
     '''Test search for sitemaps'''
     assert not sitemaps.sitemap_search('12345')
@@ -162,6 +168,7 @@ def test_extraction():
     assert (sitemap.sitemap_urls, sitemap.urls) == ([], ['https://test.org/en/1', 'https://test.org/en/2'])


+@pytest.mark.skipif(not IS_INTERNET_AVAILABLE, reason="No internet in CI")
 def test_robotstxt():
     '''Check if sitemaps can be found over robots.txt'''
     assert not sitemaps.find_robots_sitemaps('https://http.org')
@@ -172,6 +179,7 @@ def test_robotstxt():
     assert sitemaps.extract_robots_sitemaps('sitemap: https://example.org/sitemap.xml', baseurl) == ['https://example.org/sitemap.xml']


+@pytest.mark.skipif(not IS_INTERNET_AVAILABLE, reason="No internet in CI")
 def test_whole():
     "Test whole process."
     results = sitemaps.sitemap_search("https://www.sitemaps.org", target_lang="de", max_sitemaps=1)
--- contrib/python/trafilatura/tests/spider_tests.py	(index)
+++ contrib/python/trafilatura/tests/spider_tests.py	(working tree)
@@ -16,10 +16,12 @@ from trafilatura import spider  # for global variables

 # from trafilatura.utils import LANGID_FLAG

+from .utils import IS_INTERNET_AVAILABLE

 logging.basicConfig(stream=sys.stdout, level=logging.DEBUG)


+@pytest.mark.skipif(not IS_INTERNET_AVAILABLE, reason="No internet in CI")
 def test_redirections():
     "Test redirection detection."
     _, _, baseurl = spider.probe_alternative_homepage("xyz")
@@ -31,6 +33,7 @@ def test_redirections():
     # _, _, baseurl = spider.probe_alternative_homepage('https://httpbin.org/redirect-to?url=https%3A%2F%2Fhttpbin.org%2Fhtml&status_code=302')


+@pytest.mark.skip("No folder docs")
 def test_meta_redirections():
     "Test redirection detection using meta tag."

@@ -149,6 +152,7 @@ def test_process_links():
     assert spider.is_still_navigation(todo) is True


+@pytest.mark.skipif(not IS_INTERNET_AVAILABLE, reason="No internet in CI")
 def test_crawl_logic():
     "Test functions related to crawling sequence and consistency."
     url = "https://httpbun.com/html"
@@ -204,6 +208,7 @@ def test_crawl_logic():
     assert params.i == 0 and params.is_on and params.known_num == 2


+@pytest.mark.skipif(not IS_INTERNET_AVAILABLE, reason="No internet in CI")
 def test_crawl_page():
     "Test page-by-page processing."
     base_url = "https://httpbun.com"
@@ -250,6 +255,7 @@ def test_crawl_page():
     ## TODO: find a better page for language tests


+@pytest.mark.skipif(not IS_INTERNET_AVAILABLE, reason="No internet in CI")
 def test_focused_crawler():
     "Test the whole focused crawler mechanism."
     spider.URL_STORE = UrlStore()
@@ -262,6 +268,7 @@ def test_focused_crawler():
     # assert len(todo) == 1 and todo[0].startswith('https://httpbun.com/links/2')


+@pytest.mark.skipif(not IS_INTERNET_AVAILABLE, reason="No internet in CI")
 def test_robots():
     "Test robots.txt parsing"
     assert spider.get_rules("1234") is None
--- contrib/python/trafilatura/tests/unit_tests.py	(index)
+++ contrib/python/trafilatura/tests/unit_tests.py	(working tree)
@@ -38,7 +38,8 @@ from trafilatura.utils import (LANGID_FLAG, detect_encoding, is_dubious_html, is
 logging.basicConfig(stream=sys.stdout, level=logging.DEBUG)


-TEST_DIR = path.abspath(path.dirname(__file__))
+import yatest.common as yc
+TEST_DIR = path.abspath(path.dirname(yc.source_path(__file__)))
 RESOURCES_DIR = path.join(TEST_DIR, 'resources')
 SAMPLE_META = Document()

@@ -1377,7 +1378,8 @@ def test_large_doc_performance():
     start = time.time()
     extract(large_html, fast=False, config=ZERO_CONFIG)
     end = time.time()
-    assert end - start < 5, "Large document performance issue"
+    # Increase upper threshold due to internal import.resources config that takes time
+    assert end - start < 10, "Large document performance issue"


 def test_lang_detection():
