--- contrib/python/flake8/flake8/main/application.py	(index)
+++ contrib/python/flake8/flake8/main/application.py	(working tree)
@@ -228,8 +228,12 @@ class Application:
         try:
             self.file_checker_manager.run()
         except exceptions.PluginExecutionFailed as plugin_failed:
-            print(str(plugin_failed))
-            print("Run flake8 with greater verbosity to see more details")
+            try:
+                print(str(plugin_failed))
+                print("Run flake8 with greater verbosity to see more details")
+            except TypeError:
+                sys.stdout.write(str(plugin_failed).encode("utf-8"))
+                sys.stdout.write(b"\nRun flake8 with greater verbosity to see more details\n")
             self.catastrophic_failure = True
         LOG.info("Finished running")
         self.file_checker_manager.stop()
