--- contrib/python/timezonefinder/timezonefinder/hex_helpers.py	(index)
+++ contrib/python/timezonefinder/timezonefinder/hex_helpers.py	(working tree)
@@ -1,4 +1,5 @@
 import struct
+import importlib.resources
 from pathlib import Path
 from typing import Dict, List
 
@@ -28,35 +29,37 @@ def export_shortcuts_binary(global_mapping: Dict[int, List[int]], path2shortcuts
 
     """
     shortcut_space = 0
-    with open(path2shortcuts, "wb") as fp:
-        for hex_id, poly_ids in global_mapping.items():
-            fp.write(struct.pack(DTYPE_FORMAT_Q, hex_id))
-            nr_polys = len(poly_ids)
-            if nr_polys > THRES_DTYPE_B:
-                raise ValueError("value overflow: more polys than data type supports")
-            fp.write(struct.pack(DTYPE_FORMAT_B, nr_polys))
-            for poly_id in poly_ids:
-                fp.write(struct.pack(DTYPE_FORMAT_H, poly_id))
+    with importlib.resources.as_file(path2shortcuts) as f:
+        with open(f, "wb") as fp:
+            for hex_id, poly_ids in global_mapping.items():
+                fp.write(struct.pack(DTYPE_FORMAT_Q, hex_id))
+                nr_polys = len(poly_ids)
+                if nr_polys > THRES_DTYPE_B:
+                    raise ValueError("value overflow: more polys than data type supports")
+                fp.write(struct.pack(DTYPE_FORMAT_B, nr_polys))
+                for poly_id in poly_ids:
+                    fp.write(struct.pack(DTYPE_FORMAT_H, poly_id))
 
-            shortcut_space += NR_BYTES_Q + NR_BYTES_B + (len(poly_ids) * NR_BYTES_I)
+                shortcut_space += NR_BYTES_Q + NR_BYTES_B + (len(poly_ids) * NR_BYTES_I)
 
     return shortcut_space
 
 
 def read_shortcuts_binary(path2shortcuts: Path) -> ShortcutMapping:
     mapping: ShortcutMapping = {}
-    with open(path2shortcuts, "rb") as fp:
-        while 1:
-            try:
-                hex_id: int = struct.unpack(DTYPE_FORMAT_Q, fp.read(NR_BYTES_Q))[0]
-            except struct.error:
-                # EOF: buffer not long enough to unpack
-                break
-            nr_polys: int = struct.unpack(DTYPE_FORMAT_B, fp.read(NR_BYTES_B))[0]
-            poly_ids: np.ndarray = np.fromfile(
-                fp, dtype=DTYPE_FORMAT_H_NUMPY, count=nr_polys
-            )
-            mapping[hex_id] = poly_ids
+    with importlib.resources.as_file(path2shortcuts) as f:
+        with open(f, "rb") as fp:
+            while 1:
+                try:
+                    hex_id: int = struct.unpack(DTYPE_FORMAT_Q, fp.read(NR_BYTES_Q))[0]
+                except struct.error:
+                    # EOF: buffer not long enough to unpack
+                    break
+                nr_polys: int = struct.unpack(DTYPE_FORMAT_B, fp.read(NR_BYTES_B))[0]
+                poly_ids: np.ndarray = np.fromfile(
+                    fp, dtype=DTYPE_FORMAT_H_NUMPY, count=nr_polys
+                )
+                mapping[hex_id] = poly_ids
 
     return mapping
 
--- contrib/python/timezonefinder/timezonefinder/timezonefinder.py	(index)
+++ contrib/python/timezonefinder/timezonefinder/timezonefinder.py	(working tree)
@@ -1,4 +1,5 @@
 import json
+import importlib.resources
 from abc import ABC, abstractmethod
 from io import BytesIO
 from pathlib import Path
@@ -79,13 +80,13 @@ class AbstractTimezoneFinder(ABC):
         # open all the files in binary reading mode
         # for more info on what is stored in which .bin file, please read the comments in file_converter.py
         if bin_file_location is None:
-            bin_file_location = Path(__file__).parent
+            self.bin_file_location = importlib.resources.files(__package__)
         else:
-            bin_file_location = Path(bin_file_location)
-        self.bin_file_location: Path = bin_file_location
+            self.bin_file_location = Path(bin_file_location)
 
-        with open(self.bin_file_location / TIMEZONE_NAMES_FILE) as json_file:
-            self.timezone_names = json.loads(json_file.read())
+        with importlib.resources.as_file(self.bin_file_location / TIMEZONE_NAMES_FILE) as f:
+            with open(f) as json_file:
+                self.timezone_names = json.loads(json_file.read())
 
         path2shortcut_bin = self.bin_file_location / SHORTCUT_FILE
         self.shortcut_mapping = read_shortcuts_binary(path2shortcut_bin)
@@ -94,13 +95,15 @@ class AbstractTimezoneFinder(ABC):
             file_name = attribute_name + BINARY_FILE_ENDING
             path2file = self.bin_file_location / file_name
             if self.in_memory:
-                with open(path2file, mode="rb") as bin_file:
-                    bf_in_mem = BytesIO(bin_file.read())
-                    bf_in_mem.seek(0)
+                with importlib.resources.as_file(path2file) as f:
+                    with open(f, mode="rb") as bin_file:
+                        bf_in_mem = BytesIO(bin_file.read())
+                        bf_in_mem.seek(0)
                 setattr(self, attribute_name, bf_in_mem)
             else:
-                bin_file = open(path2file, mode="rb")
-                setattr(self, attribute_name, bin_file)
+                with importlib.resources.as_file(path2file) as fp:
+                    bin_file = open(fp, mode="rb")
+                    setattr(self, attribute_name, bin_file)
 
     def __del__(self):
         for attribute_name in self.binary_data_attributes:
@@ -329,8 +332,9 @@ class TimezoneFinder(AbstractTimezoneFinder):
 
         # stores for which polygons (how many) holes exits and the id of the first of those holes
         # since there are very few it is feasible to keep them in the memory
-        with open(self.bin_file_location / HOLE_REGISTRY_FILE) as json_file:
-            hole_registry_tmp = json.loads(json_file.read())
+        with importlib.resources.as_file(self.bin_file_location / HOLE_REGISTRY_FILE) as f:
+            with open(f) as json_file:
+                hole_registry_tmp = json.loads(json_file.read())
 
         # convert the json string keys to int
         hole_registry = {int(k): v for k, v in hole_registry_tmp.items()}
