diff --git a/contrib/python/pydantic-settings/tests/test_source_json.py b/contrib/python/pydantic-settings/tests/test_source_json.py
--- a/contrib/python/pydantic-settings/tests/test_source_json.py
+++ b/contrib/python/pydantic-settings/tests/test_source_json.py
@@ -145,13 +145,8 @@ def test_multiple_file_json_merge(tmp_path, deep_merge):
 class TestTraversableSupport:
     FILENAME = 'example_test_config.json'
 
-    @pytest.fixture(params=['importlib_resources', 'custom', 'custom_with_path'])
+    @pytest.fixture(params=['custom_with_path'])
     def json_config_path(self, request, tmp_path):
-        tests_package_dir = importlib.resources.files('tests')
-
-        if request.param == 'importlib_resources':
-            # get Traversable object using importlib.resources
-            return tests_package_dir / self.FILENAME
 
         # Create a custom Traversable implementation
         class CustomTraversable(Traversable):
@@ -186,9 +181,6 @@ class TestTraversableSupport:
             def joinpath(self, *descendants):
                 return CustomTraversable(self._path.joinpath(*descendants))
 
-        if request.param == 'custom':
-            custom_traversable = CustomTraversable(tests_package_dir)
-            return custom_traversable / self.FILENAME
 
         filepath = tmp_path / self.FILENAME
         with filepath.open('w') as f:
