--- a/include/boost/config/user.hpp	(index)
+++ b/include/boost/config/user.hpp	(working tree)
@@ -68,7 +68,7 @@
 // BOOST_DISABLE_ABI_HEADERS: Stops boost headers from including any 
 // prefix/suffix headers that normally control things like struct 
 // packing and alignment. 
-// #define BOOST_DISABLE_ABI_HEADERS
+#define BOOST_DISABLE_ABI_HEADERS
 
 // BOOST_ABI_PREFIX: A prefix header to include in place of whatever
 // boost.config would normally select, any replacement should set up 
