# Issue https://github.com/flask-restful/flask-restful/issues/963
--- contrib/python/Flask-RESTful/py3/flask_restful/reqparse.py
+++ contrib/python/Flask-RESTful/py3/flask_restful/reqparse.py
@@ -122,7 +122,11 @@ class Argument(object):
         else:
             values = MultiDict()
             for l in self.location:
-                value = getattr(request, l, None)
+                value = None
+                if l == 'json':
+                    value = request.get_json(silent=True)
+                else:
+                    value = getattr(request, l, None)
                 if callable(value):
                     value = value()
                 if value is not None:
