--- contrib/restricted/python/aioconsole/aioconsole/rlwrap.py	(index)
+++ contrib/restricted/python/aioconsole/aioconsole/rlwrap.py	(working tree)
@@ -21,6 +21,7 @@ def rlwrap_process(args, prompt_control, use_stderr=False):
         bufsize=0,
         universal_newlines=True,
         stdin=subprocess.PIPE,
+        env={"Y_PYTHON_ENTRY_POINT": ":main"},
         **{"stderr" if use_stderr else "stdout": subprocess.PIPE},
     )
     # Readline wrapping
