--- contrib/libs/opencv/modules/videoio/include/opencv2/videoio.hpp
+++ contrib/libs/opencv/modules/videoio/include/opencv2/videoio.hpp
@@ -853,6 +853,8 @@ public:
 
     The method first calls VideoCapture::release to close the already opened file or camera.
      */
+    CV_WRAP virtual bool open(const char* filename, int apiPreference, const std::vector<int>& params);
+
     CV_WRAP virtual bool open(const String& filename, int apiPreference, const std::vector<int>& params);
 
     /** @brief  Opens a camera for video capturing
--- contrib/libs/opencv/modules/videoio/src/cap.cpp
+++ contrib/libs/opencv/modules/videoio/src/cap.cpp
@@ -117,6 +117,11 @@ bool VideoCapture::open(const String& filename, int apiPreference)
     return open(filename, apiPreference, std::vector<int>());
 }
 
+bool VideoCapture::open(const char* filename, int apiPreference, const std::vector<int>& params)
+{
+    open(String(filename), apiPreference, params);
+}
+
 bool VideoCapture::open(const String& filename, int apiPreference, const std::vector<int>& params)
 {
     CV_INSTRUMENT_REGION();
