--- a/modules/core/include/opencv2/core/types_c.h
+++ b/modules/core/include/opencv2/core/types_c.h
@@ -44,6 +44,10 @@
 #ifndef OPENCV_CORE_TYPES_H
 #define OPENCV_CORE_TYPES_H

+#if !defined(__OPENCV_BUILD) && !defined(CV__DISABLE_C_API_CTORS) && !defined(_WIN32)
+#define CV__ENABLE_C_API_CTORS // enable C API ctors (must be removed)
+#endif
+
 #ifdef CV__ENABLE_C_API_CTORS  // invalid C API ctors (must be removed)
 #if defined(_WIN32) && !defined(CV__SKIP_MESSAGE_MALFORMED_C_API_CTORS)
 #error "C API ctors don't work on Win32: https://github.com/opencv/opencv/issues/15990"
