--- a/modules/features2d/include/opencv2/features2d.hpp
+++ b/modules/features2d/include/opencv2/features2d.hpp
@@ -83,6 +83,10 @@ This section describes approaches based on local 2D features and used to categor
   @}
  */
 
+using CvArr=void;
+struct CvSeq;
+struct CvMemStorage;
+
 namespace cv
 {
 
@@ -422,6 +426,45 @@ public:
     CV_WRAP virtual String getDefaultName() const CV_OVERRIDE;
 };
 
+/*!
+ Modified old version of Maximal Stable Regions Parameters
+*/
+struct CvMSERParams
+{
+    //! delta, in the code, it compares (size_{i}-size_{i-delta})/size_{i-delta}
+    int delta;
+    //! prune the area which smaller than minArea
+    int minArea;
+    //! prune the area which bigger than maxArea
+    int maxArea;
+    //! prune the area have simliar size to its children
+    float maxVariation;
+    //! trace back to cut off mser with diversity < min_diversity
+    float minDiversity;
+
+    /////// the next few params for MSER of color image
+
+    //! for color image, the evolution steps
+    int maxEvolution;
+    //! the area threshold to cause re-initialize
+    double areaThreshold;
+    //! ignore too small margin
+    double minMargin;
+    //! the aperture size for edge blur
+    int edgeBlurSize;
+
+    CvMSERParams(int delta = 5, int min_area = 60,
+        int max_area = 14400, float max_variation = .25f,
+        float min_diversity = .2f, int max_evolution = 200,
+        double area_threshold = 1.01,
+        double min_margin = .003,
+        int edge_blur_size = 5);
+
+};
+
+//// Extracts the contours of Maximally Stable Extremal Regions
+void ExtractOldMSER( CvArr* _img, CvArr* _mask, CvSeq** contours, CvMemStorage* storage, const CvMSERParams& params, bool modified_mser);
+
 /** @overload */
 CV_EXPORTS void FAST( InputArray image, CV_OUT std::vector<KeyPoint>& keypoints,
                       int threshold, bool nonmaxSuppression=true );


