--- a/modules/cudaoptflow/src/cuda/pyrlk.cu
+++ b/modules/cudaoptflow/src/cuda/pyrlk.cu
@@ -854,21 +855,9 @@
         }
     }
 
-    void loadWinSize(int* winSize, int* halfWinSize, cudaStream_t stream)
-    {
-        cudaSafeCall( cudaMemcpyToSymbolAsync(c_winSize_x, winSize, sizeof(int), 0, cudaMemcpyHostToDevice, stream) );
-        cudaSafeCall( cudaMemcpyToSymbolAsync(c_winSize_y, winSize + 1, sizeof(int), 0, cudaMemcpyHostToDevice, stream) );
-
-        cudaSafeCall( cudaMemcpyToSymbolAsync(c_halfWin_x, halfWinSize, sizeof(int), 0, cudaMemcpyHostToDevice, stream) );
-        cudaSafeCall( cudaMemcpyToSymbolAsync(c_halfWin_y, halfWinSize + 1, sizeof(int), 0, cudaMemcpyHostToDevice, stream) );
-    }
+#ifdef PYRLK_CN
 
-    void loadIters(int* iters, cudaStream_t stream)
-    {
-        cudaSafeCall( cudaMemcpyToSymbolAsync(c_iters, iters, sizeof(int), 0, cudaMemcpyHostToDevice, stream) );
-    }
-
-    void loadConstants(int2 winSize_, int iters_, cudaStream_t stream)
+    static void loadConstantsImpl(int2 winSize_, int iters_, cudaStream_t stream)
     {
         static int2 winSize = make_int2(0,0);
         if(winSize.x != winSize_.x || winSize.y != winSize_.y)
@@ -895,6 +884,117 @@
         }
     }
 
+#define DEFINE_LOAD_CONSTANTS_(TYPE, CN) \
+    void loadConstants_ ## TYPE ## _ ## CN (int2 winSize, int iters, cudaStream_t stream) { \
+        loadConstantsImpl(winSize, iters, stream); \
+    }
+
+#define DEFINE_LOAD_CONSTANTS(TYPE, CN) DEFINE_LOAD_CONSTANTS_(TYPE, CN)
+
+    DEFINE_LOAD_CONSTANTS(PYRLK_TYPE, PYRLK_CN);
+
+    static void loadWinSizeImpl(int* winSize, int* halfWinSize, cudaStream_t stream)
+    {
+        cudaSafeCall( cudaMemcpyToSymbolAsync(c_winSize_x, winSize, sizeof(int), 0, cudaMemcpyHostToDevice, stream) );
+        cudaSafeCall( cudaMemcpyToSymbolAsync(c_winSize_y, winSize + 1, sizeof(int), 0, cudaMemcpyHostToDevice, stream) );
+
+        cudaSafeCall( cudaMemcpyToSymbolAsync(c_halfWin_x, halfWinSize, sizeof(int), 0, cudaMemcpyHostToDevice, stream) );
+        cudaSafeCall( cudaMemcpyToSymbolAsync(c_halfWin_y, halfWinSize + 1, sizeof(int), 0, cudaMemcpyHostToDevice, stream) );
+    }
+
+#define DEFINE_LOAD_WIN_SIZE_(TYPE, CN) \
+    void loadWinSize_ ## TYPE ## _ ## CN (int* winSize, int* halfWinSize, cudaStream_t stream) { \
+        loadWinSizeImpl(winSize, halfWinSize, stream); \
+    }
+
+#define DEFINE_LOAD_WIN_SIZE(TYPE, CN) DEFINE_LOAD_WIN_SIZE_(TYPE, CN)
+
+    DEFINE_LOAD_WIN_SIZE(PYRLK_TYPE, PYRLK_CN);
+
+    static void loadItersImpl(int* iters, cudaStream_t stream)
+    {
+        cudaSafeCall( cudaMemcpyToSymbolAsync(c_iters, iters, sizeof(int), 0, cudaMemcpyHostToDevice, stream) );
+    }
+
+#define DEFINE_LOAD_ITERS_(TYPE, CN) \
+    void loadIters_ ## TYPE ## _ ## CN (int* iters, cudaStream_t stream) { \
+        loadItersImpl(iters, stream); \
+    }
+
+#define DEFINE_LOAD_ITERS(TYPE, CN) DEFINE_LOAD_ITERS_(TYPE, CN)
+
+    DEFINE_LOAD_ITERS(PYRLK_TYPE, PYRLK_CN);
+
+#else
+
+#define DECLARE_LOAD_CONSTANTS(TYPE) \
+    void loadConstants_ ## TYPE ## _1(int2 winSize, int iters, cudaStream_t stream); \
+    void loadConstants_ ## TYPE ## _3(int2 winSize, int iters, cudaStream_t stream); \
+    void loadConstants_ ## TYPE ## _4(int2 winSize, int iters, cudaStream_t stream);
+
+DECLARE_LOAD_CONSTANTS(uint8_t);
+DECLARE_LOAD_CONSTANTS(uint16_t);
+DECLARE_LOAD_CONSTANTS(int);
+DECLARE_LOAD_CONSTANTS(float);
+
+#define CALL_LOAD_CONSTANTS(TYPE) \
+    loadConstants_ ## TYPE ## _1(winSize, iters, stream); \
+    loadConstants_ ## TYPE ## _3(winSize, iters, stream); \
+    loadConstants_ ## TYPE ## _4(winSize, iters, stream);
+
+    void loadConstants(int2 winSize, int iters, cudaStream_t stream) {
+        CALL_LOAD_CONSTANTS(uint8_t);
+        CALL_LOAD_CONSTANTS(uint16_t);
+        CALL_LOAD_CONSTANTS(int);
+        CALL_LOAD_CONSTANTS(float);
+    }
+
+#define DECLARE_LOAD_WIN_SIZE(TYPE) \
+    void loadWinSize_ ## TYPE ## _1(int* winSize, int* halfWinSize, cudaStream_t stream); \
+    void loadWinSize_ ## TYPE ## _3(int* winSize, int* halfWinSize, cudaStream_t stream); \
+    void loadWinSize_ ## TYPE ## _4(int* winSize, int* halfWinSize, cudaStream_t stream);
+
+DECLARE_LOAD_WIN_SIZE(uint8_t);
+DECLARE_LOAD_WIN_SIZE(uint16_t);
+DECLARE_LOAD_WIN_SIZE(int);
+DECLARE_LOAD_WIN_SIZE(float);
+
+#define CALL_LOAD_WIN_SIZE(TYPE) \
+    loadWinSize_ ## TYPE ## _1(winSize, halfWinSize, stream); \
+    loadWinSize_ ## TYPE ## _3(winSize, halfWinSize, stream); \
+    loadWinSize_ ## TYPE ## _4(winSize, halfWinSize, stream);
+
+    void loadWinSize(int* winSize, int* halfWinSize, cudaStream_t stream) {
+        CALL_LOAD_WIN_SIZE(uint8_t);
+        CALL_LOAD_WIN_SIZE(uint16_t);
+        CALL_LOAD_WIN_SIZE(int);
+        CALL_LOAD_WIN_SIZE(float);
+    }
+
+#define DECLARE_LOAD_ITERS(TYPE) \
+    void loadIters_ ## TYPE ## _1(int* iters, cudaStream_t stream); \
+    void loadIters_ ## TYPE ## _3(int* iters, cudaStream_t stream); \
+    void loadIters_ ## TYPE ## _4(int* iters, cudaStream_t stream);
+
+DECLARE_LOAD_ITERS(uint8_t);
+DECLARE_LOAD_ITERS(uint16_t);
+DECLARE_LOAD_ITERS(int);
+DECLARE_LOAD_ITERS(float);
+
+#define CALL_LOAD_ITERS(TYPE) \
+    loadIters_ ## TYPE ## _1(iters, stream); \
+    loadIters_ ## TYPE ## _3(iters, stream); \
+    loadIters_ ## TYPE ## _4(iters, stream);
+
+    void loadIters(int* iters, cudaStream_t stream) {
+        CALL_LOAD_ITERS(uint8_t);
+        CALL_LOAD_ITERS(uint16_t);
+        CALL_LOAD_ITERS(int);
+        CALL_LOAD_ITERS(float);
+    }
+
+#endif
+
     template<typename T, int cn> struct pyrLK_caller
     {
         static void sparse(PtrStepSz<typename TypeVec<T, cn>::vec_type> I, PtrStepSz<typename TypeVec<T, cn>::vec_type> J, const float2* prevPts, float2* nextPts, uchar* status, float* err, int ptcount,
@@ -943,20 +1043,14 @@
         }
     };
 
-    template class pyrLK_caller<unsigned char,1>;
-    template class pyrLK_caller<unsigned short,1>;
-    template class pyrLK_caller<int,1>;
-    template class pyrLK_caller<float,1>;
-
-    template class pyrLK_caller<unsigned char, 3>;
-    template class pyrLK_caller<unsigned short, 3>;
-    template class pyrLK_caller<int, 3>;
-    template class pyrLK_caller<float, 3>;
-
-    template class pyrLK_caller<unsigned char, 4>;
-    template class pyrLK_caller<unsigned short, 4>;
-    template class pyrLK_caller<int, 4>;
-    template class pyrLK_caller<float, 4>;
+#ifdef PYRLK_CN
+
+#define DEFINE_PYRLK_CALLER(TYPE, CN) \
+    template class pyrLK_caller<TYPE, CN>;
+
+DEFINE_PYRLK_CALLER(PYRLK_TYPE, PYRLK_CN);
+
+#endif
 }
 
 #endif /* CUDA_DISABLER */
--- a/modules/cudaoptflow/src/cuda/pyrlk.float.1.cu
+++ b/modules/cudaoptflow/src/cuda/pyrlk.float.1.cu
@@ -0,0 +1,3 @@
+#define PYRLK_TYPE float
+#define PYRLK_CN 1
+#include "pyrlk.cu"
--- a/modules/cudaoptflow/src/cuda/pyrlk.float.3.cu
+++ b/modules/cudaoptflow/src/cuda/pyrlk.float.3.cu
@@ -0,0 +1,3 @@
+#define PYRLK_TYPE float
+#define PYRLK_CN 3
+#include "pyrlk.cu"
--- a/modules/cudaoptflow/src/cuda/pyrlk.float.4.cu
+++ b/modules/cudaoptflow/src/cuda/pyrlk.float.4.cu
@@ -0,0 +1,3 @@
+#define PYRLK_TYPE float
+#define PYRLK_CN 4
+#include "pyrlk.cu"
--- a/modules/cudaoptflow/src/cuda/pyrlk.int.1.cu
+++ b/modules/cudaoptflow/src/cuda/pyrlk.int.1.cu
@@ -0,0 +1,3 @@
+#define PYRLK_TYPE int
+#define PYRLK_CN 1
+#include "pyrlk.cu"
--- a/modules/cudaoptflow/src/cuda/pyrlk.int.3.cu
+++ b/modules/cudaoptflow/src/cuda/pyrlk.int.3.cu
@@ -0,0 +1,3 @@
+#define PYRLK_TYPE int
+#define PYRLK_CN 3
+#include "pyrlk.cu"
--- a/modules/cudaoptflow/src/cuda/pyrlk.int.4.cu
+++ b/modules/cudaoptflow/src/cuda/pyrlk.int.4.cu
@@ -0,0 +1,3 @@
+#define PYRLK_TYPE int
+#define PYRLK_CN 4
+#include "pyrlk.cu"
--- a/modules/cudaoptflow/src/cuda/pyrlk.u16.1.cu
+++ b/modules/cudaoptflow/src/cuda/pyrlk.u16.1.cu
@@ -0,0 +1,3 @@
+#define PYRLK_TYPE uint16_t
+#define PYRLK_CN 1
+#include "pyrlk.cu"
--- a/modules/cudaoptflow/src/cuda/pyrlk.u16.3.cu
+++ b/modules/cudaoptflow/src/cuda/pyrlk.u16.3.cu
@@ -0,0 +1,3 @@
+#define PYRLK_TYPE uint16_t
+#define PYRLK_CN 3
+#include "pyrlk.cu"
--- a/modules/cudaoptflow/src/cuda/pyrlk.u16.4.cu
+++ b/modules/cudaoptflow/src/cuda/pyrlk.u16.4.cu
@@ -0,0 +1,3 @@
+#define PYRLK_TYPE uint16_t
+#define PYRLK_CN 4
+#include "pyrlk.cu"
--- a/modules/cudaoptflow/src/cuda/pyrlk.u8.1.cu
+++ b/modules/cudaoptflow/src/cuda/pyrlk.u8.1.cu
@@ -0,0 +1,3 @@
+#define PYRLK_TYPE uint8_t
+#define PYRLK_CN 1
+#include "pyrlk.cu"
--- a/modules/cudaoptflow/src/cuda/pyrlk.u8.3.cu
+++ b/modules/cudaoptflow/src/cuda/pyrlk.u8.3.cu
@@ -0,0 +1,3 @@
+#define PYRLK_TYPE uint8_t
+#define PYRLK_CN 3
+#include "pyrlk.cu"
--- a/modules/cudaoptflow/src/cuda/pyrlk.u8.4.cu
+++ b/modules/cudaoptflow/src/cuda/pyrlk.u8.4.cu
@@ -0,0 +1,3 @@
+#define PYRLK_TYPE uint8_t
+#define PYRLK_CN 4
+#include "pyrlk.cu"
