--- contrib/libs/opencv/modules/core/src/system.cpp
+++ contrib/libs/opencv/modules/core/src/system.cpp
@@ -1315,11 +1315,13 @@ redirectError( ErrorCallback errCallback, void* userdata, void** prevUserdata)
     return prevCallback;
 }
 
+#ifdef CV_ERROR_SET_TERMINATE_HANDLER
 void terminate(int _code, const String& _err, const char* _func, const char* _file, int _line) CV_NOEXCEPT
 {
     dumpException(cv::Exception(_code, _err, _func, _file, _line));
     std::terminate();
 }
+#endif
 
 }
 
