--- a/src/include/fst/cache.h
+++ b/src/include/fst/cache.h
@@ -33,7 +33,7 @@
 #include <fst/vector-fst.h>

 #include <unordered_map>
-
+#include <atomic>
 DECLARE_bool(fst_default_cache_gc);
 DECLARE_int64(fst_default_cache_gc_limit);

@@ -207,7 +207,7 @@ class CacheState {
   int DecrRefCount() const { return --ref_count_; }

   // Used by the ArcIterator<Fst<Arc>> efficient implementation.
-  int *MutableRefCount() const { return &ref_count_; }
+  std::atomic_int *MutableRefCount() const { return &ref_count_; }

   // Used for state class allocation.
   void *operator new(size_t size, StateAllocator *alloc) {
@@ -234,7 +234,7 @@ class CacheState {
   size_t noepsilons_;                    // # of output epsilons.
   std::vector<Arc, ArcAllocator> arcs_;  // Arcs representation.
   mutable uint8_t flags_;
-  mutable int ref_count_;  // If 0, available for GC.
+  mutable std::atomic_int ref_count_;  // If 0, available for GC.
 };

 // Cache store, allocating and storing states, providing a mapping from state
--- a/src/include/fst/fst.h
+++ b/src/include/fst/fst.h
@@ -495,7 +495,7 @@ struct ArcIteratorData {
       base;         // Specialized iterator if non-null.
   const Arc *arcs;  // O.w. arcs pointer
   size_t narcs;     // ... and arc count.
-  int *ref_count;   // ... and a reference count of the `narcs`-length `arcs`
+  std::atomic_int *ref_count;   // ... and a reference count of the `narcs`-length `arcs`
                     //     array if non-null.
 };

--- a/src/include/fst/fstlib.h
+++ b/src/include/fst/fstlib.h
@@ -34,6 +34,10 @@
 //
 // This convenience file includes all other FST header files.

+#ifdef __clang__
+#pragma clang system_header
+#endif
+
 #ifndef FST_FSTLIB_H_
 #define FST_FSTLIB_H_

--- a/src/include/fst/script/draw.h
+++ b/src/include/fst/script/draw.h
@@ -15,6 +15,10 @@
 // See www.openfst.org for extensive documentation on this weighted
 // finite-state transducer library.

+#ifdef __clang__
+#pragma clang system_header
+#endif
+
 #ifndef FST_SCRIPT_DRAW_H_
 #define FST_SCRIPT_DRAW_H_

--- a/src/include/fst/symbol-table.h
+++ b/src/include/fst/symbol-table.h
@@ -17,6 +17,10 @@
 //
 // Classes to provide symbol-to-integer and integer-to-symbol mappings.

+#ifdef __clang__
+#pragma clang system_header
+#endif
+
 #ifndef FST_SYMBOL_TABLE_H_
 #define FST_SYMBOL_TABLE_H_

--- a/src/include/fst/vector-fst.h
+++ b/src/include/fst/vector-fst.h
@@ -363,6 +363,9 @@ class VectorFstImpl : public VectorFstBaseImpl<S> {

   explicit VectorFstImpl(const Fst<Arc> &fst);

+  //XXX: yandex specific: Add for new impl const attempts
+  bool SafeConvert(const Fst<Arc> &fst, size_t& max_steps);
+
   static VectorFstImpl *Read(std::istream &strm, const FstReadOptions &opts);

   void SetStart(StateId state) {
@@ -466,6 +469,35 @@ VectorFstImpl<S>::VectorFstImpl(const Fst<Arc> &fst) {
   SetProperties(fst.Properties(kCopyProperties, false) | kStaticProperties);
 }

+template <class Arc>
+bool VectorFstImpl<Arc>::SafeConvert(const Fst<Arc> &fst, size_t& max_steps) {
+  SetType("vector");
+  SetInputSymbols(fst.InputSymbols());
+  SetOutputSymbols(fst.OutputSymbols());
+  BaseImpl::SetStart(fst.Start());
+  if (fst.Properties(kExpanded, false))
+    BaseImpl::ReserveStates(CountStates(fst));
+
+  for (StateIterator< Fst<Arc> > siter(fst);
+       !siter.Done();
+       siter.Next()) {
+    StateId s = siter.Value();
+    BaseImpl::AddState();
+    BaseImpl::SetFinal(s, fst.Final(s));
+    ReserveArcs(s, fst.NumArcs(s));
+    for (ArcIterator< Fst<Arc> > aiter(fst, s);
+         !aiter.Done();
+         aiter.Next()) {
+      if (--max_steps == 0)
+          return false;
+      const Arc &arc = aiter.Value();
+      BaseImpl::AddArc(s, arc);
+    }
+  }
+  SetProperties(fst.Properties(kCopyProperties, false) | kStaticProperties);
+  return true;
+}
+
 template <class S>
 VectorFstImpl<S> *VectorFstImpl<S>::Read(std::istream &strm,
                                          const FstReadOptions &opts) {
@@ -547,6 +579,11 @@ class VectorFst : public ImplToMutableFst<internal::VectorFstImpl<S>> {
     return new VectorFst(*this, safe);
   }

+  //XXX: yandex specific: Add for new impl const attempts
+  Impl* GetVectorFstImpl() {
+    return GetMutableImpl();
+  }
+
   VectorFst &operator=(const VectorFst &) = default;
 
   VectorFst &operator=(VectorFst &&) noexcept;
--- a/src/lib/mapped-file.cc
+++ b/src/lib/mapped-file.cc
@@ -103,6 +103,7 @@ MappedFile *MappedFile::Map(std::istream *istrm, bool memorymap,
   if (memorymap) {
     LOG(WARNING) << "File mapping at offset " << spos << " of file " << source
                  << " could not be honored, reading instead";
+    return NULL;
   }
   // Reads the file into the buffer in chunks not larger than kMaxReadChunk.
   std::unique_ptr<MappedFile> mf(Allocate(size));
