--- contrib/libs/openfst/src/include/fst/extensions/far/sttable.h       (index)
+++ contrib/libs/openfst/src/include/fst/extensions/far/sttable.h       (working tree)
@@ -254,9 +254,9 @@ class STTableReader {
       strm->seekg(positions[mid]);
       std::string key;
       ReadType(*strm, &key);
-      if (key > find_key) {
+      if (std::string_view(key) > find_key) {
         high = mid;
-      } else if (key < find_key) {
+      } else if (std::string_view(key) < find_key) {
         low = mid + 1;
       } else {
         for (size_t i = mid; i > low; --i) {
