--- contrib/python/weasyprint/weasyprint/text/ffi.py    (index)
+++ contrib/python/weasyprint/weasyprint/text/ffi.py    (working tree)
@@ -445,6 +445,10 @@ def _dlopen(ffi, *names, allow_fail=False):
         flags = 0x00001000  # LOAD_LIBRARY_SEARCH_DEFAULT_DIRS
     else:
         flags = ffi.RTLD_NOW  # default
+
+    if getattr(sys, "is_standalone_binary", False):
+        with suppress(OSError):
+            return ffi.dlopen(None, flags)
     for name in names:
         with suppress(OSError):
             return ffi.dlopen(name, flags)
