--- a/tests/env_test.py
+++ b/tests/env_test.py
@@ -523,4 +523,5 @@ class OtherMethodsTest(unittest.TestCase):
         # Start a child, open a txn, then crash the child.
-        rc = os.spawnl(os.P_WAIT, sys.executable, sys.executable,
-                       __file__, 'test_reader_check_child', path)
+        rc = os.spawnle(os.P_WAIT, sys.executable, sys.executable,
+                        'test_reader_check_child', path,
+                        {'Y_PYTHON_ENTRY_POINT': __name__})
 
