RASPFRONT-7559, django translations

REVIEW: 1088490
REVIEW: 1081266

--- contrib/python/django/django-4/django/utils/translation/trans_real.py	(index)
+++ contrib/python/django/django-4/django/utils/translation/trans_real.py	(working tree)
@@ -20,4 +20,6 @@ from django.utils.safestring import SafeData, mark_safe
 from . import to_language, to_locale

+from library.python.resource import resfs_files
+
 # Translations are cached in a dictionary for every language.
 # The active translations are stored by threadid to make them thread local.
@@ -123,2 +125,17 @@ class TranslationCatalog:

+def get_app_module_locale_path(app_module):
+    """Yandex-specific: resfs locale path"""
+    try:
+        getfilename = app_module.__loader__.get_filename
+    except AttributeError:
+        return
+    module_file = getfilename(app_module.__name__)
+    path = os.path.join(os.path.dirname(module_file), "locale")
+
+    if not resfs_files(path):
+        path = os.path.join(app_module.__name__.replace(".", "/"), "locale")
+
+    return path
+
+
 class DjangoTranslation(gettext_module.GNUTranslations):
@@ -202,2 +219,4 @@ class DjangoTranslation(gettext_module.GNUTranslations):
-        settingsfile = sys.modules[settings.__module__].__file__
-        localedir = os.path.join(os.path.dirname(settingsfile), "locale")
+        # Yandex-specific: resfs locale path
+        settingsfile_module = sys.modules[settings.__module__]
+        localedir = get_app_module_locale_path(settingsfile_module)
+        # End of Yandex-specific
@@ -216,8 +235,12 @@ class DjangoTranslation(gettext_module.GNUTranslations):
                 "gettext calls at import time.")
         for app_config in app_configs:
-            localedir = os.path.join(app_config.path, "locale")
-            if os.path.exists(localedir):
-                translation = self._new_gnu_trans(localedir)
-                self.merge(translation)
+            # Yandex-specific: resfs locale path
+            localedir = get_app_module_locale_path(app_config.module)
+            if not localedir:
+                continue
+            # End of Yandex-specific
+
+            translation = self._new_gnu_trans(localedir)
+            self.merge(translation)

     def _add_local_translations(self):
--- contrib/python/django/django-4/django/views/i18n.py	(index)
+++ contrib/python/django/django-4/django/views/i18n.py	(working tree)
@@ -14,1 +14,1 @@ from django.utils.http import is_safe_url
-from django.utils.translation.trans_real import DjangoTranslation
+from django.utils.translation.trans_real import DjangoTranslation, get_app_module_locale_path
@@ -235,3 +235,13 @@ class JavaScriptCatalog(View):
         # paths of requested packages
-        return [os.path.join(app.path, "locale") for app in app_configs]
+
+        # Yandex-specific: resfs locale path
+        paths = []
+        for app_config in app_configs:
+            path = get_app_module_locale_path(app_config.module)
+            if not path:
+                continue
+
+            paths.append(path)
+
+        return paths

