--- contrib/libs/apache/arrow_next/cpp/src/generated/parquet_types.tcc	(index)
+++ contrib/libs/apache/arrow_next/cpp/src/generated/parquet_types.tcc	(working tree)
@@ -4071,7 +4071,9 @@ uint32_t ColumnIndex::read(Protocol_* iprot) {
             uint32_t _i272;
             for (_i272 = 0; _i272 < _size268; ++_i272)
             {
-              xfer += iprot->readBool(this->null_pages[_i272]);
+              bool tmp = this->null_pages[_i272];
+              xfer += iprot->readBool(tmp);
+              this->null_pages[_i272] = tmp;
             }
             xfer += iprot->readListEnd();
           }
