--- contrib/python/aiohttp-xmlrpc/aiohttp_xmlrpc/common.py	(index)
+++ contrib/python/aiohttp-xmlrpc/aiohttp_xmlrpc/common.py	(working tree)
@@ -2,6 +2,7 @@ import asyncio
 import base64
 import logging
 import os
+import importlib.resources
 from builtins import ValueError
 from datetime import datetime
 from functools import singledispatch, wraps
@@ -21,7 +22,7 @@ TIME_FORMATS = [TIME_FORMAT, "%Y%m%dT%H%M%S"]
 PY2XML_TYPES = {}
 XML2PY_TYPES = {}
 
-schema = etree.RelaxNG(file=os.path.join(CURRENT_DIR, "xmlrpc.rng"))
+schema = etree.RelaxNG(file=importlib.resources.open_binary(__package__, "xmlrpc.rng"))
 
 
 class Binary(bytes):
