--- contrib/python/docutils/py2/docutils/utils/__init__.py	(index)
+++ contrib/python/docutils/py2/docutils/utils/__init__.py	(working tree)
@@ -533,6 +533,9 @@ def find_file_in_dirs(path, dirs):
         else:
             d = os.path.expanduser(d)
             f = os.path.join(d, path)
+        # we use '//' cause resources inside contrib/python/docutils/docutils/ya.make start with
+        if f.startswith('//'):
+            return f
         if os.path.exists(f):
             return f
     return path
--- contrib/python/docutils/py2/docutils/writers/_html_base.py	(index)
+++ contrib/python/docutils/py2/docutils/writers/_html_base.py	(working tree)
@@ -21,6 +21,7 @@ import sys
 import re
 import sys
 import warnings
+import library.python.resource as rs
 
 import docutils
 from docutils import frontend, languages, nodes, utils, writers
@@ -74,9 +75,15 @@ class Writer(writers.Writer):
         self.output = self.apply_template()
 
     def apply_template(self):
-        template_file = open(self.document.settings.template, 'rb')
-        template = unicode(template_file.read(), 'utf-8')
-        template_file.close()
+        template = None
+        path = self.document.settings.template
+        # we use '//' cause resources inside contrib/python/docutils/docutils/ya.make start with
+        if path.startswith('//'):
+            template = unicode(rs.find(path), 'utf-8')
+        if template is None:
+            template_file = open(path, 'rb')
+            template = unicode(template_file.read(), 'utf-8')
+            template_file.close()
         subs = self.interpolation_dict()
         return template % subs
 
@@ -299,1 +306,5 @@ class HTMLTranslator(nodes.NodeVisitor):
-                content = docutils.io.FileInput(source_path=path,
+                # we use '//' cause resources inside contrib/python/docutils/docutils/ya.make start with
+                if path.startswith('//'):
+                    content = rs.find(path)
+                else:
+                    content = docutils.io.FileInput(source_path=path,
--- contrib/python/docutils/py2/docutils/writers/html4css1/__init__.py	(index)
+++ contrib/python/docutils/py2/docutils/writers/html4css1/__init__.py	(working tree)
@@ -26,8 +26,6 @@ class Writer(writers._html_base.Writer):
-    default_stylesheet_dirs = ['.',
-                               os.path.abspath(os.path.dirname(__file__)),
-                               os.path.abspath(os.path.join(
-                                   os.path.dirname(os.path.dirname(__file__)),
-                                   'html5_polyglot')) # for math.css
+    default_stylesheet_dirs = [
+        # we use '//' cause resources inside contrib/python/docutils/docutils/ya.make start with
+        '//docutils/writers/html4css1',
+        '//docutils/writers/html5_polyglot',
                               ]
-    default_template = os.path.join(
-        os.path.dirname(os.path.abspath(__file__)), 'template.txt')
+    default_template = '//docutils/writers/html4css1/template.txt'
--- contrib/python/docutils/py2/docutils/writers/html5_polyglot/__init__.py	(index)
+++ contrib/python/docutils/py2/docutils/writers/html5_polyglot/__init__.py	(working tree)
@@ -39,3 +39,2 @@ class Writer(writers._html_base.Writer):
-    default_stylesheet_dirs = ['.', os.path.abspath(os.path.dirname(__file__))]
+    default_stylesheet_dirs = ['//docutils/writers/html5_polyglot']
-    default_template = os.path.join(
-        os.path.dirname(os.path.abspath(__file__)), 'template.txt')
+    default_template = '//docutils/writers/html5_polyglot/template.txt'
--- contrib/python/docutils/py2/docutils/writers/latex2e/__init__.py	(index)
+++ contrib/python/docutils/py2/docutils/writers/latex2e/__init__.py	(working tree)
@@ -27,6 +27,7 @@ from docutils import frontend, nodes, languages, writers, utils, io
 from docutils.utils.error_reporting import SafeString
 from docutils.transforms import writer_aux
 from docutils.utils.math import pick_math_environment, unichar2tex
+import library.python.resource as rs
 
 if sys.version_info >= (3, 0):
     from urllib.request import url2pathname
@@ -34,7 +35,7 @@ class Writer(writers.Writer):
     """Formats this writer supports."""
 
     default_template = 'default.tex'
-    default_template_path = os.path.dirname(os.path.abspath(__file__))
+    default_template_path = '//docutils/writers/latex2e' # os.path.dirname(os.path.abspath(__file__))
     default_preamble = '\n'.join([r'% PDF Standard Fonts',
                                   r'\usepackage{mathptmx} % Times',
                                   r'\usepackage[scaled=.90]{helvet}',
@@ -573,9 +573,9 @@ def _read_block(fp):
         block.append(line)
     return ''.join(block).rstrip()
 
-_du_sty = os.path.join(os.path.dirname(os.path.abspath(__file__)),
-                      'docutils.sty')
-with open(_du_sty, encoding='utf8') as fp:
+import io
+_du_sty = "//docutils/writers/latex2e/docutils.sty"
+with io.StringIO(rs.find(_du_sty).decode('utf8')) as fp:
     for line in fp:
         line = line.strip('% \n')
         if not line.endswith('::'):
@@ -1410,7 +1411,11 @@ class LaTeXTranslator(nodes.NodeVisitor):
             if is_package:
                 path = base + '.sty' # ensure extension
             try:
-                content = docutils.io.FileInput(source_path=path,
+                # we use '//' cause resources inside contrib/python/docutils/docutils/ya.make start with
+                if path.startswith('//'):
+                    content = rs.find(path)
+                else:
+                    content = docutils.io.FileInput(source_path=path,
                                        encoding='utf-8').read()
                 self.settings.record_dependencies.add(path)
             except IOError as err:
