--- contrib/python/pytest-httpretty/pytest_httpretty.py	(index)
+++ contrib/python/pytest-httpretty/pytest_httpretty.py	(working tree)
@@ -12,14 +12,14 @@ def pytest_configure(config):
 
 
 def pytest_runtest_setup(item):
-    marker = item.get_marker('httpretty')
+    marker = item.get_closest_marker('httpretty')
     if marker is not None:
         httpretty.reset()
         httpretty.enable()
 
 
 def pytest_runtest_teardown(item, nextitem):
-    marker = item.get_marker('httpretty')
+    marker = item.get_closest_marker('httpretty')
     if marker is not None:
         httpretty.disable()
 
