From 2a0c8065c7048c48c1f7af71af8e8715c70604a5 Mon Sep 17 00:00:00 2001
From: ejqodbf6ix <yunweizu@ecton.com.cn>
Date: Thu, 9 Jun 2022 14:08:59 +0800
Subject: [PATCH] add to flask2.x version support

---
 flask_uwsgi_websocket/websocket.py | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/flask_uwsgi_websocket/websocket.py b/flask_uwsgi_websocket/websocket.py
index 248d56d..b6a868c 100644
--- a/flask_uwsgi_websocket/websocket.py
+++ b/flask_uwsgi_websocket/websocket.py
@@ -135,1 +135,4 @@ def add_url_rule(self, rule, endpoint=None, view_func=None, **options):
-        rule = Rule(rule, methods=methods, **options)
+        try:
+            rule = Rule(rule, methods=methods, websocket=True, **options)
+        except TypeError:
+            rule = Rule(rule, methods=methods, **options)
