--- contrib/python/xmlschema/xmlschema/extras/codegen.py	(index)
+++ contrib/python/xmlschema/xmlschema/extras/codegen.py	(working tree)
@@ -12,6 +12,7 @@
 This module contains abstact base class and helper
 functions for building XSD based code generators.
 """
+import importlib.resources
 import importlib.util
 import os
 import re
@@ -570,7 +571,7 @@ class PythonGenerator(AbstractGenerator):
 
     formal_language = 'Python'
 
-    searchpaths = ['templates/python/']
+    searchpaths = [importlib.resources.files(__package__) / "templates/python"]
 
     builtin_types = {
         'string': 'str',
--- contrib/python/xmlschema/xmlschema/loaders.py	(index)
+++ contrib/python/xmlschema/xmlschema/loaders.py	(working tree)
@@ -10,6 +10,7 @@
 import logging
 import warnings
 import functools
+from importlib.resources.abc import Traversable
 from operator import attrgetter
 from types import MappingProxyType
 from typing import Any, Optional, TYPE_CHECKING, cast
@@ -105,7 +106,7 @@ class SchemaLoader:
 
         if namespace in self.fallback_locations:
             values = self.fallback_locations[namespace]
-            if isinstance(values, str):
+            if isinstance(values, (str, Traversable)):
                 locations.append(values)
             else:
                 locations.extend(values)
@@ -144,7 +145,12 @@ class SchemaLoader:
                     continue
 
                 schema.imported_namespaces.append(namespace)
-                url = normalize_url(location, base_url) if location else None
+                if isinstance(base_url, Traversable) and location:
+                    url = base_url / location
+                elif isinstance(base_url, Traversable) and not location:
+                    url = None
+                else:
+                    url = normalize_url(location, base_url) if location else None
                 if self.is_missing(namespace, location, base_url):
                     self.import_namespace(schema, namespace, url)
 
--- contrib/python/xmlschema/xmlschema/resources/xml_resource.py	(index)
+++ contrib/python/xmlschema/xmlschema/resources/xml_resource.py	(working tree)
@@ -7,6 +7,8 @@
 #
 # @author Davide Brunato <brunato@sissa.it>
 #
+import importlib.resources
+from importlib.resources.abc import Traversable
 import io
 import os.path
 import threading
@@ -145,6 +147,10 @@ class XMLResource(XMLResourceLoader):
             raise XMLSchemaValueError(msg)
 
         # set and validate arguments
+        if isinstance(source, Traversable):
+            self.traversal_source = source
+            source = source.read_text()
+
         self.base_url = base_url
         self.allow = allow
         self.defuse = defuse
@@ -393,7 +399,9 @@ class XMLResource(XMLResourceLoader):
         """
         def open_url(url: str) -> IOType:
             try:
-                if self._opener is not None:
+                if isinstance(url, Traversable):
+                    return url.open()
+                elif self._opener is not None:
                     return cast(IOType, self._opener.open(url, timeout=self._timeout))
                 return cast(IOType, urlopen(url, timeout=self._timeout))
             except URLError as err:
--- contrib/python/xmlschema/xmlschema/validators/schemas.py	(index)
+++ contrib/python/xmlschema/xmlschema/validators/schemas.py	(working tree)
@@ -26,6 +26,7 @@ from pathlib import Path
 from urllib.request import OpenerDirector
 from xml.etree import ElementTree
 from xml.etree.ElementTree import Element
+from importlib.resources.abc import Traversable
 
 from elementpath import XPathToken, SchemaElementNode, build_schema_node_tree
 
@@ -98,12 +99,13 @@ class XMLSchemaMeta(ABCMeta):
             if not isinstance(meta_schema_file, str):
                 raise XMLSchemaTypeError("META_SCHEMA must be a string defining the "
                                          "location of the XSD meta-schema file")
+        elif isinstance(dict_.get('META_SCHEMA'), Traversable):
+            meta_schema_file = dict_.get('META_SCHEMA')
         elif isinstance(dict_.get('meta_schema'), str):
             meta_schema_file = dict_.pop('meta_schema')  # For backward compatibility
         else:
             meta_schema_file = None
-
-        if isinstance(meta_schema_file, str):
+        if isinstance(meta_schema_file, (str, Traversable)):
             try:
                 base_schemas = dict_.get('BASE_SCHEMAS', {})
             except KeyError as e:
@@ -845,16 +843,17 @@ class XMLSchemaBase(XsdValidator, ElementPathMixin[Union[SchemaType, XsdElement]
                 defuse='never',
                 partial=True,
             )
-
         for ns, location in base_schemas.items():
             if ns == nm.XSD_NAMESPACE:
                 # Process the patch schema for XSD 1.1 meta-schema
-                patch_schema = schema.include_schema(location=location, partial=True)
+                base_url = SCHEMAS_DIR / str(location).split("/")[-2]
+
+                patch_schema = schema.include_schema(location=location, partial=True, base_url=base_url)
                 base_url = patch_schema.base_url
                 for child in patch_schema.source.root:
                     if child.tag == nm.XSD_OVERRIDE:
                         patch_schema.include_schema(
-                            child.attrib['schemaLocation'],
+                            base_url / child.attrib['schemaLocation'],
                             base_url=base_url,
                             partial=True
                         )
--- contrib/python/xmlschema/xmlschema/validators/xsd_globals.py	(index)
+++ contrib/python/xmlschema/xmlschema/validators/xsd_globals.py	(working tree)
@@ -15,6 +15,7 @@ from contextlib import contextmanager
 from itertools import dropwhile
 from typing import Any, cast, Optional
 from elementpath import XPathToken, XPath2Parser
+from importlib.resources.abc import Traversable
 
 import xmlschema.names as nm
 from xmlschema.aliases import SchemaType, BaseXsdType, SchemaGlobalType, \
@@ -376,6 +377,10 @@ class XsdGlobals(XsdValidator, Collection[SchemaType]):
         if schemas is not None:
             if source is None:
                 return schemas[0]
+            elif isinstance(source, Traversable):
+                for schema in schemas:
+                    if schema.source.traversal_source is source:
+                        return schema
             elif (url := get_url(source)) is not None:
                 url = normalize_url(url, base_url)
                 for schema in schemas:

--- contrib/python/xmlschema/xmlschema/arguments.py	(index)
+++ contrib/python/xmlschema/xmlschema/arguments.py	(working tree)
@@ -16,6 +16,7 @@ from pathlib import Path
 from typing import Any, cast, Generic, Optional, overload, Union
 from urllib.request import OpenerDirector
 from xml.etree.ElementTree import Element
+from importlib.resources.abc import Traversable
 
 from xmlschema.aliases import T, XMLSourceType, UriMapperType, IterParseType, BlockType, \
     FillerType, DepthFillerType, ExtraValidatorType, ValidationHookType, ValueHookType, \
@@ -265,6 +266,8 @@ class BaseUrlOption(Option[Optional[str]]):
     def validated_value(self, value: Any) -> Optional[str]:
         if value is None:
             return None
+        elif isinstance(value, Traversable):
+            return value
         elif not isinstance(value, (str, bytes, Path)):
             msg = _("invalid type {!r} for {}, must be of type {!r}")
             raise XMLSchemaTypeError(msg.format(type(value), self, (str, bytes, Path)))
