--- contrib/python/flex/flex/formats.py	(index)
+++ contrib/python/flex/flex/formats.py	(working tree)
@@ -3,8 +3,7 @@ import re
 
 import six
 
-import rfc3987
-import strict_rfc3339
+import iso8601
 
 from flex.utils import is_value_of_any_type
 from flex.exceptions import ValidationError
@@ -22,11 +21,17 @@ from flex.constants import (
 )
 from flex.error_messages import MESSAGES
 
-if six.PY2:
-    import validate_email
-else:
-    # TODO: when a new version is released, this can be removed.
-    from flex.compat import validate_email
+# XXX: PATCH
+# if six.PY2:
+#     import validate_email
+# else:
+#     # TODO: when a new version is released, this can be removed.
+from flex.compat import validate_email
+
+try:
+    from urlparse import urlparse
+except ImportError:
+    from urllib.parse import urlparse
 
 
 class FormatRegistry(object):
@@ -76,12 +81,14 @@ register = registry.register
 
 @register(URI, STRING)
 def uri_validator(value, **kwargs):
+    # XXX: PATCH changed using module rfc3987 licensed under GPL to standard urlparse
     try:
-        parts = rfc3987.parse(value, rule='URI')
+        # parts = rfc3987.parse(value, rule='URI')
+        parts = urlparse(value)
     except ValueError:
         raise ValidationError(MESSAGES['format']['invalid_uri'].format(value))
 
-    if not parts['scheme'] or not parts['authority']:
+    if not parts.scheme or not parts.netloc:
         raise ValidationError(MESSAGES['format']['invalid_uri'].format(value))
 
 
@@ -105,8 +112,13 @@ def email_validator(value, **kwargs):
 
 @register(DATETIME, STRING)
 def date_time_format_validator(value, **kwargs):
-    if not strict_rfc3339.validate_rfc3339(value):
+    # XXX: PATCH changed rfc3339 on iso8601 because of minor differences
+    try:
+        iso8601.parse_date(value)
+    except (iso8601.ParseError, ValueError):
         raise ValidationError(MESSAGES['format']['invalid_datetime'].format(value))
+    # if not strict_rfc3339.validate_rfc3339(value):
+    #     raise ValidationError(MESSAGES['format']['invalid_datetime'].format(value))
 
 
 @register(DATE, STRING)
@@ -121,7 +133,7 @@ UUID_PATTERN = re.compile(
     '^'
     '[a-f0-9]{8}-'
     '[a-f0-9]{4}-'
-    '[1345][a-f0-9]{3}-'
+    '[a-f0-9]{4}-'
     '[a-f0-9]{4}'
     '-[a-f0-9]{12}'
     '$'
