--- contrib/python/pytest-django/py3/pytest_django/plugin.py	(index)
+++ contrib/python/pytest-django/py3/pytest_django/plugin.py	(working tree)
@@ -561,7 +561,10 @@ def _django_setup_unittest(
 
         # Don't set up the DB if the unittest does not require DB.
         # The `databases` propery seems like the best indicator for that.
-        if request.cls.databases:
+        if not hasattr(request.cls, "databases"):
+            request.getfixturevalue("django_db_setup")
+            db_unblock = django_db_blocker.unblock()
+        elif request.cls.databases:
             request.getfixturevalue("django_db_setup")
             db_unblock = django_db_blocker.unblock()
         else:
