--- contrib/python/ydb/py3/ydb/_grpc/grpcwrapper/common_utils.py	(index)
+++ contrib/python/ydb/py3/ydb/_grpc/grpcwrapper/common_utils.py	(working tree)
@@ -27,9 +27,9 @@ from google.protobuf.timestamp_pb2 import Timestamp as ProtoTimeStamp
 import ydb.aio

 try:
-    from ..v4.protos import ydb_topic_pb2, ydb_issue_message_pb2
-else:
-    from ..common.protos import ydb_topic_pb2, ydb_issue_message_pb2
+    from ydb.public.api.protos import ydb_topic_pb2, ydb_issue_message_pb2
+except ImportError:
+    from contrib.ydb.public.api.protos import ydb_topic_pb2, ydb_issue_message_pb2

 from ... import issues, connection

--- contrib/python/ydb/py3/ydb/_grpc/grpcwrapper/ydb_topic.py	(index)
+++ contrib/python/ydb/py3/ydb/_grpc/grpcwrapper/ydb_topic.py	(working tree)
@@ -13,9 +13,9 @@ from ... import scheme
 from ... import issues

 try:
-    from ..v4.protos import ydb_scheme_pb2, ydb_topic_pb2
-else:
-    from ..common.protos import ydb_scheme_pb2, ydb_topic_pb2
+    from ydb.public.api.protos import ydb_scheme_pb2, ydb_topic_pb2
+except ImportError:
+    from contrib.ydb.public.api.protos import ydb_scheme_pb2, ydb_topic_pb2

 from .common_utils import (
     IFromProto,
--- contrib/python/ydb/py3/ydb/_grpc/grpcwrapper/ydb_topic_public_types.py	(index)
+++ contrib/python/ydb/py3/ydb/_grpc/grpcwrapper/ydb_topic_public_types.py	(working tree)
@@ -5,9 +5,9 @@ from enum import IntEnum
 from typing import Optional, List, Union, Dict

 try:
-    from ..v4.protos import ydb_topic_pb2
-else:
-    from ..common.protos import ydb_topic_pb2
+    from ydb.public.api.protos import ydb_topic_pb2
+except ImportError:
+    from contrib.ydb.public.api.protos import ydb_topic_pb2

 from .common_utils import IToProto
 from ...scheme import SchemeEntry
