--- contrib/python/ydb/py3/ydb/_grpc/grpcwrapper/ydb_query.py	(index)
+++ contrib/python/ydb/py3/ydb/_grpc/grpcwrapper/ydb_query.py	(working tree)
@@ -3,11 +3,10 @@ import typing
 from typing import Optional


-# Workaround for good IDE and universal for runtime
-if typing.TYPE_CHECKING:
-    from ..v4.protos import ydb_query_pb2
-else:
-    from ..common.protos import ydb_query_pb2
+try:
+    from ydb.public.api.protos import ydb_query_pb2
+except ImportError:
+    from contrib.ydb.public.api.protos import ydb_query_pb2

 from . import ydb_query_public_types as public_types

--- contrib/python/ydb/py3/ydb/_grpc/grpcwrapper/ydb_query_public_types.py	(index)
+++ contrib/python/ydb/py3/ydb/_grpc/grpcwrapper/ydb_query_public_types.py	(working tree)
@@ -3,11 +3,10 @@ import typing

 from .common_utils import IToProto

-# Workaround for good IDE and universal for runtime
-if typing.TYPE_CHECKING:
-    from ..v4.protos import ydb_query_pb2
-else:
-    from ..common.protos import ydb_query_pb2
+try:
+    from ydb.public.api.protos import ydb_query_pb2
+except ImportError:
+    from contrib.ydb.public.api.protos import ydb_query_pb2


 class BaseQueryTxMode(IToProto):
--- contrib/python/ydb/py3/ydb/draft/_apis.py	(index)
+++ contrib/python/ydb/py3/ydb/draft/_apis.py	(working tree)
@@ -1,28 +1,22 @@
 # -*- coding: utf-8 -*-
 import typing

-# Workaround for good IDE and universal for runtime
-if typing.TYPE_CHECKING:
-    from .._grpc.v4.draft import (
+try:
+    from ydb.public.api.grpc.draft import (
         ydb_dynamic_config_v1_pb2_grpc,
     )

-    from .._grpc.v4.draft.protos import (
+    from ydb.public.api.protos.draft import (
         ydb_dynamic_config_pb2,
     )
-else:
-    from .._grpc.common.draft import (
+except ImportError:
+    from contrib.ydb.public.api.grpc.draft import (
         ydb_dynamic_config_v1_pb2_grpc,
     )

-    try:
-        from .._grpc.common.draft.protos import (
-            ydb_dynamic_config_pb2,
-        )
-    except ImportError:
-        from .._grpc.common.protos.draft import (
-            ydb_dynamic_config_pb2,
-        )
+    from contrib.ydb.public.api.protos.draft import (
+        ydb_dynamic_config_pb2,
+    )


 ydb_dynamic_config = ydb_dynamic_config_pb2
