--- a/src/google/protobuf/port_def.inc
+++ b/src/google/protobuf/port_def.inc
@@ -375,26 +375,13 @@ static_assert(PROTOBUF_CPLUSPLUS_MIN(201402L), "Protobuf only supports C++14 and
 #if defined(PROTOBUF_DEPRECATED_MSG)
 #error PROTOBUF_DEPRECATED_MSG was previously defined
 #endif
-#if __has_attribute(deprecated) || defined(__GNUC__)
-# define PROTOBUF_DEPRECATED __attribute__((deprecated))
-# define PROTOBUF_DEPRECATED_MSG(msg) __attribute__((deprecated(msg)))
-#elif defined(_MSC_VER)
-# define PROTOBUF_DEPRECATED __declspec(deprecated)
-# define PROTOBUF_DEPRECATED_MSG(msg) __declspec(deprecated(msg))
-#else
 # define PROTOBUF_DEPRECATED
 # define PROTOBUF_DEPRECATED_MSG(msg)
-#endif
 
 #if defined(PROTOBUF_DEPRECATED_ENUM)
 #error PROTOBUF_DEPRECATED_ENUM was previously defined
 #endif
-#if defined(__clang__) || defined(__GNUC__)
-// https://gcc.gnu.org/gcc-6/changes.html
-# define PROTOBUF_DEPRECATED_ENUM __attribute__((deprecated))
-#else
 # define PROTOBUF_DEPRECATED_ENUM
-#endif
 
 #if defined(__clang__)
 #define PROTOBUF_IGNORE_DEPRECATION_START  