--- contrib/libs/protobuf/src/google/protobuf/arenastring.h	(d40b078e35f492f4ad417c8dd4a0103d4e577049)
+++ contrib/libs/protobuf/src/google/protobuf/arenastring.h	(2babb01d30e285a96044041b98053d9edbb48b2a)
@@ -266,7 +266,6 @@ struct PROTOBUF_EXPORT ArenaStringPtr {
   void Set(const char* s, Arena* arena);
   void Set(const char* s, size_t n, Arena* arena);
 
-  void SetBytes(absl::string_view value, Arena* arena);
   void SetBytes(TProtoStringType&& value, Arena* arena);
   template <typename... OverloadDisambiguator>
   void SetBytes(const TProtoStringType& value, Arena* arena);
@@ -417,10 +416,6 @@ inline void ArenaStringPtr::Set(const char* s, size_t n, Arena* arena) {
   Set(TProtoStringType(s, s + n), arena);
 }
 
-inline void ArenaStringPtr::SetBytes(absl::string_view value, Arena* arena) {
-  Set(TProtoStringType(value.data(), value.size()), arena);
-}
-
 template <>
 PROTOBUF_EXPORT void ArenaStringPtr::Set(const TProtoStringType& value,
                                          Arena* arena);
