--- contrib/libs/protobuf/src/google/protobuf/text_format.h	(d7393489b80ddf99e790c8ca67a83547f368a584)
+++ contrib/libs/protobuf/src/google/protobuf/text_format.h	(working tree)
@@ -390,6 +390,9 @@ class PROTOBUF_EXPORT TextFormat {
       truncate_string_field_longer_than_ = truncate_string_field_longer_than;
     }
 
+    // Sets whether strings will be redacted and thus unparsable.
+    void SetRedactDebugString(bool redact) { redact_debug_string_ = redact; }
+
     // Register a custom field-specific FastFieldValuePrinter for fields
     // with a particular FieldDescriptor.
     // Returns "true" if the registration succeeded, or "false", if there is
@@ -428,9 +431,6 @@ class PROTOBUF_EXPORT TextFormat {
     // Sets whether silent markers will be inserted.
     void SetInsertSilentMarker(bool v) { insert_silent_marker_ = v; }
 
-    // Sets whether strings will be redacted and thus unparsable.
-    void SetRedactDebugString(bool redact) { redact_debug_string_ = redact; }
-
     // Sets whether the output string should be made non-deterministic.
     // This discourages equality checks based on serialized string comparisons.
     void SetRandomizeDebugString(bool randomize) {
