--- a/src/google/protobuf/compiler/cpp/field_generators/string_field.cc	(192810303a994f7cbd6bf596a3d56393bfefee6c)
+++ b/src/google/protobuf/compiler/cpp/field_generators/string_field.cc	(2b6f1fdeb6f4d5c82840dbe69e4f0d3149c8fe5a)
@@ -781,13 +781,11 @@ void RepeatedString::GenerateAccessorDeclarations(io::Printer* p) const {
     $DEPRECATED$ void $set_name$(int index, TProtoStringType&& value);
     $DEPRECATED$ void $set_name$(int index, const char* value);
     $DEPRECATED$ void $set_name$(int index, const $byte$* value, std::size_t size);
-    $DEPRECATED$ void $set_name$(int index, absl::string_view value);
     $DEPRECATED$ TProtoStringType* $add_name$();
     $DEPRECATED$ void $add_name$(const TProtoStringType& value);
     $DEPRECATED$ void $add_name$(TProtoStringType&& value);
     $DEPRECATED$ void $add_name$(const char* value);
     $DEPRECATED$ void $add_name$(const $byte$* value, std::size_t size);
-    $DEPRECATED$ void $add_name$(absl::string_view value);
     $DEPRECATED$ const $pb$::RepeatedPtrField<TProtoStringType>& $name$() const;
     $DEPRECATED$ $pb$::RepeatedPtrField<TProtoStringType>* $mutable_name$();
 
@@ -852,11 +850,6 @@ void RepeatedString::GenerateInlineAccessorDefinitions(io::Printer* p) const {
               $annotate_set$;
               // @@protoc_insertion_point(field_set_pointer:$pkg.Msg.field$)
             }
-            inline void $Msg$::set_$name$(int index, absl::string_view value) {
-              $field_$.Mutable(index)->assign(value.data(), value.size());
-              $annotate_set$;
-              // @@protoc_insertion_point(field_set_string_piece:$pkg.Msg.field$)
-            }
             inline TProtoStringType* $Msg$::_internal_add_$name$() { return $field_$.Add(); }
             inline void $Msg$::add_$name$(const TProtoStringType& value) {
               $field_$.Add()->assign(value);
@@ -879,11 +872,6 @@ void RepeatedString::GenerateInlineAccessorDefinitions(io::Printer* p) const {
               $annotate_add$;
               // @@protoc_insertion_point(field_add_pointer:$pkg.Msg.field$)
             }
-            inline void $Msg$::add_$name$(absl::string_view value) {
-              $field_$.Add()->assign(value.data(), value.size());
-              $annotate_add$;
-              // @@protoc_insertion_point(field_add_string_piece:$pkg.Msg.field$)
-            }
             inline const ::$proto_ns$::RepeatedPtrField<TProtoStringType>&
             $Msg$::$name$() const {
               $annotate_list$;
