--- a/src/google/protobuf/compiler/cpp/message.cc	(index)
+++ b/src/google/protobuf/compiler/cpp/message.cc	(working tree)
@@ -1151,7 +1151,7 @@ void MessageGenerator::GenerateClassDefinition(io::Printer* p) {
         HasDescriptorMethods(descriptor_->file(), options_) ? "" : "Lite";
     auto v = p->WithVars(std::move(vars));
     format(
-        "class $classname$ final : public "
+        "class $classname$ : public "
         "::$proto_ns$::internal::MapEntry$lite$<$classname$, \n"
         "    $key_cpp$, $val_cpp$,\n"
         "    ::$proto_ns$::internal::WireFormatLite::$key_wire_type$,\n
