--- contrib/python/pq/pq/__init__.py	(index)
+++ contrib/python/pq/pq/__init__.py	(working tree)
@@ -13,6 +13,8 @@ from .utils import (
     Literal, prepared, transaction, convert_time_spec, utc_format
 )
 
+import library.python.resource as rs
+
 
 __title__ = 'pq'
 __version__ = '1.9.0'
@@ -57,8 +59,7 @@ class PQ(object):
     def create(self):
         queue = self['']
 
-        with open(os.path.join(self.template_path, 'create.sql'), 'r') as f:
-            sql = f.read()
+        sql = rs.resfs_read('contrib/python/pq/pq/create.sql')
 
         with queue._transaction() as cursor:
             cursor.execute(sql, {'name': Literal(queue.table)})
--- contrib/python/pq/pq/tasks.py	(index)
+++ contrib/python/pq/pq/tasks.py	(working tree)
@@ -16,7 +16,7 @@ def task(
     retry_in='30s',
 ):
     def decorator(f):
-        f._path = "%s.%s" % (f.__module__, f.__qualname__)
+        f._path = "%s.%s" % (f.__module__, f.__name__)
         f._max_retries = max_retries
         f._retry_in = retry_in
 
