--- contrib/restricted/aws/aws-c-cal/source/unix/openssl_platform_init.c	(index)
+++ contrib/restricted/aws/aws-c-cal/source/unix/openssl_platform_init.c	(working tree)
@@ -489,7 +489,11 @@ static enum aws_libcrypto_version s_resolve_libcrypto(void) {
     /* Try to auto-resolve against what's linked in/process space */
     AWS_LOGF_DEBUG(AWS_LS_CAL_LIBCRYPTO_RESOLVE, "searching process and loaded modules");
     void *process = dlopen(NULL, RTLD_NOW);
+#if 0
+    // dlopen is not supported in musl. It's ok to pass NULL to s_resolve_libcrypto_symbols,
+    // as dlsym handles it well according to man.
     AWS_FATAL_ASSERT(process && "Unable to load symbols from process space");
+#endif
     enum aws_libcrypto_version result = s_resolve_libcrypto_symbols(AWS_LIBCRYPTO_LC, process);
     if (result == AWS_LIBCRYPTO_NONE) {
         AWS_LOGF_DEBUG(AWS_LS_CAL_LIBCRYPTO_RESOLVE, "did not find aws-lc symbols linked");
