--- contrib/python/pyproj/py2/pyproj/datadir.py	(index)
+++ contrib/python/pyproj/py2/pyproj/datadir.py	(working tree)
@@ -86,13 +86,8 @@ def get_data_dir():
     elif valid_data_dirs(proj_lib_dirs):
         _VALIDATED_PROJ_DATA = proj_lib_dirs
     else:
-        proj_exe = find_executable("proj")
-        if proj_exe is not None:
-            system_proj_dir = os.path.join(
-                os.path.dirname(os.path.dirname(proj_exe)), "share", "proj"
-            )
-            if valid_data_dir(system_proj_dir):
-                _VALIDATED_PROJ_DATA = system_proj_dir
+        # Use Arcadia resource file
+        _VALIDATED_PROJ_DATA = os.sep.join([os.path.dirname(__file__), 'data'])
 
     if _VALIDATED_PROJ_DATA is None:
         raise DataDirError(
