--- contrib/python/pyproj/py3/pyproj/network.py	(index)
+++ contrib/python/pyproj/py3/pyproj/network.py	(working tree)
@@ -8,1 +8,1 @@ import os
-import certifi
+import certifi.source as certifi
--- contrib/python/pyproj/py3/pyproj/datadir.py (index)
+++ contrib/python/pyproj/py3/pyproj/datadir.py (working tree)
@@ -102,13 +102,13 @@ def get_data_dir() -> str:
     elif valid_data_dir(conda_windows_prefix_datadir):
         _VALIDATED_PROJ_DATA = str(conda_windows_prefix_datadir)
     else:
-        proj_exe = shutil.which("proj", path=sys.prefix)
-        if proj_exe is None:
-            proj_exe = shutil.which("proj")
-        if proj_exe is not None:
-            system_proj_dir = Path(proj_exe).parent.parent / "share" / "proj"
-            if valid_data_dir(system_proj_dir):
-                _VALIDATED_PROJ_DATA = str(system_proj_dir)
+        # Use Arcadia resource file
+        import tempfile
+        import __res
+        tmpdir = tempfile.mkdtemp()
+        with open(os.path.join(tmpdir, 'proj.db'), 'wb') as f:
+            f.write(__res.find('/proj.db'))
+        _VALIDATED_PROJ_DATA = tmpdir
 
     if _VALIDATED_PROJ_DATA is None:
         raise DataDirError(
