--- contrib/python/sqlalchemy/sqlalchemy-1.2/sqlalchemy/dialects/mysql/base.py	(index)
+++ contrib/python/sqlalchemy/sqlalchemy-1.2/sqlalchemy/dialects/mysql/base.py	(working tree)
@@ -2135,7 +2135,7 @@ class MySQLDialect(default.DefaultDialect):
         try:
             dbapi_connection.commit()
         except Exception:
-            if self.server_version_info < (3, 23, 15):
+            if self.server_version_info and self.server_version_info < (3, 23, 15):
                 args = sys.exc_info()[1].args
                 if args and args[0] == 1064:
                     return
@@ -2147,7 +2147,7 @@ class MySQLDialect(default.DefaultDialect):
         try:
             dbapi_connection.rollback()
         except Exception:
-            if self.server_version_info < (3, 23, 15):
+            if self.server_version_info and self.server_version_info < (3, 23, 15):
                 args = sys.exc_info()[1].args
                 if args and args[0] == 1064:
                     return
