--- contrib/python/uvloop/tests/test_context.py	(index)
+++ contrib/python/uvloop/tests/test_context.py	(working tree)
@@ -687,11 +687,14 @@ class _ContextBaseTests(tb.SSLTestCase):
         async def test():
             self.assertEqual(cvar.get(), 'outer')
             cvar.set('inner')
+            import os
+            env = os.environ.copy()
+            env["Y_PYTHON_ENTRY_POINT"] = ":main"
             await self.loop.subprocess_exec(
                 lambda: proto, sys.executable, b'-c',
                 b';'.join((b'import sys',
                            b'data = sys.stdin.buffer.read()',
-                           b'sys.stdout.buffer.write(data)')))
+                           b'sys.stdout.buffer.write(data)')), env=env)
 
             try:
                 inner = await proto.connection_made_fut
--- contrib/python/uvloop/uvloop/_testbase.py	(index)
+++ contrib/python/uvloop/uvloop/_testbase.py	(working tree)
@@ -229,6 +229,8 @@ class BaseTestCase(unittest.TestCase, metaclass=BaseTestCaseMeta):
 
 
 def _cert_fullname(test_file_name, cert_file_name):
+    import yatest.common as yc
+    test_file_name = yc.source_path(test_file_name)
     fullname = os.path.abspath(os.path.join(
         os.path.dirname(test_file_name), 'certs', cert_file_name))
     assert os.path.isfile(fullname)
