--- contrib/python/uvloop/tests/test_tcp.py     (index)
+++ contrib/python/uvloop/tests/test_tcp.py     (working tree)
@@ -405,7 +405,7 @@ class _TestTCP:
             self.assertEqual(await reader.readexactly(2), b'OK')

             re = r'(a bytes-like object)|(must be byte-ish)'
-            if sys.version_info >= (3, 14):
+            if sys.version_info >= (3, 13, 9):
                 re += r'|(must be a bytes, bytearray, or memoryview object)'
             with self.assertRaisesRegex(TypeError, re):
                 writer.write('AAAA')
