https://github.com/MagicStack/uvloop/issues/702

--- contrib/python/uvloop/uvloop/__init__.py	(index)
+++ contrib/python/uvloop/uvloop/__init__.py	(working tree)
@@ -193,6 +193,7 @@ def __getattr__(name: str) -> _typing.Any:
 
         class _Local(threading.local):
             _loop: _typing.Optional[_AbstractEventLoop] = None
+            _set_called: bool = False
 
         def __init__(self) -> None:
             self._local = self._Local()
@@ -202,6 +203,28 @@ def __getattr__(name: str) -> _typing.Any:
 
             Returns an instance of EventLoop or raises an exception.
             """
+            if (self._local._loop is None and
+                    not self._local._set_called and
+                    threading.current_thread() is threading.main_thread()):
+                stacklevel = 2
+                try:
+                    f = _sys._getframe(1)
+                except AttributeError:
+                    pass
+                else:
+                    # Move up the call stack so that the warning is attached
+                    # to the line outside asyncio itself.
+                    while f:
+                        module = f.f_globals.get('__name__')
+                        if not (module == 'asyncio' or module.startswith('asyncio.')):
+                            break
+                        f = f.f_back
+                        stacklevel += 1
+                import warnings
+                warnings.warn('There is no current event loop',
+                              DeprecationWarning, stacklevel=stacklevel)
+                self.set_event_loop(self.new_event_loop())
+
             if self._local._loop is None:
                 raise RuntimeError(
                     'There is no current event loop in thread %r.'
@@ -214,6 +237,7 @@ def __getattr__(name: str) -> _typing.Any:
             self, loop: _typing.Optional[_AbstractEventLoop]
         ) -> None:
             """Set the event loop."""
+            self._local._set_called = True
             if loop is not None and not isinstance(loop, _AbstractEventLoop):
                 raise TypeError(
                     f"loop must be an instance of AbstractEventLoop or None, "
