--- contrib/python/distro/py3/tests/test_distro.py	(index)
+++ contrib/python/distro/py3/tests/test_distro.py	(working tree)
@@ -23,7 +23,13 @@ except ImportError:
 
 import pytest
 
-BASE = os.path.abspath(os.path.dirname(__file__))
+
+try:
+    import yatest.common
+    BASE = yatest.common.test_source_path()
+except ImportError:
+    BASE = os.path.abspath(os.path.dirname(__file__))
+
 RESOURCES = os.path.join(BASE, "resources")
 DISTROS_DIR = os.path.join(RESOURCES, "distros")
 TESTDISTROS = os.path.join(RESOURCES, "testdistros")
@@ -50,6 +56,14 @@ class TestNonLinuxPlatform:
 
 @pytest.mark.skipif(not IS_LINUX, reason="Irrelevant on non-linux")
 class TestCli:
+
+    def _get_command(self):
+        try:
+            import yatest.common
+            return [yatest.common.binary_path('contrib/python/distro/py3/bin/distro')]
+        except ImportError:
+            return [sys.executable, '-m', 'distro']
+
     def _parse(self, command: str) -> None:
         sys.argv = command.split()
         distro.main()
@@ -72,7 +86,7 @@ class TestCli:
         self._parse(f"distro --root-dir {root_dir}")
 
     def test_cli(self) -> None:
-        command = [sys.executable, "-m", "distro"]
+        command = self._get_command()
         desired_output = f"Name: {distro.name(pretty=True)}"
         distro_version = distro.version(pretty=True)
         distro_codename = distro.codename()
@@ -82,18 +96,18 @@ class TestCli:
         assert self._run(command) == desired_output
 
     def test_cli_json(self) -> None:
-        command = [sys.executable, "-m", "distro", "-j"]
+        command = self._get_command() + ["-j"]
         assert ast.literal_eval(self._run(command)) == distro.info()
 
     def test_cli_with_root_dir(self) -> None:
         root_dir = os.path.join(RESOURCES, "cli", "fedora30")
-        command = [sys.executable, "-m", "distro", "--root-dir", root_dir]
+        command = self._get_command() + ["--root-dir", root_dir]
         desired_output = "Name: Fedora 30 (Thirty)\nVersion: 30\nCodename: \n"
         assert desired_output == self._run(command)
 
     def test_cli_with_root_dir_as_json(self) -> None:
         root_dir = os.path.join(RESOURCES, "cli", "fedora30")
-        command = [sys.executable, "-m", "distro", "-j", "--root-dir", root_dir]
+        command = self._get_command() + ["-j", "--root-dir", root_dir]
         desired_output = {
             "codename": "",
             "id": "fedora",
@@ -116,12 +130,13 @@ class DistroTestCase(object):
         # The environment stays the same across all testcases, so we
         # save and restore the PATH env var in each test case that
         # changes it:
-        self._saved_path = os.environ["PATH"]
+        self._saved_path = os.environ.get("PATH")
         self._saved_UNIXCONFDIR = distro._UNIXCONFDIR
         self._saved_UNIXUSRLIBDIR = distro._UNIXUSRLIBDIR
 
     def teardown_method(self, test_method: FunctionType) -> None:
-        os.environ["PATH"] = self._saved_path
+        if self._saved_path:
+            os.environ["PATH"] = self._saved_path
         distro._UNIXCONFDIR = self._saved_UNIXCONFDIR
         distro._UNIXUSRLIBDIR = self._saved_UNIXUSRLIBDIR
 
