diff --git a/mypy/build.py b/mypy/build.py
index ac6471d..1f43c00 100644
--- a/mypy/build.py
+++ b/mypy/build.py
@@ -662,7 +662,7 @@ class BuildManager:
         self.find_module_cache = FindModuleCache(
             self.search_paths, self.fscache, self.options, source_set=self.source_set
         )
-        for module in CORE_BUILTIN_MODULES:
+        for module in sorted(CORE_BUILTIN_MODULES):
             if options.use_builtins_fixtures:
                 continue
             path = self.find_module_cache.find_module(module, fast_path=True)
diff --git a/mypy/config_parser.py b/mypy/config_parser.py
index a6bf021..a782500 100644
--- a/mypy/config_parser.py
+++ b/mypy/config_parser.py
@@ -237,6 +237,15 @@ def _parse_individual_file(
     config_file: str, stderr: TextIO | None = None
 ) -> tuple[MutableMapping[str, Any], dict[str, _INI_PARSER_CALLABLE], str] | None:
 
+    if config_package := os.getenv('YA_MYPY_PKG', None):
+        import pkgutil
+        _data = pkgutil.get_data(config_package, config_file)
+        assert _data is not None
+        _data = _data.decode('utf-8')
+        parser = configparser.RawConfigParser()
+        parser.read_string(_data)
+        return parser, ini_config_types, config_file
+
     if not os.path.exists(config_file):
         return None
 
@@ -254,6 +254,15 @@ def parse_config_file(
 
     current_dir = os.path.abspath(os.getcwd())
 
+    if config_package := os.getenv('YA_MYPY_PKG', None):
+        import pkgutil
+        _data = pkgutil.get_data(config_package, config_file)
+        assert _data is not None
+        _data = _data.decode('utf-8')
+        parser = configparser.RawConfigParser()
+        parser.read_string(_data, source=config_package)
+        return parser, config_types, config_package
+
     while True:
         for name in defaults.CONFIG_NAMES + defaults.SHARED_CONFIG_NAMES:
             config_file = os.path.relpath(os.path.join(current_dir, name))
diff --git a/mypy/fscache.py b/mypy/fscache.py
index 8251f4b..b94b449 100644
--- a/mypy/fscache.py
+++ b/mypy/fscache.py
@@ -37,6 +37,7 @@ from mypy_extensions import mypyc_attr

 from mypy.util import hash_digest

+import pkg_resources

 @mypyc_attr(allow_interpreted_subclasses=True)  # for tests
 class FileSystemCache:
@@ -76,6 +77,12 @@ class FileSystemCache:
                 except OSError:
                     pass

+        if not st:
+            try:
+                st = self._resource_stat(path)
+            except (pkg_resources.ResolutionError, pkg_resources.ExtractionError):
+                pass
+
         self.stat_or_none_cache[path] = st
         return st

@@ -131,6 +138,29 @@ class FileSystemCache:
                     ok = True
         return ok

+    # Along the lines of _fake_init
+    def _resource_stat(self, path: str) -> os.stat_result:
+        if not pkg_resources.resource_exists(__package__, path) and not pkg_resources.resource_isdir(__package__, path):
+            raise pkg_resources.ResolutionError()
+
+        isdir = pkg_resources.resource_isdir(__package__, path)
+
+        try:
+            st = os.stat(os.curdir)
+        except OSError:
+            raise pkg_resources.ResolutionError()
+
+        tpl = tuple(st)  # type: ignore
+        seq = list(tpl)  # type: List[float]
+        seq[stat.ST_MODE] =  stat.S_IFDIR | 0o777 if isdir else stat.S_IFREG | 0o444
+        seq[stat.ST_INO] = 1
+        seq[stat.ST_NLINK] = 2 if isdir else 1
+        seq[stat.ST_SIZE] = 0
+        tpl = tuple(seq)
+        st = os.stat_result(tpl)
+        self.stat_or_none_cache[path] = st
+        return st
+
     def _fake_init(self, path: str) -> os.stat_result:
         """Prime the cache with a fake __init__.py file.

@@ -167,9 +197,12 @@ class FileSystemCache:
         try:
             results = os.listdir(path)
         except OSError as err:
-            # Like above, take a copy to reduce memory use.
-            self.listdir_error_cache[path] = copy_os_error(err)
-            raise err
+            try:
+                results = pkg_resources.resource_listdir(__package__, path)
+            except (pkg_resources.ResolutionError, pkg_resources.ExtractionError):
+                # Like above, take a copy to reduce memory use.
+                self.listdir_error_cache[path] = copy_os_error(err)
+                raise err
         self.listdir_cache[path] = results
         # Check the fake cache.
         if path in self.fake_package_cache and "__init__.py" not in results:
@@ -272,8 +305,11 @@ class FileSystemCache:
                 with open(path, "rb") as f:
                     data = f.read()
             except OSError as err:
-                self.read_error_cache[path] = err
-                raise
+                try:
+                    data = pkg_resources.resource_string(__package__, path)
+                except (pkg_resources.ResolutionError, pkg_resources.ExtractionError):
+                    self.read_error_cache[path] = err
+                    raise

         self.read_cache[path] = data
         self.hash_cache[path] = hash_digest(data)
diff --git a/mypy/main.py b/mypy/main.py
index f177bb1..32b6fce 100644
--- a/mypy/main.py
+++ b/mypy/main.py
@@ -1249,8 +1249,6 @@ def process_options(
     config_file = dummy.config_file
     # Don't explicitly test if "config_file is not None" for this check.
     # This lets `--config-file=` (an empty string) be used to disable all config files.
-    if config_file and not os.path.exists(config_file):
-        parser.error(f"Cannot find config file '{config_file}'")

     options = Options()
     strict_option_set = False
diff --git a/mypy/modulefinder.py b/mypy/modulefinder.py
index 49c39a9..d88b61e 100644
--- a/mypy/modulefinder.py
+++ b/mypy/modulefinder.py
@@ -5,6 +5,8 @@ This builds on fscache.py; find_sources.py builds on top of this.

 from __future__ import annotations

+import importlib
+import __res as ress
 import ast
 import collections
 import functools
@@ -25,6 +27,8 @@ from mypy.stubinfo import approved_stub_package_exists
 from mypy.util import os_path_join


+import pkg_resources
+
 # Paths to be searched in find_module().
 class SearchPaths(NamedTuple):
     python_path: tuple[str, ...]  # where user code is found
@@ -456,12 +460,13 @@ class FindModuleCache:
             third_party_stubs_dirs.clear()
             found_possible_third_party_missing_type_hints = False
         python_mypy_path = self.search_paths.mypy_path + self.search_paths.python_path
-        candidate_base_dirs = self.find_lib_path_dirs(id, python_mypy_path)
+        candidate_base_dirs = []
         if use_typeshed:
             # Search for stdlib stubs in typeshed before installed
             # stubs to avoid picking up backports (dataclasses, for
             # example) when the library is included in stdlib.
             candidate_base_dirs += self.find_lib_path_dirs(id, self.search_paths.typeshed_path)
+        candidate_base_dirs += self.find_lib_path_dirs(id, python_mypy_path)
         candidate_base_dirs += third_party_stubs_dirs + third_party_inline_dirs

         # If we're looking for a module like 'foo.bar.baz', then candidate_base_dirs now
@@ -689,2 +694,2 @@ def default_lib_path(
-        typeshed_dir = os.path.join(data_dir, "typeshed", "stdlib")
-        mypy_extensions_dir = os.path.join(data_dir, "typeshed", "stubs", "mypy-extensions")
+        typeshed_dir = "typeshed/stdlib"
+        mypy_extensions_dir = "typeshed/stubs/mypy-extensions"
@@ -852,3 +857,4 @@ def load_stdlib_py_versions(custom_typeshed_dir: str | None) -> StdlibVersions:
-    assert os.path.isfile(versions_path), (custom_typeshed_dir, versions_path, __file__)
-    with open(versions_path) as f:
-        for line in f:
+    # assert os.path.isfile(versions_path), (custom_typeshed_dir, versions_path, __file__)
+    f = ress.resfs_read(versions_path).decode()
+    if True:
+        for line in f.split('\n'):
diff --git a/mypy/options.py b/mypy/options.py
index 56bd929..60f8fdf 100644
--- a/mypy/options.py
+++ b/mypy/options.py
@@ -193,7 +193,7 @@ class Options:
         self.error_summary = True

         # Assume arguments with default values of None are Optional
-        self.implicit_optional = False
+        self.implicit_optional = True

         # Don't re-export names unless they are imported with `from ... as ...`
         self.implicit_reexport = True
diff --git a/mypy/util.py b/mypy/util.py
index 289740b..322b283 100644
--- a/mypy/util.py
+++ b/mypy/util.py
@@ -864,6 +864,9 @@ class FancyFormatter:


 def is_typeshed_file(typeshed_dir: str | None, file: str) -> bool:
+    if file.endswith('.pyi'):
+        return True
+
     typeshed_dir = typeshed_dir if typeshed_dir is not None else TYPESHED_DIR
     try:
         return os.path.commonpath((typeshed_dir, os.path.abspath(file))) == typeshed_dir
