--- a/mypy/modulefinder.py	(index)
+++ b/mypy/modulefinder.py	(working tree)
@@ -39,13 +39,13 @@ class SearchPaths:
         typeshed_path: tuple[str, ...],
     ) -> None:
         # where user code is found
-        self.python_path = tuple(map(os.path.abspath, python_path))
+        self.python_path = tuple(python_path)
         # from $MYPYPATH or config variable
-        self.mypy_path = tuple(map(os.path.abspath, mypy_path))
+        self.mypy_path = tuple(mypy_path)
         # from get_site_packages_dirs()
-        self.package_path = tuple(map(os.path.abspath, package_path))
+        self.package_path = tuple(package_path)
         # paths in typeshed
-        self.typeshed_path = tuple(map(os.path.abspath, typeshed_path))
+        self.typeshed_path = tuple(typeshed_path)
 
     def asdict(self) -> dict[str, tuple[str, ...]]:
         return {
