--- contrib/python/ijson/py2/tests/test_misc.py	(index)
+++ contrib/python/ijson/py2/tests/test_misc.py	(working tree)
@@ -1 +1,4 @@
-import importlib.util
+try:
+    import importlib.util
+except ImportError:
+    pass
@@ -3,1 +3,1 @@ import unittest
-from test.test_base import JSON, JSON_EVENTS, JSON_PARSE_EVENTS, JSON_OBJECT,\
+from .test_base import JSON, JSON_EVENTS, JSON_PARSE_EVENTS, JSON_OBJECT,\
