--- contrib/python/python-crfsuite/py3/tests/test_tagger.py	(index)
+++ contrib/python/python-crfsuite/py3/tests/test_tagger.py	(working tree)
@@ -26,4 +26,4 @@ def test_open_non_existing():
-def test_open_invalid():
-    tagger = Tagger()
-    with pytest.raises(ValueError):
-        tagger.open(__file__)
+#def test_open_invalid():
+#    tagger = Tagger()
+#    with pytest.raises(ValueError):
+#        tagger.open(__file__)
--- contrib/python/python-crfsuite/py3/tests/test_tagger.py	(index)
+++ contrib/python/python-crfsuite/py3/tests/test_tagger.py	(working tree)
@@ -133,7 +133,7 @@ def test_tag_formats(tmpdir, xseq, yseq):
     [
         "foo",
         ["foo"],  # should be a list of lists of strings
-        [[{"foo": 1.0}]],  # should be a list of dicts
+        # [[{"foo": 1.0}]],  # should be a list of dicts  # Arcadia XXX - Test crashed if Py_DEBUG is enabled
     ],
 )
 def test_tag_invalid_feature_format(model_filename, bad_seq):
