--- opentelemetry-cpp/api/include/opentelemetry/logs/noop.h	(index)
+++ opentelemetry-cpp/api/include/opentelemetry/logs/noop.h	(working tree)
@@ -91,6 +91,8 @@ private:
 };
 
 #if OPENTELEMETRY_ABI_VERSION_NO < 2
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
 class NoopEventLogger final : public EventLogger
 {
 public:
@@ -125,6 +127,7 @@ public:
 private:
   nostd::shared_ptr<EventLogger> event_logger_;
 };
+#pragma GCC diagnostic pop
 #endif
 
 }  // namespace logs
