--- contrib/python/ConfigArgParse/py2/tests/test_configargparse.py	(index)
+++ contrib/python/ConfigArgParse/py2/tests/test_configargparse.py	(working tree)
@@ -263,7 +263,7 @@ class TestBasicUseCases(TestCase):
 
         if not use_groups:
             self.assertRegex(self.format_help(),
-                'usage: .* \\[-h\\] --genome GENOME \\[-v\\] -g MY_CFG_FILE\n?'
+                'usage: .* \\[-h\\] --genome GENOME \\[-v\\] -g\n?\s+MY_CFG_FILE\n?'
                 '\\s+\\[-d DBSNP\\]\\s+\\[-f FRMT\\]\\s+vcf \\[vcf ...\\]\n\n'
                 'positional arguments:\n'
                 '  vcf \\s+ Variant file\\(s\\)\n\n'
@@ -277,7 +277,7 @@ class TestBasicUseCases(TestCase):
                 7*r'(.+\s*)')
         else:
             self.assertRegex(self.format_help(),
-                'usage: .* \\[-h\\] --genome GENOME \\[-v\\] -g MY_CFG_FILE\n?'
+                'usage: .* \\[-h\\] --genome GENOME \\[-v\\] -g\n?\s+MY_CFG_FILE\n?'
                 '\\s+\\[-d DBSNP\\]\\s+\\[-f FRMT\\]\\s+vcf \\[vcf ...\\]\n\n'
                 'positional arguments:\n'
                 '  vcf \\s+ Variant file\\(s\\)\n\n'
@@ -872,7 +872,7 @@ class TestMisc(TestCase):
         self.assertEqual(ns.genome, "hg20")
 
         self.assertRegex(self.format_help(),
-            'usage: .* \\[-h\\] -c CONFIG_FILE --genome GENOME\n\n'
+            'usage: .* \\[-h\\] -c CONFIG_FILE --genome\n?\s*GENOME\n\n'
             '%s:\n'
             '  -h, --help\\s+ show this help message and exit\n'
             '  -c CONFIG_FILE, --config CONFIG_FILE\\s+ my config file\n'
