--- contrib/python/prophet/prophet/tests/test_diagnostics.py	(index)
+++ contrib/python/prophet/prophet/tests/test_diagnostics.py	(working tree)
@@ -20,8 +20,10 @@ import datetime
 from prophet import Prophet
 from prophet import diagnostics
 
+import yatest.common as yc
+
 DATA_all = pd.read_csv(
-    os.path.join(os.path.dirname(__file__), 'data.csv'), parse_dates=['ds']
+    os.path.join(os.path.dirname(yc.source_path(__file__)), 'data.csv'), parse_dates=['ds']
 )
 DATA = DATA_all.head(100)
 
--- contrib/python/prophet/prophet/tests/test_prophet.py	(index)
+++ contrib/python/prophet/prophet/tests/test_prophet.py	(working tree)
@@ -16,13 +16,14 @@ import numpy as np
 import pandas as pd
 from prophet import Prophet
 
+import yatest.common as yc
 
 DATA = pd.read_csv(
-    os.path.join(os.path.dirname(__file__), 'data.csv'),
+    os.path.join(os.path.dirname(yc.source_path(__file__)), 'data.csv'),
     parse_dates=['ds'],
 )
 DATA2 = pd.read_csv(
-    os.path.join(os.path.dirname(__file__), 'data2.csv'),
+    os.path.join(os.path.dirname(yc.source_path(__file__)), 'data2.csv'),
     parse_dates=['ds'],
 )
 
--- contrib/python/prophet/prophet/tests/test_serialize.py	(index)
+++ contrib/python/prophet/prophet/tests/test_serialize.py	(working tree)
@@ -18,9 +18,10 @@ import pandas as pd
 from prophet import Prophet
 from prophet.serialize import model_to_json, model_from_json, PD_SERIES, PD_DATAFRAME
 
+import yatest.common as yc
 
 DATA = pd.read_csv(
-    os.path.join(os.path.dirname(__file__), 'data.csv'),
+    os.path.join(os.path.dirname(yc.source_path(__file__)), 'data.csv'),
     parse_dates=['ds'],
 )
 
@@ -145,7 +146,7 @@ class TestSerialize(TestCase):
         }
         for v, (pred_val, v_str) in old_versions.items():
             fname = os.path.join(
-                os.path.dirname(__file__),
+                os.path.dirname(yc.source_path(__file__)),
                 'serialized_model_v{}.json'.format(v)
             )
             with open(fname, 'r') as fin:
--- contrib/python/prophet/prophet/tests/test_utilities.py	(index)
+++ contrib/python/prophet/prophet/tests/test_utilities.py	(working tree)
@@ -16,9 +16,10 @@ import pandas as pd
 from prophet import Prophet
 from prophet.utilities import regressor_coefficients
 
+import yatest.common as yc
 
 DATA = pd.read_csv(
-    os.path.join(os.path.dirname(__file__), 'data.csv'),
+    os.path.join(os.path.dirname(yc.source_path(__file__)), 'data.csv'),
     parse_dates=['ds'],
 )
 
