--- contrib/python/prophet/prophet/tests/test_serialize.py	(index)
+++ contrib/python/prophet/prophet/tests/test_serialize.py	(working tree)
@@ -14,4 +14,5 @@ import sys
 from unittest import TestCase, skipUnless
 
+import pytest
 import numpy as np
 import pandas as pd
@@ -29,4 +30,5 @@ DATA = pd.read_csv(
 class TestSerialize(TestCase):
 
+    @pytest.mark.xfail
     def test_simple_serialize(self):
         m = Prophet()
@@ -73,4 +75,5 @@ class TestSerialize(TestCase):
         self.assertTrue(np.array_equal(fcst['yhat'].values, fcst2['yhat'].values))
 
+    @pytest.mark.xfail
     def test_full_serialize(self):
         # Construct a model with all attributes
@@ -140,4 +143,5 @@ class TestSerialize(TestCase):
         self.assertTrue(np.array_equal(fcst['yhat'].values, fcst2['yhat'].values))
 
+    @pytest.mark.xfail
     def test_backwards_compatibility(self):
         old_versions = {
