--- contrib/python/sacrebleu/sacrebleu/__init__.py	(index)
+++ contrib/python/sacrebleu/sacrebleu/__init__.py	(working tree)
@@ -24,6 +24,7 @@ from .utils import get_available_testsets, get_langpairs_for_testset
 from .dataset import DATASETS
 from .tokenizers import TOKENIZERS, DEFAULT_TOKENIZER
 from .metrics import BLEU, CHRF
+from .sacrebleu import main
 
 # Backward compatibility functions for old style API access (<= 1.4.10)
 from .compat import *
--- contrib/python/sacrebleu/tests/test_api.py	(index)
+++ contrib/python/sacrebleu/tests/test_api.py	(working tree)
@@ -20,7 +20,7 @@ test_api_get_data = [
     ("mtnt1.1/train", "ja-en", 10, "0歳から100歳の女性が登場する海外のスキンケアCM", "The overseas skin care commercial in which 0 to 100 year old females appear."),
     ("wmt19/google/ar", "en-de", 1, "Welsh AMs worried about 'looking like muppets'", "Walisische Abgeordnete befürchten als ,Idioten’ dazustehen."),
 ]
-
+@pytest.mark.skip('no net in arcadia')
 @pytest.mark.parametrize("testset, langpair, sentno, source, reference", test_api_get_data)
 def test_api_get_source(testset, langpair, sentno, source, reference):
     with open(sacrebleu.get_source_file(testset, langpair)) as fh:
@@ -28,6 +28,7 @@ def test_api_get_source(testset, langpair, sentno, source, reference):
 
         assert line == source
 
+@pytest.mark.skip('no net in arcadia')
 @pytest.mark.parametrize("testset, langpair, sentno, source, reference", test_api_get_data)
 def test_api_get_reference(testset, langpair, sentno, source, reference):
     with open(sacrebleu.get_reference_files(testset, langpair)[0]) as fh:
