--- contrib/python/gpxpy/test.py	(index)
+++ contrib/python/gpxpy/test.py	(working tree)
@@ -60,6 +60,9 @@ from typing import *
 mod_logging.basicConfig(level=mod_logging.DEBUG,
                         format='%(asctime)s %(name)-12s %(levelname)-8s %(message)s')
 
+import yatest.common as yc
+mod_os.chdir(yc.source_path("contrib/python/gpxpy"))
+
 def equals(object1: Any, object2: Any, ignore: Any=None) -> bool:
     """ Testing purposes only """
 
--- contrib/python/gpxpy/test.py	(index)
+++ contrib/python/gpxpy/test.py	(working tree)
@@ -1881 +1881 @@ class GPXTests(mod_unittest.TestCase):
-        with open('test_files/validation_gpx10.gpx', 'w') as f:
+        with open(yc.work_path('validation_gpx10.gpx'), 'w') as f:
@@ -2572 +2572 @@ class GPXTests(mod_unittest.TestCase):
-        with open('test_files/validation_gpx11.gpx', 'w') as f:
+        with open(yc.work_path('validation_gpx11.gpx'), 'w') as f:
