--- contrib/python/PyNaCl/py2/tests/test_aead.py	(index)
+++ contrib/python/PyNaCl/py2/tests/test_aead.py	(working tree)
@@ -26,1 +26,1 @@ from hypothesis.strategies import binary, sampled_from
-from utils import read_kv_test_vectors
+from .utils import read_kv_test_vectors
--- contrib/python/PyNaCl/py2/tests/test_bindings.py	(index)
+++ contrib/python/PyNaCl/py2/tests/test_bindings.py	(working tree)
@@ -25,1 +25,1 @@ from hypothesis.strategies import binary, integers
-from test_signing import ed25519_known_answers
+from .test_signing import ed25519_known_answers
@@ -27,1 +27,1 @@ from hypothesis.strategies import binary, integers
-from utils import flip_byte, read_crypto_test_vectors
+from .utils import flip_byte, read_crypto_test_vectors
--- contrib/python/PyNaCl/py2/tests/test_box.py	(index)
+++ contrib/python/PyNaCl/py2/tests/test_box.py	(working tree)
@@ -21,1 +21,1 @@ import binascii
-from test_bindings import _box_from_seed_vectors
+from .test_bindings import _box_from_seed_vectors
--- contrib/python/PyNaCl/py2/tests/test_generichash.py	(index)
+++ contrib/python/PyNaCl/py2/tests/test_generichash.py	(working tree)
@@ -24,1 +24,1 @@ import os
-from utils import read_crypto_test_vectors
+from .utils import read_crypto_test_vectors
@@ -54,1 +54,1 @@ def blake2_salt_pers_vectors():
-    path = os.path.join(os.path.dirname(__file__), "data", DATA)
+    path = os.path.join(__import__('yatest').common.source_path(os.path.dirname(__file__)), "data", DATA)
--- contrib/python/PyNaCl/py2/tests/test_public.py	(index)
+++ contrib/python/PyNaCl/py2/tests/test_public.py	(working tree)
@@ -19,1 +19,1 @@ import binascii
-from utils import assert_equal, assert_not_equal
+from .utils import assert_equal, assert_not_equal
--- contrib/python/PyNaCl/py2/tests/test_pwhash.py	(index)
+++ contrib/python/PyNaCl/py2/tests/test_pwhash.py	(working tree)
@@ -47,1 +47,1 @@ PASSWD_CHARS = u''.join(c for c in _all_unicode
-    path = os.path.join(os.path.dirname(__file__), "data", DATA)
+    path = os.path.join(__import__('yatest').common.source_path(os.path.dirname(__file__)), "data", DATA)
@@ -56,1 +56,1 @@ def argon2i_modular_crypt_ref():
-    path = os.path.join(os.path.dirname(__file__), "data", DATA)
+    path = os.path.join(__import__('yatest').common.source_path(os.path.dirname(__file__)), "data", DATA)
@@ -66,1 +66,1 @@ def argon2i_raw_ref():
-    path = os.path.join(os.path.dirname(__file__), "data", DATA)
+    path = os.path.join(__import__('yatest').common.source_path(os.path.dirname(__file__)), "data", DATA)
@@ -77,1 +77,1 @@ def argon2id_modular_crypt_ref():
-    path = os.path.join(os.path.dirname(__file__), "data", DATA)
+    path = os.path.join(__import__('yatest').common.source_path(os.path.dirname(__file__)), "data", DATA)
--- contrib/python/PyNaCl/py2/tests/test_sealed_box.py	(index)
+++ contrib/python/PyNaCl/py2/tests/test_sealed_box.py	(working tree)
@@ -21,1 +21,1 @@ import binascii
-from utils import read_crypto_test_vectors
+from .utils import read_crypto_test_vectors
--- contrib/python/PyNaCl/py2/tests/test_secret.py	(index)
+++ contrib/python/PyNaCl/py2/tests/test_secret.py	(working tree)
@@ -21,1 +21,1 @@ import binascii
-from utils import flip_byte
+from .utils import flip_byte
--- contrib/python/PyNaCl/py2/tests/test_secretstream.py	(index)
+++ contrib/python/PyNaCl/py2/tests/test_secretstream.py	(working tree)
@@ -49,1 +49,1 @@ from nacl.utils import random as randombytes
-    path = os.path.join(os.path.dirname(__file__), "data", DATA)
+    path = os.path.join(__import__('yatest').common.source_path(os.path.dirname(__file__)), "data", DATA)
--- contrib/python/PyNaCl/py2/tests/test_signing.py	(index)
+++ contrib/python/PyNaCl/py2/tests/test_signing.py	(working tree)
@@ -21,1 +21,1 @@ import binascii
-from utils import assert_equal, assert_not_equal, read_crypto_test_vectors
+from .utils import assert_equal, assert_not_equal, read_crypto_test_vectors
--- contrib/python/PyNaCl/py2/tests/utils.py	(index)
+++ contrib/python/PyNaCl/py2/tests/utils.py	(working tree)
@@ -35,1 +35,1 @@ def assert_not_equal(x, y):
-    path = os.path.join(os.path.dirname(__file__), "data", fname)
+    path = os.path.join(__import__('yatest').common.source_path(os.path.dirname(__file__)), "data", fname)
@@ -51,1 +51,1 @@ def read_kv_test_vectors(fname, delimiter=None, newrecord=None):
-    path = os.path.join(os.path.dirname(__file__), "data", fname)
+    path = os.path.join(__import__('yatest').common.source_path(os.path.dirname(__file__)), "data", fname)
