--- contrib/python/pathy/pathy/_tests/test_base.py	(index)
+++ contrib/python/pathy/pathy/_tests/test_base.py	(working tree)
@@ -3,1 +3,1 @@ from typing import Any
-from .. import (
+from pathy import (
@@ -16,1 +16,1 @@ from .. import (
-from ..about import __version__
+from pathy.about import __version__
--- contrib/python/pathy/pathy/_tests/test_pathy.py	(index)
+++ contrib/python/pathy/pathy/_tests/test_pathy.py	(working tree)
@@ -10,7 +10,7 @@ try:
 except ModuleNotFoundError:
     has_spacy = False
 
-from .. import (
+from pathy import (
     BasePath,
     BlobStat,
     BucketClientFS,
@@ -550,13 +550,15 @@ def test_pathy_mkdir(with_adapter: str) -> None:
     assert not path.exists()
 
 
+@pytest.mark.skip
 @pytest.mark.parametrize("adapter", TEST_ADAPTERS)
 def test_pathy_ignore_extension(with_adapter: str, bucket: str) -> None:
     """The smart_open library does automatic decompression based
     on the filename. We disable that to avoid errors, e.g. if you
     have a .tar.gz file that isn't gzipped."""
+    import yatest.common
     not_targz = Pathy(f"{with_adapter}://{bucket}/{ENV_ID}/ignore_ext/one.tar.gz")
-    fixture_tar = Path(__file__).parent / "fixtures" / "tar_but_not_gzipped.tar.gz"
+    fixture_tar = Path(yatest.common.source_path("contrib/python/pathy/pathy/_tests")) / "fixtures" / "tar_but_not_gzipped.tar.gz"
     not_targz.write_bytes(fixture_tar.read_bytes())
     again = not_targz.read_bytes()
     assert again is not None
--- contrib/python/pathy/pathy/_tests/test_pure_pathy.py	(index)
+++ contrib/python/pathy/pathy/_tests/test_pure_pathy.py	(working tree)
@@ -4,7 +4,7 @@ from pathlib import Path, PurePosixPath, PureWindowsPath
 
 import pytest
 
-from .. import PurePathy
+from pathy import PurePathy
 
 
 def test_pure_pathy_paths_of_a_different_flavour() -> None:
