--- contrib/python/GDAL/py2/extensions/gdal_array_wrap.cpp	(index)
+++ contrib/python/GDAL/py2/extensions/gdal_array_wrap.cpp	(working tree)
@@ -3581,31 +3581,11 @@ class NUMPYDataset : public GDALDataset
                  NUMPYDataset();
                  ~NUMPYDataset();
 
-    virtual const char *_GetProjectionRef(void) override;
-    const OGRSpatialReference* GetSpatialRef() const override {
-        return GetSpatialRefFromOldGetProjectionRef();
-    }
-    virtual CPLErr _SetProjection( const char * ) override;
-    CPLErr SetSpatialRef(const OGRSpatialReference* poSRS) override {
-        return OldSetProjectionFromSetSpatialRef(poSRS);
-    }
-
     virtual CPLErr GetGeoTransform( double * ) override;
     virtual CPLErr SetGeoTransform( double * ) override;
 
     virtual int    GetGCPCount() override;
-    virtual const char *_GetGCPProjection() override;
-    const OGRSpatialReference* GetGCPSpatialRef() const override {
-        return GetGCPSpatialRefFromOldGetGCPProjection();
-    }
     virtual const GDAL_GCP *GetGCPs() override;
-    virtual CPLErr _SetGCPs( int nGCPCount, const GDAL_GCP *pasGCPList,
-                            const char *pszGCPProjection ) override;
-    using GDALDataset::SetGCPs;
-    CPLErr SetGCPs( int nGCPCount, const GDAL_GCP *pasGCPList,
-                    const OGRSpatialReference* poSRS ) override {
-        return OldSetGCPsFromNew(nGCPCount, pasGCPList, poSRS);
-    }
 
     static GDALDataset *Open( PyArrayObject *psArray, bool binterleave = true );
     static GDALDataset *Open( GDALOpenInfo * );
@@ -3687,29 +3667,6 @@ NUMPYDataset::~NUMPYDataset()
 }
 
 /************************************************************************/
-/*                          GetProjectionRef()                          */
-/************************************************************************/
-
-const char *NUMPYDataset::_GetProjectionRef()
-
-{
-    return( pszProjection );
-}
-
-/************************************************************************/
-/*                           SetProjection()                            */
-/************************************************************************/
-
-CPLErr NUMPYDataset::_SetProjection( const char * pszNewProjection )
-
-{
-    CPLFree( pszProjection );
-    pszProjection = CPLStrdup( pszNewProjection );
-
-    return CE_None;
-}
-
-/************************************************************************/
 /*                          GetGeoTransform()                           */
 /************************************************************************/
 
@@ -3746,16 +3703,6 @@ int NUMPYDataset::GetGCPCount()
 }
 
 /************************************************************************/
-/*                          GetGCPProjection()                          */
-/************************************************************************/
-
-const char *NUMPYDataset::_GetGCPProjection()
-
-{
-    return pszGCPProjection;
-}
-
-/************************************************************************/
 /*                               GetGCPs()                              */
 /************************************************************************/
 
@@ -3766,30 +3713,6 @@ const GDAL_GCP *NUMPYDataset::GetGCPs()
 }
 
 /************************************************************************/
-/*                              SetGCPs()                               */
-/************************************************************************/
-
-CPLErr NUMPYDataset::_SetGCPs( int nGCPCount, const GDAL_GCP *pasGCPList,
-                              const char *pszGCPProjection )
-
-{
-    CPLFree( this->pszGCPProjection );
-    if( this->nGCPCount > 0 )
-    {
-        GDALDeinitGCPs( this->nGCPCount, this->pasGCPList );
-        CPLFree( this->pasGCPList );
-    }
-
-    this->pszGCPProjection = CPLStrdup(pszGCPProjection);
-
-    this->nGCPCount = nGCPCount;
-
-    this->pasGCPList = GDALDuplicateGCPs( nGCPCount, pasGCPList );
-
-    return CE_None;
-}
-
-/************************************************************************/
 /*                                Open()                                */
 /************************************************************************/
 
