--- contrib/python/Flask-SQLAlchemy/py2/tests/test_config.py	(index)
+++ contrib/python/Flask-SQLAlchemy/py2/tests/test_config.py	(working tree)
@@ -79,6 +79,7 @@ class TestConfigKeys:
                  ' "sqlite:///:memory:".'
         assert recwarn[0].message.args[0] == expect
 
+    @pytest.mark.skip
     def test_engine_creation_ok(self, app, recwarn):
         """ create_engine() isn't called until needed.  Let's make sure we can do that without
             errors or warnings.
--- contrib/python/Flask-SQLAlchemy/py2/tests/test_sessions.py	(index)
+++ contrib/python/Flask-SQLAlchemy/py2/tests/test_sessions.py	(working tree)
@@ -1,3 +1,5 @@
+import sys
+import pytest
 import sqlalchemy as sa
 from sqlalchemy.orm import sessionmaker
 
@@ -16,6 +18,7 @@ def test_default_session_scoping(app, db):
         assert fb in db.session
 
 
+@pytest.mark.skipif(sys.version_info < (3, 0), reason="Skip Python 2 in Arcadia")
 def test_session_scoping_changing(app):
     def scopefunc():
         return id(dict())
