--- contrib/python/django-filebrowser-no-grappelli/tests/__init__.py	(index)
+++ contrib/python/django-filebrowser-no-grappelli/tests/__init__.py	(working tree)
@@ -1,52 +1,3 @@
-import os
-import shutil
-
-from django.conf import settings
-from django.contrib.auth import get_user_model
-from django.test import TestCase
-
-from filebrowser.settings import DIRECTORY, VERSIONS_BASEDIR
-from filebrowser.base import FileObject
-from filebrowser.sites import site
-
-
-class FilebrowserTestCase(TestCase):
-
-    @classmethod
-    def setUpClass(cls):
-        super(FilebrowserTestCase, cls).setUpClass()
-        User = get_user_model()
-        cls.user = User.objects.create_user('testuser', 'test@domain.com', 'password')
-        cls.user.is_staff = True
-        cls.user.save()
-
-    def setUp(self):
-        self.DIRECTORY = DIRECTORY
-        self.TEST_PATH = os.path.join(site.storage.location, '_test')
-        self.DIRECTORY_PATH = os.path.join(site.storage.location, DIRECTORY)
-        self.VERSIONS_PATH = os.path.join(site.storage.location, VERSIONS_BASEDIR)
-
-        if os.path.exists(self.TEST_PATH):
-            raise Exception('TEST_PATH Already Exists')
-
-        self.TEMP_PATH = os.path.join(self.TEST_PATH, 'tempfolder')
-        self.FOLDER_PATH = os.path.join(self.DIRECTORY_PATH, 'folder')
-        self.SUBFOLDER_PATH = os.path.join(self.FOLDER_PATH, 'subfolder')
-        self.CREATEFOLDER_PATH = os.path.join(self.DIRECTORY_PATH, 'create')
-        self.PLACEHOLDER_PATH = os.path.join(self.DIRECTORY_PATH, 'placeholders')
-
-        self.STATIC_IMG_PATH = os.path.join(settings.BASE_DIR, 'filebrowser', "static", "filebrowser", "img", "testimage.jpg")
-        self.STATIC_IMG_BAD_NAME_PATH = os.path.join(settings.BASE_DIR, 'filebrowser', "static", "filebrowser", "img", "TEST_IMAGE_000.jpg")
-
-        self.F_IMAGE = FileObject(os.path.join(DIRECTORY, 'folder', "testimage.jpg"), site=site)
-        self.F_MISSING = FileObject(os.path.join(DIRECTORY, 'folder', "missing.jpg"), site=site)
-        self.F_FOLDER = FileObject(os.path.join(DIRECTORY, 'folder'), site=site)
-        self.F_SUBFOLDER = FileObject(os.path.join(DIRECTORY, 'folder', 'subfolder'), site=site)
-        self.F_CREATEFOLDER = FileObject(os.path.join(DIRECTORY, 'create'), site=site)
-        self.F_TEMPFOLDER = FileObject(os.path.join('_test', 'tempfolder'), site=site)
-
-        os.makedirs(self.FOLDER_PATH)
-        os.makedirs(self.SUBFOLDER_PATH)
-
-    def tearDown(self):
-        shutil.rmtree(self.TEST_PATH)
+# Yandex specific
+import tests.settings
+# End Yandex specific
--- contrib/python/django-filebrowser-no-grappelli/tests/base.py	(index)
+++ contrib/python/django-filebrowser-no-grappelli/tests/base.py	(working tree)
@@ -1,6 +1,8 @@
 import os
+import random
 import shutil
 
+import yatest.common
 from django.conf import settings
 from django.contrib.auth.models import User
 from django.test import TestCase
@@ -30,7 +32,7 @@ class FilebrowserTestCase(TestCase):
         self.VERSIONS_PATH = os.path.join(site.storage.location, VERSIONS_BASEDIR)
 
         if os.path.exists(self.TEST_PATH):
-            raise Exception('TEST_PATH Already Exists')
+            raise Exception('TEST_PATH Already Exists.' + self.TEST_PATH)
 
         self.TEMP_PATH = os.path.join(self.TEST_PATH, 'tempfolder')
         self.FOLDER_PATH = os.path.join(self.DIRECTORY_PATH, 'folder')
@@ -38,9 +40,14 @@ class FilebrowserTestCase(TestCase):
         self.CREATEFOLDER_PATH = os.path.join(self.DIRECTORY_PATH, 'create')
         self.PLACEHOLDER_PATH = os.path.join(self.DIRECTORY_PATH, 'placeholders')
 
-        self.STATIC_IMG_PATH = os.path.join(settings.BASE_DIR, 'filebrowser', "static", "filebrowser", "img", "testimage.jpg")
-        self.STATIC_IMG_BAD_NAME_PATH = os.path.join(settings.BASE_DIR, 'filebrowser', "static", "filebrowser", "img", "TEST_IMAGE_000.jpg")
-        self.STATIC_IMG_BAD_PATH = os.path.join(settings.BASE_DIR, 'filebrowser', "static", "filebrowser", "img", "test_not_an_image.jpg")
+        # Yandex specific
+        base_dir = yatest.common.source_path(
+            'contrib/python/django-filebrowser-no-grappelli/filebrowser/static/filebrowser/img'
+        )
+        # End yandex specific
+        self.STATIC_IMG_PATH = os.path.join(base_dir, "testimage.jpg")
+        self.STATIC_IMG_BAD_NAME_PATH = os.path.join(base_dir, "TEST_IMAGE_000.jpg")
+        self.STATIC_IMG_BAD_PATH = os.path.join(base_dir, "test_not_an_image.jpg")
 
         self.F_IMAGE = FileObject(os.path.join(DIRECTORY, 'folder', "testimage.jpg"), site=site)
         self.F_IMAGE_BAD = FileObject(os.path.join(DIRECTORY, 'folder', "test_not_an_image.jpg"), site=site)
--- contrib/python/django-filebrowser-no-grappelli/tests/models.py	(index)
+++ contrib/python/django-filebrowser-no-grappelli/tests/models.py	(working tree)
@@ -1,7 +1,7 @@
-from .test_base import *
-from .test_commands import *
-from .test_decorators import *
-from .test_settings import *
-from .test_sites import *
-from .test_templatetags import *
-from .test_versions import *
+from __tests__.test_base import *
+from __tests__.test_commands import *
+from __tests__.test_decorators import *
+from __tests__.test_settings import *
+from __tests__.test_sites import *
+from __tests__.test_templatetags import *
+from __tests__.test_versions import *
--- contrib/python/django-filebrowser-no-grappelli/tests/settings.py	(index)
+++ contrib/python/django-filebrowser-no-grappelli/tests/settings.py	(working tree)
@@ -12,6 +12,7 @@ https://docs.djangoproject.com/en/1.8/ref/settings/
 
 # Build paths inside the project like this: os.path.join(BASE_DIR, ...)
 import os
+import yatest.common
 
 BASE_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
 
@@ -25,7 +26,7 @@ SECRET_KEY = '_4wa$(v5wpr7q(_8h1-1xs^0inp62jav%1$cuo0$r54h%17m^^'
 # SECURITY WARNING: don't run with debug turned on in production!
 DEBUG = True
 
-ALLOWED_HOSTS = []
+ALLOWED_HOSTS = ['*']
 
 
 # Application definition
@@ -36,9 +37,10 @@ INSTALLED_APPS = (
     'django.contrib.sessions',
     'django.contrib.messages',
     'django.contrib.staticfiles',
-    'filebrowser',
     'django.contrib.admin',
-    'tests',
+    'django.forms',
+    'filebrowser',
+    'contrib.python.django-filebrowser-no-grappelli.tests',
 )
 
 MIDDLEWARE_CLASSES = MIDDLEWARE = (
@@ -51,22 +53,49 @@ MIDDLEWARE_CLASSES = MIDDLEWARE = (
 
 ROOT_URLCONF = 'tests.urls'
 
+FORM_RENDERER = 'library.python.django.template.backends.forms_renderer.ArcadiaRenderer'
+
 TEMPLATES = [
+    # {
+    #     'BACKEND': 'django.template.backends.django.DjangoTemplates',
+    #     'DIRS': [],
+    #     'APP_DIRS': True,
+    #     'OPTIONS': {
+    #         'context_processors': [
+    #             'django.template.context_processors.debug',
+    #             'django.template.context_processors.request',
+    #             'django.contrib.auth.context_processors.auth',
+    #             'django.contrib.messages.context_processors.messages',
+    #         ],
+    #     },
+    # },
+    # Yandex specific
     {
-        'BACKEND': 'django.template.backends.django.DjangoTemplates',
-        'DIRS': [],
-        'APP_DIRS': True,
+        'BACKEND': 'library.python.django.template.backends.arcadia.ArcadiaTemplates',
         'OPTIONS': {
-            'context_processors': [
-                'django.template.context_processors.debug',
-                'django.template.context_processors.request',
-                'django.contrib.auth.context_processors.auth',
-                'django.contrib.messages.context_processors.messages',
-            ],
+            'debug': False,
+            'loaders': [
+                'library.python.django.template.loaders.resource.Loader',
+                'library.python.django.template.loaders.app_resource.Loader',
+                # 'django.template.loaders.filesystem.Loader'
+            ]
         },
-    },
+        'DIRS': [],
+    }
+    # End yandex specific
 ]
 
+STATICFILES_FINDERS = (
+    'django.contrib.staticfiles.finders.FileSystemFinder',
+    'django.contrib.staticfiles.finders.AppDirectoriesFinder',
+    # Yandex specific
+    'library.python.django.contrib.staticfiles.finders.ArcadiaAppFinder',
+    # End yandex specific
+)
+
+STATICFILES_DIRS = (
+    yatest.common.source_path('contrib/python/django-filebrowser-no-grappelli/filebrowser/'),
+)
 
 # WSGI_APPLICATION = 'tests.wsgi.application'
 
@@ -76,7 +105,7 @@ TEMPLATES = [
 DATABASES = {
     'default': {
         'ENGINE': 'django.db.backends.sqlite3',
-        'NAME': os.path.join(BASE_DIR, 'db.sqlite3'),
+        'NAME': 'tmp.db',
     }
 }
 
@@ -107,3 +136,16 @@ MEDIA_URL = '/media/'
 FILEBROWSER_DIRECTORY = '_test/uploads/'
 FILEBROWSER_VERSIONS_BASEDIR = '_test/_versions/'
 FILEBROWSER_PLACEHOLDER = '_test/uploads/placeholders/testimage.jpg'
+
+
+# Yandex specific
+from shutil import copyfile
+copyfile(
+    yatest.common.source_path('contrib/python/django-filebrowser-no-grappelli/tests/tmp.db'),
+    DATABASES['default']['NAME']
+)
+import django
+from django.test.utils import setup_test_environment
+django.setup()
+setup_test_environment()
+# End yandex specific
--- contrib/python/django-filebrowser-no-grappelli/tests/test_fields.py	(index)
+++ contrib/python/django-filebrowser-no-grappelli/tests/test_fields.py	(working tree)
@@ -1,6 +1,6 @@
 # coding: utf-8
 
-from tests import FilebrowserTestCase as TestCase
+from tests.base import FilebrowserTestCase as TestCase
 
 from six import string_types
 
--- contrib/python/django-filebrowser-no-grappelli/tests/test_namers.py	(index)
+++ contrib/python/django-filebrowser-no-grappelli/tests/test_namers.py	(working tree)
@@ -5,7 +5,7 @@ import shutil
 from mock import patch
 
 from filebrowser.settings import VERSIONS
-from tests import FilebrowserTestCase as TestCase
+from tests.base import FilebrowserTestCase as TestCase
 
 from filebrowser.namers import OptionsNamer
 
--- contrib/python/django-filebrowser-no-grappelli/tests/test_versions.py	(index)
+++ contrib/python/django-filebrowser-no-grappelli/tests/test_versions.py	(working tree)
@@ -2,6 +2,7 @@
 import os
 import shutil
 
+import pytest
 from django.conf import settings
 from django.template import Context, Template, TemplateSyntaxError
 from mock import patch
@@ -47,8 +48,8 @@ class ImageProcessorsTests(TestCase):
         self.assertEqual(version.size, (500, 375))
 
     @patch('filebrowser.utils.VERSION_PROCESSORS', [
-        'tests.test_versions.processor_mark_1',
-        'tests.test_versions.processor_mark_2',
+        '__tests__.test_versions.processor_mark_1',
+        '__tests__.test_versions.processor_mark_2',
     ])
     def test_process_image_calls_the_stack_of_processors_in_settings(self):
         version = process_image(self.im, {})
@@ -56,7 +57,7 @@ class ImageProcessorsTests(TestCase):
         self.assertTrue(hasattr(version, 'mark_2'))
 
     @patch('filebrowser.utils.VERSION_PROCESSORS', [
-        'tests.test_versions.processor_mark_1',
+        '__tests__.test_versions.processor_mark_1',
     ])
     def test_process_image_calls_only_explicit_provided_processors(self):
         version = process_image(self.im, {}, processors=[processor_mark_2])
@@ -362,7 +363,6 @@ class VersionObjectTemplateTagTests(TestCase):
         super(VersionObjectTemplateTagTests, self).setUp()
         shutil.copy(self.STATIC_IMG_PATH, self.FOLDER_PATH)
         shutil.copy(self.STATIC_IMG_BAD_PATH, self.FOLDER_PATH)
-
         os.makedirs(self.PLACEHOLDER_PATH)
         shutil.copy(self.STATIC_IMG_PATH, self.PLACEHOLDER_PATH)
 
