diff --git a/tcmalloc/allocation_sampling.h b/tcmalloc/allocation_sampling.h
index 2af67c8..023263a 100644
--- a/tcmalloc/allocation_sampling.h
+++ b/tcmalloc/allocation_sampling.h
@@ -193,6 +193,7 @@ SampleifyAllocation(Static& state, Policy policy, size_t requested_size,
   stack_trace.allocation_time = absl::Now();
   stack_trace.guarded_status = alloc_with_status.status;
   stack_trace.allocation_type = policy.allocation_type();
+  stack_trace.user_data = SampleUserDataSupport::UserData::Make();
 
   // How many allocations does this sample represent, given the sampling
   // frequency (weight) and its size.
diff --git a/tcmalloc/internal/logging.cc b/tcmalloc/internal/logging.cc
index 1cd8d18..13aa0e1 100644
--- a/tcmalloc/internal/logging.cc
+++ b/tcmalloc/internal/logging.cc
@@ -43,6 +43,11 @@ GOOGLE_MALLOC_SECTION_BEGIN
 namespace tcmalloc {
 namespace tcmalloc_internal {
 
+ABSL_CONST_INIT SampleUserDataSupport::CreateSampleUserDataCallback* SampleUserDataSupport::create_sample_user_data_callback_ = nullptr;
+ABSL_CONST_INIT SampleUserDataSupport::CopySampleUserDataCallback* SampleUserDataSupport::copy_sample_user_data_callback_ = nullptr;
+ABSL_CONST_INIT SampleUserDataSupport::DestroySampleUserDataCallback* SampleUserDataSupport::destroy_sample_user_data_callback_ = nullptr;
+ABSL_CONST_INIT SampleUserDataSupport::ComputeSampleUserDataHashCallback* SampleUserDataSupport::compute_sample_user_data_hash_callback_ = nullptr;
+
 // Variables for storing crash output.  Allocated statically since we
 // may not be able to heap-allocate while crashing.
 ABSL_CONST_INIT static absl::base_internal::SpinLock crash_lock(
diff --git a/tcmalloc/internal/logging.h b/tcmalloc/internal/logging.h
index 2a5c761..23b3f2a 100644
--- a/tcmalloc/internal/logging.h
+++ b/tcmalloc/internal/logging.h
@@ -51,6 +51,109 @@ GOOGLE_MALLOC_SECTION_BEGIN
 namespace tcmalloc {
 namespace tcmalloc_internal {
 
+class SampleUserDataSupport {
+public:
+  using CreateSampleUserDataCallback = void*();
+  using CopySampleUserDataCallback = void*(void*);
+  using DestroySampleUserDataCallback = void(void*);
+  using ComputeSampleUserDataHashCallback = size_t(void*);
+
+  class UserData {
+  public:
+    static UserData Make() {
+      return UserData{CreateSampleUserData()};
+    }
+    // must be matched with preceding Release
+    static void DestroyRaw(void* ptr) {
+      DestroySampleUserData(ptr);
+    }
+
+    constexpr UserData() noexcept : ptr_(nullptr) {}
+
+    UserData(const UserData& that) noexcept : ptr_(CopySampleUserData(that.ptr_)) {}
+    UserData& operator=(const UserData& that) noexcept {
+      DestroySampleUserData(ptr_);
+      ptr_ = CopySampleUserData(that.ptr_);
+      return *this;
+    }
+
+    UserData(UserData&& that) noexcept : ptr_(that.ptr_) {
+      that.ptr_ = nullptr;
+    }
+    UserData& operator=(UserData&& that) noexcept {
+      if (this == &that) {
+        return *this;
+      }
+      DestroySampleUserData(ptr_);
+      ptr_ = that.ptr_;
+      that.ptr_ = nullptr;
+      return *this;
+    }
+    void Reset() {
+      DestroySampleUserData(ptr_);
+      ptr_ = nullptr;
+    }
+
+    ~UserData() {
+      DestroySampleUserData(ptr_);
+    }
+
+    // should be paired with subsequent DestroyRaw
+    void* Release() && {
+      void* p = ptr_;
+      ptr_ = nullptr;
+      return p;
+    }
+  private:
+    UserData(void* ptr) noexcept : ptr_(ptr) {}
+  private:
+    void* ptr_;
+  };
+
+  static void Enable(CreateSampleUserDataCallback create,
+                     CopySampleUserDataCallback copy,
+                     DestroySampleUserDataCallback destroy,
+                     ComputeSampleUserDataHashCallback compute_hash) {
+    create_sample_user_data_callback_ = create;
+    copy_sample_user_data_callback_ = copy;
+    destroy_sample_user_data_callback_ = destroy;
+    compute_sample_user_data_hash_callback_ = compute_hash;
+  }
+
+  static size_t ComputeSampleUserDataHash(void* ptr) noexcept {
+    if (compute_sample_user_data_hash_callback_ != nullptr) {
+      return compute_sample_user_data_hash_callback_(ptr);
+    }
+    return 0;
+  }
+
+private:
+  static void* CreateSampleUserData() {
+    if (create_sample_user_data_callback_ != nullptr) {
+      return create_sample_user_data_callback_();
+    }
+    return nullptr;
+  }
+
+  static void* CopySampleUserData(void* ptr) noexcept {
+    if (copy_sample_user_data_callback_ != nullptr) {
+      return copy_sample_user_data_callback_(ptr);
+    }
+    return nullptr;
+  }
+
+  static void DestroySampleUserData(void* ptr) noexcept {
+    if (destroy_sample_user_data_callback_ != nullptr) {
+      destroy_sample_user_data_callback_(ptr);
+    }
+  }
+
+  ABSL_CONST_INIT static CreateSampleUserDataCallback* create_sample_user_data_callback_;
+  ABSL_CONST_INIT static CopySampleUserDataCallback* copy_sample_user_data_callback_;
+  ABSL_CONST_INIT static DestroySampleUserDataCallback* destroy_sample_user_data_callback_;
+  ABSL_CONST_INIT static ComputeSampleUserDataHashCallback* compute_sample_user_data_hash_callback_;
+};
+
 static constexpr int kMaxStackDepth = 64;
 
 // An opaque handle type used to identify allocations.
@@ -84,6 +187,8 @@ struct StackTrace {
   // between the previous sample and this one
   size_t weight;
 
+  SampleUserDataSupport::UserData user_data;
+
   // Timestamp of allocation.
   absl::Time allocation_time;
 
diff --git a/tcmalloc/internal/profile_builder.cc b/tcmalloc/internal/profile_builder.cc
index a74e3c7..5887249 100644
--- a/tcmalloc/internal/profile_builder.cc
+++ b/tcmalloc/internal/profile_builder.cc
@@ -128,10 +128,11 @@ struct SampleEqWithSubFields {
 
 struct SampleHashWithSubFields {
   size_t operator()(const Profile::Sample& s) const {
+    size_t user_data_hash = SampleUserDataSupport::ComputeSampleUserDataHash(s.user_data);
     return absl::HashOf(
         absl::MakeConstSpan(s.stack, s.depth), s.depth, s.requested_size,
         s.requested_alignment, s.requested_size_returning, s.allocated_size,
-        s.access_hint, s.access_allocated, s.guarded_status, s.type);
+        s.access_hint, s.access_allocated, s.guarded_status, s.type, user_data_hash);
   }
 };
 
diff --git a/tcmalloc/internal/sampled_allocation_recorder.h b/tcmalloc/internal/sampled_allocation_recorder.h
index 8e1ec85..1b75f0c 100644
--- a/tcmalloc/internal/sampled_allocation_recorder.h
+++ b/tcmalloc/internal/sampled_allocation_recorder.h
@@ -169,6 +169,19 @@ void SampleRecorder<T, Allocator>::PushDead(T* sample) {
   if (auto* dispose = dispose_.load(std::memory_order_relaxed)) {
     dispose(*sample);
   }
+  {
+    // Here we need to synchronize with any possible Copy call on the same
+    // user_data object.
+    // Currently there is only one source of such calls: from callback in Iterate()
+    // (i.e. user has requested a profile concurently with
+    // deallocation of a sampled allocation).
+    // That one holds sample->lock, so we acquire it here too.
+    // This critical section can not be merged with next, because
+    // graveyard_.lock is too constraining (e.g. if destroy callback tries to
+    // deallocate something, it may try to lock graveyard again and deadlock).
+    AllocationGuardSpinLockHolder sample_lock(&sample->lock);
+    sample->sampled_stack.user_data.Reset();
+  }
 
   AllocationGuardSpinLockHolder graveyard_lock(&graveyard_.lock);
   AllocationGuardSpinLockHolder sample_lock(&sample->lock);
@@ -235,6 +248,8 @@ void SampleRecorder<T, Allocator>::Iterate(
     const absl::FunctionRef<void(const T& sample)>& f) {
   T* s = all_.load(std::memory_order_acquire);
   while (s != nullptr) {
+    // Note that this lock also protects Unregister from
+    // creating a Reset-Copy data race.
     AllocationGuardSpinLockHolder l(&s->lock);
     if (s->dead == nullptr) {
       f(*s);
diff --git a/tcmalloc/internal_malloc_extension.h b/tcmalloc/internal_malloc_extension.h
index 190d742..0a93995 100644
--- a/tcmalloc/internal_malloc_extension.h
+++ b/tcmalloc/internal_malloc_extension.h
@@ -157,6 +157,13 @@ ABSL_ATTRIBUTE_WEAK void MallocExtension_Internal_SetMaxTotalThreadCacheBytes(
 
 ABSL_ATTRIBUTE_WEAK void
 MallocExtension_EnableForkSupport();
+
+ABSL_ATTRIBUTE_WEAK void
+MallocExtension_SetSampleUserDataCallbacks(
+    tcmalloc::MallocExtension::CreateSampleUserDataCallback create,
+    tcmalloc::MallocExtension::CopySampleUserDataCallback copy,
+    tcmalloc::MallocExtension::DestroySampleUserDataCallback destroy,
+    tcmalloc::MallocExtension::ComputeSampleUserDataHashCallback compute_hash);
 }
 
 #endif
diff --git a/tcmalloc/malloc_extension.cc b/tcmalloc/malloc_extension.cc
index cee8ba3..47008d0 100644
--- a/tcmalloc/malloc_extension.cc
+++ b/tcmalloc/malloc_extension.cc
@@ -804,6 +804,33 @@ void MallocExtension::EnableForkSupport() {
 #endif
 }
 
+void MallocExtension::SetSampleUserDataCallbacks(
+    CreateSampleUserDataCallback create,
+    CopySampleUserDataCallback copy,
+    DestroySampleUserDataCallback destroy,
+    ComputeSampleUserDataHashCallback compute_hash) {
+#if ABSL_INTERNAL_HAVE_WEAK_MALLOCEXTENSION_STUBS
+  if (&MallocExtension_SetSampleUserDataCallbacks != nullptr) {
+    MallocExtension_SetSampleUserDataCallbacks(create, copy, destroy, compute_hash);
+  }
+#else
+  (void)create;
+  (void)copy;
+  (void)destroy;
+  (void)compute_hash;
+#endif
+}
+
+void MallocExtension::SetSampleUserDataCallbacks(
+  CreateSampleUserDataCallback create,
+  CopySampleUserDataCallback copy,
+  DestroySampleUserDataCallback destroy) {
+    SetSampleUserDataCallbacks(create, copy, destroy, [](void* ptr) -> size_t {
+      (void) ptr;
+      return 0;
+    });
+}
+
 }  // namespace tcmalloc
 
 // Default implementation just returns size. The expectation is that
diff --git a/tcmalloc/malloc_extension.h b/tcmalloc/malloc_extension.h
index 36fd433..3534226 100644
--- a/tcmalloc/malloc_extension.h
+++ b/tcmalloc/malloc_extension.h
@@ -214,6 +214,8 @@ class Profile final {
     int depth;
     void* stack[kMaxStackDepth];
 
+    void* user_data;
+
     // The following vars are used by the lifetime (deallocation) profiler.
     uint64_t profile_id;
 
@@ -664,6 +666,32 @@ class MallocExtension final {
   // Enables fork support.
   // Allocator will continue to function correctly in the child, after calling fork().
   static void EnableForkSupport();
+
+  using CreateSampleUserDataCallback = void*();
+  using CopySampleUserDataCallback = void*(void*);
+  using DestroySampleUserDataCallback = void(void*);
+  using ComputeSampleUserDataHashCallback = size_t(void*);
+
+  // Sets callbacks for lifetime control of custom user data attached to allocation samples.
+  // In general, create, copy and destroy callbacks are used with shared_ptr-like semantics:
+  // - Each non-null user_data returned will be result of the create or copy callbacks.
+  // - For each create or copy call (with result R), there will be matching destroy call with argument R.
+  // - Copy and compute_hash callbacks can be called concurrently, but any such call happens before matching destroy call.
+  // Provided functions must satisfy following requirements:
+  // - create and destroy callbacks may not call back into the allocator except for allocating or deallocating memory.
+  // - copy callback may not call back into the allocator at all.
+  // - only compute_hash may throw exceptions (TODO: can this be relaxed safely?)
+  static void SetSampleUserDataCallbacks(
+    CreateSampleUserDataCallback create,
+    CopySampleUserDataCallback copy,
+    DestroySampleUserDataCallback destroy,
+    ComputeSampleUserDataHashCallback compute_hash);
+
+  // Temporary compat shim. Use 4-argument overload instead.
+  static void SetSampleUserDataCallbacks(
+    CreateSampleUserDataCallback create,
+    CopySampleUserDataCallback copy,
+    DestroySampleUserDataCallback destroy);
 };
 
 }  // namespace tcmalloc
diff --git a/tcmalloc/stack_trace_table.cc b/tcmalloc/stack_trace_table.cc
index cf57148..c6b6867 100644
--- a/tcmalloc/stack_trace_table.cc
+++ b/tcmalloc/stack_trace_table.cc
@@ -39,6 +39,7 @@ StackTraceTable::StackTraceTable(ProfileType type)
 StackTraceTable::~StackTraceTable() {
   LinkedSample* cur = all_;
   while (cur != nullptr) {
+    SampleUserDataSupport::UserData::DestroyRaw(cur->sample.user_data);
     LinkedSample* next = cur->next;
     tc_globals.linked_sample_allocator().Delete(cur);
     cur = next;
@@ -88,6 +89,7 @@ void StackTraceTable::AddTrace(double sample_weight, const StackTrace& t) {
   s->sample.span_start_address = t.span_start_address;
   s->sample.guarded_status = t.guarded_status;
   s->sample.type = t.allocation_type;
+  s->sample.user_data = SampleUserDataSupport::UserData{t.user_data}.Release();
 
   static_assert(kMaxStackDepth <= Profile::Sample::kMaxStackDepth,
                 "Profile stack size smaller than internal stack sizes");
diff --git a/tcmalloc/static_vars.h b/tcmalloc/static_vars.h
index 58f706c..1a5b0aa 100644
--- a/tcmalloc/static_vars.h
+++ b/tcmalloc/static_vars.h
@@ -26,6 +26,7 @@
 #include "absl/base/attributes.h"
 #include "absl/base/optimization.h"
 #include "absl/base/thread_annotations.h"
+#include "internal/logging.h"
 #include "tcmalloc/allocation_sample.h"
 #include "tcmalloc/arena.h"
 #include "tcmalloc/central_freelist.h"
@@ -182,6 +183,15 @@ class Static final {
     fork_support_enabled_ = true;
   }
 
+
+  static void SetSampleUserDataCallbacks(
+      SampleUserDataSupport::CreateSampleUserDataCallback create,
+      SampleUserDataSupport::CopySampleUserDataCallback copy,
+      SampleUserDataSupport::DestroySampleUserDataCallback destroy,
+      SampleUserDataSupport::ComputeSampleUserDataHashCallback compute_hash) {
+    SampleUserDataSupport::Enable(create, copy, destroy, compute_hash);
+  }
+
   static bool ABSL_ATTRIBUTE_ALWAYS_INLINE HaveHooks() {
     return false;
   }
diff --git a/tcmalloc/tcmalloc.cc b/tcmalloc/tcmalloc.cc
index 7332d4f..b930ec9 100644
--- a/tcmalloc/tcmalloc.cc
+++ b/tcmalloc/tcmalloc.cc
@@ -394,6 +394,13 @@ void TCMallocPostFork() {
   Static::sampled_allocation_recorder().ReleaseInternalLocks();
 }
 
+extern "C" void MallocExtension_SetSampleUserDataCallbacks(
+    MallocExtension::CreateSampleUserDataCallback create,
+    MallocExtension::CopySampleUserDataCallback copy,
+    MallocExtension::DestroySampleUserDataCallback destroy,
+    MallocExtension::ComputeSampleUserDataHashCallback compute_hash) {
+  Static::SetSampleUserDataCallbacks(create, copy, destroy, compute_hash);
+}
 
 // nallocx slow path.
 // Moved to a separate function because size_class_with_alignment is not inlined
