diff --git a/tcmalloc/internal/profile_builder.h b/tcmalloc/internal/profile_builder.h
index 06f2abe..d50992e 100644
--- a/tcmalloc/internal/profile_builder.h
+++ b/tcmalloc/internal/profile_builder.h
@@ -24,7 +24,7 @@
 #include <memory>
 #include <string>
 
-#include "tcmalloc/internal/profile.pb.h"
+// #include "tcmalloc/internal/profile.pb.h"
 #include "absl/container/btree_map.h"
 #include "absl/container/flat_hash_map.h"
 #include "absl/status/status.h"
