diff --git a/tcmalloc/malloc_extension.h b/tcmalloc/malloc_extension.h
index 0505971..26e15b4 100644
--- a/tcmalloc/malloc_extension.h
+++ b/tcmalloc/malloc_extension.h
@@ -476,11 +476,9 @@ class MallocExtension final {
 
   // The old names to get and set profile sampling intervals used "rate" to
   // refer to intervals. Use of the below is deprecated to avoid confusion.
-  ABSL_DEPRECATE_AND_INLINE()
   static int64_t GetProfileSamplingRate() {
     return GetProfileSamplingInterval();
   }
-  ABSL_DEPRECATE_AND_INLINE()
   static void SetProfileSamplingRate(int64_t rate) {
     SetProfileSamplingInterval(rate);
   }
@@ -488,7 +486,6 @@ class MallocExtension final {
   static int64_t GetGuardedSamplingRate() {
     return GetGuardedSamplingInterval();
   }
-  ABSL_DEPRECATE_AND_INLINE()
   static void SetGuardedSamplingRate(int64_t rate) {
     SetGuardedSamplingInterval(rate);
   }
