--- contrib/python/jusText/justext/core.py	(index)
+++ contrib/python/jusText/justext/core.py	(working tree)
@@ -18,7 +18,7 @@ try:
 except ImportError:
     from backports.functools_lru_cache import lru_cache

-from lxml.html.clean import Cleaner
+from lxml_html_clean import Cleaner
 from xml.sax.handler import ContentHandler
 from .paragraph import Paragraph
 from ._compat import unicode, ignored
