--- contrib/python/jusText/justext/utils.py	(index)
+++ contrib/python/jusText/justext/utils.py	(working tree)
@@ -6,1 +6,2 @@ from __future__ import division, print_function, unicode_literals
-import pkgutil
+import importlib.resources
+from importlib.resources.abc import TraversalError
@@ -36,12 +36,11 @@ def is_blank(string):

 def get_stoplists():
     """Returns a collection of built-in stop-lists."""
-    path_to_stoplists = os.path.dirname(sys.modules["justext"].__file__)
-    path_to_stoplists = os.path.join(path_to_stoplists, "stoplists")
+    stoplists_dir = importlib.resources.files(__package__) / "stoplists"

     stoplist_names = []
-    for filename in os.listdir(path_to_stoplists):
-        name, extension = os.path.splitext(filename)
+    for filename in stoplists_dir.iterdir():
+        name, extension = os.path.splitext(filename.name)
         if extension == ".txt":
             stoplist_names.append(name)

@@ -50,10 +49,11 @@ def get_stoplists():

 def get_stoplist(language):
     """Returns an built-in stop-list for the language as a set of words."""
-    file_path = os.path.join("stoplists", "%s.txt" % language)
+    stoplists_dir = importlib.resources.files(__package__) / "stoplists"
     try:
-        stopwords = pkgutil.get_data("justext", file_path)
-    except IOError:
+        file_path = stoplists_dir / f"{language}.txt"
+        stopwords = file_path.read_bytes()
+    except (IOError, TraversalError):
         raise ValueError(
             "Stoplist for language '%s' is missing. "
