--- contrib/python/Twisted/twisted/internet/iocpreactor/iocpsupport/iocpsupport.c	(index)
+++ contrib/python/Twisted/twisted/internet/iocpreactor/iocpsupport/iocpsupport.c	(working tree)
@@ -567,7 +567,7 @@ static CYTHON_INLINE float __PYX_NAN() {
 #include "winsock2.h"
 #include "ws2tcpip.h"
 #include "windows.h"
-#include "python.h"
+#include "Python.h"
 #include "string.h"
 #include "wchar.h"
 #include "winsock_pointers.h"
