--- a/drf_spectacular/contrib/django_filters.py
+++ b/drf_spectacular/contrib/django_filters.py
@@ -230,7 +230,7 @@ def _get_explicit_filter_choices(self, filter_field):
         else:
             choices = [c for c, _ in filter_field.extra['choices']]
 
-            if filter_field.field.null_label:
+            if getattr(filter_field.field, 'null_label', None) is not None:
                 choices.append(filter_field.field.null_value)
 
             return choices

