--- contrib/python/paramiko/py3/paramiko/pkey.py	(index)
+++ contrib/python/paramiko/py3/paramiko/pkey.py	(working tree)
@@ -345,7 +345,9 @@
             m = self.END_TAG.match(lines[end])
 
         if keytype == tag:
-            data = self._read_private_key_pem(lines, end, password)
+            # NOTE (torkve@) lines[start - 1:] is old fix and should be preserved
+            # on every update until it appears in upstream
+            data = self._read_private_key_pem(lines[start - 1:], end, password)
             pkformat = self._PRIVATE_KEY_FORMAT_ORIGINAL
         elif keytype == "OPENSSH":
             data = self._read_private_key_openssh(lines[start:end], password)
