--- contrib/python/graphene/py2/graphene/types/enum.py	(index)
+++ contrib/python/graphene/py2/graphene/types/enum.py	(working tree)
@@ -40,6 +40,10 @@ class EnumMeta(SubclassWithMeta_Meta):
     def __getitem__(cls, value):
         return cls._meta.enum[value]
 
+    def __iter__(cls):
+        for item in cls._meta.enum.__members__.items():
+            yield item
+
     def __prepare__(name, bases, **kwargs):  # noqa: N805
         return OrderedDict()
 
