--- contrib/python/graphene/py2/graphene/types/mutation.py	(index)
+++ contrib/python/graphene/py2/graphene/types/mutation.py	(working tree)
@@ -124,7 +124,7 @@ class Mutation(ObjectType):
         else:
             _meta.fields = fields
 
-        _meta.interfaces = interfaces
+        #_meta.interfaces = interfaces
         _meta.output = output
         _meta.resolver = resolver
         _meta.arguments = arguments

