--- contrib/python/PyNaCl/py3/tests/test_generichash.py	(index)
+++ contrib/python/PyNaCl/py3/tests/test_generichash.py	(working tree)
@@ -60,1 +60,1 @@ def blake2_salt_pers_vectors() -> List[
-    path = os.path.join(os.path.dirname(__file__), "data", DATA)
+    path = os.path.join(__import__('yatest').common.source_path(os.path.dirname(__file__)), "data", DATA)
--- contrib/python/PyNaCl/py3/tests/test_pwhash.py	(index)
+++ contrib/python/PyNaCl/py3/tests/test_pwhash.py	(working tree)
@@ -46,1 +46,1 @@ PASSWD_CHARS = "".join(
-    path = os.path.join(os.path.dirname(__file__), "data", DATA)
+    path = os.path.join(__import__('yatest').common.source_path(os.path.dirname(__file__)), "data", DATA)
@@ -56,1 +56,1 @@ def argon2i_modular_crypt_ref() -> List[Tuple[str, str]]:
-    path = os.path.join(os.path.dirname(__file__), "data", DATA)
+    path = os.path.join(__import__('yatest').common.source_path(os.path.dirname(__file__)), "data", DATA)
@@ -75,1 +75,1 @@ def argon2i_raw_ref() -> List[Tuple[int, str, str, int, int, str]]:
-    path = os.path.join(os.path.dirname(__file__), "data", DATA)
+    path = os.path.join(__import__('yatest').common.source_path(os.path.dirname(__file__)), "data", DATA)
@@ -87,1 +87,1 @@ def argon2id_modular_crypt_ref() -> List[Tuple[str, str]]:
-    path = os.path.join(os.path.dirname(__file__), "data", DATA)
+    path = os.path.join(__import__('yatest').common.source_path(os.path.dirname(__file__)), "data", DATA)
--- contrib/python/PyNaCl/py3/tests/test_secretstream.py	(index)
+++ contrib/python/PyNaCl/py3/tests/test_secretstream.py	(working tree)
@@ -51,1 +51,1 @@ Chunk = Tuple[int, Optional[bytes], bytes, bytes]
-    path = os.path.join(os.path.dirname(__file__), "data", DATA)
+    path = os.path.join(__import__('yatest').common.source_path(os.path.dirname(__file__)), "data", DATA)
--- contrib/python/PyNaCl/py3/tests/utils.py	(index)
+++ contrib/python/PyNaCl/py3/tests/utils.py	(working tree)
@@ -37,1 +37,1 @@ def read_crypto_test_vectors(
-    path = os.path.join(os.path.dirname(__file__), "data", fname)
+    path = os.path.join(__import__('yatest').common.source_path(os.path.dirname(__file__)), "data", fname)
@@ -57,1 +57,1 @@ def read_kv_test_vectors(
-    path = os.path.join(os.path.dirname(__file__), "data", fname)
+    path = os.path.join(__import__('yatest').common.source_path(os.path.dirname(__file__)), "data", fname)
