--- contrib/python/ephem/ephem/tests/test_jpl.py	(index)
+++ contrib/python/ephem/ephem/tests/test_jpl.py	(working tree)
@@ -25,9 +25,9 @@ def cleanup(s):
 class JPLDatum(object):
     pass
 
-class JPLTest(unittest.TestCase):
+class JPLTest(object):
 
-    def runTest(self):
+    def test_jpl(self):
         in_data = False
 
         f = open(self.path)
@@ -78,12 +78,24 @@ class JPLTest(unittest.TestCase):
                                           attr, jpl_value,
                                           ephem.degrees(difference)))
 
-re, traceback, datetime, strptime, ephem
-
-def load_tests(loader, tests, pattern):
-    suite = unittest.TestSuite()
-    for path in glob.glob(os.path.dirname(__file__) + '/jpl/*.txt'):
-        case = JPLTest()
-        case.path = path
-        suite.addTest(case)
-    return suite
+# re, traceback, datetime, strptime, ephem
+#
+# def load_tests(loader, tests, pattern):
+#     suite = unittest.TestSuite()
+#     for path in glob.glob(os.path.dirname(__file__) + '/jpl/*.txt'):
+#         case = JPLTest()
+#         case.path = path
+#         suite.addTest(case)
+#     return suite
+
+from yatest.common import source_path
+txts = source_path('contrib/python/ephem/ephem/tests')
+class T1(unittest.TestCase, JPLTest): path = os.path.join(txts, 'jpl', 'europa.txt')
+class T2(unittest.TestCase, JPLTest): path = os.path.join(txts, 'jpl', 'hyperion.txt')
+class T3(unittest.TestCase, JPLTest): path = os.path.join(txts, 'jpl', 'jupiter.txt')
+class T4(unittest.TestCase, JPLTest): path = os.path.join(txts, 'jpl', 'mars.txt')
+class T5(unittest.TestCase, JPLTest): path = os.path.join(txts, 'jpl', 'neptune.txt')
+class T6(unittest.TestCase, JPLTest): path = os.path.join(txts, 'jpl', 'oberon.txt')
+class T7(unittest.TestCase, JPLTest): path = os.path.join(txts, 'jpl', 'phobos.txt')
+class T8(unittest.TestCase, JPLTest): path = os.path.join(txts, 'jpl', 'saturn.txt')
+class T9(unittest.TestCase, JPLTest): path = os.path.join(txts, 'jpl', 'uranus.txt')
--- contrib/python/ephem/ephem/tests/test_usno.py	(index)
+++ contrib/python/ephem/ephem/tests/test_usno.py	(working tree)
@@ -482,7 +482,25 @@ class Mixin(object):
 # Auto-detect files
 #
 
-for path in glob.glob(os.path.dirname(__file__) + '/usno/*.txt'):
-    name = os.path.basename(path).replace('.txt', '')
-    exec('class Test_%s(unittest.TestCase, Mixin): path = %r' % (name, path))
-del path, name
+#for path in glob.glob(os.path.dirname(__file__) + '/usno/*.txt'):
+#    name = os.path.basename(path).replace('.txt', '')
+#    exec('class Test_%s(unittest.TestCase, Mixin): path = %r' % (name, path))
+#del path, name
+
+from yatest.common import source_path
+txts = source_path('contrib/python/ephem/ephem/tests')
+class T1(unittest.TestCase, Mixin): path = os.path.join(txts, 'usno', 'appgeo_deneb.txt')
+class T2(unittest.TestCase, Mixin): path = os.path.join(txts, 'usno', 'appgeo_jupiter.txt')
+class T3(unittest.TestCase, Mixin): path = os.path.join(txts, 'usno', 'appgeo_moon.txt')
+class T4(unittest.TestCase, Mixin): path = os.path.join(txts, 'usno', 'appgeo_sun.txt')
+class T5(unittest.TestCase, Mixin): path = os.path.join(txts, 'usno', 'apptopo_deneb.txt')
+class T6(unittest.TestCase, Mixin): path = os.path.join(txts, 'usno', 'apptopo_moon.txt')
+class T7(unittest.TestCase, Mixin): path = os.path.join(txts, 'usno', 'apptopo_sun.txt')
+class T8(unittest.TestCase, Mixin): path = os.path.join(txts, 'usno', 'astrom_antares.txt')
+class T9(unittest.TestCase, Mixin): path = os.path.join(txts, 'usno', 'astrom_mercury.txt')
+class T10(unittest.TestCase, Mixin): path = os.path.join(txts, 'usno', 'astrom_neptune.txt')
+class T11(unittest.TestCase, Mixin): path = os.path.join(txts, 'usno', 'moon_phases.txt')
+class T12(unittest.TestCase, Mixin): path = os.path.join(txts, 'usno', 'riset_moon.txt')
+class T13(unittest.TestCase, Mixin): path = os.path.join(txts, 'usno', 'riset_sun.txt')
+class T14(unittest.TestCase, Mixin): path = os.path.join(txts, 'usno', 'risettran_moon.txt')
+class T15(unittest.TestCase, Mixin): path = os.path.join(txts, 'usno', 'risettran_rigel.txt')
