--- contrib/python/bokeh/bokeh/core/templates.py	(index)
+++ contrib/python/bokeh/bokeh/core/templates.py	(working tree)
@@ -45 +45 @@ from typing import Any, Callable
-from jinja2 import Environment, FileSystemLoader, Template
+from jinja2 import Environment, PackageLoader, Template
@@ -83 +83 @@ def get_env() -> Environment:
-    return Environment(loader=FileSystemLoader(templates_path), trim_blocks=True, lstrip_blocks=True)
+    return Environment(loader=PackageLoader(__package__, '_templates'), trim_blocks=True, lstrip_blocks=True)
--- contrib/python/bokeh/bokeh/resources.py	(index)
+++ contrib/python/bokeh/bokeh/resources.py	(working tree)
@@ -141 +141,2 @@ def get_all_sri_hashes() -> dict[str, Hashes]:
-            with open(ROOT_DIR / "_sri" / f"{version}.json") as f:
+            import importlib.resources
+            with (importlib.resources.files(__package__) / "_sri" / f"{version}.json").open() as f:
