--- contrib/python/dnspython/py2/tests/test_zone.py	(index)
+++ contrib/python/dnspython/py2/tests/test_zone.py	(working tree)
@@ -30,8 +30,16 @@ import dns.rrset
 import dns.zone
 import dns.node
 
+import yatest.common
+
+
 def here(filename):
-    return os.path.join(os.path.dirname(__file__), filename)
+    return yatest.common.test_source_path(filename)
+
+
+def here_out(filename):
+    return yatest.common.test_output_path(filename)
+
 
 example_text = """$TTL 3600
 $ORIGIN example.
@@ -141,24 +149,24 @@ class ZoneTestCase(unittest.TestCase):
         z = dns.zone.from_file(here('example'), 'example')
         ok = False
         try:
-            z.to_file(here('example1.out'), nl=b'\x0a')
-            ok = filecmp.cmp(here('example1.out'),
+            z.to_file(here_out('example1.out'), nl=b'\x0a')
+            ok = filecmp.cmp(here_out('example1.out'),
                              here('example1.good'))
         finally:
             if not _keep_output:
-                os.unlink(here('example1.out'))
+                os.unlink(here_out('example1.out'))
         self.failUnless(ok)
 
     def testFromFile2(self): # type: () -> None
         z = dns.zone.from_file(here('example'), 'example', relativize=False)
         ok = False
         try:
-            z.to_file(here('example2.out'), relativize=False, nl=b'\x0a')
-            ok = filecmp.cmp(here('example2.out'),
+            z.to_file(here_out('example2.out'), relativize=False, nl=b'\x0a')
+            ok = filecmp.cmp(here_out('example2.out'),
                              here('example2.good'))
         finally:
             if not _keep_output:
-                os.unlink(here('example2.out'))
+                os.unlink(here_out('example2.out'))
         self.failUnless(ok)
 
     def testToFileTextualStream(self): # type: () -> None
@@ -180,38 +188,38 @@ class ZoneTestCase(unittest.TestCase):
     def testToFileTextual(self): # type: () -> None
         z = dns.zone.from_file(here('example'), 'example')
         try:
-            f = open(here('example3-textual.out'), 'w')
+            f = open(here_out('example3-textual.out'), 'w')
             z.to_file(f)
             f.close()
-            ok = filecmp.cmp(here('example3-textual.out'),
+            ok = filecmp.cmp(here_out('example3-textual.out'),
                              here('example3.good'))
         finally:
             if not _keep_output:
-                os.unlink(here('example3-textual.out'))
+                os.unlink(here_out('example3-textual.out'))
         self.failUnless(ok)
 
     def testToFileBinary(self): # type: () -> None
         z = dns.zone.from_file(here('example'), 'example')
         try:
-            f = open(here('example3-binary.out'), 'wb')
+            f = open(here_out('example3-binary.out'), 'wb')
             z.to_file(f)
             f.close()
-            ok = filecmp.cmp(here('example3-binary.out'),
+            ok = filecmp.cmp(here_out('example3-binary.out'),
                              here('example3.good'))
         finally:
             if not _keep_output:
-                os.unlink(here('example3-binary.out'))
+                os.unlink(here_out('example3-binary.out'))
         self.failUnless(ok)
 
     def testToFileFilename(self): # type: () -> None
         z = dns.zone.from_file(here('example'), 'example')
         try:
-            z.to_file(here('example3-filename.out'))
-            ok = filecmp.cmp(here('example3-filename.out'),
+            z.to_file(here_out('example3-filename.out'))
+            ok = filecmp.cmp(here_out('example3-filename.out'),
                              here('example3.good'))
         finally:
             if not _keep_output:
-                os.unlink(here('example3-filename.out'))
+                os.unlink(here_out('example3-filename.out'))
         self.failUnless(ok)
 
     def testToText(self): # type: () -> None
@@ -219,14 +227,14 @@ class ZoneTestCase(unittest.TestCase):
         ok = False
         try:
             text_zone = z.to_text(nl=b'\x0a')
-            f = open(here('example3.out'), 'wb')
+            f = open(here_out('example3.out'), 'wb')
             f.write(text_zone)
             f.close()
-            ok = filecmp.cmp(here('example3.out'),
+            ok = filecmp.cmp(here_out('example3.out'),
                              here('example3.good'))
         finally:
             if not _keep_output:
-                os.unlink(here('example3.out'))
+                os.unlink(here_out('example3.out'))
         self.failUnless(ok)
 
     def testFromText(self): # type: () -> None
