--- contrib/python/coverage/py3/coverage/collector.py	(index)
+++ contrib/python/coverage/py3/coverage/collector.py	(working tree)
@@ -419,6 +419,10 @@ class Collector(object):
             assert isinstance(runtime_err, Exception)
             raise runtime_err
 
+        if getattr(sys, 'is_standalone_binary', False):
+            # filenames should stay relative to the arcadia root, because files may not exist
+            return dict((k, v) for k, v in items if v)
+
         return {self.cached_mapped_file(k): v for k, v in items if v}
 
     def plugin_was_disabled(self, plugin):
--- contrib/python/coverage/py3/coverage/config.py	(index)
+++ contrib/python/coverage/py3/coverage/config.py	(working tree)
@@ -8,6 +8,7 @@ import copy
 import os
 import os.path
 import re
+import sys
 from collections.abc import Callable, Iterable, Mapping
 from typing import Any, Final
 
@@ -236,6 +236,7 @@ class CoverageConfig(object):
 
         # Options for plugins
         self.plugin_options: dict[str, TConfigSectionOut] = {}
+        self.suppress_plugin_errors = True
 
     MUST_BE_LIST = {
         "debug", "concurrency", "plugins",
@@ -251,6 +252,11 @@ class CoverageConfig(object):
                     v = [v]
                 setattr(self, k, v)
 
+    def from_resource(self, resource_name):
+        assert getattr(sys, 'is_standalone_binary', False), 'You have used method by mistake in script, not binary'
+        cp, self._config_contents = _load_config_from_resource(resource_name)
+        return self._parse_config(cp, resource_name, True)
+
     def from_file(self, filename: str, warn: Callable[[str], None], our_file: bool) -> bool:
         """Read configuration from a .rc file.
 
@@ -282,6 +287,9 @@ class CoverageConfig(object):
 
         self.config_files_read.extend(map(os.path.abspath, files_read))
 
+        return self._parse_config(cp, filename, our_file)
+
+    def _parse_config(self, cp, filename, our_file):
         any_set = False
         try:
             for option_spec in self.CONFIG_FILE_OPTIONS:
@@ -329,8 +339,9 @@ class CoverageConfig(object):
 
         if used:
             self.config_file = os.path.abspath(filename)
-            with open(filename, "rb") as f:
-                self._config_contents = f.read()
+            if not getattr(sys, 'is_standalone_binary', False):
+                with open(filename, "rb") as f:
+                    self._config_contents = f.read()
 
         return used
 
@@ -367,6 +373,7 @@ class CoverageConfig(object):
         ("source_pkgs", "run:source_pkgs", "list"),
         ("timid", "run:timid", "boolean"),
         ("_crash", "run:_crash"),
+        ("suppress_plugin_errors", "run:suppress_plugin_errors", "boolean"),
         #
         # [report]
         ("exclude_list", "report:exclude_lines", "regexlist"),
@@ -540,6 +547,10 @@ def read_coverage_config(config_file, **kwargs):
     # 1) defaults:
     config = CoverageConfig()
 
+    # 1.1 built-in config
+    if getattr(sys, 'is_standalone_binary', False):
+        config.from_resource("/coverage_plugins/coveragerc.txt")
+
     # 2) from a file:
     if config_file:
         files_to_try = config_files_to_try(config_file)
@@ -545,6 +545,8 @@ def read_coverage_config(config_file, **kwargs):
         files_to_try = config_files_to_try(config_file)
 
         for fname, our_file, specified_file in files_to_try:
+            if getattr(sys, 'is_standalone_binary', False) and fname == "/coverage_plugins/coveragerc.txt":
+                continue
             config_read = config.from_file(fname, warn, our_file=our_file)
             if config_read:
                 break
@@ -568,3 +579,20 @@ def read_coverage_config(config_file, **kwargs):
     config.post_process()
 
     return config
+
+
+def _load_config_from_resource(resource_name):
+    from io import StringIO
+    from library.python import resource
+
+    config_data = resource.find(resource_name)
+    if config_data is None:
+        raise IOError("No such resource: " + resource_name)
+
+    config_data = config_data.decode('utf-8')
+    cp = HandyConfigParser(True)
+    try:
+        cp.read_file(StringIO(config_data))
+    except configparser.Error as err:
+        raise CoverageException("Couldn't read config %s: %s" % (resource_name, err))
+    return cp, config_data
--- contrib/python/coverage/py3/coverage/control.py	(index)
+++ contrib/python/coverage/py3/coverage/control.py	(working tree)
@@ -11,6 +11,7 @@ import re
 import threading
 import time
 import warnings
+import json
 from collections.abc import Callable, Iterable, Iterator
 from types import FrameType
 from typing import IO, Any, cast
@@ -21,1 +21,1 @@ import re
-from coverage.files import PathAliases, abs_file, relative_filename, set_relative_directory
+from coverage.files import PathAliases, abs_file, canonical_filename, relative_filename, set_relative_directory
@@ -242,6 +243,14 @@ class Coverage(object):
         if not env.METACOV:
             _prevent_sub_process_measurement()
 
+        # Store constructor args to reproduce Coverage object in a subprocess created via multiprocessing.Process
+        self._dumped_args = json.dumps(dict(
+            data_file=data_file, data_suffix=data_suffix, cover_pylib=cover_pylib,
+            auto_data=auto_data, timid=timid, branch=branch, config_file=config_file,
+            source=source, omit=omit, include=include, debug=debug,
+            concurrency=concurrency
+        ))
+
     def __repr__(self) -> str:
         core_name = self._core.tracer_class.__name__ if self._core is not None else "-none-"
         data_file = repr(self._data._filename) if self._data is not None else "-none-"
@@ -269,8 +269,10 @@ class Coverage(object):
         self._exclude_re = {}
 
         set_relative_directory()
-        if self.config.relative_files:
-            self._file_mapper = relative_filename
+        if getattr(sys, 'is_standalone_binary', False):
+            self._file_mapper = canonical_filename
+        elif self.config.relative_files:
+            self._file_mapper = relative_filename
 
         # Load plugins
         self._plugins = Plugins.load_plugins(self.config.plugins, self.config, self._debug)
@@ -434,7 +443,7 @@ class Coverage(object):
                )
             if self.config.config_file is None:
                 raise ConfigError("multiprocessing requires a configuration file")
-            patch_multiprocessing(rcfile=self.config.config_file)
+            patch_multiprocessing(rcfile=self.config.config_file, coverage_args=self._dumped_args)
 
         dycon = self.config.dynamic_context
         if not dycon or dycon == "none":
@@ -813,7 +812,11 @@ class Coverage(object):
         file_reporter = "python"
 
         if isinstance(morf, str):
-            mapped_morf = self._file_mapper(morf)
+            if getattr(sys, 'is_standalone_binary', False):
+                # Leave morf in canonical format - relative to the arcadia root
+                mapped_morf = morf
+            else:
+                mapped_morf = self._file_mapper(morf)
             plugin_name = self._data.file_tracer(mapped_morf)
             if plugin_name:
                 plugin = self._plugins.get(plugin_name)
--- contrib/python/coverage/py3/coverage/inorout.py	(index)
+++ contrib/python/coverage/py3/coverage/inorout.py	(working tree)
@@ -132,6 +132,7 @@ class InOrOut(object):
 
         self.plugins: Plugins
         self.disp_class: type[TFileDisposition] = FileDisposition
+        self.config = config
 
         self.source_pkgs: list[str] = list(config.source_pkgs)
         self.source_dirs: list[str] = list(config.source_dirs)
@@ -141,10 +142,19 @@ class InOrOut(object):
 
         self.include = prep_patterns(config.run_include)
         self.omit = prep_patterns(config.run_omit)
+        if getattr(sys, 'is_standalone_binary', False):
+            # don't trace contrib
+            self.omit.append('contrib/python/*')
+            self.omit.append('contrib/libs/protobuf/*')
+            self.omit.append('library/python/pytest/*')
 
         # The directories for files considered "installed with the interpreter".
         self.pylib_paths: set[str] = set()
-        if not config.cover_pylib:
+        if getattr(sys, 'is_standalone_binary', False):
+            self.pylib_paths.add('contrib/tools/python')
+            self.pylib_paths.add('contrib/tools/python3')
+            self.pylib_paths.add('contrib/tools/python3_prev')
+        if not self.pylib_paths and not config.cover_pylib:
             _add_stdlib_paths(self.pylib_paths)
 
          # To avoid tracing the coverage.py code itself, we skip anything
@@ -166,7 +178,10 @@ class InOrOut(object):
         # To avoid tracing the coverage.py code itself, we skip anything
         # located where we are.
         self.cover_paths: set[str] = set()
-        _add_coverage_paths(self.cover_paths)
+        if getattr(sys, 'is_standalone_binary', False):
+            self.cover_paths.add("contrib/python/coverage")
+        else:
+            _add_coverage_paths(self.cover_paths)
 
         # Find where third-party packages are installed.
         self.third_paths: set[str] = set()
@@ -266,6 +281,11 @@ class InOrOut(object):
             # can't do anything with the data later anyway.
             return nope(disp, "file name is not real")
 
+        # XXX maybe we need to support both at the same time?
+        # Don't trace modules imported from environment in standalone mode
+        if getattr(sys, 'is_standalone_binary', False) and filename.startswith("/"):
+            return nope(disp, "skip modules from environment")
+
         canonical = canonical_filename(filename)
         disp.canonical_filename = canonical
 
@@ -289,6 +309,8 @@ class InOrOut(object):
                         )
                     break
             except Exception:
+                if not self.config.suppress_plugin_errors:
+                    raise
                 plugin_name = plugin._coverage_plugin_name
                 tb = traceback.format_exc()
                 self.warn(f"Disabling plug-in {plugin_name!r} due to an exception:\n{tb}")
--- contrib/python/coverage/py3/coverage/files.py	(index)
+++ contrib/python/coverage/py3/coverage/files.py	(working tree)
@@ -76,6 +76,13 @@ def canonical_filename(filename):
         CANONICAL_FILENAME_CACHE[filename] = cf
     return CANONICAL_FILENAME_CACHE[filename]
 
+if getattr(sys, 'is_standalone_binary', False):
+    # filename for py files in binary is already canonical,
+    # it's relative to the arcadia root
+    def canonical_filename(filename):
+        # next assert does not needed in case when we load coverage from not arcadia source in arcadia binary
+        # assert not filename.startswith("/"), filename
+        return filename
 
 def flat_rootname(filename: str) -> str:
     """A base for a flat file name to correspond to this file.
--- contrib/python/coverage/py3/coverage/html.py	(index)
+++ contrib/python/coverage/py3/coverage/html.py	(working tree)
@@ -12,6 +12,7 @@ import functools
 import json
 import os
 import re
+import sys
 import string
 from collections.abc import Iterable
 from dataclasses import dataclass, field
@@ -52,8 +53,18 @@ def data_filename(fname: str) -> str:
     return static_filename
 
 
+def get_htmlfiles_resource(name):
+    import pkgutil
+    return pkgutil.get_data(__package__, 'htmlfiles/' + name)
+
+
 def read_data(fname: str) -> str:
     """Return the contents of a data file of ours."""
+    if getattr(sys, 'is_standalone_binary', False):
+        res_buf = get_htmlfiles_resource(fname).decode()
+        if res_buf is not None:
+            return res_buf
+
     with open(data_filename(fname), encoding="utf-8") as data_file:
         return data_file.read()
 
@@ -418,7 +429,13 @@ class HtmlReporter:
 
     def copy_static_file(self, src: str, slug: str = "") -> None:
         """Copy a static file into the output directory with cache busting."""
-        dest = copy_with_cache_bust(src, self.directory)
+        data = get_htmlfiles_resource(os.path.basename(src))
+        if data is not None:
+            dest = os.path.join(self.directory, os.path.basename(src))
+            with open(dest, "wb") as afile:
+                afile.write(data)
+        else:
+            dest = copy_with_cache_bust(src, self.directory)
         if not slug:
             slug = os.path.basename(src).replace(".", "_")
         self.template_globals["statics"][slug] = dest  # type: ignore
--- contrib/python/coverage/py3/coverage/multiproc.py	(index)
+++ contrib/python/coverage/py3/coverage/multiproc.py	(working tree)
@@ -17,6 +17,8 @@ PATCHED_MARKER = "_coverage$patched"
 # monkey-patched.
 PATCHED_MARKER = "_coverage$patched"
 
+COVERAGE_CONFIGURATION_ENV = "_COVERAGE_CONFIGURATION_ENV"
+
 
 OriginalProcess = multiprocessing.process.BaseProcess
 original_bootstrap = OriginalProcess._bootstrap
@@ -32,7 +34,9 @@ original_bootstrap = OriginalProcess._bootstrap
         try:
             from coverage import Coverage  # avoid circular import
 
-            cov = Coverage(data_suffix=True, auto_data=True)
+            import json
+            kwconf = json.loads(os.environ[COVERAGE_CONFIGURATION_ENV])
+            cov = Coverage(**kwconf)
             cov._warn_preimported_source = False
             cov.start()
             debug = cov._debug
@@ -66,5 +68,5 @@ class Stowaway(object):
 
 
-def patch_multiprocessing(rcfile: str) -> None:
+def patch_multiprocessing(rcfile: str, coverage_args) -> None:
     """Monkey-patch the multiprocessing module.
 
@@ -88,6 +92,8 @@ def patch_multiprocessing(rcfile):
     # process.
     os.environ["COVERAGE_RCFILE"] = os.path.abspath(rcfile)
 
+    os.environ[COVERAGE_CONFIGURATION_ENV] = coverage_args
+
     # When spawning processes rather than forking them, we have no state in the
     # new process.  We sneak in there with a Stowaway: we stuff one of our own
     # objects into the data that gets pickled and sent to the sub-process. When
--- contrib/python/coverage/py3/coverage/python.py	(index)
+++ contrib/python/coverage/py3/coverage/python.py	(working tree)
@@ -3,6 +3,7 @@
 
 from __future__ import annotations
 
+import sys
 import os.path
 import types
 import zipimport
@@ -35,8 +36,19 @@ def read_python_source(filename):
      return source.replace(b"\r\n", b"\n").replace(b"\r", b"\n")
 
 
-def get_python_source(filename: str) -> str:
+def get_python_source(filename: str, force_fs=False) -> str:
     """Return the source code, as unicode."""
+    if getattr(sys, 'is_standalone_binary', False) and not force_fs:
+        import __res
+
+        modname = __res.importer.file_source(filename)
+        if modname:
+            source = __res.find(modname)
+            source = source.replace(b"\r\n", b"\n").replace(b"\r", b"\n")
+            return source.decode('utf-8')
+        else:
+            # it's fake generated package
+            return u''
     base, ext = os.path.splitext(filename)
     if ext == ".py" and env.WINDOWS:
         exts = [".py", ".pyw"]
