--- contrib/python/coverage/py3/coverage/parser.py	(index)
+++ contrib/python/coverage/py3/coverage/parser.py	(working tree)
@@ -98,6 +98,16 @@ class PythonParser:
         self._missing_arc_fragments: TArcFragments | None = None
         self._with_jump_fixers: dict[TArc, tuple[TArc, TArc]] = {}
 
+        self._raw_funcdefs: set[tuple[TLineNo, TLineNo, str]] | None = None
+    
+    @property
+    def raw_funcdefs(self) -> set[tuple[TLineNo, TLineNo, str]]:
+        if self._raw_funcdefs is None:
+            self._analyze_ast()
+        assert self._raw_funcdefs is not None
+        return self._raw_funcdefs
+        
+
     def lines_matching(self, regex: str) -> set[TLineNo]:
         """Find the lines matching a regex.
 
@@ -311,6 +321,7 @@ class PythonParser:
                 self._all_arcs.add((fl1, fl2))
 
         self._missing_arc_fragments = aaa.missing_arc_fragments
+        self._raw_funcdefs = aaa.funcdefs
 
     def fix_with_jumps(self, arcs: Iterable[TArc]) -> set[TArc]:
         """Adjust arcs to fix jumps leaving `with` statements.
@@ -647,6 +658,8 @@ class AstArcAnalyzer:
         self.missing_arc_fragments: TArcFragments = collections.defaultdict(list)
         self.block_stack: list[Block] = []
 
+        self.funcdefs: set[tuple[TLineNo, TLineNo, str]] = set()
+
         # If `with` clauses jump to their start on the way out, we need
         # information to be able to skip over that jump.  We record the arcs
         # from `with` into the clause (with_entries), and the arcs from the
@@ -707,12 +720,16 @@ class AstArcAnalyzer:
         else:
             # Empty module.
             self.add_arc(start, -start)
+    
+    def _process_function_def(self, start, node):
+        self.funcdefs.add((start, node.body[-1].lineno, node.name))
 
     def _code_object__FunctionDef(self, node: ast.FunctionDef) -> None:
         start = self.line_for_node(node)
         self.block_stack.append(FunctionBlock(start=start, name=node.name))
         exits = self.process_body(node.body)
         self.process_return_exits(exits)
+        self._process_function_def(start, node)
         self.block_stack.pop()
 
     _code_object__AsyncFunctionDef = _code_object__FunctionDef
