--- contrib/python/coverage/py3/coverage/python.py	(index)
+++ contrib/python/coverage/py3/coverage/python.py	(working tree)
@@ -48,11 +48,9 @@ def get_python_source(filename: str, force_fs=False) -> str:
     if getattr(sys, 'is_standalone_binary', False) and not force_fs:
         import __res
 
-        modname = __res.importer.file_source(filename)
-        if modname:
-            source = __res.find(modname)
-            source = source.replace(b"\r\n", b"\n").replace(b"\r", b"\n")
-            return source.decode('utf-8')
+        if src_code := __res.resfs_read(__res.py_src_key(filename)):
+            src_code = src_code.replace(b"\r\n", b"\n").replace(b"\r", b"\n")
+            return src_code.decode('utf-8')
         else:
             # it's fake generated package
             return u''
