--- contrib/python/pydub/pydub/utils.py	(index)
+++ contrib/python/pydub/pydub/utils.py	(working tree)
@@ -149,7 +149,7 @@ def which(program):
     if os.name == "nt" and not program.endswith(".exe"):
         program += ".exe"
 
-    envdir_list = [os.curdir] + os.environ["PATH"].split(os.pathsep)
+    envdir_list = [os.curdir] + os.environ.get("PATH", "").split(os.pathsep)
 
     for envdir in envdir_list:
         program_path = os.path.join(envdir, program)
