--- contrib/python/gino/README.rst	(950e9229c0861492fce25aa3aec3fcfb7406d504)
+++ contrib/python/gino/README.rst	(0fe8277b31a46182f056caeea2c6904a7a3ff845)
@@ -0,0 +1,116 @@
+======
+|GINO|
+======
+
+`ВНИМАНИЕ`
+В аркадии лежит пропатченная версия, обновляйте осторожно.
+В чем отличие - вместе стандартного pool из asyncpg можно использовать
+pool из библиотеки pyscopg2 https://a.yandex-team.ru/arc/trunk/arcadia/library/python/pyscopg2 , так как
+стандартный не знает ничего про мастера/реплику и (на текущий момент) не умеет переживать их переключение, а
+использование cname, которые предоставляет MDB - не является надежным решением
+так как при переключении мастера ваш сервис несколько минут будет в read-only пока
+DNS не обновится.
+Как работает - при использовании Pyscopg2Pool - get запросы автоматически будут уходить
+на мастер, а create/update/delete - на реплики (если они есть и доступны).
+При этом можно явно сказать о том, что прочитать нужно с мастера - вот так
+:code:`domain = await Domain.query.gino.first(read_only=False)`
+
+Сделать аналогичный ПР в gino на гитхабе пока нельзя, так как pyscopg2 только
+планируют выложить в opensource.
+
+Пример использования:
+
+.. code-block:: python
+
+    from gino.ext.starlette import Gino
+    from sqlalchemy.engine.url import URL
+    from gino.dialects.asyncpg import Pyscopg2Pool
+
+
+    def get_engine_config():
+           return {
+            'dsn': URL(
+                drivername=config.db_drivername,
+                username=config.db_username,
+                password=config.db_password,
+                host=config.db_host, # строка вида 'master-host,replica-host-1,replica-host-2'
+                port=config.db_port,
+                database=config.db_name,
+            ),
+            'ssl': config.db_ssl,
+            'echo': config.db_echo,
+            'pool_min_size': config.db_pool_min_size,
+            'pool_max_size': config.db_pool_max_size,
+            'use_connection_for_request': config.db_use_connection_for_request,
+            'retry_limit': config.db_retry_limit,
+            'retry_interval': config.db_retry_interval,
+            'kwargs': {'pool_class': Pyscopg2Pool, }, # вот тут нужно передать новый класс Pool
+        }
+
+
+    db = Gino(**get_engine_config())
+
+GINO - GINO Is Not ORM - is a lightweight asynchronous ORM built on top of
+SQLAlchemy_ core for Python asyncio_. GINO 1.0 supports only PostgreSQL_ with asyncpg_.
+
+* Free software: BSD license
+* Requires: Python 3.5
+* GINO is developed proudly with |PyCharm|.
+
+
+Home
+----
+
+`python-gino.org <https://python-gino.org/>`__
+
+
+Documentation
+-------------
+
+* English_
+* Chinese_
+
+
+Installation
+------------
+
+.. code-block:: console
+
+    $ pip install gino
+
+
+Features
+--------
+
+* Robust SQLAlchemy-asyncpg bi-translator with no hard hack
+* Asynchronous SQLAlchemy-alike engine and connection
+* Asynchronous dialect API
+* Asynchronous-friendly CRUD objective models
+* Well-considered contextual connection and transaction management
+* Reusing native SQLAlchemy core to build queries with grammar sugars
+* Support SQLAlchemy ecosystem, e.g. Alembic_ for migration
+* `Community support <https://github.com/python-gino/>`_ for Starlette_/FastAPI_, aiohttp_, Sanic_, Tornado_ and Quart_
+* Rich PostgreSQL JSONB support
+
+.. role:: red
+.. _SQLAlchemy: https://www.sqlalchemy.org/
+.. _asyncpg: https://github.com/MagicStack/asyncpg
+.. _PostgreSQL: https://www.postgresql.org/
+.. _asyncio: https://docs.python.org/3/library/asyncio.html
+.. _Alembic: https://bitbucket.org/zzzeek/alembic
+.. _Sanic: https://github.com/channelcat/sanic
+.. _Tornado: http://www.tornadoweb.org/
+.. _Quart: https://gitlab.com/pgjones/quart/
+.. _English: https://python-gino.org/docs/en/
+.. _Chinese: https://python-gino.org/docs/zh/
+.. _aiohttp: https://github.com/aio-libs/aiohttp
+.. _Starlette: https://www.starlette.io/
+.. _FastAPI: https://fastapi.tiangolo.com/
+.. |PyCharm| image:: ./docs/images/pycharm.svg
+        :height: 20px
+        :target: https://www.jetbrains.com/?from=GINO
+
+.. |GINO| image:: ./docs/theme/static/logo.svg
+        :alt: GINO
+        :height: 64px
+        :target: https://python-gino.org/
--- contrib/python/gino/gino/crud.py	(950e9229c0861492fce25aa3aec3fcfb7406d504)
+++ contrib/python/gino/gino/crud.py	(0fe8277b31a46182f056caeea2c6904a7a3ff845)
@@ -167,7 +167,7 @@ class UpdateRequest:
         )
         if bind is None:
             bind = cls.__metadata__.bind
-        row = await bind.first(clause)
+        row = await bind.first(clause, read_only=False, reuse=False)
         if not row:
             raise NoSuchRowError()
         for k, v in row.items():
@@ -474,7 +474,7 @@ class CRUDModel(Model):
         )
         if bind is None:
             bind = cls.__metadata__.bind
-        row = await bind.first(q)
+        row = await bind.first(q, read_only=False, reuse=False)
         for k, v in row.items():
             self.__values__[self._column_name_map.invert_get(k)] = v
         self.__profile__ = None
@@ -596,7 +596,7 @@ class CRUDModel(Model):
             clause = clause.execution_options(timeout=timeout)
         if bind is None:
             bind = self.__metadata__.bind
-        return (await bind.status(clause))[0]
+        return (await bind.status(clause, read_only=False, reuse=False))[0]
 
     def to_dict(self):
         """
--- contrib/python/gino/gino/dialects/asyncpg.py	(950e9229c0861492fce25aa3aec3fcfb7406d504)
+++ contrib/python/gino/gino/dialects/asyncpg.py	(0fe8277b31a46182f056caeea2c6904a7a3ff845)
@@ -20,6 +20,7 @@ from sqlalchemy.dialects.postgresql.base import (
     PGExecutionContext,
 )
 from sqlalchemy.sql import sqltypes
+from pyscopg2.asyncpg import PoolManager
 
 from . import base
 
@@ -225,7 +226,7 @@ class Pool(base.Pool):
     def raw_pool(self):
         return self._pool
 
-    async def acquire(self, *, timeout=None):
+    async def acquire(self, *, timeout=None, read_only=False):
         return await self._pool.acquire(timeout=timeout)
 
     async def release(self, conn):
@@ -314,7 +315,7 @@ class NullPool(base.Pool):
     def raw_pool(self):
         return self
 
-    async def acquire(self, *, timeout=None):
+    async def acquire(self, *, timeout=None, read_only=False):
         args = self._kwargs.copy()
         if timeout is not None:
             args.update(timeout=timeout)
@@ -612,3 +613,36 @@ class AsyncpgDialect(PGDialect, base.AsyncDialectMixin):
                 ),
             )
         return bool(await connection.scalar(query))
+
+
+class Pyscopg2Pool(Pool):
+    async def _init(self):
+        pool_config = {
+            'ssl': self._kwargs.get('ssl', False),
+            'max_size': self._kwargs.get('max_size', 10),
+            'min_size': self._kwargs.get('min_size', 5),
+            'loop': self._loop,
+            'init': None,
+            'fallback_master': self._kwargs.get('fallback_master', True),
+        }
+        host_str = ','.join([f'{host}:{self._url.port}' for host in self._url.host.split(',')])
+        dsn = f'postgresql://{self._url.username}:{self._url.password}@{host_str}/{self._url.database}'
+
+        self._pool = await self.init_pool(dsn, pool_config)
+        return self
+
+    async def acquire(self, *, timeout=None, read_only=False):
+        return await self._pool.acquire(timeout=timeout, read_only=read_only)
+
+    async def init_pool(self, dsn, pool_config) -> PoolManager:
+        return PoolManager(dsn, fallback_master=pool_config.pop('fallback_master'), pool_factory_kwargs=pool_config)
+
+    def repr(self, color=None):
+        return "<{classname} max={max} min={min} cur={cur}>".format(
+            classname=self._pool.__class__.__module__
+                      + "."
+                      + self._pool.__class__.__name__,
+            max=self._pool._pool_factory_kwargs['max_size'],
+            min=self._pool._pool_factory_kwargs['min_size'],
+            cur=len(self._pool._pools)
+        )
--- contrib/python/gino/gino/engine.py	(950e9229c0861492fce25aa3aec3fcfb7406d504)
+++ contrib/python/gino/gino/engine.py	(0fe8277b31a46182f056caeea2c6904a7a3ff845)
@@ -33,10 +33,10 @@ class _BaseDBAPIConnection:
     def raw_connection(self):
         raise NotImplementedError
 
-    async def acquire(self, *, timeout=None):
+    async def acquire(self, *, timeout=None, read_only=False):
         if self._closed:
             raise ValueError("This connection is already released permanently.")
-        return await self._acquire(timeout)
+        return await self._acquire(timeout, read_only)
 
     async def _acquire(self, timeout):
         raise NotImplementedError
@@ -61,7 +61,7 @@ class _DBAPIConnection(_BaseDBAPIConnection):
     def raw_connection(self):
         return self._conn
 
-    async def _acquire(self, timeout):
+    async def _acquire(self, timeout, read_only):
         try:
             if timeout is None:
                 await self._lock.acquire()
@@ -71,7 +71,7 @@ class _DBAPIConnection(_BaseDBAPIConnection):
                 after = time.monotonic()
                 timeout -= after - before
             if self._conn is None:
-                self._conn = await self._pool.acquire(timeout=timeout)
+                self._conn = await self._pool.acquire(timeout=timeout, read_only=read_only)
             return self._conn
         finally:
             self._lock.release()
@@ -93,8 +93,8 @@ class _ReusingDBAPIConnection(_BaseDBAPIConnection):
     def raw_connection(self):
         return self._root.raw_connection
 
-    async def _acquire(self, timeout):
-        return await self._root.acquire(timeout=timeout)
+    async def _acquire(self, timeout, read_only):
+        return await self._root.acquire(timeout=timeout, read_only=read_only)
 
     async def _release(self):
         pass
@@ -619,7 +619,7 @@ class GinoEngine:
         """
         return self._pool.raw_pool
 
-    def acquire(self, *, timeout=None, reuse=False, lazy=False, reusable=True):
+    def acquire(self, *, timeout=None, reuse=False, lazy=False, reusable=True, read_only=False):
         """
         Acquire a connection from the pool.
 
@@ -682,10 +682,10 @@ class GinoEngine:
 
         """
         return _AcquireContext(
-            functools.partial(self._acquire, timeout, reuse, lazy, reusable)
+            functools.partial(self._acquire, timeout, reuse, lazy, reusable, read_only)
         )
 
-    async def _acquire(self, timeout, reuse, lazy, reusable):
+    async def _acquire(self, timeout, reuse, lazy, reusable, read_only):
         stack = _ContextualStack(self._ctx)
         if reuse and stack:
             dbapi_conn = _ReusingDBAPIConnection(self._dialect.cursor_cls, stack.top)
@@ -699,7 +699,7 @@ class GinoEngine:
         )
         dbapi_conn.gino_conn = rv
         if not lazy:
-            await dbapi_conn.acquire(timeout=timeout)
+            await dbapi_conn.acquire(timeout=timeout, read_only=read_only)
         if reusable:
             stack.push(dbapi_conn)
         return rv
@@ -736,7 +736,8 @@ class GinoEngine:
         The same applies for other query methods.
 
         """
-        async with self.acquire(reuse=True) as conn:
+        read_only = params.pop('read_only', True)
+        async with self.acquire(reuse=True, read_only=read_only) as conn:
             return await conn.all(clause, *multiparams, **params)
 
     async def first(self, clause, *multiparams, **params):
@@ -744,7 +745,9 @@ class GinoEngine:
         Runs :meth:`~.GinoConnection.first`, See :meth:`.all`.
 
         """
-        async with self.acquire(reuse=True) as conn:
+        read_only = params.pop('read_only', True)
+        reuse = params.pop('reuse', True)
+        async with self.acquire(reuse=reuse, read_only=read_only) as conn:
             return await conn.first(clause, *multiparams, **params)
 
     async def one_or_none(self, clause, *multiparams, **params):
@@ -752,7 +755,8 @@ class GinoEngine:
         Runs :meth:`~.GinoConnection.one_or_none`, See :meth:`.all`.
 
         """
-        async with self.acquire(reuse=True) as conn:
+        read_only = params.pop('read_only', True)
+        async with self.acquire(reuse=True, read_only=read_only) as conn:
             return await conn.one_or_none(clause, *multiparams, **params)
 
     async def one(self, clause, *multiparams, **params):
@@ -760,7 +764,8 @@ class GinoEngine:
         Runs :meth:`~.GinoConnection.one`, See :meth:`.all`.
 
         """
-        async with self.acquire(reuse=True) as conn:
+        read_only = params.pop('read_only', True)
+        async with self.acquire(reuse=True, read_only=read_only) as conn:
             return await conn.one(clause, *multiparams, **params)
 
     async def scalar(self, clause, *multiparams, **params):
@@ -768,7 +773,8 @@ class GinoEngine:
         Runs :meth:`~.GinoConnection.scalar`, See :meth:`.all`.
 
         """
-        async with self.acquire(reuse=True) as conn:
+        read_only = params.pop('read_only', True)
+        async with self.acquire(reuse=True, read_only=read_only) as conn:
             return await conn.scalar(clause, *multiparams, **params)
 
     async def status(self, clause, *multiparams, **params):
@@ -776,7 +782,9 @@ class GinoEngine:
         Runs :meth:`~.GinoConnection.status`. See also :meth:`.all`.
 
         """
-        async with self.acquire(reuse=True) as conn:
+        read_only = params.pop('read_only', True)
+        reuse = params.pop('reuse', True)
+        async with self.acquire(reuse=reuse, read_only=read_only) as conn:
             return await conn.status(clause, *multiparams, **params)
 
     def compile(self, clause, *multiparams, **params):
@@ -822,8 +830,9 @@ class GinoEngine:
           :class:`~gino.transaction.GinoTransaction`
 
         """
+        read_only = kwargs.pop('read_only', False)
         return _TransactionContext(
-            self.acquire(timeout=timeout, reuse=reuse, reusable=reusable),
+            self.acquire(timeout=timeout, reuse=reuse, reusable=reusable, read_only=read_only),
             (args, kwargs),
         )
 
--- contrib/python/gino/ya.make	(950e9229c0861492fce25aa3aec3fcfb7406d504)
+++ contrib/python/gino/ya.make	(0fe8277b31a46182f056caeea2c6904a7a3ff845)
@@ -8,6 +8,7 @@ LICENSE(BSD-3-Clause)
 
 PEERDIR(
     contrib/python/asyncpg
+    library/python/pyscopg2
     contrib/python/sqlalchemy/sqlalchemy-1.3
     contrib/python/importlib-metadata
 )
