--- contrib/python/rasterio/rasterio/_env.pyx	(index)
+++ contrib/python/rasterio/rasterio/_env.pyx	(working tree)
@@ -55,8 +55,8 @@ code_map = {
 log = logging.getLogger(__name__)
 
 try:
-    import certifi
-    ca_bundle = certifi.where()
+    import certifi.source
+    ca_bundle = certifi.source.where()
     os.environ.setdefault("GDAL_CURL_CA_BUNDLE", ca_bundle)
     os.environ.setdefault("PROJ_CURL_CA_BUNDLE", ca_bundle)
 except ImportError:
