--- contrib/python/langchain-core/langchain_core/prompts/prompt.py	(index)
+++ contrib/python/langchain-core/langchain_core/prompts/prompt.py	(working tree)
@@ -245,7 +245,10 @@
         Returns:
             The prompt loaded from the file.
         """
-        template = Path(template_file).read_text(encoding=encoding)
+        if isinstance(template_file, str):
+            template_file = Path(template_file)
+        with template_file.open(encoding=encoding) as f:
+            template = f.read()
         return cls.from_template(template=template, **kwargs)

     @classmethod
