--- contrib/python/xlwt/py2/tests/test_bitmaps.py	(index)
+++ contrib/python/xlwt/py2/tests/test_bitmaps.py	(working tree)
@@ -6,7 +6,7 @@ import unittest
 import filecmp
 from datetime import datetime
 
-from utils import in_tst_dir, in_tst_output_dir
+from .utils import in_tst_dir, in_tst_output_dir
 
 import xlwt
 
@@ -15,7 +15,7 @@ class TestBitmaps(unittest.TestCase):
         book = xlwt.Workbook()
         sheet = book.add_sheet('bitmap test')
         
-        with open ("tests/python.bmp", "rb") as bmpfile:
+        with open (in_tst_dir("python.bmp"), "rb") as bmpfile:
             bmpdata = bmpfile.read()
             sheet.insert_bitmap_data(bmpdata, 1, 1)
         
@@ -29,7 +29,7 @@ class TestBitmaps(unittest.TestCase):
         book = xlwt.Workbook()
         sheet = book.add_sheet('bitmap test')
         
-        sheet.insert_bitmap("tests/python.bmp", 1, 1)
+        sheet.insert_bitmap(in_tst_dir("python.bmp"), 1, 1)
         
         book.save(in_tst_output_dir('bitmaps.xls'))
         
--- contrib/python/xlwt/py2/tests/test_easyxf.py	(index)
+++ contrib/python/xlwt/py2/tests/test_easyxf.py	(working tree)
@@ -2,7 +2,7 @@ import unittest
 import filecmp
 import datetime
 
-from utils import in_tst_dir, in_tst_output_dir
+from .utils import in_tst_dir, in_tst_output_dir
 
 import xlwt as xlwt
 ezxf = xlwt.easyxf
--- contrib/python/xlwt/py2/tests/test_mini.py	(index)
+++ contrib/python/xlwt/py2/tests/test_mini.py	(working tree)
@@ -3,7 +3,7 @@ import os
 import unittest
 import filecmp
 
-from utils import in_tst_dir, in_tst_output_dir
+from .utils import in_tst_dir, in_tst_output_dir
 
 import xlwt
 
--- contrib/python/xlwt/py2/tests/test_simple.py	(index)
+++ contrib/python/xlwt/py2/tests/test_simple.py	(working tree)
@@ -6,7 +6,7 @@ import unittest
 import filecmp
 from datetime import datetime
 
-from utils import in_tst_dir, in_tst_output_dir
+from .utils import in_tst_dir, in_tst_output_dir
 
 import xlwt
 
--- contrib/python/xlwt/py2/tests/test_unicode1.py	(index)
+++ contrib/python/xlwt/py2/tests/test_unicode1.py	(working tree)
@@ -4,7 +4,7 @@ import filecmp
 
 import xlwt
 
-from utils import in_tst_dir, in_tst_output_dir
+from .utils import in_tst_dir, in_tst_output_dir
 
 def create_example_xls(filename):
     w = xlwt.Workbook()
--- contrib/python/xlwt/py2/tests/utils.py	(index)
+++ contrib/python/xlwt/py2/tests/utils.py	(working tree)
@@ -2,10 +2,18 @@ import os
 from os.path import join, dirname, abspath, isdir
 
 def in_tst_dir(filename):
-    return join(dirname(abspath(__file__)), filename)
+    try:
+        import yatest.common
+        return yatest.common.test_source_path(join(filename))
+    except ImportError:
+        return join(dirname(abspath(__file__)), filename)
 
 def in_tst_output_dir(filename):
-    output_dir = join(dirname(abspath(__file__)), 'output')
-    if not isdir(output_dir):
-        os.mkdir(output_dir, 0o755)
-    return join(output_dir, filename)
\ No newline at end of file
+    try:
+        import yatest.common
+        return yatest.common.test_output_path(filename)
+    except ImportError:
+        output_dir = join(dirname(abspath(__file__)), 'output')
+        if not isdir(output_dir):
+            os.mkdir(output_dir, 0o755)
+        return join(output_dir, filename)
--- contrib/python/xlwt/py2/xlwt/BIFFRecords.py	(index)
+++ contrib/python/xlwt/py2/xlwt/BIFFRecords.py	(working tree)
@@ -1,4 +1,4 @@
-# -*- coding: cp1252 -*-
+# -*- coding: utf-8 -*-
 from struct import pack
 from .UnicodeUtils import upack1, upack2, upack2rt
 from .compat import basestring, unicode, unicode_type, xrange, iteritems
@@ -1728,10 +1728,10 @@ class RefModeRecord(BiffRecord):
     """
     This record is part of the Calculation Settings Block.
     It stores which method is used to show cell addresses in formulas.
-    The ìRCî mode uses numeric indexes for rows and columns,
-    i.e. ìR(1)C(-1)î, or ìR1C1:R2C2î.
-    The ìA1î mode uses characters for columns and numbers for rows,
-    i.e. ìB1î, or ì$A$1:$B$2î.
+    The ‚ÄúRC‚Äù mode uses numeric indexes for rows and columns,
+    i.e. ‚ÄúR(1)C(-1)‚Äù, or ‚ÄúR1C1:R2C2‚Äù.
+    The ‚ÄúA1‚Äù mode uses characters for columns and numbers for rows,
+    i.e. ‚ÄúB1‚Äù, or ‚Äú$A$1:$B$2‚Äù.
 
     Record REFMODE, BIFF2-BIFF8:
 
@@ -1779,7 +1779,7 @@ class DeltaRecord(BiffRecord):
 class SaveRecalcRecord(BiffRecord):
     """
     This record is part of the Calculation Settings Block.
-    It contains the ìRecalculate before saveî option in
+    It contains the ‚ÄúRecalculate before save‚Äù option in
     Excel's calculation settings dialogue.
 
     Record SAVERECALC, BIFF3-BIFF8:
@@ -2445,7 +2445,7 @@ class ExternnameRecord(BiffRecord):
     0       0001H   0 = Standard name; 1 = Built-in name
     1       0002H   0 = Manual link; 1 = Automatic link (DDE links and OLE links only)
     2       0004H   1 = Picture link (DDE links and OLE links only)
-    3       0008H   1 = This is the ìStdDocumentNameî identifier (DDE links only)
+    3       0008H   1 = This is the ‚ÄúStdDocumentName‚Äù identifier (DDE links only)
     4       0010H   1 = OLE link
     14-5    7FE0H   Clipboard format of last successful update (DDE links and OLE links only)
     15      8000H   1 = Iconified picture link (BIFF8 OLE links only)
--- contrib/python/xlwt/py2/xlwt/Bitmap.py	(index)
+++ contrib/python/xlwt/py2/xlwt/Bitmap.py	(working tree)
@@ -1,4 +1,4 @@
-# -*- coding: windows-1251 -*-
+# -*- coding: utf-8 -*-
 
 #  Portions are Copyright (C) 2005 Roman V. Kiseliov
 #  Portions are Copyright (c) 2004 Evgeny Filatov <fufff@users.sourceforge.net>
--- contrib/python/xlwt/py2/xlwt/Cell.py	(index)
+++ contrib/python/xlwt/py2/xlwt/Cell.py	(working tree)
@@ -1,4 +1,4 @@
-# -*- coding: windows-1252 -*-
+# -*- coding: utf-8 -*-
 
 from struct import unpack, pack
 from . import BIFFRecords
--- contrib/python/xlwt/py2/xlwt/Column.py	(index)
+++ contrib/python/xlwt/py2/xlwt/Column.py	(working tree)
@@ -1,4 +1,4 @@
-# -*- coding: windows-1252 -*-
+# -*- coding: utf-8 -*-
 
 from .BIFFRecords import ColInfoRecord
 
--- contrib/python/xlwt/py2/xlwt/CompoundDoc.py	(index)
+++ contrib/python/xlwt/py2/xlwt/CompoundDoc.py	(working tree)
@@ -1,4 +1,4 @@
-# -*- coding: windows-1252 -*-
+# -*- coding: utf-8 -*-
 
 import struct
 from .compat import xrange
--- contrib/python/xlwt/py2/xlwt/ExcelFormula.py	(index)
+++ contrib/python/xlwt/py2/xlwt/ExcelFormula.py	(working tree)
@@ -1,4 +1,4 @@
-# -*- coding: windows-1252 -*-
+# -*- coding: utf-8 -*-
 
 from . import ExcelFormulaParser, ExcelFormulaLexer
 import struct
--- contrib/python/xlwt/py2/xlwt/ExcelFormulaLexer.py	(index)
+++ contrib/python/xlwt/py2/xlwt/ExcelFormulaLexer.py	(working tree)
@@ -1,5 +1,5 @@
 from __future__ import print_function
-# -*- coding: windows-1252 -*-
+# -*- coding: utf-8 -*-
 
 from .antlr import EOF, CommonToken as Tok, TokenStream, TokenStreamException
 from . import ExcelFormulaParser
--- contrib/python/xlwt/py2/xlwt/ExcelMagic.py	(index)
+++ contrib/python/xlwt/py2/xlwt/ExcelMagic.py	(working tree)
@@ -1,4 +1,4 @@
-# -*- coding: ascii -*-
+# -*- coding: utf-8 -*-
 """
 lots of Excel Magic Numbers
 """
--- contrib/python/xlwt/py2/xlwt/Row.py	(index)
+++ contrib/python/xlwt/py2/xlwt/Row.py	(working tree)
@@ -1,4 +1,4 @@
-# -*- coding: windows-1252 -*-
+# -*- coding: utf-8 -*-
 
 from decimal import Decimal
 from . import BIFFRecords
--- contrib/python/xlwt/py2/xlwt/Style.py	(index)
+++ contrib/python/xlwt/py2/xlwt/Style.py	(working tree)
@@ -1,5 +1,5 @@
 from __future__ import print_function
-# -*- coding: windows-1252 -*-
+# -*- coding: utf-8 -*-
 
 from . import Formatting
 from .BIFFRecords import NumberFormatRecord, XFRecord, StyleRecord
--- contrib/python/xlwt/py2/xlwt/UnicodeUtils.py	(index)
+++ contrib/python/xlwt/py2/xlwt/UnicodeUtils.py	(working tree)
@@ -1,4 +1,4 @@
-# -*- coding: windows-1252 -*-
+# -*- coding: utf-8 -*-
 
 '''
 From BIFF8 on, strings are always stored using UTF-16LE  text encoding. The
@@ -34,8 +34,8 @@ Offset  Size    Contents
 [2 or 3] 2      (optional, only if richtext=1) Number of Rich-Text formatting runs (rt)
 [var.]   4      (optional, only if phonetic=1) Size of Asian phonetic settings block (in bytes, sz)
 var.     ln or 
-         2∑ln   Character array (8-bit characters or 16-bit characters, dependent on ccompr)
-[var.]   4∑rt   (optional, only if richtext=1) List of rt formatting runs 
+         2¬∑ln   Character array (8-bit characters or 16-bit characters, dependent on ccompr)
+[var.]   4¬∑rt   (optional, only if richtext=1) List of rt formatting runs 
 [var.]   sz     (optional, only if phonetic=1) Asian Phonetic Settings Block 
 '''
 
--- contrib/python/xlwt/py2/xlwt/Workbook.py	(index)
+++ contrib/python/xlwt/py2/xlwt/Workbook.py	(working tree)
@@ -1,4 +1,4 @@
-# -*- coding: windows-1252 -*-
+# -*- coding: utf-8 -*-
 # Record Order in BIFF8
 #   Workbook Globals Substream
 #       BOF Type = workbook globals
--- contrib/python/xlwt/py2/xlwt/Worksheet.py	(index)
+++ contrib/python/xlwt/py2/xlwt/Worksheet.py	(working tree)
@@ -1,4 +1,4 @@
-# -*- coding: windows-1252 -*-
+# -*- coding: utf-8 -*-
 #             BOF
 #             UNCALCED
 #             INDEX
