--- contrib/python/google-api-core/py2/google/api_core/operations_v1/operations_client.py	(index)
+++ contrib/python/google-api-core/py2/google/api_core/operations_v1/operations_client.py	(working tree)
@@ -40,7 +40,7 @@ import functools
 from google.api_core import gapic_v1
 from google.api_core import page_iterator
 from google.api_core.operations_v1 import operations_client_config
-from google.longrunning import operations_pb2
+from google.longrunning import operations_pb2, operations_pb2_grpc
 
 
 class OperationsClient(object):
@@ -56,7 +56,7 @@ class OperationsClient(object):
 
     def __init__(self, channel, client_config=operations_client_config.config):
         # Create the gRPC client stub.
-        self.operations_stub = operations_pb2.OperationsStub(channel)
+        self.operations_stub = operations_pb2_grpc.OperationsStub(channel)
 
         # Create all wrapped methods using the interface configuration.
         # The interface config contains all of the default settings for retry
