--- contrib/python/thrift/tests/test_socket.py	(index)
+++ contrib/python/thrift/tests/test_socket.py	(working tree)
@@ -1,9 +1,7 @@
 import errno
 import unittest
 
-from test_sslsocket import ServerAcceptor
-
-import _import_local_thrift  # noqa
+from .test_sslsocket import ServerAcceptor
 
 from thrift.transport.TSocket import TServerSocket
 from thrift.transport.TSocket import TSocket
--- contrib/python/thrift/tests/test_sslsocket.py	(index)
+++ contrib/python/thrift/tests/test_sslsocket.py	(working tree)
@@ -29,9 +29,11 @@ import unittest
 import warnings
 from contextlib import contextmanager
 
-import _import_local_thrift  # noqa
+from thrift.transport.TSSLSocket import TSSLSocket, TSSLServerSocket, _match_has_ipaddress  # FIX_FOR_YANDEX moved here
+from thrift.transport.TTransport import TTransportException  # FIX_FOR_YANDEX moved here
+import yatest.common as yc
 
-SCRIPT_DIR = os.path.realpath(os.path.dirname(__file__))
+SCRIPT_DIR = os.path.realpath(os.path.dirname(yc.source_path(__file__)))
 ROOT_DIR = os.path.dirname(os.path.dirname(os.path.dirname(SCRIPT_DIR)))
 SERVER_PEM = os.path.join(ROOT_DIR, 'test', 'keys', 'server.pem')
 SERVER_CERT = os.path.join(ROOT_DIR, 'test', 'keys', 'server.crt')
