--- contrib/python/pyogrio/pyogrio/_io.pyx	(index)
+++ contrib/python/pyogrio/pyogrio/_io.pyx	(working tree)
@@ -270 +270 @@ cdef ogr_close(GDALDatasetH ogr_dataset):
-        IF CTE_GDAL_VERSION >= (3, 7, 0):
+        IF CTE_GDAL_VERSION >= 30700:
@@ -983 +983 @@ cdef process_fields(
-                IF CTE_GDAL_VERSION >= (3, 7, 0):
+                IF CTE_GDAL_VERSION >= 30700:
@@ -1647 +1647 @@ def ogr_open_arrow(
-    IF CTE_GDAL_VERSION < (3, 8, 0):
+    IF CTE_GDAL_VERSION < 30800:
@@ -1735 +1735 @@ def ogr_open_arrow(
-        IF CTE_GDAL_VERSION < (3, 8, 3):
+        IF CTE_GDAL_VERSION < 30803:
@@ -1764 +1764 @@ def ogr_open_arrow(
-            IF CTE_GDAL_VERSION < (3, 8, 0):
+            IF CTE_GDAL_VERSION < 30800:
@@ -1826 +1826 @@ def ogr_open_arrow(
-        IF CTE_GDAL_VERSION >= (3, 8, 0):
+        IF CTE_GDAL_VERSION >= 30800:
@@ -1835 +1835 @@ def ogr_open_arrow(
-        IF CTE_GDAL_VERSION >= (3, 11, 0):
+        IF CTE_GDAL_VERSION >= 31100:
@@ -2197 +2197 @@ cdef void * ogr_create(
-    IF CTE_GDAL_VERSION < (3, 8, 0):
+    IF CTE_GDAL_VERSION < 30800:
@@ -2873 +2873 @@ def ogr_write_arrow(
-    IF CTE_GDAL_VERSION < (3, 8, 0):
+    IF CTE_GDAL_VERSION < 30800:
@@ -3054 +3054 @@ cdef create_fields_from_arrow_schema(
-    IF CTE_GDAL_VERSION < (3, 8, 0):
+    IF CTE_GDAL_VERSION < 30800:
--- contrib/python/pyogrio/pyogrio/_ogr.pxd	(index)
+++ contrib/python/pyogrio/pyogrio/_ogr.pxd	(working tree)
@@ -426 +426 @@ cdef extern from "ogr_api.h":
-IF CTE_GDAL_VERSION >= (3, 7, 0):
+IF CTE_GDAL_VERSION >= 30700:
@@ -434 +434 @@ IF CTE_GDAL_VERSION >= (3, 7, 0):
-IF CTE_GDAL_VERSION >= (3, 8, 0):
+IF CTE_GDAL_VERSION >= 30800:
@@ -520 +520 @@ cdef extern from "gdal.h":
-IF CTE_GDAL_VERSION >= (3, 7, 0):
+IF CTE_GDAL_VERSION >= 30700:
