https://github.com/mongodb/motor/pull/185

--- contrib/python/motor/motor/frameworks/asyncio/__init__.py	(index)
+++ contrib/python/motor/motor/frameworks/asyncio/__init__.py	(working tree)
@@ -25,7 +25,6 @@ import multiprocessing
 import os
 import warnings
 
-from asyncio import coroutine  # For framework interface.
 from concurrent.futures import ThreadPoolExecutor
 
 
@@ -35,6 +34,16 @@ except ImportError:
     contextvars = None
 
 
+try:
+    from asyncio import coroutine
+except ImportError:
+
+    def coroutine():
+        raise RuntimeError(
+            "The coroutine decorator was removed in Python 3.11.  Use 'async def' instead"
+        )
+
+
 CLASS_PREFIX = 'AsyncIO'
 
 
