--- contrib/python/py-asciimath/py_asciimath/transformer/transformer.py	(index)
+++ contrib/python/py-asciimath/py_asciimath/transformer/transformer.py	(working tree)
@@ -375,6 +375,11 @@ class Tex2ASCIIMathTransformer(MathTransformer):  # pragma: no cover
 
         return decorator
 
+    def _add_brackets(self, item):
+        if item.isdigit() or len(item.split()) == 1:
+            return item
+        return "(" + item + ")"
+
     @log
     def exp(self, items):
         return " ".join(items)
@@ -387,26 +392,26 @@ class Tex2ASCIIMathTransformer(MathTransformer):  # pragma: no cover
     def exp_frac(self, items):
         items[0] = self.remove_parenthesis(items[0])
         items[1] = self.remove_parenthesis(items[1])
-        return "(" + items[0] + ")/(" + items[1] + ")"
+        return self._add_brackets(items[0]) + "/" + self._add_brackets(items[1])
 
     @log
     def exp_under(self, items):
         items[0] = self.remove_parenthesis(items[0])
         items[1] = self.remove_parenthesis(items[1])
-        return "(" + items[0] + ")_(" + items[1] + ")"
+        return self._add_brackets(items[0]) + "_" + self._add_brackets(items[1])
 
     @log
     def exp_super(self, items):
         items[0] = self.remove_parenthesis(items[0])
         items[1] = self.remove_parenthesis(items[1])
-        return "(" + items[0] + ")^(" + items[1] + ")"
+        return self._add_brackets(items[0]) + "^" + self._add_brackets(items[1])
 
     @log
     def exp_under_super(self, items):
         items[0] = self.remove_parenthesis(items[0])
         items[1] = self.remove_parenthesis(items[1])
         items[2] = self.remove_parenthesis(items[2])
-        return "(" + items[0] + ")_(" + items[1] + ")^(" + items[2] + ")"
+        return self._add_brackets(items[0]) + "_" + self._add_brackets(items[1]) + "^" + self._add_brackets(items[2])
 
     @log
     def exp_par(self, items):
@@ -433,7 +438,9 @@ class Tex2ASCIIMathTransformer(MathTransformer):  # pragma: no cover
     @log
     def exp_binary(self, items):
         if items[0].startswith("\\sqrt"):
-            return "root(" + "".join(items[1:-1]) + ")(" + items[-1] + ")"
+            return "root(" + "".join(items[1:-1]) + ", " + items[-1] + ")"
+        if items[0].startswith("\\frac"):
+            return self._add_brackets(items[1]) + "/" + self._add_brackets(items[2])
         return l2mml_bin[items[0]] + "(" + items[1] + ")(" + items[2] + ")"
 
     @log
--- contrib/python/py-asciimath/py_asciimath/latex.py	(index)
+++ contrib/python/py-asciimath/py_asciimath/latex.py	(working tree)
@@ -151,6 +151,8 @@ relation_symbols = {
     ">": {"asciimath": [">", "gt"], "mathml": "&gt;"},
     "\\le": {"asciimath": ["<=", "le"], "mathml": "&leq;"},
     "\\ge": {"asciimath": [">=", "ge"], "mathml": "&geq;"},
+    "\\leqslant": {"asciimath": ["<=", "le"], "mathml": "&leq;"},
+    "\\geqslant": {"asciimath": [">=", "ge"], "mathml": "&geq;"},
     "\\prec": {"asciimath": ["-<", "prec"], "mathml": "&Precedes;"},
     "\\preceq": {"asciimath": ["-<=", "preceq"], "mathml": "&PrecedesEqual;"},
     "\\succ": {"asciimath": [">-", "succ"], "mathml": "&Succeeds;"},
@@ -192,6 +194,7 @@ function_symbols = {
     "\\exp": {"asciimath": "exp", "mathml": "exp"},
     "\\log": {"asciimath": "log", "mathml": "log"},
     "\\ln": {"asciimath": "ln", "mathml": "ln"},
+    "\\lg": {"asciimath": "lg", "mathml": "lg"},
     "\\det": {"asciimath": "det", "mathml": "det"},
     "\\dim": {"asciimath": "dim", "mathml": "dim"},
     "\\mod": {"asciimath": "mod", "mathml": "mod"},
@@ -246,7 +249,6 @@ greek_letters = {
 }
 
 left_parenthesis = {
-    "(": {"asciimath": "(", "mathml": "("},
     "[": {"asciimath": "[", "mathml": "["},
     # ".": {"asciimath": "{:", "mathml": ""},
     "\\{": {"asciimath": "{", "mathml": "{"},
@@ -256,7 +258,6 @@ left_parenthesis = {
 }
 
 right_parenthesis = {
-    ")": {"asciimath": ")", "mathml": ")"},
     "]": {"asciimath": "]", "mathml": "]"},
     # ".": {"asciimath": ":}", "mathml": ""},
     "\\}": {"asciimath": "}", "mathml": "}"},
@@ -303,6 +304,8 @@ colors = {
 }
 
 misc_symbols = {
+    "(": {"asciimath": "(", "mathml": "("},
+    ")": {"asciimath": ")", "mathml": ")"},
     "^": {"asciimath": "^", "mathml": "&#x5E;"},
     ",": {"asciimath": ",", "mathml": ","},
     ".": {"asciimath": ".", "mathml": "."},
@@ -344,6 +347,7 @@ misc_symbols = {
     "\\mathbb{Q}": {"asciimath": "QQ", "mathml": "&Qopf;"},
     "\\mathbb{R}": {"asciimath": "RR", "mathml": "&Ropf;"},
     "\\mathbb{Z}": {"asciimath": "ZZ", "mathml": "&Zopf;"},
+    "\\prime": {"asciimath": "'", "mathml": "&prime;"},
 }
 
 # matrix2par = {
