--- contrib/python/pyphen/tests/test_pyphen.py	(index)
+++ contrib/python/pyphen/tests/test_pyphen.py	(working tree)
@@ -11,6 +11,7 @@ Tests can be launched with Pytest.
 from pathlib import Path
 
 import pyphen
+import yatest.common as yc
 
 
 def test_inserted():
@@ -64 +65 @@ def test_dict_from_filename():
-    dic_path = Path(__file__).parents[1] / 'pyphen' / 'dictionaries' / 'hyph_fr.dic'
+    dic_path = Path(yc.source_path(__file__)).parents[1] / 'pyphen' / 'dictionaries' / 'hyph_fr.dic'
@@ -71 +72 @@ def test_dict_from_path():
-    dic_path = Path(__file__).parents[1] / 'pyphen' / 'dictionaries' / 'hyph_fr.dic'
+    dic_path = Path(yc.source_path(__file__)).parents[1] / 'pyphen' / 'dictionaries' / 'hyph_fr.dic'
