--- contrib/python/pycryptodome/py2/ya.make	(index)
+++ contrib/python/pycryptodome/py2/ya.make	(working tree)
@@ -41,12 +42,20 @@ CFLAGS(
     -Dblake2s_digest=pycryptodome_blake2s_digest
     -Dblake2s_copy=pycryptodome_blake2s_copy
-    -DUSE_SSE2
-    -DHAVE_EMMINTRIN_H
-    -DHAVE_WMMINTRIN_H
-    -DHAVE_TMMINTRIN_H
-    -DPYCRYPTO_LITTLE_ENDIAN
-    -DHAVE_UINT128
 )
 
+IF (ARCH_X86_64)
+    CFLAGS(
+        -DUSE_SSE2
+        -DHAVE_EMMINTRIN_H
+        -DHAVE_WMMINTRIN_H
+        -DHAVE_TMMINTRIN_H
+    )
+ELSEIF (OS_DARWIN AND ARCH_ARM64)
+    CFLAGS(
+        -DPYCRYPTO_LITTLE_ENDIAN
+        -DHAVE_UINT128
+    )
+ENDIF()
+
 SRCS(
     src/AES.c
@@ -100,2 +100,0 @@ SRCS(
-    src/AESNI.c
-    GLOBAL src/AESNI_syms.cpp
@@ -102,2 +102,0 @@ SRCS(
-    src/ghash_clmul.c
-    GLOBAL src/ghash_clmul_syms.cpp
@@ -138,6 +143,17 @@ SRCS(
     GLOBAL src/strxor_syms.cpp
 )
 
+IF (ARCH_X86_64)
+    SRCS(
+        # src/AESNI.c - see below
+        GLOBAL src/AESNI_syms.cpp
+        # src/ghash_clmul.c - see below
+        GLOBAL src/ghash_clmul_syms.cpp
+    )
+    SRC(src/AESNI.c -maes)
+    SRC(src/ghash_clmul.c -mpclmul -mssse3)
+ENDIF()
+
 PY_SRCS(
     TOP_LEVEL
     Crypto/Cipher/AES.py
