--- contrib/python/Flask-Migrate/py3/flask_migrate/__init__.py	(index)
+++ contrib/python/Flask-Migrate/py3/flask_migrate/__init__.py	(working tree)
@@ -37,8 +37,25 @@ class Config(AlembicConfig):
     def get_template_directory(self):
         if self.template_directory:
             return self.template_directory
-        package_dir = os.path.abspath(os.path.dirname(__file__))
-        return os.path.join(package_dir, 'templates')
+        import tempfile
+        from library.python import resource
+
+        package_dir = os.path.dirname(__file__)
+        prefix = 'resfs/file/{}/templates/'.format(package_dir)
+
+        self.tmpdir = tempfile.TemporaryDirectory()
+
+        for key, data in resource.iteritems(prefix=prefix, strip_prefix=True):
+            filename = os.path.join(self.tmpdir.name, key)
+            os.makedirs(os.path.dirname(filename), exist_ok=True)
+            with open(filename, 'wb') as f_data:
+                f_data.write(data)
+
+        return self.tmpdir.name
+
+    def __del__(self):
+        if hasattr(self, 'tmpdir'):
+            self.tmpdir.cleanup()
 
 
 class Migrate(object):
