--- contrib/python/pandas/py3/pandas/_libs/src/klib/khash_python.h	(index)
+++ contrib/python/pandas/py3/pandas/_libs/src/klib/khash_python.h	(working tree)
@@ -23,1 +23,1 @@ typedef npy_complex128 khcomplex128_t;
-void *traced_malloc(size_t size){
+static void *traced_malloc(size_t size){
@@ -31,1 +31,1 @@ void *traced_malloc(size_t size){
-void *traced_calloc(size_t num, size_t size){
+static void *traced_calloc(size_t num, size_t size){
@@ -39,1 +39,1 @@ void *traced_calloc(size_t num, size_t size){
-void *traced_realloc(void* old_ptr, size_t size){
+static void *traced_realloc(void* old_ptr, size_t size){
@@ -50,1 +50,1 @@ void *traced_realloc(void* old_ptr, size_t size){
-void traced_free(void* ptr){
+static void traced_free(void* ptr){
--- contrib/python/pandas/py3/pandas/_libs/src/ujson/python/date_conversions.c	(index)
+++ contrib/python/pandas/py3/pandas/_libs/src/ujson/python/date_conversions.c	(working tree)
@@ -9,2 +9,2 @@ The full license is in the LICENSE file, distributed with this software.
-#include <../../../tslibs/src/datetime/np_datetime.h>
-#include <../../../tslibs/src/datetime/np_datetime_strings.h>
+#include "../../../tslibs/src/datetime/np_datetime.h"
+#include "../../../tslibs/src/datetime/np_datetime_strings.h"
--- contrib/python/pandas/py3/pandas/compat/_optional.py	(index)
+++ contrib/python/pandas/py3/pandas/compat/_optional.py	(working tree)
@@ -41,1 +41,1 @@ VERSIONS = {
-    "sqlalchemy": "1.4.16",
+    "sqlalchemy": "1.2.0",
--- contrib/python/pandas/py3/ya.make	(index)
+++ contrib/python/pandas/py3/ya.make	(working tree)
@@ -44,2 +44,4 @@ CFLAGS(
 
+INCLUDE(symbols.cmake)
+
 SRCS(
