--- contrib/python/sqlalchemy/sqlalchemy-1.3/sqlalchemy/dialects/postgresql/base.py	(index)
+++ contrib/python/sqlalchemy/sqlalchemy-1.3/sqlalchemy/dialects/postgresql/base.py	(working tree)
@@ -2400,8 +2400,7 @@ class PGTypeCompiler(compiler.GenericTypeCompiler):
 
     def visit_ARRAY(self, type_, **kw):
 
-        # TODO: pass **kw?
-        inner = self.process(type_.item_type)
+        inner = self.process(type_.item_type, **kw)
         return re.sub(
             r"((?: COLLATE.*)?)$",
             (
--- contrib/python/sqlalchemy/sqlalchemy-1.3/sqlalchemy/sql/compiler.py	(index)
+++ contrib/python/sqlalchemy/sqlalchemy-1.3/sqlalchemy/sql/compiler.py	(working tree)
@@ -952,6 +952,7 @@ class SQLCompiler(Compiled):
 
     def visit_typeclause(self, typeclause, **kw):
         kw["type_expression"] = typeclause
+        kw["identifier_preparer"] = self.preparer
         return self.dialect.type_compiler.process(typeclause.type, **kw)
 
     def post_process_text(self, text):
