--- contrib/python/dask/dask/hashing.py	(index)
+++ contrib/python/dask/dask/hashing.py	(working tree)
@@ -20,7 +20,7 @@ else:
     # CityHash disabled unless the reference leak in
     # https://github.com/escherba/python-cityhash/pull/16
     # is fixed.
-    if cityhash.__version__ >= "0.2.2":
+    if getattr(cityhash, "__version__", "") >= "0.2.2":
 
         def _hash_cityhash(buf):
             """
