--- contrib/python/dask/dask/config.py	(index)
+++ contrib/python/dask/dask/config.py	(working tree)
@@ -712,1 +712,2 @@ refresh()
-    with open(fn) as f:
+    import importlib.resources
+    with (importlib.resources.files(__package__) / "dask.yaml").open(mode="rb") as f:
--- contrib/python/dask/dask/widgets/widgets.py	(index)
+++ contrib/python/dask/dask/widgets/widgets.py	(working tree)
@@ -5,1 +5,1 @@ import datetime
-from jinja2 import Environment, FileSystemLoader, Template
+from jinja2 import Environment, FileSystemLoader, Template, PackageLoader
@@ -25,1 +25,1 @@ TEMPLATE_PATHS = [os.path.join(os.path.dirname(os.path.abspath(__file__)), "temp
-    loader = FileSystemLoader(TEMPLATE_PATHS)
+    loader = PackageLoader(__package__)
