https://github.com/kmike/pymorphy2/py2/pull/157
--- contrib/python/pymorphy2/py2/pymorphy2/units/base.py	(index)
+++ contrib/python/pymorphy2/py2/pymorphy2/units/base.py	(working tree)
@@ -1,6 +1,7 @@
 # -*- coding: utf-8 -*-
 from __future__ import absolute_import, unicode_literals, division
 import inspect
+import sys
 
 from pymorphy2.utils import kwargs_repr
 from pymorphy2.units.utils import (
@@ -67,7 +68,10 @@ class BaseAnalyzerUnit(object):
         """
         if cls.__init__ is object.__init__:
             return []
-        args, varargs, kw, default = inspect.getargspec(cls.__init__)
+        if sys.version_info[0] == 2:
+            args, _, _, _ = inspect.getargspec(cls.__init__)
+        else:
+            args, _, _, _, _, _, _ = inspect.getfullargspec(cls.__init__)
         return sorted(args[1:])
 
     def _get_params(self):
