--- contrib/python/setuptools/py3/pkg_resources/__init__.py	(index)
+++ contrib/python/setuptools/py3/pkg_resources/__init__.py	(working tree)
@@ -3330,6 +3330,9 @@ class UnionProvider(EmptyProvider):

         return False

+    def get_resource_filename(self, manager, resource_name):
+        return self._fn(self.module_path, resource_name)[0][1]
+

 class ResProvider(EmptyProvider):
     _resource_fs = {}
