--- contrib/python/transliterate/py2/transliterate/discover.py	(index)
+++ contrib/python/transliterate/py2/transliterate/discover.py	(working tree)
@@ -19,27 +19,9 @@ __all__ = ('autodiscover',)
 
 def autodiscover():
     """Auto-discover the language packs in contrib/apps."""
-    languages_dir = get_setting('LANGUAGES_DIR')
-    language_pack_module_name = get_setting('LANGUAGE_PACK_MODULE_NAME')
-    debug = get_setting('DEBUG')
-
-    for app_path in os.listdir(PROJECT_DIR(languages_dir)):
-        full_app_path = list(languages_dir)
-        full_app_path.append(app_path)
-        if os.path.isdir(PROJECT_DIR(full_app_path)):
-            try:
-                import_module(
-                    "transliterate.{0}.{1}.{2}".format(
-                        '.'.join(languages_dir),
-                        app_path,
-                        language_pack_module_name
-                    )
-                )
-            except ImportError as err:
-                if debug:
-                    print_(err)
-            except Exception as err:
-                if debug:
-                    print_(err)
-        else:
-            pass
+
+    import sys
+
+    for module_name in sys.extra_modules:
+        if module_name.startswith('transliterate.contrib.languages') and module_name.endswith('translit_language_pack'):
+            import_module(module_name)
