--- contrib/python/Flask-Migrate/py2/flask_migrate/__init__.py	(index)
+++ contrib/python/Flask-Migrate/py2/flask_migrate/__init__.py	(working tree)
@@ -35,8 +35,29 @@ class _MigrateConfig(object):
 
 class Config(AlembicConfig):
     def get_template_directory(self):
-        package_dir = os.path.abspath(os.path.dirname(__file__))
-        return os.path.join(package_dir, 'templates')
+        import tempfile
+        from library.python import resource
+
+        package_dir = os.path.dirname(__file__)
+        prefix = 'resfs/file/{}/templates/'.format(package_dir)
+
+        self.tmpdir = tempfile.mkdtemp()
+
+        for key, data in resource.iteritems(prefix=prefix, strip_prefix=True):
+            filename = os.path.join(self.tmpdir, key)
+            try:
+                os.makedirs(os.path.dirname(filename))
+            except:
+                pass
+            with open(filename, 'wb') as f_data:
+                f_data.write(data)
+
+        return self.tmpdir
+
+    def __del__(self):
+        if hasattr(self, 'tmpdir'):
+            import shutil
+            shutil.rmtree(self.tmpdir, ignore_errors=True)
 
 
 class Migrate(object):
