--- contrib/python/scales/greplin/scales/formats.py	(index)
+++ contrib/python/scales/greplin/scales/formats.py	(working tree)
@@ -16,1 +16,1 @@
-import cgi
+import html
@@ -105,1 +109,1 @@ def _htmlRenderDict(pathParts, statDict, output):
-    keyStr = cgi.escape(_utf8str(key))
+    keyStr = html.escape(_utf8str(key))
@@ -119,1 +123,1 @@ def _htmlRenderDict(pathParts, statDict, output):
-                   (keyStr, type(value).__name__, cgi.escape(_utf8str(value)).replace('\n', '<br/>')))
+                   (keyStr, type(value).__name__, html.escape(_utf8str(value)).replace('\n', '<br/>')))
