--- contrib/python/scales/greplin/scales/aggregation_test.py	(index)
+++ contrib/python/scales/greplin/scales/aggregation_test.py	(working tree)
@@ -48,2 +48,2 @@ class AggregationTest(unittest.TestCase):
-    self.assertEquals(result['a']['success']['sum'], 20)
-    self.assertEquals(result['a']['error']['sum'], 4)
+    self.assertEqual(result['a']['success']['sum'], 20)
+    self.assertEqual(result['a']['error']['sum'], 4)
--- contrib/python/scales/greplin/scales/formats_test.py	(index)
+++ contrib/python/scales/greplin/scales/formats_test.py	(working tree)
@@ -67 +67 @@ class StatsTest(unittest.TestCase):
-    self.assertEquals('{"here": {"count": 1}}\n', out.getvalue())
+    self.assertEqual('{"here": {"count": 1}}\n', out.getvalue())
@@ -94 +94 @@ class UnicodeFormatTest(unittest.TestCase):
-    self.assertEquals(stats, json.loads(out.getvalue()))
+    self.assertEqual(stats, json.loads(out.getvalue()))
@@ -102 +102 @@ class UnicodeFormatTest(unittest.TestCase):
-    self.assertEquals(json.loads(out.getvalue()), {'garbage': six.u('\xc2\xc2 ROAR!! \0\0')})
+    self.assertEqual(json.loads(out.getvalue()), {'garbage': six.u('\xc2\xc2 ROAR!! \0\0')})
--- contrib/python/scales/greplin/scales/scales_test.py	(index)
+++ contrib/python/scales/greplin/scales/scales_test.py	(working tree)
@@ -133 +133 @@ class StatsTest(unittest.TestCase):
-    self.assertEquals({
+    self.assertEqual({
@@ -162 +162 @@ class StatsTest(unittest.TestCase):
-    self.assertEquals({
+    self.assertEqual({
@@ -187 +187 @@ class StatsTest(unittest.TestCase):
-    self.assertEquals({
+    self.assertEqual({
@@ -216 +216 @@ class StatsTest(unittest.TestCase):
-    self.assertEquals({
+    self.assertEqual({
@@ -224 +224 @@ class StatsTest(unittest.TestCase):
-    self.assertEquals({
+    self.assertEqual({
@@ -234 +234 @@ class StatsTest(unittest.TestCase):
-    self.assertEquals({
+    self.assertEqual({
@@ -246 +246 @@ class StatsTest(unittest.TestCase):
-    self.assertEquals({
+    self.assertEqual({
@@ -269 +269 @@ class StatsTest(unittest.TestCase):
-      self.assertEquals({
+      self.assertEqual({
@@ -286 +286 @@ class StatsTest(unittest.TestCase):
-    self.assertEquals({
+    self.assertEqual({
@@ -316 +316 @@ class StatsTest(unittest.TestCase):
-    self.assertEquals({
+    self.assertEqual({
@@ -344 +344 @@ class StatsTest(unittest.TestCase):
-    self.assertEquals({
+    self.assertEqual({
@@ -363 +363 @@ class StatsTest(unittest.TestCase):
-    self.assertEquals(100, scales.getStats()['dynamic']())
+    self.assertEqual(100, scales.getStats()['dynamic']())
@@ -366 +366 @@ class StatsTest(unittest.TestCase):
-    self.assertEquals(200, scales.getStats()['dynamic']())
+    self.assertEqual(200, scales.getStats()['dynamic']())
@@ -376 +376 @@ class StatsTest(unittest.TestCase):
-    self.assertEquals({
+    self.assertEqual({
