--- contrib/python/marisa-trie/py2/marisa-trie/include/marisa/exception.h	(index)
+++ contrib/python/marisa-trie/py2/marisa-trie/include/marisa/exception.h	(working tree)
@@ -19,7 +21,7 @@ class Exception : public std::exception {
   Exception(const Exception &ex)
       : std::exception(), filename_(ex.filename_), line_(ex.line_),
         error_code_(ex.error_code_), error_message_(ex.error_message_) {}
-  virtual ~Exception() throw() {}
+  virtual ~Exception() {}
 
   Exception &operator=(const Exception &rhs) {
     filename_ = rhs.filename_;
@@ -42,7 +44,7 @@ class Exception : public std::exception {
     return error_message_;
   }
 
-  virtual const char *what() const throw() {
+  virtual const char *what() const noexcept {
     return error_message_;
   }
 
