--- a/include/oneapi/tbb/detail/_config.h      (index)
+++ b/include/oneapi/tbb/detail/_config.h      (working tree)
@@ -246,1 +246,6 @@
-    #define __TBB_CPP20_CONCEPTS_PRESENT           (_MSC_VER >= 1923 && __TBB_LANG >= 202002L) // TODO: INTEL_COMPILER?
+    #if !defined(__cpp_lib_concepts)
+        // TBB's concepts use STL's concepts
+        #define __TBB_CPP20_CONCEPTS_PRESENT       0
+    #else
+        #define __TBB_CPP20_CONCEPTS_PRESENT       (_MSC_VER >= 1923 && __TBB_LANG >= 202002L) // TODO: INTEL_COMPILER?
+    #endif
@@ -258,2 +258,1 @@
-#define __TBB_CPP17_MEMORY_RESOURCE_PRESENT             (_MSC_VER >= 1913 && (__TBB_LANG > 201402L) || \
-                                                        __TBB_GLIBCXX_VERSION >= 90000 && __TBB_LANG >= 201703L)
+#define __TBB_CPP17_MEMORY_RESOURCE_PRESENT             0
