--- contrib/python/typed-argument-parser/tests/test_utils.py	(index)
+++ contrib/python/typed-argument-parser/tests/test_utils.py	(working tree)
@@ -9,1 +9,1 @@ import subprocess
-from unittest import TestCase
+from unittest import TestCase, skip
@@ -24,4 +24,5 @@ from tap.utils import (
 
 
+@skip("Skip git tests")
 class GitTests(TestCase):
     def setUp(self) -> None:
--- contrib/python/typed-argument-parser/tests/test_to_tap_class.py	(index)
+++ contrib/python/typed-argument-parser/tests/test_to_tap_class.py	(working tree)
@@ -339 +339 @@ def test_subclasser_simple_help_message(class_or_function_: Any):
-    usage: pytest --arg_int ARG_INT [--arg_bool] [--arg_list [ARG_LIST {_ARG_LIST_DOTS}]] [-h]
+    usage: contrib-python-typed-argument-parser-tests --arg_int ARG_INT [--arg_bool] [--arg_list [ARG_LIST {_ARG_LIST_DOTS}]] [-h]
@@ -413 +413 @@ def test_subclasser_complex_help_message(class_or_function_: Any):
-    usage: pytest [-arg ARGUMENT_WITH_REALLY_LONG_NAME] --arg_int ARG_INT [--arg_bool]
+    usage: contrib-python-typed-argument-parser-tests [-arg ARGUMENT_WITH_REALLY_LONG_NAME] --arg_int ARG_INT [--arg_bool]
@@ -500 +500 @@ def test_subclasser_subparser(
-            usage: pytest [--foo] --arg_int ARG_INT [--arg_bool] [--arg_list [ARG_LIST {_ARG_LIST_DOTS}]] [-h]
+            usage: contrib-python-typed-argument-parser-tests [--foo] --arg_int ARG_INT [--arg_bool] [--arg_list [ARG_LIST {_ARG_LIST_DOTS}]] [-h]
@@ -523 +523 @@ def test_subclasser_subparser(
-            usage: pytest a --bar BAR [-h]
+            usage: contrib-python-typed-argument-parser-tests a --bar BAR [-h]
@@ -536 +536 @@ def test_subclasser_subparser(
-            usage: pytest b --baz {{X,Y,Z}} [-h]
+            usage: contrib-python-typed-argument-parser-tests b --baz {{X,Y,Z}} [-h]
