--- contrib/python/grafanalib/grafanalib/core.py	(index)
+++ contrib/python/grafanalib/grafanalib/core.py	(working tree)
@@ -581,6 +581,7 @@ class Target(object):
     target = attr.ib(default="")
     instant = attr.ib(validator=instance_of(bool), default=False)
     datasource = attr.ib(default=None)
+    tags = attr.ib(default={})
 
     def to_json_data(self):
         return {
@@ -597,6 +598,7 @@ class Target(object):
             'step': self.step,
             'instant': self.instant,
             'datasource': self.datasource,
+            'tags': self.tags,
         }
 
 
