--- contrib/python/swagger-spec-validator/py2/swagger_spec_validator/common.py	(index)
+++ contrib/python/swagger-spec-validator/py2/swagger_spec_validator/common.py	(working tree)
@@ -11,7 +11,7 @@ import sys
 
 import six
 import yaml
-from pkg_resources import resource_filename
+from pkg_resources import resource_filename, resource_string
 from six.moves.urllib.parse import urljoin
 from six.moves.urllib.request import pathname2url
 from six.moves.urllib.request import urlopen
@@ -60,7 +60,7 @@ def read_file(file_path):
 @lru_cache()
 def read_resource_file(resource_path):
     schema_path = resource_filename('swagger_spec_validator', resource_path)
-    return read_file(schema_path), schema_path
+    return yaml.load(resource_string('swagger_spec_validator', resource_path).decode('utf-8'), Loader=SafeLoader), schema_path
 
 
 def read_url(url, timeout=TIMEOUT_SEC):
