diff --git a/src/support/win32/thread_win32.cpp b/src/support/win32/thread_win32.cpp
index a651a3c..8736a53 100644
--- a/src/support/win32/thread_win32.cpp
+++ b/src/support/win32/thread_win32.cpp
@@ -11,9 +11,9 @@
 
 #define NOMINMAX
 #define WIN32_LEAN_AND_MEAN
-#include <fibersapi.h>
-#include <process.h>
 #include <windows.h>
+#include <process.h>
+#include <fibersapi.h>
 
 _LIBCPP_BEGIN_NAMESPACE_STD
 
