diff --git a/include/__configuration/abi.h b/include/__configuration/abi.h
index 5e40e30..c81ccee 100644
--- a/include/__configuration/abi.h
+++ b/include/__configuration/abi.h
@@ -64,7 +64,12 @@
 // Previously libc++ used "unsigned int" exclusively.
 #  define _LIBCPP_ABI_VARIANT_INDEX_TYPE_OPTIMIZATION
 // Unstable attempt to provide a more optimized std::function
-#  define _LIBCPP_ABI_OPTIMIZED_FUNCTION
+#  ifdef __EMSCRIPTEN__
+// XXX EMSCRIPTEN https://github.com/emscripten-core/emscripten/issues/11022
+// #  define _LIBCPP_ABI_OPTIMIZED_FUNCTION
+#  else
+#    define _LIBCPP_ABI_OPTIMIZED_FUNCTION
+#  endif
 // All the regex constants must be distinct and nonzero.
 #  define _LIBCPP_ABI_REGEX_CONSTANTS_NONZERO
 // Re-worked external template instantiations for std::string with a focus on
diff --git a/include/typeinfo b/include/typeinfo
index 799c6eb..41a76ad 100644
--- a/include/typeinfo
+++ b/include/typeinfo
@@ -101,7 +101,13 @@ public:
 
   size_t hash_code() const _NOEXCEPT;
 
-  _LIBCPP_HIDE_FROM_ABI _LIBCPP_CONSTEXPR_SINCE_CXX23 bool operator==(const type_info& __arg) const _NOEXCEPT {
+#    ifdef __EMSCRIPTEN__
+  // XXX Emscripten: adding `always_inline` fixes
+  // https://github.com/emscripten-core/emscripten/issues/13330
+  __attribute__((always_inline))
+#    endif
+  _LIBCPP_HIDE_FROM_ABI _LIBCPP_CONSTEXPR_SINCE_CXX23 bool
+  operator==(const type_info& __arg) const _NOEXCEPT {
     // When evaluated in a constant expression, both type infos simply can't come
     // from different translation units, so it is sufficient to compare their addresses.
     if (__libcpp_is_constant_evaluated()) {
diff --git a/src/filesystem/operations.cpp b/src/filesystem/operations.cpp
index 23c1c28..9037313 100644
--- a/src/filesystem/operations.cpp
+++ b/src/filesystem/operations.cpp
@@ -51,7 +51,7 @@
 #elif defined(__FreeBSD__)
 #  define _LIBCPP_FILESYSTEM_USE_COPY_FILE_RANGE
 #endif
-#if __has_include(<sys/sendfile.h>)
+#if __has_include(<sys/sendfile.h>) && !defined(__EMSCRIPTEN__)
 #  include <sys/sendfile.h>
 #  define _LIBCPP_FILESYSTEM_USE_SENDFILE
 #elif defined(__APPLE__) || __has_include(<copyfile.h>)
diff --git a/src/new.cpp b/src/new.cpp
index e010fe4..eb1471a 100644
--- a/src/new.cpp
+++ b/src/new.cpp
@@ -18,7 +18,13 @@
 // The code below is copied as-is into libc++abi's libcxxabi/src/stdlib_new_delete.cpp
 // file. The version in this file is the canonical one.
 
-inline void __throw_bad_alloc_shim() { std::__throw_bad_alloc(); }
+inline void __throw_bad_alloc_shim() {
+# ifdef __EMSCRIPTEN__
+  abort();
+#else
+  std::__throw_bad_alloc();
+#endif
+}
 
 #  define _LIBCPP_ASSERT_SHIM(expr, str) _LIBCPP_ASSERT(expr, str)
 
diff --git a/src/support/runtime/exception_fallback.ipp b/src/support/runtime/exception_fallback.ipp
index ba283ae..f03c0fb 100644
--- a/src/support/runtime/exception_fallback.ipp
+++ b/src/support/runtime/exception_fallback.ipp
@@ -33,6 +33,7 @@ terminate_handler set_terminate(terminate_handler func) noexcept {
 
 terminate_handler get_terminate() noexcept { return __libcpp_atomic_load(&__terminate_handler); }
 
+#ifndef __EMSCRIPTEN__ // We provide this in JS
 [[noreturn]] void terminate() noexcept {
 #if _LIBCPP_HAS_EXCEPTIONS
   try {
@@ -47,13 +48,16 @@ terminate_handler get_terminate() noexcept { return __libcpp_atomic_load(&__term
   }
 #endif // _LIBCPP_HAS_EXCEPTIONS
 }
+#endif // !__EMSCRIPTEN__
 
+#if !defined(__EMSCRIPTEN__)
 bool uncaught_exception() noexcept { return uncaught_exceptions() > 0; }
 
 int uncaught_exceptions() noexcept {
 #warning uncaught_exception not yet implemented
   __libcpp_verbose_abort("uncaught_exceptions not yet implemented\n");
 }
+#endif // !__EMSCRIPTEN__
 
 exception::~exception() noexcept {}
 
