diff --git a/include/locale b/include/locale
index fa15302..ed99f98 100644
--- a/include/locale
+++ b/include/locale
@@ -2123,7 +2123,11 @@ public:
       : time_put<_CharT, _OutputIterator>(__nm, __refs) {}
 
 protected:
+#if defined(_MSC_VER) && defined(__clang__)
+  ~time_put_byname() {}
+#else
   _LIBCPP_HIDE_FROM_ABI_VIRTUAL ~time_put_byname() override {}
+#endif
 };
 
 extern template class _LIBCPP_EXTERN_TEMPLATE_TYPE_VIS time_put_byname<char>;
