diff --git a/include/cstdlib b/include/cstdlib
index adeee29..4194772 100644
--- a/include/cstdlib
+++ b/include/cstdlib
@@ -148,7 +148,7 @@ using ::wcstombs _LIBCPP_USING_IF_EXISTS;
 using ::at_quick_exit _LIBCPP_USING_IF_EXISTS;
 using ::quick_exit _LIBCPP_USING_IF_EXISTS;
 #  endif
-#  if _LIBCPP_STD_VER >= 17
+#  if _LIBCPP_STD_VER >= 17 && !defined(_LIBCPP_MSVCRT)
 using ::aligned_alloc _LIBCPP_USING_IF_EXISTS;
 #  endif
 
