diff --git a/include/__locale b/include/__locale
index fa987cb..de7c2c6 100644
--- a/include/__locale
+++ b/include/__locale
@@ -17,6 +17,7 @@
 #include <__mutex/once_flag.h>
 #include <__type_traits/make_unsigned.h>
 #include <__utility/no_destroy.h>
+#include <atomic>
 #include <cctype>
 #include <clocale>
 #include <cstdint>
@@ -126,8 +127,7 @@ private:
 };
 
 class _LIBCPP_EXPORTED_FROM_ABI locale::id {
-  once_flag __flag_;
-  int32_t __id_;
+  std::atomic<int32_t> __id_;
 
   static int32_t __next_id;
 
