diff --git a/include/__algorithm/simd_utils.h b/include/__algorithm/simd_utils.h
index 4e03723..658763c 100644
--- a/include/__algorithm/simd_utils.h
+++ b/include/__algorithm/simd_utils.h
@@ -28,7 +28,7 @@ _LIBCPP_PUSH_MACROS
 #include <__undef_macros>
 
 // TODO: Find out how altivec changes things and allow vectorizations there too.
-#if _LIBCPP_STD_VER >= 14 && defined(_LIBCPP_CLANG_VER) && !defined(__ALTIVEC__)
+#if _LIBCPP_STD_VER >= 14 && defined(_LIBCPP_CLANG_VER) && !defined(__ALTIVEC__) && !defined(__CUDACC__)
 #  define _LIBCPP_HAS_ALGORITHM_VECTOR_UTILS 1
 #else
 #  define _LIBCPP_HAS_ALGORITHM_VECTOR_UTILS 0
diff --git a/include/__iterator/reverse_iterator.h b/include/__iterator/reverse_iterator.h
index 5bd1f86..f0434f8 100644
--- a/include/__iterator/reverse_iterator.h
+++ b/include/__iterator/reverse_iterator.h
@@ -132,7 +132,7 @@ public:
     return *--__tmp;
   }
 
-#if _LIBCPP_STD_VER >= 20
+#if _LIBCPP_STD_VER >= 20 && !defined(__CUDACC__)
   _LIBCPP_HIDE_FROM_ABI constexpr pointer operator->() const
     requires is_pointer_v<_Iter> || requires(const _Iter __i) { __i.operator->(); }
   {
diff --git a/include/__iterator/segmented_iterator.h b/include/__iterator/segmented_iterator.h
index 7a8e1ad..19f6537 100644
--- a/include/__iterator/segmented_iterator.h
+++ b/include/__iterator/segmented_iterator.h
@@ -68,8 +68,10 @@ struct __segmented_iterator_traits;
 template <class _Tp, size_t = 0>
 struct __has_specialization : false_type {};
 
+#ifndef __CUDACC__
 template <class _Tp>
 struct __has_specialization<_Tp, sizeof(_Tp) * 0> : true_type {};
+#endif
 
 template <class _Iterator>
 using __is_segmented_iterator _LIBCPP_NODEBUG = __has_specialization<__segmented_iterator_traits<_Iterator> >;
diff --git a/include/__math/traits.h b/include/__math/traits.h
index 0c96f76..dcc54f2 100644
--- a/include/__math/traits.h
+++ b/include/__math/traits.h
@@ -92,7 +92,7 @@ template <class _A1, __enable_if_t<is_integral<_A1>::value, int> = 0>
 }
 
 [[__nodiscard__]] inline _LIBCPP_CONSTEXPR_SINCE_CXX23 _LIBCPP_HIDE_FROM_ABI
-#ifdef _LIBCPP_PREFERRED_OVERLOAD
+#ifdef _LIBCPP_PREFERRED_OVERLOAD && !defined(__CUDACC__)
 _LIBCPP_PREFERRED_OVERLOAD
 #endif
     bool
@@ -111,12 +111,14 @@ template <class _A1, __enable_if_t<is_integral<_A1>::value, int> = 0>
   return false;
 }
 
+#if defined(_LIBCPP_PREFERRED_OVERLOAD) && !defined(__CUDACC__)
 [[__nodiscard__]] inline _LIBCPP_CONSTEXPR_SINCE_CXX23 _LIBCPP_HIDE_FROM_ABI bool isnan(float __x) _NOEXCEPT {
   return __builtin_isnan(__x);
 }
+#endif
 
 [[__nodiscard__]] inline _LIBCPP_CONSTEXPR_SINCE_CXX23 _LIBCPP_HIDE_FROM_ABI
-#ifdef _LIBCPP_PREFERRED_OVERLOAD
+#ifdef _LIBCPP_PREFERRED_OVERLOAD && !defined(__CUDACC__)
 _LIBCPP_PREFERRED_OVERLOAD
 #endif
     bool
@@ -124,9 +126,11 @@ _LIBCPP_PREFERRED_OVERLOAD
   return __builtin_isnan(__x);
 }
 
+#if defined(_LIBCPP_PREFERRED_OVERLOAD) && !defined(__CUDACC__)
 [[__nodiscard__]] inline _LIBCPP_CONSTEXPR_SINCE_CXX23 _LIBCPP_HIDE_FROM_ABI bool isnan(long double __x) _NOEXCEPT {
   return __builtin_isnan(__x);
 }
+#endif
 
 // isnormal
 
diff --git a/include/complex b/include/complex
index df18159..cef958a 100644
--- a/include/complex
+++ b/include/complex
@@ -1457,6 +1457,7 @@ _LIBCPP_HIDE_FROM_ABI inline constexpr complex<double> operator""i(unsigned long
   return {0.0, static_cast<double>(__im)};
 }
 
+#  if !defined(__CUDACC__)
 _LIBCPP_HIDE_FROM_ABI inline constexpr complex<float> operator""if(long double __im) {
   return {0.0f, static_cast<float>(__im)};
 }
@@ -1464,6 +1465,7 @@ _LIBCPP_HIDE_FROM_ABI inline constexpr complex<float> operator""if(long double _
 _LIBCPP_HIDE_FROM_ABI inline constexpr complex<float> operator""if(unsigned long long __im) {
   return {0.0f, static_cast<float>(__im)};
 }
+#  endif
 } // namespace complex_literals
 } // namespace literals
 #  endif
diff --git a/include/tuple b/include/tuple
index 5d968bf..dbd27da 100644
--- a/include/tuple
+++ b/include/tuple
@@ -309,7 +309,7 @@ class __tuple_leaf {
 
   template <class _Tp>
   static _LIBCPP_HIDE_FROM_ABI constexpr bool __can_bind_reference() {
-#    if __has_keyword(__reference_binds_to_temporary)
+#    if __has_keyword(__reference_binds_to_temporary) && !defined(__CUDACC__)
     return !__reference_binds_to_temporary(_Hp, _Tp);
 #    else
     return true;
@@ -617,8 +617,16 @@ public:
               is_constructible<_Tp, _Up>... > {};
 
   template <class... _Up,
-            __enable_if_t< _And< _BoolConstant<sizeof...(_Up) == sizeof...(_Tp)>, _EnableUTypesCtor<_Up...> >::value,
-                           int> = 0>
+      __enable_if_t< _And< _BoolConstant<sizeof...(_Up) == sizeof...(_Tp)>,
+                           _EnableUTypesCtor<_Up...>
+// nvcc 12.2 cannot choose between tuple(const T& ... t) and tuple(U&&... u)
+// so we have to added an explicit requires in enable_if
+#  ifdef __CUDACC__
+                           ,
+                           _Not<_And<is_copy_constructible<_Tp>..., _Lazy<_And, is_convertible<_Up, const _Tp&>...>> >
+#  endif
+                           >::value,
+                     int> = 0>
   _LIBCPP_HIDE_FROM_ABI _LIBCPP_CONSTEXPR_SINCE_CXX14 explicit(_Not<_Lazy<_And, is_convertible<_Up, _Tp>...> >::value)
       tuple(_Up&&... __u) noexcept(_And<is_nothrow_constructible<_Tp, _Up>...>::value)
       : __base_(typename __make_tuple_indices<sizeof...(_Up)>::type(),
