diff --git a/include/optional b/include/optional
index 294c3dd..74e531f 100644
--- a/include/optional
+++ b/include/optional
@@ -295,7 +295,7 @@ struct __optional_destruct_base<_Tp, false> {
   };
   bool __engaged_;
 
-  _LIBCPP_HIDE_FROM_ABI _LIBCPP_CONSTEXPR_SINCE_CXX20 ~__optional_destruct_base() {
+  _LIBCPP_HIDE_FROM_ABI ~__optional_destruct_base() {
     if (__engaged_)
       __val_.~value_type();
   }
