diff --git a/src/filesystem/path_parser.h b/src/filesystem/path_parser.h
index 28a8f24..a546604 100644
--- a/src/filesystem/path_parser.h
+++ b/src/filesystem/path_parser.h
@@ -323,15 +323,17 @@ private:
       return consumeNSeparators(consumeName(P, End), End, 2);
   }
 
-  PosPtr consumeRootName(PosPtr P, PosPtr End) const noexcept {
 #if defined(_LIBCPP_WIN32API)
+  PosPtr consumeRootName(PosPtr P, PosPtr End) const noexcept {
     if (PosPtr Ret = consumeDriveLetter(P, End))
       return Ret;
     if (PosPtr Ret = consumeNetworkRoot(P, End))
       return Ret;
-#endif
     return nullptr;
   }
+#else
+  PosPtr consumeRootName(PosPtr /*P*/, PosPtr /*End*/) const noexcept { return nullptr; }
+#endif
 };
 
 inline string_view_pair separate_filename(string_view_t const& s) {
