diff --git a/src/locale.cpp b/src/locale.cpp
index b5908bf..013eb7d 100644
--- a/src/locale.cpp
+++ b/src/locale.cpp
@@ -900,6 +900,11 @@ extern "C" const int** __ctype_tolower_loc();
 extern "C" const int** __ctype_toupper_loc();
 #endif
 
+#if defined(__ANDROID__)
+// See src/support/android/android_locale.cpp
+extern "C" const unsigned short* const _ctype_android;
+#endif
+
 #ifdef _LIBCPP_PROVIDES_DEFAULT_RUNE_TABLE
 const ctype<char>::mask* ctype<char>::classic_table() noexcept {
   // clang-format off
@@ -995,6 +1000,8 @@ const ctype<char>::mask* ctype<char>::classic_table() noexcept {
 #  elif defined(_NEWLIB_VERSION)
   // Newlib has a 257-entry table in ctype_.c, where (char)0 starts at [1].
   return _ctype_ + 1;
+#  elif defined(__ANDROID__)
+  return _ctype_android;
 #  elif defined(_AIX)
   return (const unsigned int*)__lc_ctype_ptr->obj->mask;
 #  elif defined(__MVS__)
@@ -3649,13 +3656,23 @@ __codecvt_utf16<char32_t, true>::result __codecvt_utf16<char32_t, true>::do_out(
     extern_type* to,
     extern_type* to_end,
     extern_type*& to_nxt) const {
+#if defined(_LIBCPP_SHORT_WCHAR)
+  const uint16_t* _frm     = reinterpret_cast<const uint16_t*>(frm);
+  const uint16_t* _frm_end = reinterpret_cast<const uint16_t*>(frm_end);
+  const uint16_t* _frm_nxt = _frm;
+#else
   const uint32_t* _frm     = reinterpret_cast<const uint32_t*>(frm);
   const uint32_t* _frm_end = reinterpret_cast<const uint32_t*>(frm_end);
   const uint32_t* _frm_nxt = _frm;
+#endif
   uint8_t* _to             = reinterpret_cast<uint8_t*>(to);
   uint8_t* _to_end         = reinterpret_cast<uint8_t*>(to_end);
   uint8_t* _to_nxt         = _to;
+#if defined(_LIBCPP_SHORT_WCHAR)
+  result r                 = ucs2_to_utf16le(_frm, _frm_end, _frm_nxt, _to, _to_end, _to_nxt, __maxcode_, __mode_);
+#else
   result r                 = ucs4_to_utf16le(_frm, _frm_end, _frm_nxt, _to, _to_end, _to_nxt, __maxcode_, __mode_);
+#endif
   frm_nxt                  = frm + (_frm_nxt - _frm);
   to_nxt                   = to + (_to_nxt - _to);
   return r;
@@ -3717,7 +3734,7 @@ __codecvt_utf8_utf16<wchar_t>::result __codecvt_utf8_utf16<wchar_t>::do_out(
     extern_type* to,
     extern_type* to_end,
     extern_type*& to_nxt) const {
-#  if defined(_LIBCPP_SHORT_WCHAR)
+#  ifdef _WIN32
   const uint16_t* _frm     = reinterpret_cast<const uint16_t*>(frm);
   const uint16_t* _frm_end = reinterpret_cast<const uint16_t*>(frm_end);
   const uint16_t* _frm_nxt = _frm;
@@ -3746,7 +3763,7 @@ __codecvt_utf8_utf16<wchar_t>::result __codecvt_utf8_utf16<wchar_t>::do_in(
   const uint8_t* _frm     = reinterpret_cast<const uint8_t*>(frm);
   const uint8_t* _frm_end = reinterpret_cast<const uint8_t*>(frm_end);
   const uint8_t* _frm_nxt = _frm;
-#  if defined(_LIBCPP_SHORT_WCHAR)
+#  ifdef _WIN32
   uint16_t* _to     = reinterpret_cast<uint16_t*>(to);
   uint16_t* _to_end = reinterpret_cast<uint16_t*>(to_end);
   uint16_t* _to_nxt = _to;
@@ -3885,9 +3902,15 @@ __codecvt_utf8_utf16<char32_t>::result __codecvt_utf8_utf16<char32_t>::do_in(
   const uint8_t* _frm     = reinterpret_cast<const uint8_t*>(frm);
   const uint8_t* _frm_end = reinterpret_cast<const uint8_t*>(frm_end);
   const uint8_t* _frm_nxt = _frm;
+#if defined(_LIBCPP_SHORT_WCHAR)
+  uint16_t* _to           = reinterpret_cast<uint16_t*>(to);
+  uint16_t* _to_end       = reinterpret_cast<uint16_t*>(to_end);
+  uint16_t* _to_nxt       = _to;
+#else
   uint32_t* _to           = reinterpret_cast<uint32_t*>(to);
   uint32_t* _to_end       = reinterpret_cast<uint32_t*>(to_end);
   uint32_t* _to_nxt       = _to;
+#endif
   result r                = utf8_to_utf16(_frm, _frm_end, _frm_nxt, _to, _to_end, _to_nxt, __maxcode_, __mode_);
   frm_nxt                 = frm + (_frm_nxt - _frm);
   to_nxt                  = to + (_to_nxt - _to);
@@ -5128,7 +5151,7 @@ static void __init_pat(
           // We insert the space into the symbol instead of
           // setting pat.field[2]=space so that when
           // showbase is not set, the space goes away too.
-          __curr_symbol_.insert(0, 1, space_char);
+          __curr_symbol_.insert((size_t)0, 1, space_char);
         }
         return;
       default:
@@ -5150,7 +5173,7 @@ static void __init_pat(
           // We insert the space into the symbol instead of
           // setting pat.field[2]=space so that when
           // showbase is not set, the space goes away too.
-          __curr_symbol_.insert(0, 1, space_char);
+          __curr_symbol_.insert((size_t)0, 1, space_char);
         }
         return;
       case 2: // Space between sign and currency or value.
@@ -5179,7 +5202,7 @@ static void __init_pat(
           // We insert the space into the symbol instead of
           // setting pat.field[1]=space so that when
           // showbase is not set, the space goes away too.
-          __curr_symbol_.insert(0, 1, space_char);
+          __curr_symbol_.insert((size_t)0, 1, space_char);
         }
         pat.field[1] = none;
         pat.field[2] = symbol;
@@ -5221,7 +5244,7 @@ static void __init_pat(
           // We insert the space into the symbol instead of
           // setting pat.field[2]=space so that when
           // showbase is not set, the space goes away too.
-          __curr_symbol_.insert(0, 1, space_char);
+          __curr_symbol_.insert((size_t)0, 1, space_char);
         }
         return;
       default:
@@ -5243,7 +5266,7 @@ static void __init_pat(
           // We insert the space into the symbol instead of
           // setting pat.field[1]=space so that when
           // showbase is not set, the space goes away too.
-          __curr_symbol_.insert(0, 1, space_char);
+          __curr_symbol_.insert((size_t)0, 1, space_char);
         }
         return;
       case 2: // Space between sign and currency or value.
@@ -5282,7 +5305,7 @@ static void __init_pat(
           // We insert the space into the symbol instead of
           // setting pat.field[2]=space so that when
           // showbase is not set, the space goes away too.
-          __curr_symbol_.insert(0, 1, space_char);
+          __curr_symbol_.insert((size_t)0, 1, space_char);
         }
         return;
       default:
