diff --git a/include/chrono b/include/chrono
index 5bab3f8..956badd 100644
--- a/include/chrono
+++ b/include/chrono
@@ -853,7 +853,7 @@ constexpr chrono::year                                  operator ""y(unsigned lo
 // [time.syn]
 #include <compare>
 
-#if !defined(_LIBCPP_HAS_NO_LOCALIZATION) && _LIBCPP_STD_VER >= 20
+#if 0
 #  include <__chrono/formatter.h>
 #  include <__chrono/ostream.h>
 #  include <__chrono/parser_std_format_spec.h>
