diff --git a/include/__config b/include/__config
index e9c86c8..de02565 100644
--- a/include/__config
+++ b/include/__config
@@ -1039,10 +1039,12 @@ typedef __char32_t char32_t;
 #  ifndef _LIBCPP_HAS_NO_ASAN
 extern "C" _LIBCPP_EXPORTED_FROM_ABI void
 __sanitizer_annotate_contiguous_container(const void*, const void*, const void*, const void*);
+#    if !defined(_LIBCPP_NO_ASAN_CONTIGUOUS_CONTAINER_FEATURES)
 extern "C" _LIBCPP_EXPORTED_FROM_ABI void __sanitizer_annotate_double_ended_contiguous_container(
     const void*, const void*, const void*, const void*, const void*, const void*);
 extern "C" _LIBCPP_EXPORTED_FROM_ABI int
 __sanitizer_verify_double_ended_contiguous_container(const void*, const void*, const void*, const void*);
+#    endif
 #  endif
 
 // Try to find out if RTTI is disabled.
diff --git a/include/deque b/include/deque
index 6389627..314d5bd 100644
--- a/include/deque
+++ b/include/deque
@@ -888,7 +888,7 @@ private:
     (void)__old_con_end;
     (void)__new_con_beg;
     (void)__new_con_end;
-#ifndef _LIBCPP_HAS_NO_ASAN
+#if !defined(_LIBCPP_HAS_NO_ASAN) && !defined(_LIBCPP_NO_ASAN_CONTIGUOUS_CONTAINER_FEATURES)
     if (__beg != nullptr && __asan_annotate_container_with_allocator<_Allocator>::value)
       __sanitizer_annotate_double_ended_contiguous_container(
           __beg, __end, __old_con_beg, __old_con_end, __new_con_beg, __new_con_end);
@@ -1052,7 +1052,7 @@ private:
           __block_start, __block_end, __block_start, __block_start, __block_start, __block_end);
     }
   }
-#if !defined(_LIBCPP_HAS_NO_ASAN)
+#if !defined(_LIBCPP_HAS_NO_ASAN) && !defined(_LIBCPP_NO_ASAN_CONTIGUOUS_CONTAINER_FEATURES)
 
 public:
   _LIBCPP_HIDE_FROM_ABI bool __verify_asan_annotations() const _NOEXCEPT {
diff --git a/include/vector b/include/vector
index fb92f68..9e512b3 100644
--- a/include/vector
+++ b/include/vector
@@ -849,7 +849,7 @@ private:
     (void)__end;
     (void)__old_mid;
     (void)__new_mid;
-#ifndef _LIBCPP_HAS_NO_ASAN
+#if !defined(_LIBCPP_HAS_NO_ASAN) && !defined(_LIBCPP_NO_ASAN_CONTIGUOUS_CONTAINER_FEATURES)
     if (!__libcpp_is_constant_evaluated() && __beg != nullptr &&
         __asan_annotate_container_with_allocator<_Allocator>::value)
       __sanitizer_annotate_contiguous_container(__beg, __end, __old_mid, __new_mid);
