diff --git a/include/__debug_utils/sanitizers.h b/include/__debug_utils/sanitizers.h
index d8547e3..f72b399 100644
--- a/include/__debug_utils/sanitizers.h
+++ b/include/__debug_utils/sanitizers.h
@@ -57,7 +57,7 @@ _LIBCPP_HIDE_FROM_ABI void __annotate_double_ended_contiguous_container(
     const void* __last_old_contained,
     const void* __first_new_contained,
     const void* __last_new_contained) {
-#ifdef _LIBCPP_HAS_NO_ASAN
+#if !defined(_LIBCPP_HAS_NO_ASAN) || defined(_LIBCPP_NO_ASAN_CONTIGUOUS_CONTAINER_FEATURES)
   (void)__first_storage;
   (void)__last_storage;
   (void)__first_old_contained;
@@ -86,7 +86,7 @@ _LIBCPP_HIDE_FROM_ABI _LIBCPP_CONSTEXPR_SINCE_CXX14 void __annotate_contiguous_c
     const void* __last_storage,
     const void* __old_last_contained,
     const void* __new_last_contained) {
-#ifdef _LIBCPP_HAS_NO_ASAN
+#if !defined(_LIBCPP_HAS_NO_ASAN) || defined(_LIBCPP_NO_ASAN_CONTIGUOUS_CONTAINER_FEATURES)
   (void)__first_storage;
   (void)__last_storage;
   (void)__old_last_contained;
