diff --git a/include/__atomic/atomic_ref.h b/include/__atomic/atomic_ref.h
index 177ea64..a0ed1e8 100644
--- a/include/__atomic/atomic_ref.h
+++ b/include/__atomic/atomic_ref.h
@@ -261,7 +261,7 @@ template <class _Tp>
 struct atomic_ref<_Tp> : public __atomic_ref_base<_Tp> {
   using __base _LIBCPP_NODEBUG = __atomic_ref_base<_Tp>;
 
-  using difference_type = __base::value_type;
+  using difference_type = typename __base::value_type;
 
   _LIBCPP_HIDE_FROM_ABI explicit atomic_ref(_Tp& __obj) : __base(__obj) {
     _LIBCPP_ASSERT_ARGUMENT_WITHIN_DOMAIN(
@@ -307,7 +307,7 @@ template <class _Tp>
 struct atomic_ref<_Tp> : public __atomic_ref_base<_Tp> {
   using __base _LIBCPP_NODEBUG = __atomic_ref_base<_Tp>;
 
-  using difference_type = __base::value_type;
+  using difference_type = typename __base::value_type;
 
   _LIBCPP_HIDE_FROM_ABI explicit atomic_ref(_Tp& __obj) : __base(__obj) {
     _LIBCPP_ASSERT_ARGUMENT_WITHIN_DOMAIN(
