diff --git a/include/__charconv/from_chars_floating_point.h b/include/__charconv/from_chars_floating_point.h
index 811e518..55c4101 100644
--- a/include/__charconv/from_chars_floating_point.h
+++ b/include/__charconv/from_chars_floating_point.h
@@ -26,7 +26,7 @@ _LIBCPP_PUSH_MACROS
 
 _LIBCPP_BEGIN_NAMESPACE_STD
 
-#if _LIBCPP_STD_VER >= 17
+#if _LIBCPP_STD_VER >= 17 && !defined(_YNDX_LIBCPP_DISABLE_CHARS_FLOATING_POINT)
 
 template <class _Fp>
 struct __from_chars_result {
diff --git a/src/charconv.cpp b/src/charconv.cpp
index 5e8cb7d..433e513 100644
--- a/src/charconv.cpp
+++ b/src/charconv.cpp
@@ -9,7 +9,7 @@
 #include <charconv>
 #include <string.h>
 
-#include "include/from_chars_floating_point.h"
+// #include "include/from_chars_floating_point.h"
 #include "include/to_chars_floating_point.h"
 
 _LIBCPP_BEGIN_NAMESPACE_STD
@@ -75,6 +75,7 @@ to_chars_result to_chars(char* __first, char* __last, long double __value, chars
       __first, __last, static_cast<double>(__value), __fmt, __precision);
 }
 
+#ifndef _YNDX_LIBCPP_DISABLE_CHARS_FLOATING_POINT
 template <class _Fp>
 __from_chars_result<_Fp> __from_chars_floating_point(
     _LIBCPP_NOESCAPE const char* __first, _LIBCPP_NOESCAPE const char* __last, chars_format __fmt) {
@@ -86,4 +87,6 @@ template __from_chars_result<float> __from_chars_floating_point(
 
 template __from_chars_result<double> __from_chars_floating_point(
     _LIBCPP_NOESCAPE const char* __first, _LIBCPP_NOESCAPE const char* __last, chars_format __fmt);
+#endif
+
 _LIBCPP_END_NAMESPACE_STD
