diff --git a/include/__config b/include/__config
index 85d1aab..c0d1010 100644
--- a/include/__config
+++ b/include/__config
@@ -1121,7 +1121,11 @@ typedef __char32_t char32_t;
 // Optional attributes - these are useful for a better QoI, but not required to be available
 
 #  define _LIBCPP_NOALIAS __attribute__((__malloc__))
-#  define _LIBCPP_NODEBUG [[__gnu__::__nodebug__]]
+#  if __has_attribute(__nodebug__) && _LIBCPP_CLANG_VER > 1400
+#    define _LIBCPP_NODEBUG [[__gnu__::__nodebug__]]
+#  else
+#    define _LIBCPP_NODEBUG
+#  endif
 #  define _LIBCPP_NO_SANITIZE(...) __attribute__((__no_sanitize__(__VA_ARGS__)))
 #  define _LIBCPP_INIT_PRIORITY_MAX __attribute__((__init_priority__(100)))
 #  define _LIBCPP_ATTRIBUTE_FORMAT(archetype, format_string_index, first_format_arg_index)                             \
diff --git a/include/__format/buffer.h b/include/__format/buffer.h
index c88b7f3..7cb338b 100644
--- a/include/__format/buffer.h
+++ b/include/__format/buffer.h
@@ -496,7 +496,7 @@ private:
 // Selects the type of the buffer used for the output iterator.
 template <class _OutIt, __fmt_char_type _CharT>
 class _LIBCPP_TEMPLATE_VIS __buffer_selector {
-  using _Container _LIBCPP_NODEBUG = __back_insert_iterator_container<_OutIt>::type;
+  using _Container _LIBCPP_NODEBUG = typename __back_insert_iterator_container<_OutIt>::type;
 
 public:
   using type _LIBCPP_NODEBUG =
@@ -511,7 +511,7 @@ public:
 template <class _OutIt, __fmt_char_type _CharT>
 class _LIBCPP_TEMPLATE_VIS __format_to_n_buffer : private __buffer_selector<_OutIt, _CharT>::type {
 public:
-  using _Base _LIBCPP_NODEBUG = __buffer_selector<_OutIt, _CharT>::type;
+  using _Base _LIBCPP_NODEBUG = typename __buffer_selector<_OutIt, _CharT>::type;
 
   [[nodiscard]] _LIBCPP_HIDE_FROM_ABI __format_to_n_buffer(_OutIt __out_it, iter_difference_t<_OutIt> __n)
       : _Base{std::move(__out_it), std::addressof(__max_output_size_)},
