diff --git a/include/__format/format_arg_store.h b/include/__format/format_arg_store.h
index 87557aa..ae61105 100644
--- a/include/__format/format_arg_store.h
+++ b/include/__format/format_arg_store.h
@@ -175,7 +175,7 @@ _LIBCPP_HIDE_FROM_ABI basic_format_arg<_Context> __create_format_arg(_Tp& __valu
   static_assert(__arg != __arg_t::__none, "the supplied type is not formattable");
   static_assert(__formattable_with<_Tp, _Context>);
 
-  using __context_char_type = _Context::char_type;
+  using __context_char_type = typename _Context::char_type;
   // Not all types can be used to directly initialize the
   // __basic_format_arg_value.  First handle all types needing adjustment, the
   // final else requires no adjustment.
