diff --git a/include/__random/binomial_distribution.h b/include/__random/binomial_distribution.h
index e8774bb..54f321a 100644
--- a/include/__random/binomial_distribution.h
+++ b/include/__random/binomial_distribution.h
@@ -97,10 +97,6 @@ public:
   }
 };
 
-#ifndef _LIBCPP_MSVCRT_LIKE
-extern "C" double lgamma_r(double, int*);
-#endif
-
 inline _LIBCPP_HIDE_FROM_ABI double __libcpp_lgamma(double __d) {
 #if defined(_LIBCPP_MSVCRT_LIKE)
   return lgamma(__d);
