diff --git a/include/__ranges/lazy_split_view.h b/include/__ranges/lazy_split_view.h
index db031fe..9dc3ca4 100644
--- a/include/__ranges/lazy_split_view.h
+++ b/include/__ranges/lazy_split_view.h
@@ -75,6 +75,7 @@ class lazy_split_view : public view_interface<lazy_split_view<_View, _Pattern>>
   using _MaybeCurrent = _If<!forward_range<_View>, __non_propagating_cache<iterator_t<_View>>, __empty_cache>;
   _LIBCPP_NO_UNIQUE_ADDRESS _MaybeCurrent __current_ = _MaybeCurrent();
 
+private:
   template <bool>
   struct __outer_iterator;
   template <bool>
