diff --git a/include/__expected/bad_expected_access.h b/include/__expected/bad_expected_access.h
index 1b73438..d954545 100644
--- a/include/__expected/bad_expected_access.h
+++ b/include/__expected/bad_expected_access.h
@@ -20,7 +20,7 @@
 _LIBCPP_PUSH_MACROS
 #include <__undef_macros>
 
-#if _LIBCPP_STD_VER >= 23
+#if _LIBCPP_STD_VER >= 20
 
 _LIBCPP_BEGIN_NAMESPACE_STD
 
diff --git a/include/__expected/expected.h b/include/__expected/expected.h
index 03bbd16..dfded2c 100644
--- a/include/__expected/expected.h
+++ b/include/__expected/expected.h
@@ -55,7 +55,7 @@
 _LIBCPP_PUSH_MACROS
 #include <__undef_macros>
 
-#if _LIBCPP_STD_VER >= 23
+#if _LIBCPP_STD_VER >= 20
 
 _LIBCPP_BEGIN_NAMESPACE_STD
 
diff --git a/include/__expected/unexpect.h b/include/__expected/unexpect.h
index df52787..fc3be4c 100644
--- a/include/__expected/unexpect.h
+++ b/include/__expected/unexpect.h
@@ -15,7 +15,7 @@
 #  pragma GCC system_header
 #endif
 
-#if _LIBCPP_STD_VER >= 23
+#if _LIBCPP_STD_VER >= 20
 
 _LIBCPP_BEGIN_NAMESPACE_STD
 
diff --git a/include/__expected/unexpected.h b/include/__expected/unexpected.h
index 6904889..85edcf9 100644
--- a/include/__expected/unexpected.h
+++ b/include/__expected/unexpected.h
@@ -34,7 +34,7 @@
 _LIBCPP_PUSH_MACROS
 #include <__undef_macros>
 
-#if _LIBCPP_STD_VER >= 23
+#if _LIBCPP_STD_VER >= 20
 
 _LIBCPP_BEGIN_NAMESPACE_STD
 
diff --git a/include/__utility/forward_like.h b/include/__utility/forward_like.h
index 409f716..8c89f01 100644
--- a/include/__utility/forward_like.h
+++ b/include/__utility/forward_like.h
@@ -23,7 +23,7 @@
 
 _LIBCPP_BEGIN_NAMESPACE_STD
 
-#if _LIBCPP_STD_VER >= 23
+#if _LIBCPP_STD_VER >= 20
 
 template <class _Ap, class _Bp>
 using _CopyConst _LIBCPP_NODEBUG = _If<is_const_v<_Ap>, const _Bp, _Bp>;
diff --git a/include/expected b/include/expected
index 38038d3..d06fc3b 100644
--- a/include/expected
+++ b/include/expected
@@ -43,7 +43,7 @@ namespace std {
 #else
 #  include <__config>
 
-#  if _LIBCPP_STD_VER >= 23
+#  if _LIBCPP_STD_VER >= 20
 #    include <__expected/bad_expected_access.h>
 #    include <__expected/expected.h>
 #    include <__expected/unexpect.h>
