--- contrib/python/invoke/py2/invoke/util.py	(index)
+++ contrib/python/invoke/py2/invoke/util.py	(working tree)
@@ -20,18 +20,11 @@ import sys
 # attribute access, since six.moves does import shenanigans.)
 try:
     from .vendor.lexicon import Lexicon  # noqa
-    from .vendor import six
-    from .vendor.six.moves import reduce  # noqa
-
-    if six.PY3:
-        from .vendor import yaml3 as yaml  # noqa
-    else:
-        from .vendor import yaml2 as yaml  # noqa
 except ImportError:
     from lexicon import Lexicon  # noqa
-    import six
-    from six.moves import reduce  # noqa
-    import yaml  # noqa
+import six
+from six.moves import reduce  # noqa
+import yaml  # noqa
 
 
 LOG_FORMAT = "%(name)s.%(module)s.%(funcName)s: %(message)s"
