--- contrib/libs/glog/sources/src/glog/vlog_is_on.h	(5aa1c9aeaa66327ca62e43736886b69b88b7576d)
+++ contrib/libs/glog/sources/src/glog/vlog_is_on.h	(working tree)
@@ -81,13 +81,15 @@
 // matching the current source file that represents results of
 // parsing of --vmodule flag and/or SetVLOGLevel calls.
 #define VLOG_IS_ON(verboselevel)                                \
+  /* NOLINTBEGIN(bugprone-reserved-identifier)*/                               \
   __extension__  \
   ({ static google::int32* vlocal__ = &google::kLogSiteUninitialized;           \
      google::int32 verbose_level__ = (verboselevel);                    \
      (*vlocal__ >= verbose_level__) &&                          \
      ((vlocal__ != &google::kLogSiteUninitialized) ||                   \
       (google::InitVLOG3__(&vlocal__, &FLAGS_v,                         \
-                   __FILE__, verbose_level__))); })
+                   __FILE__, verbose_level__))); })                      \
+/* NOLINTEND(bugprone-reserved-identifier)*/                            
 #else
 // GNU extensions not available, so we do not support --vmodule.
 // Dynamic value of FLAGS_v always controls the logging level.
