--- contrib/python/furl/py3/tests/test_furl.py	(index)
+++ contrib/python/furl/py3/tests/test_furl.py	(working tree)
@@ -1669,7 +1669,8 @@ class TestFurl(unittest.TestCase):
         # Invalid, but well-formed, IPv6 addresses shouldn't raise an
         # exception because urlparse.urlsplit() doesn't raise an
         # exception on invalid IPv6 addresses.
-        furl.furl('http://[0:0:0:0:0:0:0:1:1:1:1:1:1:1:1:9999999999999]/')
+        if sys.version_info < (3, 11, 4):
+            furl.furl('http://[0:0:0:0:0:0:0:1:1:1:1:1:1:1:1:9999999999999]/')
 
         # Malformed IPv6 should raise an exception because urlparse.urlsplit()
         # raises an exception on malformed IPv6 addresses.
@@ -1695,11 +1696,12 @@ class TestFurl(unittest.TestCase):
         assert f.host == '1.2.3.4.5.6'
         assert f.port == 999
 
-        netloc = '[0:0:0:0:0:0:0:1:1:1:1:1:1:1:1:9999999999999]:888'
-        f.netloc = netloc
-        assert f.netloc == netloc
-        assert f.host == '[0:0:0:0:0:0:0:1:1:1:1:1:1:1:1:9999999999999]'
-        assert f.port == 888
+        if sys.version_info < (3, 11, 4):
+            netloc = '[0:0:0:0:0:0:0:1:1:1:1:1:1:1:1:9999999999999]:888'
+            f.netloc = netloc
+            assert f.netloc == netloc
+            assert f.host == '[0:0:0:0:0:0:0:1:1:1:1:1:1:1:1:9999999999999]'
+            assert f.port == 888
 
         # Malformed IPv6 should raise an exception because
         # urlparse.urlsplit() raises an exception
@@ -1715,8 +1717,9 @@ class TestFurl(unittest.TestCase):
             f.netloc = 'pump2pump.org:777777777777'
 
         # No side effects.
-        assert f.host == '[0:0:0:0:0:0:0:1:1:1:1:1:1:1:1:9999999999999]'
-        assert f.port == 888
+        if sys.version_info < (3, 11, 4):
+            assert f.host == '[0:0:0:0:0:0:0:1:1:1:1:1:1:1:1:9999999999999]'
+            assert f.port == 888
 
         # Empty netloc.
         f = furl.furl('//')
