--- contrib/python/furl/py3/tests/test_furl.py	(index)
+++ contrib/python/furl/py3/tests/test_furl.py	(working tree)
@@ -1639,7 +1639,10 @@ class TestFurl(unittest.TestCase):
         # of a netloc, even if the netloc is empty.
         f = furl.furl('////path')
         assert f.netloc == '' and str(f.path) == '//path'
-        assert f.url == '////path'
+        if sys.version_info >= (3, 12, 4):
+            assert f.url == '//////path'
+        else:
+            assert f.url == '////path'
 
         # TODO(grun): Test more odd URLs.
 
