--- contrib/python/pylxd/py2/pylxd/__init__.py	(index)
+++ contrib/python/pylxd/py2/pylxd/__init__.py	(working tree)
@@ -12,8 +12,9 @@
 # License for the specific language governing permissions and limitations
 # under the License.
 
-import pbr.version
+# import pbr.version
 
-__version__ = pbr.version.VersionInfo('pylxd').version_string()
+# __version__ = pbr.version.VersionInfo('pylxd').version_string()
+__version__ = '2.2.11'
 
 from pylxd.client import Client,EventType  # NOQA
