--- contrib/python/pylzma/compat/LzmaCompatDecode.h	(index)
+++ contrib/python/pylzma/compat/LzmaCompatDecode.h	(working tree)
@@ -9,7 +9,7 @@ Converted to a state machine by Amir Szekely
 #ifndef __LZMADECODE_H
 #define __LZMADECODE_H
 
-#include "../sdk/C/7zTypes.h"
+#include "7zTypes.h"
 
 /***********************
  *    Configuration    *
--- contrib/python/pylzma/pylzma/pylzma.c	(index)
+++ contrib/python/pylzma/pylzma/pylzma.c	(working tree)
@@ -25,10 +25,10 @@
 
 #include <Python.h>
 
-#include "../sdk/C/7zVersion.h"
-#include "../sdk/C/Sha256.h"
-#include "../sdk/C/Aes.h"
-#include "../sdk/C/Bra.h"
+#include "7zVersion.h"
+#include "Sha256.h"
+#include "Aes.h"
+#include "Bra.h"
 
 #include "pylzma.h"
 #include "pylzma_compress.h"
--- contrib/python/pylzma/pylzma/pylzma.h	(index)
+++ contrib/python/pylzma/pylzma/pylzma.h	(working tree)
@@ -28,7 +28,7 @@
 
 #include <Python.h>
 
-#include "../sdk/C/7zTypes.h"
+#include "7zTypes.h"
 
 #ifndef min
 #define min(a, b) ((a) < (b) ? (a) : (b))
--- contrib/python/pylzma/pylzma/pylzma_aes.c	(index)
+++ contrib/python/pylzma/pylzma/pylzma_aes.c	(working tree)
@@ -25,8 +25,8 @@
 
 #include <Python.h>
 
-#include "../sdk/C/Aes.h"
-#include "../sdk/C/7zTypes.h"
+#include "Aes.h"
+#include "7zTypes.h"
 
 #include "pylzma.h"
 
--- contrib/python/pylzma/pylzma/pylzma_compress.c	(index)
+++ contrib/python/pylzma/pylzma/pylzma_compress.c	(working tree)
@@ -25,7 +25,7 @@
 
 #include <Python.h>
 
-#include "../sdk/C/LzmaEnc.h"
+#include "LzmaEnc.h"
 
 #include "pylzma.h"
 #include "pylzma_streams.h"
--- contrib/python/pylzma/pylzma/pylzma_compressfile.c	(index)
+++ contrib/python/pylzma/pylzma/pylzma_compressfile.c	(working tree)
@@ -28,8 +28,8 @@
 #include <cStringIO.h>
 #endif
 
-#include "../sdk/C/LzmaEnc.h"
-#include "../sdk/C/7zTypes.h"
+#include "LzmaEnc.h"
+#include "7zTypes.h"
 
 #include "pylzma.h"
 #include "pylzma_streams.h"
--- contrib/python/pylzma/pylzma/pylzma_decompress.c	(index)
+++ contrib/python/pylzma/pylzma/pylzma_decompress.c	(working tree)
@@ -25,8 +25,8 @@
 
 #include <Python.h>
 
-#include "../sdk/C/LzmaDec.h"
-#include "../sdk/C/Lzma2Dec.h"
+#include "LzmaDec.h"
+#include "Lzma2Dec.h"
 
 #include "pylzma.h"
 #include "pylzma_streams.h"
--- contrib/python/pylzma/pylzma/pylzma_decompressobj.h	(index)
+++ contrib/python/pylzma/pylzma/pylzma_decompressobj.h	(working tree)
@@ -28,8 +28,8 @@
 
 #include <Python.h>
 
-#include "../sdk/C/LzmaDec.h"
-#include "../sdk/C/Lzma2Dec.h"
+#include "LzmaDec.h"
+#include "Lzma2Dec.h"
 
 typedef struct {
     PyObject_HEAD
--- contrib/python/pylzma/pylzma/pylzma_streams.c	(index)
+++ contrib/python/pylzma/pylzma/pylzma_streams.c	(working tree)
@@ -89,7 +89,7 @@ CreateMemoryInOutStream(CMemoryInOutStream *stream)
     stream->avail = INITIAL_BLOCKSIZE;
 }
 
-Bool
+BoolInt
 MemoryInOutStreamAppend(CMemoryInOutStream *stream, Byte *data, size_t size)
 {
     if (!size) {
--- contrib/python/pylzma/pylzma/pylzma_streams.h	(index)
+++ contrib/python/pylzma/pylzma/pylzma_streams.h	(working tree)
@@ -28,7 +28,7 @@
 
 #include <Python.h>
 
-#include "../sdk/C/7zTypes.h"
+#include "7zTypes.h"
 
 typedef struct
 {
@@ -43,7 +43,7 @@ void CreateMemoryInStream(CMemoryInStream *stream, Byte *data, size_t size);
 typedef CMemoryInStream CMemoryInOutStream;
 
 void CreateMemoryInOutStream(CMemoryInOutStream *stream);
-Bool MemoryInOutStreamAppend(CMemoryInOutStream *stream, Byte *data, size_t size);
+BoolInt MemoryInOutStreamAppend(CMemoryInOutStream *stream, Byte *data, size_t size);
 
 typedef struct
 {
