--- contrib/python/temporalio/temporalio/testing/_workflow.py	(index)
+++ contrib/python/temporalio/temporalio/testing/_workflow.py	(working tree)
@@ -2,6 +2,29 @@
 
 from __future__ import annotations
 
+try:
+    import yatest.common
+    import yatest.common.runtime
+except ImportError:
+    test_server_path = None
+    dev_server_path = None
+else:
+    try:
+        test_server_path = yatest.common.binary_path('infra/temporal/tools/temporal-test-server/temporal-test-server.sh')
+        dev_server_path = yatest.common.binary_path('infra/temporal/tools/temporal/temporal')
+    except yatest.common.runtime.NoRuntimeFormed:
+        test_server_path = None
+        dev_server_path = None
+    except Exception as e:
+        raise RuntimeError('Have you forgot to INCLUDE(${ARCADIA_ROOT}/infra/temporal/lib/py/tests.inc)?') from e
+    else:
+        import os
+
+        if not os.environ.get("JAVA_HOME"):
+            global_resources = yatest.common.runtime.global_resources()
+            if global_resources and 'JDK_DEFAULT_RESOURCE_GLOBAL' in global_resources:
+                os.environ["JAVA_HOME"] = global_resources['JDK_DEFAULT_RESOURCE_GLOBAL']
+
 import asyncio
 import logging
 from contextlib import asynccontextmanager, contextmanager
@@ -92,7 +115,7 @@ class WorkflowEnvironment:
         ui: bool = False,
         runtime: Optional[temporalio.runtime.Runtime] = None,
         search_attributes: Sequence[temporalio.common.SearchAttributeKey] = (),
-        dev_server_existing_path: Optional[str] = None,
+        dev_server_existing_path: Optional[str] = dev_server_path,
         dev_server_database_filename: Optional[str] = None,
         dev_server_log_format: str = "pretty",
         dev_server_log_level: Optional[str] = "warn",
@@ -246,7 +269,7 @@ class WorkflowEnvironment:
         port: Optional[int] = None,
         download_dest_dir: Optional[str] = None,
         runtime: Optional[temporalio.runtime.Runtime] = None,
-        test_server_existing_path: Optional[str] = None,
+        test_server_existing_path: Optional[str] = test_server_path,
         test_server_download_version: str = "default",
         test_server_extra_args: Sequence[str] = [],
         test_server_download_ttl: Optional[timedelta] = None,
