--- contrib/python/temporalio/ya.make	(index)
+++ contrib/python/temporalio/ya.make	(working tree)
@@ -29,8 +29,20 @@ NO_CHECK_IMPORTS(
     temporalio.contrib.pydantic
 )
 
+SET_RESOURCE_URI_FROM_JSON(SANDBOX_RESOURCE_URI resources.json)
+
+FROM_SANDBOX(FILE ${SANDBOX_RESOURCE_URI} OUT libtemporal_sdk_bridge.a)
+
+IF (OS_DARWIN)
+    LDFLAGS(
+        -framework CoreFoundation
+        -framework IOKit
+        -framework Security
+    )
+ENDIF()
+
 SRCS(
-    temporalio/bridge/py_wrapper.cpp
+    GLOBAL temporalio/bridge/py_wrapper.cpp
 )
 
 PY_SRCS(
