--- contrib/python/meld3/meld3/test_meld3.py	(index)
+++ contrib/python/meld3/meld3/test_meld3.py	(working tree)
@@ -304,11 +304,11 @@ _FILLMELDFORM_HTML = """\
 
 class MeldAPITests(unittest.TestCase):
     def _makeElement(self, string):
-        from . import parse_xmlstring
+        from meld3 import parse_xmlstring
         return parse_xmlstring(string)
 
     def _makeElementFromHTML(self, string):
-        from . import parse_htmlstring
+        from meld3 import parse_htmlstring
         return parse_htmlstring(string)
 
     def test_findmeld(self):
@@ -423,7 +423,7 @@ class MeldAPITests(unittest.TestCase):
         self.assertEqual(favoritecolor[1].attrib.get('checked'), None)
 
     def test_replace_removes_all_elements(self):
-        from . import Replace
+        from meld3 import Replace
         root = self._makeElement(_SIMPLE_XML)
         L = root.findmeld('list')
         L.replace('this is a textual replacement')
@@ -432,7 +432,7 @@ class MeldAPITests(unittest.TestCase):
         self.assertEqual(len(root.getchildren()), 1)
 
     def test_replace_replaces_the_right_element(self):
-        from . import Replace
+        from meld3 import Replace
         root = self._makeElement(_SIMPLE_XML)
         D = root.findmeld('description')
         D.replace('this is a textual replacement')
@@ -453,7 +453,7 @@ class MeldAPITests(unittest.TestCase):
         self.assertEqual(D.structure, False)
 
     def test_content(self):
-        from . import Replace
+        from meld3 import Replace
         root = self._makeElement(_SIMPLE_XML)
         D = root.findmeld('description')
         D.content('this is a textual replacement')
@@ -477,7 +477,7 @@ class MeldAPITests(unittest.TestCase):
         self.assertEqual(T.structure, False)
 
     def test_attributes(self):
-        from . import _MELD_ID
+        from meld3 import _MELD_ID
         root = self._makeElement(_COMPLEX_XHTML)
         D = root.findmeld('form1')
         D.attributes(foo='bar', baz='1', g='2', action='#')
@@ -487,8 +487,8 @@ class MeldAPITests(unittest.TestCase):
             _MELD_ID: 'form1'})
 
     def test_attributes_unicode(self):
-        from . import _MELD_ID
-        from ._compat import _u
+        from meld3 import _MELD_ID
+        from meld3._compat import _u
         root = self._makeElement(_COMPLEX_XHTML)
         D = root.findmeld('form1')
         D.attributes(foo=_u('bar'), action=_u('#'))
@@ -503,7 +503,7 @@ class MeldAPITests(unittest.TestCase):
 
 class MeldElementInterfaceTests(unittest.TestCase):
     def _getTargetClass(self):
-        from . import _MeldElementInterface
+        from meld3 import _MeldElementInterface
         return _MeldElementInterface
 
     def _makeOne(self, *arg, **kw):
@@ -512,7 +512,7 @@ class MeldElementInterfaceTests(unittest.TestCase):
 
     def test_repeat(self):
         root = self._makeOne('root', {})
-        from . import _MELD_ID
+        from meld3 import _MELD_ID
         item = self._makeOne('item', {_MELD_ID:'item'})
         record = self._makeOne('record', {_MELD_ID:'record'})
         name = self._makeOne('name', {_MELD_ID:'name'})
@@ -557,7 +557,7 @@ class MeldElementInterfaceTests(unittest.TestCase):
         self.assertEqual(replacenode.parent, el)
         self.assertEqual(replacenode.text, 'hello')
         self.assertEqual(replacenode.structure, False)
-        from . import Replace
+        from meld3 import Replace
         self.assertEqual(replacenode.tag, Replace)
 
     def test_content_simple_structure(self):
@@ -568,23 +568,23 @@ class MeldElementInterfaceTests(unittest.TestCase):
         self.assertEqual(replacenode.parent, el)
         self.assertEqual(replacenode.text, 'hello')
         self.assertEqual(replacenode.structure, True)
-        from . import Replace
+        from meld3 import Replace
         self.assertEqual(replacenode.tag, Replace)
 
     def test_findmeld_simple(self):
-        from . import _MELD_ID
+        from meld3 import _MELD_ID
         el = self._makeOne('div', {_MELD_ID:'thediv'})
         self.assertEqual(el.findmeld('thediv'), el)
 
     def test_findmeld_simple_oneleveldown(self):
-        from . import _MELD_ID
+        from meld3 import _MELD_ID
         el = self._makeOne('div', {_MELD_ID:'thediv'})
         span = self._makeOne('span', {_MELD_ID:'thespan'})
         el.append(span)
         self.assertEqual(el.findmeld('thespan'), span)
 
     def test_findmeld_simple_twolevelsdown(self):
-        from . import _MELD_ID
+        from meld3 import _MELD_ID
         el = self._makeOne('div', {_MELD_ID:'thediv'})
         span = self._makeOne('span', {_MELD_ID:'thespan'})
         a = self._makeOne('a', {_MELD_ID:'thea'})
@@ -758,7 +758,7 @@ class MeldElementInterfaceTests(unittest.TestCase):
         self.assertEqual(len(parent._children), 0)
 
     def test_lineage(self):
-        from . import _MELD_ID
+        from meld3 import _MELD_ID
         div1 = self._makeOne('div', {_MELD_ID:'div1'})
         span1 = self._makeOne('span', {_MELD_ID:'span1'})
         span2 = self._makeOne('span', {_MELD_ID:'span2'})
@@ -800,7 +800,7 @@ class MeldElementInterfaceTests(unittest.TestCase):
 
 
     def test_shortrepr(self):
-        from ._compat import _b
+        from meld3._compat import _b
         div = self._makeOne('div', {'id':'div1'})
         span = self._makeOne('span', {})
         span2 = self._makeOne('span', {'id':'2'})
@@ -818,8 +818,8 @@ class MeldElementInterfaceTests(unittest.TestCase):
                '<div2 id="div2"></div2></div>'))
 
     def test_shortrepr2(self):
-        from . import parse_xmlstring
-        from ._compat import _b
+        from meld3 import parse_xmlstring
+        from meld3._compat import _b
         root = parse_xmlstring(_COMPLEX_XHTML)
         r = root.shortrepr()
         self.assertEqual(r,
@@ -834,8 +834,8 @@ class MeldElementInterfaceTests(unittest.TestCase):
 
 
     def test_diffmeld1(self):
-        from . import parse_xmlstring
-        from . import _MELD_ID
+        from meld3 import parse_xmlstring
+        from meld3 import _MELD_ID
         root = parse_xmlstring(_COMPLEX_XHTML)
         clone = root.clone()
         div = self._makeOne('div', {_MELD_ID:'newdiv'})
@@ -884,7 +884,7 @@ class MeldElementInterfaceTests(unittest.TestCase):
           <b meld:id="b"></b>
         </root>
         """
-        from . import parse_htmlstring
+        from meld3 import parse_htmlstring
         source_root = parse_htmlstring(source)
         target_root = parse_htmlstring(target)
         changes = source_root.diffmeld(target_root)
@@ -940,7 +940,7 @@ class MeldElementInterfaceTests(unittest.TestCase):
           </d>
         </root>
         """
-        from . import parse_htmlstring
+        from meld3 import parse_htmlstring
         source_root = parse_htmlstring(source)
         target_root = parse_htmlstring(target)
         changes = source_root.diffmeld(target_root)
@@ -1000,7 +1000,7 @@ class MeldElementInterfaceTests(unittest.TestCase):
           </p>
         </root>
         """
-        from . import parse_htmlstring
+        from meld3 import parse_htmlstring
         source_root = parse_htmlstring(source)
         target_root = parse_htmlstring(target)
         changes = source_root.diffmeld(target_root)
@@ -1064,7 +1064,7 @@ class MeldElementInterfaceTests(unittest.TestCase):
           </z>
         </root>
         """
-        from . import parse_htmlstring
+        from meld3 import parse_htmlstring
         source_root = parse_htmlstring(source)
         target_root = parse_htmlstring(target)
         changes = source_root.diffmeld(target_root)
@@ -1099,17 +1099,17 @@ class MeldElementInterfaceTests(unittest.TestCase):
 
 class ParserTests(unittest.TestCase):
     def _parse(self, *args):
-        from . import parse_xmlstring
+        from meld3 import parse_xmlstring
         root = parse_xmlstring(*args)
         return root
 
     def _parse_html(self, *args):
-        from . import parse_htmlstring
+        from meld3 import parse_htmlstring
         root = parse_htmlstring(*args)
         return root
 
     def test_parse_simple_xml(self):
-        from . import _MELD_ID
+        from meld3 import _MELD_ID
         root = self._parse(_SIMPLE_XML)
         self.assertEqual(root.tag, 'root')
         self.assertEqual(root.parent, None)
@@ -1132,7 +1132,7 @@ class ParserTests(unittest.TestCase):
 
     def test_parse_simple_xhtml(self):
         xhtml_ns = '{http://www.w3.org/1999/xhtml}%s'
-        from . import _MELD_ID
+        from meld3 import _MELD_ID
 
         root = self._parse(_SIMPLE_XHTML)
         self.assertEqual(root.tag, xhtml_ns % 'html')
@@ -1145,7 +1145,7 @@ class ParserTests(unittest.TestCase):
 
     def test_parse_complex_xhtml(self):
         xhtml_ns = '{http://www.w3.org/1999/xhtml}%s'
-        from . import _MELD_ID
+        from meld3 import _MELD_ID
         root = self._parse(_COMPLEX_XHTML)
         self.assertEqual(root.tag, xhtml_ns % 'html')
         self.assertEqual(root.attrib, {})
@@ -1218,8 +1218,8 @@ class ParserTests(unittest.TestCase):
         self.assertEqual(td2.parent, tr)
 
     def test_nvu_html(self):
-        from . import _MELD_ID
-        from . import Comment
+        from meld3 import _MELD_ID
+        from meld3 import Comment
         root = self._parse_html(_NVU_HTML)
         self.assertEqual(root.tag, 'html')
         self.assertEqual(root.attrib, {})
@@ -1280,47 +1280,47 @@ class ParserTests(unittest.TestCase):
 class UtilTests(unittest.TestCase):
 
     def test_insert_xhtml_doctype(self):
-        from . import insert_doctype
+        from meld3 import insert_doctype
         orig = '<root></root>'
         actual = insert_doctype(orig)
         expected = '<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd"><root></root>'
         self.assertEqual(actual, expected)
 
     def test_insert_doctype_after_xmldecl(self):
-        from . import insert_doctype
+        from meld3 import insert_doctype
         orig = '<?xml version="1.0" encoding="latin-1"?><root></root>'
         actual = insert_doctype(orig)
         expected = '<?xml version="1.0" encoding="latin-1"?><!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd"><root></root>'
         self.assertEqual(actual, expected)
 
     def test_insert_meld_ns_decl(self):
-        from . import insert_meld_ns_decl
+        from meld3 import insert_meld_ns_decl
         orig = '<?xml version="1.0" encoding="latin-1"?><root></root>'
         actual = insert_meld_ns_decl(orig)
         expected = '<?xml version="1.0" encoding="latin-1"?><root xmlns:meld="http://www.plope.com/software/meld3"></root>'
         self.assertEqual(actual, expected)
 
     def test_prefeed_preserves_existing_meld_ns(self):
-        from . import prefeed
+        from meld3 import prefeed
         orig = '<?xml version="1.0" encoding="latin-1"?><root xmlns:meld="#"></root>'
         actual = prefeed(orig)
         expected = '<?xml version="1.0" encoding="latin-1"?><!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd"><root xmlns:meld="#"></root>'
         self.assertEqual(actual, expected)
 
     def test_prefeed_preserves_existing_doctype(self):
-        from . import prefeed
+        from meld3 import prefeed
         orig = '<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd"><root xmlns:meld="http://www.plope.com/software/meld3"></root>'
         actual = prefeed(orig)
         self.assertEqual(actual, orig)
 
 class WriterTests(unittest.TestCase):
     def _parse(self, xml):
-        from . import parse_xmlstring
+        from meld3 import parse_xmlstring
         root = parse_xmlstring(xml)
         return root
 
     def _parse_html(self, xml):
-        from . import parse_htmlstring
+        from meld3 import parse_htmlstring
         root = parse_htmlstring(xml)
         return root
 
@@ -1345,13 +1345,13 @@ class WriterTests(unittest.TestCase):
         return self._write(node.write_xhtml, **kw)
 
     def assertNormalizedXMLEqual(self, a, b):
-        from ._compat import _u
+        from meld3._compat import _u
         a = normalize_xml(_u(a))
         b = normalize_xml(_u(b))
         self.assertEqual(a, b)
 
     def assertNormalizedHTMLEqual(self, a, b):
-        from ._compat import _u
+        from meld3._compat import _u
         a = normalize_xml(_u(a))
         b = normalize_xml(_u(b))
         self.assertEqual(a, b)
@@ -1487,7 +1487,7 @@ class WriterTests(unittest.TestCase):
         self.assertNormalizedXMLEqual(actual, expected)
 
     def test_write_emptytags_html(self):
-        from ._compat import _u
+        from meld3._compat import _u
         root = self._parse(_EMPTYTAGS_HTML)
         actual = self._write_html(root)
         expected = """<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
@@ -1569,7 +1569,7 @@ class WriterTests(unittest.TestCase):
 
     def test_write_simple_xml_with_doctype(self):
         root = self._parse(_SIMPLE_XML)
-        from . import doctype
+        from meld3 import doctype
         actual = self._write_xml(root, doctype=doctype.xhtml)
         expected = """<?xml version="1.0"?>
         <!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd"><root>
@@ -1584,7 +1584,7 @@ class WriterTests(unittest.TestCase):
 
     def test_write_simple_xml_doctype_nodeclaration(self):
         root = self._parse(_SIMPLE_XML)
-        from . import doctype
+        from meld3 import doctype
         actual = self._write_xml(root, declaration=False,
                                  doctype=doctype.xhtml)
         expected = """<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd"><root>
@@ -1599,7 +1599,7 @@ class WriterTests(unittest.TestCase):
 
     def test_write_simple_xml_fragment_kills_doctype_and_declaration(self):
         root = self._parse(_SIMPLE_XML)
-        from . import doctype
+        from meld3 import doctype
         actual = self._write_xml(root, declaration=True,
                                  doctype=doctype.xhtml, fragment=True)
         expected = """<root>
@@ -1626,14 +1626,14 @@ class WriterTests(unittest.TestCase):
 
     def test_write_simple_xhtml_with_doctype(self):
         root = self._parse(_SIMPLE_XHTML)
-        from . import doctype
+        from meld3 import doctype
         actual = self._write_xhtml(root, doctype=doctype.xhtml)
         expected = """<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd"><html><body>Hello!</body></html>"""
         self.assertNormalizedXMLEqual(actual, expected)
 
     def test_write_simple_xhtml_doctype_nodeclaration(self):
         root = self._parse(_SIMPLE_XHTML)
-        from . import doctype
+        from meld3 import doctype
         actual = self._write_xhtml(root, declaration=False,
                                  doctype=doctype.xhtml)
         expected = """<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd"><html><body>Hello!</body></html>"""
@@ -1641,7 +1641,7 @@ class WriterTests(unittest.TestCase):
 
     def test_write_simple_xhtml_fragment_kills_doctype_and_declaration(self):
         root = self._parse(_SIMPLE_XHTML)
-        from . import doctype
+        from meld3 import doctype
         actual = self._write_xhtml(root, declaration=True,
                                  doctype=doctype.xhtml, fragment=True)
         expected = """<html><body>Hello!</body></html>"""
@@ -1663,7 +1663,7 @@ class WriterTests(unittest.TestCase):
 
     def test_write_simple_xhtml_as_html_new_doctype(self):
         root = self._parse(_SIMPLE_XHTML)
-        from . import doctype
+        from meld3 import doctype
         actual = self._write_html(root, doctype=doctype.html_strict)
         expected = """
 <!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01//EN" "http://www.w3.org/TR/html4/strict.dtd">
@@ -1766,8 +1766,8 @@ class WriterTests(unittest.TestCase):
         self.assertNormalizedXMLEqual(actual, expected)
 
     def test_escape_cdata(self):
-        from ._compat import _b
-        from . import _escape_cdata
+        from meld3._compat import _b
+        from meld3 import _escape_cdata
         a = ('< > &lt;&amp; &&apos; && &foo "" '
              'http://www.plope.com?foo=bar&bang=baz &#123;')
         self.assertEqual(
@@ -1776,15 +1776,15 @@ class WriterTests(unittest.TestCase):
             _escape_cdata(a))
 
     def test_escape_cdata_unicodeerror(self):
-        from . import _escape_cdata
-        from ._compat import _b
-        from ._compat import _u
+        from meld3 import _escape_cdata
+        from meld3._compat import _b
+        from meld3._compat import _u
         a = _u(_b('\x80'))
         self.assertEqual(_b('&#128;'), _escape_cdata(a, 'ascii'))
 
     def test_escape_attrib(self):
-        from . import _escape_attrib
-        from ._compat import _b
+        from meld3 import _escape_attrib
+        from meld3._compat import _b
         a = ('< > &lt;&amp; &&apos; && &foo "" '
              'http://www.plope.com?foo=bar&bang=baz &#123;')
         self.assertEqual(
@@ -1794,9 +1794,9 @@ class WriterTests(unittest.TestCase):
             _escape_attrib(a, None))
 
     def test_escape_attrib_unicodeerror(self):
-        from . import _escape_attrib
-        from ._compat import _b
-        from ._compat import _u
+        from meld3 import _escape_attrib
+        from meld3._compat import _b
+        from meld3._compat import _u
         a = _u(_b('\x80'))
         self.assertEqual(_b('&#128;'), _escape_attrib(a, 'ascii'))
 
@@ -1811,11 +1811,11 @@ def normalize_xml(s):
     s = re.sub(r"(?s)>\s+", ">", s)
     return s
 
-def test_suite():
+def _test_suite():
     return unittest.findTestCases(sys.modules[__name__])
 
 def main():
-    unittest.main(defaultTest='test_suite')
+    unittest.main(defaultTest='_test_suite')
 
 if __name__ == '__main__':
     main()
