--- contrib/python/caio/caio/linux_aio.c	(index)
+++ contrib/python/caio/caio/linux_aio.c	(working tree)
@@ -48 +48 @@ inline static long io_cancel(aio_context_t ctx, struct iocb *aiocb, struct io_ev
-inline int io_cancel_error(int result) {
+inline static int io_cancel_error(int result) {
@@ -85 +85 @@ inline int io_cancel_error(int result) {
-inline int io_submit_error(int result) {
+inline static int io_submit_error(int result) {
--- contrib/python/caio/caio/thread_aio.c	(index)
+++ contrib/python/caio/caio/thread_aio.c	(working tree)
@@ -205 +205 @@ void worker(void *arg) {
-inline int process_pool_error(int code) {
+inline static int process_pool_error(int code) {
