--- contrib/python/caio/ya.make	(index)
+++ contrib/python/caio/ya.make	(working tree)
@@ -12,18 +12,24 @@ NO_COMPILER_WARNINGS()
 
 NO_LINT()
 
-CFLAGS(
-    -DHAVE_FDATASYNC
-)
+IF (OS_LINUX)
+    CFLAGS(
+        -DHAVE_FDATASYNC
+    )
+    SRCS(
+        caio/linux_aio.c
+    )
+    PY_REGISTER(
+        caio.linux_aio
+    )
+ENDIF()
 
 SRCS(
-    caio/linux_aio.c
     caio/src/threadpool/threadpool.c
     caio/thread_aio.c
 )
 
 PY_REGISTER(
-    caio.linux_aio
     caio.thread_aio
 )
 
