--- contrib/python/typed-ast/ast27/Include/compile.h	(index)
+++ contrib/python/typed-ast/ast27/Include/compile.h	(working tree)
@@ -8,9 +8,10 @@
 extern "C" {
 #endif
 
+#if PY_VERSION_HEX < 0x030d0000
 /* Public interface */
 PyAPI_FUNC(PyFutureFeatures *) PyFuture_FromAST(struct _mod *, const char *);
-
+#endif
 
 #ifdef __cplusplus
 }
--- contrib/python/typed-ast/ast3/Python/Python-ast.c	(index)
+++ contrib/python/typed-ast/ast3/Python/Python-ast.c	(working tree)
@@ -855,7 +855,11 @@ static int obj2ast_int(PyObject* obj, int* out, PyArena* arena)
         return 1;
     }
 
+#if PY_VERSION_HEX < 0x030d0000
     i = _PyLong_AsInt(obj);
+#else
+    i = PyLong_AsInt(obj);
+#endif
     if (i == -1 && PyErr_Occurred())
         return 1;
     *out = i;
--- contrib/python/typed-ast/ast3/Python/ast.c	(index)
+++ contrib/python/typed-ast/ast3/Python/ast.c	(working tree)
@@ -20,7 +20,7 @@ typedef int bool;
 #endif
 
 
-#if PY_MINOR_VERSION < 6
+#if PY_MINOR_VERSION < 6 || PY_MINOR_VERSION > 12
 static PyObject *
 _PyObject_FastCall(PyObject *func, PyObject *const *args, int nargs)
 {
