--- contrib/python/tablib/tests/test_tablib.py	(index)
+++ contrib/python/tablib/tests/test_tablib.py	(working tree)
@@ -19,6 +19,8 @@ from tablib.core import Row, detect_format
 from tablib.exceptions import UnsupportedFormat
 from tablib.formats import registry
 
+import yatest.common as yc
+
 try:
     import pandas
 except ImportError:  # pragma: no cover
@@ -382,1 +382,1 @@ class TablibTestCase(BaseTestCase):
-        xlsx_source = Path(__file__).parent / 'files' / 'founders.xlsx'
+        xlsx_source = Path(yc.source_path(__file__)).parent / 'files' / 'founders.xlsx'
@@ -388,1 +388,1 @@ class TablibTestCase(BaseTestCase):
-        xlsx_source = Path(__file__).parent / 'files' / 'founders.xlsx'
+        xlsx_source = Path(yc.source_path(__file__)).parent / 'files' / 'founders.xlsx'
@@ -1191,1 +1191,1 @@ class ODSTests(BaseTestCase):
-        ods_source = Path(__file__).parent / 'files' / 'book.ods'
+        ods_source = Path(yc.source_path(__file__)).parent / 'files' / 'book.ods'
@@ -1205,1 +1205,1 @@ class ODSTests(BaseTestCase):
-        ods_source = Path(__file__).parent / 'files' / 'ragged.ods'
+        ods_source = Path(yc.source_path(__file__)).parent / 'files' / 'ragged.ods'
@@ -1213,1 +1213,1 @@ class ODSTests(BaseTestCase):
-        ods_source = Path(__file__).parent / 'files' / 'unknown_value_type.ods'
+        ods_source = Path(yc.source_path(__file__)).parent / 'files' / 'unknown_value_type.ods'
@@ -1226,1 +1226,1 @@ class XLSTests(BaseTestCase):
-        xls_source = Path(__file__).parent / 'files' / 'dates.xls'
+        xls_source = Path(yc.source_path(__file__)).parent / 'files' / 'dates.xls'
@@ -1241,1 +1241,1 @@ class XLSTests(BaseTestCase):
-        xls_source = Path(__file__).parent / 'files' / 'errors.xls'
+        xls_source = Path(yc.source_path(__file__)).parent / 'files' / 'errors.xls'
@@ -1306,1 +1306,1 @@ class XLSXTests(BaseTestCase):
-        xlsx_source = Path(__file__).parent / 'files' / 'ragged.xlsx'
+        xlsx_source = Path(yc.source_path(__file__)).parent / 'files' / 'ragged.xlsx'
@@ -1313,1 +1313,1 @@ class XLSXTests(BaseTestCase):
-        xlsx_source = Path(__file__).parent / 'files' / 'ragged.xlsx'
+        xlsx_source = Path(yc.source_path(__file__)).parent / 'files' / 'ragged.xlsx'
@@ -1328,1 +1328,1 @@ class XLSXTests(BaseTestCase):
-        xls_source = Path(__file__).parent / 'files' / 'xlsx_cell_values.xlsx'
+        xls_source = Path(yc.source_path(__file__)).parent / 'files' / 'xlsx_cell_values.xlsx'
@@ -1390,1 +1390,1 @@ class XLSXTests(BaseTestCase):
-        xls_source = Path(__file__).parent / 'files' / 'bad_dimensions.xlsx'
+        xls_source = Path(yc.source_path(__file__)).parent / 'files' / 'bad_dimensions.xlsx'
@@ -1531,1 +1531,1 @@ class YAMLTests(BaseTestCase):
-        yaml_source = Path(__file__).parent / 'files' / 'issue_524.yaml'
+        yaml_source = Path(yc.source_path(__file__)).parent / 'files' / 'issue_524.yaml'
--- contrib/python/tablib/tests/test_tablib.py	(index)
+++ contrib/python/tablib/tests/test_tablib.py	(working tree)
@@ -1380 +1380 @@ class XLSXTests(BaseTestCase):
-        xls_source = Path(__file__).parent / 'files' / 'xlsx_cell_values.xlsx'
+        xls_source = Path(yc.source_path(__file__)).parent / 'files' / 'xlsx_cell_values.xlsx'
