--- contrib/python/flake8-commas/flake8_commas/_base.py	(index)
+++ contrib/python/flake8-commas/flake8_commas/_base.py	(working tree)
@@ -1,5 +1,6 @@
 from __future__ import annotations
 
+import sys
 import collections
 import importlib.metadata
 import token as mod_token
@@ -20,6 +21,8 @@ try:
    col: int
 
 
+_IS_PEP701 = sys.version_info >= (3, 12)
+
 # A parenthesized expression list yields whatever that expression list
 # yields: if the list contains at least one comma, it yields a tuple;
 # otherwise, it yields the single expression that makes up the expression
@@ -95,12 +98,25 @@ LAMBDA = 'lambda'
 COLON = ':'
 NONE = SimpleToken(token=None, type=None)
 
+fstring_nesting = 0
+
 
 def get_type(token: Token) -> str | None:
     type = token.type
     if type == tokenize.NL:
         return NEW_LINE
 
+    if _IS_PEP701:
+        global fstring_nesting
+        if type == tokenize.FSTRING_START:
+            fstring_nesting += 1
+        elif type == tokenize.FSTRING_END:
+            fstring_nesting -= 1
+
+        # we don't check if inside f-strings
+        if fstring_nesting > 0:
+            return
+
     string = token.string
     if type == tokenize.NAME and string == 'for':
         return FOR
