diff --git a/contrib/restricted/patched/replxx/src/replxx_impl.cxx b/contrib/restricted/patched/replxx/src/replxx_impl.cxx
--- a/contrib/restricted/patched/replxx/src/replxx_impl.cxx
+++ b/contrib/restricted/patched/replxx/src/replxx_impl.cxx
@@ -576,7 +576,7 @@ void Replxx::ReplxxImpl::set_preload_buffer( std::string const& preloadText ) {
 		if ( whitespaceSeen > 0 ) {
 			it -= whitespaceSeen;
 			*it = ' ';
-			_preloadedBuffer.erase( it + 1, it + whitespaceSeen - 1 );
+			_preloadedBuffer.erase( it + 1, it + whitespaceSeen );
 		}
 		if ( is_control_code( c ) ) { // remove other control characters, flag for message
 			controlsStripped = true;
