--- contrib/python/aiohttp/aiohttp/_http_parser.pyx	(index)
+++ contrib/python/aiohttp/aiohttp/_http_parser.pyx	(working tree)
@@ -393,1 +393,1 @@ cdef class HttpParser:
-    cdef _on_header_field(self, char* at, size_t length):
+    cdef _on_header_field(self, const char* at, size_t length):
@@ -404,1 +404,1 @@ cdef class HttpParser:
-    cdef _on_header_value(self, char* at, size_t length):
+    cdef _on_header_value(self, const char* at, size_t length):
