--- contrib/python/aiohttp/aiohttp/web_response.py      (ddcb92de87597ba3c0a8961e7fdf04a184c227ce)
+++ contrib/python/aiohttp/aiohttp/web_response.py      (0978c4fe84e8994e041f045b1447dd8058efa52c)
@@ -487,8 +487,7 @@ class StreamResponse(BaseClass, HeadersMixin):
             # https://datatracker.ietf.org/doc/html/rfc9112#section-6.1-13
             if hdrs.TRANSFER_ENCODING in headers:
                 del headers[hdrs.TRANSFER_ENCODING]
-        elif self.content_length != 0:
-            # https://www.rfc-editor.org/rfc/rfc9110#section-8.3-5
+        else:
             headers.setdefault(hdrs.CONTENT_TYPE, "application/octet-stream")
         headers.setdefault(hdrs.DATE, rfc822_formatted_time())
         headers.setdefault(hdrs.SERVER, SERVER_SOFTWARE)
