--- contrib/python/aiohttp/aiohttp/client.py	(index)
+++ contrib/python/aiohttp/aiohttp/client.py	(working tree)
@@ -574,7 +574,7 @@ class ClientSession:
             with timer:
                 # https://www.rfc-editor.org/rfc/rfc9112.html#name-retrying-requests
                 retry_persistent_connection = (
-                    self._retry_connection and method in IDEMPOTENT_METHODS
+                    False  # self._retry_connection and method in IDEMPOTENT_METHODS
                 )
                 while True:
                     url, auth_from_url = strip_auth_from_url(url)
