--- contrib/python/aiohttp/aiohttp/client.py	(index)
+++ contrib/python/aiohttp/aiohttp/client.py	(working tree)
@@ -574,7 +574,7 @@ class ClientSession:
         try:
             with timer:
                 # https://www.rfc-editor.org/rfc/rfc9112.html#name-retrying-requests
-                retry_persistent_connection = method in IDEMPOTENT_METHODS
+                retry_persistent_connection = False #method in IDEMPOTENT_METHODS
                 while True:
                     url, auth_from_url = strip_auth_from_url(url)
                     if not url.raw_host:
