--- contrib/python/safehttpx/safehttpx/__init__.py	(a387bf64ceaef1f356cbfabaa871bfd91355f8cd)
+++ contrib/python/safehttpx/safehttpx/__init__.py	(index)
@@ -6,14 +6,14 @@ import os
 from functools import lru_cache, wraps
 from typing import Any, Awaitable, Callable, Coroutine, Literal, T, Tuple
 
+import importlib.resources
+
 import httpx
 
 def get_version():
-    version_file = Path(__file__).parent / 'version.txt'
-    with open(version_file, 'r') as f:
-        return f.read().strip()
+    return (importlib.resources.files(__package__) / 'version.txt').read_text().strip()
 
-__version__ = get_version() 
+__version__ = get_version()
 
 def is_public_ip(ip: str) -> bool:
     try:
