--- contrib/python/jupyterhub/jupyterhub/auth.py	(index)
+++ contrib/python/jupyterhub/jupyterhub/auth.py	(working tree)
@@ -5 +5 @@ import inspect
-import pipes
+import shlex
@@ -904 +904 @@ class LocalAuthenticator(Authenticator):
-        self.log.info("Creating user: %s", ' '.join(map(pipes.quote, cmd)))
+        self.log.info("Creating user: %s", ' '.join(map(shlex.quote, cmd)))
--- contrib/python/jupyterhub/jupyterhub/services/service.py	(index)
+++ contrib/python/jupyterhub/jupyterhub/services/service.py	(working tree)
@@ -43 +43 @@ import os
-import pipes
+import shlex
@@ -121 +121 @@ class _ServiceSpawner(LocalProcessSpawner):
-        self.log.info("Spawning %s", ' '.join(pipes.quote(s) for s in cmd))
+        self.log.info("Spawning %s", ' '.join(shlex.quote(s) for s in cmd))
--- contrib/python/jupyterhub/jupyterhub/spawner.py	(index)
+++ contrib/python/jupyterhub/jupyterhub/spawner.py	(working tree)
@@ -9 +9 @@ import os
-import pipes
+import shlex
@@ -1481 +1481 @@ class LocalProcessSpawner(Spawner):
-            cmd = self.shell_cmd + [' '.join(pipes.quote(s) for s in cmd)]
+            cmd = self.shell_cmd + [' '.join(shlex.quote(s) for s in cmd)]
@@ -1483 +1483 @@ class LocalProcessSpawner(Spawner):
-        self.log.info("Spawning %s", ' '.join(pipes.quote(s) for s in cmd))
+        self.log.info("Spawning %s", ' '.join(shlex.quote(s) for s in cmd))
