--- contrib/python/lazy_loader/lazy_loader/__init__.py	(index)
+++ contrib/python/lazy_loader/lazy_loader/__init__.py	(working tree)
@@ -341,5 +341,3 @@ def attach_stub(package_name: str, filename: str):
-    if not os.path.exists(stubfile):
-        raise ValueError(f"Cannot load imports from non-existent stub {stubfile!r}")
-
-    with open(stubfile) as f:
-        stub_node = ast.parse(f.read())
+    from library.python import resource
+    stub_data = resource.find(os.path.join("resfs/file/py", package_name.replace(".", "/"), os.path.basename(stubfile)))
+    stub_node = ast.parse(stub_data)
