--- contrib/python/python-pptx/py2/pptx/oxml/__init__.py	(index)
+++ contrib/python/python-pptx/py2/pptx/oxml/__init__.py	(working tree)
@@ -12,6 +12,7 @@ import os
 from lxml import etree
 
 from .ns import NamespacePrefixedTag
+from ..util import get_file
 
 
 # configure etree XML parser -------------------------------
@@ -25,10 +26,8 @@ def parse_from_template(template_name):
     Return an element loaded from the XML in the template file identified by
     *template_name*.
     """
-    thisdir = os.path.split(__file__)[0]
-    filename = os.path.join(thisdir, "..", "templates", "%s.xml" % template_name)
-    with open(filename, "rb") as f:
-        xml = f.read()
+    f = get_file('templates', '%s.xml' % template_name)
+    xml = f.read()
     return parse_xml(xml)
 
 
--- contrib/python/python-pptx/py2/pptx/shapes/shapetree.py	(index)
+++ contrib/python/python-pptx/py2/pptx/shapes/shapetree.py	(working tree)
@@ -31,7 +31,7 @@ from pptx.shapes.placeholder import (
     TablePlaceholder,
 )
 from pptx.shared import ParentedElementProxy
-from pptx.util import Emu, lazyproperty
+from pptx.util import Emu, lazyproperty, get_file
 
 # +-- _BaseShapes
 # |   |
@@ -1085,7 +1085,7 @@ class _OleObjectElementCreator(object):
         )
 
         _thisdir = os.path.split(__file__)[0]
-        return os.path.abspath(os.path.join(_thisdir, "..", "templates", icon_filename))
+        return get_file("templates", icon_filename)
 
     @lazyproperty
     def _icon_rId(self):
--- contrib/python/python-pptx/py2/pptx/api.py	(index)
+++ contrib/python/python-pptx/py2/pptx/api.py	(working tree)
@@ -13,6 +13,7 @@ import os
 
 from .opc.constants import CONTENT_TYPE as CT
 from .package import Package
+from .util import get_file
 
 
 def Presentation(pptx=None):
@@ -23,7 +24,7 @@ def Presentation(pptx=None):
     "template" is loaded.
     """
     if pptx is None:
-        pptx = _default_pptx_path()
+        pptx = _default_pptx()
 
     presentation_part = Package.open(pptx).main_document_part
 
@@ -34,12 +35,11 @@ def Presentation(pptx=None):
     return presentation_part.presentation
 
 
-def _default_pptx_path():
+def _default_pptx():
     """
     Return the path to the built-in default .pptx package.
     """
-    _thisdir = os.path.split(__file__)[0]
-    return os.path.join(_thisdir, "templates", "default.pptx")
+    return get_file('templates', 'default.pptx')
 
 
 def _is_pptx_package(prs_part):
--- contrib/python/python-pptx/py2/pptx/util.py	(index)
+++ contrib/python/python-pptx/py2/pptx/util.py	(working tree)
@@ -236,3 +236,11 @@ def lazyproperty(f):
         quite snappy and probably not a rich target for optimization efforts.
         """
         raise AttributeError("can't set attribute")  # pragma: no cover
+
+
+import io
+import pkgutil
+
+
+def get_file(*args):
+    return io.BytesIO(pkgutil.get_data('pptx', '/'.join(args)))
