https://st.yandex-team.ru/DEVTOOLSSUPPORT-2233
Отключаем проверку F821 для type comments, чтобы не ломать людям тесты,
но востановить проверку F821, как было в старой проверке до перехода на flake8

--- contrib/python/pyflakes/pyflakes/checker.py	(index)
+++ contrib/python/pyflakes/pyflakes/checker.py	(working tree)
@@ -1146,1 +1146,2 @@ class Checker(object):
-            self.report(messages.UndefinedName, node, name)
+            if not getattr(node, '_xxx_skip_F821_for_type_comment', False):
+                self.report(messages.UndefinedName, node, name)
@@ -1234,6 +1235,8 @@ class Checker(object):
 
     def handleChildren(self, tree, omit=None):
         for node in iter_child_nodes(tree, omit=omit):
+            if node is not None:
+                node._xxx_skip_F821_for_type_comment = getattr(tree, '_xxx_skip_F821_for_type_comment', False)
             self.handleNode(node, tree)
 
     def isLiteralTupleUnpacking(self, node):
@@ -1340,6 +1343,7 @@ class Checker(object):
                 descendant.lineno = ref_lineno
                 descendant.col_offset = ref_col_offset
 
+        parsed_annotation._xxx_skip_F821_for_type_comment = True
         self.handleNode(parsed_annotation, node)
 
     @in_annotation
