--- contrib/python/augmentex/augmentex/base.py	(index)
+++ contrib/python/augmentex/augmentex/base.py	(working tree)
@@ -4,12 +4,20 @@ from abc import ABC, abstractmethod
 from typing import List, Union, Dict

 import numpy as np
+import library.python.resource

 from augmentex.variables import SUPPORT_LANGUAGES, SUPPORT_PLATFORMS


 class BaseAug(ABC):
-    def __init__(self, min_aug: int = 1, max_aug: int = 5, random_seed: int = None, lang: str = "rus", platform: str = "pc") -> None:
+    def __init__(
+        self,
+        min_aug: int = 1,
+        max_aug: int = 5,
+        random_seed: int = None,
+        lang: str = "rus",
+        platform: str = "pc",
+    ) -> None:
         """
         Args:
             min_aug (int, optional): The minimum amount of augmentation. Defaults to 1.
@@ -30,11 +38,13 @@ class BaseAug(ABC):
         if self.lang not in SUPPORT_LANGUAGES:
             raise ValueError(
                 f"""Augmentex support only {', '.join(SUPPORT_LANGUAGES)} languages.
-                You put {self.lang}.""")
+                You put {self.lang}."""
+            )
         if self.platform not in SUPPORT_PLATFORMS:
             raise ValueError(
                 f"""Augmentex support only {', '.join(SUPPORT_PLATFORMS)} platforms.
-                You put {self.platform}.""")
+                You put {self.platform}."""
+            )

     def _read_json(self, path: str) -> Dict:
         """Read JSON to Dict.
@@ -45,8 +55,7 @@ class BaseAug(ABC):
         Returns:
             Dict: dict with data.
         """
-        with open(path, encoding='utf-8') as f:
-            data = json.load(f)
+        data = json.loads(library.python.resource.resfs_read(path).decode("utf-8"))

         return data

--- contrib/python/augmentex/augmentex/char.py	(index)
+++ contrib/python/augmentex/augmentex/char.py	(working tree)
@@ -35,24 +35,34 @@ class CharAug(BaseAug):
             correct_texts_path (str, optional): Path to txt file with correct texts. Defaults to None.
             error_texts_path (str, optional): Path to txt file with error texts. Default to None.
         """
-        super().__init__(min_aug=min_aug, max_aug=max_aug,
-                         random_seed=random_seed, lang=lang, platform=platform)
-        dir_path = os.path.dirname(os.path.abspath(__file__))
+        super().__init__(
+            min_aug=min_aug,
+            max_aug=max_aug,
+            random_seed=random_seed,
+            lang=lang,
+            platform=platform,
+        )
+        dir_path = "contrib/python/augmentex/augmentex"

-        self.typo_dict = self._read_json(os.path.join(
-            dir_path, "static_data", f"{self.platform}_typos_chars.json"))
+        self.typo_dict = self._read_json(
+            os.path.join(dir_path, "static_data", f"{self.platform}_typos_chars.json")
+        )
         self.shift_dict = self._read_json(
-            os.path.join(dir_path, "static_data", "shift.json"))
-        self.vocab = self._read_json(os.path.join(
-            dir_path, "static_data", self.lang, "vocab.json"))
+            os.path.join(dir_path, "static_data", "shift.json")
+        )
+        self.vocab = self._read_json(
+            os.path.join(dir_path, "static_data", self.lang, "vocab.json")
+        )

         if correct_texts_path is not None or error_texts_path is not None:
-            cs = ComputeStatistic(correct_texts_path,
-                                  error_texts_path, self.lang)
+            cs = ComputeStatistic(correct_texts_path, error_texts_path, self.lang)
             self.orfo_dict = cs.compute_char_statistic()
         else:
-            self.orfo_dict = self._read_json(os.path.join(
-                dir_path, "static_data", self.lang, self.platform, "orfo_chars.json"))
+            self.orfo_dict = self._read_json(
+                os.path.join(
+                    dir_path, "static_data", self.lang, self.platform, "orfo_chars.json"
+                )
+            )

         self.mult_num = mult_num
         self.unit_prob = unit_prob
@@ -104,9 +114,7 @@ class CharAug(BaseAug):
         if self.orfo_dict.get(char, None) == None:
             orfo_char = char
         else:
-            orfo_char = np.random.choice(
-                self.vocab, p=self.orfo_dict.get(char, None)
-            )
+            orfo_char = np.random.choice(self.vocab, p=self.orfo_dict.get(char, None))

         return orfo_char

--- contrib/python/augmentex/augmentex/preprocessor.py	(index)
+++ contrib/python/augmentex/augmentex/preprocessor.py	(working tree)
@@ -1,25 +1,27 @@
 import os
 import re
 import json
+import library.python.resource
 from typing import List, Dict, Tuple, Union
 from collections import defaultdict
-
-import Levenshtein as levenshtein
+from pylev import levenshtein

 from augmentex.variables import SUPPORT_LANGUAGES


-class ComputeStatistic():
+class ComputeStatistic:
     """A class for counting spelling errors."""

-    def __init__(self, correct_texts_path: str, error_texts_path: str, lang: str) -> None:
+    def __init__(
+        self, correct_texts_path: str, error_texts_path: str, lang: str
+    ) -> None:
         """
         Args:
             correct_texts_path (str): Path to txt file with correct texts.
             error_texts_path (str): Path to txt file with error texts.
             lang (str): Language of texts.
         """
-        dir_path = os.path.dirname(os.path.abspath(__file__))
+        dir_path = "contrib/python/augmentex/augmentex"
         self.lang = lang

         self.__check_txt_file(correct_texts_path)
@@ -28,16 +30,19 @@ class ComputeStatistic():
         if self.lang not in SUPPORT_LANGUAGES:
             raise ValueError(
                 f"""Augmentex support only {', '.join(SUPPORT_LANGUAGES)} languages.
-                You put {self.lang}.""")
+                You put {self.lang}."""
+            )

-        self.vocab = self.__read_json(os.path.join(
-            dir_path, "static_data", self.lang, "vocab.json"))
+        self.vocab = self.__read_json(
+            os.path.join(dir_path, "static_data", self.lang, "vocab.json")
+        )
         self.correct_texts = self.__read_txt_file(correct_texts_path)
         self.error_texts = self.__read_txt_file(error_texts_path)

         if len(self.correct_texts) != len(self.error_texts):
             raise ValueError(
-                f"Correct texts and texts with errors must be same length. Your length is {len(self.correct_texts)} and {len(self.error_texts)} respectively.")
+                f"Correct texts and texts with errors must be same length. Your length is {len(self.correct_texts)} and {len(self.error_texts)} respectively."
+            )

     def __read_json(self, path: str) -> Dict:
         """Read JSON to Dict.
@@ -48,8 +53,7 @@ class ComputeStatistic():
         Returns:
             Dict: dict with data.
         """
-        with open(path) as f:
-            data = json.load(f)
+        data = json.loads(library.python.resource.resfs_read(path).decode("utf-8"))

         return data

@@ -94,12 +98,13 @@ class ComputeStatistic():
         else:
             regex = r"[a-z]+"

-        data = list(map(lambda x: " ".join(
-            re.findall(regex, x.lower())), data))
+        data = list(map(lambda x: " ".join(re.findall(regex, x.lower())), data))

         return data

-    def __filter(self, correct_texts: List[str], error_texts: List[str]) -> List[Tuple[str, str]]:
+    def __filter(
+        self, correct_texts: List[str], error_texts: List[str]
+    ) -> List[Tuple[str, str]]:
         """Matches the correct word with the word with an error.

         Args:
@@ -118,12 +123,17 @@ class ComputeStatistic():
                     for word_idx in range(len(correct_texts_i_split)):
                         true_word = correct_texts_i_split[word_idx]
                         broke_word = error_texts_i_split[word_idx]
-                        if true_word != broke_word and levenshtein.distance(true_word, broke_word) <= 2:
+                        if (
+                            true_word != broke_word
+                            and levenshtein(true_word, broke_word) <= 2
+                        ):
                             pairs.append((true_word, broke_word))

         return pairs

-    def __compute_word_statistic(self, pairs: List[Tuple[str, str]]) -> Dict[str, List[Union[List[str], List[float]]]]:
+    def __compute_word_statistic(
+        self, pairs: List[Tuple[str, str]]
+    ) -> Dict[str, List[Union[List[str], List[float]]]]:
         """From pairs of words , it compiles statistics on the use of words with errors.

         Args:
@@ -148,11 +158,13 @@ class ComputeStatistic():
                 word_statistic[true_word] = [[broke_word], [count]]

         for item in word_statistic.items():
-            item[1][1] = [float(i)/sum(item[1][1]) for i in item[1][1]]
+            item[1][1] = [float(i) / sum(item[1][1]) for i in item[1][1]]

         return word_statistic

-    def __compute_char_statistic(self, pairs: List[Tuple[str, str]]) -> Dict[str, List[float]]:
+    def __compute_char_statistic(
+        self, pairs: List[Tuple[str, str]]
+    ) -> Dict[str, List[float]]:
         """From pairs of words , statistics on the use of the wrong char are compiled.

         Args:
@@ -195,12 +207,16 @@ class ComputeStatistic():
                 tmp[broke_char] = count
                 char_statistic[true_char] = tmp

-        char_statistic = {k: [float(i)/sum(list(v.values()))
-                              for i in list(v.values())] for k, v in char_statistic.items()}
+        char_statistic = {
+            k: [float(i) / sum(list(v.values())) for i in list(v.values())]
+            for k, v in char_statistic.items()
+        }

         return char_statistic

-    def __compute_ngram_statistic(self, pairs: List[Tuple[str, str]], n: int = 3) -> Dict[str, List[Union[List[str], List[float]]]]:
+    def __compute_ngram_statistic(
+        self, pairs: List[Tuple[str, str]], n: int = 3
+    ) -> Dict[str, List[Union[List[str], List[float]]]]:
         """From pairs of words , it compiles statistics on the use of ngrams with errors.

         Args:
@@ -217,14 +233,17 @@ class ComputeStatistic():
             if len(true_word) == len(broke_word):
                 if len(true_word) < n:
                     continue
-                true_word_ngram = [true_word[i:i+n]
-                                   for i in range(len(true_word)-n+1)]
-                broke_word_ngram = [broke_word[i:i+n]
-                                    for i in range(len(broke_word)-n+1)]
+                true_word_ngram = [
+                    true_word[i : i + n] for i in range(len(true_word) - n + 1)
+                ]
+                broke_word_ngram = [
+                    broke_word[i : i + n] for i in range(len(broke_word) - n + 1)
+                ]
                 for gram_id in range(len(true_word_ngram)):
                     if true_word_ngram[gram_id] != broke_word_ngram[gram_id]:
                         ngram_pairs.append(
-                            (true_word_ngram[gram_id], broke_word_ngram[gram_id]))
+                            (true_word_ngram[gram_id], broke_word_ngram[gram_id])
+                        )

         count_ngram_pairs = defaultdict(int)
         for pair in ngram_pairs:
@@ -242,7 +261,7 @@ class ComputeStatistic():
                 ngram_statistic[true_gram] = [[broke_gram], [count]]

         for item in ngram_statistic.items():
-            item[1][1] = [float(i)/sum(item[1][1]) for i in item[1][1]]
+            item[1][1] = [float(i) / sum(item[1][1]) for i in item[1][1]]

         return ngram_statistic

--- contrib/python/augmentex/augmentex/variables.py	(index)
+++ contrib/python/augmentex/augmentex/variables.py	(working tree)
@@ -1,6 +1,13 @@
 SUPPORT_LANGUAGES = ["rus", "eng"]
 SUPPORT_PLATFORMS = ["pc", "mobile"]
-WORD_ACTIONS = ["replace", "delete", "swap", "stopword",
-                "reverse", "text2emoji", "split", "ngram"]
-CHAR_ACTIONS = ["shift", "orfo", "typo",
-                "delete", "multiply", "swap", "insert"]
+WORD_ACTIONS = [
+    "replace",
+    "delete",
+    "swap",
+    "stopword",
+    "reverse",
+    "text2emoji",
+    "split",
+    "ngram",
+]
+CHAR_ACTIONS = ["shift", "orfo", "typo", "delete", "multiply", "swap", "insert"]
--- contrib/python/augmentex/augmentex/word.py	(index)
+++ contrib/python/augmentex/augmentex/word.py	(working tree)
@@ -34,24 +34,40 @@ class WordAug(BaseAug):
             correct_texts_path (str, optional): Path to txt file with correct texts. Defaults to None.
             error_texts_path (str, optional): Path to txt file with error texts. Default to None.
         """
-        super().__init__(min_aug=min_aug, max_aug=max_aug,
-                         random_seed=random_seed, lang=lang, platform=platform)
-        dir_path = os.path.dirname(os.path.abspath(__file__))
+        super().__init__(
+            min_aug=min_aug,
+            max_aug=max_aug,
+            random_seed=random_seed,
+            lang=lang,
+            platform=platform,
+        )
+        dir_path = "contrib/python/augmentex/augmentex"

-        self.stopwords = self._read_json(os.path.join(
-            dir_path, "static_data", self.lang, "stopwords.json"))
-        self.text2emoji_map = self._read_json(os.path.join(
-            dir_path, "static_data", self.lang, "text2emoji.json"))
+        self.stopwords = self._read_json(
+            os.path.join(dir_path, "static_data", self.lang, "stopwords.json")
+        )
+        self.text2emoji_map = self._read_json(
+            os.path.join(dir_path, "static_data", self.lang, "text2emoji.json")
+        )

         if correct_texts_path is not None or error_texts_path is not None:
-            cs = ComputeStatistic(correct_texts_path,
-                                  error_texts_path, self.lang)
+            cs = ComputeStatistic(correct_texts_path, error_texts_path, self.lang)
             self.orfo_dict, self.ngram_dict = cs.compute_word_statistic()
         else:
-            self.orfo_dict = self._read_json(os.path.join(
-                dir_path, "static_data", self.lang, self.platform, "orfo_words.json"))
-            self.ngram_dict = self._read_json(os.path.join(
-                dir_path, "static_data", self.lang, self.platform, "orfo_ngrams.json"))
+            self.orfo_dict = self._read_json(
+                os.path.join(
+                    dir_path, "static_data", self.lang, self.platform, "orfo_words.json"
+                )
+            )
+            self.ngram_dict = self._read_json(
+                os.path.join(
+                    dir_path,
+                    "static_data",
+                    self.lang,
+                    self.platform,
+                    "orfo_ngrams.json",
+                )
+            )

         self.unit_prob = unit_prob

@@ -66,12 +82,12 @@ class WordAug(BaseAug):

     def __ngram(self, word: str, n: int = 3) -> str:
         if len(word) > 3:
-            word_ngrams = [word[i:i+n] for i in range(len(word)-n+1)]
+            word_ngrams = [word[i : i + n] for i in range(len(word) - n + 1)]
             random_ngram = np.random.choice(word_ngrams)
             ngram_probas = self.ngram_dict.get(
-                random_ngram.lower(), [[random_ngram], [1.0]])
-            ngram_for_replace = np.random.choice(
-                ngram_probas[0], p=ngram_probas[1])
+                random_ngram.lower(), [[random_ngram], [1.0]]
+            )
+            ngram_for_replace = np.random.choice(ngram_probas[0], p=ngram_probas[1])
             word = word.replace(random_ngram, ngram_for_replace)

         return word
