--- contrib/python/pyreadstat/pyreadstat/_readstat_parser.pyx	(index)
+++ contrib/python/pyreadstat/pyreadstat/_readstat_parser.pyx	(working tree)
@@ -267,1 +267,1 @@ cdef object convert_readstat_to_python_value(readstat_value_t value, int index, 
-    cdef char * c_str_value
+    cdef const char * c_str_value
@@ -356,2 +356,2 @@ cdef int handle_metadata(readstat_metadata_t *metadata, void *ctx) except READST
-    cdef char * flabel_orig
-    cdef char * fencoding_orig
+    cdef const char * flabel_orig
+    cdef const char * fencoding_orig
@@ -360,1 +360,1 @@ cdef int handle_metadata(readstat_metadata_t *metadata, void *ctx) except READST
-    cdef char * table
+    cdef const char * table
@@ -410,3 +410,3 @@ cdef int handle_variable(int index, readstat_variable_t *variable,
-    cdef char * var_name, 
-    cdef char * var_label
-    cdef char * var_format
+    cdef const char * var_name, 
+    cdef const char * var_label
+    cdef const char * var_format
@@ -670,1 +670,1 @@ cdef int handle_value_label(char *val_labels, readstat_value_t value, char *labe
-    cdef char * c_str_value
+    cdef const char * c_str_value
@@ -792,1 +792,1 @@ cdef void check_exit_status(readstat_error_t retcode) except *:
-    cdef char * err_readstat
+    cdef const char * err_readstat
--- contrib/python/pyreadstat/pyreadstat/_readstat_writer.pyx	(index)
+++ contrib/python/pyreadstat/pyreadstat/_readstat_writer.pyx	(working tree)
@@ -507,1 +507,1 @@ cdef void _check_exit_status(readstat_error_t retcode) except *:
-    cdef char * err_readstat
+    cdef const char * err_readstat
