--- contrib/python/pyreadstat/pyreadstat/conditional_includes.h	(index)
+++ contrib/python/pyreadstat/pyreadstat/conditional_includes.h	(working tree)
@@ -21,15 +21,15 @@
-    int _wsopen(const wchar_t *filename, int oflag, int shflag, int pmode){ return 0; };
-    int _O_RDONLY;
-    int _O_BINARY;
-    int _O_WRONLY;
-    int _O_TRUNC;
-    int _O_CREAT;
-    int _SH_DENYRW;  
-    int _SH_DENYWR; 
-    int _SH_DENYRD;
-    int _SH_DENYNO;
-    int _S_IWRITE;
-    int _S_IREAD;
-    void assign_fd(void *io_ctx, int fd){};
-    int _close(int fd){ return 0; };
-    ssize_t _write(int fd, const void *buf, size_t nbyte){return 0;};
+    #define _wsopen(...) 0
+    #define _O_RDONLY 0
+    #define _O_BINARY 0
+    #define _O_WRONLY 0
+    #define _O_TRUNC 0
+    #define _O_CREAT 0
+    #define _SH_DENYRW 0
+    #define _SH_DENYWR 0
+    #define _SH_DENYRD 0
+    #define _SH_DENYNO 0
+    #define _S_IWRITE 0
+    #define _S_IREAD 0
+    #define assign_fd(...) {}
+    #define _close(...) 0
+    #define _write(...) 0
