diff -r -u grpc-old/src/core/tsi/alts/crypt/gsec.h grpc/src/core/tsi/alts/crypt/gsec.h
--- a/src/core/tsi/alts/crypt/gsec.h
+++ b/src/core/tsi/alts/crypt/gsec.h
@@ -98,8 +98,10 @@

 }  // namespace grpc_core

-#ifndef _STRUCT_IOVEC
+#if !defined(_STRUCT_IOVEC) && !defined(__DEFINED_struct_iovec)
 #if !defined(GRPC_EVENT_ENGINE_POSIX)
+#define _STRUCT_IOVEC
+#define __DEFINED_struct_iovec
 struct iovec {
   void* iov_base;
   size_t iov_len;
