--- contrib/libs/grpc/src/core/lib/event_engine/thread_pool/thread_pool_factory.cc	(index)
+++ contrib/libs/grpc/src/core/lib/event_engine/thread_pool/thread_pool_factory.cc	(working tree)
@@ -23,6 +23,8 @@
 #include "src/core/lib/event_engine/thread_pool/thread_pool.h"
 #include "src/core/lib/event_engine/thread_pool/work_stealing_thread_pool.h"
 #include "src/core/lib/gprpp/no_destruct.h"
+#include "src/core/lib/gprpp/env.h"
+#include "src/core/lib/gpr/string.h"
 
 namespace grpc_event_engine {
 namespace experimental {
@@ -36,15 +38,32 @@ class ThreadPoolForkCallbackMethods {
   static void PostforkParent() { g_thread_pool_fork_manager->PostforkParent(); }
   static void PostforkChild() { g_thread_pool_fork_manager->PostforkChild(); }
 };
+
+size_t GetMaxPoolThreadFromEnv() {
+  auto value = grpc_core::GetEnv("GRPC_MAX_THREADS_THREAD_POOL_ENV");
+  if (!value.has_value()) return 0;
+  int parse_succeeded = gpr_parse_nonnegative_int(value->c_str());
+
+  if (parse_succeeded <= 0) {
+     return 0;
+  }
+  return static_cast<size_t>(parse_succeeded);
+}
+
 }  // namespace
 
 void ThreadPool::SetThreadsLimit(size_t count) {
+    const auto threads_limit_env = GetMaxPoolThreadFromEnv();
+    if (threads_limit_env && threads_limit_env < count) {
+        gpr_log(GPR_INFO, "Threads limit changed via env from %u to %u", count, threads_limit_env);
+        count = threads_limit_env;
+    }
     threads_limit_ = count;
 }
 
 size_t ThreadPool::GetThreadsLimit() { return threads_limit_; }
 
-size_t ThreadPool::threads_limit_ = grpc_core::Clamp(gpr_cpu_num_cores(), 2u, 32u);
+size_t ThreadPool::threads_limit_ = GetMaxPoolThreadFromEnv() ? GetMaxPoolThreadFromEnv() : grpc_core::Clamp(gpr_cpu_num_cores(), 2u, 32u);
 
 std::shared_ptr<ThreadPool> MakeThreadPool(size_t reserve_threads) {
   auto thread_pool = std::make_shared<WorkStealingThreadPool>(reserve_threads);
