--- contrib/libs/grpc/src/core/lib/event_engine/forkable.cc	(index)
+++ contrib/libs/grpc/src/core/lib/event_engine/forkable.cc	(working tree)
@@ -29,6 +29,25 @@

 #include "src/core/lib/config/config_vars.h"

+#include "src/core/lib/debug/trace.h"
+
+namespace grpc_event_engine {
+namespace experimental {
+
+extern grpc_core::TraceFlag grpc_trace_fork;
+
+#define GRPC_FORK_TRACE_LOG(format, ...)                 \
+  do {                                                   \
+    if (GRPC_TRACE_FLAG_ENABLED(grpc_trace_fork)) {      \
+      gpr_log(GPR_DEBUG, "[fork] " format, __VA_ARGS__); \
+    }                                                    \
+  } while (0)
+
+#define GRPC_FORK_TRACE_LOG_STRING(format) GRPC_FORK_TRACE_LOG("%s", format)
+
+}
+}
+
 namespace grpc_event_engine {
 namespace experimental {

--- contrib/libs/grpc/src/core/lib/event_engine/forkable.h	(index)
+++ contrib/libs/grpc/src/core/lib/event_engine/forkable.h	(working tree)
@@ -19,24 +19,10 @@
 #include <memory>
 #include <vector>

-#include <grpc/support/log.h>
-
-#include "src/core/lib/debug/trace.h"

 namespace grpc_event_engine {
 namespace experimental {

-extern grpc_core::TraceFlag grpc_trace_fork;
-
-#define GRPC_FORK_TRACE_LOG(format, ...)                 \
-  do {                                                   \
-    if (GRPC_TRACE_FLAG_ENABLED(grpc_trace_fork)) {      \
-      gpr_log(GPR_DEBUG, "[fork] " format, __VA_ARGS__); \
-    }                                                    \
-  } while (0)
-
-#define GRPC_FORK_TRACE_LOG_STRING(format) GRPC_FORK_TRACE_LOG("%s", format)
-
 // We've faced flickering crash in grpc during fork.
 // Crash is observed in different places before 'execv' is called.
 // It is mentioned in https://man7.org/linux/man-pages/man2/fork.2.html
