--- contrib/libs/grpc/src/core/lib/surface/init.cc	(index)
+++ contrib/libs/grpc/src/core/lib/surface/init.cc	(working tree)
@@ -124,8 +124,8 @@ void grpc_shutdown_internal_locked(void)
     grpc_core::ExecCtx exec_ctx(0);
     grpc_iomgr_shutdown_background_closure();
     grpc_timer_manager_set_threading(false);  // shutdown timer_manager thread
-    grpc_resolver_dns_ares_shutdown();
     grpc_iomgr_shutdown();
+    grpc_resolver_dns_ares_shutdown();
   }
   g_shutting_down = false;
   g_shutting_down_cv->SignalAll();
