--- a/src/compiler/ruby_generator_string-inl.h	(index)
+++ b/src/compiler/ruby_generator_string-inl.h	(working tree)
@@ -36,9 +36,9 @@ namespace grpc_ruby_generator {
 inline std::vector<TString>& Split(const TString& s, char delim,
                                        std::vector<TString>* elems) {
   std::stringstream ss(s);
-  TString item;
+  std::string item;
   while (getline(ss, item, delim)) {
-    elems->push_back(item);
+    elems->emplace_back(item);
   }
   return *elems;
 }
