--- contrib/libs/grpc/include/grpcpp/impl/codegen/config_protobuf.h
+++ contrib/libs/grpc/include/grpcpp/impl/codegen/config_protobuf.h
@@ -68,8 +68,9 @@
 #ifndef GRPC_CUSTOM_JSONUTIL
 #include <google/protobuf/util/json_util.h>
 #include <google/protobuf/util/type_resolver_util.h>
+#include <y_absl/status/status.h>
 #define GRPC_CUSTOM_JSONUTIL ::google::protobuf::util
-#define GRPC_CUSTOM_UTIL_STATUS ::google::protobuf::util::Status
+#define GRPC_CUSTOM_UTIL_STATUS y_absl::Status
 #endif
