--- a/ogr/ogrct.cpp
+++ b/ogr/ogrct.cpp
@@ -1478,7 +1478,7 @@ int OGRProjCT::Initialize(const OGRSpatialReference *poSourceIn,
 
     // To easy quick&dirty compatibility with GDAL < 3.0
     if (CPLTestBool(
-            CPLGetConfigOption("OGR_CT_FORCE_TRADITIONAL_GIS_ORDER", "NO")))
+            CPLGetConfigOption("OGR_CT_FORCE_TRADITIONAL_GIS_ORDER", "YES")))
     {
         if (poSRSSource)
             poSRSSource->SetAxisMappingStrategy(OAMS_TRADITIONAL_GIS_ORDER);
