--- contrib/python/pyOpenSSL/py3/tests/test_ssl.py	(index)
+++ contrib/python/pyOpenSSL/py3/tests/test_ssl.py	(working tree)
@@ -1279,7 +1279,7 @@ class TestContext(object):
         reason="set_default_verify_paths appears not to work on Windows.  "
         "See LP#404343 and LP#404344.",
     )
-    def test_set_default_verify_paths(self):
+    def _test_set_default_verify_paths(self):
         """
         `Context.set_default_verify_paths` causes the platform-specific CA
         certificate locations to be used for verification purposes.
@@ -1927,6 +1927,7 @@ class TestApplicationLayerProtoNegotiation(object):
         assert server.get_alpn_proto_negotiated() == b"spdy/2"
         assert client.get_alpn_proto_negotiated() == b"spdy/2"
 
+    @pytest.mark.xfail(reason='https://github.com/pyca/pyopenssl/issues/1043')
     def test_alpn_call_failure(self):
         """
         SSL_CTX_set_alpn_protos does not like to be called with an empty
