--- contrib/python/django/django-2.2/django/utils/autoreload.py	(index)
+++ contrib/python/django/django-2.2/django/utils/autoreload.py	(working tree)
@@ -203,7 +203,7 @@ def get_child_arguments():
     """
     import django.__main__
 
-    args = [sys.executable] + ['-W%s' % o for o in sys.warnoptions]
+    args = []
     if sys.argv[0] == django.__main__.__file__:
         # The server was started with `python -m django runserver`.
         args += ['-m', 'django']
