--- contrib/python/django/django-2.2/django/core/mail/backends/smtp.py	(index)
+++ contrib/python/django/django-2.2/django/core/mail/backends/smtp.py	(working tree)
@@ -8,6 +8,7 @@ from django.conf import settings
 from django.core.mail.backends.base import BaseEmailBackend
 from django.core.mail.message import sanitize_address
 from django.core.mail.utils import DNS_NAME
+from django.utils.functional import cached_property
 
 
 class EmailBackend(BaseEmailBackend):
@@ -39,6 +40,15 @@ class EmailBackend(BaseEmailBackend):
     def connection_class(self):
         return smtplib.SMTP_SSL if self.use_ssl else smtplib.SMTP
 
+    @cached_property
+    def ssl_context(self):
+        ssl_context = ssl.SSLContext(protocol=ssl.PROTOCOL_TLS_CLIENT)
+        ssl_context.check_hostname = False
+        ssl_context.verify_mode = ssl.CERT_NONE
+        if self.ssl_certfile or self.ssl_keyfile:
+            ssl_context.load_cert_chain(self.ssl_certfile, self.ssl_keyfile)
+        return ssl_context
+
     def open(self):
         """
         Ensure an open connection to the email server. Return whether or not a
@@ -55,17 +65,14 @@ class EmailBackend(BaseEmailBackend):
         if self.timeout is not None:
             connection_params['timeout'] = self.timeout
         if self.use_ssl:
-            connection_params.update({
-                'keyfile': self.ssl_keyfile,
-                'certfile': self.ssl_certfile,
-            })
+            connection_params["context"] = self.ssl_context
         try:
             self.connection = self.connection_class(self.host, self.port, **connection_params)
 
             # TLS/SSL are mutually exclusive, so only attempt TLS over
             # non-secure connections.
             if not self.use_ssl and self.use_tls:
-                self.connection.starttls(keyfile=self.ssl_keyfile, certfile=self.ssl_certfile)
+                self.connection.starttls(context=self.ssl_context)
             if self.username and self.password:
                 self.connection.login(self.username, self.password)
             return True
