--- contrib/deprecated/python/billiard/billiard/einfo.py	(index)
+++ contrib/deprecated/python/billiard/billiard/einfo.py	(working tree)
@@ -24,6 +24,13 @@ class _Code(object):
         self.co_nlocals = code.co_nlocals
         self.co_stacksize = code.co_stacksize
         self.co_varnames = ()
+        if sys.version_info >= (3, 11):
+            self._co_positions = list(code.co_positions())
+
+    if sys.version_info >= (3, 11):
+        @property
+        def co_positions(self):
+            return self._co_positions.__iter__
 
 
 class _Frame(object):
