--- contrib/python/starlette-admin/starlette_admin/base.py	(index)
+++ contrib/python/starlette-admin/starlette_admin/base.py	(working tree)
@@ -1,3 +1,4 @@
+import os
 import json
 from json import JSONDecodeError
 from typing import Any, Awaitable, Callable, Dict, List, Optional, Sequence, Type, Union
@@ -138,7 +139,7 @@ class BaseAdmin:
             self.auth_provider.setup_admin(self)
 
     def init_routes(self) -> None:
-        statics = StaticFiles(directory=self.statics_dir, packages=["starlette_admin"])
+        statics = StaticFiles(directory=os.path.join(os.path.dirname(__file__), self.statics_dir))
         self.routes.extend(
             [
                 Mount("/statics", app=statics, name="statics"),
