--- contrib/python/marshmallow/py2/marshmallow/marshalling.py	(index)
+++ contrib/python/marshmallow/py2/marshmallow/marshalling.py	(working tree)
@@ -248,7 +248,7 @@ class Unmarshaller(ErrorStore):
 
         ret = dict_class()
 
-        if not isinstance(data, Mapping):
+        if not isinstance(data, Mapping) and not hasattr(data, "get"):
             errors = self.get_errors(index=index)
             msg = 'Invalid input type.'
             self.error_field_names = [SCHEMA]
--- contrib/python/marshmallow/py2/marshmallow/validate.py	(index)
+++ contrib/python/marshmallow/py2/marshmallow/validate.py	(working tree)
@@ -18,6 +18,8 @@ class Validator(object):
         add a useful `__repr__` implementation for validators.
     """
 
+    error = None  # type: typing.Optional[str]
+
     def __repr__(self):
         args = self._repr_args()
         args = '{0}, '.format(args) if args else ''
--- contrib/python/marshmallow/py2/tests/conftest.py	(index)
+++ contrib/python/marshmallow/py2/tests/conftest.py	(working tree)
@@ -2,7 +2,7 @@
 """Pytest fixtures that are available in all test modules."""
 import pytest
 
-from tests.base import User, UserSchema, Blog
+from .base import User, UserSchema, Blog
 
 @pytest.fixture
 def user():
--- contrib/python/marshmallow/py2/tests/test_fields.py	(index)
+++ contrib/python/marshmallow/py2/tests/test_fields.py	(working tree)
@@ -4,7 +4,7 @@ import pytest
 from marshmallow import fields, Schema, ValidationError
 from marshmallow.marshalling import missing
 
-from tests.base import ALL_FIELDS, User
+from .base import ALL_FIELDS, User
 
 class TestFieldAliases:
 
--- contrib/python/marshmallow/py2/tests/test_options.py	(index)
+++ contrib/python/marshmallow/py2/tests/test_options.py	(working tree)
@@ -5,7 +5,7 @@ from marshmallow import fields, Schema, SchemaOpts
 from marshmallow.exceptions import ValidationError
 from marshmallow.compat import OrderedDict
 
-from tests.base import *  # noqa
+from .base import *  # noqa
 
 class TestStrict:
 
--- contrib/python/marshmallow/py2/tests/test_deserialization.py	(index)
+++ contrib/python/marshmallow/py2/tests/test_deserialization.py	(working tree)
@@ -10,7 +10,7 @@ from marshmallow.exceptions import ValidationError
 from marshmallow.compat import basestring
 from marshmallow.validate import Equal
 
-from tests.base import (
+from .base import (
     assert_almost_equal,
     assert_date_equal,
     assert_datetime_equal,
--- contrib/python/marshmallow/py2/tests/test_marshalling.py	(index)
+++ contrib/python/marshmallow/py2/tests/test_marshalling.py	(working tree)
@@ -6,7 +6,7 @@ from marshmallow import fields, Schema, validates
 from marshmallow.marshalling import Marshaller, Unmarshaller, missing
 from marshmallow.exceptions import ValidationError
 
-from tests.base import User
+from .base import User
 
 def test_missing_is_falsy():
     assert bool(missing) is False
--- contrib/python/marshmallow/py2/tests/test_registry.py	(index)
+++ contrib/python/marshmallow/py2/tests/test_registry.py	(working tree)
@@ -17,8 +17,8 @@ def test_serializer_has_class_registry():
     assert 'MySubSchema' in class_registry._registry
 
     # by fullpath
-    assert 'tests.test_registry.MySchema' in class_registry._registry
-    assert 'tests.test_registry.MySubSchema' in class_registry._registry
+    assert '__tests__.test_registry.MySchema' in class_registry._registry
+    assert '__tests__.test_registry.MySubSchema' in class_registry._registry
 
 
 def test_serializer_class_registry_register_same_classname_different_module():
@@ -98,15 +98,15 @@ class C:
 
 class ASchema(Schema):
     id = fields.Integer()
-    b = fields.Nested('BSchema', exclude=('a', ))
+    b = fields.Nested('__tests__.test_registry.BSchema', exclude=('a', ))
 
 class BSchema(Schema):
     id = fields.Integer()
-    a = fields.Nested('ASchema')
+    a = fields.Nested('__tests__.test_registry.ASchema')
 
 class CSchema(Schema):
     id = fields.Integer()
-    bs = fields.Nested('BSchema', many=True)
+    bs = fields.Nested('__tests__.test_registry.BSchema', many=True)
 
 def test_two_way_nesting():
     a_obj = A(1)
@@ -167,7 +167,7 @@ def test_can_use_full_module_path_to_class():
     # Using full paths is ok
 
     class Schema1(Schema):
-        foo = fields.Nested('tests.foo_serializer.FooSerializer')
+        foo = fields.Nested('__tests__.foo_serializer.FooSerializer')
 
     sch = Schema1()
 
@@ -176,6 +176,6 @@ def test_can_use_full_module_path_to_class():
     assert sch.dump({'foo': {'_id': 42}}).data
 
     class Schema2(Schema):
-        foo = fields.Nested('tests.test_registry.FooSerializer')
+        foo = fields.Nested('__tests__.test_registry.FooSerializer')
     sch = Schema2()
     assert sch.dump({'foo': {'_id': 42}}).data
--- contrib/python/marshmallow/py2/tests/test_schema.py	(index)
+++ contrib/python/marshmallow/py2/tests/test_schema.py	(working tree)
@@ -15,7 +15,7 @@ from marshmallow import (
 from marshmallow.exceptions import ValidationError
 from marshmallow.compat import OrderedDict
 
-from tests.base import *  # noqa
+from .base import *  # noqa
 
 
 random.seed(1)
--- contrib/python/marshmallow/py2/tests/test_serialization.py	(index)
+++ contrib/python/marshmallow/py2/tests/test_serialization.py	(working tree)
@@ -13,7 +13,7 @@ from marshmallow.exceptions import ValidationError
 from marshmallow.compat import basestring, OrderedDict
 from marshmallow.utils import missing as missing_
 
-from tests.base import User, ALL_FIELDS
+from .base import User, ALL_FIELDS
 
 class DateTimeList:
     def __init__(self, dtimes):
--- contrib/python/marshmallow/py2/tests/test_utils.py	(index)
+++ contrib/python/marshmallow/py2/tests/test_utils.py	(working tree)
@@ -8,7 +8,7 @@ import pytest
 
 from marshmallow import utils
 from marshmallow import warnings
-from tests.base import (
+from .base import (
     assert_datetime_equal,
     central,
     assert_time_equal,
