--- contrib/deprecated/python/pymongo/pymongo/periodic_executor.py	(b52eea6b8119dd6a3a549aa6fc10cc5c8401b59b)
+++ contrib/deprecated/python/pymongo/pymongo/periodic_executor.py	(0c0f01b4286dbb31db9d2c6c46ff2d49028e0f64)
@@ -14,6 +14,7 @@
 
 """Run a target function on a background thread."""
 
+import sys
 import threading
 import time
 import weakref
@@ -83,7 +84,15 @@ class PeriodicExecutor(object):
             thread.daemon = True
             self._thread = weakref.proxy(thread)
             _register_executor(self)
-            thread.start()
+            # Mitigation to RuntimeError firing when thread starts on shutdown
+            # https://github.com/python/cpython/issues/114570
+            try:
+                thread.start()
+            except RuntimeError as e:
+                if "interpreter shutdown" in str(e) or sys.is_finalizing():
+                    self._thread = None
+                    return
+                raise
 
     def close(self, dummy=None):
         """Stop. To restart, call open().
