--- contrib/python/textual-serve/textual_serve/server.py	(index)
+++ contrib/python/textual-serve/textual_serve/server.py	(working tree)
@@ -1,6 +1,7 @@
 from __future__ import annotations
 
 import asyncio
+import importlib
 
 import logging
 import os
@@ -100,9 +101,9 @@ class Server:
         else:
             self.public_url = public_url
 
-        base_path = (Path(__file__) / "../").resolve().absolute()
+        base_path = importlib.resources.files(__package__)
         self.statics_path = base_path / statics_path
-        self.templates_path = base_path / templates_path
+        self.templates_path = templates_path
         self.console = Console()
         self.download_manager = DownloadManager()
 
@@ -140,7 +141,7 @@ class Server:
         """
         app = web.Application()
 
-        aiohttp_jinja2.setup(app, loader=jinja2.FileSystemLoader(self.templates_path))
+        aiohttp_jinja2.setup(app, loader=jinja2.PackageLoader(__package__, self.templates_path))
 
         ROUTES = [
             web.get("/", self.handle_index, name="index"),
