--- contrib/python/flup/py3/flup/client/fcgi_app.py	(index)
+++ contrib/python/flup/py3/flup/client/fcgi_app.py	(working tree)
@@ -32,6 +32,7 @@ import struct
 import socket
 import errno
 import types
+import sys
 
 __all__ = ['FCGIApp']
 
@@ -387,7 +388,11 @@ class FCGIApp(object):
         if self._connect is not None:
             # The simple case. Create a socket and connect to the
             # application.
-            if isinstance(self._connect, types.StringTypes):
+            try:
+                is_string = isinstance(self._connect, types.StringTypes)
+            except AttributeError:
+                is_string = isinstance(self._connect, str)
+            if is_string:
                 sock = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
                 sock.connect(self._connect)
             elif hasattr(socket, 'create_connection'):
--- contrib/python/flup/py3/flup/client/scgi_app.py	(index)
+++ contrib/python/flup/py3/flup/client/scgi_app.py	(working tree)
@@ -137,7 +137,11 @@ class SCGIApp(object):
         return [result]
 
     def _getConnection(self):
-        if isinstance(self._connect, types.StringTypes):
+        try:
+            is_string = isinstance(self._connect, types.StringTypes)
+        except AttributeError:
+            is_string = isinstance(self._connect, str)
+        if is_string:
             sock = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
             sock.connect(self._connect)
         elif hasattr(socket, 'create_connection'):
--- contrib/python/flup/py3/flup/server/ajp_base.py	(index)
+++ contrib/python/flup/py3/flup/server/ajp_base.py	(working tree)
@@ -39,7 +39,10 @@ import time
 import traceback
 
 # Unfortunately, for now, threads are required.
-import _thread
+try:
+    import thread as _thread
+except ImportError:
+    import _thread
 import threading
 
 from flup.server import NoDefault
@@ -433,6 +436,8 @@ class InputStream(object):
             raise StopIteration
         return r
 
+    next = __next__
+
     def setDataLength(self, length):
         """
         Once Content-Length is known, Request calls this method to set it.
--- contrib/python/flup/py3/flup/server/fcgi_base.py	(index)
+++ contrib/python/flup/py3/flup/server/fcgi_base.py	(working tree)
@@ -38,7 +38,10 @@ import errno
 import traceback
 
 try:
-    import _thread
+    try:
+        import thread as _thread
+    except ImportError:
+        import _thread
     import threading
     thread_available = True
 except ImportError:
@@ -224,6 +227,8 @@ class InputStream(object):
             raise StopIteration
         return r
 
+    next = __next__
+
     def add_data(self, data):
         if not data:
             self._eof = True
@@ -306,7 +311,7 @@ class OutputStream(object):
 
         if not data:
             return
-        if type(data) == str:
+        if sys.version_info[0] != 2 and type(data) == str:
             data = data.encode()
 
         self.dataWritten = True
@@ -369,7 +374,10 @@ class StdoutWrapper(object):
     def write(self, data):
         if data:
             self.dataWritten = True
-        self._file.buffer.write(data)
+        if hasattr(self._file, 'buffer'):
+            self._file.buffer.write(data)
+        else:
+            self._file.write(data)
 
     def writelines(self, lines):
         for line in lines:
@@ -385,14 +393,20 @@ def decode_pair(s, pos=0):
     The number of bytes decoded as well as the name/value pair
     are returned.
     """
-    nameLength = s[pos]
+    if sys.version_info[0] == 2:
+        nameLength = ord(s[pos])
+    else:
+        nameLength = s[pos]
     if nameLength & 128:
         nameLength = struct.unpack('!L', s[pos:pos+4])[0] & 0x7fffffff
         pos += 4
     else:
         pos += 1
 
-    valueLength = s[pos]
+    if sys.version_info[0] == 2:
+        valueLength = ord(s[pos])
+    else:
+        valueLength = s[pos]
     if valueLength & 128:
         valueLength = struct.unpack('!L', s[pos:pos+4])[0] & 0x7fffffff
         pos += 4
@@ -414,13 +428,19 @@ def encode_pair(name, value):
     """
     nameLength = len(name)
     if nameLength < 128:
-        s = bytes([nameLength])
+        if sys.version_info[0] == 2:
+            s = chr(nameLength)
+        else:
+            s = bytes([nameLength])
     else:
         s = struct.pack('!L', nameLength | 0x80000000)
 
     valueLength = len(value)
     if valueLength < 128:
-        s += bytes([valueLength])
+        if sys.version_info[0] == 2:
+            s += chr(valueLength)
+        else:
+            s += bytes([valueLength])
     else:
         s += struct.pack('!L', valueLength | 0x80000000)
 
@@ -755,7 +775,10 @@ class Connection(object):
             pos, (name, value) = decode_pair(inrec.contentData, pos)
             cap = self.server.capability.get(name)
             if cap is not None:
-                outrec.contentData += encode_pair(name, str(cap).encode('latin-1'))
+                if sys.version_info[0] == 2:
+                    outrec.contentData += encode_pair(name, str(cap))
+                else:
+                    outrec.contentData += encode_pair(name, str(cap).encode('latin-1'))
 
         outrec.contentLength = len(outrec.contentData)
         self.writeRecord(outrec)
@@ -802,7 +825,10 @@ class Connection(object):
                 pos = 0
                 while pos < inrec.contentLength:
                     pos, (name, value) = decode_pair(inrec.contentData, pos)
-                    req.params[name.decode('latin-1')] = value.decode('latin-1')
+                    if sys.version_info[0] == 2:
+                        req.params[name] = value
+                    else:
+                        req.params[name.decode('latin-1')] = value.decode('latin-1')
             else:
                 self._start_request(req)
 
@@ -1105,7 +1131,7 @@ class BaseFCGIServer(object):
         result = None
 
         def write(data):
-            if type(data) is str:
+            if sys.version_info[0] != 2 and type(data) is str:
                 data = data.encode('latin-1')
 
             assert type(data) is bytes, 'write() argument must be bytes'
@@ -1121,8 +1147,12 @@ class BaseFCGIServer(object):
                 if not found and result is not None:
                     try:
                         if len(result) == 1:
-                            responseHeaders.append((b'Content-Length',
-                                                    str(len(data)).encode('latin-1')))
+                            if sys.version_info[0] ==2:
+                                responseHeaders.append((b'Content-Length',
+                                                        str(len(data))))
+                            else:
+                                responseHeaders.append((b'Content-Length',
+                                                        str(len(data)).encode('latin-1')))
                     except:
                         pass
                 s = b'Status: ' + status + b'\r\n'
@@ -1145,19 +1175,22 @@ class BaseFCGIServer(object):
             else:
                 assert not headers_set, 'Headers already set!'
 
-            if type(status) is str:
+            if sys.version_info[0] != 2 and type(status) is str:
                 status = status.encode('latin-1')
 
             assert type(status) is bytes, 'Status must be a string'
             assert len(status) >= 4, 'Status must be at least 4 characters'
             assert int(status[:3]), 'Status must begin with 3-digit code'
-            assert status[3] == 0x20, 'Status must have a space after code'
+            if sys.version_info[0] == 2:
+                assert status[3] == ' ', 'Status must have a space after code'
+            else:
+                assert status[3] == 0x20, 'Status must have a space after code'
             assert type(response_headers) is list, 'Headers must be a list'
             new_response_headers = []
             for name,val in response_headers:
-                if type(name) is str:
+                if sys.version_info[0] != 2 and type(name) is str:
                     name = name.encode('latin-1')
-                if type(val) is str:
+                if sys.version_info[0] != 2 and type(val) is str:
                     val = val.encode('latin-1')
 
                 assert type(name) is bytes, 'Header name "%s" must be bytes' % name
--- contrib/python/flup/py3/flup/server/scgi_base.py	(index)
+++ contrib/python/flup/py3/flup/server/scgi_base.py	(working tree)
@@ -41,7 +41,10 @@ import traceback
 
 # Threads are required. If you want a non-threaded (forking) version, look at
 # SWAP <http://www.idyll.org/~t/www-tools/wsgi/>.
-import _thread
+try:
+    import thread as _thread
+except ImportError:
+    import _thread
 import threading
 
 __all__ = ['BaseSCGIServer']
