--- contrib/python/plotly/py3/plotly/validator_cache.py	(index)
+++ contrib/python/plotly/py3/plotly/validator_cache.py	(working tree)
@@ -16,15 +16,10 @@ class ValidatorCache(object):
     @staticmethod
     def get_validator(parent_path, prop_name):
         if ValidatorCache._json_cache is None:
+            import importlib.resources
             # Load the JSON validator params from the file
-            validator_json_path = opath.join(
-                opath.dirname(__file__), "validators", "_validators.json"
-            )
-            if not opath.exists(validator_json_path):
-                raise FileNotFoundError(
-                    f"Validator JSON file not found: {validator_json_path}"
-                )
-            with open(validator_json_path, "r") as f:
+            validator_json_path = importlib.resources.files(__package__).joinpath("validators", "_validators.json")
+            with validator_json_path.open("r") as f:
                 ValidatorCache._json_cache = json.load(f)
 
         key = (parent_path, prop_name)
