--- contrib/python/Wand/wand/api.py	(index)
+++ contrib/python/Wand/wand/api.py	(working tree)
@@ -36,6 +36,7 @@ def library_paths():
     :rtype: :class:`tuple`
 
     """
+    yield 'ImageMagick', 'ImageMagick'
     libwand = None
     libmagick = None
     versions = '', '-7', '-7.Q8', '-7.Q16', '-6', '-Q16', '-Q8', '-6.Q16'
@@ -131,7 +132,7 @@ def load_library():
             continue
         try:
             tried_paths.append(libwand_path)
-            libwand = ctypes.CDLL(str(libwand_path))
+            libwand = ctypes.CDLL(ctypes.util.find_library(libwand_path))
             if libwand_path == libmagick_path:
                 libmagick = libwand
             else:
