--- a/lib/obstack.c	(index)
+++ b/lib/obstack.c	(working tree)
@@ -51,10 +51,6 @@
 /* If GCC, or if an oddball (testing?) host that #defines __alignof__,
    use the already-supplied __alignof__.  Otherwise, this must be Gnulib
    (as glibc assumes GCC); defer to Gnulib's alignof_type.  */
-# if !defined __GNUC__ && !defined __alignof__
-#  include <alignof.h>
-#  define __alignof__(type) alignof_type (type)
-# endif
 # include <stdlib.h>
 # include <stdint.h>
 
