--- contrib/python/graphene-sqlalchemy/py3/graphene_sqlalchemy/tests/conftest.py	(index)
+++ contrib/python/graphene-sqlalchemy/py3/graphene_sqlalchemy/tests/conftest.py	(working tree)
@@ -4,2 +4,2 @@ from sqlalchemy.orm import sessionmaker
-from ..converter import convert_sqlalchemy_composite
-from ..registry import reset_global_registry
+from graphene_sqlalchemy.converter import convert_sqlalchemy_composite
+from graphene_sqlalchemy.registry import reset_global_registry
--- contrib/python/graphene-sqlalchemy/py3/graphene_sqlalchemy/tests/test_batching.py	(index)
+++ contrib/python/graphene-sqlalchemy/py3/graphene_sqlalchemy/tests/test_batching.py	(working tree)
@@ -7,4 +7,4 @@ import pytest
-from ..fields import (BatchSQLAlchemyConnectionField,
-                      default_connection_field_factory)
-from ..types import ORMField, SQLAlchemyObjectType
-from ..utils import is_sqlalchemy_version_less_than
+from graphene_sqlalchemy.fields import (BatchSQLAlchemyConnectionField,
+                      default_connection_field_factory)
+from graphene_sqlalchemy.types import ORMField, SQLAlchemyObjectType
+from graphene_sqlalchemy.utils import is_sqlalchemy_version_less_than
--- contrib/python/graphene-sqlalchemy/py3/graphene_sqlalchemy/tests/test_benchmark.py	(index)
+++ contrib/python/graphene-sqlalchemy/py3/graphene_sqlalchemy/tests/test_benchmark.py	(working tree)
@@ -3,2 +3,2 @@ import pytest
-from ..types import SQLAlchemyObjectType
-from ..utils import is_sqlalchemy_version_less_than
+from graphene_sqlalchemy.types import SQLAlchemyObjectType
+from graphene_sqlalchemy.utils import is_sqlalchemy_version_less_than
--- contrib/python/graphene-sqlalchemy/py3/graphene_sqlalchemy/tests/test_converter.py	(index)
+++ contrib/python/graphene-sqlalchemy/py3/graphene_sqlalchemy/tests/test_converter.py	(working tree)
@@ -15,8 +15,8 @@ import graphene
-from ..converter import (convert_sqlalchemy_column,
-                         convert_sqlalchemy_composite,
-                         convert_sqlalchemy_hybrid_method,
-                         convert_sqlalchemy_relationship)
-from ..fields import (UnsortedSQLAlchemyConnectionField,
-                      default_connection_field_factory)
-from ..registry import Registry, get_global_registry
-from ..types import ORMField, SQLAlchemyObjectType
+from graphene_sqlalchemy.converter import (convert_sqlalchemy_column,
+                         convert_sqlalchemy_composite,
+                         convert_sqlalchemy_hybrid_method,
+                         convert_sqlalchemy_relationship)
+from graphene_sqlalchemy.fields import (UnsortedSQLAlchemyConnectionField,
+                      default_connection_field_factory)
+from graphene_sqlalchemy.registry import Registry, get_global_registry
+from graphene_sqlalchemy.types import ORMField, SQLAlchemyObjectType
--- contrib/python/graphene-sqlalchemy/py3/graphene_sqlalchemy/tests/test_enums.py	(index)
+++ contrib/python/graphene-sqlalchemy/py3/graphene_sqlalchemy/tests/test_enums.py	(working tree)
@@ -5,2 +5,2 @@ from sqlalchemy.types import Enum as SQLAlchemyEnumType
-from ..enums import _convert_sa_to_graphene_enum, enum_for_field
-from ..types import SQLAlchemyObjectType
+from graphene_sqlalchemy.enums import _convert_sa_to_graphene_enum, enum_for_field
+from graphene_sqlalchemy.types import SQLAlchemyObjectType
--- contrib/python/graphene-sqlalchemy/py3/graphene_sqlalchemy/tests/test_fields.py	(index)
+++ contrib/python/graphene-sqlalchemy/py3/graphene_sqlalchemy/tests/test_fields.py	(working tree)
@@ -4,3 +4,3 @@ from promise import Promise
-from ..fields import (SQLAlchemyConnectionField,
-                      UnsortedSQLAlchemyConnectionField)
-from ..types import SQLAlchemyObjectType
+from graphene_sqlalchemy.fields import (SQLAlchemyConnectionField,
+                      UnsortedSQLAlchemyConnectionField)
+from graphene_sqlalchemy.types import SQLAlchemyObjectType
--- contrib/python/graphene-sqlalchemy/py3/graphene_sqlalchemy/tests/test_query.py	(index)
+++ contrib/python/graphene-sqlalchemy/py3/graphene_sqlalchemy/tests/test_query.py	(working tree)
@@ -1,3 +1,3 @@
-from ..converter import convert_sqlalchemy_composite
-from ..fields import SQLAlchemyConnectionField
-from ..types import ORMField, SQLAlchemyObjectType
+from graphene_sqlalchemy.converter import convert_sqlalchemy_composite
+from graphene_sqlalchemy.fields import SQLAlchemyConnectionField
+from graphene_sqlalchemy.types import ORMField, SQLAlchemyObjectType
--- contrib/python/graphene-sqlalchemy/py3/graphene_sqlalchemy/tests/test_query_enums.py	(index)
+++ contrib/python/graphene-sqlalchemy/py3/graphene_sqlalchemy/tests/test_query_enums.py	(working tree)
@@ -1,1 +1,1 @@
-from ..types import SQLAlchemyObjectType
+from graphene_sqlalchemy.types import SQLAlchemyObjectType
--- contrib/python/graphene-sqlalchemy/py3/graphene_sqlalchemy/tests/test_reflected.py	(index)
+++ contrib/python/graphene-sqlalchemy/py3/graphene_sqlalchemy/tests/test_reflected.py	(working tree)
@@ -1,2 +1,2 @@
-from ..registry import Registry
-from ..types import SQLAlchemyObjectType
+from graphene_sqlalchemy.registry import Registry
+from graphene_sqlalchemy.types import SQLAlchemyObjectType
--- contrib/python/graphene-sqlalchemy/py3/graphene_sqlalchemy/tests/test_registry.py	(index)
+++ contrib/python/graphene-sqlalchemy/py3/graphene_sqlalchemy/tests/test_registry.py	(working tree)
@@ -4,3 +4,3 @@ from sqlalchemy.types import Enum as SQLAlchemyEnum
-from ..registry import Registry
-from ..types import SQLAlchemyObjectType
-from ..utils import EnumValue
+from graphene_sqlalchemy.registry import Registry
+from graphene_sqlalchemy.types import SQLAlchemyObjectType
+from graphene_sqlalchemy.utils import EnumValue
--- contrib/python/graphene-sqlalchemy/py3/graphene_sqlalchemy/tests/test_sort_enums.py	(index)
+++ contrib/python/graphene-sqlalchemy/py3/graphene_sqlalchemy/tests/test_sort_enums.py	(working tree)
@@ -4,3 +4,3 @@ import sqlalchemy as sa
-from ..fields import SQLAlchemyConnectionField
-from ..types import SQLAlchemyObjectType
-from ..utils import to_type_name
+from graphene_sqlalchemy.fields import SQLAlchemyConnectionField
+from graphene_sqlalchemy.types import SQLAlchemyObjectType
+from graphene_sqlalchemy.utils import to_type_name
--- contrib/python/graphene-sqlalchemy/py3/graphene_sqlalchemy/tests/test_types.py	(index)
+++ contrib/python/graphene-sqlalchemy/py3/graphene_sqlalchemy/tests/test_types.py	(working tree)
@@ -8,7 +8,7 @@ from graphene import (Boolean, Dynamic, Field, Float, GlobalID, Int, List,
-from .. import utils
-from ..converter import convert_sqlalchemy_composite
-from ..fields import (SQLAlchemyConnectionField,
-                      UnsortedSQLAlchemyConnectionField, createConnectionField,
-                      registerConnectionFieldFactory,
-                      unregisterConnectionFieldFactory)
-from ..types import ORMField, SQLAlchemyObjectType, SQLAlchemyObjectTypeOptions
+from graphene_sqlalchemy import utils
+from graphene_sqlalchemy.converter import convert_sqlalchemy_composite
+from graphene_sqlalchemy.fields import (SQLAlchemyConnectionField,
+                      UnsortedSQLAlchemyConnectionField, createConnectionField,
+                      registerConnectionFieldFactory,
+                      unregisterConnectionFieldFactory)
+from graphene_sqlalchemy.types import ORMField, SQLAlchemyObjectType, SQLAlchemyObjectTypeOptions
--- contrib/python/graphene-sqlalchemy/py3/graphene_sqlalchemy/tests/test_utils.py	(index)
+++ contrib/python/graphene-sqlalchemy/py3/graphene_sqlalchemy/tests/test_utils.py	(working tree)
@@ -3,1 +3,1 @@ import sqlalchemy as sa
-from ..utils import (DummyImport, get_session, sort_argument_for_model,
+from graphene_sqlalchemy.utils import (DummyImport, get_session, sort_argument_for_model,
--- contrib/python/graphene-sqlalchemy/py3/graphene_sqlalchemy/utils.py	(index)
+++ contrib/python/graphene-sqlalchemy/py3/graphene_sqlalchemy/utils.py	(working tree)
@@ -7,7 +7,7 @@ import pkg_resources
 from sqlalchemy.exc import ArgumentError
 from sqlalchemy.orm import class_mapper, object_mapper
 from sqlalchemy.orm.exc import UnmappedClassError, UnmappedInstanceError
-
+from graphene.utils.str_converters import to_camel_case, to_snake_case
 
 def get_session(context):
     return context.get("session")
@@ -26,6 +26,46 @@ def get_query(model, context):
     return query
 
 
+_operator_aliases = {
+    "==": lambda x, y: x == y,
+    "=": lambda x, y: x == y,
+    ">=": lambda x, y: x >= y,
+    ">": lambda x, y: x > y,
+    "<=": lambda x, y: x <= y,
+    "<": lambda x, y: x < y,
+    "!=": lambda x, y: x != y
+}
+
+
+def get_operator_function(operator=None):
+    if not operator:
+        def operator(x, y):
+            return x == y
+
+    if not callable(operator):
+        operator_attr = operator
+
+        if operator in _operator_aliases:
+            operator = _operator_aliases[operator_attr]
+        else:
+            def operator(x, y):
+                return getattr(x, operator_attr)(y)
+
+    return operator
+
+
+def get_snake_or_camel_attr(model, attr):
+    try:
+        return getattr(model, to_snake_case(attr))
+    except Exception:
+        pass
+    try:
+        return getattr(model, to_camel_case(attr))
+    except Exception:
+        pass
+    return getattr(model, attr)
+
+
 def is_mapped_class(cls):
     try:
         class_mapper(cls)
\ No newline at end of file
