diff --git a/libraw/libraw_types.h b/libraw/libraw_types.h
index 230c877..62b070e 100644
--- a/libraw/libraw_types.h
+++ b/libraw/libraw_types.h
@@ -99,13 +98,1 @@ typedef unsigned long long UINT64;
-#ifdef LIBRAW_WIN32_DLLDEFS
-#ifdef LIBRAW_NODLL
-#define DllDef
-#else
-#ifdef LIBRAW_BUILDLIB
-#define DllDef __declspec(dllexport)
-#else
-#define DllDef __declspec(dllimport)
-#endif
-#endif
-#else
-#define DllDef
-#endif
+#define DllDef
