--- a/src/metadata/tiff.cpp
+++ b/src/metadata/tiff.cpp
@@ -1407,6 +1407,16 @@ int CLASS parse_tiff_ifd(int base)
       break;
     case 0xc640: // 50752
       read_shorts(cr2_slice, 3);
+      if (cr2_slice[0] && (!cr2_slice[1] || !cr2_slice[2])) {
+        /*
+          This tag used in segmented Canon CR2 images. It contains 3 numbers:
+            1. Number of segments minus one;
+            2. Pixel width of segments except last;
+            3. Pixel width of last segmen.
+          Width of segments can't be equal to zero.
+        */
+        throw LIBRAW_EXCEPTION_DECODE_RAW;
+      }
       break;
     case 0xc68b: /* 50827, OriginalRawFileName */
       stmread(imgdata.color.OriginalRawFileName, len, ifp);
