--- contrib/python/pydantic-core/ya.make	(index)
+++ contrib/python/pydantic-core/ya.make	(working tree)
@@ -14,6 +14,33 @@ PEERDIR(

 NO_LINT()

+IF (OS_DARWIN AND ARCH_ARM64)
+    COPY_FILE(a/aarch64-apple-darwin/release/lib_pydantic_core.a lib_pydantic_core.a)
+ELSEIF (OS_DARWIN AND ARCH_X86_64)
+    COPY_FILE(a/x86_64-apple-darwin/release/lib_pydantic_core.a lib_pydantic_core.a)
+ELSEIF (OS_LINUX AND ARCH_AARCH64)
+    COPY_FILE(a/aarch64-unknown-linux-gnu/release/lib_pydantic_core.a lib_pydantic_core.a)
+ELSEIF (OS_LINUX AND ARCH_X86_64 AND MUSL)
+    COPY_FILE(a/x86_64-unknown-linux-musl/release/lib_pydantic_core.a lib_pydantic_core.a)
+ELSEIF (OS_LINUX AND ARCH_X86_64 AND NOT MUSL)
+    COPY_FILE(a/x86_64-unknown-linux-gnu/release/lib_pydantic_core.a lib_pydantic_core.a)
+ELSEIF (OS_WINDOWS AND ARCH_X86_64)
+    COPY_FILE(a/x86_64-pc-windows-msvc/release/lib_pydantic_core.lib lib_pydantic_core.lib)
+# ELSE()
+#     MESSAGE(FATAL_ERROR "We don't have lib_pydantic_core for this platform")
+ENDIF()
+
+IF (OS_DARWIN)
+    LDFLAGS(
+        -framework CoreFoundation
+        -framework Security
+    )
+ELSEIF (OS_WINDOWS)
+    LDFLAGS(
+        ntdll.lib
+    )
+ENDIF()
+
 PY_REGISTER(
     pydantic_core._pydantic_core
 )
