--- contrib/libs/nghttp2/lib/includes/nghttp2/nghttp2.h	(index)
+++ contrib/libs/nghttp2/lib/includes/nghttp2/nghttp2.h	(working tree)
@@ -54,7 +54,12 @@ extern "C" {
 
 #include <nghttp2/nghttp2ver.h>
 
+#if defined(_MSC_VER)
+#  include <basetsd.h>
+   typedef SSIZE_T ssize_t;
+#endif
+
+#if 1
-#ifdef NGHTTP2_STATICLIB
 #  define NGHTTP2_EXTERN
 #elif defined(WIN32) || (__has_declspec_attribute(dllexport) &&                \
                          __has_declspec_attribute(dllimport))
 
