--- contrib/python/pandas/py2/pandas/_libs/tslibs/timedeltas.pyx	(index)
+++ contrib/python/pandas/py2/pandas/_libs/tslibs/timedeltas.pyx	(working tree)
@@ -27,8 +27,6 @@ from pandas._libs.tslibs.util cimport (
     is_timedelta64_object, is_datetime64_object, is_integer_object,
     is_float_object, is_string_object)
 
-from pandas._libs.tslibs.ccalendar import DAY_SECONDS
-
 from pandas._libs.tslibs.np_datetime cimport (
     cmp_scalar, reverse_ops, td64_to_tdstruct, pandas_timedeltastruct)
 
@@ -262,10 +260,10 @@ cpdef inline object precision_from_unit(object unit):
         m = 1000000000L * 2629746
         p = 9
     elif unit == 'W':
-        m = 1000000000L * DAY_SECONDS * 7
+        m = 1000000000L * 86400 * 7
         p = 9
     elif unit == 'D' or unit == 'd':
-        m = 1000000000L * DAY_SECONDS
+        m = 1000000000L * 86400
         p = 9
     elif unit == 'h':
         m = 1000000000L * 3600
--- contrib/python/pandas/py2/pandas/_libs/reduction.pyx	(index)
+++ contrib/python/pandas/py2/pandas/_libs/reduction.pyx	(working tree)
@@ -28,6 +28,15 @@ cdef _get_result_array(object obj, Py_ssize_t size, Py_ssize_t cnt):
 
     return np.empty(size, dtype='O')
 
+# Accessing the data member of ndarray is deprecated, but we depend on it.
+cdef extern from *:
+    """
+    static void PyArray_SET_DATA(PyArrayObject *arr, void * data) {
+        arr->data = data;
+    }
+    """
+    void PyArray_SET_DATA(ndarray arr, void * data)
+
 
 cdef class Reducer:
     """
@@ -103,7 +112,7 @@ cdef class Reducer:
         arr = self.arr
         chunk = self.dummy
         dummy_buf = chunk.data
-        chunk.data = arr.data
+        PyArray_SET_DATA(chunk, arr.data)
         labels = self.labels
         has_labels = labels is not None
         has_index = self.index is not None
@@ -156,7 +165,7 @@ cdef class Reducer:
                     it = <flatiter>PyArray_IterNew(result)
 
                 PyArray_SETITEM(result, PyArray_ITER_DATA(it), res)
-                chunk.data = chunk.data + self.increment
+                PyArray_SET_DATA(chunk, chunk.data + self.increment)
                 PyArray_ITER_NEXT(it)
         except Exception, e:
             if hasattr(e, 'args'):
@@ -164,7 +173,7 @@ cdef class Reducer:
             raise
         finally:
             # so we don't free the wrong memory
-            chunk.data = dummy_buf
+            PyArray_SET_DATA(chunk, dummy_buf)
 
         if result.dtype == np.object_:
             result = maybe_convert_objects(result)
@@ -466,17 +475,17 @@ cdef class Slider:
         self.orig_len = self.buf.shape[0]
         self.orig_stride = self.buf.strides[0]
 
-        self.buf.data = self.values.data
+        PyArray_SET_DATA(self.buf, self.values.data)
         self.buf.strides[0] = self.stride
 
     cpdef advance(self, Py_ssize_t k):
-        self.buf.data = <char*>self.buf.data + self.stride * k
+        PyArray_SET_DATA(self.buf, <char*>self.buf.data + self.stride * k)
 
     cdef move(self, int start, int end):
         """
         For slicing
         """
-        self.buf.data = self.values.data + self.stride * start
+        PyArray_SET_DATA(self.buf, self.values.data + self.stride * start)
         self.buf.shape[0] = end - start
 
     cpdef set_length(self, Py_ssize_t length):
@@ -485,7 +494,7 @@ cdef class Slider:
     cpdef reset(self):
 
         self.buf.shape[0] = self.orig_len
-        self.buf.data = self.orig_data
+        PyArray_SET_DATA(self.buf, self.orig_data)
         self.buf.strides[0] = self.orig_stride
 
 
@@ -595,7 +604,7 @@ cdef class BlockSlider:
             arr = self.blocks[i]
 
             # axis=1 is the frame's axis=0
-            arr.data = self.base_ptrs[i] + arr.strides[1] * start
+            PyArray_SET_DATA(arr, self.base_ptrs[i] + arr.strides[1] * start)
             arr.shape[1] = end - start
 
         # move and set the index
@@ -613,7 +622,7 @@ cdef class BlockSlider:
             arr = self.blocks[i]
 
             # axis=1 is the frame's axis=0
-            arr.data = self.base_ptrs[i]
+            PyArray_SET_DATA(arr, self.base_ptrs[i])
             arr.shape[1] = 0
 
 
--- contrib/python/pandas/py2/pandas/_libs/window.pyx	(index)
+++ contrib/python/pandas/py2/pandas/_libs/window.pyx	(working tree)
@@ -13,6 +13,15 @@ from numpy cimport ndarray, int64_t, float64_t, float32_t
 cnp.import_array()
 
 
+# Accessing the data member of ndarray is deprecated, but we depend on it.
+cdef extern from *:
+    """
+    static void PyArray_SET_DATA(PyArrayObject *arr, void * data) {
+        arr->data = (char *)data;
+    }
+    """
+    void PyArray_SET_DATA(ndarray arr, void * data)
+
 cdef extern from "src/headers/cmath" namespace "std":
     bint isnan(float64_t) nogil
     bint notnan(float64_t) nogil
@@ -1645,12 +1654,12 @@ def roll_generic(object obj,
         oldbuf = <float64_t *>bufarr.data
         for i from (win - offset) <= i < (N - offset):
             buf = buf + 1
-            bufarr.data = <char *>buf
+            PyArray_SET_DATA(bufarr, buf)
             if counts[i] >= minp:
                 output[i] = func(bufarr, *args, **kwargs)
             else:
                 output[i] = NaN
-        bufarr.data = <char *>oldbuf
+        PyArray_SET_DATA(bufarr, oldbuf)
 
         # truncated windows at the end
         for i from int_max(N - offset, 0) <= i < N:
