--- contrib/python/scancode-toolkit/licensedcode/models.py	(index)
+++ contrib/python/scancode-toolkit/licensedcode/models.py	(working tree)
@@ -19,6 +19,7 @@ from os.path import abspath
 from os.path import dirname
 from os.path import exists
 from os.path import join
+import sys
 
 import attr
 import saneyaml
@@ -43,1 +44,1 @@ data file and one or more text files containing license or notice texts.
-data_dir = join(abspath(dirname(__file__)), 'data')
+data_dir = join(abspath(dirname(sys.executable)), 'data')
--- contrib/python/scancode-toolkit/licensedcode/spans.py	(index)
+++ contrib/python/scancode-toolkit/licensedcode/spans.py	(working tree)
@@ -28,1 +28,1 @@
-from collections import Set
+from collections.abc import Set
