--- contrib/python/django/django-1.11/django/db/backends/postgresql/utils.py	(index)
+++ contrib/python/django/django-1.11/django/db/backends/postgresql/utils.py	(working tree)
@@ -1,7 +1,10 @@
 from django.utils.timezone import utc
+import datetime
 
 
 def utc_tzinfo_factory(offset):
-    if offset != 0:
+    if isinstance(offset, int) and offset != 0:
+        raise AssertionError("database connection isn't set to UTC")
+    if isinstance(offset, datetime.timedelta) and offset != datetime.timedelta(0):
         raise AssertionError("database connection isn't set to UTC")
     return utc
