--- contrib/python/django/django-1.11/django/db/backends/mysql/operations.py	(index)
+++ contrib/python/django/django-1.11/django/db/backends/mysql/operations.py	(working tree)
@@ -115,10 +115,13 @@ class DatabaseOperations(BaseDatabaseOperations):
         return 'MATCH (%s) AGAINST (%%s IN BOOLEAN MODE)' % field_name
 
     def last_executed_query(self, cursor, sql, params):
-        # With MySQLdb, cursor objects have an (undocumented) "_last_executed"
+        # With MySQLdb, cursor objects have an (undocumented) "_executed"
         # attribute where the exact query sent to the database is saved.
         # See MySQLdb/cursors.py in the source distribution.
-        return force_text(getattr(cursor, '_last_executed', None), errors='replace')
+        query = getattr(cursor, '_executed', None)
+        if query is not None:
+            query = force_text(query, errors='replace')
+        return query
 
     def no_limit_value(self):
         # 2**64 - 1, as recommended by the MySQL documentation
