--- contrib/python/pytest-responsemock/tests/test_module.py	(index)
+++ contrib/python/pytest-responsemock/tests/test_module.py	(working tree)
@@ -1,3 +1,4 @@
+import pytest
 import requests
 
 
@@ -68,6 +69,7 @@ def test_status(response_mock):
         assert result.content == b'Bad'
 
 
+@pytest.mark.skip('Arcadia. Network access on distbuild is disabled')
 def test_bypass(response_mock):
 
     with response_mock('GET http://yandex.ru -> 500 :Nice', bypass=True):
