--- contrib/python/Flask-Humanize/flask_humanize/__init__.py	(index)
+++ contrib/python/Flask-Humanize/flask_humanize/__init__.py	(working tree)
@@ -16,15 +16,18 @@ import humanize
 from flask import current_app
 from werkzeug.datastructures import ImmutableDict
 
-from . import compat
+from .compat import (
+    text_type, string_types
+)
 
 
 __version__ = '0.3.0'
 
 
 def force_unicode(value):
-    if not isinstance(value, compat.text_type):
-        return value.decode('utf-8')
+    if isinstance(value, string_types):
+        if not isinstance(value, text_type):
+            return value.decode('utf-8')
     return value
 
 
--- contrib/python/Flask-Humanize/flask_humanize/compat.py	(index)
+++ contrib/python/Flask-Humanize/flask_humanize/compat.py	(working tree)
@@ -7,5 +7,7 @@ PY2 = sys.version_info[0] == 2
 
 if PY2:
     text_type = unicode
+    string_types = basestring,
 else:
     text_type = str
+    string_types = str,
