--- contrib/python/Flask-Humanize/tests/test_humanize.py	(index)
+++ contrib/python/Flask-Humanize/tests/test_humanize.py	(working tree)
@@ -100,7 +100,7 @@ class TestL10N:
     def test_fallback_to_default_if_no_translations_for_locale(self, client, h):
         @h.localeselector
         def get_locale():
-            return 'tlh'
+            return 'af'
 
         assert client.get(url_for('naturaltime')).data == b'now'
 
