--- a/test/MemoryManagerSuite.cpp
+++ b/test/MemoryManagerSuite.cpp
@@ -180,1 +180,1 @@ TEST(MemoryManagerCompletenessSuite, MallocAndFreeRequiredOnly) {
-TEST(MemoryManagerTestingSuite, DefaultMemoryManager) {
+TEST(MemoryManagerTestingSuite, DISABLED_DefaultMemoryManager) {
@@ -202,1 +202,1 @@ TEST(MemoryManagerTestingSuite, CompleteMemoryManager) {
-TEST(MemoryManagerTestingSuite, EmulateCalloc) {
+TEST(MemoryManagerTestingSuite, DISABLED_EmulateCalloc) {
