--- contrib/libs/librdkafka/src/nanopb/pb.h	(index)
+++ contrib/libs/librdkafka/src/nanopb/pb.h	(working tree)
@@ -79,7 +79,7 @@
  * define PB_SYSTEM_HEADER to the path of this file.
  */
 #ifdef PB_SYSTEM_HEADER
-#include PB_SYSTEM_HEADER
+#error PB_SYSTEM_HEADER
 #else
 #include <stdint.h>
 #include <stddef.h>
@@ -147,7 +147,7 @@ extern "C" {
  * field information in program memory. */
 #ifndef PB_PROGMEM
 #ifdef __AVR__
-#include <avr/pgmspace.h>
+#error <avr/pgmspace.h>
 #define PB_PROGMEM             PROGMEM
 #define PB_PROGMEM_READU32(x)  pgm_read_dword(&x)
 #else
