--- contrib/python/fallocate/py3/tests/test_fallocate.py	(index)
+++ contrib/python/fallocate/py3/tests/test_fallocate.py	(working tree)
@@ -5,13 +5,13 @@ import platform
 import tempfile
 from fallocate import fallocate
 
-def simple_fallocate_1kb_test():
+def test_simple_fallocate_1kb_test():
     with tempfile.NamedTemporaryFile() as ntf:
         assert os.path.getsize(ntf.name) == 0
         fallocate(ntf, 0, 1024)
         assert os.path.getsize(ntf.name) == 1024
 
-def simple_fallocate_1mb_test():
+def test_simple_fallocate_1mb_test():
     with tempfile.NamedTemporaryFile() as ntf:
         assert os.path.getsize(ntf.name) == 0
         fallocate(ntf, 0, 1024*1024)
@@ -19,7 +19,7 @@ def simple_fallocate_1mb_test():
 
 if platform.system() == "Linux":
     from fallocate import FALLOC_FL_KEEP_SIZE, FALLOC_FL_PUNCH_HOLE
-    def fallocate_punch_hole_test():
+    def test_fallocate_punch_hole_test():
         with tempfile.NamedTemporaryFile() as ntf:
             assert os.path.getsize(ntf.name) == 0
             ntf.write(b"Hello World")
@@ -30,7 +30,7 @@ if platform.system() == "Linux":
             ntf.seek(0)
             assert ntf.read() == b"Hello \x00\x00\x00\x00d"
 
-    def fallocate_collapse_size_test():
+    def test_fallocate_collapse_size_test():
         try:
             from fallocate import FALLOC_FL_COLLAPSE_SIZE
         except:
