--- contrib/python/requests/py3/requests/adapters.py	(index)
+++ contrib/python/requests/py3/requests/adapters.py	(working tree)
@@ -76,7 +76,8 @@ DEFAULT_POOL_TIMEOUT = None
 
 try:
     import ssl  # noqa: F401
-
+    import urllib3.util.ssl_
+    urllib3.util.ssl_.SSLContext = ssl.SSLContext
     _preloaded_ssl_context = create_urllib3_context()
     _preloaded_ssl_context.load_verify_locations(
         extract_zipped_paths(DEFAULT_CA_BUNDLE_PATH)
