--- contrib/python/requests/py3/.dist-info/METADATA	(index)
+++ contrib/python/requests/py3/.dist-info/METADATA	(working tree)
@@ -30,7 +30,7 @@ Classifier: Topic :: Software Development :: Libraries
 Requires-Python: >=3.7, <4
 Description-Content-Type: text/markdown
 License-File: LICENSE
-Requires-Dist: charset-normalizer (~=2.0.0)
+Requires-Dist: charset-normalizer (>=2.0.0)
 Requires-Dist: idna (<4,>=2.5)
 Requires-Dist: urllib3 (<1.27,>=1.21.1)
 Requires-Dist: certifi (>=2017.4.17)
