--- contrib/python/Cartopy/cartopy/tests/mpl/test_examples.py	(index)
+++ contrib/python/Cartopy/cartopy/tests/mpl/test_examples.py	(working tree)
@@ -42 +42 @@ def test_contour_label():
-    from cartopy.tests.mpl.test_caching import sample_data
+    from ..mpl.test_caching import sample_data
--- contrib/python/Cartopy/cartopy/tests/mpl/test_feature_artist.py	(index)
+++ contrib/python/Cartopy/cartopy/tests/mpl/test_feature_artist.py	(working tree)
@@ -69,2 +68,0 @@ def test_feature_artist_draw(feature):
-    return fig
-
@@ -77,2 +74,0 @@ def test_feature_artist_draw_facecolor_list(feature):
-    return fig
-
@@ -87,2 +82,0 @@ def test_feature_artist_draw_cmap(feature):
-    return fig
-
@@ -98,2 +91,0 @@ def test_feature_artist_draw_styled_feature(feature):
-    return fig
-
@@ -114,2 +105,0 @@ def test_feature_artist_draw_styler(feature):
-    return fig
-
--- contrib/python/Cartopy/cartopy/tests/test_shapereader.py	(index)
+++ contrib/python/Cartopy/cartopy/tests/test_shapereader.py	(working tree)
@@ -18 +18,2 @@ class TestLakes:
-        LAKES_PATH = (Path(__file__).parent / 'lakes_shapefile'
+        import yatest.common as yc
+        LAKES_PATH = (Path(yc.source_path(__file__)).parent / 'lakes_shapefile'
--- contrib/python/Cartopy/cartopy/tests/mpl/test_boundary.py	(index)
+++ contrib/python/Cartopy/cartopy/tests/mpl/test_boundary.py	(working tree)
@@ -84,5 +84,3 @@ def test_multi_path_boundary():
     ax.set_extent((bnds[0], bnds[2], bnds[1], bnds[3]))
     ax.coastlines()
     ax.set_boundary(Path(vertices, codes))
-
-    return f
