--- contrib/python/prometheus-client/tests/test_multiprocess.py	(index)
+++ contrib/python/prometheus-client/tests/test_multiprocess.py	(working tree)
@@ -19,2 +19,3 @@ from prometheus_client.values import (
 
+@unittest.skipIf(True, "warnings disabled")
 class TestMultiProcessDeprecation(unittest.TestCase):
--- contrib/python/prometheus-client/tests/test_process_collector.py	(index)
+++ contrib/python/prometheus-client/tests/test_process_collector.py	(working tree)
@@ -10,1 +10,2 @@ import unittest
-        self.test_proc = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'proc')
+        import yatest.common as yc
+        self.test_proc = os.path.join(os.path.dirname(os.path.realpath(yc.source_path(__file__))), 'proc')
--- contrib/python/prometheus-client/tests/test_exposition.py	(index)
+++ contrib/python/prometheus-client/tests/test_exposition.py	(working tree)
@@ -348,1 +350,2 @@ class TestPushGateway(unittest.TestCase):
-            certs_dir = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'certs')
+            import yatest.common as yc
+            certs_dir = os.path.join(os.path.dirname(os.path.realpath(yc.source_path(__file__))), 'certs')
--- contrib/python/prometheus-client/tests/test_parser.py	(index)
+++ contrib/python/prometheus-client/tests/test_parser.py	(working tree)
@@ -378 +378 @@ prometheus_local_storage_chunk_ops_total{type="unpin"} 32662.0
-def test_benchmark_text_string_to_metric_families(benchmark):
+def _test_benchmark_text_string_to_metric_families(benchmark):
