--- contrib/python/falcon/py2/falcon/media/handlers.py	(index)
+++ contrib/python/falcon/py2/falcon/media/handlers.py	(working tree)
@@ -1,7 +1,7 @@
 from falcon import errors
 from falcon.media import JSONHandler
 from falcon.util.compat import UserDict
-from falcon.vendor import mimeparse
+import mimeparse
 
 
 class Handlers(UserDict):
--- contrib/python/falcon/py2/falcon/request.py	(index)
+++ contrib/python/falcon/py2/falcon/request.py	(working tree)
@@ -26,7 +26,7 @@ from falcon.util import compat
 from falcon.util import json
 from falcon.util import structures
 from falcon.util.uri import parse_host, parse_query_string
-from falcon.vendor import mimeparse
+import mimeparse
 
 DEFAULT_ERROR_LOG_FORMAT = (u'{0:%Y-%m-%d %H:%M:%S} [FALCON] [ERROR]'
                             u' {1} {2}{3} => ')
