--- contrib/python/Faker/py2/tests/providers/test_address.py	(index)
+++ contrib/python/Faker/py2/tests/providers/test_address.py	(working tree)
@@ -23,7 +23,7 @@ from faker.providers.address.hy_AM import Provider as HyAmProvider
 from faker.providers.address.ja_JP import Provider as JaProvider
 from faker.providers.address.ne_NP import Provider as NeProvider
 from faker.providers.address.pt_PT import Provider as PtPtProvider
-from ukpostcodeparser.parser import parse_uk_postcode
+#from ukpostcodeparser.parser import parse_uk_postcode
 
 
 class TestBaseProvider(unittest.TestCase):
@@ -332,6 +332,7 @@ class TestEnCA(unittest.TestCase):
         assert isinstance(secondary_address, string_types)
 
 
+@unittest.skip("no parse_uk_postcode")
 class TestEnGB(unittest.TestCase):
     """ Tests addresses in the en_GB locale """
 
--- contrib/python/Faker/py2/tests/providers/test_ssn.py	(index)
+++ contrib/python/Faker/py2/tests/providers/test_ssn.py	(working tree)
@@ -10,7 +10,7 @@ from itertools import cycle
 
 import freezegun
 import pytest
-import random2
+#import random2
 
 from faker import Faker
 from faker.providers.ssn.en_CA import checksum as ca_checksum
@@ -249,6 +249,7 @@ class TestEnUS(unittest.TestCase):
             assert 1 <= int(serial) <= 9999
             assert area != '666'
 
+    @unittest.skip("no random2")
     def test_invalid_ssn(self):
         self.fake.random = random2.Random()
         # Magic Numbers below generate '666-92-7944', '000-54-2963', '956-GG-9478', '436-00-1386',
@@ -576,6 +577,7 @@ class TestEtEE(unittest.TestCase):
         assert et_checksum([4, 7, 0, 0, 4, 2, 1, 5, 0, 1]) == 2
         assert et_checksum([3, 9, 7, 0, 3, 0, 4, 3, 3, 6]) == 0
 
+    @unittest.skip("no random2")
     @freezegun.freeze_time('2019-03-11')
     def test_ssn(self):
         self.fake.random = random2.Random()
--- contrib/python/Faker/py2/tests/utils/test_utils.py	(index)
+++ contrib/python/Faker/py2/tests/utils/test_utils.py	(working tree)
@@ -10,7 +10,8 @@ from faker.utils.datasets import add_dicts
 from faker.utils.distribution import choices_distribution, choices_distribution_unique
 from faker.utils.loading import find_available_locales, find_available_providers
 
-TEST_DIR = os.path.dirname(__file__)
+import yatest.common
+TEST_DIR = yatest.common.test_source_path('utils')
 
 
 class UtilsTestCase(unittest.TestCase):
