--- contrib/python/aioftp/tests/conftest.py	(index)
+++ contrib/python/aioftp/tests/conftest.py	(working tree)
@@ -14,1 +14,1 @@
-import trustme
+# import trustme
@@ -18,1 +18,1 @@
-from siosocks.io.asyncio import socks_server_handler
+# from siosocks.io.asyncio import socks_server_handler
@@ -22,8 +22,8 @@
-ca = trustme.CA()
-server_cert = ca.issue_server_cert("127.0.0.1", "::1")
-
-ssl_server = ssl.create_default_context(ssl.Purpose.CLIENT_AUTH)
-server_cert.configure_cert(ssl_server)
-
-ssl_client = ssl.create_default_context(ssl.Purpose.SERVER_AUTH)
-ca.configure_trust(ssl_client)
+# ca = trustme.CA()
+# server_cert = ca.issue_server_cert("127.0.0.1", "::1")
+#
+# ssl_server = ssl.create_default_context(ssl.Purpose.CLIENT_AUTH)
+# server_cert.configure_cert(ssl_server)
+#
+# ssl_client = ssl.create_default_context(ssl.Purpose.SERVER_AUTH)
+# ca.configure_trust(ssl_client)
@@ -208,25 +208,25 @@ def skip_sleep(monkeypatch):
-@pytest_asyncio.fixture(
-    params=[
-        ("127.0.0.1", socket.AF_INET),
-        ("::1", socket.AF_INET6),
-    ],
-)
-async def socks(request, unused_tcp_port):
-    handler = functools.partial(
-        socks_server_handler,
-        allowed_versions={5},
-        username="foo",
-        password="bar",
-    )
-    Socks = collections.namedtuple("Socks", "host port server")
-    host, family = request.param
-    port = unused_tcp_port
-    server = await asyncio.start_server(
-        handler,
-        host=host,
-        port=port,
-        family=family,
-    )
-    yield Socks(host, port, server)
-    server.close()
-    await server.wait_closed()
+# @pytest_asyncio.fixture(
+#     params=[
+#         ("127.0.0.1", socket.AF_INET),
+#         ("::1", socket.AF_INET6),
+#     ],
+# )
+# async def socks(request, unused_tcp_port):
+#     handler = functools.partial(
+#         socks_server_handler,
+#         allowed_versions={5},
+#         username="foo",
+#         password="bar",
+#     )
+#     Socks = collections.namedtuple("Socks", "host port server")
+#     host, family = request.param
+#     port = unused_tcp_port
+#     server = await asyncio.start_server(
+#         handler,
+#         host=host,
+#         port=port,
+#         family=family,
+#     )
+#     yield Socks(host, port, server)
+#     server.close()
+#     await server.wait_closed()
