--- a/unsupported/Eigen/CXX11/src/Tensor/TensorContraction.h
+++ b/unsupported/Eigen/CXX11/src/Tensor/TensorContraction.h
@@ -254,2 +254,3 @@ struct TensorContractionKernel {
     // Default GEBP kernel does not support beta.
+    EIGEN_ONLY_USED_FOR_DEBUG(beta);
     eigen_assert(beta == ResScalar(1));
--- a/unsupported/Eigen/CXX11/src/Tensor/TensorIndexList.h	2019-05-13 23:04:35.000000000 +0300
+++ b/unsupported/Eigen/CXX11/src/Tensor/TensorIndexList.h	2020-09-08 20:34:21.000000000 +0300
@@ -42,6 +42,7 @@
   static const Index value = n;
   EIGEN_DEVICE_FUNC constexpr operator Index() const { return n; }
   EIGEN_DEVICE_FUNC void set(Index val) {
+    EIGEN_ONLY_USED_FOR_DEBUG(val);
     eigen_assert(val == n);
   }
 };
@@ -58,6 +59,7 @@
   }
 
   EIGEN_DEVICE_FUNC void set(const IndexPair<Index>& val) {
+    EIGEN_ONLY_USED_FOR_DEBUG(val);
     eigen_assert(val.first == f);
     eigen_assert(val.second == s);
   }
@@ -274,6 +276,7 @@
   }
   template <typename... T>
   EIGEN_DEVICE_FUNC static void set(const Index i, IndexTuple<T...>& t, const ValueT value) {
+    EIGEN_ONLY_USED_FOR_DEBUG(i);
     eigen_assert (i == 0);
     update_value(array_get<0>(t), value);
   }
--- a/Eigen/src/Core/arch/NEON/GeneralBlockPanelKernel.h	(index)
+++ b/eigen/Eigen/src/Core/arch/NEON/GeneralBlockPanelKernel.h	(working tree)
@@ -24,7 +24,7 @@ struct gebp_traits <float,float,false,false,Architecture::NEON,GEBPPacketFull>
 
   template <typename LaneIdType>
   EIGEN_STRONG_INLINE void madd(const Packet4f& a, const Packet4f& b,
-                                Packet4f& c, Packet4f& tmp,
+                                Packet4f& c, Packet4f& /* tmp */,
                                 const LaneIdType&) const {
     acc(a, b, c);
   }
