--- a/Eigen/src/Core/products/Parallelizer.h
+++ b/Eigen/src/Core/products/Parallelizer.h
@@ -23,6 +23,8 @@ inline void manage_multi_threading(Action action, int* v)
 {
   static int m_maxThreads = -1;
   EIGEN_UNUSED_VARIABLE(m_maxThreads)
+  #if !defined(__IOS__) && !defined(_android_)
+  thread_local EIGEN_UNUSED int m_maxThreadsForThisThread = -1;
 
   if(action==SetAction)
   {
@@ -33,7 +35,9 @@ inline void manage_multi_threading(Action action, int* v)
   {
     eigen_internal_assert(v!=0);
     #ifdef EIGEN_HAS_OPENMP
-    if(m_maxThreads>0)
+    if(m_maxThreadsForThisThread>0)
+      *v = m_maxThreadsForThisThread;
+    else if(m_maxThreads>0)
       *v = m_maxThreads;
     else
       *v = omp_get_max_threads();
@@ -41,6 +45,30 @@ inline void manage_multi_threading(Action action, int* v)
     *v = 1;
     #endif
   }
+  else if(action==SetForThreadAction) {
+    eigen_internal_assert(v!=0);
+    m_maxThreadsForThisThread = *v;
+  }
+  #else  // !defined(__IOS__) && !defined(_android_)
+  if(action==SetAction || action==SetForThreadAction)
+  {
+    eigen_internal_assert(v!=0);
+    m_maxThreads = *v;
+  }
+  else if(action==GetAction)
+  {
+    eigen_internal_assert(v!=0);
+    #ifdef EIGEN_HAS_OPENMP
+    if(m_maxThreads>0) {
+      *v = m_maxThreads;
+    } else {
+      *v = omp_get_max_threads();
+    }
+    #else
+    *v = 1;
+    #endif
+  }
+  #endif  // !defined(__IOS__) && !defined(_android_)
   else
   {
     eigen_internal_assert(false);
@@ -74,6 +102,13 @@ inline void setNbThreads(int v)
   internal::manage_multi_threading(SetAction, &v);
 }
 
+/** Sets the max number of threads for the current thread reserved for Eigen
+ **/
+inline void setNbThreadsForCurrentThread(int v)
+{
+  internal::manage_multi_threading(SetForThreadAction, &v);
+}
+
 namespace internal {
 
 template<typename Index> struct GemmParallelInfo
--- a/Eigen/src/Core/util/Constants.h
+++ b/Eigen/src/Core/util/Constants.h
@@ -497,7 +497,7 @@ enum ProductImplType
 
 /** \internal \ingroup enums
   * Enum used in experimental parallel implementation. */
-enum Action {GetAction, SetAction};
+enum Action {GetAction, SetAction, SetForThreadAction = 9999};
 
 /** The type used to identify a dense storage. */
 struct Dense {};
