--- a/unsupported/Eigen/CXX11/src/Tensor/TensorTraits.h
+++ b/unsupported/Eigen/CXX11/src/Tensor/TensorTraits.h
@@ -256,7 +256,10 @@ struct nested<const TensorRef<PlainObjectType> >
 // Pc=0.
 typedef enum {
   PADDING_VALID = 1,
-  PADDING_SAME = 2
+  PADDING_SAME = 2,
+/** Arcadia specific section: begin */
+  PADDING_DANET = 837,  // TODO (splinter|2021-03-26): forbid to use danet style padding with eigen
+/** Arcadia specific section: end */
 } PaddingType;

 }  // end namespace Eigen
