--- contrib/python/path.py/py2/path.py	(index)
+++ contrib/python/path.py/py2/path.py	(working tree)
@@ -1732,7 +1732,7 @@ def _permission_mask(mode):
     True
     """
     # parse the symbolic mode
-    parsed = re.match('(?P<who>[ugoa]+)(?P<op>[-+=])(?P<what>[rwx]*)$', mode)
+    parsed = re.match('(?P<who>[ugoa]+)?(?P<op>[-+=])(?P<what>[rwx]*)$', mode)
     if not parsed:
         raise ValueError("Unrecognized symbolic mode", mode)
 
@@ -1743,7 +1743,7 @@ def _permission_mask(mode):
 
     # now apply spec to each subject in who
     shift_map = dict(u=6, g=3, o=0)
-    who = parsed.group('who').replace('a', 'ugo')
+    who = (parsed.group('who') or 'a').replace('a', 'ugo')
     masks = (spec << shift_map[subj] for subj in who)
     mask = functools.reduce(operator.or_, masks)
 
