Index: contrib/python/bandit/bandit/core/blacklisting.py
===================================================================
--- contrib/python/bandit/bandit/core/blacklisting.py	(revision 6984347)
+++ contrib/python/bandit/bandit/core/blacklisting.py	(working copy)
@@ -4,6 +4,8 @@
 #
 # SPDX-License-Identifier: Apache-2.0
 import ast
+import fnmatch
+import re
 
 from bandit.core import issue
 
@@ -53,9 +54,13 @@
                     name = context.call_args[0]
                 else:
                     name = context.call_keywords["name"]
-        for check in blacklists[node_type]:
-            for qn in check["qualnames"]:
-                if name is not None and name == qn:
+
+        if name is not None:
+            reqn = "qualnames_re_cache"
+            for check in blacklists[node_type]:
+                if reqn not in check:
+                    check[reqn] = re.compile("|".join(fnmatch.translate(x) for x in check["qualnames"]))
+                if check[reqn].match(name):
                     return report_issue(check, name)
 
     if node_type.startswith("Import"):
