--- contrib/python/entrypoints/entrypoints.py	(index)
+++ contrib/python/entrypoints/entrypoints.py	(working tree)
@@ -186,6 +186,26 @@ def iter_files_distros(path=None, repeated_distro='first'):
                         cp.read_file(fu, source=osp.join(folder, info.filename))
                     yield cp, distro
 
+        # Arcadia importer.
+        elif folder == sys.executable:
+            from pkg_resources import find_in_res
+            for d in find_in_res(0, 0):
+                if not d.has_metadata('entry_points.txt'):
+                    continue
+
+                distro = Distribution(d.project_name, d.version)
+                if (repeated_distro == 'first') and (distro.name in distro_names_seen):
+                    continue
+                distro_names_seen.add(distro.name)
+
+                text = d.get_metadata('entry_points.txt')
+                if isinstance(text, bytes):
+                    text = text.decode('UTF-8')
+
+                cp = CaseSensitiveConfigParser(delimiters=('=',))
+                cp.read_string(text, source=d._provider._get_metadata_path('entry_points.txt'))
+                yield cp, distro
+
         # Regular file imports (not egg, not zip file)
         for path in itertools.chain(
             glob.iglob(osp.join(glob.escape(folder), '*.dist-info', 'entry_points.txt')),
