--- contrib/python/marshmallow-jsonschema/marshmallow_jsonschema/base.py	(c13d656f861b7af70b9d5a28834df76b00db3356)
+++ contrib/python/marshmallow-jsonschema/marshmallow_jsonschema/base.py	(0e10a9c969f42e76ccafed4d5991b86a809c043a)
@@ -4,13 +4,22 @@ import uuid
 from enum import Enum
 from inspect import isclass
 import typing
+import importlib.metadata
 
 from marshmallow import fields, missing, Schema, validate
 from marshmallow.class_registry import get_class
 from marshmallow.decorators import post_dump
 from marshmallow.utils import _Missing
-
 from marshmallow import INCLUDE, EXCLUDE, RAISE
+from packaging.version import Version
+
+
+def marshmallow_version_supports_native_enums() -> bool:
+    """
+    returns true if and only if the version of marshmallow installed supports enums natively
+    """
+    return Version(importlib.metadata.version("marshmallow")) >= Version("3.18")
+
 
 try:
     from marshmallow_union import Union
@@ -20,11 +29,15 @@ except ImportError:
     ALLOW_UNIONS = False
 
 try:
-    from marshmallow_enum import EnumField, LoadDumpOptions
+    from marshmallow_enum import EnumField as MarshmallowEnumEnumField, LoadDumpOptions
 
-    ALLOW_ENUMS = True
+    ALLOW_MARSHMALLOW_ENUM_ENUMS = True
 except ImportError:
-    ALLOW_ENUMS = False
+    ALLOW_MARSHMALLOW_ENUM_ENUMS = False
+
+ALLOW_MARSHMALLOW_NATIVE_ENUMS = marshmallow_version_supports_native_enums()
+if ALLOW_MARSHMALLOW_NATIVE_ENUMS:
+    from marshmallow.fields import Enum as MarshmallowNativeEnumField
 
 from .exceptions import UnsupportedValueError
 from .validation import (
@@ -92,10 +105,12 @@ MARSHMALLOW_TO_PY_TYPES_PAIRS = [
     (fields.Nested, dict),
 ]
 
-if ALLOW_ENUMS:
+if ALLOW_MARSHMALLOW_NATIVE_ENUMS:
+    MARSHMALLOW_TO_PY_TYPES_PAIRS.append((MarshmallowNativeEnumField, Enum))
+if ALLOW_MARSHMALLOW_ENUM_ENUMS:
     # We currently only support loading enum's from their names. So the possible
     # values will always map to string in the JSONSchema
-    MARSHMALLOW_TO_PY_TYPES_PAIRS.append((EnumField, Enum))
+    MARSHMALLOW_TO_PY_TYPES_PAIRS.append((MarshmallowEnumEnumField, Enum))
 
 
 FIELD_VALIDATORS = {
@@ -191,8 +206,10 @@ class JSONSchema(Schema):
         if field.default is not missing and not callable(field.default):
             json_schema["default"] = field.default
 
-        if ALLOW_ENUMS and isinstance(field, EnumField):
-            json_schema["enum"] = self._get_enum_values(field)
+        if ALLOW_MARSHMALLOW_NATIVE_ENUMS and isinstance(field, MarshmallowNativeEnumField):
+            json_schema["enum"] = self._get_marshmallow_native_enum_values(field)
+        elif ALLOW_MARSHMALLOW_ENUM_ENUMS and isinstance(field, MarshmallowEnumEnumField):
+            json_schema["enum"] = self._get_marshmallow_enum_enum_values(field)
 
         if field.allow_none:
             previous_type = json_schema["type"]
@@ -218,8 +235,8 @@ class JSONSchema(Schema):
             )
         return json_schema
 
-    def _get_enum_values(self, field) -> typing.List[str]:
-        assert ALLOW_ENUMS and isinstance(field, EnumField)
+    def _get_marshmallow_enum_enum_values(self, field) -> typing.List[str]:
+        assert ALLOW_MARSHMALLOW_ENUM_ENUMS and isinstance(field, MarshmallowEnumEnumField)
 
         if field.load_by == LoadDumpOptions.value:
             # Python allows enum values to be almost anything, so it's easier to just load from the
@@ -229,6 +246,17 @@ class JSONSchema(Schema):
             )
 
         return [value.name for value in field.enum]
+    def _get_marshmallow_native_enum_values(self, field) -> typing.List[str]:
+        assert ALLOW_MARSHMALLOW_NATIVE_ENUMS and isinstance(field, MarshmallowNativeEnumField)
+
+        if field.by_value:
+            # Python allows enum values to be almost anything, so it's easier to just load from the
+            # names of the enum's which will have to be strings.
+            raise NotImplementedError(
+                "Currently do not support JSON schema for enums loaded by value"
+            )
+
+        return [value.name for value in field.enum]
 
     def _from_union_schema(
         self, obj, field
--- contrib/python/marshmallow-jsonschema/ya.make	(c13d656f861b7af70b9d5a28834df76b00db3356)
+++ contrib/python/marshmallow-jsonschema/ya.make	(0e10a9c969f42e76ccafed4d5991b86a809c043a)
@@ -10,6 +10,7 @@ LICENSE(MIT)
 
 PEERDIR(
     contrib/python/marshmallow
+    contrib/python/packaging
 )
 
 NO_LINT()
