--- contrib/python/snowflake-connector-python/snowflake/connector/arrow_iterator.pyx   (index)
+++ contrib/python/snowflake-connector-python/snowflake/connector/arrow_iterator.pyx   (working tree)
@@ -23,10 +23,8 @@ from pyarrow.includes.libarrow cimport (
     Seekable,
 )

-IF ARROW_LESS_THAN_8:
-    from pyarrow.includes.libarrow cimport PyReadableFile
-ELSE:
-    from pyarrow.includes.libarrow_python cimport PyReadableFile
+# todo: update when pyarrow >= 8
+from pyarrow.includes.libarrow cimport PyReadableFile

 from .constants import IterUnit
 from .errorcode import (
