--- contrib/python/ua-parser/py2/ua_parser/user_agent_parser_test.py	(index)
+++ contrib/python/ua-parser/py2/ua_parser/user_agent_parser_test.py	(working tree)
@@ -48,9 +48,8 @@ except ImportError:
 
 from ua_parser import user_agent_parser
 
-TEST_RESOURCES_DIR = os.path.join(
-    os.path.abspath(os.path.dirname(__file__)), "../uap-core"
-)
+import yatest.common as yc
+TEST_RESOURCES_DIR = yc.work_path('test_data/')
 
 
 class ParseTest(unittest.TestCase):
