--- contrib/python/chdb/cpp/src/Common/Config/ConfigProcessor.cpp	(index)
+++ contrib/python/chdb/cpp/src/Common/Config/ConfigProcessor.cpp	(working tree)
@@ -9,6 +9,7 @@
 #include <algorithm>
 #include <functional>
 #include <filesystem>
+#include <fstream>
 #include <boost/algorithm/string.hpp>
 #include <Poco/DOM/Text.h>
 #include <Poco/DOM/Attr.h>
--- contrib/python/chdb/cpp/src/Common/getNumberOfPhysicalCPUCores.cpp	(index)
+++ contrib/python/chdb/cpp/src/Common/getNumberOfPhysicalCPUCores.cpp	(working tree)
@@ -13,6 +13,7 @@
 #include <filesystem>
 #include <thread>
 #include <set>
+#include <vector>
 
 namespace
 {
--- contrib/python/chdb/cpp/src/Disks/ObjectStorages/StoredObject.cpp	(index)
+++ contrib/python/chdb/cpp/src/Disks/ObjectStorages/StoredObject.cpp	(working tree)
@@ -1,5 +1,7 @@
 #include <Disks/ObjectStorages/StoredObject.h>
 
+#include <vector>
+
 namespace DB
 {
 
