--- contrib/python/chdb/cpp/src/Dictionaries/MongoDBDictionarySource.cpp	(index)
+++ contrib/python/chdb/cpp/src/Dictionaries/MongoDBDictionarySource.cpp	(working tree)
@@ -1,3 +1,4 @@
+#if 0
 #include "MongoDBDictionarySource.h"
 #include "DictionarySourceFactory.h"
 #include "DictionaryStructure.h"
@@ -278,3 +279,4 @@ std::string MongoDBDictionarySource::toString() const
 }
 
 }
+#endif
--- contrib/python/chdb/cpp/src/Dictionaries/registerDictionaries.cpp	(index)
+++ contrib/python/chdb/cpp/src/Dictionaries/registerDictionaries.cpp	(working tree)
@@ -43,7 +43,7 @@ void registerDictionaries()
         registerDictionarySourceFile(source_factory);
         registerDictionarySourceMysql(source_factory);
         registerDictionarySourceClickHouse(source_factory);
-        registerDictionarySourceMongoDB(source_factory);
+        // registerDictionarySourceMongoDB(source_factory);
         registerDictionarySourceRedis(source_factory);
         registerDictionarySourceCassandra(source_factory);
         registerDictionarySourceXDBC(source_factory);
--- contrib/python/chdb/cpp/src/Processors/Sources/MongoDBSource.cpp	(index)
+++ contrib/python/chdb/cpp/src/Processors/Sources/MongoDBSource.cpp	(working tree)
@@ -1,3 +1,4 @@
+#if 0
 #include "MongoDBSource.h"
 
 #include <string>
@@ -575,3 +576,4 @@ Chunk MongoDBSource::generate()
 }
 
 }
+#endif
--- contrib/python/chdb/cpp/src/Storages/StorageMongoDB.cpp	(index)
+++ contrib/python/chdb/cpp/src/Storages/StorageMongoDB.cpp	(working tree)
@@ -1,3 +1,4 @@
+#if 0
 #include <Storages/StorageMongoDB.h>
 #include <Storages/StorageMongoDBSocketFactory.h>
 #include <Storages/StorageFactory.h>
@@ -316,3 +317,4 @@ void registerStorageMongoDB(StorageFactory & factory)
 }
 
 }
+#endif
--- contrib/python/chdb/cpp/src/Storages/StorageMongoDBSocketFactory.cpp	(index)
+++ contrib/python/chdb/cpp/src/Storages/StorageMongoDBSocketFactory.cpp	(working tree)
@@ -1,3 +1,4 @@
+#if 0
 #include "StorageMongoDBSocketFactory.h"
 
 #include <Common/Exception.h>
@@ -53,3 +54,4 @@ Poco::Net::StreamSocket StorageMongoDBSocketFactory::createSecureSocket(const st
 }
 
 }
+#endif
--- contrib/python/chdb/cpp/src/Storages/registerStorages.cpp	(index)
+++ contrib/python/chdb/cpp/src/Storages/registerStorages.cpp	(working tree)
@@ -167,7 +167,7 @@ void registerStorages()
     registerStorageMySQL(factory);
     #endif
 
-    registerStorageMongoDB(factory);
+    // registerStorageMongoDB(factory);
     registerStorageRedis(factory);
 
     #if USE_RDKAFKA
--- contrib/python/chdb/cpp/src/TableFunctions/TableFunctionMongoDB.cpp	(index)
+++ contrib/python/chdb/cpp/src/TableFunctions/TableFunctionMongoDB.cpp	(working tree)
@@ -1,3 +1,4 @@
+#if 0
 #include <Storages/StorageMongoDB.h>
 
 #include <Common/Exception.h>
@@ -126,3 +127,4 @@ void registerTableFunctionMongoDB(TableFunctionFactory & factory)
 }
 
 }
+#endif
--- contrib/python/chdb/cpp/src/TableFunctions/registerTableFunctions.cpp	(index)
+++ contrib/python/chdb/cpp/src/TableFunctions/registerTableFunctions.cpp	(working tree)
@@ -23,7 +23,7 @@ void registerTableFunctions()
     registerTableFunctionValues(factory);
     registerTableFunctionInput(factory);
     registerTableFunctionGenerate(factory);
-    registerTableFunctionMongoDB(factory);
+    // registerTableFunctionMongoDB(factory);
     registerTableFunctionRedis(factory);
     registerTableFunctionMergeTreeIndex(factory);
     registerTableFunctionFuzzQuery(factory);
