From 055af4b755fc60f6876c43471f466acf99727f14 Mon Sep 17 00:00:00 2001
From: Aleksandr Mikhnenko <mikhnenko@yandex-team.com>
Date: Sun, 12 Jan 2025 18:18:51 +0300
Subject: [PATCH 1/2] Add missed include <stdlib.h>

---
 base/base/defines.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/base/base/defines.h b/base/base/defines.h
index a0c3c0d1de54..239196c6b9f7 100644
--- a/base/base/defines.h
+++ b/base/base/defines.h
@@ -107,6 +107,7 @@
 #    if defined(DEBUG_OR_SANITIZER_BUILD)
         // clang-format off
         #include <base/types.h>
+        #include <stdlib.h>
         namespace DB
         {
             [[noreturn]] void abortOnFailedAssertion(const String & description);

From 8d1ddb83bc79fab0ad0b98691dae66db307789af Mon Sep 17 00:00:00 2001
From: Aleksandr Mikhnenko <mikhnenko@yandex-team.com>
Date: Wed, 29 Jan 2025 17:45:22 +0300
Subject: [PATCH 2/2] add missed vector too

---
 src/Backups/BackupStatus.h | 2 +-
 src/IO/HTTPHeaderEntries.h | 1 +
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/Backups/BackupStatus.h b/src/Backups/BackupStatus.h
index 398f2086cfa8..ce9bd8b90d64 100644
--- a/src/Backups/BackupStatus.h
+++ b/src/Backups/BackupStatus.h
@@ -1,7 +1,7 @@
 #pragma once
 
 #include <base/types.h>
-
+#include <vector>
 
 namespace DB
 {
diff --git a/src/IO/HTTPHeaderEntries.h b/src/IO/HTTPHeaderEntries.h
index 36b2ccc4ba5e..3bcffeba4c89 100644
--- a/src/IO/HTTPHeaderEntries.h
+++ b/src/IO/HTTPHeaderEntries.h
@@ -1,5 +1,6 @@
 #pragma once
 #include <string>
+#include <vector>
 
 namespace DB
 {
