--- contrib/python/chdb/cpp/src/Parsers/Kusto/KustoFunctions/KQLDateTimeFunctions.cpp	(index)
+++ contrib/python/chdb/cpp/src/Parsers/Kusto/KustoFunctions/KQLDateTimeFunctions.cpp	(working tree)
@@ -250,7 +250,7 @@ bool EndOfYear::convertImpl(String & out, IParser::Pos & pos)
         offset = getConvertedArgument(fn_name, pos);
         if (offset.empty())
             throw Exception(ErrorCodes::SYNTAX_ERROR, "Number of arguments do not match in function: {}", fn_name);
-        offset.erase(remove(offset.begin(), offset.end(), ' '), offset.end());
+        offset.erase(std::remove(offset.begin(), offset.end(), ' '), offset.end());
     }
 
     out = std::format(
--- contrib/python/chdb/cpp/src/Parsers/Kusto/KustoFunctions/KQLDynamicFunctions.cpp	(index)
+++ contrib/python/chdb/cpp/src/Parsers/Kusto/KustoFunctions/KQLDynamicFunctions.cpp	(working tree)
@@ -267,8 +267,8 @@ bool Repeat::convertImpl(String & out, IParser::Pos & pos)
     String value = getArgument(function_name, pos);
     String count = getArgument(function_name, pos);
 
-    value.erase(remove(value.begin(), value.end(), ' '), value.end());
-    count.erase(remove(count.begin(), count.end(), ' '), count.end());
+    value.erase(std::remove(value.begin(), value.end(), ' '), value.end());
+    count.erase(std::remove(count.begin(), count.end(), ' '), count.end());
 
     if (count.empty())
         throw Exception(ErrorCodes::SYNTAX_ERROR, "number of arguments do not match in function: {}", function_name);
--- contrib/python/chdb/cpp/src/Parsers/Kusto/KustoFunctions/KQLStringFunctions.cpp	(index)
+++ contrib/python/chdb/cpp/src/Parsers/Kusto/KustoFunctions/KQLStringFunctions.cpp	(working tree)
@@ -559,7 +559,7 @@ bool Split::convertImpl(String & out, IParser::Pos & pos)
         ++pos;
         auto arg = getConvertedArgument(fn_name, pos);
         // remove space between minus and value
-        arg.erase(remove_if(arg.begin(), arg.end(), isspace), arg.end());
+        arg.erase(std::remove_if(arg.begin(), arg.end(), isspace), arg.end());
         requested_index = std::stoi(arg);
         requested_index += 1;
         out = std::format(
