--- contrib/python/chdb/cpp/base/poco/Foundation/include/Poco/Dynamic/VarHolder.h	(index)
+++ contrib/python/chdb/cpp/base/poco/Foundation/include/Poco/Dynamic/VarHolder.h	(working tree)
@@ -417,7 +417,7 @@ namespace Dynamic
             {
                 throw RangeException("Value too large.");
             }
-            else if (from > std::numeric_limits<T>::max())
+            else if (static_cast<T>(from) > std::numeric_limits<T>::max())
             {
                 throw RangeException("Value too large.");
             }
--- contrib/python/chdb/cpp/src/Common/PageCache.cpp	(index)
+++ contrib/python/chdb/cpp/src/Common/PageCache.cpp	(working tree)
@@ -260,7 +260,7 @@ PageCache::MemoryStats PageCache::getResidentSetSize() const
             if (!in.eof() && *in.position() == '-')
             {
                 if (s.size() < 16)
-                    s.insert(0, 16 - s.size(), '0');
+                    s.insert(0ul, 16 - s.size(), '0');
                 UInt64 addr = unhexUInt<UInt64>(s.c_str());
                 current_range_is_cache = cache_mmap_addrs.contains(addr);
             }
