--- contrib/python/chdb/cpp/src/Client/ClientApplicationBase.cpp	(index)
+++ contrib/python/chdb/cpp/src/Client/ClientApplicationBase.cpp	(working tree)
@@ -47,13 +47,18 @@ static ClientInfo::QueryKind parseQueryKind(const String & query_kind)
 /// This signal handler is set only for SIGINT and SIGQUIT.
 void interruptSignalHandler(int signum)
 {
+    // https://github.com/ClickHouse/ClickHouse/commit/c7c1f10720cd194d85de6d81156cbd37304ab52b#diff-88cd216d19afe73d27f1f21466b8624663594a4d6e106040bce1bd448322f41bR458
+    // https://github.com/ClickHouse/ClickHouse/commit/49dc30d5c28392d361ce0ef1e18f7db73841617f#diff-26fd521d2157408b67e75d02cebffb1c05a913f88f2a94679b1a0d7b9e24854cR53
+
     /// Signal handler might be called even before the setup is fully finished
     /// and client application started to process the query.
     /// Because of that we have to manually check it.
-    if (auto * instance = ClientApplicationBase::instanceRawPtr(); instance)
-        if (auto * base = dynamic_cast<ClientApplicationBase *>(instance); base)
-            if (base->tryStopQuery())
-                safeExit(128 + signum);
+    if (ClientApplicationBase::getInstance().tryStopQuery())
+        safeExit(128 + signum);
+//     if (auto * instance = ClientApplicationBase::instanceRawPtr(); instance)
+//         if (auto * base = dynamic_cast<ClientApplicationBase *>(instance); base)
+//             if (base->tryStopQuery())
+//                 safeExit(128 + signum);
 }
 
 ClientApplicationBase::~ClientApplicationBase()
--- contrib/python/chdb/cpp/src/Processors/Formats/Impl/AvroRowOutputFormat.cpp	(index)
+++ contrib/python/chdb/cpp/src/Processors/Formats/Impl/AvroRowOutputFormat.cpp	(working tree)
@@ -520,8 +520,8 @@ static avro::Codec getCodec(const std::string & codec_name)
 
     if (codec_name == "null")    return avro::Codec::NULL_CODEC;
     if (codec_name == "deflate") return avro::Codec::DEFLATE_CODEC;
-    if (codec_name == "zstd")
-        return avro::Codec::ZSTD_CODEC;
+    // if (codec_name == "zstd")
+    //     return avro::Codec::ZSTD_CODEC;
 #ifdef SNAPPY_CODEC_AVAILABLE
     if (codec_name == "snappy")
         return avro::Codec::SNAPPY_CODEC;
--- contrib/python/chdb/cpp/src/Storages/ObjectStorage/DataLakes/IcebergMetadata.cpp	(index)
+++ contrib/python/chdb/cpp/src/Storages/ObjectStorage/DataLakes/IcebergMetadata.cpp	(working tree)
@@ -487,18 +487,18 @@ Strings IcebergMetadata::getDataFiles() const
 
         /// Manifest file should always have table schema in avro file metadata. By now we don't support tables with evolved schema,
         /// so we should check if all manifest files have the same schema as in table metadata.
-        auto avro_metadata = manifest_file_reader->metadata();
-        std::vector<uint8_t> schema_json = avro_metadata["schema"];
-        String schema_json_string = String(reinterpret_cast<char *>(schema_json.data()), schema_json.size());
-        Poco::JSON::Parser parser;
-        Poco::Dynamic::Var json = parser.parse(schema_json_string);
-        Poco::JSON::Object::Ptr schema_object = json.extract<Poco::JSON::Object::Ptr>();
-        if (!context->getSettingsRef().iceberg_engine_ignore_schema_evolution && schema_object->getValue<int>("schema-id") != current_schema_id)
-            throw Exception(
-                ErrorCodes::UNSUPPORTED_METHOD,
-                "Cannot read Iceberg table: the table schema has been changed at least 1 time, reading tables with evolved schema is not "
-                "supported. If you want to ignore schema evolution and read all files using latest schema saved on table creation, enable setting "
-                "iceberg_engine_ignore_schema_evolution (Note: enabling this setting can lead to incorrect result)");
+        // auto avro_metadata = manifest_file_reader->metadata();
+        // std::vector<uint8_t> schema_json = avro_metadata["schema"];
+        // String schema_json_string = String(reinterpret_cast<char *>(schema_json.data()), schema_json.size());
+        // Poco::JSON::Parser parser;
+        // Poco::Dynamic::Var json = parser.parse(schema_json_string);
+        // Poco::JSON::Object::Ptr schema_object = json.extract<Poco::JSON::Object::Ptr>();
+        // if (!context->getSettingsRef().iceberg_engine_ignore_schema_evolution && schema_object->getValue<int>("schema-id") != current_schema_id)
+        //     throw Exception(
+        //         ErrorCodes::UNSUPPORTED_METHOD,
+        //         "Cannot read Iceberg table: the table schema has been changed at least 1 time, reading tables with evolved schema is not "
+        //         "supported. If you want to ignore schema evolution and read all files using latest schema saved on table creation, enable setting "
+        //         "iceberg_engine_ignore_schema_evolution (Note: enabling this setting can lead to incorrect result)");
 
         avro::NodePtr root_node = manifest_file_reader->dataSchema().root();
         size_t leaves_num = root_node->leaves();
