--- contrib/python/chdb/cpp/src/Compression/CompressionCodecEncrypted.cpp	(index)
+++ contrib/python/chdb/cpp/src/Compression/CompressionCodecEncrypted.cpp	(working tree)
@@ -136,7 +136,6 @@ size_t encrypt(std::string_view plaintext, char * ciphertext_and_tag, Encryption
     int out_len;
     int ciphertext_len;
     EVP_CIPHER_CTX * ctx;
-    EVP_CIPHER * cipher;
 
     ctx = EVP_CIPHER_CTX_new();
     if (ctx == nullptr)
@@ -144,7 +143,7 @@ size_t encrypt(std::string_view plaintext, char * ciphertext_and_tag, Encryption
 
     try
     {
-        cipher = EVP_CIPHER_fetch(nullptr, getMethod(method), nullptr);
+        const EVP_CIPHER* cipher = EVP_get_cipherbyname(getMethod(method));
         if (cipher == nullptr)
             throw Exception::createDeprecated(lastErrorString(), ErrorCodes::OPENSSL_ERROR);
 
@@ -185,11 +184,9 @@ size_t encrypt(std::string_view plaintext, char * ciphertext_and_tag, Encryption
     }
     catch (...)
     {
-        EVP_CIPHER_free(cipher);
         EVP_CIPHER_CTX_free(ctx);
         throw;
     }
-    EVP_CIPHER_free(cipher);
     EVP_CIPHER_CTX_free(ctx);
     return ciphertext_len + tag_size;
 }
@@ -203,7 +200,7 @@ size_t decrypt(std::string_view ciphertext, char * plaintext, EncryptionMethod m
     int out_len;
     int plaintext_len;
     EVP_CIPHER_CTX * ctx;
-    EVP_CIPHER * cipher;
+    // EVP_CIPHER * cipher;
 
     ctx = EVP_CIPHER_CTX_new();
     if (ctx == nullptr)
@@ -211,7 +208,8 @@ size_t decrypt(std::string_view ciphertext, char * plaintext, EncryptionMethod m
 
     try
     {
-        cipher = EVP_CIPHER_fetch(nullptr, getMethod(method), nullptr);
+        const EVP_CIPHER* cipher = EVP_get_cipherbyname(getMethod(method));
+        // cipher = EVP_CIPHER_fetch(nullptr, getMethod(method), nullptr);
         if (cipher == nullptr)
             throw Exception::createDeprecated(lastErrorString(), ErrorCodes::OPENSSL_ERROR);
 
@@ -252,11 +250,11 @@ size_t decrypt(std::string_view ciphertext, char * plaintext, EncryptionMethod m
     }
     catch (...)
     {
-        EVP_CIPHER_free(cipher);
+        // EVP_CIPHER_free(cipher);
         EVP_CIPHER_CTX_free(ctx);
         throw;
     }
-    EVP_CIPHER_free(cipher);
+    // EVP_CIPHER_free(cipher);
     EVP_CIPHER_CTX_free(ctx);
 
     return plaintext_len + out_len;
