--- contrib/python/chdb/cpp/src/Disks/ObjectStorages/MetadataStorageFromPlainRewritableObjectStorage.cpp    (index)
+++ contrib/python/chdb/cpp/src/Disks/ObjectStorages/MetadataStorageFromPlainRewritableObjectStorage.cpp    (working tree)
@@ -147,7 +147,7 @@ void getDirectChildrenOnDiskImpl(
             if (!k.starts_with(local_path))
                 break;
 
-            auto slash_num = count(k.begin() + local_path.size(), k.end(), '/');
+            auto slash_num = std::count(k.begin() + local_path.size(), k.end(), '/');
             /// The local_path_prefixes comparator ensures that the paths with the smallest number of
             /// hops from the local_path are iterated first. The paths do not end with '/', hence
             /// break the loop if the number of slashes is greater than 0.
--- contrib/python/chdb/cpp/src/Storages/MergeTree/RowOrderOptimizer.cpp	(index)
+++ contrib/python/chdb/cpp/src/Storages/MergeTree/RowOrderOptimizer.cpp	(working tree)
@@ -39,7 +39,7 @@ std::vector<size_t> getOtherColumnIndexes(const Block & block, const SortDescrip
     if (sorting_key_columns_count == 0)
     {
         other_column_indexes.resize(block.columns());
-        iota(other_column_indexes.begin(), other_column_indexes.end(), 0);
+        std::iota(other_column_indexes.begin(), other_column_indexes.end(), 0);
     }
     else
     {
@@ -127,9 +127,9 @@ void updatePermutationInEqualRange(
     LOG_TEST(log, "Starting optimization in equal range");
 
     std::vector<size_t> column_order(other_column_indexes.size());
-    iota(column_order.begin(), column_order.end(), 0);
+    std::iota(column_order.begin(), column_order.end(), 0);
     auto cmp = [&](size_t lhs, size_t rhs) -> bool { return cardinalities[lhs] < cardinalities[rhs]; };
-    stable_sort(column_order.begin(), column_order.end(), cmp);
+    std::stable_sort(column_order.begin(), column_order.end(), cmp);
 
     std::vector<EqualRange> ranges = {equal_range};
     LOG_TEST(log, "equal_range: .from: {}, .to: {}", equal_range.from, equal_range.to);
