--- contrib/python/chdb/cpp/src/Backups/IBackupCoordination.h	(5ea1aa02ce8cbd9fe4f7ad9850bbc1f482443b12)
+++ contrib/python/chdb/cpp/src/Backups/IBackupCoordination.h	(working tree)
@@ -1,5 +1,6 @@
 #pragma once
 
+#include <chrono>
 #include <Core/Types.h>
 
 
--- contrib/python/chdb/cpp/src/Backups/IRestoreCoordination.h	(5ea1aa02ce8cbd9fe4f7ad9850bbc1f482443b12)
+++ contrib/python/chdb/cpp/src/Backups/IRestoreCoordination.h	(working tree)
@@ -1,5 +1,6 @@
 #pragma once
 
+#include <chrono>
 #include <Core/Types.h>
 
 
--- contrib/python/chdb/cpp/src/Disks/IO/getThreadPoolReader.h	(5ea1aa02ce8cbd9fe4f7ad9850bbc1f482443b12)
+++ contrib/python/chdb/cpp/src/Disks/IO/getThreadPoolReader.h	(working tree)
@@ -1,5 +1,7 @@
 #pragma once
 
+#include <memory>
+
 namespace CHDBPoco::Util { class AbstractConfiguration; }
 
 namespace DB_CHDB
--- contrib/python/chdb/cpp/src/Disks/ObjectStorages/IMetadataOperation.h	(5ea1aa02ce8cbd9fe4f7ad9850bbc1f482443b12)
+++ contrib/python/chdb/cpp/src/Disks/ObjectStorages/IMetadataOperation.h	(working tree)
@@ -1,5 +1,6 @@
 #pragma once
 
+#include <memory>
 #include <mutex>
 #include <Common/SharedMutex.h>
 
