From 0a037633fbe1430e6103863ba1a9cbf4306ddfc3 Mon Sep 17 00:00:00 2001
From: Yuriy Chernyshov <thegeorg@yandex-team.com>
Date: Thu, 21 Aug 2025 12:50:56 +0300
Subject: [PATCH] Make StringRefHash compatible with abseil-cpp 20250814.0

---
 base/base/StringRef.h | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/base/base/StringRef.h b/base/base/StringRef.h
index 8dbce1e4c61f..98b3b64ce24d 100644
--- a/base/base/StringRef.h
+++ b/base/base/StringRef.h
@@ -310,7 +310,7 @@ inline size_t hashLessThan16(const char * data, size_t size)
 
 struct CRC32Hash
 {
-    unsigned operator() (StringRef x) const
+    size_t operator() (StringRef x) const
     {
         const char * pos = x.data;
         size_t size = x.size;
@@ -326,7 +326,7 @@ struct CRC32Hash
         }
 
         const char * end = pos + size;
-        unsigned res = -1U;
+        size_t res = -1U;
 
         do
         {
@@ -339,7 +339,11 @@ struct CRC32Hash
         UInt64 word = unalignedLoadLittleEndian<UInt64>(end - 8);    /// I'm not sure if this is normal.
         res = static_cast<unsigned>(CRC_INT(res, word));
 
-        return res;
+        // abseil-cpp and std require hash functions to return 64-bit values,
+        // though we intentionally use crc32 for the sake of speed.
+        //
+        // Fill upper bits with the same value as the lower ones.
+        return (res << 32) | res;
     }
 };
 
