--- contrib/python/chdb/cpp/programs/local/FormatHelper.cpp (index)
+++ contrib/python/chdb/cpp/programs/local/FormatHelper.cpp (working tree)
@@ -1,5 +1,7 @@
 #include "FormatHelper.h"
 
+#include <algorithm>
+
 namespace CHDB {
 
 static bool is_json_supported = true;
--- contrib/python/chdb/cpp/programs/local/QueryResult.h    (index)
+++ contrib/python/chdb/cpp/programs/local/QueryResult.h    (working tree)
@@ -1,6 +1,7 @@
 #pragma once
 
-#include "chdb-internal.h"
+#include <memory> 
+#include <vector>
 
 #include <base/types.h>
 #include <utility>