--- contrib/python/chdb/cpp/src/Processors/Formats/Impl/ORCBlockInputFormat.cpp (index)
+++ contrib/python/chdb/cpp/src/Processors/Formats/Impl/ORCBlockInputFormat.cpp (working tree)
@@ -44,7 +44,7 @@ Chunk ORCBlockInputFormat::read()
         return {};
 
     if (need_only_count)
-        return getChunkForCount(file_reader->GetRawORCReader()->getStripe(stripe_current++)->getNumberOfRows());
+        throw Exception(500, "Not implemented");
 
     auto batch_result = file_reader->ReadStripe(stripe_current, include_indices);
     if (!batch_result.ok())
@@ -59,22 +59,7 @@ Chunk ORCBlockInputFormat::read()
         throw Exception(
             ErrorCodes::CANNOT_READ_ALL_DATA, "Error while reading batch of ORC data: {}", table_result.status().ToString());
 
-    /// We should extract the number of rows directly from the stripe, because in case when
-    /// record batch contains 0 columns (for example if we requested only columns that
-    /// are not presented in data) the number of rows in record batch will be 0.
-    size_t num_rows = file_reader->GetRawORCReader()->getStripe(stripe_current)->getNumberOfRows();
-
-    auto table = table_result.ValueOrDie();
-    if (!table || !num_rows)
-        return {};
-
-    approx_bytes_read_for_chunk = file_reader->GetRawORCReader()->getStripe(stripe_current)->getDataLength();
-    ++stripe_current;
-
-    /// If defaults_for_omitted_fields is true, calculate the default values from default expression for omitted fields.
-    /// Otherwise fill the missing columns with zero values of its type.
-    BlockMissingValues * block_missing_values_ptr = format_settings.defaults_for_omitted_fields ? &block_missing_values : nullptr;
-    return arrow_column_to_ch_column->arrowTableToCHChunk(table, num_rows, block_missing_values_ptr);
+    throw Exception(500, "Not implemented");
 }
 
 void ORCBlockInputFormat::resetParser()
--- contrib/python/chdb/cpp/src/Processors/Formats/Impl/ArrowBlockInputFormat.cpp       (index)
+++ contrib/python/chdb/cpp/src/Processors/Formats/Impl/ArrowBlockInputFormat.cpp       (working tree)
@@ -62,11 +62,7 @@ Chunk ArrowBlockInputFormat::read()
 
         if (need_only_count)
         {
-            auto rows = file_reader->RecordBatchCountRows(record_batch_current++);
-            if (!rows.ok())
-                throw Exception(
-                    ErrorCodes::CANNOT_READ_ALL_DATA, "Error while reading batch of Arrow data: {}", rows.status().ToString());
-            return getChunkForCount(*rows);
+            throw Exception(500, "Not implemented");
         }
 
         batch_result = file_reader->ReadRecordBatch(record_batch_current);