--- contrib/python/chdb/cpp/src/Formats/ProtobufSerializer.cpp  (index)
+++ contrib/python/chdb/cpp/src/Formats/ProtobufSerializer.cpp  (working tree)
@@ -1785,7 +1785,7 @@ namespace
                 return parse<UUID>(text_buffer);
             };
 
-            default_function = [this]() -> UUID { return parse<UUID>(field_descriptor.default_value_string()); };
+            default_function = [this]() -> UUID { return parse<UUID>(String(field_descriptor.default_value_string())); };
         }
 
         static void uuidToString(const UUID & uuid, String & str)
@@ -1868,7 +1868,7 @@ namespace
                 return value;
             };
 
-            default_function = [this]() -> IPv6 { return parse<IPv6>(field_descriptor.default_value_string()); };
+            default_function = [this]() -> IPv6 { return parse<IPv6>(String(field_descriptor.default_value_string())); };
         }
 
         std::function<void(IPv6)> write_function;
--- contrib/python/chdb/cpp/src/Formats/ProtobufSchemas.cpp     (index)
+++ contrib/python/chdb/cpp/src/Formats/ProtobufSchemas.cpp     (working tree)
@@ -46,7 +46,7 @@ public:
         if (descriptor)
             return descriptor;
 
-        const auto * file_descriptor = importer.Import(schema_path);
+        const auto * file_descriptor = importer.Import(TProtoStringType(schema_path));
         if (error)
         {
             auto info = error.value();
@@ -89,7 +89,7 @@ public:
 
 private:
     // Overrides google::protobuf::compiler::MultiFileErrorCollector:
-    void AddError(const String & filename, int line, int column, const String & message) override
+    void AddError(const TProtoStringType & filename, int line, int column, const TProtoStringType & message) override
     {
         /// Protobuf library code is not exception safe, we should
         /// remember the error and throw it later from our side.
@@ -102,10 +102,10 @@ private:
 
     struct ErrorInfo
     {
-        String filename;
+        TProtoStringType filename;
         int line;
         int column;
-        String message;
+        TProtoStringType message;
     };
 
     std::optional<ErrorInfo> error;
--- contrib/python/chdb/cpp/contrib/orc/c++/src/sargs/PredicateLeaf.cc  (index)
+++ contrib/python/chdb/cpp/contrib/orc/c++/src/sargs/PredicateLeaf.cc  (working tree)
@@ -502,6 +502,16 @@ namespace orc_chdb {
     return result;
   }
 
+  static std::vector<TProtoStringType> literal2TProtoStringType(const std::vector<Literal>& values) {
+    std::vector<TProtoStringType> result;
+    std::for_each(values.cbegin(), values.cend(), [&](const Literal& val) {
+      if (!val.isNull()) {
+        result.emplace_back(TProtoStringType(val.getString()));
+      }
+    });
+    return result;
+  }
+
   TruthValue PredicateLeaf::evaluatePredicateMinMax(const proto::ColumnStatistics& colStats) const {
     TruthValue result = TruthValue::YES_NO_NULL;
     switch (mType) {
@@ -532,7 +542,7 @@ namespace orc_chdb {
         if (colStats.has_stringstatistics() && colStats.stringstatistics().has_minimum() &&
             colStats.stringstatistics().has_maximum()) {
           const auto& stats = colStats.stringstatistics();
-          result = evaluatePredicateRange(mOperator, literal2String(mLiterals), stats.minimum(),
+          result = evaluatePredicateRange(mOperator, literal2TProtoStringType(mLiterals), stats.minimum(),
                                           stats.maximum(), colStats.hasnull());
         }
         break;
--- contrib/python/chdb/cpp/contrib/orc/c++/src/Writer.cc       (index)
+++ contrib/python/chdb/cpp/contrib/orc/c++/src/Writer.cc       (working tree)
@@ -686,7 +686,7 @@ namespace orc_chdb {
     for (uint64_t i = 0; i < t.getSubtypeCount(); ++i) {
       // only add subtypes' field names if this type is STRUCT
       if (t.getKind() == STRUCT) {
-        footer.mutable_types(pos)->add_fieldnames(t.getFieldName(i));
+        footer.mutable_types(pos)->add_fieldnames(TProtoStringType(t.getFieldName(i)));
       }
       footer.mutable_types(pos)->add_subtypes(++index);
       buildFooterType(*t.getSubtype(i), footer, index);
--- contrib/python/chdb/cpp/contrib/orc/c++/src/Reader.cc       (index)
+++ contrib/python/chdb/cpp/contrib/orc/c++/src/Reader.cc       (working tree)
@@ -564,7 +564,7 @@ namespace orc_chdb {
     mutableFooter->CopyFrom(*footer);
     tail.set_filelength(fileLength);
     tail.set_postscriptlength(postscriptLength);
-    std::string result;
+    TProtoStringType result;
     if (!tail.SerializeToString(&result)) {
       throw ParseError("Failed to serialize file tail");
     }
