--- contrib/deprecated/python/celery/celery/app/amqp.py	(index)
+++ contrib/deprecated/python/celery/celery/app/amqp.py	(working tree)
@@ -3,7 +3,11 @@
 from __future__ import absolute_import, unicode_literals
 
 import numbers
-from collections import Mapping, namedtuple
+from collections import namedtuple
+try:
+    from collections.abc import Mapping
+except ImportError:
+    from collections import Mapping
 from datetime import timedelta
 from weakref import WeakValueDictionary
 
--- contrib/deprecated/python/celery/celery/canvas.py	(index)
+++ contrib/deprecated/python/celery/celery/canvas.py	(working tree)
@@ -442,6 +442,12 @@ class Signature(dict):
             return _chain(self, other, app=self._app)
         return NotImplemented
 
+    def __ior__(self, other):
+        # Python 3.9 introduces | as the merge operator for dicts.
+        # We override the in-place version of that operator
+        # so that canvases continue to work as they did before.
+        return self.__or__(other)
+
     def election(self):
         type = self.type
         app = type.app
--- contrib/deprecated/python/celery/celery/utils/saferepr.py	(index)
+++ contrib/deprecated/python/celery/celery/utils/saferepr.py	(working tree)
@@ -208,7 +208,7 @@ def _reprseq(val, lit_start, lit_end, builtin_type, chainer):
 
 def reprstream(stack, seen=None, maxlevels=3, level=0, isinstance=isinstance):
     """Streaming repr, yielding tokens."""
-    # type: (deque, Set, int, int, Callable) -> Iterator[Any]
+    # (broken) type: (deque, Set, int, int, Callable) -> Iterator[Any]
     seen = seen or set()
     append = stack.append
     popleft = stack.popleft
--- contrib/deprecated/python/celery/celery/utils/timer2.py	(index)
+++ contrib/deprecated/python/celery/celery/utils/timer2.py	(working tree)
@@ -102,7 +102,7 @@ class Timer(threading.Thread):
             self.running = False
 
     def ensure_started(self):
-        if not self.running and not self.isAlive():
+        if not self.running and not self.is_alive():
             if self.on_start:
                 self.on_start(self)
             self.start()
