--- contrib/python/django-closuretree/closuretree/models.py	(index)
+++ contrib/python/django-closuretree/closuretree/models.py	(working tree)
@@ -30,7 +30,7 @@ from django.db import models
 from django.db.models.base import ModelBase
 from django.db.models.signals import post_save, pre_delete
 from django.dispatch import receiver
-from django.utils.six import with_metaclass
+from six import with_metaclass
 import sys
 
 def _closure_model_unicode(self):
