commit 2e34e623f4199a0e97f68fc8919867a2f6d84bae
author: pulkit
date: 2019-12-13T18:12:32+03:00

    [libgit2] introduce public function to apply patch to buffer

--- a/include/git2/apply.h
+++ b/include/git2/apply.h
@@ -124,6 +124,20 @@
 	git_apply_location_t location,
 	const git_apply_options *options);
 
+/**
+ * Apply a patch to a buffer.
+ *
+ * This is not part of official libgit2 and is patched by Arc team.
+ */
+GIT_EXTERN(int) git_apply_to_buf(
+    git_buf *contents_out,
+    char **filename_out,
+    unsigned int *mode_out,
+    const char *source,
+    size_t source_len,
+    git_diff *diff,
+    const git_apply_options *given_opts);
+
 /** @} */
 GIT_END_DECL
 #endif
--- a/src/libgit2/apply.c
+++ b/src/libgit2/apply.c
@@ -363,6 +363,27 @@
 	return error;
 }
 
+int git_apply_to_buf(
+	git_buf *contents_out,
+	char **filename_out,
+	unsigned int *mode_out,
+	const char *source,
+	size_t source_len,
+	git_diff *diff,
+	const git_apply_options *given_opts)
+{
+    int error = 0;
+    git_patch *patch;
+    error = git_patch_from_diff(&patch, diff, 0);
+    if (error) {
+        git_patch_free(patch);
+        return error;
+    }
+    error = git_apply__patch(contents_out, filename_out, mode_out, source, source_len, patch, given_opts);
+    git_patch_free(patch);
+    return error;
+}
+
 int git_apply__patch(
 	git_str *contents_out,
 	char **filename_out,
