diff --git a/src/libgit2/signature.c b/src/libgit2/signature.c
--- a/src/libgit2/signature.c
+++ b/src/libgit2/signature.c
@@ -42,7 +42,7 @@ static bool contains_angle_brackets(const char *input)
 
 static bool is_crud(unsigned char c)
 {
-	return  c <= 32  ||
+	return  c < 32  ||
 		c == ',' ||
 		c == ':' ||
 		c == ';' ||
@@ -225,6 +225,15 @@ int git_signature__parse(git_signature *sig, const char **buffer_out,
 
 	email_start += 1;
 	sig->name = extract_trimmed(buffer, email_start - buffer - 1);
+
+	// Cut only one last space
+	// "Linus Torvalds <torvalds@ppc970.osdl.org>" -> "Linus Torvalds"
+	// "Linus Torvalds  <torvalds@ppc970.osdl.org>" -> "Linus Torvalds "
+	const size_t name_len = strlen(sig->name);
+	if (name_len > 0 && isspace(sig->name[name_len - 1])) {
+		sig->name[name_len - 1] = 0;
+	}
+
 	sig->email = extract_trimmed(email_start, email_end - email_start);
 
 	/* Do we even have a time at the end of the signature? */
