https://github.com/click-contrib/click-repl/pull/109
--- contrib/python/click-repl/click_repl/_completer.py	(index)
+++ contrib/python/click-repl/click_repl/_completer.py	(working tree)
@@ -254,7 +254,10 @@ class ClickCompleter(Completer):
 
         if self.parsed_args != args:
             self.parsed_args = args
-            self.parsed_ctx = _resolve_context(args, self.ctx)
+            try:
+                self.parsed_ctx = _resolve_context(args, self.ctx)
+            except Exception:
+                return []
             self.ctx_command = self.parsed_ctx.command
 
         if getattr(self.ctx_command, "hidden", False):
