--- contrib/restricted/boost/regex/ya.make	(index)
+++ contrib/restricted/boost/regex/ya.make	(working tree)
@@ -16,7 +16,6 @@ VERSION(1.87.0)
 ORIGINAL_SOURCE(https://github.com/boostorg/regex/archive/boost-1.87.0.tar.gz)
 
 PEERDIR(
-    contrib/libs/icu
     contrib/restricted/boost/assert
     contrib/restricted/boost/concept_check
     contrib/restricted/boost/config
@@ -40,7 +39,6 @@ NO_COMPILER_WARNINGS()
 NO_UTIL()
 
 CFLAGS(
-    -DBOOST_HAS_ICU
     -DBOOST_NO_CXX98_BINDERS
 )
 
--- contrib/restricted/thrift/ya.make	(index)
+++ contrib/restricted/thrift/ya.make	(working tree)
@@ -24,7 +24,6 @@ PEERDIR(
     contrib/libs/openssl
     contrib/libs/zlib
     contrib/restricted/boost/interprocess
-    contrib/restricted/boost/locale
     contrib/restricted/boost/math
     contrib/restricted/boost/system
     contrib/restricted/boost/thread
@@ -49,7 +48,6 @@ SRCS(
     thrift/protocol/TBase64Utils.cpp
     thrift/protocol/TDebugProtocol.cpp
     thrift/protocol/THeaderProtocol.cpp
-    thrift/protocol/TJSONProtocol.cpp
     thrift/protocol/TMultiplexedProtocol.cpp
     thrift/protocol/TProtocol.cpp
     thrift/server/TConnectedClient.cpp
