--- contrib/python/types-protobuf/google-stubs/protobuf/descriptor.pyi	(index)
+++ contrib/python/types-protobuf/google-stubs/protobuf/descriptor.pyi	(working tree)
@@ -162,6 +162,7 @@ class FieldDescriptor(DescriptorBase):
     number: int
     type: int
     cpp_type: int
+    label: int
     @property
     def is_required(self) -> bool: ...
     @property
--- contrib/python/types-protobuf/google-stubs/protobuf/json_format.pyi	(index)
+++ contrib/python/types-protobuf/google-stubs/protobuf/json_format.pyi	(working tree)
@@ -12,24 +12,22 @@ class EnumStringValueParseError(ParseError): ...
 
 def MessageToJson(
     message: Message,
+    including_default_value_fields: bool = False,
     preserving_proto_field_name: bool = False,
     indent: int | None = 2,
     sort_keys: bool = False,
     use_integers_for_enums: bool = False,
     descriptor_pool: DescriptorPool | None = None,
+    float_precision: int | None = None,
     ensure_ascii: bool = True,
-    always_print_fields_with_no_presence: bool = False,
-    *,
-    unquote_int64_if_possible: bool = False,
 ) -> str: ...
 def MessageToDict(
     message: Message,
-    always_print_fields_with_no_presence: bool = False,
+    including_default_value_fields: bool = False,
     preserving_proto_field_name: bool = False,
     use_integers_for_enums: bool = False,
     descriptor_pool: DescriptorPool | None = None,
-    *,
-    unquote_int64_if_possible: bool = False,
+    float_precision: int | None = None,
 ) -> dict[str, Any]: ...
 def Parse(
     text: bytes | str,
--- contrib/python/types-protobuf/google-stubs/protobuf/message_factory.pyi	(index)
+++ contrib/python/types-protobuf/google-stubs/protobuf/message_factory.pyi	(working tree)
@@ -11,5 +11,8 @@ def GetMessageClassesForFiles(files: Iterable[str], pool: DescriptorPool) -> dic
 class MessageFactory:
     pool: DescriptorPool
     def __init__(self, pool: DescriptorPool | None = None) -> None: ...
+    def GetPrototype(self, descriptor: Descriptor) -> type[Message]: ...
+    def CreatePrototype(self, descriptor: Descriptor) -> type[Message]: ...
+    def GetMessages(self, files: Iterable[str]) -> dict[str, type[Message]]: ...
 
 def GetMessages(file_protos: Iterable[FileDescriptorProto], pool: DescriptorPool | None = None) -> dict[str, type[Message]]: ...
