--- contrib/python/Markdown/py3/tests/test_legacy.py	(index)
+++ contrib/python/Markdown/py3/tests/test_legacy.py	(working tree)
@@ -32,1 +32,2 @@ warnings.simplefilter('error')
-parent_test_dir = os.path.abspath(os.path.dirname(__file__))
+import yatest.common
+parent_test_dir = yatest.common.test_source_path()
--- contrib/python/Markdown/py3/tests/test_syntax/extensions/test_md_in_html.py	(index)
+++ contrib/python/Markdown/py3/tests/test_syntax/extensions/test_md_in_html.py	(working tree)
@@ -23,1 +23,1 @@ Copyright 2004 Manfred Stienstra (the original version)
-from unittest import TestSuite
+#from unittest import TestSuite
