From 53cef01075f50f6f5e0a56bea5a4a288f1da642b Mon Sep 17 00:00:00 2001
From: Kanak Kshetri <kanakkshetri@fastmail.fm>
Date: Thu, 30 Nov 2023 19:45:12 -0600
Subject: [PATCH] wyhash.h: add missing static inline

---
 wyhash.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/wyhash.h b/wyhash.h
index f066782..5e66d8b 100644
--- a/wyhash.h
+++ b/wyhash.h
@@ -200,7 +200,7 @@ static inline unsigned long long pow_mod(unsigned long long a, unsigned long lon
     }
     return r;
 }
-unsigned sprp(unsigned long long n, unsigned long long a) {
+static inline unsigned sprp(unsigned long long n, unsigned long long a) {
     unsigned long long d=n-1;
     unsigned char s=0;
     while (!(d & 0xff)) { d>>=8; s+=8; }
@@ -217,7 +217,7 @@ unsigned sprp(unsigned long long n, unsigned long long a) {
     }
     return 0;
 }
-unsigned is_prime(unsigned long long n) {
+static inline unsigned is_prime(unsigned long long n) {
     if (n<2||!(n&1)) return 0;
     if (n<4) return 1;
     if (!sprp(n,2)) return 0;
