--- contrib/python/paramiko/py2/paramiko/common.py	(index)
+++ contrib/python/paramiko/py2/paramiko/common.py	(working tree)
@@ -29,7 +29,8 @@ from paramiko.py3compat import byte_chr, PY2, long, b
     MSG_DEBUG,
     MSG_SERVICE_REQUEST,
     MSG_SERVICE_ACCEPT,
-) = range(1, 7)
+    MSG_EXT_INFO,
+) = range(1, 8)
 (MSG_KEXINIT, MSG_NEWKEYS) = range(20, 22)
 (
     MSG_USERAUTH_REQUEST,
@@ -68,6 +69,7 @@ cMSG_UNIMPLEMENTED = byte_chr(MSG_UNIMPLEMENTED)
 cMSG_DEBUG = byte_chr(MSG_DEBUG)
 cMSG_SERVICE_REQUEST = byte_chr(MSG_SERVICE_REQUEST)
 cMSG_SERVICE_ACCEPT = byte_chr(MSG_SERVICE_ACCEPT)
+cMSG_EXT_INFO = byte_chr(MSG_EXT_INFO)
 cMSG_KEXINIT = byte_chr(MSG_KEXINIT)
 cMSG_NEWKEYS = byte_chr(MSG_NEWKEYS)
 cMSG_USERAUTH_REQUEST = byte_chr(MSG_USERAUTH_REQUEST)
@@ -108,6 +110,7 @@ MSG_NAMES = {
     MSG_DEBUG: "debug",
     MSG_SERVICE_REQUEST: "service-request",
     MSG_SERVICE_ACCEPT: "service-accept",
+    MSG_EXT_INFO: "ext-info",
     MSG_KEXINIT: "kexinit",
     MSG_NEWKEYS: "newkeys",
     30: "kex30",
--- contrib/python/paramiko/py2/paramiko/transport.py	(index)
+++ contrib/python/paramiko/py2/paramiko/transport.py	(working tree)
@@ -53,6 +53,7 @@ from paramiko.common import (
     cMSG_UNIMPLEMENTED,
     INFO,
     cMSG_KEXINIT,
+    cMSG_EXT_INFO,
     cMSG_NEWKEYS,
     MSG_NEWKEYS,
     cMSG_REQUEST_SUCCESS,
@@ -178,6 +179,7 @@ class Transport(threading.Thread, ClosingContextManager):
         "ssh-rsa",
         "ssh-dss",
     )
+    _kex_ext_info_c = "ext-info-c"
     _preferred_kex = (
         "ecdh-sha2-nistp256",
         "ecdh-sha2-nistp384",
@@ -455,6 +457,8 @@ class Transport(threading.Thread, ClosingContextManager):
         self.kex_engine = None
         self.H = None
         self.K = None
+        # server flag indicating that client supports MSG2_EXT_INFO
+        self.ext_info_c_supported = None
 
         self.initial_kex_done = False
         self.in_kex = False
@@ -2363,6 +2367,9 @@ class Transport(threading.Thread, ClosingContextManager):
         self.kex_engine = self._kex_info[agreed_kex[0]](self)
         self._log(DEBUG, "Kex agreed: {}".format(agreed_kex[0]))
 
+        self.ext_info_c_supported = self._kex_ext_info_c in kex_algo_list
+        self._log(DEBUG, "ext-info-c agreed: {}".format(self.ext_info_c_supported))
+
         if self.server_mode:
             available_server_keys = list(
                 filter(
@@ -2579,6 +2586,14 @@ class Transport(threading.Thread, ClosingContextManager):
             self.packetizer.set_outbound_compressor(compress_out())
         if not self.packetizer.need_rekey():
             self.in_kex = False
+        if self.server_mode and self.ext_info_c_supported:
+            # RFC 8308
+            m = Message()
+            m.add_byte(cMSG_EXT_INFO)
+            m.add_int(1)
+            m.add_string("server-sig-algs")
+            m.add_string("rsa-sha2-256,rsa-sha2-512")
+            self._send_message(m)
         # we always expect to receive NEWKEYS now
         self._expect_packet(MSG_NEWKEYS)
 
