--- contrib/python/bson/py3/bson/objectid.py	(d3136ea390bf7a768368f88b9559af5baadef08e)
+++ contrib/python/bson/py3/bson/objectid.py	(index)
@@ -27,8 +27,8 @@ import struct
 import threading
 import time

-from bson.py3compat import PY3, bytes_from_hex, string_type, text_type
-from bson.tz_util import utc
+from .py3compat import PY3, bytes_from_hex, string_type, text_type
+from .tz_util import utc


 # fnv_1a_24 adaptation taken from MongoDB Python Driver at https://github.com/mongodb/mongo-python-driver/commit/61850357a0e0eeec1a30e1adc0bbf7ebee807358

