--- contrib/python/ibm-db/py2/ibm_db.c	(5a69d9e6ca0b7eba642278c3788c2cefb44c061c)
+++ contrib/python/ibm-db/py2/ibm_db.c	(index)
@@ -1723,7 +1723,7 @@ static int _python_ibm_db_set_decfloat_rounding_mode_client(SQLHANDLE hdbc)
     if (strcmp(decflt_rounding, "ROUND_DOWN") == 0) rounding_mode = ROUND_DOWN;
     if (strcmp(decflt_rounding, "ROUND_CEILING") == 0) rounding_mode = ROUND_CEILING;
     if (strcmp(decflt_rounding, "ROUND_FLOOR") == 0) rounding_mode = ROUND_FLOOR;
-	
+
     Py_BEGIN_ALLOW_THREADS;
 #ifndef PASE
     rc = SQLSetConnectAttr(hdbc, SQL_ATTR_DECFLOAT_ROUNDING_MODE, (SQLPOINTER)rounding_mode, SQL_NTS);
@@ -1731,7 +1731,7 @@ static int _python_ibm_db_set_decfloat_rounding_mode_client(SQLHANDLE hdbc)
     rc = SQLSetConnectAttr(hdbc, SQL_ATTR_DECFLOAT_ROUNDING_MODE, (SQLPOINTER)&rounding_mode, SQL_NTS);
 #endif
     Py_END_ALLOW_THREADS;
-	
+
     return rc;
 
 }
@@ -5683,7 +5683,7 @@ static int _python_ibm_db_bind_data( stmt_handle *stmt_res, param_node *curr, Py
                                 curr->param_type, SQL_C_LONG, curr->data_type,
                                 curr->param_size, curr->scale, curr->ivalueArray, 0, NULL);
                     Py_END_ALLOW_THREADS;
-                }                
+                }
                 else
                 {
                     curr->ivalue = (SQLINTEGER) PyLong_AsLong(bind_data);
@@ -5736,7 +5736,7 @@ static int _python_ibm_db_bind_data( stmt_handle *stmt_res, param_node *curr, Py
                 _python_ibm_db_check_sql_errors(stmt_res->hstmt, SQL_HANDLE_STMT,
                                                     rc, 1, NULL, -1, 1);
             }
-            
+
             curr->data_type = SQL_C_LONG;
             break;
 
@@ -5756,7 +5756,7 @@ static int _python_ibm_db_bind_data( stmt_handle *stmt_res, param_node *curr, Py
                             curr->param_type, SQL_C_LONG, curr->data_type,
                             curr->param_size, curr->scale, curr->ivalueArray, 0, NULL);
                 Py_END_ALLOW_THREADS;
-            }            
+            }
             else
             {
                 curr->ivalue = 1;
@@ -5818,7 +5818,7 @@ static int _python_ibm_db_bind_data( stmt_handle *stmt_res, param_node *curr, Py
                     int isNewBuffer, param_size = 0;
                     Py_ssize_t n = PyList_Size(bind_data);
                     curr->uvalue = (SQLWCHAR *)ALLOC_N(SQLWCHAR, curr->param_size * (n));
-                    curr->bind_indicator_array = (SQLINTEGER *) ALLOC_N(SQLINTEGER, n);         
+                    curr->bind_indicator_array = (SQLINTEGER *) ALLOC_N(SQLINTEGER, n);
                     memset(curr->uvalue , 0, sizeof(SQLWCHAR) * curr->param_size * n);
 
                     for (i = 0; i < n; i++)
@@ -5828,8 +5828,8 @@ static int _python_ibm_db_bind_data( stmt_handle *stmt_res, param_node *curr, Py
                         tmp_uvalue = NULL;
                         dest_uvalue = NULL;
 
-                        if(PyObject_CheckBuffer(item) && (curr->data_type == SQL_BLOB   || 
-                                                          curr->data_type == SQL_BINARY || 
+                        if(PyObject_CheckBuffer(item) && (curr->data_type == SQL_BLOB   ||
+                                                          curr->data_type == SQL_BINARY ||
                                                           curr->data_type == SQL_VARBINARY) )
                         {
 #if  PY_MAJOR_VERSION >= 3
@@ -5842,7 +5842,7 @@ static int _python_ibm_db_bind_data( stmt_handle *stmt_res, param_node *curr, Py
                             curr->ivalue = buffer_len;
 #endif
                         }
-                        else 
+                        else
                         {
                             if(tmp_uvalue != NULL)
                             {
@@ -5867,7 +5867,7 @@ static int _python_ibm_db_bind_data( stmt_handle *stmt_res, param_node *curr, Py
 #ifndef PASE /* i5/OS SQL_LONGVARBINARY is SQL_VARBINARY */
                                         || (curr->data_type == SQL_LONGVARBINARY)
 #endif /* PASE */
-                                        || (curr->data_type == SQL_VARBINARY) || (curr->data_type == SQL_XML)) 
+                                        || (curr->data_type == SQL_VARBINARY) || (curr->data_type == SQL_XML))
                                 {
                                     if (curr->ivalue <= curr->param_size)
                                     {
@@ -5890,11 +5890,11 @@ static int _python_ibm_db_bind_data( stmt_handle *stmt_res, param_node *curr, Py
                             dest_uvalue = memcpy(dest_uvalue, tmp_uvalue, (param_length + sizeof(SQLWCHAR)));
                             param_size = curr->param_size;
                         }
-                        else if (param_length <= curr->param_size) 
+                        else if (param_length <= curr->param_size)
                         {
                             dest_uvalue = (char *)&curr->uvalue[0] + (curr->param_size * i);
                             dest_uvalue = memcpy(dest_uvalue, tmp_uvalue, (param_length + sizeof(SQLWCHAR)));
-                            PyMem_Del(tmp_uvalue);                  
+                            PyMem_Del(tmp_uvalue);
                             param_size = curr->param_size;
                         }
                         else if(curr->data_type == SQL_TYPE_TIMESTAMP)
@@ -5986,8 +5986,8 @@ static int _python_ibm_db_bind_data( stmt_handle *stmt_res, param_node *curr, Py
                     }
 
                     Py_BEGIN_ALLOW_THREADS;
-                    rc = SQLBindParameter(stmt_res->hstmt, curr->param_num, curr->param_type, valueType, 
-                            curr->data_type, curr->param_size, curr->scale, paramValuePtr, 
+                    rc = SQLBindParameter(stmt_res->hstmt, curr->param_num, curr->param_type, valueType,
+                            curr->data_type, curr->param_size, curr->scale, paramValuePtr,
                             param_size, &curr->bind_indicator_array[0]);
                     Py_END_ALLOW_THREADS;
 
@@ -6151,7 +6151,7 @@ static int _python_ibm_db_bind_data( stmt_handle *stmt_res, param_node *curr, Py
 
                         tmp_svalue = NULL;
                         dest_svalue = NULL;
-                        if (PyObject_CheckBuffer(item) && (curr->data_type == SQL_BLOB      || 
+                        if (PyObject_CheckBuffer(item) && (curr->data_type == SQL_BLOB      ||
                                                            curr->data_type == SQL_BINARY   ||
                                                            curr->data_type == SQL_VARBINARY) )
                         {
@@ -6225,13 +6225,13 @@ static int _python_ibm_db_bind_data( stmt_handle *stmt_res, param_node *curr, Py
                                 break;
 
                             case SQL_BLOB:
-                                if (curr->param_type == SQL_PARAM_OUTPUT || curr->param_type == SQL_PARAM_INPUT_OUTPUT) 
+                                if (curr->param_type == SQL_PARAM_OUTPUT || curr->param_type == SQL_PARAM_INPUT_OUTPUT)
                                 {
                                     curr->ivalue = curr->ivalue - 1;
                                     curr->bind_indicator_array[i] = param_length;
                                     paramValuePtr = (SQLPOINTER)curr->svalue;
-                                } 
-                                else 
+                                }
+                                else
                                 {
                                     curr->bind_indicator_array[i] = curr->ivalue;
 #ifndef PASE
@@ -6251,7 +6251,7 @@ static int _python_ibm_db_bind_data( stmt_handle *stmt_res, param_node *curr, Py
                             case SQL_XML:
                                 /* account for bin_mode settings as well */
                                 curr->bind_indicator_array[i] = curr->ivalue;
-                                if (curr->param_type == SQL_PARAM_OUTPUT || curr->param_type == SQL_PARAM_INPUT_OUTPUT) 
+                                if (curr->param_type == SQL_PARAM_OUTPUT || curr->param_type == SQL_PARAM_INPUT_OUTPUT)
                                 {
                                     curr->ivalue = curr->ivalue - 1;
                                     curr->bind_indicator_array[i] = param_length;
@@ -6267,7 +6267,7 @@ static int _python_ibm_db_bind_data( stmt_handle *stmt_res, param_node *curr, Py
                             case SQL_TYPE_TIMESTAMP:
                                 valueType = SQL_C_CHAR;
                                 curr->bind_indicator_array[i] = curr->ivalue;
-                                if (curr->param_type == SQL_PARAM_OUTPUT || curr->param_type == SQL_PARAM_INPUT_OUTPUT) 
+                                if (curr->param_type == SQL_PARAM_OUTPUT || curr->param_type == SQL_PARAM_INPUT_OUTPUT)
                                 {
                                     if( param_length == 0)
                                     {
@@ -6288,7 +6288,7 @@ static int _python_ibm_db_bind_data( stmt_handle *stmt_res, param_node *curr, Py
                             default:
                                 valueType = SQL_C_CHAR;
                                 curr->bind_indicator_array[i] = curr->ivalue;
-                                if (curr->param_type == SQL_PARAM_OUTPUT || curr->param_type == SQL_PARAM_INPUT_OUTPUT) 
+                                if (curr->param_type == SQL_PARAM_OUTPUT || curr->param_type == SQL_PARAM_INPUT_OUTPUT)
                                 {
                                     curr->bind_indicator_array[i] = SQL_NTS;
                                 }
@@ -6311,7 +6311,7 @@ static int _python_ibm_db_bind_data( stmt_handle *stmt_res, param_node *curr, Py
                 else /* To bind scalar values */
                 {
                     char* tmp;
-                    if (PyObject_CheckBuffer(bind_data) && (curr->data_type == SQL_BLOB   || 
+                    if (PyObject_CheckBuffer(bind_data) && (curr->data_type == SQL_BLOB   ||
                                                             curr->data_type == SQL_BINARY ||
                                                             curr->data_type == SQL_VARBINARY) )
                     {
@@ -6341,12 +6341,12 @@ static int _python_ibm_db_bind_data( stmt_handle *stmt_res, param_node *curr, Py
                     * string returned. The string is then truncate past that size.
                     * If no size is given then use BUFSIZ to return the string.
                     */
-                    if (curr->size != 0) 
+                    if (curr->size != 0)
                     {
                         curr->ivalue = curr->size;
                     }
 
-                    if (curr->param_type == SQL_PARAM_OUTPUT || curr->param_type == SQL_PARAM_INPUT_OUTPUT) 
+                    if (curr->param_type == SQL_PARAM_OUTPUT || curr->param_type == SQL_PARAM_INPUT_OUTPUT)
                     {
                         if (curr->size == 0)
                         {
@@ -6371,7 +6371,7 @@ static int _python_ibm_db_bind_data( stmt_handle *stmt_res, param_node *curr, Py
                             {
                                 curr->bind_indicator = param_length;
                                 paramValuePtr = (SQLPOINTER)curr->svalue;
-                            } 
+                            }
                             else
                             {
                                 curr->bind_indicator = SQL_DATA_AT_EXEC;
@@ -6477,7 +6477,7 @@ static int _python_ibm_db_bind_data( stmt_handle *stmt_res, param_node *curr, Py
             if (curr->data_type == SQL_DECIMAL || curr->data_type == SQL_DECFLOAT)
             {
                 if(TYPE(bind_data) == PYTHON_LIST)
-                {   
+                {
                     char *svalue = NULL;
                     Py_ssize_t n = PyList_Size(bind_data);
                     int max_precn = 0;
@@ -6512,7 +6512,7 @@ static int _python_ibm_db_bind_data( stmt_handle *stmt_res, param_node *curr, Py
 #endif
                         svalue = PyBytes_AsString(tempobj);
                         curr->ivalue = strlen(svalue);
-                        memcpy(curr->svalue + (i * max_precn), svalue, curr->ivalue);           
+                        memcpy(curr->svalue + (i * max_precn), svalue, curr->ivalue);
                         valueType = SQL_C_CHAR;
                         paramValuePtr = (SQLPOINTER)(curr->svalue);
                         curr->bind_indicator_array[i] = curr->ivalue;
@@ -6520,8 +6520,8 @@ static int _python_ibm_db_bind_data( stmt_handle *stmt_res, param_node *curr, Py
                     }
 
                     Py_BEGIN_ALLOW_THREADS;
-                    rc = SQLBindParameter(stmt_res->hstmt, curr->param_num, curr->param_type, valueType, 
-                            curr->data_type, curr->param_size, curr->scale, 
+                    rc = SQLBindParameter(stmt_res->hstmt, curr->param_num, curr->param_type, valueType,
+                            curr->data_type, curr->param_size, curr->scale,
                             paramValuePtr, max_precn, curr->bind_indicator_array);
                     Py_END_ALLOW_THREADS;
 
@@ -6556,8 +6556,8 @@ static int _python_ibm_db_bind_data( stmt_handle *stmt_res, param_node *curr, Py
                     curr->bind_indicator = curr->ivalue;
 
                     Py_BEGIN_ALLOW_THREADS;
-                    rc = SQLBindParameter(stmt_res->hstmt, curr->param_num, curr->param_type, valueType, 
-                            curr->data_type, curr->param_size, curr->scale, 
+                    rc = SQLBindParameter(stmt_res->hstmt, curr->param_num, curr->param_type, valueType,
+                            curr->data_type, curr->param_size, curr->scale,
                             paramValuePtr, curr->ivalue, &(curr->bind_indicator));
                     Py_END_ALLOW_THREADS;
 
@@ -6573,7 +6573,7 @@ static int _python_ibm_db_bind_data( stmt_handle *stmt_res, param_node *curr, Py
 
         case PYTHON_DATE:
             if(TYPE(bind_data) == PYTHON_LIST)
-            {   
+            {
                 Py_ssize_t n = PyList_Size(bind_data);
                 curr->date_value = ALLOC_N(DATE_STRUCT, n);
 
@@ -6643,12 +6643,12 @@ static int _python_ibm_db_bind_data( stmt_handle *stmt_res, param_node *curr, Py
 
         case PYTHON_TIMESTAMP:
             if(TYPE(bind_data) == PYTHON_LIST)
-            {   
+            {
                 Py_ssize_t n = PyList_Size(bind_data);
                 curr->ts_value = ALLOC_N(TIMESTAMP_STRUCT, n);
 
                 for (i = 0; i < n; i++)
-                {       
+                {
                     item = PyList_GetItem(bind_data, i);
                     (curr->ts_value + i)->year = PyDateTime_GET_YEAR(item);
                     (curr->ts_value + i)->month = PyDateTime_GET_MONTH(item);
@@ -7545,7 +7545,7 @@ static PyObject *ibm_db_next_result(PyObject *self, PyObject *args)
                 _python_ibm_db_check_sql_errors( stmt_res->hstmt, SQL_HANDLE_STMT,
                                                 rc, 1, NULL, -1, 1);
             }
-#ifndef __MVS__            
+#ifndef __MVS__
 	    Py_BEGIN_ALLOW_THREADS;
 		SQLFreeHandle(SQL_HANDLE_STMT, new_hstmt);
 		Py_END_ALLOW_THREADS;
@@ -8949,7 +8949,7 @@ static PyObject *ibm_db_result(PyObject *self, PyObject *args)
                          &out_length);
             if ( rc == SQL_ERROR ) {
                 PyErr_Clear();
-                Py_RETURN_FALSE; 
+                Py_RETURN_FALSE;
             }
             if (out_length == SQL_NULL_DATA) {
                 Py_RETURN_NONE;
