--- contrib/deprecated/python/audioop-lts/audioop/_audioop.c	(index)
+++ contrib/deprecated/python/audioop-lts/audioop/_audioop.c	(working tree)
@@ -1981,7 +1981,9 @@ audioop_exec(PyObject* module)
 static PyModuleDef_Slot audioop_slots[] = {
     {Py_mod_exec, audioop_exec},
     {Py_mod_multiple_interpreters, Py_MOD_PER_INTERPRETER_GIL_SUPPORTED},
+#ifdef Py_mod_gil  // Python 3.13+
     {Py_mod_gil, Py_MOD_GIL_NOT_USED},
+#endif
     {0, NULL}
 };
 
