diff --git a/PreLoad.cmake b/PreLoad.cmake
index 3437a89..4ee158b 100644
--- a/PreLoad.cmake
+++ b/PreLoad.cmake
@@ -42,7 +42,7 @@ if (NOT "$ENV{CFLAGS}" STREQUAL ""
     message("CMAKE_SHARED_LINKER_FLAGS_INIT: ${CMAKE_SHARED_LINKER_FLAGS_INIT}")
     message("CMAKE_MODULE_LINKER_FLAGS_INIT: ${CMAKE_MODULE_LINKER_FLAGS_INIT}")
 
-    message(FATAL_ERROR "
+    message("
         Some of the variables like CFLAGS, CXXFLAGS, LDFLAGS are not empty.
         It is not possible to build ClickHouse with custom flags.
         These variables can be set up by previous invocation of some other build tools.