diff --git a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -421,11 +421,6 @@
     target_compile_options(global-group INTERFACE "-Werror")
 endif ()
 
-# Make this extra-checks for correct library dependencies.
-if (OS_LINUX AND NOT SANITIZE)
-    target_link_options(global-group INTERFACE "LINKER:--no-undefined")
-endif ()
-
 ######################################
 ### Add targets below this comment ###
 ######################################
diff --git a/contrib/arrow-cmake/CMakeLists.txt b/contrib/arrow-cmake/CMakeLists.txt
--- a/contrib/arrow-cmake/CMakeLists.txt	(revision 7a4096f7bf1fd146df6ed1337656a6b925b3335e)
+++ b/contrib/arrow-cmake/CMakeLists.txt	(date 1756221795206)
@@ -29,12 +29,6 @@
     message (FATAL_ERROR "Using internal parquet library on FreeBSD is not supported")
 endif()
 
-if (OS_LINUX)
-    set (ARROW_JEMALLOC ON)
-    add_definitions(-DARROW_JEMALLOC=1)
-    message(STATUS "Using jemalloc in arrow lib")
-endif()
-
 set (CMAKE_CXX_STANDARD 17)
 
 set(ARROW_VERSION "11.0.0")
@@ -207,9 +201,6 @@
 target_include_directories(_orc SYSTEM BEFORE PUBLIC
     ${ORC_INCLUDE_DIR}
     "${ClickHouse_SOURCE_DIR}/contrib/arrow-cmake/cpp/src/orc/c++/include")
-if (OS_LINUX)
-    target_link_libraries(_orc PRIVATE ch_contrib::jemalloc)
-endif ()
 target_include_directories(_orc SYSTEM BEFORE PUBLIC ${ORC_BUILD_INCLUDE_DIR})
 target_include_directories(_orc SYSTEM PRIVATE
     ${ORC_SOURCE_SRC_DIR}
@@ -441,10 +432,6 @@
         "${ARROW_SRC_DIR}/arrow/adapters/orc/options.cc"
         )
 
-if (OS_LINUX)
-    SET(ARROW_SRCS "${LIBRARY_DIR}/memory_pool_jemalloc.cc" ${ARROW_SRCS})
-endif()
-
 add_definitions(-DARROW_WITH_LZ4)
 SET(ARROW_SRCS "${LIBRARY_DIR}/util/compression_lz4.cc" ${ARROW_SRCS})
 
@@ -475,9 +462,6 @@
     ch_contrib::zstd
     ch_contrib::brotli
 )
-if (OS_LINUX)
-    target_link_libraries(_arrow PRIVATE ch_contrib::jemalloc)
-endif ()
 
 target_link_libraries(_arrow PUBLIC _orc)
 
@@ -562,10 +546,6 @@
         boost::regex
         OpenSSL::Crypto OpenSSL::SSL)
 
-if (OS_LINUX)
-    target_link_libraries(_parquet PRIVATE ch_contrib::jemalloc)
-endif ()
-
 if (SANITIZE STREQUAL "undefined")
     target_compile_options(_parquet PRIVATE -fno-sanitize=undefined)
     target_compile_options(_arrow PRIVATE -fno-sanitize=undefined)