diff --git a/src/configure_config.cmake b/src/configure_config.cmake
index 5a1aa179e01..e9115509113 100644
--- a/src/configure_config.cmake
+++ b/src/configure_config.cmake
@@ -100,7 +100,7 @@ endif()
 if (TARGET ch_contrib::llvm)
     set(USE_EMBEDDED_COMPILER ${ENABLE_EMBEDDED_COMPILER})
     set(USE_DWARF_PARSER ${ENABLE_DWARF_PARSER})
-    set(USE_BLAKE3 ${ENABLE_LIBRARIES})
+    set(USE_BLAKE3 ${ENABLE_BLAKE3})
 endif()
 if (TARGET ch_contrib::unixodbc)
     set(USE_ODBC 1)