--- contrib/python/demjson/py3/demjson.py	(index)
+++ contrib/python/demjson/py3/demjson.py	(working tree)
@@ -367 +367 @@ class _undefined_class(object):
-    def __nonzero__(self):
+    def __bool__(self):
@@ -645 +645 @@ del _nonnumber_float_constants
-class json_int( (1L).__class__ ):    # Have to specify base this way to satisfy 2to3
+class json_int( (1).__class__ ):    # Have to specify base this way to satisfy 2to3
@@ -810 +810 @@ class utf32(codecs.CodecInfo):
-                import cStringIO as sio
+                import io as sio
@@ -812 +812 @@ class utf32(codecs.CodecInfo):
-                import StringIO as sio
+                import io as sio
@@ -927 +927 @@ class utf32(codecs.CodecInfo):
-                    chars.append( unichr(0xFFFD) )
+                    chars.append( chr(0xFFFD) )
@@ -943 +943 @@ class utf32(codecs.CodecInfo):
-        return (u''.join( chars ), num_bytes)
+        return (''.join( chars ), num_bytes)
@@ -963,2 +963,2 @@ def _make_unsafe_string_chars():
-    for c in [unichr(i) for i in range(0x100)]:
-        if c == u'"' or c == u'\\' \
+    for c in [chr(i) for i in range(0x100)]:
+        if c == '"' or c == '\\' \
@@ -967 +967 @@ def _make_unsafe_string_chars():
-    return u''.join( unsafe )
+    return ''.join( unsafe )
@@ -1032 +1032 @@ class helpers(object):
-                c = unichr(w1)
+                c = chr(w1)
@@ -1034 +1034 @@ class helpers(object):
-                c = unichr(w1) + unichr(w2)
+                c = chr(w1) + chr(w2)
@@ -1036 +1036 @@ class helpers(object):
-            c = unichr(codepoint)
+            c = chr(codepoint)
@@ -1042,3 +1042,3 @@ class helpers(object):
-        if not isinstance(c,unicode):
-            c = unicode(c)
-        if c in u' \t\n\r\f\v':
+        if not isinstance(c,str):
+            c = str(c)
+        if c in ' \t\n\r\f\v':
@@ -1061 +1061 @@ class helpers(object):
-        return c in u'\r\n\u2028\u2029'
+        return c in '\r\n\u2028\u2029'
@@ -1075 +1075 @@ class helpers(object):
-        return c.isalnum() or c in u'_$\u200c\u200d'
+        return c.isalnum() or c in '_$\u200c\u200d'
@@ -1104 +1104 @@ class helpers(object):
-        txt2 = filter( lambda c: unicodedata.category(unicode(c)) != 'Cf', txt )
+        txt2 = [c for c in txt if unicodedata.category(str(c)) != 'Cf']
@@ -1109,2 +1109,2 @@ class helpers(object):
-        if not isinstance(txt2,basestring):
-            txt2 = u''.join(txt2)
+        if not isinstance(txt2,str):
+            txt2 = ''.join(txt2)
@@ -1154 +1154 @@ class helpers(object):
-            if isinstance(x, basestring):
+            if isinstance(x, str):
@@ -1180 +1180 @@ class helpers(object):
-        if isinstance(z, basestring):
+        if isinstance(z, str):
@@ -1238 +1238 @@ class helpers(object):
-        if isinstance(txt, unicode):
+        if isinstance(txt, str):
@@ -1259 +1259 @@ class helpers(object):
-            if len(unitxt) > 0 and unitxt[0] == u'\uFEFF':
+            if len(unitxt) > 0 and unitxt[0] == '\uFEFF':
@@ -1262 +1262 @@ class helpers(object):
-            elif len(unitxt) > 0 and unitxt[0] == u'\uFFFE': # Reversed BOM
+            elif len(unitxt) > 0 and unitxt[0] == '\uFFFE': # Reversed BOM
@@ -1302 +1302 @@ class helpers(object):
-            return (unichr(w1),)
+            return (chr(w1),)
@@ -1304 +1304 @@ class helpers(object):
-            return (unichr(w1), unichr(w2))
+            return (chr(w1), chr(w2))
@@ -1321 +1321 @@ class helpers(object):
-        return isinstance(obj, (int,long,float)) \
+        return isinstance(obj, (int,float)) \
@@ -1359 +1359 @@ class helpers(object):
-        if isinstance(obj, basestring):
+        if isinstance(obj, str):
@@ -1362,3 +1362,3 @@ class helpers(object):
-        import types, UserString
-        return isinstance(obj, types.StringTypes) \
-               or isinstance(obj, UserString.UserString)
+        import types, collections
+        return isinstance(obj, (str,)) \
+               or isinstance(obj, collections.UserString)
@@ -1643 +1643 @@ class buffered_stream(object):
-        except IndexError, err:
+        except IndexError as err:
@@ -1672 +1672 @@ class buffered_stream(object):
-        self.__rawbuf = u''
+        self.__rawbuf = ''
@@ -1678 +1678 @@ class buffered_stream(object):
-        except Exception, err:
+        except Exception as err:
@@ -2066 +2066 @@ class JSONError(JSONException):
-        for kw,val in kwargs.items():
+        for kw,val in list(kwargs.items()):
@@ -2342 +2342 @@ class decode_statistics(object):
-                charname = unicodedata.name(unichr(self.min_codepoint))
+                charname = unicodedata.name(chr(self.min_codepoint))
@@ -2352 +2352 @@ class decode_statistics(object):
-                charname = unicodedata.name(unichr(self.max_codepoint))
+                charname = unicodedata.name(chr(self.max_codepoint))
@@ -2494 +2494 @@ class decode_state(object):
-        for kw, val in kwargs.items():
+        for kw, val in list(kwargs.items()):
@@ -2779 +2779 @@ def smart_sort_transform( key ):
-    elif isinstance( key, (int,long) ):
+    elif isinstance( key, int ):
@@ -2781 +2781 @@ def smart_sort_transform( key ):
-    elif isinstance( key, basestring ):
+    elif isinstance( key, str ):
@@ -2813 +2813 @@ except ImportError:
-class json_options(object):
+class json_options(object, metaclass=_behaviors_metaclass):
@@ -2815,2 +2814,0 @@ class json_options(object):
-
-    __metaclass__ = _behaviors_metaclass
@@ -2965 +2963 @@ class json_options(object):
-        for kw,val in kwargs.items():
+        for kw,val in list(kwargs.items()):
@@ -2976 +2974 @@ class json_options(object):
-                        self.always_escape_chars = set(u'<>/&')
+                        self.always_escape_chars = set('<>/&')
@@ -2978 +2976 @@ class json_options(object):
-                        self.always_escape_chars.update( set(u'<>/&') )
+                        self.always_escape_chars.update( set('<>/&') )
@@ -3004 +3002 @@ class json_options(object):
-                    elif isinstance(val,(int,long)) or val[0].isdigit:
+                    elif isinstance(val,int) or val[0].isdigit:
@@ -3011 +3009 @@ class json_options(object):
-                if isinstance(val,basestring):
+                if isinstance(val,str):
@@ -3180 +3178 @@ class json_options(object):
-        if isinstance(sign, (int,long)):
+        if isinstance(sign, int):
@@ -3185 +3183 @@ class json_options(object):
-        if isinstance(s,basestring):
+        if isinstance(s,str):
@@ -3192 +3190 @@ class json_options(object):
-            if isinstance(s, (int,long)):
+            if isinstance(s, int):
@@ -3216 +3214 @@ class json_options(object):
-        elif isinstance( s, (int,long) ):
+        elif isinstance( s, int ):
@@ -3236 +3234 @@ class json_options(object):
-        if isinstance(n,(int,long)) and self.keep_format:
+        if isinstance(n,int) and self.keep_format:
@@ -3249 +3247 @@ class json_options(object):
-        elif isinstance(sign, (int,long)):
+        elif isinstance(sign, int):
@@ -3277 +3275 @@ class json_options(object):
-        elif isinstance(sign, (int,long)):
+        elif isinstance(sign, int):
@@ -3299 +3297 @@ class json_options(object):
-        if isinstance(radix,basestring):
+        if isinstance(radix,str):
@@ -3418 +3416 @@ class JSON(object):
-    json_syntax_characters = u"{}[]\"\\,:0123456789.-+abcdefghijklmnopqrstuvwxyz \t\n\r"
+    json_syntax_characters = "{}[]\"\\,:0123456789.-+abcdefghijklmnopqrstuvwxyz \t\n\r"
@@ -3461,2 +3459,2 @@ class JSON(object):
-                 and not self._rev_escapes.has_key(chr(c)) \
-                 and not unicodedata.category(unichr(c)) in ['Cc','Cf','Zl','Zp']
+                 and chr(c) not in self._rev_escapes \
+                 and not unicodedata.category(chr(c)) in ['Cc','Cf','Zl','Zp']
@@ -3591 +3589 @@ class JSON(object):
-        except Exception, err:
+        except Exception as err:
@@ -3624,3 +3622,3 @@ class JSON(object):
-            if not isinstance(c,unicode):
-                c = unicode(c)
-            if c in u' \t\n\r\f\v':
+            if not isinstance(c,str):
+                c = str(c)
+            if c in ' \t\n\r\f\v':
@@ -3639 +3637 @@ class JSON(object):
-        if c == u'\u2028' or c == u'\u2029': # unicodedata.category(c) in  ['Zl', 'Zp']
+        if c == '\u2028' or c == '\u2029': # unicodedata.category(c) in  ['Zl', 'Zp']
@@ -3737 +3735 @@ class JSON(object):
-                    except JSONError, err:
+                    except JSONError as err:
@@ -3798 +3796 @@ class JSON(object):
-        elif c == '0' and (buf.peek(1) in [u'x',u'X']):
+        elif c == '0' and (buf.peek(1) in ['x','X']):
@@ -3813 +3811 @@ class JSON(object):
-        elif c == '0' and (buf.peek(1) in [u'o','O']):
+        elif c == '0' and (buf.peek(1) in ['o','O']):
@@ -3828 +3826 @@ class JSON(object):
-        elif c == '0' and (buf.peek(1) in [u'b','B']):
+        elif c == '0' and (buf.peek(1) in ['b','B']):
@@ -4002 +4000 @@ class JSON(object):
-        if isinstance(n, (int,long)):
+        if isinstance(n, int):
@@ -4104 +4102 @@ class JSON(object):
-                        uc = u'\ufffd' # replacement char
+                        uc = '\ufffd' # replacement char
@@ -4114 +4112 @@ class JSON(object):
-                    _append( u'\ufffd' ) # replacement char
+                    _append( '\ufffd' ) # replacement char
@@ -4156 +4154 @@ class JSON(object):
-                elif escapes.has_key(c):
+                elif c in escapes:
@@ -4217 +4215 @@ class JSON(object):
-                        low_surrogate = unichr(codepoint)
+                        low_surrogate = chr(codepoint)
@@ -4225 +4223 @@ class JSON(object):
-                            uc = u'\ufffd' # replacement char
+                            uc = '\ufffd' # replacement char
@@ -4239 +4237 @@ class JSON(object):
-                        high_surrogate = unichr(codepoint)  # remember until we get to the low surrogate
+                        high_surrogate = chr(codepoint)  # remember until we get to the low surrogate
@@ -4246 +4244 @@ class JSON(object):
-                        _append( u'\ufffd' ) # replacement char
+                        _append( '\ufffd' ) # replacement char
@@ -4304 +4302 @@ class JSON(object):
-            _append( u'\ufffd' ) # replacement char
+            _append( '\ufffd' ) # replacement char
@@ -4326 +4324 @@ class JSON(object):
-            except JSONError, err:
+            except JSONError as err:
@@ -4339 +4337 @@ class JSON(object):
-        import UserString
+        import collections
@@ -4341 +4339 @@ class JSON(object):
-        if isinstance(s, UserString.UserString):
+        if isinstance(s, collections.UserString):
@@ -4344 +4342 @@ class JSON(object):
-                if py2strenc and not isinstance(c2,unicode):
+                if py2strenc and not isinstance(c2,str):
@@ -4348 +4346 @@ class JSON(object):
-        elif py2strenc and not isinstance(s,unicode):
+        elif py2strenc and not isinstance(s,str):
@@ -4389,2 +4387,2 @@ class JSON(object):
-                chunks.append( unicode(s[j:i]) )
-            elif revesc.has_key(c):
+                chunks.append( str(s[j:i]) )
+            elif c in revesc:
@@ -4434 +4432 @@ class JSON(object):
-                    if optrevesc.has_key(c):
+                    if c in optrevesc:
@@ -4501 +4499 @@ class JSON(object):
-                except JSONError, err:
+                except JSONError as err:
@@ -4511 +4509 @@ class JSON(object):
-                except JSONError, err:
+                except JSONError as err:
@@ -4796 +4794 @@ class JSON(object):
-                except JSONError, err:
+                except JSONError as err:
@@ -4806 +4804 @@ class JSON(object):
-                except JSONError, err:
+                except JSONError as err:
@@ -4891 +4889 @@ class JSON(object):
-            except JSONException, err:
+            except JSONException as err:
@@ -4893 +4891 @@ class JSON(object):
-            except Exception, err:   # Mainly here to catch maximum recursion depth exceeded
+            except Exception as err:   # Mainly here to catch maximum recursion depth exceeded
@@ -4958,2 +4956,2 @@ class JSON(object):
-                    catfirst = unicodedata.category(unicode(first))
-                    catsecond = unicodedata.category(unicode(second))
+                    catfirst = unicodedata.category(str(first))
+                    catsecond = unicodedata.category(str(second))
@@ -4999 +4997 @@ class JSON(object):
-        elif isinstance(obj, (int,long,float,complex)) or\
+        elif isinstance(obj, (int,float,complex)) or\
@@ -5002 +5000 @@ class JSON(object):
-        elif isinstance(obj, basestring) or helpers.isstringtype(obj):
+        elif isinstance(obj, str) or helpers.isstringtype(obj):
@@ -5107 +5105 @@ class JSON(object):
-            except UnicodeError, err:
+            except UnicodeError as err:
@@ -5122 +5120 @@ class JSON(object):
-            except UnicodeEncodeError, err:
+            except UnicodeEncodeError as err:
@@ -5169 +5167 @@ class JSON(object):
-            except JSONEncodeError, err1:
+            except JSONEncodeError as err1:
@@ -5174 +5172 @@ class JSON(object):
-                except Exception, err2:
+                except Exception as err2:
@@ -5303 +5301 @@ class JSON(object):
-                it = obj.iterkeys()
+                it = iter(obj.keys())
@@ -5351 +5349 @@ class JSON(object):
-                    obj2 = it.next()
+                    obj2 = next(it)
@@ -5413 +5411 @@ class JSON(object):
-                    part_keys.sort( key=(lambda t: (unicode(t[0]).upper(),t[0])) )
+                    part_keys.sort( key=(lambda t: (str(t[0]).upper(),t[0])) )
@@ -5415 +5413 @@ class JSON(object):
-                    part_keys.sort( key=(lambda t: unicode(t[0])) )
+                    part_keys.sort( key=(lambda t: str(t[0])) )
@@ -5675 +5673 @@ def decode( txt, encoding=None, **kwargs ):
-    for kw,val in kwargs.items():
+    for kw,val in list(kwargs.items()):
@@ -5739 +5737 @@ def encode_to_file( filename, obj, encoding='utf-8', overwrite=False, **kwargs )
-    if not isinstance(filename,basestring) or not filename:
+    if not isinstance(filename,str) or not filename:
@@ -5764 +5762 @@ def decode_file( filename, encoding=None, **kwargs ):
-    if isinstance(filename,basestring):
+    if isinstance(filename,str):
@@ -5950 +5948 @@ MORE INFORMATION:
-        except JSONError, err:
+        except JSONError as err:
@@ -5954 +5952 @@ MORE INFORMATION:
-        except Exception, err:
+        except Exception as err:
@@ -6001 +5999 @@ MORE INFORMATION:
-            except IOError, err:
+            except IOError as err:
@@ -6021 +6019 @@ MORE INFORMATION:
-                except IOError, err:
+                except IOError as err:
@@ -6093 +6091 @@ MORE INFORMATION:
-        except getopt.GetoptError, err:
+        except getopt.GetoptError as err:
@@ -6273 +6271 @@ the options --allow, --warn, or --forbid ; for example:
-            except KeyboardInterrupt, err:
+            except KeyboardInterrupt as err:
--- contrib/python/demjson/py3/tests/test_demjson.py	(index)
+++ contrib/python/demjson/py3/tests/test_demjson.py	(working tree)
@@ -61 +61 @@ else:
-            print "\nSKIPPING TEST %s: Requires Python 3" % method.__name__
+            print("\nSKIPPING TEST %s: Requires Python 3" % method.__name__)
@@ -66 +66 @@ else:
-            print "\nSKIPPING TEST %s: Requires Python 2.7 or greater" % method.__name__
+            print("\nSKIPPING TEST %s: Requires Python 2.7 or greater" % method.__name__)
@@ -71 +71 @@ else:
-            print "\nSKIPPING TEST %s: Requires Python with wide Unicode support (maxunicode > U+FFFF)" % method.__name__
+            print("\nSKIPPING TEST %s: Requires Python with wide Unicode support (maxunicode > U+FFFF)" % method.__name__)
@@ -176 +176 @@ class rot_one(codecs.CodecInfo):
-                chars.append( unichr(b-1) )
+                chars.append( chr(b-1) )
@@ -178,3 +178,3 @@ class rot_one(codecs.CodecInfo):
-                chars.append( u'Z' )
-            elif b <= 0x7fL:
-                chars.append( unichr(b) )
+                chars.append( 'Z' )
+            elif b <= 0x7f:
+                chars.append( chr(b) )
@@ -183 +183 @@ class rot_one(codecs.CodecInfo):
-        return (u''.join(chars), i+1)
+        return (''.join(chars), i+1)
@@ -216,2 +216,2 @@ class no_curly_braces(codecs.CodecInfo):
-                chars.append( unichr(b) )
-        return (u''.join(chars), i+1)
+                chars.append( chr(b) )
+        return (''.join(chars), i+1)
@@ -414,2 +414 @@ class DemjsonTest(unittest.TestCase):
-            raise self.failureException, \
-                  "can't compare non-string to regex: %r" % value
+            raise self.failureException("can't compare non-string to regex: %r" % value)
@@ -417,2 +416 @@ class DemjsonTest(unittest.TestCase):
-            raise self.failureException, \
-                  (msg or '%r !~ /%s/' %(value,pattern))
+            raise self.failureException(msg or '%r !~ /%s/' %(value,pattern))
@@ -471 +469 @@ class DemjsonTest(unittest.TestCase):
-        self.assertTrue( isinstance( demjson.json_int(0), (int,long) ) )
+        self.assertTrue( isinstance( demjson.json_int(0), int ) )
@@ -573,4 +571,4 @@ class DemjsonTest(unittest.TestCase):
-        self.assertEqual(demjson.decode(r'"\u2012"'), u'\u2012')
-        self.assertEqual(demjson.decode(r'"\u1eDc"'), u'\u1edc')
-        self.assertEqual(demjson.decode(r'"\uffff"'), u'\uffff')
-        self.assertEqual(demjson.decode(r'"\u00a012"'), u'\u00a0' + '12')
+        self.assertEqual(demjson.decode(r'"\u2012"'), '\u2012')
+        self.assertEqual(demjson.decode(r'"\u1eDc"'), '\u1edc')
+        self.assertEqual(demjson.decode(r'"\uffff"'), '\uffff')
+        self.assertEqual(demjson.decode(r'"\u00a012"'), '\u00a0' + '12')
@@ -584,2 +582,2 @@ class DemjsonTest(unittest.TestCase):
-        self.assertEqual(demjson.encode(u'\u00e0', escape_unicode=True), r'"\u00e0"')
-        self.assertEqual(demjson.encode(u'\u2012', escape_unicode=True), r'"\u2012"')
+        self.assertEqual(demjson.encode('\u00e0', escape_unicode=True), r'"\u00e0"')
+        self.assertEqual(demjson.encode('\u2012', escape_unicode=True), r'"\u2012"')
@@ -596,8 +594,8 @@ class DemjsonTest(unittest.TestCase):
-        self.assertEqual(demjson.decode(r'"\u{0041}"',allow_extended_unicode_escapes=True), u'A')
-        self.assertEqual(demjson.decode(r'"\u{1aFe}"',allow_extended_unicode_escapes=True), u'\u1afe')
-        self.assertEqual(demjson.decode(r'"\u{41}"',allow_extended_unicode_escapes=True), u'A')
-        self.assertEqual(demjson.decode(r'"\u{1}"',allow_extended_unicode_escapes=True), u'\u0001')
-        self.assertEqual(demjson.decode(r'"\u{00000000000041}"',allow_extended_unicode_escapes=True), u'A')
-        self.assertEqual(demjson.decode(r'"\u{1000a}"',allow_extended_unicode_escapes=True), u'\U0001000a')
-        self.assertEqual(demjson.decode(r'"\u{10ffff}"',allow_extended_unicode_escapes=True), u'\U0010FFFF')
-        self.assertEqual(demjson.decode(r'"\u{0000010ffff}"',allow_extended_unicode_escapes=True), u'\U0010FFFF')
+        self.assertEqual(demjson.decode(r'"\u{0041}"',allow_extended_unicode_escapes=True), 'A')
+        self.assertEqual(demjson.decode(r'"\u{1aFe}"',allow_extended_unicode_escapes=True), '\u1afe')
+        self.assertEqual(demjson.decode(r'"\u{41}"',allow_extended_unicode_escapes=True), 'A')
+        self.assertEqual(demjson.decode(r'"\u{1}"',allow_extended_unicode_escapes=True), '\u0001')
+        self.assertEqual(demjson.decode(r'"\u{00000000000041}"',allow_extended_unicode_escapes=True), 'A')
+        self.assertEqual(demjson.decode(r'"\u{1000a}"',allow_extended_unicode_escapes=True), '\U0001000a')
+        self.assertEqual(demjson.decode(r'"\u{10ffff}"',allow_extended_unicode_escapes=True), '\U0010FFFF')
+        self.assertEqual(demjson.decode(r'"\u{0000010ffff}"',allow_extended_unicode_escapes=True), '\U0010FFFF')
@@ -681 +679 @@ class DemjsonTest(unittest.TestCase):
-                                        encoding='utf-8'), u'\u00e0')
+                                        encoding='utf-8'), '\u00e0')
@@ -683 +681 @@ class DemjsonTest(unittest.TestCase):
-                                        encoding='ucs4le'), u'\u00e0')
+                                        encoding='ucs4le'), '\u00e0')
@@ -685 +683 @@ class DemjsonTest(unittest.TestCase):
-                                        encoding='ucs4be'), u'\u00e0')
+                                        encoding='ucs4be'), '\u00e0')
@@ -687 +685 @@ class DemjsonTest(unittest.TestCase):
-                                        encoding='utf-32be'), u'\u00e0')
+                                        encoding='utf-32be'), '\u00e0')
@@ -689 +687 @@ class DemjsonTest(unittest.TestCase):
-                                        encoding='ucs4'), u'\u00e0')
+                                        encoding='ucs4'), '\u00e0')
@@ -693 +691 @@ class DemjsonTest(unittest.TestCase):
-        self.assertEqual(demjson.encode(u'\u00e0', escape_unicode=False, encoding='utf-8'),
+        self.assertEqual(demjson.encode('\u00e0', escape_unicode=False, encoding='utf-8'),
@@ -695 +693 @@ class DemjsonTest(unittest.TestCase):
-        self.assertEqual(demjson.encode(u'\u00e0', escape_unicode=False, encoding='ucs4le'),
+        self.assertEqual(demjson.encode('\u00e0', escape_unicode=False, encoding='ucs4le'),
@@ -697 +695 @@ class DemjsonTest(unittest.TestCase):
-        self.assertEqual(demjson.encode(u'\u00e0', escape_unicode=False, encoding='ucs4be'),
+        self.assertEqual(demjson.encode('\u00e0', escape_unicode=False, encoding='ucs4be'),
@@ -699 +697 @@ class DemjsonTest(unittest.TestCase):
-        self.assertEqual(demjson.encode(u'\u00e0', escape_unicode=False, encoding='utf-32be'),
+        self.assertEqual(demjson.encode('\u00e0', escape_unicode=False, encoding='utf-32be'),
@@ -701 +699 @@ class DemjsonTest(unittest.TestCase):
-        self.assertTrue(demjson.encode(u'\u00e0', escape_unicode=False, encoding='ucs4')
+        self.assertTrue(demjson.encode('\u00e0', escape_unicode=False, encoding='ucs4')
@@ -710 +708 @@ class DemjsonTest(unittest.TestCase):
-        self.assertEqual(demjson.encode(u'\u001B', escape_unicode=False, encoding='utf-8'),
+        self.assertEqual(demjson.encode('\u001B', escape_unicode=False, encoding='utf-8'),
@@ -713 +711 @@ class DemjsonTest(unittest.TestCase):
-        self.assertEqual(demjson.encode(u'\u007F', escape_unicode=False, encoding='utf-8'),
+        self.assertEqual(demjson.encode('\u007F', escape_unicode=False, encoding='utf-8'),
@@ -716 +714 @@ class DemjsonTest(unittest.TestCase):
-        self.assertEqual(demjson.encode(u'\u00AD', escape_unicode=False, encoding='utf-8'),
+        self.assertEqual(demjson.encode('\u00AD', escape_unicode=False, encoding='utf-8'),
@@ -719 +717 @@ class DemjsonTest(unittest.TestCase):
-        self.assertEqual(demjson.encode(u'\u200F', escape_unicode=False, encoding='utf-8'),
+        self.assertEqual(demjson.encode('\u200F', escape_unicode=False, encoding='utf-8'),
@@ -722 +720 @@ class DemjsonTest(unittest.TestCase):
-        self.assertEqual(demjson.encode(u'\u2028', escape_unicode=False, encoding='utf-8'),
+        self.assertEqual(demjson.encode('\u2028', escape_unicode=False, encoding='utf-8'),
@@ -725 +723 @@ class DemjsonTest(unittest.TestCase):
-        self.assertEqual(demjson.encode(u'\u2029', escape_unicode=False, encoding='utf-8'),
+        self.assertEqual(demjson.encode('\u2029', escape_unicode=False, encoding='utf-8'),
@@ -728 +726 @@ class DemjsonTest(unittest.TestCase):
-        self.assertEqual(demjson.encode(u'\U000E007F', escape_unicode=False, encoding='utf-8'),
+        self.assertEqual(demjson.encode('\U000E007F', escape_unicode=False, encoding='utf-8'),
@@ -735 +733 @@ class DemjsonTest(unittest.TestCase):
-                             u'\U00102345')
+                             '\U00102345')
@@ -737 +735 @@ class DemjsonTest(unittest.TestCase):
-                             u'\U00010000')
+                             '\U00010000')
@@ -739 +737 @@ class DemjsonTest(unittest.TestCase):
-                             u'\U0010ffff')
+                             '\U0010ffff')
@@ -752 +750 @@ class DemjsonTest(unittest.TestCase):
-            self.assertEqual(demjson.encode(u'\U00010000',encoding='ascii'),
+            self.assertEqual(demjson.encode('\U00010000',encoding='ascii'),
@@ -754 +752 @@ class DemjsonTest(unittest.TestCase):
-            self.assertEqual(demjson.encode(u'\U00102345',encoding='ascii'),
+            self.assertEqual(demjson.encode('\U00102345',encoding='ascii'),
@@ -756 +754 @@ class DemjsonTest(unittest.TestCase):
-            self.assertEqual(demjson.encode(u'\U0010ffff',encoding='ascii'),
+            self.assertEqual(demjson.encode('\U0010ffff',encoding='ascii'),
@@ -775 +773 @@ class DemjsonTest(unittest.TestCase):
-                             u"a\u00e0\u0152")
+                             "a\u00e0\u0152")
@@ -783 +781 @@ class DemjsonTest(unittest.TestCase):
-                             u"A{@\u0152")
+                             "A{@\u0152")
@@ -791 +789 @@ class DemjsonTest(unittest.TestCase):
-                             u"a\u00e0\u0152")
+                             "a\u00e0\u0152")
@@ -798 +796 @@ class DemjsonTest(unittest.TestCase):
-                         u"ZAXY 5")
+                         "ZAXY 5")
@@ -805 +803 @@ class DemjsonTest(unittest.TestCase):
-                         u"abc")
+                         "abc")
@@ -813 +811 @@ class DemjsonTest(unittest.TestCase):
-            s = u'a\u00e0\u0152'
+            s = 'a\u00e0\u0152'
@@ -820 +818 @@ class DemjsonTest(unittest.TestCase):
-            s = u"A{@\u0152"
+            s = "A{@\u0152"
@@ -827 +825 @@ class DemjsonTest(unittest.TestCase):
-            s = u'a\u00e0\u0152'
+            s = 'a\u00e0\u0152'
@@ -834 +832 @@ class DemjsonTest(unittest.TestCase):
-        d = u"ABYZ 5"
+        d = "ABYZ 5"
@@ -841,2 +839,2 @@ class DemjsonTest(unittest.TestCase):
-        self.assertRaises(demjson.JSONEncodeError, demjson.encode, u'"abc"', encoding=ci )
-        self.assertRaises(demjson.JSONEncodeError, demjson.encode, u'{"abc":42}', encoding=ci )
+        self.assertRaises(demjson.JSONEncodeError, demjson.encode, '"abc"', encoding=ci )
+        self.assertRaises(demjson.JSONEncodeError, demjson.encode, '{"abc":42}', encoding=ci )
@@ -906 +904 @@ class DemjsonTest(unittest.TestCase):
-        self.assertRaises(demjson.JSONDecodeError, demjson.decode, '{"a":1,"\u0061":1}', prevent_duplicate_keys=True)
+        self.assertRaises(demjson.JSONDecodeError, demjson.decode, '{"a":1,"\\u0061":1}', prevent_duplicate_keys=True)
@@ -951 +949 @@ class DemjsonTest(unittest.TestCase):
-                    def next(self):
+                    def __next__(self):
@@ -962,2 +960,2 @@ class DemjsonTest(unittest.TestCase):
-        import UserString
-        class LikeString(UserString.UserString):
+        import collections
+        class LikeString(collections.UserString):
@@ -967 +965 @@ class DemjsonTest(unittest.TestCase):
-        mystring = LikeString(u'hi\u2012there')
+        mystring = LikeString('hi\u2012there')
@@ -989 +987 @@ class DemjsonTest(unittest.TestCase):
-        self.assertRaises(demjson.JSONDecodeError, demjson.decode, u'\u00a0[]', strict=True)
+        self.assertRaises(demjson.JSONDecodeError, demjson.decode, '\u00a0[]', strict=True)
@@ -1087 +1085 @@ class DemjsonTest(unittest.TestCase):
-                         {u'a': 42, u'b': {u'c': 99}} )
+                         {'a': 42, 'b': {'c': 99}} )
@@ -1089 +1087 @@ class DemjsonTest(unittest.TestCase):
-                         [(u'a', 42), (u'b', [(u'c', 99)])] )
+                         [('a', 42), ('b', [('c', 99)])] )
@@ -1113 +1111 @@ class DemjsonTest(unittest.TestCase):
-            if isinstance(k,basestring):
+            if isinstance(k,str):
@@ -1119 +1117 @@ class DemjsonTest(unittest.TestCase):
-            if isinstance(k, (int,basestring)):
+            if isinstance(k, (int,str)):
@@ -1205 +1203 @@ class DemjsonTest(unittest.TestCase):
-            elif isinstance(val, basestring):
+            elif isinstance(val, str):
@@ -1214 +1212 @@ class DemjsonTest(unittest.TestCase):
-                         u'{"ASOF":"YEAR 2014 MONTH 01 DAY 17","NUMBER":{"IMAGINARY":7.25,"REAL":3.0},"TEN":10}' )
+                         '{"ASOF":"YEAR 2014 MONTH 01 DAY 17","NUMBER":{"IMAGINARY":7.25,"REAL":3.0},"TEN":10}' )
@@ -1219 +1217 @@ class DemjsonTest(unittest.TestCase):
-            return "/".join( sorted([ str(k) for k in d.keys() ]) )
+            return "/".join( sorted([ str(k) for k in list(d.keys()) ]) )
@@ -1238 +1236 @@ class DemjsonTest(unittest.TestCase):
-                         u'["abc",123,"%s","%s",{"a":42,"wow":true}]' % ( repr(vals[2]), repr(vals[3])) )
+                         '["abc",123,"%s","%s",{"a":42,"wow":true}]' % ( repr(vals[2]), repr(vals[3])) )
@@ -1241 +1239 @@ class DemjsonTest(unittest.TestCase):
-                         u'["abc",123,"%s","%s","a/wow"]' % ( repr(vals[2]), repr(vals[3])) )
+                         '["abc",123,"%s","%s","a/wow"]' % ( repr(vals[2]), repr(vals[3])) )
@@ -1244 +1242 @@ class DemjsonTest(unittest.TestCase):
-                         u'["abc",123,"%s","%s","%s"]' % ( repr(vals[2]), repr(vals[3]), repr(magic(vals[4])) ) )
+                         '["abc",123,"%s","%s","%s"]' % ( repr(vals[2]), repr(vals[3]), repr(magic(vals[4])) ) )
@@ -1252 +1250 @@ class DemjsonTest(unittest.TestCase):
-        d = datetime.date(2014,01,04)
+        d = datetime.date(2014,0o1,0o4)
@@ -1257 +1255 @@ class DemjsonTest(unittest.TestCase):
-        d = datetime.datetime(2014,01,04,13,22,15)
+        d = datetime.datetime(2014,0o1,0o4,13,22,15)
@@ -1420,2 +1418,2 @@ class DemjsonTest(unittest.TestCase):
-        self.assertEqual( self.decode_stats( u'"\u2020"' ).max_string_length, 1 )
-        self.assertEqual( self.decode_stats( u'"\U0010ffff"' ).max_string_length, (1 if is_wide_python else 2) )
+        self.assertEqual( self.decode_stats( '"\u2020"' ).max_string_length, 1 )
+        self.assertEqual( self.decode_stats( '"\U0010ffff"' ).max_string_length, (1 if is_wide_python else 2) )
@@ -1431,2 +1429,2 @@ class DemjsonTest(unittest.TestCase):
-        self.assertEqual( self.decode_stats( u'"\u0000"' ).min_codepoint, 0 )
-        self.assertEqual( self.decode_stats( u'"\u0000"' ).max_codepoint, 0 )
+        self.assertEqual( self.decode_stats( '"\u0000"' ).min_codepoint, 0 )
+        self.assertEqual( self.decode_stats( '"\u0000"' ).max_codepoint, 0 )
@@ -1445,2 +1443,2 @@ class DemjsonTest(unittest.TestCase):
-        self.assertEqual( self.decode_stats( u'"\uffff"' ).min_codepoint, 0xffff )
-        self.assertEqual( self.decode_stats( u'"\uffff"' ).max_codepoint, 0xffff )
+        self.assertEqual( self.decode_stats( '"\uffff"' ).min_codepoint, 0xffff )
+        self.assertEqual( self.decode_stats( '"\uffff"' ).max_codepoint, 0xffff )
@@ -1449,2 +1447,2 @@ class DemjsonTest(unittest.TestCase):
-        self.assertEqual( self.decode_stats( u'"\U0010ffff"' ).min_codepoint, (0x10ffff if is_wide_python else 0xdbff) )
-        self.assertEqual( self.decode_stats( u'"\U0010ffff"' ).max_codepoint, (0x10ffff if is_wide_python else 0xdfff) )
+        self.assertEqual( self.decode_stats( '"\U0010ffff"' ).min_codepoint, (0x10ffff if is_wide_python else 0xdbff) )
+        self.assertEqual( self.decode_stats( '"\U0010ffff"' ).max_codepoint, (0x10ffff if is_wide_python else 0xdfff) )
@@ -1496 +1494 @@ def run_all_tests():
-    print 'Running with demjson version %s, Python version %s with %s-Unicode' % (demjson.__version__, sys.version.split(' ',1)[0],unicode_width)
+    print('Running with demjson version %s, Python version %s with %s-Unicode' % (demjson.__version__, sys.version.split(' ',1)[0],unicode_width))
@@ -1498 +1496 @@ def run_all_tests():
-        print 'WARNING: TESTING AGAINST AN OLD VERSION!'
+        print('WARNING: TESTING AGAINST AN OLD VERSION!')
