--- contrib/python/smmap/smmap/test/test_tutorial.py	(index)
+++ contrib/python/smmap/smmap/test/test_tutorial.py	(working tree)
@@ -19,8 +19,8 @@ class TestTutorial(TestBase):
 
         # Cursors
         ##########
-        import smmap.test.lib
-        with smmap.test.lib.FileCreator(1024 * 1024 * 8, "test_file") as fc:
+        from .lib import FileCreator
+        with FileCreator(1024 * 1024 * 8, "test_file") as fc:
             # obtain a cursor to access some file.
             c = mman.make_cursor(fc.path)
 
