--- a/tools/llvm-xray/xray-account.cpp
+++ b/tools/llvm-xray/xray-account.cpp
@@ -239,10 +239,10 @@ ResultRow getStats(std::vector<uint64_t> &Timings) {

-  auto Pct90Off = std::floor(Timings.size() * 0.9);
+  size_t Pct90Off = std::floor(Timings.size() * 0.9);
   std::nth_element(Timings.begin(), Timings.begin() + (uint64_t)Pct90Off,
                    Timings.end());
   R.Pct90 = Timings[Pct90Off];

-  auto Pct99Off = std::floor(Timings.size() * 0.99);
+  size_t Pct99Off = std::floor(Timings.size() * 0.99);
   std::nth_element(Timings.begin(), Timings.begin() + (uint64_t)Pct99Off,
                    Timings.end());
   R.Pct99 = Timings[Pct99Off];
