--- contrib/python/google-auth-oauthlib/tests/unit/test_flow.py	(index)
+++ contrib/python/google-auth-oauthlib/tests/unit/test_flow.py	(working tree)
@@ -28,1 +28,2 @@ from six.moves import urllib
-DATA_DIR = os.path.join(os.path.dirname(__file__), "data")
+import yatest.common as yc
+DATA_DIR = os.path.join(os.path.dirname(yc.source_path(__file__)), "data")
--- contrib/python/google-auth-oauthlib/tests/unit/test_helpers.py	(index)
+++ contrib/python/google-auth-oauthlib/tests/unit/test_helpers.py	(working tree)
@@ -21,1 +21,2 @@ import pytest
-DATA_DIR = os.path.join(os.path.dirname(__file__), "data")
+import yatest.common as yc
+DATA_DIR = os.path.join(os.path.dirname(yc.source_path(__file__)), "data")
--- contrib/python/google-auth-oauthlib/tests/unit/test_tool.py	(index)
+++ contrib/python/google-auth-oauthlib/tests/unit/test_tool.py	(working tree)
@@ -25,1 +25,2 @@ import pytest
-DATA_DIR = os.path.join(os.path.dirname(__file__), "data")
+import yatest.common as yc
+DATA_DIR = os.path.join(os.path.dirname(yc.source_path(__file__)), "data")
