commit 1dfb7c5883f95f5e61123cfc74b62833c087c9c5 (HEAD -> GEOQ-2214-use-compressed-fasttext-models)
author: mkrutyakov
date: 2024-03-12T15:17:14+03:00

    GEOQ-2214: Fix compress-fasttext zero embeddings

--- contrib/python/compress-fasttext/compress_fasttext/compress.py	(c937f389ea1ac0b31a5a5a326eb84ecb3c51fa1c)
+++ contrib/python/compress-fasttext/compress_fasttext/compress.py	(1dfb7c5883f95f5e61123cfc74b62833c087c9c5)
@@ -128,6 +128,13 @@ class CompressedFastTextKeyedVectors(FastTextKeyedVectors):
             fname, separately, sep_limit, ignore, pickle_protocol, compress, subname
         )
 
+    def __contains__(self, word):
+        if not super(FastTextKeyedVectors, self).__contains__(word):
+            return False
+        word_vec = self.word_vec(word, use_norm=False)
+        vec_is_not_zero = word_vec.any()
+        return vec_is_not_zero
+
 
 def make_new_fasttext_model(
         ft,
