--- contrib/python/Werkzeug/py3/werkzeug/test.py	(index)
+++ contrib/python/Werkzeug/py3/werkzeug/test.py	(working tree)
@@ -201,8 +201,8 @@ class _TestCookieJar(CookieJar):
 
         if cvals:
             environ["HTTP_COOKIE"] = "; ".join(cvals)
-        else:
-            environ.pop("HTTP_COOKIE", None)
+        #else:
+        #    environ.pop("HTTP_COOKIE", None)
 
     def extract_wsgi(
         self,
