From 8b5f3b5f655d299d956e29a52f68b4281570cc26 Mon Sep 17 00:00:00 2001
From: Erik Lax <erik@halon.se>
Date: Mon, 13 Jul 2026 21:03:03 +0200
Subject: [PATCH] Regression: Fix reentrant query removal after QID reuse

---
 src/lib/ares_process.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/src/lib/ares_process.c b/src/lib/ares_process.c
index aa2dc7f13d..9740bad16f 100644
--- a/src/lib/ares_process.c
+++ b/src/lib/ares_process.c
@@ -1598,7 +1598,12 @@ static void ares_detach_query(ares_query_t *query)
 {
   /* Remove the query from all the lists in which it is linked */
   ares_query_remove_from_conn(query);
-  ares_htable_szvp_remove(query->channel->queries_by_qid, query->qid);
+  /* A callback may queue a new query that reuses this ID. Only remove the
+   * entry if it still points to this query. */
+  if (ares_htable_szvp_get_direct(query->channel->queries_by_qid, query->qid) ==
+      query) {
+    ares_htable_szvp_remove(query->channel->queries_by_qid, query->qid);
+  }
   ares_llist_node_destroy(query->node_all_queries);
   query->node_all_queries = NULL;
 }
