--- contrib/libs/c-ares/src/lib/ares__socket.c	(index)
+++ contrib/libs/c-ares/src/lib/ares__socket.c	(working tree)
@@ -61,7 +61,7 @@
 #  define TFO_SKIP_CONNECT   0
 #  define TFO_USE_SENDTO     0
 #  define TFO_USE_CONNECTX   0
-#  define TFO_CLIENT_SOCKOPT TCP_FASTOPEN_CONNECT
+#  define TFO_CLIENT_SOCKOPT 0
 #elif defined(__FreeBSD__) && defined(TCP_FASTOPEN)
 #  define TFO_SUPPORTED      1
 #  define TFO_SKIP_CONNECT   1
