--- contrib/python/sympy/py3/sympy/parsing/latex/_parse_latex_antlr.py	(index)
+++ contrib/python/sympy/py3/sympy/parsing/latex/_parse_latex_antlr.py	(working tree)
@@ -56,8 +56,7 @@ if ErrorListener:
 def parse_latex(sympy, strict=False):
     antlr4 = import_module('antlr4')
 
-    if None in [antlr4, MathErrorListener] or \
-            not version('antlr4-python3-runtime').startswith('4.11'):
+    if None in [antlr4, MathErrorListener]:
         raise ImportError("LaTeX parsing requires the antlr4 Python package,"
                           " provided by pip (antlr4-python3-runtime) or"
                           " conda (antlr-python-runtime), version 4.11")
