Index: subvertpy/__init__.py
===================================================================
--- contrib/python/patched/subvertpy/subvertpy/__init__.py	(rev 0)
+++ contrib/python/patched/subvertpy/subvertpy/__init__.py	(working copy)
@@ -124,10 +124,11 @@ def _check_mtime(m):

 try:
     from subvertpy import client, _ra, repos, wc
-    for x in client, _ra, repos, wc:
-        if not _check_mtime(x):
-            from warnings import warn
-            warn("subvertpy extensions are outdated and need to be rebuilt")
-            break
+    # (ilezhankin): no need to check mtime in Arcadia build.
+    # for x in client, _ra, repos, wc:
+    #    if not _check_mtime(x):
+    #        from warnings import warn
+    #        warn("subvertpy extensions are outdated and need to be rebuilt")
+    #        break
 except ImportError as e:
     raise ImportError("Unable to load subvertpy extensions: %s" % e)
Index: subvertpy/_ra.c
===================================================================
--- contrib/python/patched/subvertpy/subvertpy/_ra.c	(rev 0)
+++ contrib/python/patched/subvertpy/subvertpy/_ra.c	(working copy)
@@ -3450,7 +3450,7 @@ moduleinit(void)
 	};
 	mod = PyModule_Create(&moduledef);
 #else
-	mod = Py_InitModule3("_ra", ra_module_methods, "Remote Access");
+	mod = Py_InitModule3("subvertpy._ra", ra_module_methods, "Remote Access");
 #endif
 	if (mod == NULL)
 		return NULL;
@@ -3464,7 +3464,7 @@ moduleinit(void)
 	PyModule_AddObject(mod, "Editor", (PyObject *)&Editor_Type);
 	Py_INCREF(&Editor_Type);

-	busy_exc = PyErr_NewException("_ra.BusyException", NULL, NULL);
+	busy_exc = PyErr_NewException("subvertpy._ra.BusyException", NULL, NULL);
 	PyModule_AddObject(mod, "BusyException", busy_exc);

 #if ONLY_SINCE_SVN(1, 5)
Index: subvertpy/client.c
===================================================================
--- contrib/python/patched/subvertpy/subvertpy/client.c	(rev 0)
+++ contrib/python/patched/subvertpy/subvertpy/client.c	(working copy)
@@ -364,6 +364,8 @@ static PyObject *py_info(const svn_info_t *info)
     ret->info = *svn_client_info2_dup(info, ret->pool);
     if (info->wc_info != NULL)
         ret->wc_info->info = *svn_wc_info_dup(info->wc_info, ret->pool);
+    else
+        ret->wc_info = NULL;
 #else
     ret->info = *svn_info_dup(info, ret->pool);
     if (info->has_wc_info) {
@@ -1138,7 +1140,7 @@ static PyObject *client_copy(PyObject *self, PyObject *args, PyObject *kwargs)
         apr_pool_destroy(temp_pool);
         return NULL;
     }
-    APR_ARRAY_IDX(src_paths, 0, svn_client_copy_source_t *) = &src;
+    APR_ARRAY_PUSH(src_paths, svn_client_copy_source_t *) = &src;
 #endif
 #if ONLY_SINCE_SVN(1, 6)
     RUN_SVN_WITH_POOL(temp_pool, svn_client_copy5(&commit_info, src_paths,
@@ -1199,7 +1201,7 @@ static PyObject *client_propset(PyObject *self, PyObject *args)
     /* FIXME: Support changelists */
     /* FIXME: Support depth */
     RUN_SVN_WITH_POOL(temp_pool, svn_client_propset3(&commit_info, propname,
-                &c_propval, target, recurse?svn_depth_infinity:svn_depth_files,
+                c_propval.len==0?NULL:&c_propval, target, recurse?svn_depth_infinity:svn_depth_files,
                 skip_checks, base_revision_for_url,
                 NULL, revprops, client->client, temp_pool));
     ret = py_commit_info_tuple(commit_info);
@@ -1210,7 +1212,7 @@ static PyObject *client_propset(PyObject *self, PyObject *args)
         apr_pool_destroy(temp_pool);
         return NULL;
     }
-    RUN_SVN_WITH_POOL(temp_pool, svn_client_propset2(propname, &c_propval,
+    RUN_SVN_WITH_POOL(temp_pool, svn_client_propset2(propname, c_propval.len==0?NULL:&c_propval,
                 target, recurse, skip_checks, client->client, temp_pool));
     ret = Py_None;
     Py_INCREF(ret);
@@ -1375,6 +1377,38 @@ static PyObject *client_proplist(PyObject *self, PyObject *args,
     return prop_list;
 }

+static PyObject* client_revproplist(PyObject* self, PyObject* args,
+                                    PyObject* kwargs) {
+    char* kwnames[] = {"url", "revision", NULL};
+    svn_opt_revision_t c_rev;
+    svn_revnum_t revnum;
+    apr_pool_t* temp_pool;
+    char* url;
+    PyObject* revision = Py_None;
+    ClientObject* client = (ClientObject*)self;
+
+    if (!PyArg_ParseTupleAndKeywords(args, kwargs, "sO", kwnames, &url, &revision))
+        return NULL;
+    if (!to_opt_revision(revision, &c_rev))
+        return NULL;
+    temp_pool = Pool(NULL);
+    if (temp_pool == NULL)
+        return NULL;
+
+#if ONLY_SINCE_SVN(1, 5)
+    apr_hash_t* props;
+    RUN_SVN_WITH_POOL(temp_pool,
+                      svn_client_revprop_list(&props, url, &c_rev,
+                                              &revnum, client->client, temp_pool));
+    PyObject* prop_dict = prop_hash_to_dict(props);
+    apr_pool_destroy(temp_pool);
+#else
+    error "Not implemented for SVN < 1.5"
+#endif
+
+    return prop_dict;
+}
+
 static PyObject *client_resolve(PyObject *self, PyObject *args)
 {
 #if ONLY_SINCE_SVN(1, 5)
@@ -1416,14 +1450,17 @@ static PyObject *client_update(PyObject *self, PyObject *args, PyObject *kwargs)
     int i = 0;
     ClientObject *client = (ClientObject *)self;
     bool allow_unver_obstructions = false,
-                  depth_is_sticky = false;
+                  depth_is_sticky = false,
+              adds_as_modification = true,
+	                 make_parents = false;
 	char *kwnames[] =
 		{ "path", "revision", "recurse", "ignore_externals", "depth_is_sticky",
-			"allow_unver_obstructions", NULL };
+			"allow_unver_obstructions", "adds_as_modification", "make_parents", NULL };

-	if (!PyArg_ParseTupleAndKeywords(args, kwargs, "O|Obbbb", kwnames,
+	if (!PyArg_ParseTupleAndKeywords(args, kwargs, "O|Obbbbbb", kwnames,
 			&paths, &rev, &recurse, &ignore_externals,
-			&depth_is_sticky, &allow_unver_obstructions))
+			&depth_is_sticky, &allow_unver_obstructions, &adds_as_modification,
+			&make_parents))
 		return NULL;

 	if (!to_opt_revision(rev, &c_rev))
@@ -1435,7 +1472,30 @@ static PyObject *client_update(PyObject *self, PyObject *args, PyObject *kwargs)
 		apr_pool_destroy(temp_pool);
 		return NULL;
 	}
-#if ONLY_SINCE_SVN(1, 5)
+
+#if ONLY_BEFORE_SVN(1, 7)
+	if (!adds_as_modification) {
+		PyErr_SetString(PyExc_NotImplementedError,
+						"!adds_as_modification not supported before svn 1.7");
+		apr_pool_destroy(temp_pool);
+		return NULL;
+	}
+
+	if (make_parents) {
+		PyErr_SetString(PyExc_NotImplementedError,
+						"make_parents not supported before svn 1.7");
+		apr_pool_destroy(temp_pool);
+		return NULL;
+	}
+#endif
+
+#if ONLY_SINCE_SVN(1, 7)
+	RUN_SVN_WITH_POOL(temp_pool, svn_client_update4(&result_revs,
+		apr_paths, &c_rev, recurse?svn_depth_infinity:svn_depth_files,
+		depth_is_sticky?TRUE:FALSE, ignore_externals, allow_unver_obstructions?TRUE:FALSE,
+		adds_as_modification?TRUE:FALSE, make_parents?TRUE:FALSE,
+		client->client, temp_pool));
+#elif ONLY_SINCE_SVN(1, 5)
 	RUN_SVN_WITH_POOL(temp_pool, svn_client_update3(&result_revs,
 		apr_paths, &c_rev, recurse?svn_depth_infinity:svn_depth_files,
 		depth_is_sticky?TRUE:FALSE, ignore_externals, allow_unver_obstructions?TRUE:FALSE,
@@ -1760,6 +1820,7 @@ static PyObject *client_info(PyObject *self, PyObject *args, PyObject *kwargs)
     apr_pool_t *temp_pool;
     ClientObject *client = (ClientObject *)self;

+    PyObject *py_path;
     const char *path;
     int depth = svn_depth_empty;
     bool fetch_excluded = false, fetch_actual_only = false;
@@ -1768,8 +1829,8 @@ static PyObject *client_info(PyObject *self, PyObject *args, PyObject *kwargs)
     PyObject *entry_dict;
     svn_error_t *err;

-    if (!PyArg_ParseTupleAndKeywords(args, kwargs, "s|OOibb", kwnames,
-                                     &path, &revision,
+    if (!PyArg_ParseTupleAndKeywords(args, kwargs, "O|OOibb", kwnames,
+                                     &py_path, &revision,
                                      &peg_revision, &depth,
                                      &fetch_excluded, &fetch_actual_only))
         return NULL;
@@ -1779,8 +1840,26 @@ static PyObject *client_info(PyObject *self, PyObject *args, PyObject *kwargs)
     if (!to_opt_revision(peg_revision, &c_peg_rev))
         return NULL;

-    if (c_rev.kind == svn_opt_revision_unspecified)
-        c_rev.kind = svn_opt_revision_head;
+
+    temp_pool = Pool(NULL);
+    if (temp_pool == NULL)
+        return NULL;
+
+    path = py_object_to_svn_path_or_url(py_path, temp_pool);
+    if (path == NULL) {
+        apr_pool_destroy(temp_pool);
+        return NULL;
+    }
+
+    if (svn_path_is_url(path))
+    {
+      if (c_peg_rev.kind == svn_opt_revision_unspecified)
+        c_peg_rev.kind = svn_opt_revision_head;
+    }
+    else
+    {
+        path = py_object_to_svn_abspath(py_path, temp_pool);
+    }

 #if ONLY_BEFORE_SVN(1, 5)
     if (depth != svn_depth_infinity && depth != svn_depth_empty) {
@@ -1790,10 +1869,6 @@ static PyObject *client_info(PyObject *self, PyObject *args, PyObject *kwargs)
     }
 #endif

-    temp_pool = Pool(NULL);
-    if (temp_pool == NULL)
-        return NULL;
-
     entry_dict = PyDict_New();
     if (entry_dict == NULL) {
         apr_pool_destroy(temp_pool);
@@ -1849,6 +1924,7 @@ static PyMethodDef client_methods[] = {
     { "propset", client_propset, METH_VARARGS, "S.propset(name, value, target, recurse=True, skip_checks=False)" },
     { "propget", client_propget, METH_VARARGS, "S.propget(name, target, peg_revision, revision=None, recurse=False) -> value" },
     { "proplist", (PyCFunction)client_proplist, METH_VARARGS|METH_KEYWORDS, "S.proplist(path, peg_revision, depth, revision=None)" },
+    { "revproplist", (PyCFunction)client_revproplist, METH_VARARGS | METH_KEYWORDS, "S.revproplist(url, revision)" },
     { "resolve", client_resolve, METH_VARARGS, "S.resolve(path, depth, choice)" },
     { "update", (PyCFunction)client_update, METH_VARARGS|METH_KEYWORDS, "S.update(path, rev=None, recurse=True, ignore_externals=False) -> list of revnums" },
     { "list", (PyCFunction)client_list, METH_VARARGS|METH_KEYWORDS, "S.list(path, peg_revision, depth, dirents=ra.DIRENT_ALL, revision=None) -> list of directory entries" },
@@ -2373,7 +2449,7 @@ moduleinit(void)
 	};
 	mod = PyModule_Create(&moduledef);
 #else
-	mod = Py_InitModule3("client", client_mod_methods, "Client methods");
+	mod = Py_InitModule3("subvertpy.client", client_mod_methods, "Client methods");
 #endif
 	if (mod == NULL)
 		return NULL;
Index: subvertpy/editor.c
===================================================================
--- contrib/python/patched/subvertpy/subvertpy/editor.c	(rev 0)
+++ contrib/python/patched/subvertpy/subvertpy/editor.c	(working copy)
@@ -79,10 +79,12 @@ static void py_editor_dealloc(PyObject *self)
 	PyObject_Del(self);
 }

+#ifndef _WIN64
 /* paranoia check */
 #if defined(SIZEOF_SIZE_T) && SIZEOF_SIZE_T != SIZEOF_LONG
 #error "Unable to determine PyArg_Parse format for size_t"
 #endif
+#endif

 /* svn_filesize_t is always 64 bits */
 #if SIZEOF_LONG == 8
Index: subvertpy/repos.c
===================================================================
--- contrib/python/patched/subvertpy/subvertpy/repos.c	(rev 0)
+++ contrib/python/patched/subvertpy/subvertpy/repos.c	(working copy)
@@ -30,6 +30,7 @@ extern PyTypeObject FileSystemRoot_Type;
 extern PyTypeObject Repository_Type;
 extern PyTypeObject FileSystem_Type;
 extern PyTypeObject Stream_Type;
+extern PyTypeObject Transaction_Type;

 typedef struct {
 	PyObject_VAR_HEAD
@@ -251,11 +252,75 @@ static PyObject *fs_get_revision_proplist(FileSystemObject *self, PyObject *args
 	return (PyObject *)ret;
 }

+static PyObject *fs_get_revision_prop(FileSystemObject *self, PyObject *args)
+{
+    svn_revnum_t rev;
+    char *propname;
+	apr_pool_t *pool;
+    svn_string_t *value;
+
+    if (!PyArg_ParseTuple(args, "ls", &rev, &propname)) {
+        return NULL;
+    }
+
+    pool = Pool(NULL);
+    if (pool == NULL) {
+        return NULL;
+    }
+
+    RUN_SVN_WITH_POOL(pool, svn_fs_revision_prop(&value, self->fs, rev, propname, pool));
+
+    if (value == NULL || value->data == NULL) {
+        Py_RETURN_NONE;
+    } else {
+        return PyBytes_FromStringAndSize(value->data, value->len);
+    }
+}
+
+typedef struct {
+    PyObject_HEAD
+    FileSystemObject *fs;
+    apr_pool_t *pool;
+    svn_fs_txn_t *txn;
+} TransactionObject;
+
+static PyObject *fs_open_txn(FileSystemObject *self, PyObject *args)
+{
+    char *name;
+    apr_pool_t *pool;
+    TransactionObject *ret;
+    svn_fs_txn_t *txn;
+
+    if (!PyArg_ParseTuple(args, "s", &name))
+        return NULL;
+
+    pool = Pool(NULL);
+    if (pool == NULL)
+        return NULL;
+
+    RUN_SVN_WITH_POOL(pool, svn_fs_open_txn(&txn, self->fs, name, pool));
+
+    ret = PyObject_New(TransactionObject, &Transaction_Type);
+    if (ret == NULL) {
+        apr_pool_destroy(pool);
+        return NULL;
+    }
+
+    ret->txn = txn;
+    ret->pool = pool;
+    ret->fs = self;
+    Py_INCREF(self);
+
+    return (PyObject *)ret;
+}
+
 static PyMethodDef fs_methods[] = {
 	{ "get_uuid", (PyCFunction)fs_get_uuid, METH_NOARGS, NULL },
 	{ "youngest_revision", (PyCFunction)fs_get_youngest_revision, METH_NOARGS, NULL },
 	{ "revision_root", (PyCFunction)fs_get_revision_root, METH_VARARGS, NULL },
 	{ "revision_proplist", (PyCFunction)fs_get_revision_proplist, METH_VARARGS, NULL },
+	{ "revision_prop", (PyCFunction)fs_get_revision_prop, METH_VARARGS, NULL },
+	{ "open_txn", (PyCFunction)fs_open_txn, METH_VARARGS, NULL },
 	{ NULL, }
 };

@@ -653,7 +718,7 @@ static PyObject *py_fs_path_change(svn_fs_path_change_t *val)

 #else

-static PyObject *py_fs_path_change2(svn_fs_path_change2_t *val)
+static PyObject *py_fs_path_change2(svn_fs_path_change2_t *val, svn_boolean_t copy_info)
 {
 	PyObject *ret, *py_node_id;

@@ -661,21 +726,32 @@ static PyObject *py_fs_path_change2(svn_fs_path_change2_t *val)
 	if (py_node_id == NULL) {
 		return NULL;
 	}
-	ret = Py_BuildValue("(Oibb)", py_node_id,
-						   val->change_kind, val->text_mod, val->prop_mod);
+	if (copy_info) {
+		ret = Py_BuildValue("(Oibbzl)", py_node_id,
+							val->change_kind, val->text_mod, val->prop_mod,
+							val->copyfrom_known ? val->copyfrom_path : NULL,
+							val->copyfrom_rev);
+	} else {
+		ret = Py_BuildValue("(Oibb)", py_node_id,
+							val->change_kind, val->text_mod, val->prop_mod);
+	}
 	Py_DECREF(py_node_id);
 	if (ret == NULL) {
 		return NULL;
 	}

-	/* FIXME: copyfrom information */
-
 	return ret;
 }
 #endif

-static PyObject *fs_root_paths_changed(FileSystemRootObject *self)
+static PyObject *fs_root_paths_changed(FileSystemRootObject *self, PyObject *args, PyObject *kwargs)
 {
+    char *kwnames[] = { "copy_info", NULL };
+	svn_boolean_t copy_info = FALSE;
+
+	if (!PyArg_ParseTupleAndKeywords(args, kwargs, "|b", kwnames, &copy_info))
+		return NULL;
+
 	apr_pool_t *temp_pool;
 	apr_hash_t *changed_paths;
 	const char *key;
@@ -708,7 +784,7 @@ static PyObject *fs_root_paths_changed(FileSystemRootObject *self)
 #endif
 		apr_hash_this(idx, (const void **)&key, &klen, (void **)&val);
 #if ONLY_SINCE_SVN(1, 6)
-		py_val = py_fs_path_change2(val);
+		py_val = py_fs_path_change2(val, copy_info);
 #else
 		py_val = py_fs_path_change(val);
 #endif
@@ -803,6 +879,30 @@ static PyObject *fs_node_file_proplist(FileSystemRootObject *self, PyObject *arg
 	apr_pool_destroy(temp_pool);
 	return ret;
 }
+
+static PyObject *fs_node_file_prop(FileSystemRootObject *self, PyObject *args)
+{
+    apr_pool_t *temp_pool;
+    svn_string_t *value;
+
+    char *path;
+    char *propname;
+    if (!PyArg_ParseTuple(args, "ss", &path, &propname)) {
+        return NULL;
+    }
+
+    temp_pool = Pool(NULL);
+    if (temp_pool == NULL) {
+        return NULL;
+    }
+
+    RUN_SVN_WITH_POOL(temp_pool, svn_fs_node_prop(&value, self->root, path, propname, temp_pool));
+    if (value == NULL || value->data == NULL) {
+        Py_RETURN_NONE;
+    }
+
+    return PyBytes_FromStringAndSize(value->data, value->len);
+}

 static PyObject *fs_root_file_checksum(FileSystemRootObject *self, PyObject *args)
 {
@@ -877,14 +977,110 @@ static PyObject *fs_root_file_contents(FileSystemRootObject *self, PyObject *arg
     return (PyObject *)ret;
 }

+static PyObject *fs_dir_entries(FileSystemRootObject *self, PyObject *args)
+{
+    PyObject *ret;
+    const char *path;
+    apr_pool_t *pool;
+    apr_hash_t *entries;
+    apr_hash_index_t *idx;
+    const char *key;
+    apr_ssize_t klen;
+    svn_fs_dirent_t *val;
+    PyObject *py_key;
+    PyObject *py_val;
+
+    if (!PyArg_ParseTuple(args, "s", &path)) {
+        return NULL;
+    }
+
+    pool = Pool(NULL);
+    if (pool == NULL) {
+        return NULL;
+    }
+
+    ret = PyDict_New();
+    if (ret == NULL) {
+        apr_pool_destroy(pool);
+        return NULL;
+    }
+
+    RUN_SVN_WITH_POOL(pool, svn_fs_dir_entries(&entries, self->root, path, pool));
+
+    for (idx = apr_hash_first(pool, entries); idx != NULL; idx = apr_hash_next(idx)) {
+        apr_hash_this(idx, (const void **)&key, &klen, (void **)&val);
+        py_key = PyBytes_FromStringAndSize(key, klen);
+
+        if ((val == NULL) || (py_key == NULL) || (val->name == NULL)) {
+            PyObject_Del(ret);
+            apr_pool_destroy(pool);
+            return NULL;
+        }
+
+        py_val = PyLong_FromLong(val->kind);
+        if (py_val == NULL) {
+            PyObject_Del(py_key);
+            PyObject_Del(ret);
+            apr_pool_destroy(pool);
+            return NULL;
+        }
+
+        if (PyDict_SetItemString(ret, key, py_val) != 0) {
+            PyObject_Del(py_val);
+            PyObject_Del(py_key);
+            PyObject_Del(ret);
+            apr_pool_destroy(pool);
+            return NULL;
+        }
+
+        Py_DECREF(py_val);
+        Py_DECREF(py_key);
+    }
+
+    apr_pool_destroy(pool);
+    return ret;
+}
+
+static PyObject *fs_node_created_rev(FileSystemRootObject *self, PyObject *args)
+{
+    PyObject *ret;
+    svn_revnum_t revision;
+    const char *path;
+    apr_pool_t *pool;
+
+    if (!PyArg_ParseTuple(args, "s", &path)) {
+        return NULL;
+    }
+
+    pool = Pool(NULL);
+    if (pool == NULL) {
+        return NULL;
+    }
+
+    RUN_SVN_WITH_POOL(pool, svn_fs_node_created_rev(&revision, self->root, path, pool));
+    if (revision == SVN_INVALID_REVNUM) {
+        apr_pool_destroy(pool);
+        Py_RETURN_NONE;
+    }
+
+    ret = PyLong_FromLong(revision);
+
+    apr_pool_destroy(pool);
+    return ret;
+}
+
+
 static PyMethodDef fs_root_methods[] = {
-	{ "paths_changed", (PyCFunction)fs_root_paths_changed, METH_NOARGS, NULL },
+	{ "paths_changed", (PyCFunction)fs_root_paths_changed, METH_VARARGS | METH_KEYWORDS, NULL },
 	{ "is_dir", (PyCFunction)fs_root_is_dir, METH_VARARGS, NULL },
 	{ "is_file", (PyCFunction)fs_root_is_file, METH_VARARGS, NULL },
 	{ "file_length", (PyCFunction)fs_root_file_length, METH_VARARGS, NULL },
 	{ "file_content", (PyCFunction)fs_root_file_contents, METH_VARARGS, NULL },
 	{ "file_checksum", (PyCFunction)fs_root_file_checksum, METH_VARARGS, NULL },
 	{ "proplist", (PyCFunction)fs_node_file_proplist, METH_VARARGS, NULL },
+	{ "dir_entries", (PyCFunction)fs_dir_entries, METH_VARARGS, NULL },
+	{ "prop", (PyCFunction)fs_node_file_prop, METH_VARARGS, NULL },
+	{ "created_rev", (PyCFunction)fs_node_created_rev, METH_VARARGS, NULL },
 	{ NULL, }
 };

@@ -948,6 +1144,157 @@ PyTypeObject FileSystemRoot_Type = {
 	fs_root_methods, /*	struct PyMethodDef *tp_methods;	*/
 };

+static void txn_dealloc(PyObject *self)
+{
+    TransactionObject *txnobj = (TransactionObject *)self;
+
+    apr_pool_destroy(txnobj->pool);
+    Py_DECREF(txnobj->fs);
+    PyObject_Del(txnobj);
+}
+
+static PyObject *fs_txn_set_prop(TransactionObject *self, PyObject *args)
+{
+    char *name, *value;
+    svn_string_t *cvalue;
+    int vallen;
+    if (!PyArg_ParseTuple(args, "sz#", &name, &value, &vallen))
+        return NULL;
+
+    if (value == NULL)
+        cvalue = NULL;
+    else
+        cvalue = svn_string_ncreate(value, vallen, self->pool);
+    RUN_SVN_WITH_POOL(self->pool, svn_fs_change_txn_prop(self->txn, name,
+                                                         cvalue, self->pool));
+    Py_RETURN_NONE;
+}
+
+static PyObject *fs_txn_get_prop(TransactionObject *self, PyObject *args)
+{
+    char *name;
+    svn_string_t *value;
+    if (!PyArg_ParseTuple(args, "s", &name))
+        return NULL;
+
+    RUN_SVN_WITH_POOL(self->pool, svn_fs_txn_prop(&value, self->txn, name,
+                                                  self->pool));
+
+    if (value == NULL || value->data == NULL)
+        Py_RETURN_NONE;
+    else
+        return PyBytes_FromStringAndSize(value->data, value->len);
+}
+
+static PyObject *fs_txn_list_props(TransactionObject *self)
+{
+    PyObject *ret;
+    apr_pool_t *pool;
+    apr_hash_t *props;
+
+    pool = Pool(NULL);
+    if (pool == NULL)
+        return NULL;
+
+    RUN_SVN_WITH_POOL(pool, svn_fs_txn_proplist(&props, self->txn, pool));
+
+    ret = prop_hash_to_dict(props);
+    apr_pool_destroy(pool);
+    return (PyObject *)ret;
+}
+
+static PyObject *fs_txn_root(TransactionObject *self)
+{
+	FileSystemRootObject *ret;
+	apr_pool_t *pool;
+	svn_fs_root_t *root;
+
+	pool = Pool(NULL);
+	if (pool == NULL)
+		return NULL;
+
+	RUN_SVN_WITH_POOL(pool, svn_fs_txn_root(&root, self->txn, pool));
+
+	ret = PyObject_New(FileSystemRootObject, &FileSystemRoot_Type);
+	if (ret == NULL) {
+		apr_pool_destroy(pool);
+		return NULL;
+	}
+
+	ret->root = root;
+	ret->pool = pool;
+
+	return (PyObject *)ret;
+}
+
+static PyMethodDef txn_methods[] = {
+    { "set_prop", (PyCFunction)fs_txn_set_prop, METH_VARARGS, NULL },
+    { "get_prop", (PyCFunction)fs_txn_get_prop, METH_VARARGS, NULL },
+    { "list_props", (PyCFunction)fs_txn_list_props, METH_NOARGS, NULL },
+    { "root", (PyCFunction)fs_txn_root, METH_NOARGS, NULL },
+    { NULL, }
+};
+
+PyTypeObject Transaction_Type = {
+    PyVarObject_HEAD_INIT(NULL, 0)
+	"repos.Transaction", /*	const char *tp_name;  For printing, in format "<module>.<name>" */
+	sizeof(TransactionObject),
+	0,/*	Py_ssize_t tp_basicsize, tp_itemsize;  For allocation */
+
+	/* Methods to implement standard operations */
+
+	txn_dealloc, /*	destructor tp_dealloc;	*/
+	NULL, /*	printfunc tp_print;	*/
+	NULL, /*	getattrfunc tp_getattr;	*/
+	NULL, /*	setattrfunc tp_setattr;	*/
+	NULL, /*	cmpfunc tp_compare;	*/
+	NULL, /*	reprfunc tp_repr;	*/
+
+	/* Method suites for standard classes */
+
+	NULL, /*	PyNumberMethods *tp_as_number;	*/
+	NULL, /*	PySequenceMethods *tp_as_sequence;	*/
+	NULL, /*	PyMappingMethods *tp_as_mapping;	*/
+
+	/* More standard operations (here for binary compatibility) */
+
+	NULL, /*	hashfunc tp_hash;	*/
+	NULL, /*	ternaryfunc tp_call;	*/
+	NULL, /*	reprfunc tp_str;	*/
+	NULL, /*	getattrofunc tp_getattro;	*/
+	NULL, /*	setattrofunc tp_setattro;	*/
+
+	/* Functions to access object as input/output buffer */
+	NULL, /*	PyBufferProcs *tp_as_buffer;	*/
+
+	/* Flags to define presence of optional/expanded features */
+	0, /*	long tp_flags;	*/
+
+	"Transaction", /*	const char *tp_doc;  Documentation string */
+
+	/* Assigned meaning in release 2.0 */
+	/* call function for all accessible objects */
+	NULL, /*	traverseproc tp_traverse;	*/
+
+	/* delete references to contained objects */
+	NULL, /*	inquiry tp_clear;	*/
+
+	/* Assigned meaning in release 2.1 */
+	/* rich comparisons */
+	NULL, /*	richcmpfunc tp_richcompare;	*/
+
+	/* weak reference enabler */
+	0, /*	Py_ssize_t tp_weaklistoffset;	*/
+
+	/* Added in release 2.2 */
+	/* Iterators */
+	NULL, /*	getiterfunc tp_iter;	*/
+	NULL, /*	iternextfunc tp_iternext;	*/
+
+	/* Attribute descriptor and subclassing stuff */
+	txn_methods, /*	struct PyMethodDef *tp_methods;	*/
+};
+

 static PyObject *moduleinit(void)
 {
@@ -966,6 +1313,9 @@ static PyObject *moduleinit(void)
 	if (PyType_Ready(&Stream_Type) < 0)
 		return NULL;

+	if (PyType_Ready(&Transaction_Type) < 0)
+		return NULL;
+
 	apr_initialize();
 	pool = Pool(NULL);
 	if (pool == NULL)
@@ -987,7 +1337,7 @@ static PyObject *moduleinit(void)
 	};
 	mod = PyModule_Create(&moduledef);
 #else
-	mod = Py_InitModule3("repos", repos_module_methods, "Local repository management");
+	mod = Py_InitModule3("subvertpy.repos", repos_module_methods, "Local repository management");
 #endif
 	if (mod == NULL)
 		return NULL;
@@ -1014,6 +1364,15 @@ static PyObject *moduleinit(void)
 	PyModule_AddObject(mod, "Stream", (PyObject *)&Stream_Type);
 	Py_INCREF(&Stream_Type);

+    PyModule_AddObject(mod, "FS_DIRENT_NODE_KIND_NONE", PyLong_FromLong(svn_node_none));
+    PyModule_AddObject(mod, "FS_DIRENT_NODE_KIND_FILE", PyLong_FromLong(svn_node_file));
+    PyModule_AddObject(mod, "FS_DIRENT_NODE_KIND_DIR", PyLong_FromLong(svn_node_dir));
+    PyModule_AddObject(mod, "FS_DIRENT_NODE_KIND_UNKNOWN", PyLong_FromLong(svn_node_unknown));
+
+#if ONLY_SINCE_SVN(1, 8)
+    PyModule_AddObject(mod, "FS_DIRENT_NODE_KIND_SYMLINK", PyLong_FromLong(svn_node_symlink));
+#endif
+
 	return mod;
 }

Index: subvertpy/util.c
===================================================================
--- contrib/python/patched/subvertpy/subvertpy/util.c	(rev 0)
+++ contrib/python/patched/subvertpy/subvertpy/util.c	(working copy)
@@ -42,6 +42,65 @@ void PyErr_SetAprStatus(apr_status_t status)
 		apr_strerror(status, errmsg, sizeof(errmsg)));
 }

+const char *py_object_to_svn_path_or_url(PyObject *obj, apr_pool_t *pool)
+{
+    const char *ret;
+    if (PyUnicode_Check(obj)) {
+        obj = PyUnicode_AsUTF8String(obj);
+        if (obj == NULL) {
+            return NULL;
+        }
+    } else {
+        Py_INCREF(obj);
+    }
+    if (!PyBytes_Check(obj)) {
+        PyErr_SetString(PyExc_TypeError,
+                        "URIs need to be UTF-8 bytestrings or unicode strings");
+        Py_DECREF(obj);
+        return NULL;
+    }
+    ret = PyBytes_AsString(obj);
+    if (svn_path_is_url(ret)) {
+        ret = svn_uri_canonicalize(ret, pool);
+    } else {
+        ret = svn_dirent_canonicalize(ret, pool);
+    }
+    Py_DECREF(obj);
+    return ret;
+}
+
+const char *py_object_to_svn_abspath(PyObject *obj, apr_pool_t *pool)
+{
+    const char *ret;
+    if (PyUnicode_Check(obj)) {
+        obj = PyUnicode_AsUTF8String(obj);
+        if (obj == NULL) {
+            return NULL;
+        }
+    } else {
+        Py_INCREF(obj);
+    }
+    if (!PyBytes_Check(obj)) {
+        PyErr_SetString(PyExc_TypeError,
+                        "URIs need to be UTF-8 bytestrings or unicode strings");
+        Py_DECREF(obj);
+        return NULL;
+    }
+    ret = PyBytes_AsString(obj);
+    ret = apr_pstrdup(pool, ret);
+    Py_DECREF(obj);
+    if (ret == NULL) {
+        return NULL;
+    }
+    if (svn_dirent_is_absolute(ret)) {
+        return svn_dirent_canonicalize(ret, pool);
+    } else {
+        const char *absolute;
+        RUN_SVN(svn_dirent_get_absolute(&absolute, ret, pool))
+        return svn_dirent_canonicalize(absolute, pool);
+    }
+}
+
 #if ONLY_BEFORE_SVN(1, 7)
 const char *
 svn_uri_canonicalize(const char *uri,
Index: subvertpy/util.h
===================================================================
--- contrib/python/patched/subvertpy/subvertpy/util.h	(rev 0)
+++ contrib/python/patched/subvertpy/subvertpy/util.h	(working copy)
@@ -137,6 +137,9 @@ typedef struct {

 extern PyTypeObject Stream_Type;

+const char *py_object_to_svn_path_or_url(PyObject *obj, apr_pool_t *pool);
+const char *py_object_to_svn_abspath(PyObject *obj, apr_pool_t *pool);
+
 #if ONLY_BEFORE_SVN(1, 7)
 const char *
 svn_uri_canonicalize(const char *uri,
Index: subvertpy/wc.c
===================================================================
--- contrib/python/patched/subvertpy/subvertpy/wc.c	(rev 0)
+++ contrib/python/patched/subvertpy/subvertpy/wc.c	(working copy)
@@ -379,6 +379,8 @@ static PyMemberDef entry_members[] = {
 		"Last date this was changed" },
 	{ "cmt_author", T_STRING, offsetof(EntryObject, entry.cmt_author), READONLY,
 		"Last commit author of this item" },
+	{ "changelist", T_STRING, offsetof(EntryObject, entry.changelist), READONLY,
+		"Which changelist this item is part of, or None if not part of any" },
 	{ NULL, }
 };

@@ -2253,16 +2255,22 @@ static PyObject *ra_status(PyObject *self, PyObject *args)
 	return (PyObject*)ret;
 }

+/* Subversion 1.7 has a couple of significant behaviour changes that break subvertpy.
+ * We haven't updated the code to deal with these changes in behaviour yet.
+ * */
 static PyMethodDef adm_methods[] = {
+	/*
 	{ "prop_set", adm_prop_set, METH_VARARGS, "S.prop_set(name, value, path, skip_checks=False)" },
 	{ "access_path", (PyCFunction)adm_access_path, METH_NOARGS,
 		"S.access_path() -> path\n"
 		"Returns the base path for this working copy handle." },
 	{ "prop_get", adm_prop_get, METH_VARARGS, "S.prop_get(name, path) -> value" },
 	{ "entries_read", adm_entries_read, METH_VARARGS, "S.entries_read(include_hidden=False) -> dict" },
+	*/
 	{ "walk_entries", adm_walk_entries, METH_VARARGS,
 		"S.walk_entries(path, callback, show_hidden=False)\n"
 		"callback should be a function that takes a path and a wc entry" },
+	/*
 	{ "locked", (PyCFunction)adm_locked, METH_NOARGS,
 		"S.locked() -> bool" },
 	{ "get_prop_diffs", adm_get_prop_diffs, METH_VARARGS,
@@ -2281,8 +2289,10 @@ static PyMethodDef adm_methods[] = {
 	{ "process_committed_queue", (PyCFunction)adm_process_committed_queue, METH_VARARGS, "S.process_committed_queue(queue, new_revnum, rev_date, rev_author)" },
 	{ "remove_lock", (PyCFunction)adm_remove_lock, METH_VARARGS, "S.remove_lock(path)" },
 	{ "has_binary_prop", (PyCFunction)adm_has_binary_prop, METH_VARARGS, "S.has_binary_prop(path) -> bool" },
+	*/
 	{ "text_modified", (PyCFunction)adm_text_modified, METH_VARARGS, "S.text_modified(filename, force_comparison=False) -> bool" },
 	{ "props_modified", (PyCFunction)adm_props_modified, METH_VARARGS, "S.props_modified(filename) -> bool" },
+	/*
 	{ "get_ancestry", (PyCFunction)get_ancestry, METH_VARARGS,
 		"S.get_ancestry(path) -> (url, rev)" },
 	{ "maybe_set_repos_root", (PyCFunction)maybe_set_repos_root, METH_VARARGS, "S.maybe_set_repos_root(path, repos)" },
@@ -2314,6 +2324,7 @@ static PyMethodDef adm_methods[] = {
 		"S.conflicted(path) -> (text_conflicted, prop_conflicted, tree_conflicted)" },
 	{ "resolved_conflict", (PyCFunction)resolved_conflict, METH_VARARGS,
 		"S.resolved_conflict(path, resolve_text, resolve_props, resolve_tree, depth, conflict_choice, notify_func=None, cancel=None)" },
+	*/
 	{ "status", (PyCFunction)ra_status, METH_VARARGS, "status(wc_path) -> Status" },
 	{ NULL, }
 };
@@ -3030,7 +3041,7 @@ moduleinit(void)
 	};
 	mod = PyModule_Create(&moduledef);
 #else
-	mod = Py_InitModule3("wc", wc_methods, "Working Copies");
+	mod = Py_InitModule3("subvertpy.wc", wc_methods, "Working Copies");
 #endif
 	if (mod == NULL)
 		return NULL;
@@ -3089,13 +3100,14 @@ moduleinit(void)
 	PyModule_AddIntConstant(mod, "CONFLICT_CHOOSE_MERGED", svn_wc_conflict_choose_merged);
 #endif

-#if ONLY_BEFORE_SVN(1, 7)
 	/* Subversion 1.7 has a couple of significant behaviour changes that break subvertpy.
 	 * We haven't updated the code to deal with these changes in behaviour yet.
+	 * (workfork) Some methods are working and tested though, so enable Adm_Type partially (see adm_methods).
 	 * */
 	PyModule_AddObject(mod, "WorkingCopy", (PyObject *)&Adm_Type);
 	Py_INCREF(&Adm_Type);

+#if ONLY_BEFORE_SVN(1, 7)
 	PyModule_AddObject(mod, "CommittedQueue", (PyObject *)&CommittedQueue_Type);
 	Py_INCREF(&CommittedQueue_Type);
 #endif
