--- contrib/python/namedlist/py2/tests/test_namedlist.py	(index)
+++ contrib/python/namedlist/py2/tests/test_namedlist.py	(working tree)
@@ -29,6 +29,11 @@ import unittest
 import collections
 import unicodedata
 
+try:
+    import collections.abc as collections_abc
+except ImportError:
+    import collections as collections_abc
+
 _PY2 = sys.version_info[0] == 2
 _PY3 = sys.version_info[0] == 3
 
@@ -526,10 +531,10 @@ class TestNamedList(unittest.TestCase):
     def test_ABC(self):
         Point = namedlist('Point', 'a b c')
         p = Point(1, 2, 2)
-        self.assertIsInstance(p, collections.Container)
-        self.assertIsInstance(p, collections.Iterable)
-        self.assertIsInstance(p, collections.Sized)
-        self.assertIsInstance(p, collections.Sequence)
+        self.assertIsInstance(p, collections_abc.Container)
+        self.assertIsInstance(p, collections_abc.Iterable)
+        self.assertIsInstance(p, collections_abc.Sized)
+        self.assertIsInstance(p, collections_abc.Sequence)
 
         self.assertEqual(list(reversed(p)), [2, 2, 1])
         self.assertEqual(p.count(0), 0)
@@ -817,4 +822,4 @@ def _add_unittest_methods(cls):
 
 _add_unittest_methods(TestNamedList)
 
-unittest.main()
+#unittest.main()
--- contrib/python/namedlist/py2/namedlist.py	(index)
+++ contrib/python/namedlist/py2/namedlist.py	(working tree)
@@ -43,6 +43,11 @@ from keyword import iskeyword as _iskeyword
 import collections as _collections
 import abc as _abc
 
+try:
+    import collections.abc as _collections_abc
+except ImportError:
+    import collections as _collections_abc
+
 _PY2 = _sys.version_info[0] == 2
 _PY3 = _sys.version_info[0] == 3
 _PY38_or_higher = _PY3 and _sys.version_info.minor >= 8
@@ -249,7 +254,7 @@ def _fields_and_defaults(typename, field_names, default, rename):
 
     # If field_names is a Mapping, change it to return the
     #  (field_name, default) pairs, as if it were a list.
-    if isinstance(field_names, _collections.Mapping):
+    if isinstance(field_names, _collections_abc.Mapping):
         field_names = field_names.items()
 
     # Parse and validate the field names.
@@ -365,7 +370,7 @@ def _nl_index(self, value, start=NO_DEFAULT, stop=NO_DEFAULT):
 def _nl_update(_self, _other=None, **kwds):
     if isinstance(_other, type(_self)):
         _other = zip(_self._fields, _other)
-    elif isinstance(_other, _collections.Mapping):
+    elif isinstance(_other, _collections_abc.Mapping):
         tmp = []
         for field_name in _self._fields:
             try:
@@ -420,7 +425,7 @@ def namedlist(typename, field_names, default=NO_DEFAULT, rename=False,
     t = type(typename, (object,), type_dict)
 
     # Register its ABC's
-    _collections.Sequence.register(t)
+    _collections_abc.Sequence.register(t)
 
     # And return it.
     return t
