--- contrib/python/flower/flower/__init__.py	(45d58b333abeda3e03ead03f338820022724c867)
+++ contrib/python/flower/flower/__init__.py	(925ddd19c66a913340c78704b20376e7fe8543ef)
@@ -0,0 +0,5 @@
+import os
+import pkgutil
+import tempfile
+import __res
+
@@ -2,1 +7,23 @@
 __version__ = '.'.join(map(str, VERSION))
+
+resource_temp_dir = None
+
+
+def extract_resources():
+    global resource_temp_dir
+    if resource_temp_dir is not None:
+        return
+    resource_temp_dir = tempfile.TemporaryDirectory()
+    resource_files_path = __res.resfs_files("contrib/python/flower/flower")
+    for files_path in resource_files_path:
+        rel_resource_path = os.path.relpath(files_path.decode("utf-8"), "contrib/python/flower/flower")
+        rel_new_resource_temp_dir_path = os.path.dirname(rel_resource_path)
+        abs_new_resource_temp_dir_path = os.path.join(resource_temp_dir.name, rel_new_resource_temp_dir_path)
+        os.makedirs(abs_new_resource_temp_dir_path, exist_ok=True)
+
+        resource_data = pkgutil.get_data("flower", rel_resource_path)
+        with open(os.path.join(resource_temp_dir.name, rel_resource_path), "wb") as new_resource_temp_file:
+            new_resource_temp_file.write(resource_data)
+
+
+extract_resources()
--- contrib/python/flower/flower/urls.py	(45d58b333abeda3e03ead03f338820022724c867)
+++ contrib/python/flower/flower/urls.py	(925ddd19c66a913340c78704b20376e7fe8543ef)
@@ -13,4 +13,5 @@ from .views.tasks import TaskView, TasksView, TasksDataTable
 from .views.tasks import TasksDataTable, TasksView, TaskView
 from .views.workers import WorkersView, WorkerView
+from . import resource_temp_dir
 
 settings = dict(
@@ -17,2 +17,2 @@ from .views.tasks import TaskView, TasksView, TasksDataTable
-    template_path=os.path.join(os.path.dirname(__file__), "templates"),
-    static_path=os.path.join(os.path.dirname(__file__), "static"),
+    template_path=os.path.join(resource_temp_dir.name, "templates"),
+    static_path=os.path.join(resource_temp_dir.name, "static"),
