--- contrib/python/pytest-datafixtures/datafixtures/entry.py	(index)
+++ contrib/python/pytest-datafixtures/datafixtures/entry.py	(working tree)
@@ -3,6 +3,7 @@ from pathlib import Path
 from typing import Optional, Union
 
 import pytest
+import yatest.common
 
 if False:  # pragma nocover
     from _pytest.fixtures import SubRequest  # noqa
@@ -11,7 +12,8 @@ if False:  # pragma nocover
 @pytest.fixture
 def datafix_dir(request: 'SubRequest') -> Path:
     """Returns data fixtures directory (as Path object) for test."""
-    return Path(request.module.__file__).absolute().parent / 'datafixtures'
+    subdir = request.module.__package__.removeprefix('__tests__').strip('.').replace('.', '/')
+    return Path(yatest.common.test_source_path(subdir)) / 'datafixtures'


 @pytest.fixture
