diff --git a/contrib/python/moto/moto/ec2/models.py b/contrib/python/moto/moto/ec2/models.py
--- a/contrib/python/moto/moto/ec2/models.py
+++ b/contrib/python/moto/moto/ec2/models.py
@@ -120,13 +120,11 @@ from .utils import (
     tag_filter_matches,
 )
 
-INSTANCE_TYPES = json.load(
-    open(resource_filename(__name__, 'resources/instance_types.json'), 'r')
-)
-AMIS = json.load(
-    open(os.environ.get('MOTO_AMIS_PATH') or resource_filename(
-         __name__, 'resources/amis.json'), 'r')
-)
+# load from ya-resources
+import library.python.resource as _ya_res
+
+INSTANCE_TYPES = json.loads(_ya_res.find('resource/instance_types.json'))
+AMIS = json.loads(_ya_res.find('resource/amis.json'))
 
 
 def utc_date_and_time():
