--- contrib/python/python-crfsuite/py2/pycrfsuite/crfsuite_api.pxd	(index)
+++ contrib/python/python-crfsuite/py2/pycrfsuite/crfsuite_api.pxd	(working tree)
@@ -1,7 +1,7 @@
 from libcpp.string cimport string
 from libcpp.vector cimport vector
 
-cdef extern from "../crfsuite/include/crfsuite.h":
+cdef extern from "crfsuite.h":
     ctypedef enum:
         CRFSUITE_SUCCESS
         CRFSUITEERR_UNKNOWN         # Unknown error occurred.
@@ -13,7 +13,7 @@ cdef extern from "../crfsuite/include/crfsuite.h":
         CRFSUITEERR_NOTIMPLEMENTED  # Not implemented.
 
 
-cdef extern from "../crfsuite/include/crfsuite_api.hpp" namespace "CRFSuite":
+cdef extern from "crfsuite_api.hpp" namespace "CRFSuite":
     cdef cppclass Attribute:
         string attr
         double value
