--- a/src/pcre2_jit_simd_inc.h	(index)
+++ b/src/pcre2_jit_simd_inc.h	(working tree)
@@ -766,7 +766,7 @@ if (common->match_end_ptr != 0)
 
 #endif /* SLJIT_CONFIG_X86 */
 
-#if (defined SLJIT_CONFIG_ARM_64 && SLJIT_CONFIG_ARM_64 && (defined __ARM_NEON || defined __ARM_NEON__))
+#if 0 // (defined SLJIT_CONFIG_ARM_64 && SLJIT_CONFIG_ARM_64 && (defined __ARM_NEON || defined __ARM_NEON__))
 
 #include <arm_neon.h>
 
