--- contrib/python/plyvel/py2/plyvel/_plyvel.pyx	(index)
+++ contrib/python/plyvel/py2/plyvel/_plyvel.pyx	(working tree)
@@ -182,8 +182,8 @@ cdef int parse_options(Options *options, c_bool create_if_missing,
     if block_restart_interval is not None:
         options.block_restart_interval = block_restart_interval
 
-    if max_file_size is not None:
-        options.max_file_size = max_file_size
+    #if max_file_size is not None:
+    #    options.max_file_size = max_file_size
 
     if compression is None:
         options.compression = leveldb.kNoCompression
--- contrib/python/plyvel/py2/plyvel/leveldb.pxd	(index)
+++ contrib/python/plyvel/py2/plyvel/leveldb.pxd	(working tree)
@@ -69,7 +69,7 @@ cdef extern from "leveldb/options.h" namespace "leveldb":
         Cache* block_cache
         size_t block_size
         int block_restart_interval
-        size_t max_file_size
+        #size_t max_file_size
         CompressionType compression
         FilterPolicy* filter_policy
         Options() nogil
