--- mecab/src/mmap.h	(0b9fce48f99f42f308b54bdb6b7613adb34a8437)
+++ mecab/src/mmap.h	(7ef79a02ea4ee9b9d4a4f7441abd8106246adfb8)
@@ -9,6 +9,8 @@
 #include <errno.h>
 #include <string>
 
+#include <library/cpp/resource/resource.h>
+
 #ifdef HAVE_CONFIG_H
 #include "config.h"
 #endif
@@ -61,6 +63,7 @@ template <class T> class Mmap {
   T            *text;
   size_t       length;
   std::string  fileName;
+  TString      contents;
   whatlog what_;
 
 #if defined(_WIN32) && !defined(__CYGWIN__)
@@ -142,6 +145,13 @@ template <class T> class Mmap {
     struct stat st;
     fileName = std::string(filename);
 
+    if (fileName.starts_with("resfs://")) {
+      contents = NResource::Find(TString::Join("resfs/file/", fileName.substr(8)));
+      text = reinterpret_cast<T*>(contents.begin());
+      length = contents.length();
+      return true;
+    }
+
     if      (std::strcmp(mode, "r") == 0)
       flag = O_RDONLY;
     else if (std::strcmp(mode, "r+") == 0)
@@ -179,6 +189,12 @@ template <class T> class Mmap {
   }
 
   void close() {
+    if (text && contents.size()) {
+      text = 0;
+      contents.clear();
+      return;
+    }
+
     if (fd >= 0) {
       ::close(fd);
       fd = -1;
--- mecab/src/param.cpp	(0b9fce48f99f42f308b54bdb6b7613adb34a8437)
+++ mecab/src/param.cpp	(7ef79a02ea4ee9b9d4a4f7441abd8106246adfb8)
@@ -14,6 +14,8 @@
 #include "config.h"
 #endif
 
+#include <library/cpp/resource/resource.h>
+
 namespace MeCab {
 namespace {
 void init_param(std::string *help,
@@ -64,7 +66,16 @@ void Param::dump_config(std::ostream *os) const {
 }
 
 bool Param::load(const char *filename) {
-  std::ifstream ifs(WPATH(filename));
+  std::string contents;
+  std::unique_ptr<std::istream> streamPtr;
+  if (TStringBuf resname(filename); resname.SkipPrefix("resfs://")) {
+    auto resource = NResource::Find(TString::Join("resfs/file/", resname));
+    contents = resource;
+    streamPtr = std::make_unique<std::istringstream>(contents);
+  } else {
+    streamPtr = std::make_unique<std::ifstream>(WPATH(filename));
+  }
+  std::istream& ifs = *streamPtr;
 
   CHECK_FALSE(ifs) << "no such file or directory: " << filename;
 
