--- a/python/CMakeLists.txt
+++ b/python/CMakeLists.txt
@@ -36,9 +36,12 @@ if( HAVE_PYTHON2 )
     ####### Do not invoke swig. Use our own generated C wrapper file ######
     set( _gribapi_swig "gribapi/_gribapi_swig${CMAKE_SHARED_LIBRARY_SUFFIX}" )
     # Build the extension module for use in build tree with RPATH pointing to the build tree
+    add_custom_command( OUTPUT gribapi_swig.c gribapi_swig.py
+                        COMMAND swig -python -module gribapi_swig -o gribapi_swig.c ${CMAKE_CURRENT_SOURCE_DIR}/gribapi_swig.i
+                        DEPENDS grib_errors.h gribapi_swig.i numpy.i )
     add_custom_command( OUTPUT ${_gribapi_swig}
                         COMMAND ${PYTHON_EXECUTABLE} setup.py build_ext --inplace --rpath ${CMAKE_BINARY_DIR}/${INSTALL_LIB_DIR}
-                        DEPENDS grib_interface.h grib_interface.c swig_wrap_numpy.c setup.py.in eccodes )
+                        DEPENDS grib_interface.h grib_interface.c gribapi_swig.c setup.py.in eccodes )
     add_custom_target(eccodes_build_swig_wrapper ALL DEPENDS ${_gribapi_swig})
 
     #ecbuild_add_library(TARGET    _gribapi_swig
@@ -52,7 +55,6 @@ if( HAVE_PYTHON2 )
 
     # Copy gribapi and eccodes python modules to build area
     file( COPY eccodes gribapi DESTINATION . )
-    configure_file( swig_wrap_numpy.py gribapi/gribapi_swig.py COPYONLY )
 
     # Build the extension module for use in install tree with RPATH pointing to install tree
     install(CODE "message(\"Building Python extension modules:
--- a/python/setup.py.in
+++ b/python/setup.py.in
@@ -12,15 +12,17 @@ except AttributeError:
     numpy_include = numpy.get_numpy_include()
 
 # See ECC-644
-extra_compile_args = []
+extra_compile_args = ['-DHAVE_ECCODES_CONFIG_H']
 cmake_c_compiler_id='@CMAKE_C_COMPILER_ID@'
 if cmake_c_compiler_id == 'PGI':
     extra_compile_args.append('-noswitcherror')
 
-attdict = dict(sources=['@CMAKE_CURRENT_SOURCE_DIR@/swig_wrap_numpy.c',
+attdict = dict(sources=['@CMAKE_CURRENT_BINARY_DIR@/gribapi_swig.c',
                         '@CMAKE_CURRENT_SOURCE_DIR@/grib_interface.c'],
                swig_opts=[],
                include_dirs=['.', '@CMAKE_CURRENT_BINARY_DIR@/../src',
+                             '@CMAKE_BINARY_DIR@',  # for eccodes_config.h
+                             '@CMAKE_CURRENT_SOURCE_DIR@',  # for grib_interface.h
                              '@CMAKE_CURRENT_SOURCE_DIR@/../src',
                              numpy_include],
                library_dirs=['@CMAKE_BINARY_DIR@/lib'],
