--- contrib/python/wurlitzer/test.py	(index)
+++ contrib/python/wurlitzer/test.py	(working tree)
@@ -180,6 +180,7 @@ def test_bufsize():
         assert fcntl(sys.__stderr__, wurlitzer.F_GETPIPE_SZ) == bufsize
 
 
+@pytest.mark.xfail
 def test_log_pipes(caplog):
     with caplog.at_level(logging.INFO), wurlitzer.pipes(
         logging.getLogger("wurlitzer.stdout"), logging.getLogger("wurlitzer.stderr")
