diff --git a/omp.h b/omp.h
index f2e6345..cb2fe49 100644
--- a/omp.h
+++ b/omp.h
@@ -15,7 +15,13 @@
 #ifndef __OMP_H
 #   define __OMP_H
 
+#if 0  // !defined(NORUNTIME) && !defined(USE_STL_SYSTEM)
+    // We need to put all possible dependencies to prevent blinking:
+    // on all stdlib.h that can be mentioned here within a platform.
+#   include <contrib/libs/cxxsupp/libcxx/include/stdlib.h>
+#else
 #   include <stdlib.h>
+#endif
 #   include <stdint.h>
 
 #   define KMP_VERSION_MAJOR    5
