diff --git a/implib-gen.py b/implib-gen.py
index 8ecb902..723e6a4 100755
--- a/implib-gen.py
+++ b/implib-gen.py
@@ -11,6 +11,7 @@
 Generates static import library for POSIX shared library
 """
 
+import io
 import sys
 import os.path
 import re
@@ -19,8 +20,13 @@ import argparse
 import string
 import configparser
 
+from library.python.resource import resfs_read, resfs_file_exists
+
 me = os.path.basename(__file__)
-root = os.path.dirname(__file__)
+root = ''
+
+def resfs_open(path):
+    return io.StringIO(resfs_read(path).decode())
 
 def warn(msg):
   """Emits a nicely-decorated warning."""
@@ -462,11 +468,11 @@ Examples:
 
   target_dir = os.path.join(root, 'arch', target)
 
-  if not os.path.exists(target_dir):
+  if not resfs_file_exists(target_dir + '/config.ini'):
     error(f"unknown architecture '{target}'")
 
   cfg = configparser.ConfigParser(inline_comment_prefixes=';')
-  cfg.read(target_dir + '/config.ini')
+  cfg.read_file(resfs_open(target_dir + '/config.ini'))
 
   ptr_size = int(cfg['Arch']['PointerSize'])
   symbol_reloc_types = set(re.split(r'\s*,\s*', cfg['Arch']['SymbolReloc']))
@@ -580,13 +586,13 @@ Examples:
   with open(os.path.join(outdir, tramp_file), 'w') as f:
     if not quiet:
       print(f"Generating {tramp_file}...")
-    with open(target_dir + '/table.S.tpl', 'r') as t:
+    with resfs_open(target_dir + '/table.S.tpl') as t:
       table_text = string.Template(t.read()).substitute(
         lib_suffix=lib_suffix,
         table_size=ptr_size*(len(funs) + 1))
     f.write(table_text)
 
-    with open(target_dir + '/trampoline.S.tpl', 'r') as t:
+    with resfs_open(target_dir + '/trampoline.S.tpl') as t:
       tramp_tpl = string.Template(t.read())
 
     for i, name in enumerate(funs):
@@ -603,7 +609,7 @@ Examples:
   with open(os.path.join(outdir, init_file), 'w') as f:
     if not quiet:
       print(f"Generating {init_file}...")
-    with open(os.path.join(root, 'arch/common/init.c.tpl'), 'r') as t:
+    with resfs_open(os.path.join(root, 'arch/common/init.c.tpl')) as t:
       if funs:
         sym_names = ',\n  '.join(f'"{name}"' for name in funs) + ','
       else:
