diff --git a/implib-gen.py b/implib-gen.py
index 723e6a4..8a6fd21 100755
--- a/implib-gen.py
+++ b/implib-gen.py
@@ -425,7 +425,9 @@ Examples:
   dlsym_callback = args.dlsym_callback
   dlopen = args.dlopen
   lazy_load = args.lazy_load
-  if args.target.startswith('arm'):
+  if args.target.startswith('armv8'):
+    target = 'aarch64'
+  elif args.target.startswith('arm'):
     target = 'arm'  # Handle armhf-..., armel-...
   elif re.match(r'^i[0-9]86', args.target):
     target = 'i386'
