diff --git a/arch/aarch64/trampoline.S.tpl b/arch/aarch64/trampoline.S.tpl
index cdb26a7..c0e0701 100644
--- a/arch/aarch64/trampoline.S.tpl
+++ b/arch/aarch64/trampoline.S.tpl
@@ -7,7 +7,11 @@
  * found in the LICENSE.txt file.
  */
 
+#ifndef IMPLIB_IMPORT_AS_WEAK
   .globl $sym
+#else
+  .weak $sym
+#endif
   .p2align 4
   .type $sym, %function
 #ifndef IMPLIB_EXPORT_SHIMS
diff --git a/arch/arm/trampoline.S.tpl b/arch/arm/trampoline.S.tpl
index fd01484..db2e7b3 100644
--- a/arch/arm/trampoline.S.tpl
+++ b/arch/arm/trampoline.S.tpl
@@ -7,7 +7,11 @@
  * found in the LICENSE.txt file.
  */
 
+#ifndef IMPLIB_IMPORT_AS_WEAK
   .globl $sym
+#else
+  .weak $sym
+#endif
   .p2align 4
   .type $sym, %function
 #ifndef IMPLIB_EXPORT_SHIMS
diff --git a/arch/e2k/trampoline.S.tpl b/arch/e2k/trampoline.S.tpl
index 57beccc..7a35e16 100644
--- a/arch/e2k/trampoline.S.tpl
+++ b/arch/e2k/trampoline.S.tpl
@@ -7,7 +7,11 @@
  * found in the LICENSE.txt file.
  */
 
+#ifndef IMPLIB_IMPORT_AS_WEAK
   .globl $sym
+#else
+  .weak $sym
+#endif
   .p2align 3
   .ignore strict_delay
   .type $sym, %function
diff --git a/arch/i386/trampoline.S.tpl b/arch/i386/trampoline.S.tpl
index 98b07c4..975e6e6 100644
--- a/arch/i386/trampoline.S.tpl
+++ b/arch/i386/trampoline.S.tpl
@@ -7,7 +7,11 @@
  * found in the LICENSE.txt file.
  */
 
+#ifndef IMPLIB_IMPORT_AS_WEAK
   .globl $sym
+#else
+  .weak $sym
+#endif
   .p2align 4
   .type $sym, %function
 #ifndef IMPLIB_EXPORT_SHIMS
diff --git a/arch/mips/trampoline.S.tpl b/arch/mips/trampoline.S.tpl
index a37ec56..6d636b3 100644
--- a/arch/mips/trampoline.S.tpl
+++ b/arch/mips/trampoline.S.tpl
@@ -7,7 +7,11 @@
  * found in the LICENSE.txt file.
  */
 
+#ifndef IMPLIB_IMPORT_AS_WEAK
   .globl $sym
+#else
+  .weak $sym
+#endif
   .p2align 4
   .type $sym, %function
 #ifndef IMPLIB_EXPORT_SHIMS
diff --git a/arch/mips64/trampoline.S.tpl b/arch/mips64/trampoline.S.tpl
index 057e87d..a1ee090 100644
--- a/arch/mips64/trampoline.S.tpl
+++ b/arch/mips64/trampoline.S.tpl
@@ -7,7 +7,11 @@
  * found in the LICENSE.txt file.
  */
 
+#ifndef IMPLIB_IMPORT_AS_WEAK
   .globl $sym
+#else
+  .weak $sym
+#endif
   .p2align 4
   .type $sym, %function
 #ifndef IMPLIB_EXPORT_SHIMS
diff --git a/arch/powerpc64/trampoline.S.tpl b/arch/powerpc64/trampoline.S.tpl
index 7a54c8c..1c050c3 100644
--- a/arch/powerpc64/trampoline.S.tpl
+++ b/arch/powerpc64/trampoline.S.tpl
@@ -7,7 +7,11 @@
  * found in the LICENSE.txt file.
  */
 
+#ifndef IMPLIB_IMPORT_AS_WEAK
   .globl $sym
+#else
+  .weak $sym
+#endif
   .p2align 4
   .type $sym, %function
 #ifndef IMPLIB_EXPORT_SHIMS
diff --git a/arch/powerpc64le/trampoline.S.tpl b/arch/powerpc64le/trampoline.S.tpl
index 0d697f4..a58c86d 100644
--- a/arch/powerpc64le/trampoline.S.tpl
+++ b/arch/powerpc64le/trampoline.S.tpl
@@ -7,7 +7,11 @@
  * found in the LICENSE.txt file.
  */
 
+#ifndef IMPLIB_IMPORT_AS_WEAK
   .globl $sym
+#else
+  .weak $sym
+#endif
   .p2align 4
   .type $sym, %function
 #ifndef IMPLIB_EXPORT_SHIMS
diff --git a/arch/riscv64/trampoline.S.tpl b/arch/riscv64/trampoline.S.tpl
index 9064aec..15822f3 100644
--- a/arch/riscv64/trampoline.S.tpl
+++ b/arch/riscv64/trampoline.S.tpl
@@ -7,7 +7,11 @@
  * found in the LICENSE.txt file.
  */
 
+#ifndef IMPLIB_IMPORT_AS_WEAK
   .globl $sym
+#else
+  .weak $sym
+#endif
   .p2align 4
   .type $sym, %function
 #ifndef IMPLIB_EXPORT_SHIMS
diff --git a/arch/x86_64/trampoline.S.tpl b/arch/x86_64/trampoline.S.tpl
index 1beadec..bd893e8 100644
--- a/arch/x86_64/trampoline.S.tpl
+++ b/arch/x86_64/trampoline.S.tpl
@@ -7,7 +7,11 @@
  * found in the LICENSE.txt file.
  */
 
+#ifndef IMPLIB_IMPORT_AS_WEAK
   .globl $sym
+#else
+  .weak $sym
+#endif
   .p2align 4
   .type $sym, %function
 #ifndef IMPLIB_EXPORT_SHIMS
diff --git a/implib-gen.py b/implib-gen.py
index 8a6fd21..bfacad9 100755
--- a/implib-gen.py
+++ b/implib-gen.py
@@ -421,7 +421,7 @@ Examples:
 
   input_name = args.library
   verbose = args.verbose
-  dlopen_callback = args.dlopen_callback
+  dlopen_callback = '' if args.dlopen_callback == '_' else args.dlopen_callback
   dlsym_callback = args.dlsym_callback
   dlopen = args.dlopen
   lazy_load = args.lazy_load
