--- contrib/python/msgspec/tests/unit/test_struct.py	(index)
+++ contrib/python/msgspec/tests/unit/test_struct.py	(working tree)
@@ -1967 +1967 @@ class TestDefStruct:
-        assert Point.__module__ == "tests.unit.test_struct"
+        assert Point.__module__ == "__tests__.unit.test_struct"
@@ -2029 +2029 @@ class TestDefStruct:
-        assert Test.__module__ == "tests.unit.test_struct"
+        assert Test.__module__ == "__tests__.unit.test_struct"
--- contrib/python/msgspec/tests/unit/test_raw.py	(index)
+++ contrib/python/msgspec/tests/unit/test_raw.py	(working tree)
@@ -88,8 +88,11 @@ def test_raw_copy_doesnt_leak():
         print(peak)
         """
     )
+    import os
+    env = os.environ.copy()
+    env["Y_PYTHON_ENTRY_POINT"] = ":main"
 
-    output = subprocess.check_output([sys.executable, "-c", script])
+    output = subprocess.check_output([sys.executable, "-c", script], env=env)
     peak = int(output.decode().strip())
     assert peak < 10_000  # should really be ~2000
 
