From 452fddbcc35b60bf77c6d9559627707ce534b60c Mon Sep 17 00:00:00 2001
From: Alexander Shadchin <shadchin@yandex-team.com>
Date: Thu, 27 Nov 2025 14:52:23 +0300
Subject: [PATCH] Fix segmentation fault on Python 3.13

---
 src/msgspec/_core.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/msgspec/_core.c b/msgspec/_core.c
index d4151cd8..64f2b983 100644
--- a/msgspec/_core.c
+++ b/msgspec/_core.c
@@ -5337,6 +5337,11 @@ Struct_alloc(PyTypeObject *type) {
     if (obj == NULL) return NULL;
     /* Zero out slot fields */
     memset((char *)obj + sizeof(PyObject), '\0', type->tp_basicsize - sizeof(PyObject));
+#if PY313_PLUS && !PY314_PLUS
+    if (type->tp_flags & Py_TPFLAGS_INLINE_VALUES) {
+        _PyObject_InitInlineValues(obj, type);
+    }
+#endif
     return obj;
 }
 
