--- contrib/python/opensfm/opensfm/bow.py	(index)
+++ contrib/python/opensfm/opensfm/bow.py	(working tree)
@@ -5,9 +5,10 @@ from __future__ import unicode_literals
 
 import os.path
 import numpy as np
+import tempfile
 import cv2
 from opensfm import context
-
+from library.python import resource
 
 class BagOfWords:
     def __init__(self, words, frequencies):
@@ -49,8 +50,11 @@ def load_bow_words_and_frequencies(config):
         assert config['feature_root']
         assert config['hahog_normalize_to_uchar']
 
-    bow_file = os.path.join(context.BOW_PATH, config['bow_file'])
-    bow = np.load(bow_file)
+    with tempfile.TemporaryFile() as f:
+        f.write(resource.find('bow_hahog_10000'))
+        f.seek(0)
+        bow = np.load(f)
+
     return bow['words'], bow['frequencies']
 
 
@@ -60,8 +64,11 @@ def load_vlad_words_and_frequencies(config):
         assert config['feature_root']
         assert config['hahog_normalize_to_uchar']
 
-    vlad_file = os.path.join(context.BOW_PATH, config['vlad_file'])
-    vlad = np.load(vlad_file)
+    with tempfile.TemporaryFile() as f:
+        f.write(resource.find('bow_hahog_64'))
+        f.seek(0)
+        vlad = np.load(f)
+
     return vlad['words'], vlad['frequencies']
 
 
--- contrib/python/opensfm/opensfm/reconstruction.py	(index)
+++ contrib/python/opensfm/opensfm/reconstruction.py	(working tree)
@@ -250,6 +250,11 @@ def bundle_single_view(graph, reconstruction, shot_id, config):
     ba.add_shot(shot.id, camera.id, r, t, False)
 
     for track_id in graph[shot_id]:
+        # PATCH START
+        if track_id not in reconstruction.points:
+            continue
+        # PATCH END
+
         track = reconstruction.points[track_id]
         ba.add_point(track_id, track.coordinates, True)
         point = graph[shot_id][track_id]['feature']
@@ -728,6 +733,13 @@ def bootstrap_reconstruction(data, graph, im1, im2, p1, p2):
         report['decision'] = "Re-triangulation after initial motion did not generate enough points"
         logger.info(report['decision'])
         return None, None, report
+
+    # PATCH START
+    if im2 not in graph_inliers:
+        report['decision'] = "Shot dropped after retriangulation"
+        return None, None, report
+    # PATCH END
+
     bundle_single_view(graph_inliers, reconstruction, im2, camera_priors,
                        data.config)
 
@@ -799,7 +811,15 @@ def resect(graph, graph_inliers, reconstruction, shot_id,
         shot.id = shot_id
         shot.camera = camera
         shot.pose = types.Pose()
-        shot.pose.set_rotation_matrix(R)
+
+        # PATCH START
+        try:
+            shot.pose.set_rotation_matrix(R)
+        except ValueError:
+            report['decision'] = "Rotation matrix is not orthogonal"
+            return False, report
+        # PATCH END
+
         shot.pose.translation = t
         shot.metadata = metadata
         reconstruction.add_shot(shot)
--- contrib/python/opensfm/opensfm/sensors.py	(index)
+++ contrib/python/opensfm/opensfm/sensors.py	(working tree)
@@ -2,9 +2,9 @@ import json
 
 from opensfm import context
 from opensfm import io
+from library.python import resource
 
-with io.open_rt(context.SENSOR) as f:
-    sensor_data = io.json_load(f)
+sensor_data = io.json_loads(resource.find('sensor_data'))
 
 # Convert model types to lower cases for easier query
 keys = [k.lower() for k in sensor_data.keys()]
--- contrib/python/opensfm/opensfm/src/akaze_bind.cc	(index)
+++ contrib/python/opensfm/opensfm/src/akaze_bind.cc	(working tree)
@@ -1,7 +1,7 @@
 #include "types.h"
 #include "AKAZE.h"
 #include "akaze_bind.h"
-#include <opencv2/imgproc/imgproc.hpp>
+#include <opencv2/imgproc.hpp>
 
 namespace csfm {
 
--- contrib/python/opensfm/opensfm/src/hahog.cc	(index)
+++ contrib/python/opensfm/opensfm/src/hahog.cc	(working tree)
@@ -1,6 +1,8 @@
 #include "types.h"
 #include "hahog.h"
 
+#include <library/cpp/vl_feat/vl_feat.h>
+
 #include <vector>
 #include <iostream>
 
@@ -18,87 +20,96 @@ py::object hahog(pyarray_f image,
                  float edge_threshold,
                  int target_num_features,
                  bool use_adaptive_suppression) {
-  py::gil_scoped_release release;
 
   if (image.size()) {
-    //clock_t t_start = clock();
-    // create a detector object
-    VlCovDet * covdet = vl_covdet_new(VL_COVDET_METHOD_HESSIAN);
-    // set various parameters (optional)
-    vl_covdet_set_first_octave(covdet, 0);
-    //vl_covdet_set_octave_resolution(covdet, octaveResolution);
-    vl_covdet_set_peak_threshold(covdet, peak_threshold);
-    vl_covdet_set_edge_threshold(covdet, edge_threshold);
-    vl_covdet_set_target_num_features(covdet, target_num_features);
-    vl_covdet_set_use_adaptive_suppression(covdet, use_adaptive_suppression);
-
-    // process the image and run the detector
-    vl_covdet_put_image(covdet, image.data(), image.shape(1), image.shape(0));
-
-    //clock_t t_scalespace = clock();
-
-    vl_covdet_detect(covdet);
-
-    //clock_t t_detect = clock();
-
-    // compute the affine shape of the features (optional)
-    //vl_covdet_extract_affine_shape(covdet);
-
-    //clock_t t_affine = clock();
-
-    // compute the orientation of the features (optional)
-    vl_covdet_extract_orientations(covdet);
-
-    //clock_t t_orient = clock();
-
-    // get feature descriptors
-    vl_size numFeatures = vl_covdet_get_num_features(covdet);
-    VlCovDetFeature const *feature = (VlCovDetFeature const *)vl_covdet_get_features(covdet);
-    VlSiftFilt *sift = vl_sift_new(16, 16, 1, 3, 0);
-    vl_index i;
     vl_size dimension = 128;
-    vl_index patchResolution = 15;
-    double patchRelativeExtent = 7.5;
-    double patchRelativeSmoothing = 1;
-    vl_size patchSide = 2 * patchResolution + 1;
-    double patchStep = (double)patchRelativeExtent / patchResolution;
-    std::vector<float> points(4 * numFeatures);
-    std::vector<float> desc(dimension * numFeatures);
-    std::vector<float> patch(patchSide * patchSide);
-    std::vector<float> patchXY(2 * patchSide * patchSide);
-
-    vl_sift_set_magnif(sift, 3.0);
-    for (i = 0; i < (signed)numFeatures; ++i) {
-      const VlFrameOrientedEllipse &frame = feature[i].frame;
-      float det = frame.a11 * frame.a22 - frame.a12 * frame.a21;
-      float size = sqrt(fabs(det));
-      float angle = atan2(frame.a21, frame.a11) * 180.0f / M_PI;
-      points[4 * i + 0] = frame.x;
-      points[4 * i + 1] = frame.y;
-      points[4 * i + 2] = size;
-      points[4 * i + 3] = angle;
-
-      vl_covdet_extract_patch_for_frame(covdet,
-                                        &patch[0],
-                                        patchResolution,
-                                        patchRelativeExtent,
-                                        patchRelativeSmoothing,
-                                        frame);
-
-      vl_imgradient_polar_f(&patchXY[0], &patchXY[1],
-                            2, 2 * patchSide,
-                            &patch[0], patchSide, patchSide, patchSide);
-
-      vl_sift_calc_raw_descriptor(sift,
-                                  &patchXY[0],
-                                  &desc[dimension * i],
-                                  (int)patchSide, (int)patchSide,
-                                  (double)(patchSide - 1) / 2, (double)(patchSide - 1) / 2,
-                                  (double)patchRelativeExtent / (3.0 * (4 + 1) / 2) / patchStep,
-                                  VL_PI / 2);
+    vl_size numFeatures = 0;
+
+    std::vector<float> points;
+    std::vector<float> desc;
+    {
+      py::gil_scoped_release release;
+      // initialize vlfeat
+      TVlFeatToken token;
+      //clock_t t_start = clock();
+      // create a detector object
+      VlCovDet* covdet = vl_covdet_new(VL_COVDET_METHOD_HESSIAN);
+      // set various parameters (optional)
+      vl_covdet_set_first_octave(covdet, 0);
+      //vl_covdet_set_octave_resolution(covdet, octaveResolution);
+      vl_covdet_set_peak_threshold(covdet, peak_threshold);
+      vl_covdet_set_edge_threshold(covdet, edge_threshold);
+      vl_covdet_set_target_num_features(covdet, target_num_features);
+      vl_covdet_set_use_adaptive_suppression(covdet, use_adaptive_suppression);
+
+      // process the image and run the detector
+      vl_covdet_put_image(covdet, image.data(), image.shape(1), image.shape(0));
+
+      //clock_t t_scalespace = clock();
+
+      vl_covdet_detect(covdet);
+
+      //clock_t t_detect = clock();
+
+      // compute the affine shape of the features (optional)
+      //vl_covdet_extract_affine_shape(covdet);
+
+      //clock_t t_affine = clock();
+
+      // compute the orientation of the features (optional)
+      vl_covdet_extract_orientations(covdet);
+
+      //clock_t t_orient = clock();
+
+      // get feature descriptors
+      numFeatures = vl_covdet_get_num_features(covdet);
+      VlCovDetFeature const* feature = (VlCovDetFeature const*) vl_covdet_get_features(covdet);
+      VlSiftFilt* sift = vl_sift_new(16, 16, 1, 3, 0);
+      vl_index i;
+      //vl_size dimension = 128;
+      vl_index patchResolution = 15;
+      double patchRelativeExtent = 7.5;
+      double patchRelativeSmoothing = 1;
+      vl_size patchSide = 2 * patchResolution + 1;
+      double patchStep = (double) patchRelativeExtent / patchResolution;
+      points.resize(4 * numFeatures);
+      desc.resize(dimension * numFeatures);
+      std::vector<float> patch(patchSide * patchSide);
+      std::vector<float> patchXY(2 * patchSide * patchSide);
+
+      vl_sift_set_magnif(sift, 3.0);
+      for (i = 0; i < (signed) numFeatures; ++i) {
+        const VlFrameOrientedEllipse& frame = feature[i].frame;
+        float det = frame.a11 * frame.a22 - frame.a12 * frame.a21;
+        float size = sqrt(fabs(det));
+        float angle = atan2(frame.a21, frame.a11) * 180.0f / M_PI;
+        points[4 * i + 0] = frame.x;
+        points[4 * i + 1] = frame.y;
+        points[4 * i + 2] = size;
+        points[4 * i + 3] = angle;
+
+        vl_covdet_extract_patch_for_frame(covdet,
+                                          &patch[0],
+                                          patchResolution,
+                                          patchRelativeExtent,
+                                          patchRelativeSmoothing,
+                                          frame);
+
+        vl_imgradient_polar_f(&patchXY[0], &patchXY[1],
+                              2, 2 * patchSide,
+                              &patch[0], patchSide, patchSide, patchSide);
+
+        vl_sift_calc_raw_descriptor(sift,
+                                    &patchXY[0],
+                                    &desc[dimension * i],
+                                    (int) patchSide, (int) patchSide,
+                                    (double) (patchSide - 1) / 2, (double) (patchSide - 1) / 2,
+                                    (double) patchRelativeExtent / (3.0 * (4 + 1) / 2) / patchStep,
+                                    VL_PI / 2);
+      }
+      vl_sift_delete(sift);
+      vl_covdet_delete(covdet);
     }
-    vl_sift_delete(sift);
-    vl_covdet_delete(covdet);
 
     // clock_t t_description = clock();
     // std::cout << "t_scalespace " << float(t_scalespace - t_start)/CLOCKS_PER_SEC << "\n";
--- contrib/python/opensfm/opensfm/src/third_party/akaze/lib/AKAZE.h	(index)
+++ contrib/python/opensfm/opensfm/src/third_party/akaze/lib/AKAZE.h	(working tree)
@@ -15,7 +15,7 @@
 #include "nldiffusion_functions.h"
 
 // OpenCV
-#include <opencv2/features2d/features2d.hpp>
+#include <opencv2/features2d.hpp>
 
 /* ************************************************************************* */
 namespace libAKAZE {
--- contrib/python/opensfm/opensfm/src/third_party/akaze/lib/utils.h	(index)
+++ contrib/python/opensfm/opensfm/src/third_party/akaze/lib/utils.h	(working tree)
@@ -9,12 +9,13 @@
 
 /* ************************************************************************* */
 // OpenCV
-#include <opencv2/features2d/features2d.hpp>
+#include <opencv2/features2d.hpp>
 
 // System
 #include <vector>
 #include <iostream>
 #include <iomanip>
+#include <sstream>
 
 /* ************************************************************************* */
 // Stringify common types such as int, double and others.
--- contrib/python/opensfm/opensfm/src/types.h	(index)
+++ contrib/python/opensfm/opensfm/src/types.h	(working tree)
@@ -19,21 +19,21 @@ typedef py::array_t<int, py::array::c_style | py::array::forcecast> pyarray_int;
 typedef py::array_t<unsigned char, py::array::c_style | py::array::forcecast> pyarray_uint8;
 
 template <typename T>
-py::array_t<T> py_array_from_data(const T *data, size_t shape0) {
+py::array_t<T> py_array_from_data(const T *data, py::ssize_t shape0) {
   py::array_t<T> res({shape0});
   std::copy(data, data + shape0, res.mutable_data());
   return res;
 }
 
 template <typename T>
-py::array_t<T> py_array_from_data(const T *data, size_t shape0, size_t shape1) {
+py::array_t<T> py_array_from_data(const T *data, py::ssize_t shape0, py::ssize_t shape1) {
   py::array_t<T> res({shape0, shape1});
   std::copy(data, data + shape0 * shape1, res.mutable_data());
   return res;
 }
 
 template <typename T>
-py::array_t<T> py_array_from_data(const T *data, size_t shape0, size_t shape1, size_t shape2) {
+py::array_t<T> py_array_from_data(const T *data, py::ssize_t shape0, py::ssize_t shape1, py::ssize_t shape2) {
   py::array_t<T> res({shape0, shape1, shape2});
   std::copy(data, data + shape0 * shape1 * shape2, res.mutable_data());
   return res;
@@ -42,7 +42,7 @@ py::array_t<T> py_array_from_data(const T *data, size_t shape0, size_t shape1, s
 template <typename T>
 py::array_t<T> py_array_from_vector(const std::vector<T> &v) {
   const T *data = v.size() ? &v[0] : NULL;
-  return py_array_from_data(data, v.size());
+  return py_array_from_data(data, v.ssize());
 }
 
 template <typename T>
