--- contrib/python/django-fernet-fields/fernet_fields/test/settings/base.py	(index)
+++ contrib/python/django-fernet-fields/fernet_fields/test/settings/base.py	(working tree)
@@ -1,5 +1,5 @@
 INSTALLED_APPS = [
-    'fernet_fields.test',
+    'test',
 ]
 
 SECRET_KEY = 'secret'
--- contrib/python/django-fernet-fields/fernet_fields/test/settings/sqlite.py	(index)
+++ contrib/python/django-fernet-fields/fernet_fields/test/settings/sqlite.py	(working tree)
@@ -1,9 +1,6 @@
 from .base import *  # noqa
 
-import os
-
-HERE = os.path.dirname(os.path.abspath(__file__))
-DB = os.path.join(HERE, 'testdb.sqlite')
+DB = 'mydatabase'
 
 DATABASES = {
     'default': {
--- contrib/python/django-fernet-fields/fernet_fields/test/test_fields.py	(index)
+++ contrib/python/django-fernet-fields/fernet_fields/test/test_fields.py	(working tree)
@@ -7,7 +7,7 @@ from django.utils.encoding import force_bytes, force_text
 import pytest
 
 import fernet_fields as fields
-from . import models
+from test import models
 
 
 class TestEncryptedField(object):
