--- contrib/libs/yaml-cpp/include/yaml-cpp/dll.h	(index)
+++ contrib/libs/yaml-cpp/include/yaml-cpp/dll.h	(working tree)
@@ -7,26 +7,10 @@
 // Definition yaml_cpp_EXPORTS using to building YAML-CPP as dll/so library
 // (definition created by CMake or defined manually)
 
-#ifdef YAML_CPP_STATIC_DEFINE
+#if defined(YAML_CPP_STATIC_DEFINE) || defined(_MSC_VER) || defined(__MINGW32__) || defined(__MINGW64__)
 #  define YAML_CPP_API
 #  define YAML_CPP_NO_EXPORT
 #else
-#  if defined(_MSC_VER) || defined(__MINGW32__) || defined(__MINGW64__)
-#    ifndef YAML_CPP_API
-#      ifdef yaml_cpp_EXPORTS
-         /* We are building this library */
-#        pragma message( "Defining YAML_CPP_API for DLL export" )
-#        define YAML_CPP_API __declspec(dllexport)
-#      else
-         /* We are using this library */
-#        pragma message( "Defining YAML_CPP_API for DLL import" )
-#        define YAML_CPP_API __declspec(dllimport)
-#      endif
-#    endif
-#    ifndef YAML_CPP_NO_EXPORT
-#      define YAML_CPP_NO_EXPORT
-#    endif
-#  else /* No _MSC_VER */
 #    ifndef YAML_CPP_API
 #      ifdef yaml_cpp_EXPORTS
          /* We are building this library */
@@ -39,7 +23,6 @@
 #    ifndef YAML_CPP_NO_EXPORT
 #      define YAML_CPP_NO_EXPORT __attribute__((visibility("hidden")))
 #    endif
-#  endif /* _MSC_VER */
 #endif   /* YAML_CPP_STATIC_DEFINE */
 
 #ifndef YAML_CPP_DEPRECATED
