--- contrib/python/fiona/tests/test_transactions.py	(index)
+++ contrib/python/fiona/tests/test_transactions.py	(working tree)
@@ -8,7 +8,7 @@ import fiona
 from fiona.model import Feature
 import fiona.ogrext
 
-from tests.conftest import requires_gdal2
+from .conftest import requires_gdal2
 
 has_gpkg = "GPKG" in fiona.supported_drivers.keys()
 
--- contrib/python/fiona/tests/test_version.py	(index)
+++ contrib/python/fiona/tests/test_version.py	(working tree)
@@ -3,7 +3,7 @@ import platform
 import re
 import os
 import sys
-from tests.conftest import travis_only
+from .conftest import travis_only
 from fiona._env import GDALVersion, get_gdal_release_name
 
