--- contrib/python/fiona/fiona/_vsiopener.pyx	(index)
+++ contrib/python/fiona/fiona/_vsiopener.pyx	(working tree)
@@ -503,7 +503,7 @@ class FileContainer(ABC):
         pass
 
     @abstractmethod
-    def ls(self, path: str) -> list[str]:
+    def ls(self, path: str) -> list:
         """Get a 'directory' listing.
 
         Parameters
@@ -568,7 +568,7 @@ class FileContainer(ABC):
 class MultiByteRangeResource(ABC):
     """An object that provides VSIFilesystemPluginReadMultiRangeCallback."""
     @abstractmethod
-    def get_byte_ranges(self, offsets: list[int], sizes: list[int]) -> list[bytes]:
+    def get_byte_ranges(self, offsets: list, sizes: list) -> list:
         """Get a sequence of bytes specified by a sequence of ranges."""
         pass
 
