--- contrib/restricted/abseil-cpp-tstring/y_absl/synchronization/internal/futex_waiter.cc	(f090c19d9eef233290e4fd8af760f9991fbdb714)
+++ contrib/restricted/abseil-cpp-tstring/y_absl/synchronization/internal/futex_waiter.cc	(e6361f84d455fa283366753dd2cdf0491af4aa22)
@@ -77,7 +77,7 @@ bool FutexWaiter::Wait(KernelTimeout t) {
     if (!first_pass) MaybeBecomeIdle();
     const int err = WaitUntil(&futex_, 0, t);
     if (err != 0) {
-      if (err == -EINTR || err == -EWOULDBLOCK) {
+      if (err == -EINTR || err == -EWOULDBLOCK || err == -512 /* ERESTARTSYS */ || err == -516 /* ERESTART_RESTARTBLOCK */) {
         // Do nothing, the loop will retry.
       } else if (err == -ETIMEDOUT) {
         return false;
