--- contrib/restricted/abseil-cpp-tstring/y_absl/base/internal/nullability_impl.h	(index)
+++ contrib/restricted/abseil-cpp-tstring/y_absl/base/internal/nullability_impl.h	(working tree)
@@ -34,7 +34,7 @@ using NullableImpl
 // Don't add the _Nullable attribute in Objective-C compiles. Many Objective-C
 // projects enable the `-Wnullable-to-nonnull-conversion warning`, which is
 // liable to produce false positives.
-#if Y_ABSL_HAVE_FEATURE(nullability_on_classes) && !defined(__OBJC__)
+#if Y_ABSL_HAVE_FEATURE(nullability_on_classes) && !defined(__OBJC__) && !defined(__NVCC__)
     = T _Nullable;
 #else
     = T;
