--- contrib/restricted/abseil-cpp-tstring/y_absl/time/internal/cctz/src/time_zone_format.cc	(index)
+++ contrib/restricted/abseil-cpp-tstring/y_absl/time/internal/cctz/src/time_zone_format.cc	(working tree)
@@ -18,12 +18,6 @@
 #endif
 #endif
 
-#if defined(HAS_STRPTIME) && HAS_STRPTIME
-#if !defined(_XOPEN_SOURCE) && !defined(__FreeBSD__) && !defined(__OpenBSD__)
-#define _XOPEN_SOURCE 500  // Exposes definitions for SUSv2 (UNIX 98).
-#endif
-#endif
-
 #include "y_absl/base/config.h"
 #include "y_absl/time/internal/cctz/include/cctz/time_zone.h"
 
