--- contrib/python/swagger-spec-validator/py3/swagger_spec_validator/common.py	(index)
+++ contrib/python/swagger-spec-validator/py3/swagger_spec_validator/common.py	(working tree)
@@ -15 +15 @@ from urllib.request import urlopen
-import importlib_resources
+import importlib.resources
@@ -58,2 +58,2 @@ def read_resource_file(resource_path: str) -> tuple[dict[str, Any], str]:
-    ref = importlib_resources.files("swagger_spec_validator") / resource_path
-    with importlib_resources.as_file(ref) as path:
+    ref = importlib.resources.files("swagger_spec_validator") / resource_path
+    with importlib.resources.as_file(ref) as path:
