--- contrib/python/celery/py3/celery/utils/static/__init__.py	(index)
+++ contrib/python/celery/py3/celery/utils/static/__init__.py	(working tree)
@@ -8,1 +10,1 @@ def get_file(*args):
-    return os.path.join(os.path.abspath(os.path.dirname(__file__)), *args)
+    return os.path.join(os.path.dirname(__file__), *args)
--- contrib/python/celery/py3/celery/utils/term.py	(index)
+++ contrib/python/celery/py3/celery/utils/term.py	(working tree)
@@ -169,3 +169,3 @@ def _read_as_base64(path):
-    with open(path, mode='rb') as fh:
-        encoded = base64.b64encode(fh.read())
-        return encoded.decode('ascii')
+    import __res
+    encoded = base64.b64encode(__res.resfs_read(path))
+    return encoded.decode('ascii')
