--- ./src/pendulum/locales/locale.py	(index)
+++ ./src/pendulum/locales/locale.py	(working tree)
@@ -2,8 +2,7 @@ from __future__ import annotations

 import re

-from importlib import import_module, resources
-from pathlib import Path
+from importlib import import_module, util
 from typing import Any
 from typing import ClassVar
 from typing import Dict
@@ -33,14 +32,12 @@ class Locale:

         # Checking locale existence
         actual_locale = locale
-        locale_path = cast(Path, resources.files(__package__).joinpath(actual_locale))
-        while not locale_path.exists():
-            if actual_locale == locale:
-                raise ValueError(f"Locale [{locale}] does not exist.")
+        locale_module = f"pendulum.locales.{actual_locale}.locale"

-            actual_locale = actual_locale.split("_")[0]
+        if util.find_spec(locale_module) is None:
+            raise ValueError(f"Locale [{locale}] does not exist.")

-        m = import_module(f"pendulum.locales.{actual_locale}.locale")
+        m = import_module(locale_module)

         cls._cache[locale] = cls(locale, m.locale)
