--- contrib/python/ConfigArgParse/py3/tests/test_configargparse.py	(index)
+++ contrib/python/ConfigArgParse/py3/tests/test_configargparse.py	(working tree)
@@ -263,1 +263,1 @@ class TestBasicUseCases(TestCase):
-                'usage: .* \\[-h\\] --genome GENOME \\[-v\\] -g MY_CFG_FILE\n?'
+                'usage: .* \\[-h\\] --genome GENOME \\[-v\\]\n?\s*-g\n?\s+MY_CFG_FILE\n?'
@@ -277,1 +277,1 @@ class TestBasicUseCases(TestCase):
-                'usage: .* \\[-h\\] --genome GENOME \\[-v\\] -g MY_CFG_FILE\n?'
+                'usage: .* \\[-h\\] --genome GENOME \\[-v\\]\n?\s*-g\n?\s+MY_CFG_FILE\n?'
@@ -379,2 +379,2 @@ class TestBasicUseCases(TestCase):
-            r'usage: .* \[-h\] --genome GENOME \[-v\]\s+\(-f1 TYPE1_CFG_FILE \|'
-            r'\s+-f2 TYPE2_CFG_FILE\)\s+\(-f FRMT \| -b\)\n\n'
+            r'usage: .* \[-h\] --genome GENOME \[-v\]\n?\s*\(-f1 TYPE1_CFG_FILE \|'
+            r'\n?\s*-f2 TYPE2_CFG_FILE\)\s+\(-f FRMT \|\n?\s*-b\)\n\n'
@@ -872,1 +872,1 @@ class TestMisc(TestCase):
-            r'usage: .* \[-h\] -c CONFIG_FILE --genome GENOME\n\n'
+            r'usage: .* \[-h\] -c CONFIG_FILE\n?\s*--genome\n?\s+GENOME\n\n'
