--- contrib/python/supervisor/supervisor/options.py	(index)
+++ contrib/python/supervisor/supervisor/options.py	(working tree)
@@ -56,10 +56,8 @@ from supervisor import xmlrpc
 from supervisor import poller
 
 def _read_version_txt():
-    mydir = os.path.abspath(os.path.dirname(__file__))
-    version_txt = os.path.join(mydir, 'version.txt')
-    with open(version_txt, 'r') as f:
-        return f.read().strip()
+    import pkgutil
+    return pkgutil.get_data(__package__, 'version.txt').decode().strip()
 VERSION = _read_version_txt()
 
 def normalize_path(v):
