--- contrib/python/supervisor/supervisor/compat.py	(index)
+++ contrib/python/supervisor/supervisor/compat.py	(working tree)
@@ -85,6 +85,9 @@ try: # pragma: no cover
 except ImportError: # pragma: no cover
     syslog = None
 
+import logging
+import logging.handlers
+
 try: # pragma: no cover
     import ConfigParser
 except ImportError: # pragma: no cover
--- contrib/python/supervisor/supervisor/dispatchers.py	(index)
+++ contrib/python/supervisor/supervisor/dispatchers.py	(working tree)
@@ -136,9 +136,13 @@ class POutputDispatcher(PDispatcher):
             )
 
         if to_syslog:
+            ident = getattr(config, 'syslog_ident', None)
+            format = config.name + ' %(message)s' if ident is None else '%(message)s'
+
             loggers.handle_syslog(
                 self.normallog,
-                fmt=config.name + ' %(message)s'
+                fmt=format,
+                ident=ident
             )
 
     def _init_capturelog(self):
--- contrib/python/supervisor/supervisor/loggers.py	(index)
+++ contrib/python/supervisor/supervisor/loggers.py	(working tree)
@@ -13,6 +13,7 @@ import sys
 import time
 import traceback
 
+from supervisor.compat import logging
 from supervisor.compat import syslog
 from supervisor.compat import long
 from supervisor.compat import is_text_stream
@@ -379,6 +380,52 @@ class SyslogHandler(Handler):
         except:
             self.handleError()
 
+class SyslogIdentedHandler(Handler):
+    def __init__(self, ident):
+        self.ident = ident
+
+        Handler.__init__(self)
+        self.handler = self._make_handler()
+
+    def _make_handler(self):
+        handler = logging.handlers.SysLogHandler(address='/dev/log')
+        handler.setFormatter(logging.Formatter(fr'{self.ident}: %(message)s'))
+        return handler
+
+    def _syslog(self, message, level):
+        self.handler.emit(logging.LogRecord(
+            msg = message,
+            level = level,
+            name = None,
+            pathname = None,
+            lineno = None,
+            args = (),
+            exc_info = None,
+            func = None
+        ))
+
+    def close(self):
+        self.handler.close()
+
+    def reopen(self):
+        self.close()
+        self.handler = self._make_handler()
+
+    def emit(self, record):
+        try:
+            params = record.asdict()
+            message = params['message']
+            for line in message.rstrip('\n').split('\n'):
+                params['message'] = line
+                msg = self.fmt % params
+                try:
+                    self._syslog(msg, record.level)
+                except UnicodeError:
+                    self._syslog(msg.encode("UTF-8"),  record.level)
+        except:
+            self.handleError()
+
+
 def getLogger(level=None):
     return Logger(level)
 
@@ -400,9 +447,9 @@ def handle_stdout(logger, fmt):
     handler.setLevel(logger.level)
     logger.addHandler(handler)
 
-def handle_syslog(logger, fmt):
+def handle_syslog(logger, fmt, ident=None):
     """Attach a new Syslog handler to an existing Logger"""
-    handler = SyslogHandler()
+    handler = SyslogHandler() if ident is None else SyslogIdentedHandler(ident)
     handler.setFormat(fmt)
     handler.setLevel(logger.level)
     logger.addHandler(handler)
--- contrib/python/supervisor/supervisor/options.py	(index)
+++ contrib/python/supervisor/supervisor/options.py	(working tree)
@@ -929,6 +929,8 @@ class ServerOptions(Options):
         if serverurl and serverurl.strip().upper() == 'AUTO':
             serverurl = None
 
+        syslog_ident = get(section, 'syslog_ident', None)
+
         # find uid from "user" option
         user = get(section, 'user', None)
         if user is None:
@@ -1046,6 +1048,7 @@ class ServerOptions(Options):
                 stderr_logfile_backups=logfiles['stderr_logfile_backups'],
                 stderr_logfile_maxbytes=logfiles['stderr_logfile_maxbytes'],
                 stderr_syslog=logfiles['stderr_syslog'],
+                syslog_ident=syslog_ident,
                 stopsignal=stopsignal,
                 stopwaitsecs=stopwaitsecs,
                 stopasgroup=stopasgroup,
@@ -1866,7 +1869,7 @@ class ProcessConfig(Config):
         'stderr_logfile_backups', 'stderr_logfile_maxbytes',
         'stderr_events_enabled', 'stderr_syslog',
         'stopsignal', 'stopwaitsecs', 'stopasgroup', 'killasgroup',
-        'exitcodes', 'redirect_stderr' ]
+        'exitcodes', 'redirect_stderr', 'syslog_ident' ]
     optional_param_names = [ 'environment', 'serverurl' ]
 
     def __init__(self, options, **params):
--- contrib/python/supervisor/supervisor/rpcinterface.py	(index)
+++ contrib/python/supervisor/supervisor/rpcinterface.py	(working tree)
@@ -596,6 +596,7 @@ class SupervisorNamespaceRPCInterface:
                      'stderr_logfile_maxbytes': pconfig.stderr_logfile_maxbytes,
                      'stderr_syslog': pconfig.stderr_syslog,
                      'serverurl': pconfig.serverurl,
+                     'syslog_ident': pconfig.syslog_ident,
                     }
                 # no support for these types in xml-rpc
                 d.update((k, 'auto') for k, v in d.items() if v is Automatic)
--- contrib/python/supervisor/supervisor/skel/sample.conf	(index)
+++ contrib/python/supervisor/supervisor/skel/sample.conf	(working tree)
@@ -115,6 +115,7 @@ serverurl=unix:///tmp/supervisor.sock ; use a unix:// URL  for a unix socket
 ;stderr_syslog=false           ; send stderr to syslog with process name (default false)
 ;environment=A="1",B="2"       ; process environment additions (def no adds)
 ;serverurl=AUTO                ; override serverurl computation (childutils)
+;syslog_ident=supervisord      ; ident passed to syslog messages, default is 'supervisord'
 
 ; The sample eventlistener section below shows all possible eventlistener
 ; subsection values.  Create one or more 'real' eventlistener: sections to be
