--- contrib/python/GDAL/py3/extensions/gdal_wrap.cpp	(index)
+++ contrib/python/GDAL/py3/extensions/gdal_wrap.cpp	(working tree)
@@ -3162,1 +3162,1 @@ SWIGINTERNINLINE PyObject*
-void NeverCallMePlease() {
+static void NeverCallMePlease() {
@@ -7399,1 +7399,1 @@ GDALDriverShadow *IdentifyDriverEx( const char* utf8_path,
-  char **GeneralCmdLineProcessor( char **papszArgv, int nOptions = 0 ) {
+static char **GeneralCmdLineProcessor( char **papszArgv, int nOptions = 0 ) {
--- contrib/python/GDAL/py3/extensions/gnm_wrap.cpp	(index)
+++ contrib/python/GDAL/py3/extensions/gnm_wrap.cpp	(working tree)
@@ -2957,1 +2957,1 @@ SWIGINTERNINLINE PyObject*
-void NeverCallMePlease() {
+static void NeverCallMePlease() {
--- contrib/python/GDAL/py3/extensions/ogr_wrap.cpp	(index)
+++ contrib/python/GDAL/py3/extensions/ogr_wrap.cpp	(working tree)
@@ -3023,1 +3023,1 @@ void DontUseExceptions() {
-void NeverCallMePlease() {
+static void NeverCallMePlease() {
@@ -5645,1 +5645,1 @@ OGRDriverShadow* GetDriver(int driver_number) {
-  char **GeneralCmdLineProcessor( char **papszArgv, int nOptions = 0 ) {
+static char **GeneralCmdLineProcessor( char **papszArgv, int nOptions = 0 ) {
--- contrib/python/GDAL/py3/extensions/osr_wrap.cpp	(index)
+++ contrib/python/GDAL/py3/extensions/osr_wrap.cpp	(working tree)
@@ -2986,1 +2986,1 @@ void DontUseExceptions() {
-void NeverCallMePlease() {
+static void NeverCallMePlease() {
