diff --git a/libde265/encoder/encoder-types.cc b/libde265/encoder/encoder-types.cc
index f8bbd2c..17dcc98 100644
--- a/libde265/encoder/encoder-types.cc
+++ b/libde265/encoder/encoder-types.cc
@@ -432,7 +432,7 @@ void enc_cb::reconstruct(encoder_context* ectx, de265_image* img) const
 void enc_cb::debug_dumpTree(int flags, int indent) const
 {
   std::string indentStr;
-  indentStr.insert(0,indent,' ');
+  indentStr.insert(indentStr.begin(),indent,' ');
 
   std::cout << indentStr << "CB " << x << ";" << y << " "
             << (1<<log2Size) << "x" << (1<<log2Size) << " [" << this << "]\n";
@@ -461,7 +461,7 @@ void enc_cb::debug_dumpTree(int flags, int indent) const
 void enc_tb::debug_dumpTree(int flags, int indent) const
 {
   std::string indentStr;
-  indentStr.insert(0,indent,' ');
+  indentStr.insert(indentStr.begin(),indent,' ');
 
   std::cout << indentStr << "TB " << x << ";" << y << " "
             << (1<<log2Size) << "x" << (1<<log2Size) << " [" << this << "]\n";
