--- contrib/libs/libde265/libde265/image.cc	(index)
+++ contrib/libs/libde265/libde265/image.cc	(working tree)
@@ -59,7 +59,7 @@ static inline void *ALLOC_ALIGNED(size_t alignment, size_t size) {
 };
 #define FREE_ALIGNED(mem)                      free((mem))
 #else
-#define ALLOC_ALIGNED(alignment, size)      memalign((alignment), (size))
+#define ALLOC_ALIGNED(alignment, size)      aligned_alloc((alignment), (size))
 #define FREE_ALIGNED(mem)                   free((mem))
 #endif
 
