commit 10f6fc3184240c70670f61755fcbe7e964ae39c3 (HEAD -> users/ilikepugs/fix-msvc-c20-contrib-libs-libde265)
author: ilikepugs
date: 2021-03-05T09:53:29+03:00

    Allow 'for each for CLR

--- a/libde265/util.h
+++ b/libde265/util.h
@@ -65,7 +65,8 @@
 #define ALIGNED_4( var )  LIBDE265_DECLARE_ALIGNED( var, 4 )
 
 // C++11 specific features
-#if defined(_MSC_VER) && !defined(__clang__) || (!__clang__ && __GNUC__ && GCC_VERSION < 40600)
+#if defined(_MSC_VER) && (defined(__cplusplus_winrt) || defined(__CRL_VER)) \
+    && !defined(__clang__) || (!__clang__ && __GNUC__ && GCC_VERSION < 40600)
 #define FOR_LOOP(type, var, list)   for each (type var in list)
 #undef FOR_LOOP_AUTO_SUPPORT
 #else
