diff --git a/contrib/libs/libunwind/src/Unwind-EHABI.cpp b/contrib/libs/libunwind/src/Unwind-EHABI.cpp
--- a/contrib/libs/libunwind/src/Unwind-EHABI.cpp
+++ b/contrib/libs/libunwind/src/Unwind-EHABI.cpp
@@ -261,7 +261,7 @@
                       size_t offset, size_t len) {
   bool wrotePC = false;
   bool finish = false;
-  bool hasReturnAddrAuthCode = false;
+  bool hasReturnAddrAuthCode [[maybe_unused]] = false;
   while (offset < len && !finish) {
     uint8_t byte = getByte(data, offset++);
     if ((byte & 0x80) == 0) {
diff --git a/contrib/libs/libunwind/src/UnwindLevel1.c b/contrib/libs/libunwind/src/UnwindLevel1.c
--- a/contrib/libs/libunwind/src/UnwindLevel1.c
+++ b/contrib/libs/libunwind/src/UnwindLevel1.c
@@ -195,7 +195,7 @@
 
   // uc is initialized by __unw_getcontext in the parent frame. The first stack
   // frame walked is unwind_phase2.
-  unsigned framesWalked = 1;
+  unsigned framesWalked __attribute__((unused)) = 1;
 #ifdef _LIBUNWIND_USE_CET
   unsigned long shadowStackTop = _get_ssp();
 #endif
@@ -330,7 +330,7 @@
 
   // uc is initialized by __unw_getcontext in the parent frame. The first stack
   // frame walked is unwind_phase2_forced.
-  unsigned framesWalked = 1;
+  unsigned framesWalked __attribute__((unused)) = 1;
   // Walk each frame until we reach where search phase said to stop
   while (__unw_step_stage2(cursor) > 0) {

diff --git a/contrib/libs/libunwind/src/Unwind-wasm.c b/contrib/libs/libunwind/src/Unwind-wasm.c
--- a/contrib/libs/libunwind/src/Unwind-wasm.c
+++ b/contrib/libs/libunwind/src/Unwind-wasm.c
@@ -102,8 +102,8 @@
 }
 
 /// Not used in Wasm.
-_LIBUNWIND_EXPORT void _Unwind_SetIP(struct _Unwind_Context *context,
-                                     uintptr_t value) {}
+_LIBUNWIND_EXPORT void _Unwind_SetIP(struct _Unwind_Context *context __attribute__((unused)),
+                                     uintptr_t value __attribute__((unused))) {}
 
 /// Called by personality handler to get LSDA for current frame.
 _LIBUNWIND_EXPORT uintptr_t
@@ -116,7 +116,7 @@
 
 /// Not used in Wasm.
 _LIBUNWIND_EXPORT uintptr_t
-_Unwind_GetRegionStart(struct _Unwind_Context *context) {
+_Unwind_GetRegionStart(struct _Unwind_Context *context __attribute__((unused))) {
   return 0;
 }
