commit 080afaf1ef8e81cfdc26850560b639acd0c88733
author: robot-contrib
date: 2024-12-28T08:39:56+03:00

    Update contrib/libs/libunwind to 19.1.5

--- contrib/libs/libunwind/include/__libunwind_config.h	(cc7e7e3cd08b4064e3856fcec6df5aa39b8cc0a4)
+++ contrib/libs/libunwind/include/__libunwind_config.h	(080afaf1ef8e81cfdc26850560b639acd0c88733)
@@ -180,6 +180,10 @@
 #endif
 #define _LIBUNWIND_HIGHEST_DWARF_REGISTER                                      \
   _LIBUNWIND_HIGHEST_DWARF_REGISTER_LOONGARCH
+#elif defined(__wasm__)
+// Unused
+#define _LIBUNWIND_CONTEXT_SIZE 0
+#define _LIBUNWIND_CURSOR_SIZE 0
 # else
 #  error "Unsupported architecture."
 # endif
