--- contrib/python/websocket-client/websocket/tests/test_http.py	(index)
+++ contrib/python/websocket-client/websocket/tests/test_http.py	(working tree)
@@ -74,7 +74,8 @@ class HeaderSockMock(SockMock):
 
     def __init__(self, fname):
         SockMock.__init__(self)
-        path = os.path.join(os.path.dirname(__file__), fname)
+        import yatest.common as yc
+        path = os.path.join(os.path.dirname(yc.source_path(__file__)), fname)
         with open(path, "rb") as f:
             self.add_packet(f.read())
 
--- contrib/python/websocket-client/websocket/tests/test_websocket.py	(index)
+++ contrib/python/websocket-client/websocket/tests/test_websocket.py	(working tree)
@@ -86,7 +86,8 @@ class HeaderSockMock(SockMock):
 
     def __init__(self, fname):
         SockMock.__init__(self)
-        path = os.path.join(os.path.dirname(__file__), fname)
+        import yatest.common as yc
+        path = os.path.join(os.path.dirname(yc.source_path(__file__)), fname)
         with open(path, "rb") as f:
             self.add_packet(f.read())
 
