From fbea9fc86eba9a35958b0154b1b08dbc4744fa09 Mon Sep 17 00:00:00 2001
From: Sam Clegg <sbc@chromium.org>
Date: Fri, 27 Jan 2023 09:02:49 -0800
Subject: [PATCH] Patches from emscripten  3.1.31

diff --git a/src/cxa_thread_atexit.cpp b/src/cxa_thread_atexit.cpp
--- a/src/cxa_thread_atexit.cpp
+++ b/src/cxa_thread_atexit.cpp
@@ -113,8 +113,8 @@ extern "C" {
     return __cxa_thread_atexit_impl(dtor, obj, dso_symbol);
 #else
 #ifndef __EMSCRIPTEN__
-    // Emscripten doesn't fully support weak undefined symbols yet
-    // https://github.com/emscripten-core/emscripten/issues/12819
+    // Emscripten doesn't implement __cxa_thread_atexit_impl, so we can simply
+    // avoid this check.
     if (__cxa_thread_atexit_impl) {
       return __cxa_thread_atexit_impl(dtor, obj, dso_symbol);
     } else
