--- contrib/restricted/boost/math/include/boost/math/tools/config.hpp	(index)
+++ contrib/restricted/boost/math/include/boost/math/tools/config.hpp	(working tree)
@@ -224,7 +224,7 @@
 // Generic catch all case for gcc's "double-double" long double type.
 // We do not support this as it's not even remotely IEEE conforming:
 //
-#  define BOOST_MATH_NO_LONG_DOUBLE_MATH_FUNCTIONS
+//#  define BOOST_MATH_NO_LONG_DOUBLE_MATH_FUNCTIONS
 #endif
 #if defined(unix) && defined(__INTEL_COMPILER) && (__INTEL_COMPILER <= 1000) && !defined(BOOST_MATH_NO_LONG_DOUBLE_MATH_FUNCTIONS)
 //
