--- contrib/python/Faker/py3/tests/providers/test_python.py     (index)
+++ contrib/python/Faker/py3/tests/providers/test_python.py     (working tree)
@@ -170,6 +170,7 @@ class TestPython(unittest.TestCase):
         assert some_list
         assert isinstance(some_list, list)

+    @unittest.skip("wait pydebug enable")
     def test_pylist_types(self):
         with warnings.catch_warnings(record=True) as w:
             some_list = self.factory.pylist(10, True, [int])
--- contrib/python/Faker/py2/tests/utils/test_utils.py	(index)
+++ contrib/python/Faker/py2/tests/utils/test_utils.py	(working tree)
@@ -11,1 +11,2 @@ from faker.utils.datasets import add_dicts
-TEST_DIR = Path(__file__).resolve().parent
+import yatest.common as yc
+TEST_DIR = Path(yc.source_path(__file__)).resolve().parent
