--- contrib/python/djangorestframework-csv/rest_framework_csv/tests.py	(index)
+++ contrib/python/djangorestframework-csv/rest_framework_csv/tests.py	(working tree)
@@ -10,2 +10,2 @@ from types import GeneratorType
-from .renderers import CSVRenderer, CSVStreamingRenderer, PaginatedCSVRenderer
-from .parsers import CSVParser
+from rest_framework_csv.renderers import CSVRenderer, CSVStreamingRenderer, PaginatedCSVRenderer
+from rest_framework_csv.parsers import CSVParser
@@ -244,1 +244,2 @@ class TestCSVParser(TestCase):
-        CURDIR = os.path.dirname(__file__)
+        import yatest.common
+        CURDIR = yatest.common.source_path('contrib/python/djangorestframework-csv/rest_framework_csv')
