--- contrib/python/django-simple-history/simple_history/admin.py	(index)
+++ contrib/python/django-simple-history/simple_history/admin.py	(working tree)
@@ -7,1 +7,5 @@ from django.contrib.admin.utils import unquote
-from django.urls import re_path, reverse
+from django.urls import reverse
+try:
+    from django.urls import re_path as url
+except ImportError:
+    from django.conf.urls import url
@@ -29,7 +33,7 @@ class SimpleHistoryAdmin(admin.ModelAdmin):
         opts = self.model._meta
         info = opts.app_label, opts.model_name
         history_urls = [
-            re_path(
+            url(
                 "^([^/]+)/history/([^/]+)/$",
                 admin_site.admin_view(self.history_form_view),
                 name="%s_%s_simple_history" % info,
