--- contrib/libs/fmt/include/fmt/args.h	(b92e22f2baa7a4b746902dd4641a3025ca1f424f)
+++ contrib/libs/fmt/include/fmt/args.h	(working tree)
@@ -51,6 +51,7 @@ class dynamic_arg_list {
 
  public:
   template <typename T, typename Arg> const T& push(const Arg& arg) {
+    // NOLINTNEXTLINE(clang-analyzer-cplusplus.NewDeleteLeaks)
     auto new_node = std::unique_ptr<typed_node<T>>(new typed_node<T>(arg));
     auto& value = new_node->value;
     new_node->next = std::move(head_);
