--- contrib/python/ijson/py3/tests/test_misc.py	(index)
+++ contrib/python/ijson/py3/tests/test_misc.py	(working tree)
@@ -1,1 +1,1 @@
-from tests.test_base import JSON, JSON_EVENTS, JSON_PARSE_EVENTS, JSON_OBJECT,\
+from .test_base import JSON, JSON_EVENTS, JSON_PARSE_EVENTS, JSON_OBJECT,\
--- contrib/python/ijson/py3/tests/test_benchmark.py	(index)
+++ contrib/python/ijson/py3/tests/test_benchmark.py	(working tree)
@@ -8 +8 @@ import pytest
-from tests.test_base import JSON
+from .test_base import JSON
--- contrib/python/ijson/py3/tests/test_dump.py	(index)
+++ contrib/python/ijson/py3/tests/test_dump.py	(working tree)
@@ -7 +7 @@ import pytest
-from tests.test_base import JSON
+from .test_base import JSON
