--- contrib/python/pyroute2/py2/pyroute2/netlink/rtnl/ifinfmsg/__init__.py	(index)
+++ contrib/python/pyroute2/py2/pyroute2/netlink/rtnl/ifinfmsg/__init__.py	(working tree)
@@ -340,7 +340,7 @@ class iptnl_data(nla):
                ('IFLA_IPIP_TOS', 'uint8'),
                ('IFLA_IPIP_ENCAP_LIMIT', 'uint8'),
                ('IFLA_IPIP_FLOWINFO', 'be32'),
-               ('IFLA_IPIP_FLAGS', 'be16'),
+               ('IFLA_IPIP_FLAGS', 'uint32'),
                ('IFLA_IPIP_PROTO', 'uint8'),
                ('IFLA_IPIP_PMTUDISC', 'uint8'),
                ('IFLA_IPIP_6RD_PREFIX', 'ip6addr'),
@@ -749,7 +749,7 @@ class ifinfbase(object):
                        ('IFLA_IP6TNL_TOS', 'uint8'),
                        ('IFLA_IP6TNL_ENCAP_LIMIT', 'uint8'),
                        ('IFLA_IP6TNL_FLOWINFO', 'be32'),
-                       ('IFLA_IP6TNL_FLAGS', 'be16'),
+                       ('IFLA_IP6TNL_FLAGS', 'uint32'),
                        ('IFLA_IP6TNL_PROTO', 'uint8'),
                        ('IFLA_IP6TNL_PMTUDISC', 'uint8'),
                        ('IFLA_IP6TNL_6RD_PREFIX', 'ip6addr'),
