--- contrib/python/humbledb/humbledb/maps.py	(index)
+++ contrib/python/humbledb/humbledb/maps.py	(working tree)
@@ -8,7 +8,9 @@ class NameMap(str):
     """This class is used to map attribute names to document keys internally."""
 
     def __new__(cls, value=""):
-        return super().__new__(cls, value)
+        instance = super().__new__(cls, value)
+        instance.__dict__ = {}
+        return instance
 
     def __init__(self, value=""):
         self._key = value.split(".")[-1]
