--- contrib/python/drf-ujson/drf_ujson/parsers.py	(index)
+++ contrib/python/drf-ujson/drf_ujson/parsers.py	(working tree)
@@ -2,7 +2,7 @@
 from __future__ import unicode_literals
 
 from django.conf import settings
-from rest_framework.compat import six
+import six
 from rest_framework.parsers import BaseParser, ParseError
 from rest_framework.renderers import JSONRenderer
 import ujson
--- contrib/python/drf-ujson/drf_ujson/renderers.py	(index)
+++ contrib/python/drf-ujson/drf_ujson/renderers.py	(working tree)
@@ -1,5 +1,5 @@
 from __future__ import unicode_literals
-from rest_framework.compat import six
+import six
 from rest_framework.renderers import BaseRenderer
 import ujson
 
