--- contrib/python/scikit-image/skimage/restoration/unwrap_2d_ljmu.c	(index)
+++ contrib/python/scikit-image/skimage/restoration/unwrap_2d_ljmu.c	(working tree)
@@ -100 +100 @@ typedef enum {
-yes_no find_pivot(EDGE *left, EDGE *right, double *pivot_ptr) {
+static yes_no find_pivot(EDGE *left, EDGE *right, double *pivot_ptr) {
@@ -127 +127 @@ yes_no find_pivot(EDGE *left, EDGE *right, double *pivot_ptr) {
-EDGE *partition(EDGE *left, EDGE *right, double pivot) {
+static EDGE *partition(EDGE *left, EDGE *right, double pivot) {
@@ -140 +140 @@ EDGE *partition(EDGE *left, EDGE *right, double pivot) {
-void quicker_sort(EDGE *left, EDGE *right) {
+static void quicker_sort(EDGE *left, EDGE *right) {
@@ -193 +193 @@ void initialisePIXELs(double *wrapped_image, unsigned char *input_mask,
-double wrap(double pixel_value) {
+static double wrap(double pixel_value) {
@@ -205 +205 @@ double wrap(double pixel_value) {
-int find_wrap(double pixelL_value, double pixelR_value) {
+static int find_wrap(double pixelL_value, double pixelR_value) {
@@ -220 +220 @@ int find_wrap(double pixelL_value, double pixelR_value) {
-void extend_mask(unsigned char *input_mask, unsigned char *extended_mask,
+static void extend_mask(unsigned char *input_mask, unsigned char *extended_mask,
@@ -321 +321 @@ void extend_mask(unsigned char *input_mask, unsigned char *extended_mask,
-void calculate_reliability(double *wrappedImage, PIXELM *pixel, int image_width,
+static void calculate_reliability(double *wrappedImage, PIXELM *pixel, int image_width,
@@ -436 +436 @@ void calculate_reliability(double *wrappedImage, PIXELM *pixel, int image_width,
-void horizontalEDGEs(PIXELM *pixel, EDGE *edge, int image_width,
+static void horizontalEDGEs(PIXELM *pixel, EDGE *edge, int image_width,
@@ -484 +484 @@ void horizontalEDGEs(PIXELM *pixel, EDGE *edge, int image_width,
-void verticalEDGEs(PIXELM *pixel, EDGE *edge, int image_width, int image_height,
+static void verticalEDGEs(PIXELM *pixel, EDGE *edge, int image_width, int image_height,
--- contrib/python/scikit-image/skimage/restoration/unwrap_3d_ljmu.c	(index)
+++ contrib/python/scikit-image/skimage/restoration/unwrap_3d_ljmu.c	(working tree)
@@ -104 +104 @@ typedef enum {
-yes_no find_pivot(EDGE *left, EDGE *right, double *pivot_ptr) {
+static yes_no find_pivot(EDGE *left, EDGE *right, double *pivot_ptr) {
@@ -131 +131 @@ yes_no find_pivot(EDGE *left, EDGE *right, double *pivot_ptr) {
-EDGE *partition(EDGE *left, EDGE *right, double pivot) {
+static EDGE *partition(EDGE *left, EDGE *right, double pivot) {
@@ -144 +144 @@ EDGE *partition(EDGE *left, EDGE *right, double pivot) {
-void quicker_sort(EDGE *left, EDGE *right) {
+static void quicker_sort(EDGE *left, EDGE *right) {
@@ -200 +200 @@ void initialiseVOXELs(double *WrappedVolume, unsigned char *input_mask,
-double wrap(double voxel_value) {
+static double wrap(double voxel_value) {
@@ -212 +212 @@ double wrap(double voxel_value) {
-int find_wrap(double voxelL_value, double voxelR_value) {
+static int find_wrap(double voxelL_value, double voxelR_value) {
@@ -227 +227 @@ int find_wrap(double voxelL_value, double voxelR_value) {
-void extend_mask(unsigned char *input_mask, unsigned char *extended_mask,
+static void extend_mask(unsigned char *input_mask, unsigned char *extended_mask,
@@ -461 +461 @@ void extend_mask(unsigned char *input_mask, unsigned char *extended_mask,
-void calculate_reliability(double *wrappedVolume, VOXELM *voxel,
+static void calculate_reliability(double *wrappedVolume, VOXELM *voxel,
@@ -791 +791 @@ void calculate_reliability(double *wrappedVolume, VOXELM *voxel,
-void horizontalEDGEs(VOXELM *voxel, EDGE *edge, int volume_width,
+static void horizontalEDGEs(VOXELM *voxel, EDGE *edge, int volume_width,
@@ -840 +840 @@ void horizontalEDGEs(VOXELM *voxel, EDGE *edge, int volume_width,
-void verticalEDGEs(VOXELM *voxel, EDGE *edge, int volume_width,
+static void verticalEDGEs(VOXELM *voxel, EDGE *edge, int volume_width,
