--- contrib/python/scikit-image/ya.make	(index)
+++ contrib/python/scikit-image/ya.make	(working tree)
@@ -409,6 +409,18 @@ RESOURCE_FILES(
     skimage/scripts/skivi
 )
 
+RUN_PROGRAM(
+    contrib/python/pythran/bin/pythran -E skimage/feature/_hessian_det_appx_pythran.py -o _hessian_det_appx.cpp
+    IN skimage/feature/_hessian_det_appx_pythran.py
+    OUT _hessian_det_appx.cpp
+)
+
+RUN_PROGRAM(
+    contrib/python/pythran/bin/pythran -E skimage/feature/brief_pythran.py -o brief_cy.cpp
+    IN skimage/feature/brief_pythran.py
+    OUT brief_cy.cpp
+)
+
 END()
 
 RECURSE_FOR_TESTS(
