--- contrib/python/mf2py/mf2py/backcompat.py	(index)
+++ contrib/python/mf2py/mf2py/backcompat.py	(working tree)
@@ -23,13 +23,13 @@ child['class'] = child_classes
 
 """
 
-import codecs
 import copy
 import json
 import os
 from urllib.parse import unquote
 
 import bs4
+import library.python.resource
 
 from . import mf2_classes
 from .dom_helpers import get_children
@@ -40,16 +40,12 @@ _CLASSIC_MAP = {}
 
 # populate backcompat rules from JSON files
 
-_RULES_LOC = os.path.join(
-    os.path.dirname(os.path.abspath(__file__)), "backcompat-rules"
-)
+_RULES_LOC = 'contrib/python/mf2py/mf2py/backcompat-rules'
 
-for filename in os.listdir(_RULES_LOC):
-    file_path = os.path.join(_RULES_LOC, filename)
+for filename in library.python.resource.resfs_files(_RULES_LOC):
     root = os.path.splitext(filename)[0]
-    with codecs.open(file_path, "r", "utf-8") as f:
-        rules = json.load(f)
-
+    content = library.python.resource.resfs_read(filename)
+    rules = json.loads(content.decode())
     _CLASSIC_MAP[root] = rules
 
 
