--- contrib/python/elementpath/elementpath/validators/__init__.py	(index)
+++ contrib/python/elementpath/elementpath/validators/__init__.py	(working tree)
@@ -12,6 +12,7 @@ Subpackage for validating against XPath standard schemas.
 """
 import pathlib
 from xml.etree.ElementTree import Element
+from importlib.resources import files
 from typing import Optional
 
 from elementpath.exceptions import ElementPathRuntimeError
@@ -34,12 +35,13 @@ else:
     json_to_xml_schema: Optional[xmlschema.XMLSchemaBase] = None
 
     __all__ = ['validate_analyzed_string', 'validate_json_to_xml']
+    root_node = files(__package__)
 
     def validate_analyzed_string(root: Element) -> None:
         global analyzed_string_schema
 
         if analyzed_string_schema is None:
-            xsd_file = pathlib.Path(__file__).parent.joinpath('analyze-string.xsd')
+            xsd_file = (root_node / 'analyze-string.xsd').read_text()
             analyzed_string_schema = xmlschema.XMLSchema(xsd_file)
 
         analyzed_string_schema.validate(root)
@@ -48,7 +50,7 @@ else:
         global json_to_xml_schema
 
         if json_to_xml_schema is None:
-            xsd_file = pathlib.Path(__file__).parent.joinpath('schema-for-json.xsd')
+            xsd_file = (root_node / 'schema-for-json.xsd').read_text()
             json_to_xml_schema = xmlschema.XMLSchema(xsd_file)
 
         json_to_xml_schema.validate(root, namespaces={'j': XPATH_FUNCTIONS_NAMESPACE})
