--- contrib/python/pgcli/pgcli/packages/pgliterals/main.py	(index)
+++ contrib/python/pgcli/pgcli/packages/pgliterals/main.py	(working tree)
@@ -1,11 +1,11 @@
 import os
 import json
+import pkgutil
 
 root = os.path.dirname(__file__)
 literal_file = os.path.join(root, "pgliterals.json")
 
-with open(literal_file) as f:
-    literals = json.load(f)
+literals = json.loads(pkgutil.get_data(__package__, "pgliterals.json"))
 
 
 def get_literals(literal_type, type_=tuple):
--- contrib/python/pgcli/pgcli/config.py	(index)
+++ contrib/python/pgcli/pgcli/config.py	(working tree)
@@ -55,16 +55,18 @@ def upgrade_config(config, def_config):
 
 
 def get_config(pgclirc_file=None):
-    from pgcli import __file__ as package_root
-
-    package_root = os.path.dirname(package_root)
+    import pkgutil
+    import tempfile
 
     pgclirc_file = get_config_filename(pgclirc_file)
 
-    default_config = os.path.join(package_root, "pgclirc")
-    write_default_config(default_config, pgclirc_file)
+    with tempfile.NamedTemporaryFile() as pgclirc:
+        pgclirc.write(pkgutil.get_data(__package__, "pgclirc"))
+
+        default_config = pgclirc.name
+        write_default_config(default_config, pgclirc_file)
 
-    return load_config(pgclirc_file, default_config)
+        return load_config(pgclirc_file, default_config)
 
 
 def get_casing_file(config):
