--- contrib/python/jupyter-client/py3/tests/conftest.py	(index)
+++ contrib/python/jupyter-client/py3/tests/conftest.py	(working tree)
@@ -35,2 +35,3 @@ import pytest
-def setup_environ(jp_environ):
-    pass
+def setup_environ(jp_environ, monkeypatch):
+    import yatest.common
+    monkeypatch.setenv("JUPYTER_DATA_DIR", yatest.common.work_path())
--- contrib/python/jupyter-client/py3/tests/test_kernelapp.py   (index)
+++ contrib/python/jupyter-client/py3/tests/test_kernelapp.py   (working tree)
@@ -6,12 +6,16 @@ import time
 from subprocess import Popen
 from tempfile import mkdtemp

+import yatest.common
+
+JUPYTER_KERNEL_BIN_PATH = "contrib/python/jupyter-client/py3/bin/jupyter_kernel/jupyter-kernel"
+
 
 def _launch(extra_env):
     env = os.environ.copy()
     env.update(extra_env)
     return Popen(
-        [sys.executable, "-c", "from jupyter_client.kernelapp import main; main()"],
+        [yatest.common.binary_path(JUPYTER_KERNEL_BIN_PATH)],
         env=env,
         stderr=PIPE,
     )
--- contrib/python/jupyter-client/py3/tests/test_kernelspec.py	(index)
+++ contrib/python/jupyter-client/py3/tests/test_kernelspec.py	(working tree)
@@ -21,1 +21,5 @@ from tempfile import TemporaryDirectory
 from jupyter_client import kernelspec
+
+import yatest.common
+
+JUPYTER_KERNELSPEC_BIN_PATH = "contrib/python/jupyter-client/py3/bin/jupyter_kernelspec/jupyter-kernelspec"
@@ -136,9 +140,7 @@ class KernelSpecTests(unittest.TestCase):
     def test_remove_kernel_spec_app(self):
         p = Popen(
             [
-                sys.executable,
-                "-m",
-                "jupyter_client.kernelspecapp",
+                yatest.common.binary_path(JUPYTER_KERNELSPEC_BIN_PATH),
                 "remove",
                 "sample",
                 "-f",
--- contrib/python/jupyter-client/py3/tests/test_kernelmanager.py	(index)
+++ contrib/python/jupyter-client/py3/tests/test_kernelmanager.py	(working tree)
@@ -20 +20 @@ from jupyter_client.manager import _ShutdownStatus, start_new_async_kernel, star
-from .utils import AsyncKMSubclass, SyncKMSubclass
+from tests.utils import AsyncKMSubclass, SyncKMSubclass
--- contrib/python/jupyter-client/py3/tests/test_kernelspec.py	(index)
+++ contrib/python/jupyter-client/py3/tests/test_kernelspec.py	(working tree)
@@ -21 +21 @@ from jupyter_client import kernelspec
-from .utils import install_kernel, sample_kernel_json
+from tests.utils import install_kernel, sample_kernel_json
--- contrib/python/jupyter-client/py3/tests/test_multikernelmanager.py	(index)
+++ contrib/python/jupyter-client/py3/tests/test_multikernelmanager.py	(working tree)
@@ -20 +20 @@ from jupyter_client.multikernelmanager import AsyncMultiKernelManager, MultiKern
-from .utils import (
+from tests.utils import (
--- contrib/python/jupyter-client/py3/tests/test_provisioning.py	(index)
+++ contrib/python/jupyter-client/py3/tests/test_provisioning.py	(working tree)
@@ -198,2 +198,2 @@ initial_provisioner_map = {
-    "subclassed-test-provisioner": "tests.test_provisioning:SubclassedTestProvisioner",
-    "custom-test-provisioner": "tests.test_provisioning:CustomTestProvisioner",
+    "subclassed-test-provisioner": "__tests__.test_provisioning:SubclassedTestProvisioner",
+    "custom-test-provisioner": "__tests__.test_provisioning:CustomTestProvisioner",
@@ -214 +214 @@ def mock_get_provisioner(_: str, name: str) -> EntryPoint:
-            "tests.test_provisioning:NewTestProvisioner",
+            "__tests__.test_provisioning:NewTestProvisioner",
--- contrib/python/jupyter-client/py3/tests/test_kernelspecapp.py	(index)
+++ contrib/python/jupyter-client/py3/tests/test_kernelspecapp.py	(working tree)
@@ -14,4 +15,5 @@ from jupyter_client.kernelspecapp import (
 
 
+@pytest.mark.xfail
 def test_kernelspec_sub_apps(jp_kernel_dir):
     app = InstallKernelSpec()
