--- contrib/python/kazoo/py2/kazoo/client.py	(index)
+++ contrib/python/kazoo/py2/kazoo/client.py	(working tree)
@@ -718,6 +718,15 @@ class KazooClient(object):
         .. versionadded:: 0.5
 
         """
+
+        def read_all(sock):
+            while True:
+                part = sock.recv(8192)
+                if len(part) == 0:
+                    break
+                else:
+                    yield part
+
         if not self._live.is_set():
             raise ConnectionLoss("No connection to server")
 
@@ -732,7 +741,7 @@ class KazooClient(object):
             verify_certs=self.verify_certs,
         )
         sock.sendall(cmd)
-        result = sock.recv(8192)
+        result = ''.join(read_all(sock))
         sock.close()
         return result.decode('utf-8', 'replace')
 
