--- contrib/python/sqlparse/py3/tests/conftest.py	(index)
+++ contrib/python/sqlparse/py3/tests/conftest.py	(working tree)
@@ -5,13 +5,13 @@ import os
 
 import pytest
 
-DIR_PATH = os.path.dirname(__file__)
-FILES_DIR = os.path.join(DIR_PATH, 'files')
-
 
 @pytest.fixture()
 def filepath():
     """Returns full file path for test files."""
+    import yatest.common
+    DIR_PATH = yatest.common.test_source_path()
+    FILES_DIR = os.path.join(DIR_PATH, 'files')
 
     def make_filepath(filename):
         # https://stackoverflow.com/questions/18011902/py-test-pass-a-parameter-to-a-fixture-function
