--- contrib/python/python-hcl2/hcl2/parser.py	(index)
+++ contrib/python/python-hcl2/hcl2/parser.py	(working tree)
@@ -14 +14,2 @@ def parser() -> Lark:
-    return Lark.open(
+    return Lark.open_from_package(
+        __package__,
@@ -18 +18,0 @@ def parser() -> Lark:
-        rel_to=__file__,
@@ -31 +31,2 @@ def reconstruction_parser() -> Lark:
-    return Lark.open(
+    return Lark.open_from_package(
+        __package__,
@@ -39 +39,0 @@ def reconstruction_parser() -> Lark:
-        rel_to=__file__,
