--- contrib/python/selenium/py3/selenium/webdriver/firefox/firefox_profile.py	(index)
+++ contrib/python/selenium/py3/selenium/webdriver/firefox/firefox_profile.py	(working tree)
@@ -17,6 +17,7 @@
 
 from __future__ import with_statement
 
+import pkgutil
 import base64
 import copy
 import json
@@ -62,10 +63,10 @@ class FirefoxProfile(object):
            directory when object is created.
         """
         if not FirefoxProfile.DEFAULT_PREFERENCES:
-            with open(os.path.join(os.path.dirname(__file__),
-                                   WEBDRIVER_PREFERENCES)) as default_prefs:
-                FirefoxProfile.DEFAULT_PREFERENCES = json.load(default_prefs)
-
+            _pkg = '.'.join(__name__.split('.')[:-1])
+            FirefoxProfile.DEFAULT_PREFERENCES = json.loads(
+                pkgutil.get_data(_pkg, 'webdriver_prefs.json').decode('utf8')
+            )
         self.default_preferences = copy.deepcopy(
             FirefoxProfile.DEFAULT_PREFERENCES['mutable'])
         self.native_events_enabled = True
