--- contrib/python/attrdict/py3/attrdict/default.py	(index)
+++ contrib/python/attrdict/py3/attrdict/default.py	(working tree)
@@ -1,7 +1,10 @@
 """
 A subclass of MutableAttr that has defaultdict support.
 """
-from collections import Mapping
+try:
+    from collections.abc import Mapping
+except ImportError:
+    from collections import Mapping
 
 import six
 
--- contrib/python/attrdict/py3/attrdict/mapping.py	(index)
+++ contrib/python/attrdict/py3/attrdict/mapping.py	(working tree)
@@ -1,7 +1,10 @@
 """
 An implementation of MutableAttr.
 """
-from collections import Mapping
+try:
+    from collections.abc import Mapping
+except ImportError:
+    from collections import Mapping
 
 import six
 
--- contrib/python/attrdict/py3/attrdict/merge.py	(index)
+++ contrib/python/attrdict/py3/attrdict/merge.py	(working tree)
@@ -1,7 +1,10 @@
 """
 A right-favoring Mapping merge.
 """
-from collections import Mapping
+try:
+    from collections.abc import Mapping
+except ImportError:
+    from collections import Mapping
 
 
 __all__ = ['merge']
--- contrib/python/attrdict/py3/attrdict/mixins.py	(index)
+++ contrib/python/attrdict/py3/attrdict/mixins.py	(working tree)
@@ -2,7 +2,10 @@
 Mixin Classes for Attr-support.
 """
 from abc import ABCMeta, abstractmethod
-from collections import Mapping, MutableMapping, Sequence
+try:
+    from collections.abc import Mapping, MutableMapping, Sequence
+except ImportError:
+    from collections import Mapping, MutableMapping, Sequence
 import re
 
 import six
