diff -ruN postgresql/src/backend/utils/cache/lsyscache.c postgresql.patched/src/backend/utils/cache/lsyscache.c
--- postgresql/src/backend/utils/cache/lsyscache.c	2022-05-22 11:54:23.953458311 +0300
+++ postgresql.patched/src/backend/utils/cache/lsyscache.c	2022-05-22 11:53:44.393144226 +0300
@@ -2269,7 +2269,7 @@
  *					returned is controlled by IOFuncSelector
  */
 void
-get_type_io_data(Oid typid,
+get_type_io_data_original(Oid typid,
 				 IOFuncSelector which_func,
 				 int16 *typlen,
 				 bool *typbyval,

