diff --git a/include/tuple b/include/tuple
index 1334d04..868b66d 100644
--- a/include/tuple
+++ b/include/tuple
@@ -1608,6 +1608,8 @@ pair<_T1, _T2>::pair(piecewise_construct_t,
     :  first(_VSTD::forward<_Args1>(_VSTD::get<_I1>( __first_args))...),
       second(_VSTD::forward<_Args2>(_VSTD::get<_I2>(__second_args))...)
 {
+    (void)__first_args;
+    (void)__second_args;
 }
 
 #if _LIBCPP_STD_VER > 14
