diff --git a/include/version b/include/version
index 3aed750..25d112a 100644
--- a/include/version
+++ b/include/version
@@ -47,7 +47,7 @@ __cpp_lib_bounded_array_traits                          201902L <type_traits>
 __cpp_lib_boyer_moore_searcher                          201603L <functional>
 __cpp_lib_byte                                          201603L <cstddef>
 __cpp_lib_byteswap                                      202110L <bit>
-__cpp_lib_char8_t                                       201811L <atomic> <filesystem> <istream>
+__cpp_lib_char8_t                                       201907L <atomic> <filesystem> <istream>
                                                                 <limits> <locale> <ostream>
                                                                 <string> <string_view>
 __cpp_lib_chrono                                        201611L <chrono>
@@ -231,7 +231,11 @@ __cpp_lib_void_t                                        201411L <type_traits>
 # define __cpp_lib_allocator_traits_is_always_equal     201411L
 # define __cpp_lib_any                                  201606L
 # define __cpp_lib_apply                                201603L
-# define __cpp_lib_array_constexpr                      201603L
+# if defined(_MSC_VER) && _MSC_VER >= 1926
+#   define __cpp_lib_array_constexpr                    201803L
+# else
+#   define __cpp_lib_array_constexpr                    201603L
+# endif
 # define __cpp_lib_as_const                             201510L
 # define __cpp_lib_atomic_is_always_lock_free           201603L
 # define __cpp_lib_bool_constant                        201505L
@@ -242,7 +246,7 @@ __cpp_lib_void_t                                        201411L <type_traits>
 # define __cpp_lib_enable_shared_from_this              201603L
 // # define __cpp_lib_execution                            201603L
 # if !defined(_LIBCPP_AVAILABILITY_DISABLE_FTM___cpp_lib_filesystem)
-#   define __cpp_lib_filesystem                         201703L
+# define __cpp_lib_filesystem                           201703L
 # endif
 # define __cpp_lib_gcd_lcm                              201606L
 // # define __cpp_lib_hardware_interference_size           201703L
@@ -272,7 +276,11 @@ __cpp_lib_void_t                                        201411L <type_traits>
 # endif
 # define __cpp_lib_shared_ptr_arrays                    201611L
 # define __cpp_lib_shared_ptr_weak_type                 201606L
-# define __cpp_lib_string_view                          201606L
+# if defined(_MSC_VER) && _MSC_VER >= 1926
+#   define __cpp_lib_string_view                        201803L
+# else
+#   define __cpp_lib_string_view                        201606L
+# endif
 // # define __cpp_lib_to_chars                             201611L
 # undef  __cpp_lib_transparent_operators
 # define __cpp_lib_transparent_operators                201510L
@@ -304,7 +312,7 @@ __cpp_lib_void_t                                        201411L <type_traits>
 // # define __cpp_lib_bitops                               201907L
 # define __cpp_lib_bounded_array_traits                 201902L
 # if !defined(_LIBCPP_HAS_NO_CHAR8_T)
-#   define __cpp_lib_char8_t                            201811L
+#   define __cpp_lib_char8_t                            201907L
 # endif
 # define __cpp_lib_concepts                             202002L
 # define __cpp_lib_constexpr_algorithms                 201806L
