--- contrib/python/dbfread/dbfread/test_field_parser.py	(index)
+++ contrib/python/dbfread/dbfread/test_field_parser.py	(working tree)
@@ -1,7 +1,7 @@
 import datetime
 from decimal import Decimal
 from pytest import raises
-from .field_parser import FieldParser
+from dbfread.field_parser import FieldParser
 
 class MockHeader(object):
     dbversion = 0x02
--- contrib/python/dbfread/dbfread/test_ifiles.py	(index)
+++ contrib/python/dbfread/dbfread/test_ifiles.py	(working tree)
@@ -1,5 +1,5 @@
-from .ifiles import *
-from .ifiles import ipat
+from dbfread.ifiles import *
+from dbfread.ifiles import ipat
 
 assert ipat('mixed') == '[Mm][Ii][Xx][Ee][Dd]'
 assert ifnmatch('test', 'test') == True
--- contrib/python/dbfread/dbfread/test_invalid_value.py	(index)
+++ contrib/python/dbfread/dbfread/test_invalid_value.py	(working tree)
@@ -1,4 +1,4 @@
-from .field_parser import InvalidValue
+from dbfread.field_parser import InvalidValue
 
 def test_repr():
     assert repr(InvalidValue(b'')) == "InvalidValue(b'')"
--- contrib/python/dbfread/dbfread/test_memo.py	(index)
+++ contrib/python/dbfread/dbfread/test_memo.py	(working tree)
@@ -1,13 +1,14 @@
 from pytest import raises
-from .dbf import DBF
-from .exceptions import MissingMemoFile
+from dbfread.dbf import DBF
+from dbfread.exceptions import MissingMemoFile
 
 def test_missing_memofile():
+    import yatest.common as yc
     with raises(MissingMemoFile):
-        DBF('testcases/no_memofile.dbf')
+        DBF(yc.source_path('contrib/python/dbfread/testcases/no_memofile.dbf'))
 
     # This should succeed.
-    table = DBF('testcases/no_memofile.dbf', ignore_missing_memofile=True)
+    table = DBF(yc.source_path('contrib/python/dbfread/testcases/no_memofile.dbf'), ignore_missing_memofile=True)
 
     # Memo fields should be returned as None.
     record = next(iter(table))
--- contrib/python/dbfread/dbfread/test_read_and_length.py	(index)
+++ contrib/python/dbfread/dbfread/test_read_and_length.py	(working tree)
@@ -3,15 +3,15 @@ Tests reading from database.
 """
 from pytest import fixture
 import datetime
-from .dbf import DBF
+from dbfread.dbf import DBF
 
-@fixture
 def table():
-    return DBF('testcases/memotest.dbf')
+    import yatest.common as yc
+    return DBF(yc.source_path('contrib/python/dbfread/testcases/memotest.dbf'))
 
-@fixture
 def loaded_table():
-    return DBF('testcases/memotest.dbf', load=True)
+    import yatest.common as yc
+    return DBF(yc.source_path('contrib/python/dbfread/testcases/memotest.dbf'), load=True)
 
 # This relies on people.dbf having this exact content.
 records = [{u'NAME': u'Alice',
