--- contrib/python/vcrpy/py3/vcr/stubs/aiohttp_stubs/__init__.py	(index)
+++ contrib/python/vcrpy/py3/vcr/stubs/aiohttp_stubs/__init__.py	(working tree)
@@ -169,7 +169,7 @@ def vcr_request(cassette, real_request):
         headers = kwargs.get("headers")
         auth = kwargs.get("auth")
         headers = self._prepare_headers(headers)
-        data = kwargs.get("data", kwargs.get("json"))
+        data = kwargs.get("data") or kwargs.get("json")
         params = kwargs.get("params")
 
         if auth is not None:
