--- contrib/python/pycryptodome/py3/Crypto/Cipher/AES.py	(index)
+++ contrib/python/pycryptodome/py3/Crypto/Cipher/AES.py	(working tree)
@@ -50 +50 @@ _cproto = """
-        int AES_start_operation(const uint8_t key[],
+        int pycryptodome_AES_start_operation(const uint8_t key[],
@@ -53 +53 @@ _cproto = """
-        int AES_encrypt(const void *state,
+        int pycryptodome_AES_encrypt(const void *state,
@@ -57 +57 @@ _cproto = """
-        int AES_decrypt(const void *state,
+        int pycryptodome_AES_decrypt(const void *state,
@@ -61 +61 @@ _cproto = """
-        int AES_stop_operation(void *state);
+        int pycryptodome_AES_stop_operation(void *state);
@@ -99,2 +99,2 @@ def _create_base_cipher(dict_parameters):
-        start_operation = _raw_aes_lib.AES_start_operation
-        stop_operation = _raw_aes_lib.AES_stop_operation
+        start_operation = _raw_aes_lib.pycryptodome_AES_start_operation
+        stop_operation = _raw_aes_lib.pycryptodome_AES_stop_operation
--- contrib/python/pycryptodome/py3/Crypto/Cipher/ARC2.py	(index)
+++ contrib/python/pycryptodome/py3/Crypto/Cipher/ARC2.py	(working tree)
@@ -80,2 +80,2 @@ def _create_base_cipher(dict_parameters):
-    start_operation = _raw_arc2_lib.ARC2_start_operation
-    stop_operation = _raw_arc2_lib.ARC2_stop_operation
+    start_operation = _raw_arc2_lib.pycryptodome_ARC2_start_operation
+    stop_operation = _raw_arc2_lib.pycryptodome_ARC2_stop_operation
--- contrib/python/pycryptodome/py3/Crypto/Cipher/Blowfish.py	(index)
+++ contrib/python/pycryptodome/py3/Crypto/Cipher/Blowfish.py	(working tree)
@@ -73,2 +73,2 @@ def _create_base_cipher(dict_parameters):
-    start_operation = _raw_blowfish_lib.Blowfish_start_operation
-    stop_operation = _raw_blowfish_lib.Blowfish_stop_operation
+    start_operation = _raw_blowfish_lib.pycryptodome_Blowfish_start_operation
+    stop_operation = _raw_blowfish_lib.pycryptodome_Blowfish_stop_operation
--- contrib/python/pycryptodome/py3/Crypto/Cipher/CAST.py	(index)
+++ contrib/python/pycryptodome/py3/Crypto/Cipher/CAST.py	(working tree)
@@ -72,2 +72,2 @@ def _create_base_cipher(dict_parameters):
-    start_operation = _raw_cast_lib.CAST_start_operation
-    stop_operation = _raw_cast_lib.CAST_stop_operation
+    start_operation = _raw_cast_lib.pycryptodome_CAST_start_operation
+    stop_operation = _raw_cast_lib.pycryptodome_CAST_stop_operation
--- contrib/python/pycryptodome/py3/Crypto/Cipher/DES.py	(index)
+++ contrib/python/pycryptodome/py3/Crypto/Cipher/DES.py	(working tree)
@@ -72,2 +72,2 @@ def _create_base_cipher(dict_parameters):
-    start_operation = _raw_des_lib.DES_start_operation
-    stop_operation = _raw_des_lib.DES_stop_operation
+    start_operation = _raw_des_lib.pycryptodome_DES_start_operation
+    stop_operation = _raw_des_lib.pycryptodome_DES_stop_operation
--- contrib/python/pycryptodome/py3/Crypto/Cipher/DES3.py	(index)
+++ contrib/python/pycryptodome/py3/Crypto/Cipher/DES3.py	(working tree)
@@ -101,2 +101,2 @@ def _create_base_cipher(dict_parameters):
-    start_operation = _raw_des3_lib.DES3_start_operation
-    stop_operation = _raw_des3_lib.DES3_stop_operation
+    start_operation = _raw_des3_lib.pycryptodome_DES3_start_operation
+    stop_operation = _raw_des3_lib.pycryptodome_DES3_stop_operation
--- contrib/python/pycryptodome/py3/Crypto/Cipher/_EKSBlowfish.py	(index)
+++ contrib/python/pycryptodome/py3/Crypto/Cipher/_EKSBlowfish.py	(working tree)
@@ -77,2 +77,2 @@ def _create_base_cipher(dict_parameters):
-    start_operation = _raw_blowfish_lib.EKSBlowfish_start_operation
-    stop_operation = _raw_blowfish_lib.EKSBlowfish_stop_operation
+    start_operation = _raw_blowfish_lib.pycryptodome_EKSBlowfish_start_operation
+    stop_operation = _raw_blowfish_lib.pycryptodome_EKSBlowfish_stop_operation
--- contrib/python/pycryptodome/py3/Crypto/Hash/BLAKE2b.py	(index)
+++ contrib/python/pycryptodome/py3/Crypto/Hash/BLAKE2b.py	(working tree)
@@ -44 +44 @@ _raw_blake2b_lib = load_pycryptodome_raw_lib("Crypto.Hash._BLAKE2b",
-                        int blake2b_init(void **state,
+                        int pycryptodome_blake2b_init(void **state,
@@ -48,2 +48,2 @@ _raw_blake2b_lib = load_pycryptodome_raw_lib("Crypto.Hash._BLAKE2b",
-                        int blake2b_destroy(void *state);
-                        int blake2b_update(void *state,
+                        int pycryptodome_blake2b_destroy(void *state);
+                        int pycryptodome_blake2b_update(void *state,
@@ -52 +52 @@ _raw_blake2b_lib = load_pycryptodome_raw_lib("Crypto.Hash._BLAKE2b",
-                        int blake2b_digest(const void *state,
+                        int pycryptodome_blake2b_digest(const void *state,
@@ -54 +54 @@ _raw_blake2b_lib = load_pycryptodome_raw_lib("Crypto.Hash._BLAKE2b",
-                        int blake2b_copy(const void *src, void *dst);
+                        int pycryptodome_blake2b_copy(const void *src, void *dst);
--- contrib/python/pycryptodome/py3/Crypto/Hash/BLAKE2s.py	(index)
+++ contrib/python/pycryptodome/py3/Crypto/Hash/BLAKE2s.py	(working tree)
@@ -44 +44 @@ _raw_blake2s_lib = load_pycryptodome_raw_lib("Crypto.Hash._BLAKE2s",
-                        int blake2s_init(void **state,
+                        int pycryptodome_blake2s_init(void **state,
@@ -48,2 +48,2 @@ _raw_blake2s_lib = load_pycryptodome_raw_lib("Crypto.Hash._BLAKE2s",
-                        int blake2s_destroy(void *state);
-                        int blake2s_update(void *state,
+                        int pycryptodome_blake2s_destroy(void *state);
+                        int pycryptodome_blake2s_update(void *state,
@@ -52 +52 @@ _raw_blake2s_lib = load_pycryptodome_raw_lib("Crypto.Hash._BLAKE2s",
-                        int blake2s_digest(const void *state,
+                        int pycryptodome_blake2s_digest(const void *state,
@@ -54 +54 @@ _raw_blake2s_lib = load_pycryptodome_raw_lib("Crypto.Hash._BLAKE2s",
-                        int blake2s_copy(const void *src, void *dst);
+                        int pycryptodome_blake2s_copy(const void *src, void *dst);
--- contrib/python/pycryptodome/py3/Crypto/Hash/Poly1305.py	(index)
+++ contrib/python/pycryptodome/py3/Crypto/Hash/Poly1305.py	(working tree)
@@ -38 +38 @@ _raw_poly1305 = load_pycryptodome_raw_lib("Crypto.Hash._poly1305",
-                        int poly1305_init(void **state,
+                        int pycryptodome_poly1305_init(void **state,
@@ -73 +73 @@ class Poly1305_MAC(object):
-        result = _raw_poly1305.poly1305_init(state.address_of(),
+        result = _raw_poly1305.pycryptodome_poly1305_init(state.address_of(),
--- contrib/python/pycryptodome/py3/Crypto/SelfTest/Signature/test_dss.py	(index)
+++ contrib/python/pycryptodome/py3/Crypto/SelfTest/Signature/test_dss.py	(working tree)
@@ -1180 +1180 @@ class TestVectorsDSAWycheproof(unittest.TestCase):
-    def __init__(self, wycheproof_warnings, slow_tests):
+    def __init__(self, wycheproof_warnings=False, slow_tests=False):
@@ -1239 +1239 @@ class TestVectorsECDSAWycheproof(unittest.TestCase):
-    def __init__(self, wycheproof_warnings, slow_tests):
+    def __init__(self, wycheproof_warnings=False, slow_tests=False):
--- contrib/python/pycryptodome/py3/Crypto/Util/_raw_api.py	(index)
+++ contrib/python/pycryptodome/py3/Crypto/Util/_raw_api.py	(working tree)
@@ -78,0 +79,2 @@ try:
+    raise ImportError("Disable CFFI for Arcadia")
+
@@ -179,12 +181 @@ except ImportError:
-        if not cached_architecture:
-            # platform.architecture() creates a subprocess, so caching the
-            # result makes successive imports faster.
-            import platform
-            cached_architecture[:] = platform.architecture()
-        bits, linkage = cached_architecture
-        if "." not in name and not linkage.startswith("Win"):
-            full_name = find_library(name)
-            if full_name is None:
-                raise OSError("Cannot load library '%s'" % name)
-            name = full_name
-        return CDLL(name)
+        return CDLL(find_library(name))
@@ -295,15 +286 @@ def load_pycryptodome_raw_lib(name, cdecl):
-
-    split = name.split(".")
-    dir_comps, basename = split[:-1], split[-1]
-    attempts = []
-    for ext in extension_suffixes:
-        try:
-            filename = basename + ext
-            full_name = pycryptodome_filename(dir_comps, filename)
-            if not os.path.isfile(full_name):
-                attempts.append("Not found '%s'" % filename)
-                continue
-            return load_lib(full_name, cdecl)
-        except OSError as exp:
-            attempts.append("Cannot load '%s': %s" % (filename, str(exp)))
-    raise OSError("Cannot load native module '%s': %s" % (name, ", ".join(attempts)))
+    return load_lib(name, cdecl)
--- contrib/python/pycryptodome/py3/src/AES.c	(index)
+++ contrib/python/pycryptodome/py3/src/AES.c	(working tree)
@@ -33 +33 @@ FAKE_INIT(raw_aes)
-#define MODULE_NAME AES
+#define MODULE_NAME pycryptodome_AES
--- contrib/python/pycryptodome/py3/src/AESNI.c	(index)
+++ contrib/python/pycryptodome/py3/src/AESNI.c	(working tree)
@@ -140 +140 @@ static int expand_key(__m128i *erk, __m128i *drk, const uint8_t *key, unsigned N
-static int AESNI_encrypt(const BlockBase *bb, const uint8_t *in, uint8_t *out, size_t data_len)
+int AESNI_encrypt(const BlockBase *bb, const uint8_t *in, uint8_t *out, size_t data_len)
@@ -249 +249 @@ static int AESNI_encrypt(const BlockBase *bb, const uint8_t *in, uint8_t *out, s
-static int AESNI_decrypt(const BlockBase *bb, const uint8_t *in, uint8_t *out, size_t data_len)
+int AESNI_decrypt(const BlockBase *bb, const uint8_t *in, uint8_t *out, size_t data_len)
--- contrib/python/pycryptodome/py3/src/ARC2.c	(index)
+++ contrib/python/pycryptodome/py3/src/ARC2.c	(working tree)
@@ -39 +39 @@ FAKE_INIT(raw_arc2)
-#define MODULE_NAME ARC2
+#define MODULE_NAME pycryptodome_ARC2
@@ -202 +202 @@ block_finalize(struct block_state* self)
-EXPORT_SYM int ARC2_start_operation(const uint8_t key[], size_t key_len, size_t effective_key_len, ARC2_State **pResult)
+EXPORT_SYM int pycryptodome_ARC2_start_operation(const uint8_t key[], size_t key_len, size_t effective_key_len, pycryptodome_ARC2_State **pResult)
@@ -209 +209 @@ EXPORT_SYM int ARC2_start_operation(const uint8_t key[], size_t key_len, size_t 
-    *pResult = calloc(1, sizeof(ARC2_State));
+    *pResult = calloc(1, sizeof(pycryptodome_ARC2_State));
@@ -214,3 +214,3 @@ EXPORT_SYM int ARC2_start_operation(const uint8_t key[], size_t key_len, size_t 
-    block_base->encrypt = &ARC2_encrypt;
-    block_base->decrypt = &ARC2_decrypt;
-    block_base->destructor = &ARC2_stop_operation;
+    block_base->encrypt = &pycryptodome_ARC2_encrypt;
+    block_base->decrypt = &pycryptodome_ARC2_decrypt;
+    block_base->destructor = &pycryptodome_ARC2_stop_operation;
--- contrib/python/pycryptodome/py3/src/CAST.c	(index)
+++ contrib/python/pycryptodome/py3/src/CAST.c	(working tree)
@@ -51 +51 @@ FAKE_INIT(raw_cast)
-#define MODULE_NAME CAST
+#define MODULE_NAME pycryptodome_CAST
--- contrib/python/pycryptodome/py3/src/DES.c	(index)
+++ contrib/python/pycryptodome/py3/src/DES.c	(working tree)
@@ -92 +92 @@ static void block_decrypt(struct block_state *self, const uint8_t *in, uint8_t *
-# define MODULE_NAME DES3   /* triple DES */
+# define MODULE_NAME pycryptodome_DES3   /* triple DES */
@@ -96 +96 @@ static void block_decrypt(struct block_state *self, const uint8_t *in, uint8_t *
-# define MODULE_NAME DES   /* single DES */
+# define MODULE_NAME pycryptodome_DES   /* single DES */
--- contrib/python/pycryptodome/py3/src/bignum.c	(index)
+++ contrib/python/pycryptodome/py3/src/bignum.c	(working tree)
@@ -250 +250 @@ STATIC void add_mod(uint64_t* out, const uint64_t* a, const uint64_t* b, const u
-int sub_mod(uint64_t *out, const uint64_t *a, const uint64_t *b, const uint64_t *modulus, uint64_t *tmp1, uint64_t *tmp2, size_t nw)
+STATIC int sub_mod(uint64_t *out, const uint64_t *a, const uint64_t *b, const uint64_t *modulus, uint64_t *tmp1, uint64_t *tmp2, size_t nw)
--- contrib/python/pycryptodome/py3/src/block_common.c	(index)
+++ contrib/python/pycryptodome/py3/src/block_common.c	(working tree)
@@ -47 +47 @@ typedef struct {
-static int CIPHER_ENCRYPT
+int CIPHER_ENCRYPT
@@ -69 +69 @@ static int CIPHER_ENCRYPT
-static int CIPHER_DECRYPT
+int CIPHER_DECRYPT
--- contrib/python/pycryptodome/py3/src/blowfish.c	(index)
+++ contrib/python/pycryptodome/py3/src/blowfish.c	(working tree)
@@ -40 +40 @@
-#define MODULE_NAME EKSBlowfish
+#define MODULE_NAME pycryptodome_EKSBlowfish
@@ -43 +43 @@ FAKE_INIT(raw_eksblowfish)
-#define MODULE_NAME Blowfish
+#define MODULE_NAME pycryptodome_Blowfish
--- contrib/python/pycryptodome/py3/src/mont.c	(index)
+++ contrib/python/pycryptodome/py3/src/mont.c	(working tree)
@@ -60 +60 @@
-#include "modexp_utils.c"
+#include "modexp_utils.h"
--- contrib/python/pycryptodome/py3/src/poly1305.c	(index)
+++ contrib/python/pycryptodome/py3/src/poly1305.c	(working tree)
@@ -334 +334 @@ static void poly1305_finalize(uint32_t h[5], const uint32_t s[5])
-EXPORT_SYM int poly1305_init(mac_state **pState,
+EXPORT_SYM int pycryptodome_poly1305_init(mac_state **pState,
@@ -434 +434 @@ int main(void)
-    poly1305_init(&state, r, 16, s, 16);
+    pycryptodome_poly1305_init(&state, r, 16, s, 16);
