--- contrib/python/isodate/py2/isodate/tests/test_date.py	(index)
+++ contrib/python/isodate/py2/isodate/tests/test_date.py	(working tree)
@@ -27,8 +27,10 @@
 '''
 Test cases for the isodate module.
 '''
-import unittest
 from datetime import date
+
+import pytest
+
 from isodate import parse_date, ISO8601Error, date_isoformat
 from isodate import DATE_CENTURY, DATE_YEAR
 from isodate import DATE_BAS_MONTH, DATE_EXT_MONTH
@@ -37,96 +39,62 @@ from isodate import DATE_BAS_ORD_COMPLETE, DATE_EXT_ORD_COMPLETE
 from isodate import DATE_BAS_WEEK, DATE_BAS_WEEK_COMPLETE
 from isodate import DATE_EXT_WEEK, DATE_EXT_WEEK_COMPLETE
 
-# the following list contains tuples of ISO date strings and the expected
-# result from the parse_date method. A result of None means an ISO8601Error
-# is expected. The test cases are grouped into dates with 4 digit years
-# and 6 digit years.
-TEST_CASES = {4: [('19', date(1901, 1, 1), DATE_CENTURY),
-                  ('1985', date(1985, 1, 1), DATE_YEAR),
-                  ('1985-04', date(1985, 4, 1), DATE_EXT_MONTH),
-                  ('198504', date(1985, 4, 1), DATE_BAS_MONTH),
-                  ('1985-04-12', date(1985, 4, 12), DATE_EXT_COMPLETE),
-                  ('19850412', date(1985, 4, 12), DATE_BAS_COMPLETE),
-                  ('1985102', date(1985, 4, 12), DATE_BAS_ORD_COMPLETE),
-                  ('1985-102', date(1985, 4, 12), DATE_EXT_ORD_COMPLETE),
-                  ('1985W155', date(1985, 4, 12), DATE_BAS_WEEK_COMPLETE),
-                  ('1985-W15-5', date(1985, 4, 12), DATE_EXT_WEEK_COMPLETE),
-                  ('1985W15', date(1985, 4, 8), DATE_BAS_WEEK),
-                  ('1985-W15', date(1985, 4, 8), DATE_EXT_WEEK),
-                  ('1989-W15', date(1989, 4, 10), DATE_EXT_WEEK),
-                  ('1989-W15-5', date(1989, 4, 14), DATE_EXT_WEEK_COMPLETE),
-                  ('1-W1-1', None, DATE_BAS_WEEK_COMPLETE)],
-              6: [('+0019', date(1901, 1, 1), DATE_CENTURY),
-                  ('+001985', date(1985, 1, 1), DATE_YEAR),
-                  ('+001985-04', date(1985, 4, 1), DATE_EXT_MONTH),
-                  ('+001985-04-12', date(1985, 4, 12), DATE_EXT_COMPLETE),
-                  ('+0019850412', date(1985, 4, 12), DATE_BAS_COMPLETE),
-                  ('+001985102', date(1985, 4, 12), DATE_BAS_ORD_COMPLETE),
-                  ('+001985-102', date(1985, 4, 12), DATE_EXT_ORD_COMPLETE),
-                  ('+001985W155', date(1985, 4, 12), DATE_BAS_WEEK_COMPLETE),
-                  ('+001985-W15-5', date(1985, 4, 12), DATE_EXT_WEEK_COMPLETE),
-                  ('+001985W15', date(1985, 4, 8), DATE_BAS_WEEK),
-                  ('+001985-W15', date(1985, 4, 8), DATE_EXT_WEEK)]}
 
+TEST_CASES = [
+    (4, '19', date(1901, 1, 1), DATE_CENTURY),
+    (4, '1985', date(1985, 1, 1), DATE_YEAR),
+    (4, '1985-04', date(1985, 4, 1), DATE_EXT_MONTH),
+    (4, '198504', date(1985, 4, 1), DATE_BAS_MONTH),
+    (4, '1985-04-12', date(1985, 4, 12), DATE_EXT_COMPLETE),
+    (4, '19850412', date(1985, 4, 12), DATE_BAS_COMPLETE),
+    (4, '1985102', date(1985, 4, 12), DATE_BAS_ORD_COMPLETE),
+    (4, '1985-102', date(1985, 4, 12), DATE_EXT_ORD_COMPLETE),
+    (4, '1985W155', date(1985, 4, 12), DATE_BAS_WEEK_COMPLETE),
+    (4, '1985-W15-5', date(1985, 4, 12), DATE_EXT_WEEK_COMPLETE),
+    (4, '1985W15', date(1985, 4, 8), DATE_BAS_WEEK),
+    (4, '1985-W15', date(1985, 4, 8), DATE_EXT_WEEK),
+    (4, '1989-W15', date(1989, 4, 10), DATE_EXT_WEEK),
+    (4, '1989-W15-5', date(1989, 4, 14), DATE_EXT_WEEK_COMPLETE),
+    (4, '1-W1-1', None, DATE_BAS_WEEK_COMPLETE),
+    (6, '+0019', date(1901, 1, 1), DATE_CENTURY),
+    (6, '+001985', date(1985, 1, 1), DATE_YEAR),
+    (6, '+001985-04', date(1985, 4, 1), DATE_EXT_MONTH),
+    (6, '+001985-04-12', date(1985, 4, 12), DATE_EXT_COMPLETE),
+    (6, '+0019850412', date(1985, 4, 12), DATE_BAS_COMPLETE),
+    (6, '+001985102', date(1985, 4, 12), DATE_BAS_ORD_COMPLETE),
+    (6, '+001985-102', date(1985, 4, 12), DATE_EXT_ORD_COMPLETE),
+    (6, '+001985W155', date(1985, 4, 12), DATE_BAS_WEEK_COMPLETE),
+    (6, '+001985-W15-5', date(1985, 4, 12), DATE_EXT_WEEK_COMPLETE),
+    (6, '+001985W15', date(1985, 4, 8), DATE_BAS_WEEK),
+    (6, '+001985-W15', date(1985, 4, 8), DATE_EXT_WEEK),
+]
 
-def create_testcase(yeardigits, datestring, expectation, format):
-    '''
-    Create a TestCase class for a specific test.
 
-    This allows having a separate TestCase for each test tuple from the
-    TEST_CASES list, so that a failed test won't stop other tests.
+@pytest.mark.parametrize("yeardigits,datestring,expectation,format", TEST_CASES)
+def test_parse(yeardigits, datestring, expectation, format):
     '''
-
-    class TestDate(unittest.TestCase):
-        '''
-        A test case template to parse an ISO date string into a date
-        object.
-        '''
-
-        def test_parse(self):
-            '''
-            Parse an ISO date string and compare it to the expected value.
-            '''
-            if expectation is None:
-                self.assertRaises(ISO8601Error, parse_date, datestring,
-                                  yeardigits)
-            else:
-                result = parse_date(datestring, yeardigits)
-                self.assertEqual(result, expectation)
-
-        def test_format(self):
-            '''
-            Take date object and create ISO string from it.
-            This is the reverse test to test_parse.
-            '''
-            if expectation is None:
-                self.assertRaises(AttributeError,
-                                  date_isoformat, expectation, format,
-                                  yeardigits)
-            else:
-                self.assertEqual(date_isoformat(expectation, format,
-                                                yeardigits),
-                                 datestring)
-
-    return unittest.TestLoader().loadTestsFromTestCase(TestDate)
+    Parse an ISO date string and compare it to the expected value.
+    '''
+    if expectation is None:
+        with pytest.raises(ISO8601Error):
+            parse_date(datestring, yeardigits)
+    else:
+        result = parse_date(datestring, yeardigits)
+        assert result, expectation
 
 
-def test_suite():
+@pytest.mark.parametrize("yeardigits,datestring,expectation,format", TEST_CASES)
+def test_format(yeardigits, datestring, expectation, format):
     '''
-    Construct a TestSuite instance for all test cases.
+    Take date object and create ISO string from it.
+    This is the reverse test to test_parse.
     '''
-    suite = unittest.TestSuite()
-    for yeardigits, tests in TEST_CASES.items():
-        for datestring, expectation, format in tests:
-            suite.addTest(create_testcase(yeardigits, datestring,
-                                          expectation, format))
-    return suite
-
-
-# load_tests Protocol
-def load_tests(loader, tests, pattern):
-    return test_suite()
-
-
-if __name__ == '__main__':
-    unittest.main(defaultTest='test_suite')
+    if expectation is None:
+        with pytest.raises(AttributeError):
+            date_isoformat(expectation, format, yeardigits)
+    else:
+        assert date_isoformat(
+            expectation,
+            format,
+            yeardigits
+        ) == datestring
--- contrib/python/isodate/py2/isodate/tests/test_datetime.py	(index)
+++ contrib/python/isodate/py2/isodate/tests/test_datetime.py	(working tree)
@@ -27,9 +27,10 @@
 '''
 Test cases for the isodatetime module.
 '''
-import unittest
 from datetime import datetime
 
+import pytest
+
 from isodate import parse_datetime, UTC, FixedOffset, datetime_isoformat
 from isodate import ISO8601Error
 from isodate import DATE_BAS_COMPLETE, TIME_BAS_MINUTE, TIME_BAS_COMPLETE
@@ -107,51 +108,27 @@ def create_testcase(datetimestring, expectation, format, output):
     TEST_CASES list, so that a failed test won't stop other tests.
     """
 
-    class TestDateTime(unittest.TestCase):
-        '''
-        A test case template to parse an ISO datetime string into a
-        datetime object.
-        '''
-
-        def test_parse(self):
-            '''
-            Parse an ISO datetime string and compare it to the expected value.
-            '''
-            if expectation is None:
-                self.assertRaises(ISO8601Error, parse_datetime, datetimestring)
-            else:
-                self.assertEqual(parse_datetime(datetimestring), expectation)
-
-        def test_format(self):
-            '''
-            Take datetime object and create ISO string from it.
-            This is the reverse test to test_parse.
-            '''
-            if expectation is None:
-                self.assertRaises(AttributeError,
-                                  datetime_isoformat, expectation, format)
-            else:
-                self.assertEqual(datetime_isoformat(expectation, format),
-                                 output)
-
-    return unittest.TestLoader().loadTestsFromTestCase(TestDateTime)
-
 
-def test_suite():
+@pytest.mark.parametrize("datetimestring,expectation,format,output", TEST_CASES)
+def test_parse(datetimestring, expectation, format, output):
     '''
-    Construct a TestSuite instance for all test cases.
+    Parse an ISO datetime string and compare it to the expected value.
     '''
-    suite = unittest.TestSuite()
-    for datetimestring, expectation, format, output in TEST_CASES:
-        suite.addTest(create_testcase(datetimestring, expectation,
-                                      format, output))
-    return suite
+    if expectation is None:
+        with pytest.raises(ISO8601Error):
+            parse_datetime(datetimestring)
+    else:
+        assert parse_datetime(datetimestring) == expectation
 
 
-# load_tests Protocol
-def load_tests(loader, tests, pattern):
-    return test_suite()
-
-
-if __name__ == '__main__':
-    unittest.main(defaultTest='test_suite')
+@pytest.mark.parametrize("datetimestring,expectation,format,output", TEST_CASES)
+def test_format(datetimestring, expectation, format, output):
+    '''
+    Take datetime object and create ISO string from it.
+    This is the reverse test to test_parse.
+    '''
+    if expectation is None:
+        with pytest.raises(AttributeError):
+            datetime_isoformat(expectation, format)
+    else:
+        assert datetime_isoformat(expectation, format) == output
--- contrib/python/isodate/py2/isodate/tests/test_duration.py	(index)
+++ contrib/python/isodate/py2/isodate/tests/test_duration.py	(working tree)
@@ -31,14 +31,16 @@ import unittest
 import operator
 from datetime import timedelta, date, datetime
 
+import pytest
+
 from isodate import Duration, parse_duration, ISO8601Error
 from isodate import D_DEFAULT, D_WEEK, D_ALT_EXT, duration_isoformat
 
 # the following list contains tuples of ISO duration strings and the expected
 # result from the parse_duration method. A result of None means an ISO8601Error
 # is expected.
-PARSE_TEST_CASES = {'P18Y9M4DT11H9M8S': (Duration(4, 8, 0, 0, 9, 11, 0, 9, 18),
-                                         D_DEFAULT, None),
+PARSE_TEST_CASES = [(durationstring, expectation, format, altstr) for durationstring, (expectation, format, altstr) in {'P18Y9M4DT11H9M8S': (Duration(4, 8, 0, 0, 9, 11, 0, 9, 18),
+                    D_DEFAULT, None),
                     'P2W': (timedelta(weeks=2), D_WEEK, None),
                     'P3Y6M4DT12H30M5S': (Duration(4, 5, 0, 0, 30, 12, 0, 6, 3),
                                          D_DEFAULT, None),
@@ -77,7 +79,7 @@ PARSE_TEST_CASES = {'P18Y9M4DT11H9M8S': (Duration(4, 8, 0, 0, 9, 11, 0, 9, 18),
                     'P0018-09-04T11:09:08': (Duration(4, 8, 0, 0, 9, 11, 0, 9,
                                                       18), D_ALT_EXT, None),
                     # 'PT000022.22': timedelta(seconds=22.22),
-                    }
+                    }.items()]
 
 #                       d1                    d2           '+', '-', '>'
 # A list of test cases to test addition and subtraction between datetime and
@@ -85,18 +87,21 @@ PARSE_TEST_CASES = {'P18Y9M4DT11H9M8S': (Duration(4, 8, 0, 0, 9, 11, 0, 9, 18),
 # each tuple contains 2 duration strings, and a result string for addition and
 # one for subtraction. The last value says, if the first duration is greater
 # than the second.
-MATH_TEST_CASES = (('P5Y7M1DT9H45M16.72S', 'PT27M24.68S',
-                    'P5Y7M1DT10H12M41.4S', 'P5Y7M1DT9H17M52.04S', None),
-                   ('PT28M12.73S', 'PT56M29.92S',
-                    'PT1H24M42.65S', '-PT28M17.19S', False),
-                   ('P3Y7M23DT5H25M0.33S', 'PT1H1.95S',
-                    'P3Y7M23DT6H25M2.28S', 'P3Y7M23DT4H24M58.38S', None),
-                   ('PT1H1.95S', 'P3Y7M23DT5H25M0.33S',
-                    'P3Y7M23DT6H25M2.28S', '-P3Y7M23DT4H24M58.38S', None),
-                   ('P1332DT55M0.33S', 'PT1H1.95S',
-                    'P1332DT1H55M2.28S', 'P1331DT23H54M58.38S', True),
-                   ('PT1H1.95S', 'P1332DT55M0.33S',
-                    'P1332DT1H55M2.28S', '-P1331DT23H54M58.38S', False))
+MATH_TEST_CASES = [
+    (parse_duration(dur1), parse_duration(dur2), parse_duration(resadd), parse_duration(ressub), resge)
+    for dur1, dur2, resadd, ressub, resge in (
+        ('P5Y7M1DT9H45M16.72S', 'PT27M24.68S',
+         'P5Y7M1DT10H12M41.4S', 'P5Y7M1DT9H17M52.04S', None),
+        ('PT28M12.73S', 'PT56M29.92S',
+         'PT1H24M42.65S', '-PT28M17.19S', False),
+        ('P3Y7M23DT5H25M0.33S', 'PT1H1.95S',
+         'P3Y7M23DT6H25M2.28S', 'P3Y7M23DT4H24M58.38S', None),
+        ('PT1H1.95S', 'P3Y7M23DT5H25M0.33S',
+         'P3Y7M23DT6H25M2.28S', '-P3Y7M23DT4H24M58.38S', None),
+        ('P1332DT55M0.33S', 'PT1H1.95S',
+         'P1332DT1H55M2.28S', 'P1331DT23H54M58.38S', True),
+        ('PT1H1.95S', 'P1332DT55M0.33S',
+         'P1332DT1H55M2.28S', '-P1331DT23H54M58.38S', False))]
 
 
 # A list of test cases to test addition and subtraction of date/datetime
@@ -410,42 +415,27 @@ class DurationTest(unittest.TestCase):
         self.assertEqual(dur.totimedelta(datetime(2001, 3, 25)), timedelta(61))
 
 
-def create_parsetestcase(durationstring, expectation, format, altstr):
-    """
-    Create a TestCase class for a specific test.
-
-    This allows having a separate TestCase for each test tuple from the
-    PARSE_TEST_CASES list, so that a failed test won't stop other tests.
-    """
+@pytest.mark.parametrize("durationstring, expectation, format, altstr", PARSE_TEST_CASES)
+def test_parse(durationstring, expectation, format, altstr):
+    '''
+    Parse an ISO duration string and compare it to the expected value.
+    '''
+    result = parse_duration(durationstring)
+    assert result, expectation
 
-    class TestParseDuration(unittest.TestCase):
-        '''
-        A test case template to parse an ISO duration string into a
-        timedelta or Duration object.
-        '''
 
-        def test_parse(self):
-            '''
-            Parse an ISO duration string and compare it to the expected value.
-            '''
-            result = parse_duration(durationstring)
-            self.assertEqual(result, expectation)
-
-        def test_format(self):
-            '''
-            Take duration/timedelta object and create ISO string from it.
-            This is the reverse test to test_parse.
-            '''
-            if altstr:
-                self.assertEqual(duration_isoformat(expectation, format),
-                                 altstr)
-            else:
-                # if durationstring == '-P2W':
-                #     import pdb; pdb.set_trace()
-                self.assertEqual(duration_isoformat(expectation, format),
-                                 durationstring)
-
-    return unittest.TestLoader().loadTestsFromTestCase(TestParseDuration)
+@pytest.mark.parametrize("durationstring, expectation, format, altstr", PARSE_TEST_CASES)
+def test_format(durationstring, expectation, format, altstr):
+    '''
+    Take duration/timedelta object and create ISO string from it.
+    This is the reverse test to test_parse.
+    '''
+    if altstr:
+        assert duration_isoformat(expectation, format) == altstr
+    else:
+        # if durationstring == '-P2W':
+        #     import pdb; pdb.set_trace()
+        assert duration_isoformat(expectation, format) == durationstring
 
 
 def create_mathtestcase(dur1, dur2, resadd, ressub, resge):
@@ -467,140 +457,76 @@ def create_mathtestcase(dur1, dur2, resadd, ressub, resge):
         operators for Duration objects.
         '''
 
-        def test_add(self):
-            '''
-            Test operator + (__add__, __radd__)
-            '''
-            self.assertEqual(dur1 + dur2, resadd)
-
-        def test_sub(self):
-            '''
-            Test operator - (__sub__, __rsub__)
-            '''
-            self.assertEqual(dur1 - dur2, ressub)
-
-        def test_ge(self):
-            '''
-            Test operator > and <
-            '''
-            def dogetest():
-                ''' Test greater than.'''
-                return dur1 > dur2
-
-            def doletest():
-                ''' Test less than.'''
-                return dur1 < dur2
-            if resge is None:
-                self.assertRaises(TypeError, dogetest)
-                self.assertRaises(TypeError, doletest)
-            else:
-                self.assertEqual(dogetest(), resge)
-                self.assertEqual(doletest(), not resge)
-
-    return unittest.TestLoader().loadTestsFromTestCase(TestMathDuration)
-
-
-def create_datetestcase(start, tdelta, duration):
-    """
-    Create a TestCase class for a specific test.
-
-    This allows having a separate TestCase for each test tuple from the
-    DATE_TEST_CASES list, so that a failed test won't stop other tests.
-    """
-
-    class TestDateCalc(unittest.TestCase):
-        '''
-        A test case template test addition, subtraction
-        operators for Duration objects.
-        '''
-
-        def test_add(self):
-            '''
-            Test operator +.
-            '''
-            self.assertEqual(start + tdelta, start + duration)
-
-        def test_sub(self):
-            '''
-            Test operator -.
-            '''
-            self.assertEqual(start - tdelta, start - duration)
-
-    return unittest.TestLoader().loadTestsFromTestCase(TestDateCalc)
-
 
-def create_datecalctestcase(start, duration, expectation):
-    """
-    Create a TestCase class for a specific test.
-
-    This allows having a separate TestCase for each test tuple from the
-    DATE_CALC_TEST_CASES list, so that a failed test won't stop other tests.
-    """
-
-    class TestDateCalc(unittest.TestCase):
-        '''
-        A test case template test addition operators for Duration objects.
-        '''
+@pytest.mark.parametrize("dur1,dur2,resadd,ressub,resge", MATH_TEST_CASES)
+def test_add(dur1, dur2, resadd, ressub, resge):
+    '''
+    Test operator + (__add__, __radd__)
+    '''
+    assert dur1 + dur2, resadd
 
-        def test_calc(self):
-            '''
-            Test operator +.
-            '''
-            if expectation is None:
-                self.assertRaises(ValueError, operator.add, start, duration)
-            else:
-                self.assertEqual(start + duration, expectation)
 
-    return unittest.TestLoader().loadTestsFromTestCase(TestDateCalc)
+@pytest.mark.parametrize("dur1,dur2,resadd,ressub,resge", MATH_TEST_CASES)
+def test_sub(dur1, dur2, resadd, ressub, resge):
+    '''
+    Test operator - (__sub__, __rsub__)
+    '''
+    assert dur1 - dur2, resadd
 
 
-def create_datemultestcase(operand1, operand2, expectation):
-    """
-    Create a TestCase class for a specific test.
+@pytest.mark.parametrize("dur1,dur2,resadd,ressub,resge", MATH_TEST_CASES)
+def test_ge(dur1, dur2, resadd, ressub, resge):
+    '''
+    Test operator > and <
+    '''
+    def dogetest():
+        ''' Test greater than.'''
+        return dur1 > dur2
+
+    def doletest():
+        ''' Test less than.'''
+        return dur1 < dur2
+    if resge is None:
+        with pytest.raises(TypeError):
+            dogetest()
+        with pytest.raises(TypeError):
+            doletest()
+    else:
+        assert dogetest() is resge
+        assert doletest() is not resge
+
+
+@pytest.mark.parametrize("start,tdelta,duration", DATE_TEST_CASES)
+def test_date_calc_add(start, tdelta, duration):
+    '''
+    Test operator +.
+    '''
+    assert start + tdelta == start + duration
 
-    This allows having a separate TestCase for each test tuple from the
-    DATE_CALC_TEST_CASES list, so that a failed test won't stop other tests.
-    """
 
-    class TestDateMul(unittest.TestCase):
-        '''
-        A test case template test addition operators for Duration objects.
-        '''
+@pytest.mark.parametrize("start,tdelta,duration", DATE_TEST_CASES)
+def test_date_calc_sub(start, tdelta, duration):
+    '''
+    Test operator -.
+    '''
+    assert start - tdelta == start - duration
 
-        def test_mul(self):
-            '''
-            Test operator *.
-            '''
-            self.assertEqual(operand1 * operand2, expectation)
 
-    return unittest.TestLoader().loadTestsFromTestCase(TestDateMul)
+@pytest.mark.parametrize("start,duration,expectation", DATE_CALC_TEST_CASES)
+def test_date_calc(start, duration, expectation):
+    '''
+    Test operator +.
+    '''
+    if expectation is None:
+        with pytest.raises(ValueError):
+            operator.add(start, duration)
+    else:
+        assert start + duration == expectation
 
 
-def test_suite():
+@pytest.mark.parametrize("operand1,operand2,expectation", DATE_MUL_TEST_CASES)
+def test_date_mul(operand1, operand2, expectation):
     '''
-    Return a test suite containing all test defined above.
+    Test operator *.
     '''
-    suite = unittest.TestSuite()
-    for durationstring, (expectation, format,
-                         altstr) in PARSE_TEST_CASES.items():
-        suite.addTest(create_parsetestcase(durationstring, expectation,
-                                           format, altstr))
-    for testdata in MATH_TEST_CASES:
-        suite.addTest(create_mathtestcase(*testdata))
-    for testdata in DATE_TEST_CASES:
-        suite.addTest(create_datetestcase(*testdata))
-    for testdata in DATE_CALC_TEST_CASES:
-        suite.addTest(create_datecalctestcase(*testdata))
-    for testdata in DATE_MUL_TEST_CASES:
-        suite.addTest(create_datemultestcase(*testdata))
-    suite.addTest(unittest.TestLoader().loadTestsFromTestCase(DurationTest))
-    return suite
-
-
-# load_tests Protocol
-def load_tests(loader, tests, pattern):
-    return test_suite()
-
-
-if __name__ == '__main__':
-    unittest.main(defaultTest='test_suite')
+    assert operand1 * operand2 == expectation
--- contrib/python/isodate/py2/isodate/tests/test_pickle.py	(index)
+++ contrib/python/isodate/py2/isodate/tests/test_pickle.py	(working tree)
@@ -1,8 +1,7 @@
 import unittest
 
-from six.moves import cPickle as pickle
-
 import isodate
+from six.moves import cPickle as pickle
 
 
 class TestPickle(unittest.TestCase):
@@ -43,21 +42,3 @@ class TestPickle(unittest.TestCase):
         isodate.UTC objects remain the same after pickling.
         '''
         self.assertTrue(isodate.UTC is pickle.loads(pickle.dumps(isodate.UTC)))
-
-
-def test_suite():
-    '''
-    Construct a TestSuite instance for all test cases.
-    '''
-    suite = unittest.TestSuite()
-    suite.addTest(unittest.TestLoader().loadTestsFromTestCase(TestPickle))
-    return suite
-
-
-# load_tests Protocol
-def load_tests(loader, tests, pattern):
-    return test_suite()
-
-
-if __name__ == '__main__':
-    unittest.main(defaultTest='test_suite')
--- contrib/python/isodate/py2/isodate/tests/test_strf.py	(index)
+++ contrib/python/isodate/py2/isodate/tests/test_strf.py	(working tree)
@@ -27,9 +27,11 @@
 '''
 Test cases for the isodate module.
 '''
-import unittest
 import time
 from datetime import datetime, timedelta
+
+import pytest
+
 from isodate import strftime
 from isodate import LOCAL
 from isodate import DT_EXT_COMPLETE
@@ -51,86 +53,54 @@ TEST_CASES = ((datetime(2012, 12, 25, 13, 30, 0, 0, LOCAL), DT_EXT_COMPLETE,
               )
 
 
-def create_testcase(dt, format, expectation):
-    """
-    Create a TestCase class for a specific test.
-
-    This allows having a separate TestCase for each test tuple from the
-    TEST_CASES list, so that a failed test won't stop other tests.
-    """
-
-    class TestDate(unittest.TestCase):
-        '''
-        A test case template to test ISO date formatting.
-        '''
-
-        # local time zone mock function
-        def localtime_mock(self, secs):
-            """
-            mock time.localtime so that it always returns a time_struct with
-            tm_idst=1
-            """
-            tt = self.ORIG['localtime'](secs)
-            # befor 2000 everything is dst, after 2000 no dst.
-            if tt.tm_year < 2000:
-                dst = 1
-            else:
-                dst = 0
-            tt = (tt.tm_year, tt.tm_mon, tt.tm_mday,
-                  tt.tm_hour, tt.tm_min, tt.tm_sec,
-                  tt.tm_wday, tt.tm_yday, dst)
-            return time.struct_time(tt)
-
-        def setUp(self):
-            self.ORIG = {}
-            self.ORIG['STDOFFSET'] = tzinfo.STDOFFSET
-            self.ORIG['DSTOFFSET'] = tzinfo.DSTOFFSET
-            self.ORIG['DSTDIFF'] = tzinfo.DSTDIFF
-            self.ORIG['localtime'] = time.localtime
-            # ovveride all saved values with fixtures.
-            # calculate LOCAL TZ offset, so that this test runs in
-            # every time zone
-            tzinfo.STDOFFSET = timedelta(seconds=36000)  # assume LOC = +10:00
-            tzinfo.DSTOFFSET = timedelta(seconds=39600)  # assume DST = +11:00
-            tzinfo.DSTDIFF = tzinfo.DSTOFFSET - tzinfo.STDOFFSET
-            time.localtime = self.localtime_mock
-
-        def tearDown(self):
-            # restore test fixtures
-            tzinfo.STDOFFSET = self.ORIG['STDOFFSET']
-            tzinfo.DSTOFFSET = self.ORIG['DSTOFFSET']
-            tzinfo.DSTDIFF = self.ORIG['DSTDIFF']
-            time.localtime = self.ORIG['localtime']
-
-        def test_format(self):
-            '''
-            Take date object and create ISO string from it.
-            This is the reverse test to test_parse.
-            '''
-            if expectation is None:
-                self.assertRaises(AttributeError,
-                                  strftime(dt, format))
-            else:
-                self.assertEqual(strftime(dt, format),
-                                 expectation)
-
-    return unittest.TestLoader().loadTestsFromTestCase(TestDate)
-
-
-def test_suite():
+@pytest.mark.parametrize("dt,format,expectation", TEST_CASES)
+def test_format(dt, format, expectation):
     '''
-    Construct a TestSuite instance for all test cases.
+    Take date object and create ISO string from it.
+    This is the reverse test to test_parse.
     '''
-    suite = unittest.TestSuite()
-    for dt, format, expectation in TEST_CASES:
-        suite.addTest(create_testcase(dt, format, expectation))
-    return suite
-
-
-# load_tests Protocol
-def load_tests(loader, tests, pattern):
-    return test_suite()
-
-
-if __name__ == '__main__':
-    unittest.main(defaultTest='test_suite')
+    ORIG = {}
+
+    def localtime_mock(secs):
+        """
+        mock time.localtime so that it always returns a time_struct with
+        tm_idst=1
+        """
+        tt = ORIG['localtime'](secs)
+        # befor 2000 everything is dst, after 2000 no dst.
+        if tt.tm_year < 2000:
+            dst = 1
+        else:
+            dst = 0
+        tt = (tt.tm_year, tt.tm_mon, tt.tm_mday,
+              tt.tm_hour, tt.tm_min, tt.tm_sec,
+              tt.tm_wday, tt.tm_yday, dst)
+        return time.struct_time(tt)
+
+    def setUp():
+        ORIG['STDOFFSET'] = tzinfo.STDOFFSET
+        ORIG['DSTOFFSET'] = tzinfo.DSTOFFSET
+        ORIG['DSTDIFF'] = tzinfo.DSTDIFF
+        ORIG['localtime'] = time.localtime
+        # ovveride all saved values with fixtures.
+        # calculate LOCAL TZ offset, so that this test runs in
+        # every time zone
+        tzinfo.STDOFFSET = timedelta(seconds=36000)  # assume LOC = +10:00
+        tzinfo.DSTOFFSET = timedelta(seconds=39600)  # assume DST = +11:00
+        tzinfo.DSTDIFF = tzinfo.DSTOFFSET - tzinfo.STDOFFSET
+        time.localtime = localtime_mock
+
+    def tearDown():
+        # restore test fixtures
+        tzinfo.STDOFFSET = ORIG['STDOFFSET']
+        tzinfo.DSTOFFSET = ORIG['DSTOFFSET']
+        tzinfo.DSTDIFF = ORIG['DSTDIFF']
+        time.localtime = ORIG['localtime']
+
+    setUp()
+    if expectation is None:
+        with pytest.raises(AttributeError):
+            strftime(dt, format)
+    else:
+        assert strftime(dt, format) == expectation
+    tearDown()
--- contrib/python/isodate/py2/isodate/tests/test_time.py	(index)
+++ contrib/python/isodate/py2/isodate/tests/test_time.py	(working tree)
@@ -27,9 +27,10 @@
 '''
 Test cases for the isotime module.
 '''
-import unittest
 from datetime import time
 
+import pytest
+
 from isodate import parse_time, UTC, FixedOffset, ISO8601Error, time_isoformat
 from isodate import TIME_BAS_COMPLETE, TIME_BAS_MINUTE
 from isodate import TIME_EXT_COMPLETE, TIME_EXT_MINUTE
@@ -92,59 +93,27 @@ TEST_CASES = [('232050', time(23, 20, 50), TIME_BAS_COMPLETE + TZ_BAS),
               ('1:17:30', None, TIME_EXT_COMPLETE)]
 
 
-def create_testcase(timestring, expectation, format):
-    """
-    Create a TestCase class for a specific test.
-
-    This allows having a separate TestCase for each test tuple from the
-    TEST_CASES list, so that a failed test won't stop other tests.
-    """
-
-    class TestTime(unittest.TestCase):
-        '''
-        A test case template to parse an ISO time string into a time
-        object.
-        '''
-
-        def test_parse(self):
-            '''
-            Parse an ISO time string and compare it to the expected value.
-            '''
-            if expectation is None:
-                self.assertRaises(ISO8601Error, parse_time, timestring)
-            else:
-                result = parse_time(timestring)
-                self.assertEqual(result, expectation)
-
-        def test_format(self):
-            '''
-            Take time object and create ISO string from it.
-            This is the reverse test to test_parse.
-            '''
-            if expectation is None:
-                self.assertRaises(AttributeError,
-                                  time_isoformat, expectation, format)
-            elif format is not None:
-                self.assertEqual(time_isoformat(expectation, format),
-                                 timestring)
-
-    return unittest.TestLoader().loadTestsFromTestCase(TestTime)
-
-
-def test_suite():
+@pytest.mark.parametrize("timestring,expectation,format", TEST_CASES)
+def test_parse(timestring, expectation, format):
     '''
-    Construct a TestSuite instance for all test cases.
+    Parse an ISO time string and compare it to the expected value.
     '''
-    suite = unittest.TestSuite()
-    for timestring, expectation, format in TEST_CASES:
-        suite.addTest(create_testcase(timestring, expectation, format))
-    return suite
-
+    if expectation is None:
+        with pytest.raises(ISO8601Error):
+            parse_time(timestring)
+    else:
+        result = parse_time(timestring)
+        result == expectation
 
-# load_tests Protocol
-def load_tests(loader, tests, pattern):
-    return test_suite()
 
-
-if __name__ == '__main__':
-    unittest.main(defaultTest='test_suite')
+@pytest.mark.parametrize("timestring,expectation,format", TEST_CASES)
+def test_format(timestring, expectation, format):
+    '''
+    Take time object and create ISO string from it.
+    This is the reverse test to test_parse.
+    '''
+    if expectation is None:
+        with pytest.raises(AttributeError):
+            time_isoformat(expectation, format)
+    elif format is not None:
+        assert time_isoformat(expectation, format) == timestring
