--- contrib/libs/flatbuffers/src/util.cpp	(index)
+++ contrib/libs/flatbuffers/src/util.cpp	(working tree)
@@ -409,7 +409,7 @@ ClassicLocale ClassicLocale::instance_;
   ClassicLocale::~ClassicLocale() { _free_locale(locale_); }
 #else
   ClassicLocale::ClassicLocale()
-    : locale_(newlocale(LC_ALL, "C", nullptr)) {}
+    : locale_(newlocale(LC_ALL, "C", static_cast<locale_t>(0))) {}
   ClassicLocale::~ClassicLocale() { freelocale(locale_); }
 #endif
 // clang-format on
