diff --git a/botocore/auth.py b/botocore/auth.py
index b17516d..f16bd4d 100644
--- a/botocore/auth.py
+++ b/botocore/auth.py
@@ -419,7 +419,7 @@ class SigV4Auth(BaseSigner):
     def add_auth(self, request):
         if self.credentials is None:
             raise NoCredentialsError()
-        datetime_now = datetime.datetime.utcnow()
+        datetime_now = datetime.datetime.now(datetime.UTC)
         request.context['timestamp'] = datetime_now.strftime(SIGV4_TIMESTAMP)
         # This could be a retry.  Make sure the previous
         # authorization header is removed first.
