--- a/rapidfuzz/fuzz_cpp.hpp
+++ b/rapidfuzz/fuzz_cpp.hpp
@@ -58,7 +58,7 @@ static inline RF_UncachedScorerFunc UncachedPartialRatioFuncInit()
     return scorer;
 }

-rf::ScoreAlignment<double> partial_ratio_alignment_func(const RF_String& str1, const RF_String& str2,
+static inline rf::ScoreAlignment<double> partial_ratio_alignment_func(const RF_String& str1, const RF_String& str2,
                                                         double score_cutoff)
 {
     return visitor(str1, str2, [&](auto s1, auto s2) {
