--- contrib/python/M2Crypto/py2/SWIG/libcrypto-compat.h	(index)
+++ contrib/python/M2Crypto/py2/SWIG/libcrypto-compat.h	(working tree)
@@ -13,28 +13,28 @@
-int RSA_set0_key(RSA *r, BIGNUM *n, BIGNUM *e, BIGNUM *d);
-int RSA_set0_factors(RSA *r, BIGNUM *p, BIGNUM *q);
-int RSA_set0_crt_params(RSA *r, BIGNUM *dmp1, BIGNUM *dmq1, BIGNUM *iqmp);
-void RSA_get0_key(const RSA *r, const BIGNUM **n, const BIGNUM **e, const BIGNUM **d);
-void RSA_get0_factors(const RSA *r, const BIGNUM **p, const BIGNUM **q);
-void RSA_get0_crt_params(const RSA *r, const BIGNUM **dmp1, const BIGNUM **dmq1, const BIGNUM **iqmp);
-
-void DSA_get0_pqg(const DSA *d, const BIGNUM **p, const BIGNUM **q, const BIGNUM **g);
-int DSA_set0_pqg(DSA *d, BIGNUM *p, BIGNUM *q, BIGNUM *g);
-void DSA_get0_key(const DSA *d, const BIGNUM **pub_key, const BIGNUM **priv_key);
-int DSA_set0_key(DSA *d, BIGNUM *pub_key, BIGNUM *priv_key);
-
-void DSA_SIG_get0(const DSA_SIG *sig, const BIGNUM **pr, const BIGNUM **ps);
-int DSA_SIG_set0(DSA_SIG *sig, BIGNUM *r, BIGNUM *s);
-
-void ECDSA_SIG_get0(const ECDSA_SIG *sig, const BIGNUM **pr, const BIGNUM **ps);
-int ECDSA_SIG_set0(ECDSA_SIG *sig, BIGNUM *r, BIGNUM *s);
-
-void DH_get0_pqg(const DH *dh, const BIGNUM **p, const BIGNUM **q, const BIGNUM **g);
-int DH_set0_pqg(DH *dh, BIGNUM *p, BIGNUM *q, BIGNUM *g);
-void DH_get0_key(const DH *dh, const BIGNUM **pub_key, const BIGNUM **priv_key);
-int DH_set0_key(DH *dh, BIGNUM *pub_key, BIGNUM *priv_key);
-int DH_set_length(DH *dh, long length);
-
-const unsigned char *EVP_CIPHER_CTX_iv(const EVP_CIPHER_CTX *ctx);
-unsigned char *EVP_CIPHER_CTX_iv_noconst(EVP_CIPHER_CTX *ctx);
-EVP_MD_CTX *EVP_MD_CTX_new(void);
-void EVP_MD_CTX_free(EVP_MD_CTX *ctx);
+static int RSA_set0_key(RSA *r, BIGNUM *n, BIGNUM *e, BIGNUM *d);
+static int RSA_set0_factors(RSA *r, BIGNUM *p, BIGNUM *q);
+static int RSA_set0_crt_params(RSA *r, BIGNUM *dmp1, BIGNUM *dmq1, BIGNUM *iqmp);
+static void RSA_get0_key(const RSA *r, const BIGNUM **n, const BIGNUM **e, const BIGNUM **d);
+static void RSA_get0_factors(const RSA *r, const BIGNUM **p, const BIGNUM **q);
+static void RSA_get0_crt_params(const RSA *r, const BIGNUM **dmp1, const BIGNUM **dmq1, const BIGNUM **iqmp);
+
+static void DSA_get0_pqg(const DSA *d, const BIGNUM **p, const BIGNUM **q, const BIGNUM **g);
+static int DSA_set0_pqg(DSA *d, BIGNUM *p, BIGNUM *q, BIGNUM *g);
+static void DSA_get0_key(const DSA *d, const BIGNUM **pub_key, const BIGNUM **priv_key);
+static int DSA_set0_key(DSA *d, BIGNUM *pub_key, BIGNUM *priv_key);
+
+static void DSA_SIG_get0(const DSA_SIG *sig, const BIGNUM **pr, const BIGNUM **ps);
+static int DSA_SIG_set0(DSA_SIG *sig, BIGNUM *r, BIGNUM *s);
+
+static void ECDSA_SIG_get0(const ECDSA_SIG *sig, const BIGNUM **pr, const BIGNUM **ps);
+static int ECDSA_SIG_set0(ECDSA_SIG *sig, BIGNUM *r, BIGNUM *s);
+
+static void DH_get0_pqg(const DH *dh, const BIGNUM **p, const BIGNUM **q, const BIGNUM **g);
+static int DH_set0_pqg(DH *dh, BIGNUM *p, BIGNUM *q, BIGNUM *g);
+static void DH_get0_key(const DH *dh, const BIGNUM **pub_key, const BIGNUM **priv_key);
+static int DH_set0_key(DH *dh, BIGNUM *pub_key, BIGNUM *priv_key);
+static int DH_set_length(DH *dh, long length);
+
+static const unsigned char *EVP_CIPHER_CTX_iv(const EVP_CIPHER_CTX *ctx);
+static unsigned char *EVP_CIPHER_CTX_iv_noconst(EVP_CIPHER_CTX *ctx);
+static EVP_MD_CTX *EVP_MD_CTX_new(void);
+static void EVP_MD_CTX_free(EVP_MD_CTX *ctx);
@@ -44,3 +44,3 @@ void EVP_MD_CTX_free(EVP_MD_CTX *ctx);
-int RSA_size(const RSA* rsa);
-RSA_METHOD *RSA_meth_dup(const RSA_METHOD *meth);
-int RSA_meth_set1_name(RSA_METHOD *meth, const char *name);
+//int RSA_size(const RSA* rsa);
+static RSA_METHOD *RSA_meth_dup(const RSA_METHOD *meth);
+static int RSA_meth_set1_name(RSA_METHOD *meth, const char *name);
@@ -48,4 +48,4 @@ int RSA_meth_set1_name(RSA_METHOD *meth, const char *name);
-int RSA_meth_set_priv_enc(RSA_METHOD *meth, int (*priv_enc) (int flen, const unsigned char *from, unsigned char *to, RSA *rsa, int padding));
-int RSA_meth_set_priv_dec(RSA_METHOD *meth, int (*priv_dec) (int flen, const unsigned char *from, unsigned char *to, RSA *rsa, int padding));
-int RSA_meth_set_finish(RSA_METHOD *meth, int (*finish) (RSA *rsa));
-void RSA_meth_free(RSA_METHOD *meth);
+static int RSA_meth_set_priv_enc(RSA_METHOD *meth, int (*priv_enc) (int flen, const unsigned char *from, unsigned char *to, RSA *rsa, int padding));
+static int RSA_meth_set_priv_dec(RSA_METHOD *meth, int (*priv_dec) (int flen, const unsigned char *from, unsigned char *to, RSA *rsa, int padding));
+static int RSA_meth_set_finish(RSA_METHOD *meth, int (*finish) (RSA *rsa));
+static void RSA_meth_free(RSA_METHOD *meth);
@@ -53 +53 @@ void RSA_meth_free(RSA_METHOD *meth);
-int RSA_bits(const RSA *r);
+static int RSA_bits(const RSA *r);
@@ -55 +55 @@ int RSA_bits(const RSA *r);
-RSA *EVP_PKEY_get0_RSA(EVP_PKEY *pkey);
+static RSA *EVP_PKEY_get0_RSA(EVP_PKEY *pkey);
@@ -57 +57 @@ RSA *EVP_PKEY_get0_RSA(EVP_PKEY *pkey);
-int X509_NAME_get0_der(X509_NAME *nm, const unsigned char **pder,
+static int X509_NAME_get0_der(X509_NAME *nm, const unsigned char **pder,
