--- contrib/python/prance/prance/util/url.py	(index)
+++ contrib/python/prance/prance/util/url.py	(working tree)
@@ -194,11 +194,7 @@ def fetch_url_text(url, cache={}, encoding=None):
 
         import pkg_resources
 
-        path = pkg_resources.resource_filename(package, path)
-
-        from .fs import read_file, from_posix
-
-        content = read_file(from_posix(path), encoding)
+        content = pkg_resources.resource_string(package, path).decode(encoding or 'utf-8')
     else:
         import requests
 
