--- contrib/python/svn/py2/svn/common.py	(index)
+++ contrib/python/svn/py2/svn/common.py	(working tree)
@@ -258,7 +258,7 @@ class CommonClient(svn.common_base.CommonBase):
 
         # Merge history can create nested log entries, so use iter instead of findall
         for e in root.iter('logentry'):
-            entry_info = {x.tag: x.text for x in e.getchildren()}
+            entry_info = {x.tag: x.text for x in e}
 
             date = None
             date_text = entry_info.get('date')
