--- contrib/python/pytest-allure-adaptor/py2/allure/pytest_plugin.py	(index)
+++ contrib/python/pytest-allure-adaptor/py2/allure/pytest_plugin.py	(working tree)
@@ -10,7 +10,7 @@ from allure.common import AllureImpl, StepContext
 from allure.constants import Status, AttachmentType, Severity, \
     FAILED_STATUSES, Label, SKIPPED_STATUSES
 from allure.utils import parent_module, parent_down_from_module, labels_of, \
-    all_of, get_exception_message, now, mangle_testnames
+    all_of, get_exception_message, now, mangle_testnames, marker_of_node
 from allure.structure import TestCase, TestStep, Attach, TestSuite, Failure, TestLabel
 
 
@@ -71,6 +71,16 @@ def pytest_addoption(parser):
 
 
 def pytest_configure(config):
+    pytest.allure = MASTER_HELPER
+
+    for label in (
+            Label.FEATURE, Label.STORY,
+            Label.SEVERITY, Label.ISSUE,
+            Label.TESTCASE, Label.THREAD,
+            Label.HOST, Label.FRAMEWORK,
+            Label.LANGUAGE):
+        config.addinivalue_line("markers", "%s.%s: this allure adaptor marker." % (Label.DEFAULT, label))
+
     reportdir = config.option.allurereportdir
 
     if reportdir:  # we actually record something
@@ -236,7 +246,7 @@ class AllureTestListener(object):
         """
         report = (yield).get_result()
 
-        status = self.config.hook.pytest_report_teststatus(report=report)
+        status = self.config.hook.pytest_report_teststatus(report=report, config=self.config)
         status = status and status[0]
 
         if report.when == 'call':
@@ -273,7 +283,7 @@ class AllureTestListener(object):
                     # TODO: think about that once again
                     self.test.status = Status.BROKEN
             # if a test isn't marked as "unreported" or it has failed, add it to the report.
-            if not item.get_marker("unreported") or self.test.status in FAILED_STATUSES:
+            if not marker_of_node(item, "unreported") or self.test.status in FAILED_STATUSES:
                 self.report_case(item, report)
 
 
@@ -461,10 +471,6 @@ class AllureHelper(object):
 MASTER_HELPER = AllureHelper()
 
 
-def pytest_namespace():
-    return {'allure': MASTER_HELPER}
-
-
 class AllureAgregatingListener(object):
 
     """
--- contrib/python/pytest-allure-adaptor/py2/allure/utils.py	(index)
+++ contrib/python/pytest-allure-adaptor/py2/allure/utils.py	(working tree)
@@ -82,7 +82,7 @@ def labels_of(item):
 
         markers = list()
         for suitable_name in suitable_names:
-            markers.append(item.get_marker(suitable_name))
+            markers.append(marker_of_node(item, suitable_name))
 
         return markers
 
@@ -181,3 +181,13 @@ def mangle_testnames(names):
     names = [x.replace(".py", "") for x in names if x != '()']
     names[0] = names[0].replace("/", '.')
     return names
+
+
+def marker_of_node(node, name):
+    """
+    Return marker of node by name
+    """
+    try:
+        return node.get_marker(name)
+    except AttributeError:
+        return node.get_closest_marker(name)
--- contrib/python/pytest-allure-adaptor/py2/.dist-info/METADATA	(index)
+++ contrib/python/pytest-allure-adaptor/py2/.dist-info/METADATA	(working tree)
@@ -7,7 +7,7 @@ Author: pupssman
 Author-email: pupssman@yandex-team.ru
 License: UNKNOWN
 Platform: UNKNOWN
-Requires-Dist: enum34
+Requires-Dist: enum34 ; python_version < "3.4"
 Requires-Dist: lxml (>=3.2.0)
 Requires-Dist: namedlist
 Requires-Dist: pytest (>=2.7.3)
--- contrib/python/pytest-allure-adaptor/py2/allure/utils.py	(index)
+++ contrib/python/pytest-allure-adaptor/py2/allure/utils.py	(working tree)
@@ -82,7 +82,8 @@ def labels_of(item):
 
         markers = list()
         for suitable_name in suitable_names:
-            markers.append(marker_of_node(item, suitable_name))
+            for marker in item.iter_markers(name=suitable_name):
+                markers.append(marker)
 
         return markers
 
