--- contrib/python/fissix/fissix/__init__.py	(index)
+++ contrib/python/fissix/fissix/__init__.py	(working tree)
@@ -23,13 +23,13 @@ CACHE_DIR = Path(user_cache_dir("fissix", version=__version__))
 
 
 def _generate_pickle_name(gt):
-    path = Path(gt)
+    path = Path(gt.name)
     filename = f"{path.stem}{__base_version__}.pickle"
     CACHE_DIR.mkdir(parents=True, exist_ok=True)
     return (CACHE_DIR / filename).as_posix()
 
 
-def load_grammar(gt="Grammar.txt", gp=None, save=True, force=False, logger=None):
+def load_grammar(gt="Grammar.txt", gp=None, save=False, force=True, logger=None):
     """Load the grammar (maybe from a pickle)."""
     if logger is None:
         logger = logging.getLogger()
--- contrib/python/fissix/fissix/pgen2/driver.py	(index)
+++ contrib/python/fissix/fissix/pgen2/driver.py	(working tree)
@@ -149,7 +149,7 @@ def load_packaged_grammar(package, grammar_source):
     but preserves load_grammar's automatic regeneration behavior when possible.
 
     """
-    if os.path.isfile(grammar_source):
+    if grammar_source.is_file():
         return load_grammar(grammar_source)
     pickled_name = _generate_pickle_name(os.path.basename(grammar_source))
     data = pkgutil.get_data(package, pickled_name)
--- contrib/python/fissix/fissix/pgen2/pgen.py	(index)
+++ contrib/python/fissix/fissix/pgen2/pgen.py	(working tree)
@@ -13,7 +13,7 @@ class ParserGenerator(object):
     def __init__(self, filename, stream=None):
         close_stream = None
         if stream is None:
-            stream = open(filename)
+            stream = filename.open()
             close_stream = stream.close
         self.filename = filename
         self.stream = stream
--- contrib/python/fissix/fissix/pygram.py	(index)
+++ contrib/python/fissix/fissix/pygram.py	(working tree)
@@ -12,8 +12,9 @@ from .pgen2 import driver
 from . import pytree
 
 # The grammar file
-_GRAMMAR_FILE = os.path.join(os.path.dirname(__file__), "Grammar.txt")
-_PATTERN_GRAMMAR_FILE = os.path.join(os.path.dirname(__file__), "PatternGrammar.txt")
+import importlib.resources
+_GRAMMAR_FILE = importlib.resources.files(__package__) / "Grammar.txt"
+_PATTERN_GRAMMAR_FILE = importlib.resources.files(__package__) / "PatternGrammar.txt"
 
 
 class Symbols(object):
--- contrib/python/fissix/fissix/__init__.py	(index)
+++ contrib/python/fissix/fissix/__init__.py	(working tree)
@@ -25,7 +25,7 @@ CACHE_DIR = Path(user_cache_dir("fissix", version=__version__))
 def _generate_pickle_name(gt):
     path = Path(gt.name)
     filename = f"{path.stem}{__base_version__}.pickle"
-    CACHE_DIR.mkdir(parents=True, exist_ok=True)
+    #CACHE_DIR.mkdir(parents=True, exist_ok=True)
     return (CACHE_DIR / filename).as_posix()
 
 
