--- contrib/python/lz4/py3/tests/block/conftest.py	(index)
+++ contrib/python/lz4/py3/tests/block/conftest.py	(working tree)
@@ -9,7 +9,7 @@ test_data = [
     (b'0' * 8 * 1024),
     (bytearray(b'')),
     (bytearray(os.urandom(8 * 1024))),
-    (bytearray(open(os.path.join(os.path.dirname(__file__), 'numpy_byte_array.bin'), 'rb').read()))
+    #(bytearray(open(os.path.join(os.path.dirname(__file__), 'numpy_byte_array.bin'), 'rb').read()))
 ]
 
 if sys.version_info > (2, 7):
--- contrib/python/lz4/py3/tests/stream/conftest.py	(index)
+++ contrib/python/lz4/py3/tests/stream/conftest.py	(working tree)
@@ -8,7 +8,7 @@ test_data = [
     # (b'0' * 8 * 1024),
     # (bytearray(b'')),
     # (bytearray(os.urandom(8 * 1024))),
-    (bytearray(open(os.path.join(os.path.dirname(__file__), 'numpy_byte_array.bin'), 'rb').read()))
+    #(bytearray(open(os.path.join(os.path.dirname(__file__), 'numpy_byte_array.bin'), 'rb').read()))
 ]
 
 if sys.version_info > (2, 7):
