This patch is similar with ca2a46f290b28eec6672292fbf347f6d192c5a1c

--- contrib/restricted/boost/unordered/include/boost/unordered/detail/implementation.hpp        (index)
+++ contrib/restricted/boost/unordered/include/boost/unordered/detail/implementation.hpp        (working tree)
@@ -26,7 +26,7 @@
 #include <boost/preprocessor/repetition/repeat_from_to.hpp>
 #include <boost/preprocessor/seq/enum.hpp>
 #include <boost/preprocessor/seq/size.hpp>
-#include <boost/swap.hpp>
+#include <boost/core/swap.hpp>
 #include <boost/throw_exception.hpp>
 #include <boost/tuple/tuple.hpp>
 #include <boost/type_traits/add_lvalue_reference.hpp>