commit 96742bfd3fc9c3d711da9d1ebbfb3efa21198b6c (HEAD -> murmurhash-patch)
author: thegeorg
date: 2021-06-18T11:27:17+03:00

    Make length size_t

--- a/MurmurHash1.cpp
+++ b/MurmurHash1.cpp
@@ -17,7 +17,7 @@
 
 //-----------------------------------------------------------------------------
 
-uint32_t MurmurHash1 ( const void * key, int len, uint32_t seed )
+uint32_t MurmurHash1 ( const void * key, size_t len, uint32_t seed )
 {
   const unsigned int m = 0xc6a4a793;
 
@@ -73,7 +73,7 @@ uint32_t MurmurHash1 ( const void * key, int len, uint32_t seed )
 
 // Performance should be equal to or better than the simple version.
 
-unsigned int MurmurHash1Aligned ( const void * key, int len, unsigned int seed )
+unsigned int MurmurHash1Aligned ( const void * key, size_t len, unsigned int seed )
 {
   const unsigned int m = 0xc6a4a793;
   const int r = 16;
@@ -82,7 +82,7 @@ unsigned int MurmurHash1Aligned ( const void * key, int len, unsigned int seed )
 
   unsigned int h = seed ^ (len * m);
 
-  int align = (uint64_t)data & 3;
+  size_t align = (uint64_t)data & 3;
 
   if(align && (len >= 4))
   {
@@ -122,7 +122,7 @@ unsigned int MurmurHash1Aligned ( const void * key, int len, unsigned int seed )
 
     // Handle leftover data in temp registers
 
-    int pack = len < align ? len : align;
+    size_t pack = len < align ? len : align;
 
     d = 0;
 
--- a/MurmurHash1.h
+++ b/MurmurHash1.h
@@ -5,6 +5,8 @@
 #ifndef _MURMURHASH1_H_
 #define _MURMURHASH1_H_
 
+#include <stddef.h>
+
 //-----------------------------------------------------------------------------
 // Platform-specific functions and macros
 
@@ -26,8 +28,8 @@ typedef unsigned __int64 uint64_t;
 
 //-----------------------------------------------------------------------------
 
-uint32_t MurmurHash1        ( const void * key, int len, uint32_t seed );
-uint32_t MurmurHash1Aligned ( const void * key, int len, uint32_t seed );
+uint32_t MurmurHash1        ( const void * key, size_t len, uint32_t seed );
+uint32_t MurmurHash1Aligned ( const void * key, size_t len, uint32_t seed );
 
 //-----------------------------------------------------------------------------
 
--- a/MurmurHash2.cpp
+++ b/MurmurHash2.cpp
@@ -34,7 +34,7 @@
 
 //-----------------------------------------------------------------------------
 
-uint32_t MurmurHash2 ( const void * key, int len, uint32_t seed )
+uint32_t MurmurHash2 ( const void * key, size_t len, uint32_t seed )
 {
   // 'm' and 'r' are mixing constants generated offline.
   // They're not really 'magic', they just happen to work well.
@@ -93,7 +93,7 @@ uint32_t MurmurHash2 ( const void * key, int len, uint32_t seed )
 
 // 64-bit hash for 64-bit platforms
 
-uint64_t MurmurHash64A ( const void * key, int len, uint64_t seed )
+uint64_t MurmurHash64A ( const void * key, size_t len, uint64_t seed )
 {
   const uint64_t m = BIG_CONSTANT(0xc6a4a7935bd1e995);
   const int r = 47;
@@ -139,7 +139,7 @@ uint64_t MurmurHash64A ( const void * key, int len, uint64_t seed )
 
 // 64-bit hash for 32-bit platforms
 
-uint64_t MurmurHash64B ( const void * key, int len, uint64_t seed )
+uint64_t MurmurHash64B ( const void * key, size_t len, uint64_t seed )
 {
   const uint32_t m = 0x5bd1e995;
   const int r = 24;
@@ -203,7 +203,7 @@ uint64_t MurmurHash64B ( const void * key, int len, uint64_t seed )
 
 #define mmix(h,k) { k *= m; k ^= k >> r; k *= m; h *= m; h ^= k; }
 
-uint32_t MurmurHash2A ( const void * key, int len, uint32_t seed )
+uint32_t MurmurHash2A ( const void * key, size_t len, uint32_t seed )
 {
   const uint32_t m = 0x5bd1e995;
   const int r = 24;
@@ -270,7 +270,7 @@ public:
     m_size  = 0;
   }
 
-  void Add ( const unsigned char * data, int len )
+  void Add ( const unsigned char * data, size_t len )
   {
     m_size += len;
 
@@ -306,7 +306,7 @@ private:
   static const uint32_t m = 0x5bd1e995;
   static const int r = 24;
 
-  void MixTail ( const unsigned char * & data, int & len )
+  void MixTail ( const unsigned char * & data, size_t & len )
   {
     while( len && ((len<4) || m_count) )
     {
@@ -336,7 +336,7 @@ private:
 // Same as MurmurHash2, but endian- and alignment-neutral.
 // Half the speed though, alas.
 
-uint32_t MurmurHashNeutral2 ( const void * key, int len, uint32_t seed )
+uint32_t MurmurHashNeutral2 ( const void * key, size_t len, uint32_t seed )
 {
   const uint32_t m = 0x5bd1e995;
   const int r = 24;
@@ -391,7 +391,7 @@ uint32_t MurmurHashNeutral2 ( const void * key, int len, uint32_t seed )
 #define MIX(h,k,m) { k *= m; k ^= k >> r; k *= m; h *= m; h ^= k; }
 
 
-uint32_t MurmurHashAligned2 ( const void * key, int len, uint32_t seed )
+uint32_t MurmurHashAligned2 ( const void * key, size_t len, uint32_t seed )
 {
   const uint32_t m = 0x5bd1e995;
   const int r = 24;
@@ -400,7 +400,7 @@ uint32_t MurmurHashAligned2 ( const void * key, int len, uint32_t seed )
 
   uint32_t h = seed ^ len;
 
-  int align = (uint64_t)data & 3;
+  size_t align = (uint64_t)data & 3;
 
   if(align && (len >= 4))
   {
--- a/MurmurHash2.h
+++ b/MurmurHash2.h
@@ -5,6 +5,8 @@
 #ifndef _MURMURHASH2_H_
 #define _MURMURHASH2_H_
 
+#include <stddef.h>
+
 //-----------------------------------------------------------------------------
 // Platform-specific functions and macros
 
@@ -26,12 +28,12 @@ typedef unsigned __int64 uint64_t;
 
 //-----------------------------------------------------------------------------
 
-uint32_t MurmurHash2        ( const void * key, int len, uint32_t seed );
-uint64_t MurmurHash64A      ( const void * key, int len, uint64_t seed );
-uint64_t MurmurHash64B      ( const void * key, int len, uint64_t seed );
-uint32_t MurmurHash2A       ( const void * key, int len, uint32_t seed );
-uint32_t MurmurHashNeutral2 ( const void * key, int len, uint32_t seed );
-uint32_t MurmurHashAligned2 ( const void * key, int len, uint32_t seed );
+uint32_t MurmurHash2        ( const void * key, size_t len, uint32_t seed );
+uint64_t MurmurHash64A      ( const void * key, size_t len, uint64_t seed );
+uint64_t MurmurHash64B      ( const void * key, size_t len, uint64_t seed );
+uint32_t MurmurHash2A       ( const void * key, size_t len, uint32_t seed );
+uint32_t MurmurHashNeutral2 ( const void * key, size_t len, uint32_t seed );
+uint32_t MurmurHashAligned2 ( const void * key, size_t len, uint32_t seed );
 
 //-----------------------------------------------------------------------------
 
--- a/MurmurHash3.cpp
+++ b/MurmurHash3.cpp
@@ -94,7 +94,7 @@ FORCE_INLINE uint64_t fmix64 ( uint64_t k )
 
 //-----------------------------------------------------------------------------
 
-void MurmurHash3_x86_32 ( const void * key, int len,
+void MurmurHash3_x86_32 ( const void * key, size_t len,
                           uint32_t seed, void * out )
 {
   const uint8_t * data = (const uint8_t*)key;
@@ -150,7 +150,7 @@ void MurmurHash3_x86_32 ( const void * key, int len,
 
 //-----------------------------------------------------------------------------
 
-void MurmurHash3_x86_128 ( const void * key, const int len,
+void MurmurHash3_x86_128 ( const void * key, const size_t len,
                            uint32_t seed, void * out )
 {
   const uint8_t * data = (const uint8_t*)key;
@@ -255,7 +255,7 @@ void MurmurHash3_x86_128 ( const void * key, const int len,
 
 //-----------------------------------------------------------------------------
 
-void MurmurHash3_x64_128 ( const void * key, const int len,
+void MurmurHash3_x64_128 ( const void * key, const size_t len,
                            const uint32_t seed, void * out )
 {
   const uint8_t * data = (const uint8_t*)key;
--- a/MurmurHash3.h
+++ b/MurmurHash3.h
@@ -5,6 +5,8 @@
 #ifndef _MURMURHASH3_H_
 #define _MURMURHASH3_H_
 
+#include <stddef.h>
+
 //-----------------------------------------------------------------------------
 // Platform-specific functions and macros
 
@@ -26,11 +28,11 @@ typedef unsigned __int64 uint64_t;
 
 //-----------------------------------------------------------------------------
 
-void MurmurHash3_x86_32  ( const void * key, int len, uint32_t seed, void * out );
+void MurmurHash3_x86_32  ( const void * key, size_t len, uint32_t seed, void * out );
 
-void MurmurHash3_x86_128 ( const void * key, int len, uint32_t seed, void * out );
+void MurmurHash3_x86_128 ( const void * key, size_t len, uint32_t seed, void * out );
 
-void MurmurHash3_x64_128 ( const void * key, int len, uint32_t seed, void * out );
+void MurmurHash3_x64_128 ( const void * key, size_t len, uint32_t seed, void * out );
 
 //-----------------------------------------------------------------------------
 
