commit db1177ad78c9e6290502e8a429ab14c766404810 (HEAD -> murmurhash-patch)
author: thegeorg
date: 2021-06-18T11:21:56+03:00

    Fix unaligned read in MurmurHash3.cpp

--- a/MurmurHash3.cpp
+++ b/MurmurHash3.cpp
@@ -8,6 +8,7 @@
 // non-native version will be less than optimal.
 
 #include "MurmurHash3.h"
+#include <string.h>
 
 //-----------------------------------------------------------------------------
 // Platform-specific functions and macros
@@ -54,12 +55,16 @@ inline uint64_t rotl64 ( uint64_t x, int8_t r )
 
 FORCE_INLINE uint32_t getblock32 ( const uint32_t * p, int i )
 {
-  return p[i];
+  uint32_t res;
+  memcpy(&res, p + i, sizeof(res));
+  return res;
 }

 FORCE_INLINE uint64_t getblock64 ( const uint64_t * p, int i )
 {
-  return p[i];
+  uint64_t res;
+  memcpy(&res, p + i, sizeof(res));
+  return res;
 }
 
 //-----------------------------------------------------------------------------
