commit 93d6babc4eb3c18b37b6a3288d531a7fe0bed7ec (HEAD -> murmurhash-cgo)
author: thegeorg
date: 2021-06-22T12:05:35+03:00

    Specify extern "C" for murmurhash to allow CGO bindings

--- a/MurmurHash1.h
+++ b/MurmurHash1.h
@@ -28,9 +28,17 @@ typedef unsigned __int64 uint64_t;
 
 //-----------------------------------------------------------------------------
 
+#ifdef __cplusplus
+extern "C" {
+#endif
+
 uint32_t MurmurHash1        ( const void * key, size_t len, uint32_t seed );
 uint32_t MurmurHash1Aligned ( const void * key, size_t len, uint32_t seed );
 
+#ifdef __cplusplus
+}
+#endif
+
 //-----------------------------------------------------------------------------
 
 #endif // _MURMURHASH1_H_
--- a/MurmurHash2.h
+++ b/MurmurHash2.h
@@ -28,12 +28,20 @@ typedef unsigned __int64 uint64_t;
 
 //-----------------------------------------------------------------------------
 
+#ifdef __cplusplus
+extern "C" {
+#endif
+
 uint32_t MurmurHash2        ( const void * key, size_t len, uint32_t seed );
 uint64_t MurmurHash64A      ( const void * key, size_t len, uint64_t seed );
 uint64_t MurmurHash64B      ( const void * key, size_t len, uint64_t seed );
 uint32_t MurmurHash2A       ( const void * key, size_t len, uint32_t seed );
 uint32_t MurmurHashNeutral2 ( const void * key, size_t len, uint32_t seed );
 uint32_t MurmurHashAligned2 ( const void * key, size_t len, uint32_t seed );
+ 
+#ifdef __cplusplus
+}
+#endif
 
 //-----------------------------------------------------------------------------
 
--- a/MurmurHash3.h
+++ b/MurmurHash3.h
@@ -28,12 +28,20 @@ typedef unsigned __int64 uint64_t;
 
 //-----------------------------------------------------------------------------
 
+#ifdef __cplusplus
+extern "C" {
+#endif
+
 void MurmurHash3_x86_32  ( const void * key, size_t len, uint32_t seed, void * out );
 
 void MurmurHash3_x86_128 ( const void * key, size_t len, uint32_t seed, void * out );
 
 void MurmurHash3_x64_128 ( const void * key, size_t len, uint32_t seed, void * out );
 
+#ifdef __cplusplus
+}
+#endif
+
 //-----------------------------------------------------------------------------
 
 #endif // _MURMURHASH3_H_
