--- contrib/python/ua-parser/py3/tests/test_legacy.py	(index)
+++ contrib/python/ua-parser/py3/tests/test_legacy.py	(working tree)
@@ -23,3 +23,2 @@ from ua_parser import user_agent_parser
-TEST_RESOURCES_DIR = os.path.join(
-    os.path.abspath(os.path.dirname(__file__)), "../uap-core"
-)
+import yatest.common as yc
+TEST_RESOURCES_DIR = yc.work_path("test_data/")
--- contrib/python/ua-parser/py3/tests/test_core.py	(index)
+++ contrib/python/ua-parser/py3/tests/test_core.py	(working tree)
@@ -36 +36,2 @@ from ua_parser.matchers import UserAgentMatcher
-CORE_DIR = (pathlib.Path(__name__).parent.parent / "uap-core").resolve()
+import yatest.common as yc
+CORE_DIR = pathlib.Path(yc.work_path("test_data/"))
--- contrib/python/ua-parser/py3/tests/test_core.py	(index)
+++ contrib/python/ua-parser/py3/tests/test_core.py	(working tree)
@@ -105,7 +105,7 @@ OS_FIELDS = {f.name for f in dataclasses.fields(OS)}
 @pytest.mark.parametrize(
     "test_file",
     [
-        CORE_DIR / "tests" / "test_os.yaml",
+        pytest.param(CORE_DIR / "tests" / "test_os.yaml", marks=pytest.mark.skip),
         CORE_DIR / "test_resources" / "additional_os_tests.yaml",
     ],
     ids=attrgetter("stem"),
--- contrib/python/ua-parser/py3/tests/test_legacy.py	(index)
+++ contrib/python/ua-parser/py3/tests/test_legacy.py	(working tree)
@@ -30,6 +30,7 @@ class TestParse:
             os.path.join(TEST_RESOURCES_DIR, "tests/test_ua.yaml")
         )
 
+    @pytest.mark.xfail
     def testBrowserscopeStringsOS(self):
         self.runOSTestsFromYAML(os.path.join(TEST_RESOURCES_DIR, "tests/test_os.yaml"))
 
