--- contrib/python/implicit/implicit/bpr.pyx	(index)
+++ contrib/python/implicit/implicit/bpr.pyx	(working tree)
@@ -26,7 +26,7 @@ log = logging.getLogger("implicit")
 
 # thin wrapper around omp_get_thread_num (since referencing directly will cause OSX
 # build to fail)
-cdef extern from "bpr.h" namespace "implicit" nogil:
+cdef extern from "contrib/python/implicit/implicit/bpr.h" namespace "implicit" nogil:
     cdef int get_thread_num()
 
 
--- contrib/python/implicit/implicit/cuda/_cuda.pyx	(index)
+++ contrib/python/implicit/implicit/cuda/_cuda.pyx	(working tree)
@@ -31,7 +31,7 @@ cdef extern from "als.h" namespace "implicit" nogil:
                              const CudaDenseMatrix & Y, float regularization) except +
 
 
-cdef extern from "bpr.h" namespace "implicit" nogil:
+cdef extern from "contrib/python/implicit/implicit/cuda/bpr.h" namespace "implicit" nogil:
     cdef pair[int, int] bpr_update(const CudaVector[int] & userids,
                                    const CudaVector[int] & itemids,
                                    const CudaVector[int] & indptr,
