diff --git a/include/version b/include/version
index 4fdfec4..cbdd87a 100644
--- a/include/version
+++ b/include/version
@@ -263,7 +263,11 @@ __cpp_lib_within_lifetime                               202306L <type_traits>
 # define __cpp_lib_allocator_traits_is_always_equal     201411L
 # define __cpp_lib_any                                  201606L
 # define __cpp_lib_apply                                201603L
-# define __cpp_lib_array_constexpr                      201603L
+# if defined(_MSC_VER) && _MSC_VER >= 1926
+#   define __cpp_lib_array_constexpr                    201803L
+# else
+#   define __cpp_lib_array_constexpr                    201603L
+# endif
 # define __cpp_lib_as_const                             201510L
 # define __cpp_lib_atomic_is_always_lock_free           201603L
 # define __cpp_lib_bool_constant                        201505L
@@ -308,7 +312,11 @@ __cpp_lib_within_lifetime                               202306L <type_traits>
 # endif
 # define __cpp_lib_shared_ptr_arrays                    201611L
 # define __cpp_lib_shared_ptr_weak_type                 201606L
-# define __cpp_lib_string_view                          201606L
+# if defined(_MSC_VER) && _MSC_VER >= 1926
+#   define __cpp_lib_string_view                        201803L
+# else
+#   define __cpp_lib_string_view                        201606L
+# endif
 // # define __cpp_lib_to_chars                             201611L
 # undef  __cpp_lib_transparent_operators
 # define __cpp_lib_transparent_operators                201510L
