diff --git a/include/__random/binomial_distribution.h b/include/__random/binomial_distribution.h
index 18053c7..7da71ad 100644
--- a/include/__random/binomial_distribution.h
+++ b/include/__random/binomial_distribution.h
@@ -114,10 +114,6 @@ public:
         {return !(__x == __y);}
 };
 
-#ifndef _LIBCPP_MSVCRT_LIKE
-extern "C" double lgamma_r(double, int *);
-#endif
-
 inline _LIBCPP_INLINE_VISIBILITY double __libcpp_lgamma(double __d) {
 #if defined(_LIBCPP_MSVCRT_LIKE)
   return lgamma(__d);
