--- contrib/python/fallocate/py2/ya.make	(index)
+++ contrib/python/fallocate/py2/ya.make	(working tree)
@@ -12,12 +12,19 @@ NO_COMPILER_WARNINGS()
 
 NO_LINT()
 
-CFLAGS(
-    -DHAVE_FALLOCATE=1
-    -DHAVE_POSIX_FALLOCATE=1
-    -DHAVE_POSIX_FADVISE=1
-    -DHAVE_APPLE_F_ALLOCATECONTIG=1
-)
+IF (OS_LINUX)
+    CFLAGS(
+        -DHAVE_FALLOCATE=1
+        -DHAVE_POSIX_FALLOCATE=1
+        -DHAVE_POSIX_FADVISE=1
+    )
+ENDIF()
+
+IF (OS_DARWIN)
+    CFLAGS(
+        -DHAVE_APPLE_F_ALLOCATECONTIG=1
+    )
+ENDIF()
 
 SRCS(
     fallocate/_fallocatemodule.c
