--- a/lark/load_grammar.py	(index)
+++ b/lark/load_grammar.py	(working tree)
@@ -989,7 +989,7 @@ class GrammarBuilder:
         if path_node.data == 'import_lib':  # Import from library
             base_path = None
         else:  # Relative import
-            if grammar_name == '<string>':  # Import relative to script file path if grammar is coded in script
+            if isinstance(grammar_name, str) and grammar_name == '<string>':  # Import relative to script file path if grammar is coded in script
                 try:
                     base_file = os.path.abspath(sys.modules['__main__'].__file__)
                 except AttributeError:
@@ -999,6 +999,8 @@ class GrammarBuilder:
             if base_file:
                 if isinstance(base_file, PackageResource):
                     base_path = PackageResource(base_file.pkg_name, os.path.split(base_file.path)[0])
+                elif not isinstance(base_file, str):
+                    base_path = base_file
                 else:
                     base_path = os.path.split(base_file)[0]
             else:
@@ -1096,9 +1098,14 @@ class GrammarBuilder:
                 if callable(source):
                     joined_path, text = source(base_path, grammar_path)
                 else:
-                    joined_path = os.path.join(source, grammar_path)
-                    with open(joined_path, encoding='utf8') as f:
-                        text = f.read()
+                    if isinstance(source, str):
+                        joined_path = os.path.join(source, grammar_path)
+                        with open(joined_path, encoding='utf8') as f:
+                            text = f.read()
+                    else:
+                        joined_path = source / grammar_path
+                        with joined_path.open(encoding='utf8') as f:
+                            text = f.read()
             except IOError:
                 continue
             else:
