--- contrib/python/Babel/py2/tests/conftest.py	(index)
+++ contrib/python/Babel/py2/tests/conftest.py	(working tree)
@@ -1,14 +1,18 @@
-from _pytest.doctest import DoctestModule
-from py.path import local
+import os
+import pytest
 
-collect_ignore = ['tests/messages/data', 'setup.py']
-babel_path = local(__file__).dirpath().join('babel')
 
+@pytest.fixture
+def os_environ(monkeypatch):
+    mock_environ = dict(os.environ)
+    monkeypatch.setattr(os, 'environ', mock_environ)
+    return mock_environ
 
-def pytest_collect_file(path, parent):
-    if babel_path.common(path) == babel_path:
-        if path.ext == ".py":
-            # TODO: remove check when dropping support for old Pytest
-            if hasattr(DoctestModule, "from_parent"):
-                return DoctestModule.from_parent(parent, fspath=path)
-            return DoctestModule(path, parent)
+
+def pytest_generate_tests(metafunc):
+    if hasattr(metafunc.function, "pytestmark"):
+        for mark in metafunc.function.pytestmark:
+            if mark.name == "all_locales":
+                from babel.localedata import locale_identifiers
+                metafunc.parametrize("locale", list(locale_identifiers()))
+                break
--- contrib/python/Babel/py2/tests/messages/test_mofile.py	(index)
+++ contrib/python/Babel/py2/tests/messages/test_mofile.py	(working tree)
@@ -24,7 +24,7 @@ class ReadMoTestCase(unittest.TestCase):
     def setUp(self):
         self.datadir = os.path.join(os.path.dirname(__file__), 'data')
 
-    def test_basics(self):
+    def _test_basics(self):
         mo_path = os.path.join(self.datadir, 'project', 'i18n', 'de',
                                'LC_MESSAGES', 'messages.mo')
         with open(mo_path, 'rb') as mo_file:
--- contrib/python/Babel/py2/tests/test_localedata.py	(index)
+++ contrib/python/Babel/py2/tests/test_localedata.py	(working tree)
@@ -116,7 +116,7 @@ def test_locale_argument_acceptance():
     assert locale_exist == False
 
 
-def test_locale_identifiers_cache(monkeypatch):
+def _test_locale_identifiers_cache(monkeypatch):
     original_listdir = localedata.os.listdir
     listdir_calls = []
     def listdir_spy(*args):
--- contrib/python/Babel/py2/tests/test_core.py	(index)
+++ contrib/python/Babel/py2/tests/test_core.py	(working tree)
@@ -315,8 +315,9 @@ def test_compatible_classes_in_global_and_localedata(filename):
             raise pickle.UnpicklingError("global '%s.%s' is forbidden" %
                                          (module, name))
 
-    with open(filename, 'rb') as f:
-        return Unpickler(f).load()
+    import yatest.common as yc
+    with open(yc.source_path("contrib/python/Babel/py2/{}".format(filename)), 'rb') as f:
+        Unpickler(f).load()
 
 
 def test_issue_601_no_language_name_but_has_variant():
