--- contrib/python/pystan/py3/pystan/experimental/pickling_tools.py	(index)
+++ contrib/python/pystan/py3/pystan/experimental/pickling_tools.py	(working tree)
@@ -1,4 +1,5 @@
 """Experimental module to load fit without model instance."""
+import setuptools
 from distutils.core import Extension
 from Cython.Build.Inline import _get_build_extension
 from Cython.Build.Dependencies import cythonize
--- contrib/python/pystan/py3/pystan/model.py	(index)
+++ contrib/python/pystan/py3/pystan/model.py	(working tree)
@@ -25,6 +25,7 @@ import sys
 import tempfile
 import time
 
+import setuptools
 import distutils
 from distutils.core import Extension
 
