--- a/contrib/deprecated/python/nose/py3/nose/core.py
+++ b/contrib/deprecated/python/nose/py3/nose/core.py
@@ -246,18 +246,8 @@ class TestProgram(unittest.TestProgram):
                 print()
 
     def usage(cls):
-        import nose
-        try:
-            ld = nose.__loader__
-            text = ld.get_data(os.path.join(
-                os.path.dirname(__file__), 'usage.txt'))
-        except AttributeError:
-            f = open(os.path.join(
-                os.path.dirname(__file__), 'usage.txt'), 'r')
-            try:
-                text = f.read()
-            finally:
-                f.close()
+        import importlib.resources
+        text = (importlib.resources.files(__package__) / "usage.txt").read_text()
         # Ensure that we return str, not bytes.
         if not isinstance(text, str):
             text = text.decode('utf-8')
