--- contrib/deprecated/python/nose/py3/nose/importer.py	(acf9565245292a289b5942646f51e8cea9039c2b)
+++ contrib/deprecated/python/nose/py3/nose/importer.py	(d5d0952dedbe5c6f8f946bf43bc9cfebb5e6388b)
@@ -9,8 +9,6 @@ import os
 import sys
 from nose.config import Config
 
-from imp import find_module, load_module, acquire_lock, release_lock
-
 log = logging.getLogger(__name__)
 
 try:
@@ -72,6 +70,7 @@ class Importer(object):
                 part_fqname = part
             else:
                 part_fqname = "%s.%s" % (part_fqname, part)
+            from imp import find_module, load_module, acquire_lock, release_lock
             try:
                 acquire_lock()
                 log.debug("find module part %s (%s) in %s",
--- contrib/deprecated/python/nose/py3/nose/result.py	(index)
+++ contrib/deprecated/python/nose/py3/nose/result.py	(working tree)
@@ -16 +16 @@ except ImportError:
-    from unittest import _TextTestResult
+    from unittest import TextTestResult as _TextTestResult
--- contrib/deprecated/python/nose/py3/nose/tools/trivial.py	(index)
+++ contrib/deprecated/python/nose/py3/nose/tools/trivial.py	(working tree)
@@ -49,6 +49,9 @@ for at in [ at for at in dir(_t)
     vars()[pepd] = getattr(_t, at)
     __all__.append(pepd)
 
+assert_raises_regexp = assert_raises_regex
+__all__.append('assert_raises_regexp')
+
 del Dummy
 del _t
 del pep8
