--- contrib/python/structlog/tests/test_frames.py	(index)
+++ contrib/python/structlog/tests/test_frames.py	(working tree)
@@ -18,6 +18,7 @@ from structlog._frames import (
 )
 
 
+@pytest.mark.skip
 class TestFindFirstAppFrameAndName:
     def test_ignores_structlog_by_default(self):
         """
--- contrib/python/structlog/tests/test_stdlib.py	(index)
+++ contrib/python/structlog/tests/test_stdlib.py	(working tree)
@@ -81,11 +81,12 @@ class TestLoggerFactory:
         The factory isn't called directly but from structlog._config so
         deducing has to be slightly smarter.
         """
-        assert "tests.additional_frame" == (
+        assert "__tests__.additional_frame" == (
             additional_frame(LoggerFactory()).name
         )
-        assert "tests.test_stdlib" == LoggerFactory()().name
+        assert "__tests__.test_stdlib" == LoggerFactory()().name
 
+    @pytest.mark.skip
     def test_ignores_frames(self):
         """
         The name guesser walks up the frames until it reaches a frame whose
@@ -114,1 +115,1 @@ class TestLoggerFactory:
-        assert file_name == os.path.realpath(__file__)
+        assert file_name == __file__
--- contrib/python/structlog/tests/test_output.py	(index)
+++ contrib/python/structlog/tests/test_output.py	(working tree)
@@ -255,6 +255,7 @@ class TestBytesLogger:
 
         assert b"hello" in sio.getvalue()
 
+    @pytest.mark.skip
     @pytest.mark.parametrize("file", [None, stdout.buffer, stderr.buffer])
     @pytest.mark.parametrize("proto", range(pickle.HIGHEST_PROTOCOL + 1))
     def test_pickle(self, file, proto):
--- contrib/python/structlog/tests/test_stdlib.py	(index)
+++ contrib/python/structlog/tests/test_stdlib.py	(working tree)
@@ -149,3 +149 @@ class TestLoggerFactory:
-        assert caplog.text.startswith(
-            "ERROR    tests.test_stdlib:test_stdlib.py"
-        )
+        assert "ERROR    __tests__.test_stdlib:test_stdlib.py" in caplog.text
--- contrib/python/structlog/tests/processors/test_processors.py	(index)
+++ contrib/python/structlog/tests/processors/test_processors.py	(working tree)
@@ -238,7 +238,7 @@ class TestStackInfoRenderer:
         """
         Filtering of names works.
         """
-        sir = StackInfoRenderer(["tests.additional_frame"])
+        sir = StackInfoRenderer(["__tests__.additional_frame"])
 
         ed = additional_frame(
             functools.partial(sir, None, None, {"stack_info": True})
@@ -342,7 +342,7 @@ class TestCallsiteParameterAdder:
         `additional_ignores` are ignored when determining the callsite.
         """
         test_message = "test message"
-        additional_ignores = ["tests.additional_frame"]
+        additional_ignores = ["__tests__.additional_frame"]
         processor = self.make_processor(None, additional_ignores)
         event_dict: EventDict = {"event": test_message}
 
