--- a/falcon/__init__.py
+++ b/falcon/__init__.py
@@ -77,6 +77,7 @@
 from falcon.util import IS_64_BITS
 from falcon.util import is_python_func
 from falcon.util import misc
+from falcon.util import parse_header
 from falcon.util import reader
 from falcon.util import runs_sync
 from falcon.util import secure_filename
--- a/falcon/asgi/multipart.py
+++ b/falcon/asgi/multipart.py
@@ -14,11 +14,10 @@
 
 """ASGI multipart form media handler components."""
 
-import cgi
-
 from falcon.asgi.reader import BufferedReader
 from falcon.errors import DelimiterError
 from falcon.media import multipart
+from falcon.util.mediatypes import parse_header
 
 _ALLOWED_CONTENT_HEADERS = multipart._ALLOWED_CONTENT_HEADERS
 _CRLF = multipart._CRLF
@@ -54,7 +53,7 @@ async def get_media(self):
         return self._media
 
     async def get_text(self):
-        content_type, options = cgi.parse_header(self.content_type)
+        content_type, options = parse_header(self.content_type)
         if content_type != 'text/plain':
             return None
 
--- a/falcon/media/multipart.py
+++ b/falcon/media/multipart.py
@@ -14,7 +14,6 @@
 
 """Multipart form media handler."""
 
-import cgi
 import re
 from urllib.parse import unquote_to_bytes
 
@@ -24,4 +23,5 @@
 from falcon.util import misc
 from falcon.util.deprecation import deprecated_args
+from falcon.util.mediatypes import parse_header
 
 
@@ -249,7 +249,7 @@ def get_text(self):
             str: The part decoded as a text string provided the part is
             encoded as ``text/plain``, ``None`` otherwise.
         """
-        content_type, options = cgi.parse_header(self.content_type)
+        content_type, options = parse_header(self.content_type)
         if content_type != 'text/plain':
             return None
 
@@ -275,7 +275,7 @@ def filename(self):
 
             if self._content_disposition is None:
                 value = self._headers.get(b'content-disposition', b'')
-                self._content_disposition = cgi.parse_header(value.decode())
+                self._content_disposition = parse_header(value.decode())
 
             _, params = self._content_disposition
 
@@ -311,7 +311,7 @@ def name(self):
 
             if self._content_disposition is None:
                 value = self._headers.get(b'content-disposition', b'')
-                self._content_disposition = cgi.parse_header(value.decode())
+                self._content_disposition = parse_header(value.decode())
 
             _, params = self._content_disposition
             self._name = params.get('name')
@@ -493,1 +493,1 @@ def __init__(self, parse_options=None):
-        _, options = cgi.parse_header(content_type)
+        _, options = parse_header(content_type)
--- a/falcon/testing/helpers.py
+++ b/falcon/testing/helpers.py
@@ -23,7 +23,6 @@
 """
 
 import asyncio
-import cgi
 from collections import defaultdict
 from collections import deque
 import contextlib
@@ -51,6 +50,7 @@
 from falcon.constants import SINGLETON_HEADERS
 import falcon.request
 from falcon.util import uri
+from falcon.util.mediatypes import parse_header
 
 # NOTE(kgriffs): Changed in 3.0 from 'curl/7.24.0 (x86_64-apple-darwin12.0)'
 DEFAULT_UA = 'falcon-client/' + falcon.__version__
@@ -802,7 +802,7 @@ def get_encoding_from_headers(headers):
     if not content_type:
         return None
 
-    content_type, params = cgi.parse_header(content_type)
+    content_type, params = parse_header(content_type)
 
     if 'charset' in params:
         return params['charset'].strip('\'"')
--- a/falcon/util/__init__.py
+++ b/falcon/util/__init__.py
@@ -29,2 +29,3 @@
 from falcon.util.deprecation import deprecated
+from falcon.util.mediatypes import parse_header
 from falcon.util.misc import code_to_http_status
--- /dev/null
+++ b/falcon/util/mediatypes.py
@@ -0,0 +1,89 @@
+# Copyright 2023-2024 by Vytautas Liuolia.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Media (aka MIME) type parsing and matching utilities."""
+
+import typing
+
+
+def _parse_header_old_stdlib(line):  # type: ignore
+    """Parse a Content-type like header.
+
+    Return the main content-type and a dictionary of options.
+
+    Note:
+        This method has been copied (almost) verbatim from CPython 3.8 stdlib.
+        It is slated for removal from the stdlib in 3.13.
+    """
+
+    def _parseparam(s):  # type: ignore
+        while s[:1] == ';':
+            s = s[1:]
+            end = s.find(';')
+            while end > 0 and (s.count('"', 0, end) - s.count('\\"', 0, end)) % 2:
+                end = s.find(';', end + 1)
+            if end < 0:
+                end = len(s)
+            f = s[:end]
+            yield f.strip()
+            s = s[end:]
+
+    parts = _parseparam(';' + line)
+    key = parts.__next__()
+    pdict = {}
+    for p in parts:
+        i = p.find('=')
+        if i >= 0:
+            name = p[:i].strip().lower()
+            value = p[i + 1 :].strip()
+            if len(value) >= 2 and value[0] == value[-1] == '"':
+                value = value[1:-1]
+                value = value.replace('\\\\', '\\').replace('\\"', '"')
+            pdict[name] = value
+    return key, pdict
+
+
+def parse_header(line: str) -> typing.Tuple[str, dict]:
+    """Parse a Content-type like header.
+
+    Return the main content-type and a dictionary of options.
+
+    Args:
+        line: A header value to parse.
+
+    Returns:
+        A tuple containing the main content-type and a dictionary of options.
+
+    Note:
+        This function replaces an equivalent method previously available in the
+        stdlib as ``cgi.parse_header()``.
+        It was removed from the stdlib in Python 3.13.
+    """
+    if '"' not in line and '\\' not in line:
+        key, semicolon, parts = line.partition(';')
+        if not semicolon:
+            return (key.strip(), {})
+
+        pdict = {}
+        for part in parts.split(';'):
+            name, equals, value = part.partition('=')
+            if equals:
+                pdict[name.strip().lower()] = value.strip()
+
+        return (key.strip(), pdict)
+
+    return _parse_header_old_stdlib(line)
+
+
+__all__ = ['parse_header']
--- /dev/null
+++ b/tests/test_mediatypes.py
@@ -0,0 +1,33 @@
+import pytest
+
+from falcon.util import mediatypes
+
+
+@pytest.mark.parametrize(
+    'value,expected',
+    [
+        ('', ('', {})),
+        ('strange', ('strange', {})),
+        ('text/plain', ('text/plain', {})),
+        ('text/plain ', ('text/plain', {})),
+        (' text/plain', ('text/plain', {})),
+        (' text/plain ', ('text/plain', {})),
+        ('   text/plain   ', ('text/plain', {})),
+        (
+            'falcon/peregrine;  key1; key2=value; key3',
+            ('falcon/peregrine', {'key2': 'value'}),
+        ),
+        ('"falcon/peregrine" ; key="value"', ('"falcon/peregrine"', {'key': 'value'})),
+        ('falcon/peregrine; empty=""', ('falcon/peregrine', {'empty': ''})),
+        ('falcon/peregrine; quote="', ('falcon/peregrine', {'quote': '"'})),
+        ('text/plain; charset=utf-8', ('text/plain', {'charset': 'utf-8'})),
+        ('stuff/strange; missing-value; missing-another', ('stuff/strange', {})),
+        ('stuff/strange; missing-value\\missing-another', ('stuff/strange', {})),
+        (
+            'application/falcon; P1 = "key; value"; P2="\\""',
+            ('application/falcon', {'p1': 'key; value', 'p2': '"'}),
+        ),
+    ],
+)
+def test_parse_header(value, expected):
+    assert mediatypes.parse_header(value) == expected
--- a/falcon/util/mediatypes.py
+++ b/falcon/util/mediatypes.py
@@ -17,6 +17,19 @@
 import typing
 
 
+def _parse_param_old_stdlib(s):  # type: ignore
+    while s[:1] == ';':
+        s = s[1:]
+        end = s.find(';')
+        while end > 0 and (s.count('"', 0, end) - s.count('\\"', 0, end)) % 2:
+            end = s.find(';', end + 1)
+        if end < 0:
+            end = len(s)
+        f = s[:end]
+        yield f.strip()
+        s = s[end:]
+
+
 def _parse_header_old_stdlib(line):  # type: ignore
     """Parse a Content-type like header.
 
@@ -26,20 +39,7 @@ def _parse_header_old_stdlib(line):  # type: ignore
         This method has been copied (almost) verbatim from CPython 3.8 stdlib.
         It is slated for removal from the stdlib in 3.13.
     """
-
-    def _parseparam(s):  # type: ignore
-        while s[:1] == ';':
-            s = s[1:]
-            end = s.find(';')
-            while end > 0 and (s.count('"', 0, end) - s.count('\\"', 0, end)) % 2:
-                end = s.find(';', end + 1)
-            if end < 0:
-                end = len(s)
-            f = s[:end]
-            yield f.strip()
-            s = s[end:]
-
-    parts = _parseparam(';' + line)
+    parts = _parse_param_old_stdlib(';' + line)
     key = parts.__next__()
     pdict = {}
     for p in parts:
@@ -63,7 +63,7 @@ def parse_header(line: str) -> typing.Tuple[str, dict]:
         line: A header value to parse.
 
     Returns:
-        A tuple containing the main content-type and a dictionary of options.
+        tuple: (the main content-type, dictionary of options).
 
     Note:
         This function replaces an equivalent method previously available in the
--- a/tests/test_mediatypes.py
+++ b/tests/test_mediatypes.py
@@ -17,6 +17,14 @@
             'falcon/peregrine;  key1; key2=value; key3',
             ('falcon/peregrine', {'key2': 'value'}),
         ),
+        (
+            'audio/pcm;rate=48000;encoding=float;bits=32',
+            ('audio/pcm', {'bits': '32', 'encoding': 'float', 'rate': '48000'}),
+        ),
+        (
+            'falcon/*; genus=falco; family=falconidae; class=aves; ',
+            ('falcon/*', {'class': 'aves', 'family': 'falconidae', 'genus': 'falco'}),
+        ),
         ('"falcon/peregrine" ; key="value"', ('"falcon/peregrine"', {'key': 'value'})),
         ('falcon/peregrine; empty=""', ('falcon/peregrine', {'empty': ''})),
         ('falcon/peregrine; quote="', ('falcon/peregrine', {'quote': '"'})),
--- a/ya.make	(index)
+++ b/ya.make	(working tree)
@@ -75,6 +75,7 @@ PY_SRCS(
     falcon/uri.py
     falcon/util/__init__.py
     falcon/util/deprecation.py
+    falcon/util/mediatypes.py
     falcon/util/misc.py
     falcon/util/reader.py
     falcon/util/structures.py
