--- contrib/python/h3/tests/test_cells_and_edges.py	(index)
+++ contrib/python/h3/tests/test_cells_and_edges.py	(working tree)
@@ -203,7 +203,7 @@ def test_distance_error():
         h3.h3_distance(h1, h2)
 
 
-def test_compact():
+def test_compact(in_test=False):
 
     # lat/lngs for State of Maine
     maine = [
@@ -238,12 +238,13 @@ def test_compact():
 
     assert h_comp == expected
 
-    return h_uncomp, h_comp, res
+    if in_test:
+        return h_uncomp, h_comp, res
 
 
 def test_uncompact():
 
-    h_uncomp, h_comp, res = test_compact()
+    h_uncomp, h_comp, res = test_compact(True)
 
     out = h3.uncompact(h_comp, res)
 
