--- a/contrib/python/arrow/py2/arrow/arrow.py	(revision abc18ee9cbaa457e4d5ee56e98b946fd66e22426)
+++ b/contrib/python/arrow/py2/arrow/arrow.py	(revision cc4992a32929491047c296e6bc3e8ad79378068f)
@@ -21,7 +21,7 @@
 
 if sys.version_info[:2] < (3, 6):  # pragma: no cover
     with warnings.catch_warnings():
-        warnings.simplefilter("default", DeprecationWarning)
+        warnings.simplefilter("default", DeprecationWarning, append=1)
         warnings.warn(
             "Arrow will drop support for Python 2.7 and 3.5 in the upcoming v1.0.0 release. Please upgrade to "
             "Python 3.6+ to continue receiving updates for Arrow.",
