--- contrib/python/spintop-openhtf/openhtf/output/servers/web_gui_server.py	(index)
+++ contrib/python/spintop-openhtf/openhtf/output/servers/web_gui_server.py	(working tree)
@@ -24,6 +24,8 @@ import tornado.httpserver
 import tornado.ioloop
 import tornado.netutil
 import tornado.web
+import io
+from library.python.resource import resfs_read, resfs_files
 
 _SERVER_SHUTDOWN_BUFFER_S = 0.5
 
@@ -103,6 +105,26 @@ class StaticFileHandler(tornado.web.StaticFileHandler):
     return abspath
 
 
+class YamakeFileHandler(tornado.web.StaticFileHandler):
+
+  @classmethod
+  def get_content(cls, abspath, start=None, end=None):
+    return resfs_read(abspath)
+
+  def get_content_size(self):
+    return len(resfs_read(self.absolute_path))
+
+  def get_modified_time(self):
+    return None
+
+  @classmethod
+  def get_absolute_path(cls, root, path):
+    return os.path.join(root, path)
+
+  def validate_absolute_path(self, root, abspath):
+    return abspath
+
+
 class TemplateLoader(tornado.template.Loader):
 
   def __init__(self, root_directory, **kwargs):
@@ -112,6 +134,17 @@ class TemplateLoader(tornado.template.Loader):
     return name
 
 
+class YamakeTemplateLoader(tornado.template.DictLoader):
+
+  def __init__(self, root_directory, **kwargs):
+    files_dict = {
+      os.path.relpath(path, root_directory): resfs_read(path) 
+      for path in resfs_files() 
+      if root_directory in path
+    }
+    super(YamakeTemplateLoader, self).__init__(dict=files_dict, **kwargs)
+
+
 class WebGuiServer(threading.Thread):
   """Serves the OpenHTF Angular frontend."""
 
@@ -124,9 +157,13 @@ class WebGuiServer(threading.Thread):
     # Set up routes.
     routes = [
         ('/', IndexHandler.with_config(self._get_config())),
-        (STATIC_FILE_ROUTES, StaticFileHandler, {
+        (
+          STATIC_FILE_ROUTES, 
+          (StaticFileHandler if os.path.exists(static_files_root) else YamakeFileHandler), 
+          {
             'path': static_files_root
-        }),
+          }
+        ),
     ]
     routes.extend(additional_routes)
 
@@ -142,7 +179,11 @@ class WebGuiServer(threading.Thread):
     self.application = tornado.web.Application(
         routes,
         default_handler_class=DefaultHandler,
-        template_loader=TemplateLoader(static_files_root),
+        template_loader=(
+          TemplateLoader(static_files_root)
+          if os.path.exists(static_files_root)
+          else YamakeTemplateLoader(static_files_root)
+        ),
         static_path=static_files_root,)
     self.server = tornado.httpserver.HTTPServer(self.application)
     self.server.add_sockets(sockets)
