diff --git a/include/librsvg-2.0/librsvg/rsvg-cairo.h b/include/librsvg-2.0/librsvg/rsvg-cairo.h
new file mode 100644
index 0000000..750ef16
--- /dev/null
+++ b/include/librsvg-2.0/librsvg/rsvg-cairo.h
@@ -0,0 +1 @@
+#include "../../librsvg/rsvg-cairo.h" /* generated while patching */
diff --git a/include/librsvg-2.0/librsvg/rsvg-features.h b/include/librsvg-2.0/librsvg/rsvg-features.h
new file mode 100644
index 0000000..1d2ab95
--- /dev/null
+++ b/include/librsvg-2.0/librsvg/rsvg-features.h
@@ -0,0 +1 @@
+#include "../../librsvg/rsvg-features.h" /* generated while patching */
diff --git a/include/librsvg-2.0/librsvg/rsvg-version.h b/include/librsvg-2.0/librsvg/rsvg-version.h
new file mode 100644
index 0000000..3c4db40
--- /dev/null
+++ b/include/librsvg-2.0/librsvg/rsvg-version.h
@@ -0,0 +1 @@
+#include "../../librsvg/rsvg-version.h" /* generated while patching */
diff --git a/include/librsvg-2.0/librsvg/rsvg-version.h.in b/include/librsvg-2.0/librsvg/rsvg-version.h.in
new file mode 100644
index 0000000..1fba33b
--- /dev/null
+++ b/include/librsvg-2.0/librsvg/rsvg-version.h.in
@@ -0,0 +1 @@
+#include "../../librsvg/rsvg-version.h.in" /* generated while patching */
diff --git a/include/librsvg-2.0/librsvg/rsvg.h b/include/librsvg-2.0/librsvg/rsvg.h
new file mode 100644
index 0000000..89a46e4
--- /dev/null
+++ b/include/librsvg-2.0/librsvg/rsvg.h
@@ -0,0 +1 @@
+#include "../../librsvg/rsvg.h" /* generated while patching */
