--- contrib/python/voluptuous/voluptuous/tests/tests.py	(index)
+++ contrib/python/voluptuous/voluptuous/tests/tests.py	(working tree)
@@ -28,6 +28,11 @@ def test_new_required_test():
     assert schema.required
 
 
+def _path_to_tests_md():
+    import yatest.common
+    return yatest.common.source_path("contrib/python/voluptuous/voluptuous/tests/tests.md")
+
+
 def test_exact_sequence():
     schema = Schema(ExactSequence([int, int]))
     try:
@@ -1198,19 +1203,19 @@ def test_validation_performance():
 def test_IsDir():
     schema = Schema(IsDir())
     pytest.raises(MultipleInvalid, schema, 3)
-    schema(os.path.dirname(os.path.abspath(__file__)))
+    schema(os.path.dirname(os.path.abspath(_path_to_tests_md())))
 
 
 def test_IsFile():
     schema = Schema(IsFile())
     pytest.raises(MultipleInvalid, schema, 3)
-    schema(os.path.abspath(__file__))
+    schema(os.path.abspath(_path_to_tests_md()))
 
 
 def test_PathExists():
     schema = Schema(PathExists())
     pytest.raises(MultipleInvalid, schema, 3)
-    schema(os.path.abspath(__file__))
+    schema(os.path.abspath(_path_to_tests_md()))
 
 
 def test_description():
