--- contrib/python/pyarrow/pyarrow/__init__.py	(index)
+++ contrib/python/pyarrow/pyarrow/__init__.py	(working tree)
@@ -175,7 +175,8 @@ from pyarrow.lib import (NativeFile, PythonFile,
                          create_memory_map, MockOutputStream,
                          input_stream, output_stream)
 
-from pyarrow._hdfsio import HdfsFile, have_libhdfs
+# Disable for Arcadia, arrow imported without support hdfs
+# from pyarrow._hdfsio import HdfsFile, have_libhdfs
 
 from pyarrow.lib import (ChunkedArray, RecordBatch, Table, table,
                          concat_arrays, concat_tables)
@@ -200,7 +201,7 @@ from pyarrow.lib import (deserialize_from, deserialize,
                          SerializationCallbackError,
                          DeserializationCallbackError)
 
-import pyarrow.hdfs as hdfs
+# import pyarrow.hdfs as hdfs
 
 from pyarrow.ipc import serialize_pandas, deserialize_pandas
 import pyarrow.ipc as ipc
@@ -217,7 +218,7 @@ import pyarrow.types as types
 
 from pyarrow.filesystem import FileSystem as _FileSystem
 from pyarrow.filesystem import LocalFileSystem as _LocalFileSystem
-from pyarrow.hdfs import HadoopFileSystem as _HadoopFileSystem
+# from pyarrow.hdfs import HadoopFileSystem as _HadoopFileSystem
 
 from pyarrow.lib import SerializationContext as _SerializationContext
 from pyarrow.lib import SerializedPyObject as _SerializedPyObject
@@ -239,7 +240,7 @@ _deprecated = {
     "localfs": (_localfs, "LocalFileSystem"),
     "FileSystem": (_FileSystem, "FileSystem"),
     "LocalFileSystem": (_LocalFileSystem, "LocalFileSystem"),
-    "HadoopFileSystem": (_HadoopFileSystem, "HadoopFileSystem"),
+    # "HadoopFileSystem": (_HadoopFileSystem, "HadoopFileSystem"),
 }
 
 _serialization_deprecatd = {
