--- contrib/python/uWSGI/py3/uwsgi.h	(index)
+++ contrib/python/uWSGI/py3/uwsgi.h	(working tree)
@@ -81,9 +81,18 @@ extern "C" {
 #define UWSGI_OPT_METRICS	(1 << 15)
 
 #define MAX_GENERIC_PLUGINS 128
+
+#ifndef MAX_GATEWAYS
 #define MAX_GATEWAYS 64
+#endif
+
+#ifndef MAX_TIMERS
 #define MAX_TIMERS 64
+#endif
+
+#ifndef MAX_CRONS
 #define MAX_CRONS 64
+#endif
 
 #define UWSGI_VIA_SENDFILE	1
 #define UWSGI_VIA_ROUTE	2
