--- contrib/python/uWSGI/py3/plugins/python/gil.c	(index)
+++ contrib/python/uWSGI/py3/plugins/python/gil.c	(working tree)
@@ -6,6 +6,9 @@ extern struct uwsgi_python up;
 
 void gil_real_get() {
 	//uwsgi_log("LOCK %d\n", uwsgi.mywid);
+	if (PyGILState_Check()) {
+		return;
+	}
 #if !defined(PYTHREE)
 	PyEval_AcquireLock();
 	PyThreadState_Swap((PyThreadState *) pthread_getspecific(up.upt_gil_key));
