From b71dc90b82c1e8e640d0c449d535171b19a08d9d Mon Sep 17 00:00:00 2001
From: Sam Mason <sam@samason.uk>
Date: Tue, 21 Feb 2017 17:44:43 +0000
Subject: [PATCH] seem to require this to be able to call uwsgi.route() from
 python

---
 core/routing.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/core/routing.c b/core/routing.c
index 5887ec308..ab26b399b 100644
--- a/core/routing.c
+++ b/core/routing.c
@@ -535,6 +535,7 @@ int uwsgi_route_api_func(struct wsgi_request *wsgi_req, char *router, char *args
 	return -1;
 found:
 	ur = uwsgi_calloc(sizeof(struct uwsgi_route));
+	uwsgi_fixup_routes(ur);
 	// initialize the virtual route
 	if (r->func(ur, args)) {
 		free(ur);
