Index: uwsgi/plugins/python/uwsgi_pymodule.c
===================================================================
--- uwsgi/plugins/python/uwsgi_pymodule.c	(revision 3390957)
+++ uwsgi/plugins/python/uwsgi_pymodule.c	(working copy)
@@ -914,6 +914,17 @@
 	return Py_None;
 }

+PyObject *py_uwsgi_headers_sent(PyObject * self, PyObject * args) {
+    struct wsgi_request *wsgi_req = py_current_wsgi_req();
+    if (wsgi_req->headers_sent) {
+        Py_INCREF(Py_True);
+        return Py_True;
+    }
+
+    Py_INCREF(Py_False);
+    return Py_False;
+}
+
 PyObject *py_uwsgi_is_locked(PyObject * self, PyObject * args) {

         int lock_num = 0;
@@ -2650,6 +2661,7 @@
 	{"recv", py_uwsgi_recv, METH_VARARGS, ""},
 	{"close", py_uwsgi_close, METH_VARARGS, ""},
 	{"i_am_the_spooler", py_uwsgi_i_am_the_spooler, METH_VARARGS, ""},
+	{"headers_sent", py_uwsgi_headers_sent, METH_VARARGS, ""},

 	{"parsefile", py_uwsgi_parse_file, METH_VARARGS, ""},
 	{"embedded_data", py_uwsgi_embedded_data, METH_VARARGS, ""},
