--- contrib/python/django-export-action-py3/export_action/report.py	(index)
+++ contrib/python/django-export-action-py3/export_action/report.py	(working tree)
@@ -4,6 +4,8 @@ from collections import namedtuple
 from itertools import chain
 import csv
 import re
+import datetime
+from dateutil.tz import tzlocal
 
 from django.http import HttpResponse
 from django.utils.encoding import force_text
@@ -105,6 +107,8 @@ def build_sheet(data, ws, sheet_name='report', header=None, widths=None):
                     row[i] = text_type(item)
                 except UnicodeDecodeError:
                     row[i] = text_type(item.decode('utf-8', 'ignore'))
+            elif isinstance(item, datetime.datetime):
+                row[i] = item.replace(tzinfo=None)
             elif type(item) is dict:
                 row[i] = text_type(item)
         try:
