--- contrib/python/django-export-action-py3/export_action/introspection.py	(index)
+++ contrib/python/django-export-action-py3/export_action/introspection.py	(working tree)
@@ -3,7 +3,7 @@ from __future__ import unicode_literals, absolute_import
 from itertools import chain
 
 from django.contrib.contenttypes.models import ContentType
-from django.db.models.fields import FieldDoesNotExist
+from django.core.exceptions import FieldDoesNotExist
 
 
 def _get_field_by_name(model_class, field_name):
--- contrib/python/django-export-action-py3/export_action/report.py	(index)
+++ contrib/python/django-export-action-py3/export_action/report.py	(working tree)
@@ -17,7 +17,7 @@ from openpyxl.utils import get_column_letter
 from openpyxl.workbook import Workbook
 from openpyxl.writer.excel import save_virtual_workbook
 
-from django.utils.six import BytesIO, text_type
+from six import BytesIO, text_type
 
 from .introspection import get_model_from_path_string
 
--- contrib/python/django-export-action-py3/export_action/templates/export_action/export.html	(index)
+++ contrib/python/django-export-action-py3/export_action/templates/export_action/export.html	(working tree)
@@ -1,5 +1,5 @@
 {% extends 'admin/base_site.html' %}
-{% load i18n admin_urls admin_static %}
+{% load i18n admin_urls static %}
 
 {% block extrahead %}
     {{ block.super }}
