--- contrib/python/django-export-action-py3/export_action/admin.py	(revision 43f1d23b078184fa26e4676cb90f0244d7784406)
+++ contrib/python/django-export-action-py3/export_action/admin.py	(revision 0a10deb53235aa4ce305e5d8b0379d2d912f0f47)
@@ -6,1 +6,6 @@
-from django.utils.translation import ugettext_lazy as _
+try:
+    # Django < 4.0
+    from django.utils.translation import ugettext_lazy as _
+except ImportError:
+    # Django >= 4.0
+    from django.utils.translation import gettext_lazy as _
--- contrib/python/django-export-action-py3/export_action/report.py	(revision 43f1d23b078184fa26e4676cb90f0244d7784406)
+++ contrib/python/django-export-action-py3/export_action/report.py	(revision 0a10deb53235aa4ce305e5d8b0379d2d912f0f47)
@@ -11,1 +11,6 @@
-from django.utils.encoding import force_text
+try:
+    # Django < 4.0
+    from django.utils.encoding import force_text
+except ImportError:
+    # Django >= 4.0
+    from django.utils.encoding import force_str as force_text
