--- contrib/python/zero-downtime-migrations/zero_downtime_migrations/backend/schema.py	(index)
+++ contrib/python/zero-downtime-migrations/zero_downtime_migrations/backend/schema.py	(working tree)
@@ -224,7 +224,8 @@ class ZeroDownTimeMixin(object):
         return actions
 
     def get_pk_column_name(self, model):
-        return model._meta.pk.name
+        _, db_column = model._meta.pk.get_attname_column()
+        return db_column
 
     def update_batch(self, model, field, objects_in_batch_count, value):
         pk_column_name = self.get_pk_column_name(model)
