--- contrib/libs/onnx/onnx/cpp2py_export.cc	(index)
+++ contrib/libs/onnx/onnx/cpp2py_export.cc	(working tree)
@@ -35,9 +35,9 @@ static std::tuple<bool, py::bytes, py::bytes> Parse(const char* cstr) {
   ProtoType proto{};
   OnnxParser parser(cstr);
   auto status = parser.Parse(proto);
-  std::string out;
+  TString out;
   proto.SerializeToString(&out);
-  return std::make_tuple(status.IsOK(), py::bytes(status.ErrorMessage()), py::bytes(out));
+  return std::make_tuple(status.IsOK(), py::bytes(status.ErrorMessage()), py::bytes(TProtoStringType{out}));
 }
 
 template <typename ProtoType>
@@ -101,9 +101,9 @@ std::unordered_map<std::string, py::bytes> CallNodeInferenceFunction(
   for (size_t i = 0; i < ctx.allOutputTypes_.size(); i++) {
     const auto& proto = ctx.allOutputTypes_[i];
     if (proto.IsInitialized()) {
-      std::string s;
+      TString s;
       proto.SerializeToString(&s);
-      typeProtoBytes[node.output(static_cast<int>(i))] = py::bytes(s);
+      typeProtoBytes[node.output(static_cast<int>(i))] = py::bytes(s.ConstRef());
     }
   }
 
@@ -189,9 +189,9 @@ PYBIND11_MODULE(onnx_cpp2py_export, onnx_cpp2py_export) {
       .def_property_readonly(
           "_default_value",
           [](OpSchema::Attribute* attr) -> py::bytes {
-            std::string out;
+            TString out;
             attr->default_value.SerializeToString(&out);
-            return out;
+            return out.ConstRef();
           })
       .def_readonly("required", &OpSchema::Attribute::required);
 
@@ -343,20 +343,20 @@ PYBIND11_MODULE(onnx_cpp2py_export, onnx_cpp2py_export) {
       .def_property_readonly(
           "_function_body",
           [](OpSchema* op) -> py::bytes {
-            std::string bytes = "";
+            TString bytes = "";
             if (op->HasFunction())
               op->GetFunction()->SerializeToString(&bytes);
-            return py::bytes(bytes);
+            return py::bytes(bytes.ConstRef());
           })
       .def(
           "get_function_with_opset_version",
           [](OpSchema* op, int opset_version) -> py::bytes {
-            std::string bytes = "";
+            TString bytes = "";
             const FunctionProto* function_proto = op->GetFunction(opset_version);
             if (function_proto) {
               function_proto->SerializeToString(&bytes);
             }
-            return py::bytes(bytes);
+            return py::bytes(TProtoStringType{bytes});
           })
       .def_property_readonly("has_context_dependent_function", &OpSchema::HasContextDependentFunction)
       .def(
@@ -364,7 +364,7 @@ PYBIND11_MODULE(onnx_cpp2py_export, onnx_cpp2py_export) {
           [](OpSchema* op, const py::bytes& bytes, const std::vector<py::bytes>& input_types_bytes) -> py::bytes {
             NodeProto proto{};
             ParseProtoFromPyBytes(&proto, bytes);
-            std::string func_bytes = "";
+            TString func_bytes = "";
             if (op->HasContextDependentFunction()) {
               std::vector<TypeProto> input_types;
               input_types.reserve(input_types_bytes.size());
@@ -378,7 +378,7 @@ PYBIND11_MODULE(onnx_cpp2py_export, onnx_cpp2py_export) {
               op->BuildContextDependentFunction(ctx, func_proto);
               func_proto.SerializeToString(&func_bytes);
             }
-            return py::bytes(func_bytes);
+            return py::bytes(TProtoStringType{func_bytes});
           })
       .def(
           "get_context_dependent_function_with_opset_version",
@@ -386,7 +386,7 @@ PYBIND11_MODULE(onnx_cpp2py_export, onnx_cpp2py_export) {
               -> py::bytes {
             NodeProto proto{};
             ParseProtoFromPyBytes(&proto, bytes);
-            std::string func_bytes = "";
+            TString func_bytes = "";
             if (op->HasContextDependentFunctionWithOpsetVersion(opset_version)) {
               std::vector<TypeProto> input_types;
               input_types.reserve(input_types_bytes.size());
@@ -400,7 +400,7 @@ PYBIND11_MODULE(onnx_cpp2py_export, onnx_cpp2py_export) {
               op->BuildContextDependentFunction(ctx, func_proto, opset_version);
               func_proto.SerializeToString(&func_bytes);
             }
-            return py::bytes(func_bytes);
+            return py::bytes(func_bytes.ConstRef());
           });
 
   defs.def(
@@ -596,9 +596,9 @@ PYBIND11_MODULE(onnx_cpp2py_export, onnx_cpp2py_export) {
     ParseProtoFromPyBytes(&proto, bytes);
     shape_inference::InferShapes(proto);
     auto result = version_conversion::ConvertVersion(proto, target);
-    std::string out;
+    TString out;
     result.SerializeToString(&out);
-    return py::bytes(out);
+    return py::bytes(TProtoStringType{out});
   });
 
   // Submodule `inliner`
@@ -609,9 +609,9 @@ PYBIND11_MODULE(onnx_cpp2py_export, onnx_cpp2py_export) {
     ModelProto model{};
     ParseProtoFromPyBytes(&model, bytes);
     inliner::InlineLocalFunctions(model, convert_version);
-    std::string out;
+    TString out;
     model.SerializeToString(&out);
-    return py::bytes(out);
+    return py::bytes(TProtoStringType{out});
   });
 
   // inline_selected_functions: Inlines all functions specified in function_ids, unless
@@ -624,9 +624,9 @@ PYBIND11_MODULE(onnx_cpp2py_export, onnx_cpp2py_export) {
         ParseProtoFromPyBytes(&model, bytes);
         auto function_id_set = inliner::FunctionIdSet::Create(std::move(function_ids), exclude);
         inliner::InlineSelectedFunctions(model, *function_id_set);
-        std::string out;
+        TString out;
         model.SerializeToString(&out);
-        return py::bytes(out);
+        return py::bytes(TProtoStringType{out});
       });
 
   // Submodule `shape_inference`
@@ -641,9 +641,9 @@ PYBIND11_MODULE(onnx_cpp2py_export, onnx_cpp2py_export) {
         ParseProtoFromPyBytes(&proto, bytes);
         ShapeInferenceOptions options{check_type, strict_mode ? 1 : 0, data_prop};
         shape_inference::InferShapes(proto, OpSchemaRegistry::Instance(), options);
-        std::string out;
+        TString out;
         proto.SerializeToString(&out);
-        return py::bytes(out);
+        return py::bytes(TProtoStringType{out});
       },
       "bytes"_a,
       "check_type"_a = false,
@@ -689,9 +689,9 @@ PYBIND11_MODULE(onnx_cpp2py_export, onnx_cpp2py_export) {
         std::vector<py::bytes> result;
         result.reserve(output_types.size());
         for (auto& type_proto : output_types) {
-          std::string out;
+          TString out;
           type_proto.SerializeToString(&out);
-          result.emplace_back(out);
+          result.emplace_back(TProtoStringType{out});
         }
         return result;
       });
--- contrib/libs/onnx/onnx/defs/parser.cc	(index)
+++ contrib/libs/onnx/onnx/defs/parser.cc	(working tree)
@@ -353,16 +353,17 @@ Status OnnxParser::ParseFunctionInputOutput(IdList& idlist, ValueInfoList& vilis
       // The name is added to idlist. If the optional type is present, an entry is
       // added to vilist.
 
-      std::string* name = idlist.Add();
+      std::string name;
       ValueInfoProto* vi = nullptr;
 
       if (NextIsType()) {
         vi = vilist.Add();
         PARSE(*(vi->mutable_type()));
       }
-      CHECK_PARSER_STATUS(ParseQuotableIdentifier(*name));
+      CHECK_PARSER_STATUS(ParseQuotableIdentifier(name));
+      *idlist.Add() = name;
       if (vi != nullptr)
-        vi->set_name(*name);
+        vi->set_name(name);
     } while (Matches(','));
     MATCH(')');
   }
@@ -556,7 +557,9 @@ Status OnnxParser::ParseSingleAttributeValue(AttributeProto& attr, AttributeProt
       if ((next == '{') || (next == '=') || (NextIsIdentifier())) {
         attr.set_type(AttributeProto_AttributeType_TENSOR);
         auto& tensorProto = *attr.mutable_t();
-        CHECK_PARSER_STATUS(ParseOptionalQuotableIdentifier(*tensorProto.mutable_name()));
+        std::string name;
+        CHECK_PARSER_STATUS(ParseOptionalQuotableIdentifier(name));
+        *tensorProto.mutable_name() = name;
         (void)Matches('='); // Optional, to unify handling of initializers
         CHECK_PARSER_STATUS(Parse(tensorProto, typeProto));
       } else {
@@ -722,7 +725,9 @@ Status OnnxParser::Parse(AttrList& attrlist) {
 
 Status OnnxParser::Parse(NodeProto& node) {
   if (Matches('[')) {
-    CHECK_PARSER_STATUS(ParseOptionalQuotableIdentifier(*node.mutable_name()));
+    std::string name;
+    CHECK_PARSER_STATUS(ParseOptionalQuotableIdentifier(name));
+    *node.mutable_name() = name;
     MATCH(']');
   }
   PARSE(*node.mutable_output());
--- contrib/libs/onnx/onnx/defs/parser.h	(index)
+++ contrib/libs/onnx/onnx/defs/parser.h	(working tree)
@@ -19,7 +19,7 @@ namespace ONNX_NAMESPACE {
 
 using namespace ONNX_NAMESPACE::Common;
 
-using IdList = google::protobuf::RepeatedPtrField<std::string>;
+using IdList = google::protobuf::RepeatedPtrField<TProtoStringType>;
 
 using NodeList = google::protobuf::RepeatedPtrField<NodeProto>;
 
--- contrib/libs/onnx/onnx/defs/tensor_proto_util.cc	(index)
+++ contrib/libs/onnx/onnx/defs/tensor_proto_util.cc	(working tree)
@@ -22,6 +22,15 @@ namespace ONNX_NAMESPACE {
     return t;                                       \
   }
 
+#define DEFINE_TO_TENSOR_STRING(type, enumType, field) \
+  template <>                                       \
+  TensorProto ToTensor<type>(const type& value) {   \
+    TensorProto t;                                  \
+    t.set_data_type(enumType);                      \
+    t.add_##field##_data(TProtoStringType(value));  \
+    return t;                                       \
+  }
+
 #define DEFINE_TO_TENSOR_LIST(type, enumType, field)            \
   template <>                                                   \
   TensorProto ToTensor<type>(const std::vector<type>& values) { \
@@ -34,6 +43,18 @@ namespace ONNX_NAMESPACE {
     return t;                                                   \
   }
 
+#define DEFINE_TO_TENSOR_LIST_STRING(type, enumType, field)            \
+  template <>                                                   \
+  TensorProto ToTensor<type>(const std::vector<type>& values) { \
+    TensorProto t;                                              \
+    t.clear_##field##_data();                                   \
+    t.set_data_type(enumType);                                  \
+    for (const type& val : values) {                            \
+      t.add_##field##_data(TProtoStringType{val});                                \
+    }                                                           \
+    return t;                                                   \
+  }
+
 #define DEFINE_PARSE_DATA(type, typed_data_fetch, tensorproto_datatype)                                            \
   template <>                                                                                                      \
   std::vector<type> ParseData(const TensorProto* tensor_proto) {                                                   \
@@ -121,7 +142,7 @@ DEFINE_TO_TENSOR_ONE(int32_t, TensorProto_DataType_INT32, int32)
 DEFINE_TO_TENSOR_ONE(int64_t, TensorProto_DataType_INT64, int64)
 DEFINE_TO_TENSOR_ONE(uint64_t, TensorProto_DataType_UINT64, uint64)
 DEFINE_TO_TENSOR_ONE(double, TensorProto_DataType_DOUBLE, double)
-DEFINE_TO_TENSOR_ONE(std::string, TensorProto_DataType_STRING, string)
+DEFINE_TO_TENSOR_STRING(std::string, TensorProto_DataType_STRING, string)
 
 DEFINE_TO_TENSOR_LIST(float, TensorProto_DataType_FLOAT, float)
 DEFINE_TO_TENSOR_LIST(bool, TensorProto_DataType_BOOL, int32)
@@ -129,7 +150,7 @@ DEFINE_TO_TENSOR_LIST(int32_t, TensorProto_DataType_INT32, int32)
 DEFINE_TO_TENSOR_LIST(int64_t, TensorProto_DataType_INT64, int64)
 DEFINE_TO_TENSOR_LIST(uint64_t, TensorProto_DataType_UINT64, uint64)
 DEFINE_TO_TENSOR_LIST(double, TensorProto_DataType_DOUBLE, double)
-DEFINE_TO_TENSOR_LIST(std::string, TensorProto_DataType_STRING, string)
+DEFINE_TO_TENSOR_LIST_STRING(std::string, TensorProto_DataType_STRING, string)
 
 DEFINE_PARSE_DATA(int32_t, int32_data, TensorProto_DataType_INT32)
 DEFINE_PARSE_DATA(int64_t, int64_data, TensorProto_DataType_INT64)
--- contrib/libs/onnx/onnx/inliner/inliner.cc	(index)
+++ contrib/libs/onnx/onnx/inliner/inliner.cc	(working tree)
@@ -195,8 +195,8 @@ class InliningRenamer : private MutableVisitor {
 
   template <bool isOutput>
   void Bind(
-      google::protobuf::RepeatedPtrField<std::string>& formals,
-      const google::protobuf::RepeatedPtrField<std::string>& actuals) {
+      google::protobuf::RepeatedPtrField<TProtoStringType>& formals,
+      const google::protobuf::RepeatedPtrField<TProtoStringType>& actuals) {
     // Every formal parameter name FP should be replace by the corresponding actual parameter name AP.
     // However, if AP is empty, it is a missing optional parameter. This does not make any difference
     // for inputs. However, for outputs we use a unique dummy name to handle the case that it
@@ -206,8 +206,8 @@ class InliningRenamer : private MutableVisitor {
     auto& current_scope = rename_scopes.back();
     int i = 0;
     for (; i < actuals.size(); ++i) {
-      std::string& formal = *formals.Mutable(i);
-      std::string rename_as = actuals.Get(i);
+      TProtoStringType& formal = *formals.Mutable(i);
+      TProtoStringType rename_as = actuals.Get(i);
       if (isOutput)
         if (rename_as.empty())
           rename_as = MakeUnique(formal);
@@ -216,8 +216,8 @@ class InliningRenamer : private MutableVisitor {
         formal = rename_as;
     }
     for (; i < formals.size(); ++i) {
-      std::string& formal = *formals.Mutable(i);
-      std::string rename_as = isOutput ? MakeUnique(formal) : std::string();
+      TProtoStringType& formal = *formals.Mutable(i);
+      TProtoStringType rename_as = isOutput ? MakeUnique(formal) : std::string();
       current_scope[formal] = rename_as;
       if (!rename_as.empty())
         formal = rename_as;
@@ -230,10 +230,10 @@ class InliningRenamer : private MutableVisitor {
       node->set_name(MakeUnique(node->name()));
 
     for (auto& x : *node->mutable_input()) {
-      LookupOrRename(x, false);
+      LookupOrRename(x.MutRef(), false);
     }
     for (auto& y : *node->mutable_output()) {
-      LookupOrRename(y, true);
+      LookupOrRename(y.MutRef(), true);
     }
     return true; // Process attribute subgraphs in traversal
   }
@@ -244,11 +244,11 @@ class InliningRenamer : private MutableVisitor {
   void VisitGraph(GraphProto* graph) override {
     rename_scopes.emplace_back();
     for (auto& x : *graph->mutable_input())
-      Rename(*x.mutable_name());
+      Rename(x.mutable_name()->MutRef());
     for (auto& init : *graph->mutable_initializer())
-      Rename(*init.mutable_name());
+      Rename(init.mutable_name()->MutRef());
     for (auto& y : *graph->mutable_output())
-      Rename(*y.mutable_name());
+      Rename(y.mutable_name()->MutRef());
     for (auto& n : *graph->mutable_node())
       VisitNode(&n);
     rename_scopes.pop_back();
@@ -267,7 +267,7 @@ class InliningRenamer : private MutableVisitor {
 
     renamer.VisitFunction(&callee);
     for (auto& v : *callee.mutable_value_info())
-      renamer.LookupOrRename(*v.mutable_name(), false);
+      renamer.LookupOrRename(v.mutable_name()->MutRef(), false);
   }
 };
 
--- contrib/libs/onnx/onnx/shape_inference/implementation.cc	(index)
+++ contrib/libs/onnx/onnx/shape_inference/implementation.cc	(working tree)
@@ -846,7 +846,7 @@ void InferShapes(
   // Save the inferred model to the original model path
   // Use SerializeToString instead of SerializeToOstream due to LITE_PROTO
   std::fstream output(save_path, std::ios::out | std::ios::trunc | std::ios::binary);
-  std::string model_string;
+  TProtoStringType model_string;
   ONNX_TRY {
     model.SerializeToString(&model_string);
     output << model_string;
