--- contrib/python/docutils/py3/docutils/utils/__init__.py	(index)
+++ contrib/python/docutils/py3/docutils/utils/__init__.py	(working tree)
@@ -533,5 +533,8 @@ def find_file_in_dirs(path, dirs):
         return path.as_posix()
     for d in dirs:
         f = Path(d).expanduser() / path
+        # we use '//' cause resources inside contrib/python/docutils/docutils/ya.make start with
+        if str(f).startswith('//'):
+            return f
         if f.exists():
             return f.as_posix()
--- contrib/python/docutils/py3/docutils/writers/html4css1/__init__.py	(index)
+++ contrib/python/docutils/py3/docutils/writers/html4css1/__init__.py	(working tree)
@@ -26,8 +26,6 @@ class Writer(writers._html_base.Writer):
-    default_stylesheet_dirs = ['.',
-                               os.path.abspath(os.path.dirname(__file__)),
-                               os.path.abspath(os.path.join(
-                                   os.path.dirname(os.path.dirname(__file__)),
-                                   'html5_polyglot'))  # for math.css
+    default_stylesheet_dirs = [
+        # we use '//' cause resources inside contrib/python/docutils/docutils/ya.make start with
+        '//docutils/writers/html4css1',
+        '//docutils/writers/html5_polyglot',
                                ]
-    default_template = os.path.join(
-        os.path.dirname(os.path.abspath(__file__)), 'template.txt')
+    default_template = '//docutils/writers/html4css1/template.txt'
--- contrib/python/docutils/py3/docutils/writers/html5_polyglot/__init__.py	(index)
+++ contrib/python/docutils/py3/docutils/writers/html5_polyglot/__init__.py	(working tree)
@@ -39,2 +39,2 @@ class Writer(writers._html_base.Writer):
-    default_stylesheet_dirs = ['.', str(Path(__file__).parent)]
-    default_template = Path(__file__).parent / 'template.txt'
+    default_stylesheet_dirs = ['//docutils/writers/html5_polyglot']
+    default_template = '//docutils/writers/html5_polyglot/template.txt'
--- contrib/python/docutils/py3/docutils/writers/latex2e/__init__.py	(index)
+++ contrib/python/docutils/py3/docutils/writers/latex2e/__init__.py	(working tree)
@@ -27,6 +27,7 @@ from docutils import frontend, nodes, languages, writers, utils, io
 from docutils.utils.error_reporting import SafeString
 from docutils.transforms import writer_aux
 from docutils.utils.math import pick_math_environment, unichar2tex
+import library.python.resource as rs
 
 LATEX_WRITER_DIR = Path(__file__).parent
 
@@ -34,1 +35,1 @@ class Writer(writers.Writer):
-    default_template_path = LATEX_WRITER_DIR
+    default_template_path = '//docutils/writers/latex2e'
@@ -573,1 +573,3 @@ def _read_block(fp):
-with open(LATEX_WRITER_DIR/'docutils.sty', encoding='utf-8') as fp:
+import io
+_docutils_sty = "//docutils/writers/latex2e/docutils.sty"
+with io.StringIO(rs.find(_docutils_sty).decode('utf8')) as fp:
@@ -1410,1 +1411,5 @@ class LaTeXTranslator(nodes.NodeVisitor):
-                content = path.read_text(encoding='utf-8')
+                # we use '//' cause resources inside contrib/python/docutils/docutils/ya.make start with
+                if str(path).startswith('//'):
+                    content = rs.find(str(path))
+                else:
+                    content = path.read_text(encoding='utf-8')
--- contrib/python/docutils/py3/docutils/writers/_html_base.py	(index)
+++ contrib/python/docutils/py3/docutils/writers/_html_base.py	(working tree)
@@ -25,6 +25,7 @@ import re
 import urllib
 import warnings
 import xml.etree.ElementTree as ET  # TODO: lazy import in prepare_svg()?
+import library.python.resource as rs
 
 import docutils
 from docutils import frontend, languages, nodes, utils, writers
@@ -159,8 +160,16 @@ class Writer(writers.Writer):
         self.output = self.apply_template()
 
     def apply_template(self):
-        with open(self.document.settings.template, encoding='utf-8') as fp:
-            template = fp.read()
+        template = None
+        path = str(self.document.settings.template)
+        if not path.startswith("//"):
+            path = "/" + path
+        raw = rs.find(path)
+        if raw is not None:
+            template = str(raw, 'utf-8')
+        if template is None:
+            with open(self.document.settings.template, encoding='utf-8') as fp:
+                template = fp.read()
         subs = self.interpolation_dict()
         return template % subs
 
@@ -504,8 +513,12 @@ class HTMLTranslator(nodes.NodeVisitor):
             adjust_path = bool(self.settings.stylesheet_path)
         if self.settings.embed_stylesheet:
             try:
-                with open(path, encoding='utf-8') as f:
-                    content = f.read()
+                # we use '//' cause resources inside contrib/python/docutils/docutils/ya.make start with
+                if str(path).startswith('//'):
+                    content = rs.find(str(path))
+                else:
+                    with open(path, encoding='utf-8') as f:
+                        content = f.read()
             except OSError as err:
                 msg = f'Cannot embed stylesheet: {err}'
                 self.document.reporter.error(msg)
