--- contrib/python/django-hosts/py3/tests/test_resolvers.py	(index)
+++ contrib/python/django-hosts/py3/tests/test_resolvers.py	(working tree)
@@ -9,1 +9,1 @@ from django_hosts.resolvers import (get_host, get_host_patterns, get_hostconf,
-from .hosts import simple
+from tests.hosts import simple
--- contrib/python/django-hosts/py3/tests/test_sites.py	(index)
+++ contrib/python/django-hosts/py3/tests/test_sites.py	(working tree)
@@ -10,1 +10,1 @@ from django.utils.functional import empty
-from .models import Author, BlogPost, WikiPage
+from tests.models import Author, BlogPost, WikiPage
