--- contrib/python/django/django-3/django/core/management/__init__.py
+++ contrib/python/django/django-3/django/core/management/__init__.py
@@ -19,6 +19,8 @@ from django.utils import autoreload, lru_cache, six
 from django.core.management.color import color_style
 from django.utils import autoreload
 
+from library.python.django.utils import find_modules_in_path
+
 
 def find_commands(management_dir):
     """
@@ -63,7 +65,7 @@ def get_commands():
     The dictionary is cached on the first call and reused on subsequent
     calls.
     """
-    commands = {name: 'django.core' for name in find_commands(__path__[0])}
+    commands = {name: 'django.core' for name in find_modules_in_path('django.core.management.commands')}
 
     if not settings.configured:
         return commands
@@ -72,6 +74,12 @@ def get_commands():
         path = os.path.join(app_config.path, 'management')
         commands.update({name: app_config.name for name in find_commands(path)})
 
+        try:
+            commands.update(
+                {name: app_config.name for name in find_modules_in_path('%s.management.commands' % app_config.name)})
+        except ImportError:
+            commands.update({name: app_config.name for name in find_commands(path)})
+
     return commands
 
 
