--- contrib/python/pystan/py2/pystan/model.py
+++ contrib/python/pystan/py2/pystan/model.py
@@ -12,6 +12,7 @@ if PY2:
 else:
     from collections.abc import Callable, Iterable
 import datetime
+import importlib
 import io
 import itertools
 import logging
@@ -30,3 +31,0 @@ import time
-import Cython
-from Cython.Build.Inline import _get_build_extension
-from Cython.Build.Dependencies import cythonize
@@ -103,6 +100,9 @@ class StanModel:

     Parameters
     ----------
+    module_name : string
+        Name of the Python module created by pystan_model.
+
     file : string {'filename', 'file'}
         If filename, the string passed as an argument is expected to
         be a filename containing the Stan model specification.
@@ -105,6 +102,7 @@
 
     Attributes
     ----------
+    module_name : string
     model_name : string
     model_code : string
         Stan code for the model.
@@ -171,1 +160,1 @@
-    def __init__(self, file=None, charset='utf-8', model_name="anon_model",
+    def __init__(self, module_name, file=None, charset='utf-8', model_name="anon_model",
@@ -173,6 +160,10 @@
                  boost_lib=None, eigen_lib=None, verbose=False,
                  obfuscate_model_name=True, extra_compile_args=None,
                  allow_undefined=False, include_dirs=None, includes=None):
+        self.module_name = module_name
+        self.module = importlib.import_module(module_name)
+        self.fit_class = getattr(self.module, "StanFit4Model")
+        return
 
         if stanc_ret is None:
             stanc_ret = pystan.api.stanc(file=file,
@@ -190,2 +190,2 @@
-        s = u"StanModel object '{}' coded as follows:\n{}"
-        return s.format(self.model_name, self.model_code)
+        s = u"StanModel('{}')"
+        return s.format(self.module_name)
@@ -208,6 +190,9 @@

     def __setstate__(self, state):
         self.__dict__.update(state)
+        self.module = importlib.import_module(self.module_name)
+        self.fit_class = getattr(self.module, "StanFit4Model")
+        return
         lib_dir = tempfile.mkdtemp()
         with io.open(os.path.join(lib_dir, self.module_filename), 'wb') as f:
             f.write(self.module_bytes)
