diff --git a/dgif_lib.c b/dgif_lib.c
index 82fc097..faa5edd 100644
--- a/dgif_lib.c
+++ b/dgif_lib.c
@@ -111,6 +111,14 @@ DGifOpenFileHandle(int FileHandle, int *Error)
 #endif                                  /* _WIN32 */
 
 	f = fdopen(FileHandle, "rb"); /* Make it into a stream: */
+    if (f == NULL) {
+        if (Error != NULL)
+        *Error = D_GIF_ERR_OPEN_FAILED;
+        (void)close(FileHandle);
+        free((char *)Private);
+        free((char *)GifFile);
+        return NULL;
+    }
 
 	/*@-mustfreeonly@*/
 	GifFile->Private = (void *)Private;
