--- contrib/python/aioquic/aioquic/tls.py	(index)
+++ contrib/python/aioquic/aioquic/tls.py	(working tree)
@@ -237,10 +237,10 @@ def verify_certificate(
     capath: Optional[str] = None,
 ) -> None:
     # verify dates
-    now = utcnow()
-    if now < certificate.not_valid_before_utc:
+    now = utcnow().replace(tzinfo=None)
+    if now < certificate.not_valid_before:
         raise AlertCertificateExpired("Certificate is not valid yet")
-    if now > certificate.not_valid_after_utc:
+    if now > certificate.not_valid_after:
         raise AlertCertificateExpired("Certificate is no longer valid")
 
     # verify subject
