--- contrib/python/aioquic/aioquic/h3/connection.py	(index)
+++ contrib/python/aioquic/aioquic/h3/connection.py	(working tree)
@@ -266,6 +266,14 @@ def validate_headers(
                 path = value
             elif key == b":scheme":
                 scheme = value
+            elif key == b":method" and stream is not None:
+                stream.request_method = value
+            elif key == b":status" and stream is not None:
+                try:
+                    if int(value) >= 200:  # skip interim responses
+                        stream.response_status = int(value)
+                except ValueError:
+                    raise MessageError(":status is not an integer")
         else:
             # regular headers
             after_pseudo_headers = True
@@ -359,6 +367,8 @@ class H3Stream:
         self.stream_type: Optional[int] = None
         self.expected_content_length: Optional[int] = None
         self.content_length: int = 0
+        self.request_method: Optional[bytes] = None
+        self.response_status: Optional[int] = None
 
     def is_ended(self) -> bool:
         return self.sending_ended and self.receiving_ended and not self.blocked
@@ -524,6 +534,11 @@ class H3Connection:
         #  create push stream
         push_stream_id = self._create_uni_stream(StreamType.PUSH, push_id=push_id)
         self._quic.send_stream_data(push_stream_id, encode_uint_var(push_id))
+        with self._get_or_create_stream(push_stream_id) as stream:
+            for key, value in headers:
+                if key == b":method":
+                    stream.request_method = value
+                    break
 
         return push_stream_id
 
@@ -577,6 +592,18 @@ class H3Connection:
             if end_stream:
                 stream.finish_sending()
 
+            for key, value in headers:
+                if self._is_client and key == b":method":
+                    stream.request_method = value
+                    break
+                if not self._is_client and key == b":status":
+                    try:
+                        if int(value) >= 200:  # skip interim responses
+                            stream.response_status = int(value)
+                    except ValueError:
+                        pass
+                    break
+
             frame_data = self._encode_headers(stream_id, headers)
 
             # log frame
@@ -712,6 +739,12 @@ class H3Connection:
 
     def _check_content_length(self, stream: H3Stream):
         if (
+            self._is_client
+            and (stream.request_method == b"HEAD" or stream.response_status == 304)
+        ):
+            return
+
+        if (
             stream.expected_content_length is not None
             and stream.content_length != stream.expected_content_length
         ):
@@ -786,19 +819,21 @@ class H3Connection:
                     ),
                 )
 
-            # update state and emit headers
-            if stream.headers_recv_state == HeadersState.INITIAL:
-                stream.headers_recv_state = HeadersState.AFTER_HEADERS
-            else:
-                stream.headers_recv_state = HeadersState.AFTER_TRAILERS
-            http_events.append(
-                HeadersReceived(
-                    headers=headers,
-                    push_id=stream.push_id,
-                    stream_id=stream.stream_id,
-                    stream_ended=stream_ended,
+            # FIXME: do not simply ignore interim responses here?
+            if stream.response_status is not None:  # skip interim responses
+                # update state and emit headers
+                if stream.headers_recv_state == HeadersState.INITIAL:
+                    stream.headers_recv_state = HeadersState.AFTER_HEADERS
+                else:
+                    stream.headers_recv_state = HeadersState.AFTER_TRAILERS
+                http_events.append(
+                    HeadersReceived(
+                        headers=headers,
+                        push_id=stream.push_id,
+                        stream_id=stream.stream_id,
+                        stream_ended=stream_ended,
+                    )
                 )
-            )
         elif frame_type == FrameType.PUSH_PROMISE and stream.push_id is None:
             if not self._is_client:
                 raise FrameUnexpected("Clients must not send PUSH_PROMISE")
