--- contrib/python/boto3/py2/ya.make	(index)
+++ contrib/python/boto3/py2/ya.make	(working tree)
@@ -61,9 +61,9 @@ PY_SRCS(
     PREFIX contrib/python/boto3/py2/
     .dist-info/METADATA
     .dist-info/top_level.txt
-    boto3/data/cloudformation/2010-05-15/resources-1.json
-    boto3/data/cloudwatch/2010-08-01/resources-1.json
-    boto3/data/dynamodb/2012-08-10/resources-1.json
+    #boto3/data/cloudformation/2010-05-15/resources-1.json
+    #boto3/data/cloudwatch/2010-08-01/resources-1.json
+    #boto3/data/dynamodb/2012-08-10/resources-1.json
     boto3/data/ec2/2014-10-01/resources-1.json
     boto3/data/ec2/2015-03-01/resources-1.json
     boto3/data/ec2/2015-04-15/resources-1.json
@@ -71,11 +69,11 @@ RESOURCE_FILES(
     boto3/data/ec2/2016-04-01/resources-1.json
     boto3/data/ec2/2016-09-15/resources-1.json
     boto3/data/ec2/2016-11-15/resources-1.json
-    boto3/data/glacier/2012-06-01/resources-1.json
-    boto3/data/iam/2010-05-08/resources-1.json
-    boto3/data/opsworks/2013-02-18/resources-1.json
+    #boto3/data/glacier/2012-06-01/resources-1.json
+    #boto3/data/iam/2010-05-08/resources-1.json
+    #boto3/data/opsworks/2013-02-18/resources-1.json
     boto3/data/s3/2006-03-01/resources-1.json
-    boto3/data/sns/2010-03-31/resources-1.json
+    #boto3/data/sns/2010-03-31/resources-1.json
     boto3/data/sqs/2012-11-05/resources-1.json
     boto3/examples/cloudfront.rst
     boto3/examples/s3.rst
