--- contrib/python/groovy/groovy/__init__.py	(a387bf64ceaef1f356cbfabaa871bfd91355f8cd)
+++ contrib/python/groovy/groovy/__init__.py	(index)
@@ -1,7 +1,9 @@
+import importlib.resources
+
 from pathlib import Path
 
 from groovy.transpiler import TranspilerError, transpile
 
-__version__ = Path(__file__).parent.joinpath("version.txt").read_text().strip()
+__version__ = (importlib.resources.files(__package__) / 'version.txt').read_text().strip()
 
 __all__ = ["__version__", "transpile", "TranspilerError"]
