--- contrib/python/joblib/joblib/externals/loky/backend/fork_exec.py	(index)
+++ contrib/python/joblib/joblib/externals/loky/backend/fork_exec.py	(working tree)
@@ -33,6 +33,7 @@ def fork_exec(cmd, keep_fds, env=None):
     # as bytes).
     env = env or {}
     env = {**os.environ, **env}
+    env["Y_PYTHON_ENTRY_POINT"] = ":main"
     encoded_env = []
     for key, value in env.items():
         encoded_env.append(os.fsencode(f"{key}={value}"))
