--- contrib/python/schematics/schematics/deprecated.py	(index)
+++ contrib/python/schematics/schematics/deprecated.py	(working tree)
@@ -14,15 +14,8 @@ class SchematicsDeprecationWarning(DeprecationWarning):
 
 
 def deprecated(func):
-    @functools.wraps(func)
-    def new_func(*args, **kwargs):
-        warnings.warn(
-            "Call to deprecated function {0}.".format(func.__name__),
-            category=SchematicsDeprecationWarning,
-            stacklevel=2
-        )
-        return func(*args, **kwargs)
-    return new_func
+    # Schematics seems unmaintained, no point in warning about deprecations anyway.
+    return func
 
 
 class SchemaCompatibilityMixin(object):
