--- contrib/python/whenever/ya.make	(index)
+++ contrib/python/whenever/ya.make	(working tree)
@@ -10,6 +10,23 @@ LICENSE(MIT)

 NO_LINT()

+IF (OS_DARWIN AND ARCH_ARM64)
+    COPY_FILE(a/aarch64-apple-darwin/release/lib_whenever.a lib_whenever.a)
+ELSEIF (OS_DARWIN AND ARCH_X86_64)
+    COPY_FILE(a/x86_64-apple-darwin/release/lib_whenever.a lib_whenever.a)
+ELSEIF (OS_LINUX AND ARCH_X86_64 AND NOT MUSL)
+    COPY_FILE(a/x86_64-unknown-linux-gnu/release/lib_whenever.a lib_whenever.a)
+ELSE()
+    MESSAGE(FATAL_ERROR "We don't have lib_whenever for this platform")
+ENDIF()
+
+IF (OS_DARWIN)
+    LDFLAGS(
+        -framework CoreFoundation
+        -framework Security
+    )
+ENDIF()
+
 PY_REGISTER(
     whenever.whenever
 )
