--- contrib/python/python-json-logger/tests/test_deprecation.py	(index)
+++ contrib/python/python-json-logger/tests/test_deprecation.py	(working tree)
@@ -43 +43,4 @@ def test_import(command: str):
-    output = subprocess.check_output([sys.executable, "-c", f"{command};print('OK')"])
+    import os
+    env = os.environ.copy()
+    env["Y_PYTHON_ENTRY_POINT"] = ":main"
+    output = subprocess.check_output([sys.executable, "-c", f"{command};print('OK')"], env=env)
--- contrib/python/python-json-logger/tests/test_dictconfig.py	(index)
+++ contrib/python/python-json-logger/tests/test_dictconfig.py	(working tree)
@@ -39,2 +39,2 @@ LOGGING_CONFIG = {
-            "json_default": "ext://tests.test_dictconfig.my_json_default",
-            "static_fields": {"ext-val": "ext://tests.test_dictconfig.EXT_VAL"},
+            "json_default": "ext://__tests__.test_dictconfig.my_json_default",
+            "static_fields": {"ext-val": "ext://__tests__.test_dictconfig.EXT_VAL"},
