--- contrib/python/cached-property/py2/tests/test_cached_property.py	(index)
+++ contrib/python/cached-property/py2/tests/test_cached_property.py	(working tree)
@@ -199,10 +199,14 @@ class TestCachedPropertyWithTTL(TestCachedProperty):
         self.assert_cached(check, num_threads)
 
         # The cache expires in the future
+        import freezegun.api
+        orig_value = freezegun.api.call_stack_inspection_limit
+        freezegun.api.call_stack_inspection_limit = 0
         with freeze_time("9999-01-01"):
             check.run_threads(num_threads)
             self.assert_cached(check, 2 * num_threads)
             self.assert_cached(check, 2 * num_threads)
+        freezegun.api.call_stack_inspection_limit = orig_value
 
         # Things are not reverted when we are back to the present
         self.assert_cached(check, 2 * num_threads)
