--- contrib/python/publicsuffix2/py3/publicsuffix2/__init__.py	(index)
+++ contrib/python/publicsuffix2/py3/publicsuffix2/__init__.py	(working tree)
@@ -44,6 +44,7 @@ from __future__ import unicode_literals
 
 import codecs
 from os import path
+import pkgutil
 import warnings
 
 try:
@@ -82,7 +83,9 @@ class PublicSuffixList(object):
         :param idna: boolean, whether to convert file to IDNA-encoded strings
         """
         # Note: we test for None as we accept empty lists as inputs
-        if psl_file is None or isinstance(psl_file, str):
+        if psl_file is None:
+            psl = pkgutil.get_data(__name__, 'public_suffix_list.dat').decode('utf-8').split('\n')
+        elif isinstance(psl_file, str):
             with codecs.open(psl_file or PSL_FILE, 'r', encoding='utf8') as psl:
                 psl = psl.readlines()
         else:
