--- contrib/python/WTForms/py3/wtforms/i18n.py	(index)
+++ contrib/python/WTForms/py3/wtforms/i18n.py	(working tree)
@@ -5,10 +5,8 @@ def messages_path():
     """
     Determine the path to the 'messages' directory as best possible.
     """
-    module_path = os.path.abspath(__file__)
+    module_path = __file__
     locale_path = os.path.join(os.path.dirname(module_path), "locale")
-    if not os.path.exists(locale_path):  # pragma: no cover
-        locale_path = "/usr/share/locale"
     return locale_path
 
 
