--- contrib/python/hyperopt/hyperopt/tests/test_mongoexp.py	(index)
+++ contrib/python/hyperopt/hyperopt/tests/test_mongoexp.py	(working tree)
@@ -22,7 +22,7 @@ from hyperopt.mongoexp import main_worker_helper
 from hyperopt.mongoexp import MongoJobs
 from hyperopt.fmin import fmin
 from hyperopt import hp, rand
-import hyperopt.tests.test_base
+from . import test_base
 from .test_domains import gauss_wave2
 
 
@@ -152,7 +152,7 @@ except OSError as e:
     raise nose.SkipTest()
 
 
-class TestMongoTrials(hyperopt.tests.test_base.TestTrials):
+class TestMongoTrials(test_base.TestTrials):
     def setUp(self):
         self.temp_mongo = TempMongo()
         self.temp_mongo.__enter__()
--- contrib/python/hyperopt/hyperopt/tests/test_rand.py	(index)
+++ contrib/python/hyperopt/hyperopt/tests/test_rand.py	(working tree)
@@ -1,7 +1,7 @@ from builtins import range
 import unittest
 from hyperopt.base import Trials, trials_from_docs, miscs_to_idxs_vals
 from hyperopt import rand
-from hyperopt.tests.test_base import Suggest_API
+from .test_base import Suggest_API
 from .test_domains import gauss_wave2, coin_flip
 import numpy as np
 
--- contrib/python/hyperopt/hyperopt/tests/test_tpe.py	(index)
+++ contrib/python/hyperopt/hyperopt/tests/test_tpe.py	(working tree)
@@ -546,16 +546,16 @@ class TestSuggest(unittest.TestCase, CasePerDomain):
         )
 
 
-class TestSuggestAtpe(unittest.TestCase, NonCategoricalCasePerDomain):
-    def work(self):
-        trials = Trials()
-        fmin(
-            passthrough,
-            space=self.bandit.expr,
-            algo=atpe.suggest,
-            trials=trials,
-            max_evals=10,
-        )
+#class TestSuggestAtpe(unittest.TestCase, NonCategoricalCasePerDomain):
+#    def work(self):
+#        trials = Trials()
+#        fmin(
+#            passthrough,
+#            space=self.bandit.expr,
+#            algo=atpe.suggest,
+#            trials=trials,
+#            max_evals=10,
+#        )
 
 
 class TestOpt(unittest.TestCase, CasePerDomain):
--- contrib/python/hyperopt/hyperopt/tests/unit/test_rand.py	(index)
+++ contrib/python/hyperopt/hyperopt/tests/unit/test_rand.py	(working tree)
@@ -1,7 +1,7 @@
 import unittest
 from hyperopt.base import Trials, trials_from_docs, miscs_to_idxs_vals
 from hyperopt import rand
-from hyperopt.tests.test_base import Suggest_API
+from ..test_base import Suggest_API
 from .test_domains import gauss_wave2, coin_flip
 import numpy as np
 
