--- django-6/django/utils/autoreload.py
+++ django-6/django/utils/autoreload.py
@@ -267,7 +267,7 @@ def trigger_reload(filename):
 
 
 def restart_with_reloader():
-    new_environ = {**os.environ, DJANGO_AUTORELOAD_ENV: "true"}
+    new_environ = {**os.environ, DJANGO_AUTORELOAD_ENV: "true", "Y_PYTHON_ENTRY_POINT": ":main"}
     orig = getattr(sys, "orig_argv", ())
     if any(
         (arg == "-u")
