--- a/src/geomgraph/index/SimpleMCSweepLineIntersector.cpp	(index)
+++ b/src/geomgraph/index/SimpleMCSweepLineIntersector.cpp	(working tree)
@@ -101,7 +101,7 @@ SimpleMCSweepLineIntersector::prepareEvents()
         events.push_back(&e);
     }
 
-    sort(events.begin(), events.end(), SweepLineEventLessThen());
+    std::sort(events.begin(), events.end(), SweepLineEventLessThen());
     for(size_t i = 0; i < events.size(); ++i) {
         GEOS_CHECK_FOR_INTERRUPTS();
         auto& ev = events[i];
