--- a/include/geos/operation/buffer/BufferParameters.h	(index)
+++ b/include/geos/operation/buffer/BufferParameters.h	(working tree)
@@ -92,7 +92,7 @@ public:
     /// For a max error of < 1%, use QS = 12.
     /// For a max error of < 0.1%, use QS = 18.
     ///
-    static const int DEFAULT_QUADRANT_SEGMENTS = 8;
+    static constexpr int DEFAULT_QUADRANT_SEGMENTS = 8;
 
     /// The default mitre limit
     //
