--- contrib/libs/geos/src/geom/GeometryFactory.cpp	(index)
+++ contrib/libs/geos/src/geom/GeometryFactory.cpp	(working tree)
@@ -570,7 +570,7 @@ GeometryFactory::createMultiPoint(const std::vector<Coordinate>& fromCoords) con
 std::unique_ptr<Polygon>
 GeometryFactory::createPolygon(std::size_t coordinateDimension) const
 {
-    auto cs = coordinateListFactory->create(std::size_t(0), coordinateDimension);
+    auto cs = DefaultCoordinateSequenceFactory::instance()->create(std::size_t(0), coordinateDimension);
     auto lr = createLinearRing(cs.release());
     return std::unique_ptr<Polygon>(createPolygon(lr, nullptr));
 }
