--- contrib/libs/geos/src/geom/LineSegment.cpp	(index)
+++ contrib/libs/geos/src/geom/LineSegment.cpp	(working tree)
@@ -176,6 +176,13 @@ LineSegment::equalsTopo(const LineSegment& other) const
     return (p0 == other.p0 && p1 == other.p1) || (p0 == other.p1 && p1 == other.p0);
 }
 
+Coordinate LineSegment::midPoint(const Coordinate& pt0, const Coordinate& pt1)
+{
+    return Coordinate(
+        (pt0.x + pt1.x) / 2,
+        (pt0.y + pt1.y) / 2);  
+}
+
 /*public*/
 int
 LineSegment::orientationIndex(const LineSegment& seg) const
--- contrib/libs/geos/include/geos/geom/LineSegment.h	(index)
+++ contrib/libs/geos/include/geos/geom/LineSegment.h	(working tree)
@@ -61,6 +61,9 @@ public:
 
     friend std::ostream& operator<< (std::ostream& o, const LineSegment& l);
 
+    /// Computes the midpoint of the segment
+    static Coordinate midPoint(const Coordinate& pt0, const Coordinate& pt1);
+
     Coordinate p0; /// Segment start
 
     Coordinate p1; /// Segment end
@@ -155,10 +158,7 @@ public:
     double angle() const;
 
     /// Computes the midpoint of the segment
-    //
-    /// @param ret will be set to the midpoint of the segment
-    ///
-    void midPoint(Coordinate& ret) const;
+    Coordinate midPoint() const;
 
     /// Computes the distance between this line segment and another one.
     double distance(const LineSegment& ls) const;
--- contrib/libs/geos/include/geos/geom/LineSegment.inl	(index)
+++ contrib/libs/geos/include/geos/geom/LineSegment.inl	(working tree)
@@ -165,11 +165,10 @@ LineSegment::angle() const
     return std::atan2(p1.y - p0.y, p1.x - p0.x);
 }
 
-INLINE void
-LineSegment::midPoint(Coordinate& ret) const
+INLINE Coordinate
+LineSegment::midPoint() const
 {
-    ret = Coordinate((p0.x + p1.x) / 2,
-                     (p0.y + p1.y) / 2);
+    return midPoint(p0, p1);
 }
 
 INLINE std::ostream&
--- contrib/libs/geos/src/operation/buffer/OffsetCurveSetBuilder.cpp	(index)
+++ contrib/libs/geos/src/operation/buffer/OffsetCurveSetBuilder.cpp	(working tree)
@@ -40,10 +40,12 @@
 #include <geos/geomgraph/Label.h>
 #include <geos/noding/NodedSegmentString.h>
 #include <geos/util.h>
+#include <geos/io/WKTWriter.h>
 
 #include <algorithm> // for min
 #include <cmath>
 #include <cassert>
+#include <iomanip>
 #include <memory>
 #include <vector>
 #include <typeinfo>
@@ -356,56 +358,65 @@ static const double NEARNESS_FACTOR = 0.99;
 /* private static*/
 bool
 OffsetCurveSetBuilder::isRingCurveInverted(
-    const CoordinateSequence* inputPts, double dist,
-    const CoordinateSequence* curvePts)
+    const CoordinateSequence* inputRing, double dist,
+    const CoordinateSequence* curveRing)
 {
     if (dist == 0.0) return false;
     /**
      * Only proper rings can invert.
      */
-    if (inputPts->size() <= 3) return false;
+    if (inputRing->size() <= 3) return false;
     /**
      * Heuristic based on low chance that a ring with many vertices will invert.
      * This low limit ensures this test is fairly efficient.
      */
-    if (inputPts->size() >= MAX_INVERTED_RING_SIZE) return false;
+    if (inputRing->size() >= MAX_INVERTED_RING_SIZE) return false;
 
     /**
      * An inverted curve has no more points than the input ring.
      * This also eliminates concave inputs (which will produce fillet arcs)
      */
-    if (curvePts->size() > INVERTED_CURVE_VERTEX_FACTOR * inputPts->size()) return false;
+    if (curveRing->size() > INVERTED_CURVE_VERTEX_FACTOR * inputRing->size()) return false;
 
     /**
-     * Check if the curve vertices are all closer to the input ring
-     * than the buffer distance.
-     * If so, the curve is NOT a valid buffer curve.
+     * If curve contains points which are on the buffer, 
+     * it is not inverted and can be included in the raw curves.
      */
-    double distTol = NEARNESS_FACTOR * fabs(dist);
-    double maxDist = maxDistance(curvePts, inputPts);
-    bool isCurveTooClose = maxDist < distTol;
-    return isCurveTooClose;
+    if (hasPointOnBuffer(inputRing, dist, curveRing))
+      return false;
+
+    //-- curve is inverted, so discard it
+    return true;
 }
 
-/**
- * Computes the maximum distance out of a set of points to a linestring.
- *
- * @param pts the points
- * @param line the linestring vertices
- * @return the maximum distance
- */
 /* private static */
-double
-OffsetCurveSetBuilder::maxDistance(const CoordinateSequence*  pts, const CoordinateSequence*  line) {
-    double maxDistance = 0;
-    for (size_t i = 0; i < pts->size(); i++) {
-        Coordinate p = pts->getAt(i);
-        double dist = Distance::pointToSegmentString(p, line);
-        if (dist > maxDistance) {
-        maxDistance = dist;
+bool
+OffsetCurveSetBuilder::hasPointOnBuffer(
+    const CoordinateSequence* inputRing, double dist, 
+    const CoordinateSequence* curveRing) 
+{
+    double distTol = NEARNESS_FACTOR * fabs(dist);
+
+    for (std::size_t i = 0; i < curveRing->size(); i++) {
+        const Coordinate& v = curveRing->getAt(i);
+
+        //-- check curve vertices
+        double distVertex = Distance::pointToSegmentString(v, inputRing);
+        if (distVertex > distTol) {
+            return true; 
+        }
+
+        //-- check curve segment midpoints
+        std::size_t iNext = (i < curveRing->size() - 1) ? i + 1 : 0;
+        const Coordinate& vnext = curveRing->getAt(iNext);
+        Coordinate midPt = LineSegment::midPoint(v, vnext);
+
+        double distMid = Distance::pointToSegmentString(midPt, inputRing);
+        if (distMid > distTol) {
+            return true; 
         }
     }
-    return maxDistance;
+    return false;
 }
 
 /*private*/
--- contrib/libs/geos/include/geos/operation/buffer/OffsetCurveSetBuilder.h	(index)
+++ contrib/libs/geos/include/geos/operation/buffer/OffsetCurveSetBuilder.h	(working tree)
@@ -166,14 +166,19 @@ private:
         const geom::CoordinateSequence* curvePts);
 
     /**
-     * Computes the maximum distance out of a set of points to a linestring.
-     *
-     * @param pts the points
-     * @param line the linestring vertices
-     * @return the maximum distance
+     * Tests if there are points on the raw offset curve which may
+     * lie on the final buffer curve
+     * (i.e. they are (approximately) at the buffer distance from the input ring). 
+     * For efficiency this only tests a limited set of points on the curve.
+     * 
+     * @param inputRing
+     * @param distance
+     * @param curveRing
+     * @return true if the curve contains points lying at the required buffer distance
      */
-    static double maxDistance(
-        const geom::CoordinateSequence*  pts, const geom::CoordinateSequence*  line);
+    static bool hasPointOnBuffer(
+        const geom::CoordinateSequence* inputRing, double dist,
+        const geom::CoordinateSequence* curveRing);
 
     /**
      * The ringCoord is assumed to contain no repeated points.
