--- a/src/operation/intersection/RectangleIntersectionBuilder.cpp	(index)
+++ b/src/operation/intersection/RectangleIntersectionBuilder.cpp	(working tree)
@@ -185,6 +185,12 @@ RectangleIntersectionBuilder::build()
     }
     points.clear();
 
+    // If source geometry is GeometryCollection, SRID for geometry parts may be lost after clipping.
+    // And when clipping result is single geometry GeometryFactory::buildGeometry()
+    // doesn't set SRID. Factory just returns passed pointer.
+    if (geoms->size() == 1)
+        (*geoms)[0]->setSRID((*geoms)[0]->getFactory()->getSRID());
+
     return std::unique_ptr<Geometry>(
                (*geoms)[0]->getFactory()->buildGeometry(geoms)
            );
