--- a/include/geos/operation/overlay/PolygonBuilder.h       (index)
+++ b/include/geos/operation/overlay/PolygonBuilder.h       (working tree)
@@ -147,11 +147,6 @@ private:
                             std::vector<geomgraph::EdgeRing*>& newShellList,
                             std::vector<geomgraph::EdgeRing*>& freeHoleList);
 
-    struct FastPIPRing {
-        geomgraph::EdgeRing* edgeRing;
-        algorithm::locate::IndexedPointInAreaLocator* pipLocator;
-    };
-
     /** \brief
      * This method determines finds a containing shell for all holes
      * which have not yet been assigned to a shell.
@@ -166,7 +161,7 @@ private:
      *
      * @throws TopologyException if a hole cannot be assigned to a shell
      */
-    void placeFreeHoles(std::vector<FastPIPRing>& newShellList,
+    void placeFreeHoles(std::vector<geomgraph::EdgeRing*>& newShellList,
                         std::vector<geomgraph::EdgeRing*>& freeHoleList);
     // throw(const TopologyException&)
 
@@ -189,7 +184,7 @@ private:
      * @return NULL if no containing geomgraph::EdgeRing is found
      */
     geomgraph::EdgeRing* findEdgeRingContaining(geomgraph::EdgeRing* testEr,
-            std::vector<FastPIPRing>& newShellList);
+            std::vector<geomgraph::EdgeRing*>& newShellList);
 
     std::vector<geom::Geometry*>* computePolygons(
         std::vector<geomgraph::EdgeRing*>& newShellList);
--- a/src/operation/overlay/PolygonBuilder.cpp       (index)
+++ b/src/operation/overlay/PolygonBuilder.cpp       (working tree)
@@ -115,17 +115,8 @@ PolygonBuilder::add(const vector<DirectedEdge*>* dirEdges,
 
     sortShellsAndHoles(edgeRings, shellList, freeHoleList);
 
-    vector<FastPIPRing> indexedshellist;
-    for(auto const& shell : shellList) {
-        FastPIPRing pipRing { shell, new geos::algorithm::locate::IndexedPointInAreaLocator(*shell->getLinearRing()) };
-        indexedshellist.push_back(pipRing);
-    }
-    placeFreeHoles(indexedshellist, freeHoleList);
+    placeFreeHoles(shellList, freeHoleList);
     //Assert: every hole on freeHoleList has a shell assigned to it
-
-    for(auto const& shell : indexedshellist) {
-        delete shell.pipLocator;
-    }
 }
 
 /*public*/
@@ -278,7 +269,7 @@ PolygonBuilder::sortShellsAndHoles(vector<MaximalEdgeRing*>& edgeRings,
 
 /*private*/
 void
-PolygonBuilder::placeFreeHoles(vector<FastPIPRing>& newShellList,
+PolygonBuilder::placeFreeHoles(vector<EdgeRing*>& newShellList,
                                std::vector<EdgeRing*>& freeHoleList)
 {
     for(std::vector<EdgeRing*>::iterator
@@ -320,7 +311,7 @@ PolygonBuilder::placeFreeHoles(vector<FastPIPRing>& newShellList,
 /*private*/
 EdgeRing*
 PolygonBuilder::findEdgeRingContaining(EdgeRing* testEr,
-                                       vector<FastPIPRing>& newShellList)
+                                       vector<EdgeRing*>& newShellList)
 {
     LinearRing* testRing = testEr->getLinearRing();
     const Envelope* testEnv = testRing->getEnvelopeInternal();
@@ -328,7 +319,7 @@ PolygonBuilder::findEdgeRingContaining(EdgeRing* testEr,
     const Envelope* minShellEnv = nullptr;
 
     for(auto const& tryShell : newShellList) {
-        LinearRing* tryShellRing = tryShell.edgeRing->getLinearRing();
+        LinearRing* tryShellRing = tryShell->getLinearRing();
         const Envelope* tryShellEnv = tryShellRing->getEnvelopeInternal();
         // the hole envelope cannot equal the shell envelope
         // (also guards against testing rings against themselves)
@@ -344,7 +335,7 @@ PolygonBuilder::findEdgeRingContaining(EdgeRing* testEr,
         const Coordinate& testPt = operation::polygonize::EdgeRing::ptNotInList(testRing->getCoordinatesRO(), tsrcs);
 
         bool isContained = false;
-        if(tryShell.pipLocator->locate(&testPt) != Location::EXTERIOR) {
+        if(PointLocation::isInRing(testPt, tsrcs)) {
             isContained = true;
         }
 
@@ -353,7 +344,7 @@ PolygonBuilder::findEdgeRingContaining(EdgeRing* testEr,
         if(isContained) {
             if(minShell == nullptr
                     || minShellEnv->contains(tryShellEnv)) {
-                minShell = tryShell.edgeRing;
+                minShell = tryShell;
                 minShellEnv = minShell->getLinearRing()->getEnvelopeInternal();
             }
         }
