--- contrib/python/datadiff/datadiff/tests/test_datadiff.py	(index)
+++ contrib/python/datadiff/datadiff/tests/test_datadiff.py	(working tree)
@@ -1,5 +1,6 @@
 import re
 import itertools
+import sys
 from textwrap import dedent
 from datetime import datetime
 from unittest.case import SkipTest
@@ -391,6 +392,7 @@ def test_DataDiff_change_type():
     pytest.raises(Exception, str, dd)
 
 
+@pytest.mark.xfail(sys.version_info >= (3, 12), reason="slice is hashable in Python 3.12")
 def test_unhashable_type():
     a = []
     b = [slice(1)]
