--- contrib/python/tinyhtml5/tests/test_encoding.py	(index)
+++ contrib/python/tinyhtml5/tests/test_encoding.py	(working tree)
@@ -7,9 +7,10 @@ from tinyhtml5.parser import HTMLParser
 
 from . import Data
 
+import yatest.common as yc
 _tests = tuple(
     (f"{path.stem}-{i}", test)
-    for path in (Path(__file__).parent / "encoding").glob("*.dat")
+    for path in (Path(yc.source_path(__file__)).parent / "encoding").glob("*.dat")
     for i, test in enumerate(Data(path, b"data", encoding=None)))
 
 
--- contrib/python/tinyhtml5/tests/test_tokenizer.py	(index)
+++ contrib/python/tinyhtml5/tests/test_tokenizer.py	(working tree)
@@ -122,10 +122,10 @@ def unescape(test):
         token[1] = decode(token[1])
     return test
 
-
+import yatest.common as yc
 _tests = tuple(
     (f"{path.stem}-{i}", test)
-    for path in (Path(__file__).parent / "tokenizer").glob("*.test")
+    for path in (Path(yc.source_path(__file__)).parent / "tokenizer").glob("*.test")
     for i, test in enumerate(json.loads(path.read_text())["tests"]))
 _xfails = (
     "domjs-4",
--- contrib/python/tinyhtml5/tests/test_tree_construction.py	(index)
+++ contrib/python/tinyhtml5/tests/test_tree_construction.py	(working tree)
@@ -43,9 +43,10 @@ def serialize(item, indent=0, skip=False):
             yield f'{"  " * indent}"{child.tail}"'
 
 
+import yatest.common as yc
 _tests = tuple(
     (f"{path.stem}-{i}", test)
-    for path in (Path(__file__).parent / "tree-construction").glob("*.dat")
+    for path in (Path(yc.source_path(__file__)).parent / "tree-construction").glob("*.dat")
     for i, test in enumerate(Data(path)) if "script-on" not in test)
 _xfails = (
     "adoption01-17",
