--- contrib/python/pyjavaproperties/pyjavaproperties_test.py	(index)
+++ contrib/python/pyjavaproperties/pyjavaproperties_test.py	(working tree)
@@ -15,7 +15,8 @@ class PyJavaPropertiesTest(unittest.TestCase):
   """Tests pyjavaproperties complies to java.util.Properties contract."""
 
   def setUp(self):
-    test_dir = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'testdata')
+    import yatest.common as yc
+    test_dir = os.path.join(yc.source_path(os.path.dirname(__file__)), 'tests', 'testdata')
     self.properties_file = os.path.join(test_dir, 'complex.properties')
 
   def testParsePropertiesInput(self):
