--- contrib/python/pyjavaproperties/pyjavaproperties.py	(index)
+++ contrib/python/pyjavaproperties/pyjavaproperties.py	(working tree)
@@ -41,7 +41,7 @@ class Properties(object):
         self.refre =  re.compile('{.+?}')
         # Debug flag
         self._verbose = verbose
-        
+
     def __str__(self):
         s='{'
         for key,value in self._props.items():
@@ -52,7 +52,7 @@ class Properties(object):
 
     def _debug(self, msg):
         print(msg) if self._verbose else None
-            
+
     def __parse(self, lines):
         """ Parse a list of lines and create
         an internal property dictionary """
@@ -91,7 +91,7 @@ class Properties(object):
 
         lineno=0
         i = iter(lines)
-        
+
         for line in i:
             lineno += 1
             line = line.strip()
@@ -124,7 +124,7 @@ class Properties(object):
                     # wspacere a bit, not to look for
                     # : or = characters.
                     wspacere = self.wspacere2
-                    
+
                 start, end = 0, len(line)
 
             m2 = wspacere.search(line, start, end)
@@ -189,7 +189,7 @@ class Properties(object):
         # Patch from N B @ ActiveState
         refs = self.refre.findall(value)
         # To take care of referenced properties
-        
+
         for ref in refs:
             ref_key = ref[1:-1]
             if ref_key not in self._props:
@@ -200,7 +200,7 @@ class Properties(object):
             ref_val = self._props[ref_key]
             self._debug('Overwriting value with referenced key {} value => {}'.format(ref_key, ref_val))
             value = value.replace(ref, ref_val, 1)
-                    
+
         self._props[key] = value.strip()
 
         # Check if an entry exists in pristine keys
@@ -220,17 +220,17 @@ class Properties(object):
         # Java escapes the '=' and ':' in the value
         # string with backslashes in the store method.
         # So let us do the same.
-        newvalue = value.replace(':','\:')
-        newvalue = newvalue.replace('=','\=')
+        newvalue = value.replace(':','\\:')
+        newvalue = newvalue.replace('=','\\=')
 
         return newvalue
 
     def unescape(self, value):
         """ Unescaping some chars """
-        
+
         # Reverse of escape
-        newvalue = value.replace('\:',':')
-        newvalue = newvalue.replace('\=','=')
+        newvalue = value.replace('\\:',':')
+        newvalue = newvalue.replace('\\=','=')
 
         return newvalue
 
@@ -279,7 +279,7 @@ class Properties(object):
             if timestamp:
                 tstamp = time.strftime('%a %b %d %H:%M:%S %Z %Y', time.localtime())
                 out.write(''.join(('#',tstamp,'\n')))
-                
+
             # Write properties from the pristine dictionary
             for prop in self._keyorder:
                 if prop in self._origprops:
@@ -288,7 +288,7 @@ class Properties(object):
 
     def __contains__(self, key):
         return key in self._props
-    
+
     def getPropertyDict(self):
         return self._props
 
