--- contrib/python/PyWavelets/pywt/_pytest.py	(index)
+++ contrib/python/PyWavelets/pywt/_pytest.py	(working tree)
@@ -47,1 +46,2 @@ if 'PYWT_XSLOW' in os.environ:
-    data_dir = os.path.join(os.path.dirname(__file__), 'tests', 'data')
+    import yatest.common as yc
+    data_dir = os.path.join(os.path.dirname(yc.source_path(__file__)), 'tests', 'data')
--- contrib/python/PyWavelets/pywt/tests/test_data.py	(index)
+++ contrib/python/PyWavelets/pywt/tests/test_data.py	(working tree)
@@ -3,1 +3,2 @@ import numpy as np
-data_dir = os.path.join(os.path.dirname(__file__), 'data')
+import yatest.common as yc
+data_dir = os.path.join(os.path.dirname(yc.source_path(__file__)), 'data')
