--- contrib/python/PyWavelets/pywt/_c99_config.py	(index)
+++ contrib/python/PyWavelets/pywt/_c99_config.py	(working tree)
@@ -3 +3 @@
-_have_c99_complex = @have_c99_complex@
+_have_c99_complex = True
--- contrib/python/PyWavelets/pywt/_extensions/config.pxi	(index)
+++ contrib/python/PyWavelets/pywt/_extensions/config.pxi	(working tree)
@@ -3 +3 @@
-DEF HAVE_C99_CPLX = @have_c99_complex_cy@
+DEF HAVE_C99_CPLX = 1
