--- contrib/python/Shapely/py3/shapely/tests/geometry/test_multilinestring.py	(index)
+++ contrib/python/Shapely/py3/shapely/tests/geometry/test_multilinestring.py	(working tree)
@@ -4,7 +4,7 @@ import pytest
 from shapely import LineString, MultiLineString
 from shapely.errors import EmptyPartError
 from shapely.geometry.base import dump_coords
-from shapely.tests.geometry.test_multi import MultiGeometryTestCase
+from .test_multi import MultiGeometryTestCase
 
 
 class TestMultiLineString(MultiGeometryTestCase):
--- contrib/python/Shapely/py3/shapely/tests/geometry/test_multipoint.py	(index)
+++ contrib/python/Shapely/py3/shapely/tests/geometry/test_multipoint.py	(working tree)
@@ -4,7 +4,7 @@ import pytest
 from shapely import MultiPoint, Point
 from shapely.errors import EmptyPartError
 from shapely.geometry.base import dump_coords
-from shapely.tests.geometry.test_multi import MultiGeometryTestCase
+from .test_multi import MultiGeometryTestCase
 
 
 class TestMultiPoint(MultiGeometryTestCase):
--- contrib/python/Shapely/py3/shapely/tests/geometry/test_multipolygon.py	(index)
+++ contrib/python/Shapely/py3/shapely/tests/geometry/test_multipolygon.py	(working tree)
@@ -3,7 +3,7 @@ import pytest
 
 from shapely import MultiPolygon, Polygon
 from shapely.geometry.base import dump_coords
-from shapely.tests.geometry.test_multi import MultiGeometryTestCase
+from .test_multi import MultiGeometryTestCase
 
 
 class TestMultiPolygon(MultiGeometryTestCase):
--- contrib/python/Shapely/py3/shapely/tests/test_strtree.py    (index)
+++ contrib/python/Shapely/py3/shapely/tests/test_strtree.py    (working tree)
@@ -120,6 +120,7 @@ def test_geometries_property():
     assert_geometries_equal(point, tree.geometries[0])
 
 
+@pytest.mark.skip(reason='Arcadia')
 def test_pickle_persistence(tmp_path):
     # write the pickeled tree to another process; the process should not crash
     tree = STRtree([Point(i, i).buffer(0.1) for i in range(3)])
