--- contrib/python/pydash/py3/tests/test_objects.py	(index)
+++ contrib/python/pydash/py3/tests/test_objects.py	(working tree)
@@ -14,1 +14,1 @@ from . import fixtures
-today = dt.date.today()
+today = 'today'
@@ -816,2 +816,5 @@ def test_to_pairs(case, expected):
 def test_to_string(case, expected):
+    if case is today:
+        case = dt.date.today()
+        expected = str(case)
     assert _.to_string(case) == expected
