commit 4df93b60d1ffcaaf18f59c33ca4db138cb6ed769 (HEAD -> SK-5630-sdk)
author: eninng
date: 2021-11-29T16:50:25+03:00

    [jsoncpp] TString support
    
    Task: SK-5630

--- a/include/json/value.h	(61d17e42589ccb46237fee71e44c9dc135cb0513)
+++ b/include/json/value.h	(4df93b60d1ffcaaf18f59c33ca4db138cb6ed769)
@@ -46,6 +46,8 @@
 #include <string>
 #include <vector>
 
+#include <util/generic/string.h>
+
 // Disable warning C4251: <data member>: <type> needs to have dll-interface to
 // be used by...
 #if defined(JSONCPP_DISABLE_DLL_INTERFACE_WARNING)
@@ -341,6 +343,7 @@ public:
    */
   Value(const StaticString& value);
   Value(const String& value);
+  Value(const ::TString& value);
   Value(bool value);
   Value(std::nullptr_t ptr) = delete;
   Value(const Value& other);
--- a/src/lib_json/json_value.cpp	(61d17e42589ccb46237fee71e44c9dc135cb0513)
+++ b/src/lib_json/json_value.cpp	(4df93b60d1ffcaaf18f59c33ca4db138cb6ed769)
@@ -418,6 +418,12 @@ Value::Value(const String& value) {
       value.data(), static_cast<unsigned>(value.length()));
 }
 
+Value::Value(const ::TString& value) {
+  initBasic(stringValue, true);
+  value_.string_ = duplicateAndPrefixStringValue(
+      value.data(), static_cast<unsigned>(value.length()));
+}
+
 Value::Value(const StaticString& value) {
   initBasic(stringValue);
   value_.string_ = const_cast<char*>(value.c_str());
