--- contrib/restricted/boost/odeint/include/boost/numeric/odeint/stepper/runge_kutta_dopri5.hpp (index)
+++ contrib/restricted/boost/odeint/include/boost/numeric/odeint/stepper/runge_kutta_dopri5.hpp (working tree)
@@ -84,7 +84,17 @@ public :
     #endif // DOXYGEN_SKIP
 
 
-    runge_kutta_dopri5( const algebra_type &algebra = algebra_type() ) : stepper_base_type( algebra )
+    runge_kutta_dopri5( const algebra_type &algebra = algebra_type() )
+        : stepper_base_type( algebra )
+        , m_x_tmp()
+        , m_k2()
+        , m_k3()
+        , m_k4()
+        , m_k5()
+        , m_k6()
+        , m_dxdt_tmp()
+        , m_k_x_tmp_resizer()
+        , m_dxdt_tmp_resizer()
     { }
 
 
--- contrib/restricted/boost/odeint/include/boost/numeric/odeint/stepper/base/explicit_error_stepper_fsal_base.hpp      (index)
+++ contrib/restricted/boost/odeint/include/boost/numeric/odeint/stepper/base/explicit_error_stepper_fsal_base.hpp      (working tree)
@@ -94,7 +94,7 @@ public:
     static const order_type error_order_value = ErrorOrder;
 
     explicit_error_stepper_fsal_base( const algebra_type &algebra = algebra_type() )
-    : algebra_stepper_base_type( algebra ) , m_first_call( true )
+    : algebra_stepper_base_type( algebra ) , m_first_call( true ) , m_dxdt()
     { }
 
     order_type order( void ) const

