--- contrib/restricted/boost/odeint/include/boost/numeric/odeint/stepper/generation/generation_controlled_adams_bashforth_moulton.hpp   (index)
+++ contrib/restricted/boost/odeint/include/boost/numeric/odeint/stepper/generation/generation_controlled_adams_bashforth_moulton.hpp   (working tree)
@@ -40,13 +40,13 @@ struct controller_factory< Stepper , controlled_adams_bashforth_moulton< Stepper
     typedef typename stepper_type::value_type value_type;
     typedef typename stepper_type::value_type time_type;
 
-    controller_type operator()( value_type abs_error , value_type rel_error , const stepper_type &stepper )
+    controller_type operator()( value_type abs_error , value_type rel_error , const stepper_type & /*stepper*/ )
     {
         return controller_type(step_adjuster_type(abs_error, rel_error));
     }
 
     controller_type operator()( value_type abs_error , value_type rel_error ,
-                                time_type max_dt, const stepper_type &stepper )
+                                time_type max_dt, const stepper_type & /*stepper*/ )
     {
         return controller_type( step_adjuster_type(abs_error, rel_error, max_dt));
     }
@@ -56,4 +56,4 @@ struct controller_factory< Stepper , controlled_adams_bashforth_moulton< Stepper
 }
 }
 
-#endif
\ No newline at end of file
+#endif
