--- contrib/python/python-pptx/py3/pptx/api.py	(index)
+++ contrib/python/python-pptx/py3/pptx/api.py	(working tree)
@@ -42,2 +42,2 @@ def _default_pptx_path() -> str:
-    _thisdir = os.path.split(__file__)[0]
-    return os.path.join(_thisdir, "templates", "default.pptx")
+    import importlib.resources
+    return (importlib.resources.files("pptx") / "templates" / "default.pptx").open(mode="rb")
--- contrib/python/python-pptx/py3/pptx/oxml/__init__.py	(index)
+++ contrib/python/python-pptx/py3/pptx/oxml/__init__.py	(working tree)
@@ -27,3 +27,3 @@ def parse_from_template(template_file_name: str):
-    thisdir = os.path.split(__file__)[0]
-    filename = os.path.join(thisdir, "..", "templates", "%s.xml" % template_file_name)
-    with open(filename, "rb") as f:
+    import importlib.resources
+    resource = importlib.resources.files("pptx") / "templates" / f"{template_file_name}.xml"
+    with resource.open(mode="rb") as f:
--- contrib/python/python-pptx/py3/pptx/shapes/shapetree.py	(index)
+++ contrib/python/python-pptx/py3/pptx/shapes/shapetree.py	(working tree)
@@ -1139,2 +1139,2 @@ class _OleObjectElementCreator(object):
-        _thisdir = os.path.split(__file__)[0]
-        return os.path.abspath(os.path.join(_thisdir, "..", "templates", icon_filename))
+        import importlib.resources
+        return (importlib.resources.files("pptx") / "templates" / icon_filename).open(mode="rb")
