--- contrib/python/latex2mathml/latex2mathml/symbols_parser.py	(e9102d6ef8ce421122c8182f7813b0fa5d34bd6f)
+++ contrib/python/latex2mathml/latex2mathml/symbols_parser.py	(index)
@@ -1,9 +1,9 @@
 import codecs
-import os
+import importlib.resources
 import re
 from typing import Optional, Union
 
-SYMBOLS_FILE: str = os.path.join(os.path.dirname(os.path.realpath(__file__)), "unimathsymbols.txt")
+SYMBOLS_FILE: str = importlib.resources.files(__package__) / "unimathsymbols.txt"
 SYMBOLS: Optional[dict[str, str]] = None
 
 
@@ -16,7 +16,7 @@ def convert_symbol(symbol: str) -> Union[str, None]:
 
 def parse_symbols() -> dict[str, str]:
     _symbols: dict[str, str] = {}
-    with codecs.open(SYMBOLS_FILE, encoding="utf-8") as f:
+    with SYMBOLS_FILE.open(encoding="utf-8") as f:
         for line in f:
             if line.startswith("#"):
                 continue
