--- contrib/libs/clang18-rt/lib/asan/asan_flags.inc	(index)
+++ contrib/libs/clang18-rt/lib/asan/asan_flags.inc	(working tree)
@@ -147,7 +147,7 @@ ASAN_FLAG(bool, detect_container_overflow, true,
           "If true, honor the container overflow annotations. See "
           "https://github.com/google/sanitizers/wiki/"
           "AddressSanitizerContainerOverflow")
-ASAN_FLAG(int, detect_odr_violation, 2,
+ASAN_FLAG(int, detect_odr_violation, 0,
           "If >=2, detect violation of One-Definition-Rule (ODR); "
           "If ==1, detect ODR-violation only if the two variables "
           "have different sizes")
