--- contrib/python/datamodel-code-generator/tests/test_infer_input_type.py	(index)
+++ contrib/python/datamodel-code-generator/tests/test_infer_input_type.py	(working tree)
@@ -2,1 +2,2 @@
-DATA_PATH: Path = Path(__file__).parent / "data"
+import yatest.common as yc
+DATA_PATH: Path = Path(yc.source_path(__file__)).parent / "data"
--- contrib/python/datamodel-code-generator/tests/test_main_kr.py	(index)
+++ contrib/python/datamodel-code-generator/tests/test_main_kr.py	(working tree)
@@ -13,1 +13,2 @@
-DATA_PATH: Path = Path(__file__).parent / "data"
+import yatest.common as yc
+DATA_PATH: Path = Path(yc.source_path(__file__)).parent / "data"
--- contrib/python/datamodel-code-generator/tests/parser/test_jsonschema.py	(index)
+++ contrib/python/datamodel-code-generator/tests/parser/test_jsonschema.py	(working tree)
@@ -16,1 +16,2 @@
-DATA_PATH: Path = Path(__file__).parents[1] / "data" / "jsonschema"
+import yatest.common as yc
+DATA_PATH: Path = Path(yc.source_path(__file__)).parents[1] / "data" / "jsonschema"
@@ -18,1 +18,1 @@
-EXPECTED_JSONSCHEMA_PATH = Path(__file__).parents[1] / "data" / "expected" / "parser" / "jsonschema"
+EXPECTED_JSONSCHEMA_PATH = Path(yc.source_path(__file__)).parents[1] / "data" / "expected" / "parser" / "jsonschema"
--- contrib/python/datamodel-code-generator/tests/parser/test_openapi.py	(index)
+++ contrib/python/datamodel-code-generator/tests/parser/test_openapi.py	(working tree)
@@ -11,1 +11,2 @@
-DATA_PATH: Path = Path(__file__).parents[1] / "data" / "openapi"
+import yatest.common as yc
+DATA_PATH: Path = Path(yc.source_path(__file__)).parents[1] / "data" / "openapi"
@@ -14,1 +14,1 @@
-EXPECTED_OPEN_API_PATH = Path(__file__).parents[1] / "data" / "expected" / "parser" / "openapi"
+EXPECTED_OPEN_API_PATH = Path(yc.source_path(__file__)).parents[1] / "data" / "expected" / "parser" / "openapi"
--- contrib/python/datamodel-code-generator/tests/parser/test_graphql.py	(index)
+++ contrib/python/datamodel-code-generator/tests/parser/test_graphql.py	(working tree)
@@ -1,15 +1,17 @@
 from __future__ import annotations
 
+from pathlib import Path
 from typing import TYPE_CHECKING
 
 from freezegun import freeze_time
 
 from datamodel_code_generator.__main__ import Exit, main
-from tests.main.test_main_general import DATA_PATH
 
 if TYPE_CHECKING:
-    from pathlib import Path
+    pass
 
+import yatest.common as yc
+DATA_PATH: Path = Path(yc.source_path(__file__)).parents[1] / "data"
 GRAPHQL_DATA_PATH: Path = DATA_PATH / "graphql"
 EXPECTED_GRAPHQL_PATH: Path = DATA_PATH / "expected" / "parser" / "graphql"
 
--- contrib/python/datamodel-code-generator/tests/test_format.py	(index)
+++ contrib/python/datamodel-code-generator/tests/test_format.py	(working tree)
@@ -5,1 +5,3 @@ import pytest
-EXAMPLE_LICENSE_FILE = str(Path(__file__).parent / "data/python/custom_formatters/license_example.txt")
+import yatest.common as yc
+sys.path.insert(0, yc.source_path("contrib/python/datamodel-code-generator"))
+EXAMPLE_LICENSE_FILE = str(Path(yc.source_path(__file__)).parent / "data/python/custom_formatters/license_example.txt")
--- contrib/python/datamodel-code-generator/tests/main/graphql/test_main_graphql.py	(index)
+++ contrib/python/datamodel-code-generator/tests/main/graphql/test_main_graphql.py	(working tree)
@@ -11 +11 @@ from datamodel_code_generator.__main__ import Exit, main
-from tests.main.test_main_general import DATA_PATH, EXPECTED_MAIN_PATH
+from __tests__.main.test_main_general import DATA_PATH, EXPECTED_MAIN_PATH
--- contrib/python/datamodel-code-generator/tests/main/jsonschema/test_main_jsonschema.py	(index)
+++ contrib/python/datamodel-code-generator/tests/main/jsonschema/test_main_jsonschema.py	(working tree)
@@ -21 +21 @@ from datamodel_code_generator.__main__ import Exit, main
-from tests.main.test_main_general import DATA_PATH, EXPECTED_MAIN_PATH, TIMESTAMP
+from __tests__.main.test_main_general import DATA_PATH, EXPECTED_MAIN_PATH, TIMESTAMP
--- contrib/python/datamodel-code-generator/tests/main/openapi/test_main_openapi.py	(index)
+++ contrib/python/datamodel-code-generator/tests/main/openapi/test_main_openapi.py	(working tree)
@@ -28 +28 @@ from datamodel_code_generator.__main__ import Exit, main
-from tests.main.test_main_general import DATA_PATH, EXPECTED_MAIN_PATH, TIMESTAMP
+from __tests__.main.test_main_general import DATA_PATH, EXPECTED_MAIN_PATH, TIMESTAMP
--- contrib/python/datamodel-code-generator/tests/main/test_main_csv.py	(index)
+++ contrib/python/datamodel-code-generator/tests/main/test_main_csv.py	(working tree)
@@ -9 +9 @@ from datamodel_code_generator.__main__ import Exit, main
-from tests.main.test_main_general import DATA_PATH, EXPECTED_MAIN_PATH
+from __tests__.main.test_main_general import DATA_PATH, EXPECTED_MAIN_PATH
--- contrib/python/datamodel-code-generator/tests/main/test_main_general.py	(index)
+++ contrib/python/datamodel-code-generator/tests/main/test_main_general.py	(working tree)
@@ -16 +16,2 @@ MonkeyPatch = pytest.MonkeyPatch
-DATA_PATH: Path = Path(__file__).parent.parent / "data"
+import yatest.common as yc
+DATA_PATH: Path = Path(yc.source_path(__file__)).parent.parent / "data"
--- contrib/python/datamodel-code-generator/tests/main/test_main_json.py	(index)
+++ contrib/python/datamodel-code-generator/tests/main/test_main_json.py	(working tree)
@@ -15 +15 @@ from datamodel_code_generator.__main__ import Exit, main
-from tests.main.test_main_general import DATA_PATH, EXPECTED_MAIN_PATH
+from __tests__.main.test_main_general import DATA_PATH, EXPECTED_MAIN_PATH
--- contrib/python/datamodel-code-generator/tests/main/test_main_yaml.py	(index)
+++ contrib/python/datamodel-code-generator/tests/main/test_main_yaml.py	(working tree)
@@ -9 +9 @@ from datamodel_code_generator.__main__ import Exit, main
-from tests.main.test_main_general import DATA_PATH, EXPECTED_MAIN_PATH
+from __tests__.main.test_main_general import DATA_PATH, EXPECTED_MAIN_PATH
--- contrib/python/datamodel-code-generator/tests/main/jsonschema/test_main_jsonschema.py	(index)
+++ contrib/python/datamodel-code-generator/tests/main/jsonschema/test_main_jsonschema.py	(working tree)
@@ -1103 +1103 @@ def test_main_generate() -> None:
-    input_ = (JSON_SCHEMA_DATA_PATH / "person.json").relative_to(Path.cwd())
+    input_ = (JSON_SCHEMA_DATA_PATH / "person.json").relative_to(Path.cwd(), walk_up=True)
@@ -1123 +1123 @@ def test_main_generate_non_pydantic_output() -> None:
-    input_ = (JSON_SCHEMA_DATA_PATH / "simple_string.json").relative_to(Path.cwd())
+    input_ = (JSON_SCHEMA_DATA_PATH / "simple_string.json").relative_to(Path.cwd(), walk_up=True)
@@ -1138 +1138 @@ def test_main_generate_from_directory() -> None:
-    input_ = (JSON_SCHEMA_DATA_PATH / "external_files_in_directory").relative_to(Path.cwd())
+    input_ = (JSON_SCHEMA_DATA_PATH / "external_files_in_directory").relative_to(Path.cwd(), walk_up=True)
@@ -1193 +1193 @@ def test_main_generate_custom_class_name_generator() -> None:
-    input_ = (JSON_SCHEMA_DATA_PATH / "person.json").relative_to(Path.cwd())
+    input_ = (JSON_SCHEMA_DATA_PATH / "person.json").relative_to(Path.cwd(), walk_up=True)
@@ -1296 +1296 @@ def test_main_generate_custom_class_name_generator_additional_properties(
-    input_ = (JSON_SCHEMA_DATA_PATH / "root_model_with_additional_properties.json").relative_to(Path.cwd())
+    input_ = (JSON_SCHEMA_DATA_PATH / "root_model_with_additional_properties.json").relative_to(Path.cwd(), walk_up=True)
