--- contrib/python/datamodel-code-generator/datamodel_code_generator/model/base.py	(index)
+++ contrib/python/datamodel-code-generator/datamodel_code_generator/model/base.py	(working tree)
@@ -20,1 +20,1 @@ from typing import (
-from jinja2 import Environment, FileSystemLoader, Template
+from jinja2 import Environment, PackageLoader, Template
@@ -194,1 +194,2 @@ class DataModelFieldBase(_BaseModel):
-    loader = FileSystemLoader(str(TEMPLATE_DIR / template_file_path.parent))
+    template_dir = Path("template") / template_file_path.parent
+    loader = PackageLoader(__package__, template_dir)
 
