--- contrib/python/Flask-Admin/flask_admin/base.py	(index)
+++ contrib/python/Flask-Admin/flask_admin/base.py	(working tree)
@@ -1,11 +1,16 @@
+import hashlib
 import os.path as op
+import threading
 import warnings
 
 from functools import wraps
 
-from flask import Blueprint, current_app, render_template, abort, g, url_for
+import mimetypes
+from jinja2.loaders import FileSystemLoader, DictLoader
+from flask import Blueprint, current_app, render_template, abort, g, url_for, request, Response
+from flask.helpers import locked_cached_property, send_from_directory
 from flask_admin import babel
-from flask_admin._compat import with_metaclass, as_unicode
+from flask_admin._compat import with_metaclass, as_unicode, iteritems
 from flask_admin import helpers as h
 
 # For compatibility reasons import MenuLink
@@ -102,6 +107,112 @@ class AdminViewMeta(type):
                 setattr(cls, p, _wrap_view(attr))
 
 
+class FlaskAdminBlueprint(Blueprint):
+    RESOURCE_LOCK = threading.Lock()
+    RESOURCE_STATIC_FILES = None
+    RESOURCE_TEMPLATES = None
+
+    KNOWN_MIMETYPES = {
+        'woff2': 'font/woff2',
+        'woff': 'font/woff',
+        'ttf': 'font/ttf',
+    }
+
+    def __init__(self, *args, **kwargs):
+        super(FlaskAdminBlueprint, self).__init__(*args, **kwargs)
+        self._static_resource_dict = {}
+        self._template_resource_dict = {}
+        if self._is_in_arcadia:
+            self._load_files_from_resources()
+            self._static_resource_dict.update(self.RESOURCE_STATIC_FILES)
+            self._template_resource_dict.update(self.RESOURCE_TEMPLATES)
+
+    def register_static_files(self, static_file_dict):
+        for filename, data in iteritems(static_file_dict):
+            self._static_resource_dict[filename] = self._describe_file(filename, data)
+
+    def register_templates(self, template_dict):
+        self._template_resource_dict.update(template_dict)
+
+    @locked_cached_property
+    def jinja_loader(self):
+        """
+            arcadia's method for templates
+            self.templates_folder is like 'templates/bootstrap3'
+        """
+        if self.template_folder is None:
+            return
+
+        if self._is_in_arcadia:
+            template_resource_dict = {}
+            for key, value in iteritems(self._template_resource_dict):
+                if key.startswith(self.template_folder + '/'):
+                    key = key.replace(self.template_folder + '/', '')
+                    template_resource_dict[key] = value
+            return DictLoader(template_resource_dict)
+        else:
+            return FileSystemLoader(op.join(self.root_path, self.template_folder))
+
+    def send_static_file(self, filename):
+        if not self.has_static_folder:
+            raise RuntimeError('No static folder for this object')
+        if self._is_in_arcadia:
+            file_desc = self._static_resource_dict.get(filename)
+            if file_desc is None:
+                abort(404)  # not found
+            if request.if_none_match and file_desc['etag'] in request.if_none_match:
+                return Response(status=304)  # not modified
+            response = Response(file_desc['data'], mimetype=file_desc['mimetype'])
+            response.set_etag(file_desc['etag'])
+            return response
+        else:
+            cache_timeout = self.get_send_file_max_age(filename)
+            return send_from_directory(self.static_folder, filename, cache_timeout=cache_timeout)
+
+    @classmethod
+    def _load_files_from_resources(cls):
+        from library.python import resource
+
+        with cls.RESOURCE_LOCK:
+            if cls.RESOURCE_STATIC_FILES is not None and cls.RESOURCE_TEMPLATES is not None:
+                return
+
+            static_files = {}
+            templates = {}
+            for key, data in resource.iteritems(prefix='resfs/file/contrib/python/Flask-Admin/flask_admin/', strip_prefix=True):
+                if key.startswith('static/'):
+                    filename = key[7:]  # len('static/') == 7
+                    static_files[filename] = cls._describe_file(filename, data)
+
+                if key.startswith('templates/'):
+                    templates[key] = data.decode('utf-8')
+
+            cls.RESOURCE_STATIC_FILES = static_files
+            cls.RESOURCE_TEMPLATES = templates
+
+    @classmethod
+    def _describe_file(cls, filename, data):
+        ext = filename.rpartition('.')[2]
+        if ext in cls.KNOWN_MIMETYPES:
+            mimetype = cls.KNOWN_MIMETYPES[ext]
+        else:
+            mimetype = mimetypes.guess_type(filename)[0]
+
+        return {
+            'data': data,
+            'mimetype': mimetype,
+            'etag': hashlib.sha1(data).hexdigest(),
+        }
+
+    @locked_cached_property
+    def _is_in_arcadia(self):
+        try:
+            from library.python import resource
+            return True
+        except ImportError:
+            return False
+
+
 class BaseViewClass(object):
     pass
 
@@ -262,12 +373,12 @@ class BaseView(with_metaclass(AdminViewMeta, BaseViewClass)):
             self.name = self._prettify_class_name(self.__class__.__name__)
 
         # Create blueprint and register rules
-        self.blueprint = Blueprint(self.endpoint, __name__,
-                                   url_prefix=self.url,
-                                   subdomain=self.admin.subdomain,
-                                   template_folder=op.join('templates', self.admin.template_mode),
-                                   static_folder=self.static_folder,
-                                   static_url_path=self.static_url_path)
+        self.blueprint = FlaskAdminBlueprint(self.endpoint, __name__,
+                                             url_prefix=self.url,
+                                             subdomain=self.admin.subdomain,
+                                             template_folder=op.join('templates', self.admin.template_mode),
+                                             static_folder=self.static_folder,
+                                             static_url_path=self.static_url_path)
 
         for url, name, methods in self._urls:
             self.blueprint.add_url_rule(url,
--- contrib/python/Flask-Admin/flask_admin/templates/bootstrap3/admin/static.html	(index)
+++ contrib/python/Flask-Admin/flask_admin/templates/bootstrap3/admin/static.html	(working tree)
@@ -1,3 +1,3 @@
 {% macro url() -%}
-    {{ get_url('{admin_endpoint}.static'.format(admin_endpoint=admin_view.admin.endpoint), *varargs, **kwargs) }}
+    {{ get_url('admin.static', *varargs, **kwargs) }}
 {%- endmacro %}
