--- contrib/python/wcwidth/py3/tests/test_benchmarks.py	(index)
+++ contrib/python/wcwidth/py3/tests/test_benchmarks.py	(working tree)
@@ -186 +186,2 @@ def test_iter_sequences_mixed(benchmark):
-UDHR_FILE = os.path.join(os.path.dirname(__file__), 'udhr_combined.txt')
+import yatest.common as yc
+UDHR_FILE = os.path.join(os.path.dirname(yc.source_path(__file__)), 'udhr_combined.txt')
--- contrib/python/wcwidth/py3/tests/test_emojis.py	(index)
+++ contrib/python/wcwidth/py3/tests/test_emojis.py	(working tree)
@@ -136 +136,2 @@ def read_sequences_from_file(filename):
-    with open(os.path.join(os.path.dirname(__file__), filename), encoding='utf-8') as fp:
+    import yatest.common as yc
+    with open(os.path.join(os.path.dirname(yc.source_path(__file__)), filename), encoding='utf-8') as fp:
--- contrib/python/wcwidth/py3/tests/test_grapheme.py	(index)
+++ contrib/python/wcwidth/py3/tests/test_grapheme.py	(working tree)
@@ -60 +60,2 @@ def read_grapheme_break_test():
-    test_file = os.path.join(os.path.dirname(__file__), 'GraphemeBreakTest.txt')
+    import yatest.common as yc
+    test_file = os.path.join(os.path.dirname(yc.source_path(__file__)), 'GraphemeBreakTest.txt')
