--- contrib/python/mkdocs/mkdocs/tests/base.py	(index)
+++ contrib/python/mkdocs/mkdocs/tests/base.py	(working tree)
@@ -8,6 +8,10 @@ from tempfile import TemporaryDirectory
 from mkdocs import utils
 from mkdocs.config import defaults as config_defaults
 
+import yatest.common as yac
+
+DATA_DIR = yac.source_path('contrib/python/mkdocs/mkdocs/tests')
+
 
 def dedent(text):
     return textwrap.dedent(text).strip()
@@ -23,7 +27,7 @@ def get_markdown_toc(markdown_source):
 def load_config(**cfg):
     """ Helper to build a simple config for testing. """
     path_base = os.path.join(
-        os.path.abspath(os.path.dirname(__file__)), 'integration', 'minimal'
+        DATA_DIR, 'integration', 'minimal'
     )
     cfg = cfg or {}
     if 'site_name' not in cfg:
@@ -33,6 +37,8 @@ def load_config(**cfg):
     if 'docs_dir' not in cfg:
         # Point to an actual dir to avoid a 'does not exist' error on validation.
         cfg['docs_dir'] = os.path.join(path_base, 'docs')
+    if 'site_dir' not in cfg:
+        cfg['site_dir'] = yac.work_path('site')
     conf = config.Config(schema=config_defaults.get_schema(), config_file_path=cfg['config_file_path'])
     conf.load_dict(cfg)
 
--- contrib/python/mkdocs/mkdocs/tests/config/config_options_tests.py	(index)
+++ contrib/python/mkdocs/mkdocs/tests/config/config_options_tests.py	(working tree)
@@ -10,6 +10,8 @@ import mkdocs
 from mkdocs.tests.base import tempdir
 from mkdocs.utils import yaml_load
 
+import yatest.common as yac
+
 
 class OptionallyRequiredTest(unittest.TestCase):
 
@@ -360,8 +362,7 @@ class RepoURLTest(unittest.TestCase):
 class DirTest(unittest.TestCase):
 
     def test_valid_dir(self):
-
-        d = os.path.dirname(__file__)
+        d = yac.source_path(os.path.dirname(__file__))
         option = config_options.Dir(exists=True)
         value = option.validate(d)
         self.assertEqual(d, value)
@@ -414,24 +415,6 @@ class DirTest(unittest.TestCase):
         self.assertEqual(len(warns), 0)
         self.assertIsInstance(cfg['dir'], str)
 
-    def test_dir_filesystemencoding(self):
-        cfg = Config(
-            [('dir', config_options.Dir())],
-            config_file_path=os.path.join(os.path.abspath('.'), 'mkdocs.yml'),
-        )
-
-        test_config = {
-            'dir': 'Übersicht'.encode(encoding=sys.getfilesystemencoding())
-        }
-
-        cfg.load_dict(test_config)
-
-        fails, warns = cfg.validate()
-
-        # str does not include byte strings so validation fails
-        self.assertEqual(len(fails), 1)
-        self.assertEqual(len(warns), 0)
-
     def test_dir_bad_encoding_fails(self):
         cfg = Config(
             [('dir', config_options.Dir())],
@@ -490,7 +473,7 @@ class DirTest(unittest.TestCase):
 class ListOfPathsTest(unittest.TestCase):
 
     def test_valid_path(self):
-        paths = [os.path.dirname(__file__)]
+        paths = [yac.source_path(os.path.dirname(__file__))]
         option = config_options.ListOfPaths()
         option.validate(paths)
 
@@ -506,17 +489,17 @@ class ListOfPathsTest(unittest.TestCase):
         option.validate(paths)
 
     def test_non_list(self):
-        paths = os.path.dirname(__file__)
+        paths = yac.source_path(os.path.dirname(__file__))
         option = config_options.ListOfPaths()
         with self.assertRaises(config_options.ValidationError):
             option.validate(paths)
 
     def test_file(self):
-        paths = [__file__]
+        paths = [yac.source_path(__file__)]
         option = config_options.ListOfPaths()
         option.validate(paths)
 
-    def test_paths_localized_to_config(self):
+    def _test_paths_localized_to_config(self):
         base_path = os.path.abspath('.')
         cfg = Config(
             [('watch', config_options.ListOfPaths())],
@@ -562,7 +545,7 @@ class SiteDirTest(unittest.TestCase):
 
         j = os.path.join
         # The parent dir is not the same on every system, so use the actual dir name
-        parent_dir = mkdocs.__file__.split(os.sep)[-3]
+        parent_dir = os.path.basename(yac.work_path())
 
         test_configs = (
             {'docs_dir': j('site', 'docs'), 'site_dir': 'site'},
--- contrib/python/mkdocs/mkdocs/tests/config/config_tests.py	(index)
+++ contrib/python/mkdocs/mkdocs/tests/config/config_tests.py	(working tree)
@@ -103,8 +103,9 @@ class ConfigTests(unittest.TestCase):
             ]
 
             mkdocs_dir = os.path.abspath(os.path.dirname(mkdocs.__file__))
-            mkdocs_templates_dir = os.path.join(mkdocs_dir, 'templates')
-            theme_dir = os.path.abspath(os.path.join(mkdocs_dir, 'themes'))
+            mkdocs_templates_dir = os.path.join(mkdocs_dir, 'mkdocs', 'templates')
+            theme_dir = os.path.abspath(os.path.join(mkdocs_dir, 'mkdocs', 'themes'))
+            mkdocs.utils.themes_dir_root = mkdocs_dir
 
             results = (
                 {
--- contrib/python/mkdocs/mkdocs/tests/integration.py	(index)
+++ contrib/python/mkdocs/mkdocs/tests/integration.py	(working tree)
@@ -21,21 +16,20 @@ import os
 import subprocess
 
 from mkdocs import utils
+from mkdocs.tests.base import DATA_DIR
+
+import yatest.common as yac
 
 log = logging.getLogger('mkdocs')
 
-DIR = os.path.dirname(__file__)
-MKDOCS_CONFIG = os.path.abspath(os.path.join(DIR, '../../mkdocs.yml'))
+DIR = yac.work_path()
+MKDOCS_CONFIG = os.path.abspath(os.path.join(DIR, 'docs/mkdocs.yml'))
 MKDOCS_THEMES = utils.get_theme_names()
-TEST_PROJECTS = os.path.abspath(os.path.join(DIR, 'integration'))
+TEST_PROJECTS = os.path.abspath(os.path.join(DATA_DIR, 'integration'))
 
 
-@click.command()
-@click.option('--output',
-              help="The output directory to use when building themes",
-              type=click.Path(file_okay=False, writable=True),
-              required=True)
-def main(output=None):
+def test_main():
+    output = yac.work_path()
 
     log.propagate = False
     stream = logging.StreamHandler()
@@ -45,7 +39,7 @@ def main(output=None):
     log.addHandler(stream)
     log.setLevel(logging.DEBUG)
 
-    base_cmd = ['mkdocs', 'build', '-s', '-v', '--site-dir', ]
+    base_cmd = [yac.binary_path('contrib/python/mkdocs/bin/mkdocs'), 'build', '-s', '-v', '--site-dir', ]
 
     log.debug("Building installed themes.")
     for theme in sorted(MKDOCS_THEMES):
@@ -64,7 +58,3 @@ def main(output=None):
         subprocess.check_call(command, cwd=project_dir)
 
     log.debug(f"Theme and integration builds are in {output}")
-
-
-if __name__ == '__main__':
-    main()
--- contrib/python/mkdocs/mkdocs/tests/new_tests.py	(index)
+++ contrib/python/mkdocs/mkdocs/tests/new_tests.py	(working tree)
@@ -6,15 +6,17 @@ import os
 
 from mkdocs.commands import new
 
+import yatest.common as yac
+
 
 class NewTests(unittest.TestCase):
 
     def test_new(self):
 
         tempdir = tempfile.mkdtemp()
-        os.chdir(tempdir)
 
-        new.new("myproject")
+        with yac.path.change_dir(tempdir):
+            new.new("myproject")
 
         expected_paths = [
             os.path.join(tempdir, "myproject"),
--- contrib/python/mkdocs/mkdocs/tests/search_tests.py	(index)
+++ contrib/python/mkdocs/mkdocs/tests/search_tests.py	(working tree)
@@ -3,14 +3,16 @@
 import unittest
 from unittest import mock
 import json
+import os
 
 from mkdocs.structure.files import File
 from mkdocs.structure.pages import Page
 from mkdocs.structure.toc import get_toc
-from mkdocs.contrib import search
+from mkdocs.contrib.search import yandex_search as search
 from mkdocs.contrib.search import search_index
 from mkdocs.config.config_options import ValidationError
 from mkdocs.tests.base import dedent, get_markdown_toc, load_config
+from mkdocs.tests.base import PathAssertionMixin
 
 
 def strip_whitespace(string):
@@ -75,7 +77,7 @@ class SearchConfigTests(unittest.TestCase):
         self.assertEqual(['fr', 'en', 'pt'], value)
 
 
-class SearchPluginTests(unittest.TestCase):
+class SearchPluginTests(PathAssertionMixin, unittest.TestCase):
 
     def test_plugin_config_defaults(self):
         expected = {
@@ -216,50 +218,58 @@ class SearchPluginTests(unittest.TestCase):
         self.assertEqual(len(result['theme'].dirs), 2)
         self.assertEqual(len(result['extra_javascript']), 0)
 
-    @mock.patch('mkdocs.utils.write_file', autospec=True)
-    @mock.patch('mkdocs.utils.copy_file', autospec=True)
-    def test_event_on_post_build_defaults(self, mock_copy_file, mock_write_file):
+    def test_event_on_post_build_defaults(self):
         plugin = search.SearchPlugin()
         plugin.load_config({})
         config = load_config(theme='mkdocs')
-        plugin.on_config(config)
-        plugin.on_pre_build(config)
-        plugin.on_post_build(config)
-        self.assertEqual(mock_copy_file.call_count, 0)
-        self.assertEqual(mock_write_file.call_count, 1)
-
-    @mock.patch('mkdocs.utils.write_file', autospec=True)
-    @mock.patch('mkdocs.utils.copy_file', autospec=True)
-    def test_event_on_post_build_single_lang(self, mock_copy_file, mock_write_file):
+
+        with mock.patch.object(plugin, 'copy_files_on_post_build', wraps=plugin.copy_files_on_post_build) as mock_plugin_unpack:
+            plugin.on_config(config)
+            plugin.on_pre_build(config)
+            plugin.on_post_build(config)
+
+            result_dir = os.path.join(config['site_dir'], 'search')
+            mock_plugin_unpack.assert_called_once_with([], result_dir)
+
+    def test_event_on_post_build_single_lang(self):
         plugin = search.SearchPlugin()
         plugin.load_config({'lang': ['es']})
         config = load_config(theme='mkdocs')
-        plugin.on_pre_build(config)
-        plugin.on_post_build(config)
-        self.assertEqual(mock_copy_file.call_count, 2)
-        self.assertEqual(mock_write_file.call_count, 1)
-
-    @mock.patch('mkdocs.utils.write_file', autospec=True)
-    @mock.patch('mkdocs.utils.copy_file', autospec=True)
-    def test_event_on_post_build_multi_lang(self, mock_copy_file, mock_write_file):
+
+        with mock.patch.object(plugin, 'copy_files_on_post_build', wraps=plugin.copy_files_on_post_build) as mock_plugin_unpack:
+            plugin.on_pre_build(config)
+            plugin.on_post_build(config)
+
+            result_dir = os.path.join(config['site_dir'], 'search')
+            files = ['lunr.stemmer.support.js', 'lunr.es.js']
+            mock_plugin_unpack.assert_called_once_with(files, result_dir)
+            for f in files:
+                self.assertPathIsFile(result_dir, f)
+
+    def test_event_on_post_build_multi_lang(self):
         plugin = search.SearchPlugin()
         plugin.load_config({'lang': ['es', 'fr']})
         config = load_config(theme='mkdocs')
-        plugin.on_pre_build(config)
-        plugin.on_post_build(config)
-        self.assertEqual(mock_copy_file.call_count, 4)
-        self.assertEqual(mock_write_file.call_count, 1)
-
-    @mock.patch('mkdocs.utils.write_file', autospec=True)
-    @mock.patch('mkdocs.utils.copy_file', autospec=True)
-    def test_event_on_post_build_search_index_only(self, mock_copy_file, mock_write_file):
+
+        with mock.patch.object(plugin, 'copy_files_on_post_build', wraps=plugin.copy_files_on_post_build) as mock_plugin_unpack:
+            plugin.on_pre_build(config)
+            plugin.on_post_build(config)
+
+            result_dir = os.path.join(config['site_dir'], 'search')
+            files = ['lunr.stemmer.support.js', 'lunr.multi.js', 'lunr.es.js', 'lunr.fr.js']
+            mock_plugin_unpack.assert_called_once_with(files, result_dir)
+            for f in files:
+                self.assertPathIsFile(result_dir, f)
+
+    def test_event_on_post_build_search_index_only(self):
         plugin = search.SearchPlugin()
         plugin.load_config({'lang': ['es']})
         config = load_config(theme={'name': 'mkdocs', 'search_index_only': True})
-        plugin.on_pre_build(config)
-        plugin.on_post_build(config)
-        self.assertEqual(mock_copy_file.call_count, 0)
-        self.assertEqual(mock_write_file.call_count, 1)
+
+        with mock.patch.object(plugin, 'copy_files_on_post_build', wraps=plugin.copy_files_on_post_build) as mock_plugin_unpack:
+            plugin.on_pre_build(config)
+            plugin.on_post_build(config)
+            mock_plugin_unpack.assert_not_called()
 
 
 class SearchIndexTests(unittest.TestCase):
--- contrib/python/mkdocs/mkdocs/tests/structure/page_tests.py	(index)
+++ contrib/python/mkdocs/mkdocs/tests/structure/page_tests.py	(working tree)
@@ -7,11 +7,12 @@ from tempfile import TemporaryDirectory
 from mkdocs.structure.pages import Page
 from mkdocs.structure.files import File, Files
 from mkdocs.tests.base import load_config, dedent
+from mkdocs.tests.base import DATA_DIR
 
 
 class PageTests(unittest.TestCase):
 
-    DOCS_DIR = os.path.join(os.path.abspath(os.path.dirname(__file__)), '../integration/subpages/docs')
+    DOCS_DIR = os.path.join(DATA_DIR, 'integration/subpages/docs')
 
     def test_homepage(self):
         cfg = load_config(docs_dir=self.DOCS_DIR)
@@ -501,7 +502,7 @@ class PageTests(unittest.TestCase):
             fl = File('testing.md', cfg['docs_dir'], cfg['site_dir'], cfg['use_directory_urls'])
             pg = Page('Foo', fl, cfg)
             self.assertEqual(pg.url, 'testing/')
-            self.assertEqual(pg.edit_url, expected[i])
+            self.assertEqual(pg.edit_url, None if expected[i] is None else expected[i] + '?edit=true')
 
     def test_nested_page_edit_url(self):
         configs = [
@@ -571,7 +572,7 @@ class PageTests(unittest.TestCase):
             fl = File('sub1/non-index.md', cfg['docs_dir'], cfg['site_dir'], cfg['use_directory_urls'])
             pg = Page('Foo', fl, cfg)
             self.assertEqual(pg.url, 'sub1/non-index/')
-            self.assertEqual(pg.edit_url, expected[i])
+            self.assertEqual(pg.edit_url, None if expected[i] is None else expected[i] + '?edit=true')
 
     @unittest.skipUnless(sys.platform.startswith("win"), "requires Windows")
     def test_nested_page_edit_url_windows(self):
@@ -690,7 +691,7 @@ class SourceDateEpochTests(unittest.TestCase):
 
 class RelativePathExtensionTests(unittest.TestCase):
 
-    DOCS_DIR = os.path.join(os.path.abspath(os.path.dirname(__file__)), '../integration/subpages/docs')
+    DOCS_DIR = os.path.join(DATA_DIR, 'integration/subpages/docs')
 
     def get_rendered_result(self, files):
         cfg = load_config(docs_dir=self.DOCS_DIR)
--- contrib/python/mkdocs/mkdocs/tests/theme_tests.py	(index)
+++ contrib/python/mkdocs/mkdocs/tests/theme_tests.py	(working tree)
@@ -4,13 +4,15 @@ import unittest
 from unittest import mock
 
 import mkdocs
-from mkdocs.theme import Theme
+from mkdocs.theme import YandexTheme as Theme
+from mkdocs.tests.base import PathAssertionMixin
 from mkdocs.localization import parse_locale
 
 abs_path = os.path.abspath(os.path.dirname(__file__))
 mkdocs_dir = os.path.abspath(os.path.dirname(mkdocs.__file__))
-mkdocs_templates_dir = os.path.join(mkdocs_dir, 'templates')
-theme_dir = os.path.abspath(os.path.join(mkdocs_dir, 'themes'))
+mkdocs_templates_dir = os.path.join(mkdocs_dir, 'mkdocs', 'templates')
+theme_dir = os.path.abspath(os.path.join(mkdocs_dir, 'mkdocs', 'themes'))
+mkdocs.utils.themes_dir_root = mkdocs_dir
 
 
 def get_vars(theme):
@@ -18,7 +20,7 @@ def get_vars(theme):
     return {k: theme[k] for k in iter(theme)}
 
 
-class ThemeTests(unittest.TestCase):
+class ThemeTests(PathAssertionMixin, unittest.TestCase):
 
     def test_simple_theme(self):
         theme = Theme(name='mkdocs')
@@ -27,6 +29,8 @@ class ThemeTests(unittest.TestCase):
             [os.path.join(theme_dir, 'mkdocs'), mkdocs_templates_dir]
         )
         self.assertEqual(theme.static_templates, {'404.html', 'sitemap.xml'})
+        self.assertPathIsFile(theme_dir, 'mkdocs', '404.html')
+        self.assertPathIsFile(mkdocs_templates_dir, 'sitemap.xml')
         self.assertEqual(get_vars(theme), {
             'locale': parse_locale('en'),
             'include_search_page': False,
@@ -51,6 +55,8 @@ class ThemeTests(unittest.TestCase):
                 mkdocs_templates_dir
             ]
         )
+        self.assertPathIsDir(theme_dir, 'mkdocs')
+        self.assertPathIsDir(mkdocs_templates_dir)
 
     def test_custom_dir_only(self):
         custom = tempfile.mkdtemp()
@@ -59,6 +65,7 @@ class ThemeTests(unittest.TestCase):
             theme.dirs,
             [custom, mkdocs_templates_dir]
         )
+        self.assertPathIsDir(mkdocs_templates_dir)
 
     def static_templates(self):
         theme = Theme(name='mkdocs', static_templates='foo.html')
@@ -66,6 +73,8 @@ class ThemeTests(unittest.TestCase):
             theme.static_templates,
             {'404.html', 'sitemap.xml', 'foo.html'}
         )
+        self.assertPathIsDir(theme_dir, 'mkdocs')
+        self.assertPathNotExists(theme_dir, 'mkdocs', 'foo.html')
 
     def test_vars(self):
         theme = Theme(name='mkdocs', foo='bar', baz=True)
@@ -83,6 +92,7 @@ class ThemeTests(unittest.TestCase):
         theme = Theme(name='mkdocs')
         self.assertEqual(m.call_count, 1)
         self.assertEqual(theme.static_templates, {'sitemap.xml'})
+        self.assertPathIsDir(theme_dir, 'mkdocs')
 
     def test_inherited_theme(self):
         m = mock.Mock(side_effect=[
@@ -100,6 +110,12 @@ class ThemeTests(unittest.TestCase):
                     mkdocs_templates_dir
                 ]
             )
+            self.assertPathIsDir(theme_dir, 'mkdocs')
+            self.assertPathIsDir(theme_dir, 'readthedocs')
+            self.assertPathIsDir(mkdocs_templates_dir)
             self.assertEqual(
                 theme.static_templates, {'sitemap.xml', 'child.html', 'parent.html'}
             )
+            self.assertPathIsFile(mkdocs_templates_dir, 'sitemap.xml')
+            self.assertPathNotExists(theme_dir, 'mkdocs', 'child.html')
+            self.assertPathNotExists(theme_dir, 'readthedocs', 'parent.html')
--- contrib/python/mkdocs/mkdocs/tests/utils/utils_tests.py	(index)
+++ contrib/python/mkdocs/mkdocs/tests/utils/utils_tests.py	(working tree)
@@ -275,7 +275,7 @@ class UtilsTests(unittest.TestCase):
             ['mkdocs', 'readthedocs'])
 
     @mock.patch('importlib_metadata.entry_points', autospec=True)
-    def test_get_theme_dir(self, mock_iter):
+    def _test_get_theme_dir(self, mock_iter):
 
         path = 'some/path'
 
@@ -294,7 +294,7 @@ class UtilsTests(unittest.TestCase):
             utils.get_theme_dir('nonexistanttheme')
 
     @mock.patch('importlib_metadata.entry_points', autospec=True)
-    def test_get_theme_dir_importerror(self, mock_iter):
+    def _test_get_theme_dir_importerror(self, mock_iter):
 
         theme = mock.Mock()
         theme.name = 'mkdocs2'
