--- contrib/python/holoviews/holoviews/ipython/__init__.py	(index)
+++ contrib/python/holoviews/holoviews/ipython/__init__.py	(working tree)
@@ -253,7 +253,7 @@
         """
         import jinja2
 
-        templateLoader = jinja2.FileSystemLoader(os.path.dirname(os.path.abspath(__file__)))
+        templateLoader = jinja2.PackageLoader(package_name='holoviews', package_path='ipython')
         jinjaEnv = jinja2.Environment(loader=templateLoader)
         template = jinjaEnv.get_template('load_notebook.html')
         html = template.render({'logo':        logo,
