--- contrib/libs/nanopb/generator/nanopb_generator.py   (index)
+++ contrib/libs/nanopb/generator/nanopb_generator.py   (working tree)
@@ -17,6 +17,8 @@ import shlex
 import os
 from functools import reduce
 
+import google.protobuf.compiler.plugin_pb2 as plugin_pb2
+
 # Python-protobuf breaks easily with protoc version differences if
 # using the cpp or upb implementation. Force it to use pure Python
 # implementation. Performance is not very important in the generator.
@@ -33,7 +35,6 @@ except:
 try:
     import google.protobuf.text_format as text_format
     import google.protobuf.descriptor_pb2 as descriptor
-    import google.protobuf.compiler.plugin_pb2 as plugin_pb2
     import google.protobuf.descriptor
     import google.protobuf.message_factory as message_factory
 except:
@@ -60,7 +61,7 @@ else:
 
 if getattr(sys, 'frozen', False):
     # Binary package, just import the file
-    from proto import nanopb_pb2
+    from .proto import nanopb_pb2
 else:
     # Import nanopb_pb2.py, rebuilds if necessary and not disabled
     # by env variable NANOPB_PB2_NO_REBUILD
@@ -2612,7 +2613,8 @@ def main_plugin():
         msvcrt.setmode(sys.stdin.fileno(), os.O_BINARY)
         msvcrt.setmode(sys.stdout.fileno(), os.O_BINARY)
 
-    data = io.open(sys.stdin.fileno(), "rb").read()
+    with io.open(sys.stdin.fileno(), "rb") as input:
+        data = input.read()
 
     request = plugin_pb2.CodeGeneratorRequest.FromString(data)
 
@@ -2676,7 +2678,8 @@ def main_plugin():
     if hasattr(plugin_pb2.CodeGeneratorResponse, "FEATURE_PROTO3_OPTIONAL"):
         response.supported_features = plugin_pb2.CodeGeneratorResponse.FEATURE_PROTO3_OPTIONAL
 
-    io.open(sys.stdout.fileno(), "wb").write(response.SerializeToString())
+    with io.open(sys.stdout.fileno(), "wb") as output:
+        output.write(response.SerializeToString())
 
 if __name__ == '__main__':
     # Check if we are running as a plugin under protoc
