--- a/coders/icon.c
+++ b/coders/icon.c
@@ -394,6 +394,9 @@ static Image *ReadICONImage(const ImageInfo *image_info,
     /*
       Verify Icon identifier.
     */
+    MagickSizeType blob_size = GetBlobSize(image);
+    if (blob_size > 0 && directory->icons[i]->offset >= blob_size)
+      ThrowReaderException(CorruptImageError,"ImproperImageHeader");
     offset=(ssize_t) SeekBlob(image,(MagickOffsetType)
       directory->icons[i]->offset,SEEK_SET);
     if (offset < 0)
@@ -513,6 +516,14 @@ static Image *ReadICONImage(const ImageInfo *image_info,
           /*
             Read Icon raster colormap.
           */
+          // Fix preserving memory on broken ICOs with enormous image->colors
+          MagickSizeType blob_size = GetBlobSize(image);
+          if (blob_size > 0){
+            MagickOffsetType space_after_read = blob_size-TellBlob(image)-(4*image->colors);
+            if (space_after_read < 0)
+              ThrowReaderException(CorruptImageError,
+                "InsufficientImageDataInFile");
+          }
           if (image->colors > GetBlobSize(image))
             ThrowICONReaderException(CorruptImageError,
               "InsufficientImageDataInFile");
