--- a/coders/png.c
+++ b/coders/png.c
@@ -120,9 +120,9 @@
 #undef MNG_BASI_SUPPORTED
 #define MNG_COALESCE_LAYERS /* In 5.4.4, this interfered with MMAP'ed files. */
 #define MNG_INSERT_LAYERS   /* Troublesome, but seem to work as of 5.4.4 */
-#if defined(MAGICKCORE_JPEG_DELEGATE)
-#  define JNG_SUPPORTED /* Not finished as of 5.5.2.  See "To do" comments. */
-#endif
+// #if defined(MAGICKCORE_JPEG_DELEGATE)
+// #  define JNG_SUPPORTED /* Not finished as of 5.5.2.  See "To do" comments. */
+// #endif
 #if !defined(RGBColorMatchExact)
 #define IsPNGColorEqual(color,target) \
        (((color).red == (target).red) && \
