--- a/coders/svg.c
+++ b/coders/svg.c
@@ -241,6 +241,10 @@ static MagickBooleanType IsSVG(const uns
     return(MagickFalse);
   if (LocaleNCompare((const char *) magick+1,"?xml",4) == 0)
     return(MagickTrue);
+  if (length < 14)
+    return(MagickFalse);
+  if (LocaleNCompare((const char *) magick+1,"!DOCTYPE svg",13) == 0)
+    return(MagickTrue);
   return(MagickFalse);
 }
 
--- a/magick/magic.c
+++ b/magick/magic.c
@@ -210,6 +210,8 @@ static const MagicMapInfo
     { "SVG", 1, MagicPattern("?xml") },
     { "SVG", 1, MagicPattern("SVG") },
     { "SVG", 1, MagicPattern("svg") },
+    { "SVG", 1, MagicPattern("!DOCTYPE SVG"), MagickFalse },
+    { "SVG", 1, MagicPattern("!DOCTYPE svg"), MagickFalse },
     { "TIFF", 0, MagicPattern("\115\115\000\052") },
     { "TIFF", 0, MagicPattern("\111\111\052\000") },
     { "TIFF64", 0, MagicPattern("\115\115\000\053\000\010\000\000") },
