--- contrib/python/eccodes/gribapi/bindings.py	(index)
+++ contrib/python/eccodes/gribapi/bindings.py	(working tree)
@@ -25,4 +25,1 @@ __version__ = "1.7.0"
-library_path = find_binary_libs("eccodes")
-
-if library_path is None:
-    raise RuntimeError("Cannot find the ecCodes library")
+library_path = None
--- contrib/python/eccodes/ya.make	(index)
+++ contrib/python/eccodes/ya.make	(working tree)
@@ -39,6 +39,13 @@ PY_SRCS(
     gribapi/gribapi.py
 )
 
+RUN_PROGRAM(
+    contrib/python/cffi/py3/gen builder.py:ffibuilder gribapi._bindings.cffi.py3.c
+    IN builder.py gribapi/eccodes.h gribapi/grib_api.h
+    OUT gribapi._bindings.cffi.py3.c
+    OUTPUT_INCLUDES eccodes.h
+)
+
 RESOURCE_FILES(
     PREFIX contrib/python/eccodes/
     .dist-info/METADATA
