--- contrib/python/spdx-tools/spdx/config.py	(index)
+++ contrib/python/spdx-tools/spdx/config.py	(working tree)
@@ -33,8 +33,16 @@ def _load_list(file_name, object_type='licenses', id_attribute='licenseId'):
     from https://github.com/spdx/license-list-data
     """
     licenses_map = {}
-    with codecs.open(file_name, 'rb', encoding='utf-8') as lics:
-        licenses = json.load(lics)
+    if not os.path.exists(file_name):
+        # try to load from resource
+        import pkgutil
+        data = pkgutil.get_data(__package__, os.path.basename(file_name))
+        if not data:
+            raise FileNotFoundError('file {} not found'.format(file_name))
+        licenses = json.loads(data)
+    else:
+        with codecs.open(file_name, 'rb', encoding='utf-8') as lics:
+            licenses = json.load(lics)
     version = tuple(licenses['licenseListVersion'].split('.'))
     for lic in licenses[object_type]:
         if lic.get('isDeprecatedLicenseId'):
