--- contrib/deprecated/python/opentracing/py3/tests/scope_managers/test_asyncio.py	(index)
+++ contrib/deprecated/python/opentracing/py3/tests/scope_managers/test_asyncio.py	(working tree)
@@ -35,8 +35,7 @@ class AsyncioCompabilityCheck(TestCase, ScopeCompatibilityCheckMixin):
         return AsyncioScopeManager()
 
     def run_test(self, test_fn):
-        @asyncio.coroutine
-        def async_test_fn():
+        async def async_test_fn():
             test_fn()
         asyncio.get_event_loop().run_until_complete(async_test_fn())
 
--- contrib/deprecated/python/opentracing/py3/tests/scope_managers/test_contextvars.py	(index)
+++ contrib/deprecated/python/opentracing/py3/tests/scope_managers/test_contextvars.py	(working tree)
@@ -36,8 +36,7 @@ class AsyncioContextVarsCompabilityCheck(
         return ContextVarsScopeManager()
 
     def run_test(self, test_fn):
-        @asyncio.coroutine
-        def async_test_fn():
+        async def async_test_fn():
             test_fn()
         asyncio.get_event_loop().run_until_complete(async_test_fn())
 
