--- a/Lib/ctypes/__init__.py	(index)
+++ b/Lib/ctypes/__init__.py	(working tree)
@@ -372,3 +373,9 @@ class CDLL(object):
+        self._builtin = {}
+        if isinstance(name, dict):
+            self._builtin = name['symbols']
+            self._name = name = name['name']
+            self._handle = 0
+            return
         if name:
             name = _os.fspath(name)
 
@@ -391,7 +399,13 @@ class CDLL(object):
         return func
 
     def __getitem__(self, name_or_ordinal):
-        func = self._FuncPtr((name_or_ordinal, self))
+        if self._builtin:
+            if name_or_ordinal not in self._builtin:
+                raise AttributeError("function %r not found" % name_or_ordinal)
+            func = self._FuncPtr(self._builtin[name_or_ordinal])
+        else:
+            func = self._FuncPtr((name_or_ordinal, self))
+
         if not isinstance(name_or_ordinal, int):
             func.__name__ = name_or_ordinal
         return func
@@ -458,7 +472,8 @@ class LibraryLoader(object):
-if _os.name == "nt":
-    pythonapi = PyDLL("python dll", None, _sys.dllhandle)
-elif _sys.platform in ["android", "cygwin"]:
-    # These are Unix-like platforms which use a dynamically-linked libpython.
-    pythonapi = PyDLL(_sysconfig.get_config_var("LDLIBRARY"))
-else:
+try:
     pythonapi = PyDLL(None)
+except:
+    from .util import find_library as _find_library
+    try:
+        pythonapi = PyDLL(_find_library('python'))
+    except:
+        pythonapi = PyDLL(dict(name='python', symbols={}))
--- a/Lib/ctypes/util.py	(index)
+++ b/Lib/ctypes/util.py	(working tree)
@@ -329,6 +329,16 @@ elif os.name == "posix":
             return _findSoname_ldconfig(name) or \
                    _get_soname(_findLib_gcc(name)) or _get_soname(_findLib_ld(name))
 
+try:
+    from library.python.symbols.module import find_library as _find_library
+except ImportError:
+    pass
+else:
+    _next_find_library = find_library
+
+    def find_library(name):
+        return _find_library(name, _next_find_library)
+
 ################################################################
 # test code
 
