commit 9b726d183d9b023f472f788f4e87136eaee3edf8
author: shadchin
date: 2023-01-09T20:49:50+03:00

    Fix devtools/dummy_arcadia/test/system_python

--- contrib/tools/python3/Include/cpython/pythread.h	(a09217b707768f106bfed2d616636b38cfce2f13)
+++ contrib/tools/python3/Include/cpython/pythread.h	(9b726d183d9b023f472f788f4e87136eaee3edf8)
@@ -21,7 +21,7 @@ PyAPI_FUNC(int) _PyThread_at_fork_reinit(PyThread_type_lock *lock);
     */
 #   define NATIVE_TSS_KEY_T     unsigned long
 #elif defined(HAVE_PTHREAD_STUBS)
-#   include "cpython/pthread_stubs.h"
+#error #   include "cpython/pthread_stubs.h"
 #   define NATIVE_TSS_KEY_T     pthread_key_t
 #else
 #   error "Require native threads. See https://bugs.python.org/issue31370"
