--- contrib/python/awscli/awscli/customizations/configure/__init__.py	(index)
+++ contrib/python/awscli/awscli/customizations/configure/__init__.py	(working tree)
@@ -14,1 +14,1 @@
-from botocore.vendored.six.moves import shlex_quote
+from six.moves import shlex_quote
--- contrib/python/awscli/awscli/customizations/awslambda.py	(index)
+++ contrib/python/awscli/awscli/customizations/awslambda.py	(working tree)
@@ -17,1 +17,1 @@ import zipfile
-from botocore.vendored import six
+import six
--- contrib/python/awscli/awscli/__init__.py	(index)
+++ contrib/python/awscli/awscli/__init__.py	(working tree)
@@ -32,1 +32,1 @@ if 'AWS_DATA_PATH' in os.environ:
-    os.path.join(os.path.dirname(os.path.abspath(__file__)), 'data')
+    os.path.join(os.path.dirname(__file__), 'data')
--- contrib/python/awscli/awscli/clidocs.py	(index)
+++ contrib/python/awscli/awscli/clidocs.py	(working tree)
@@ -720,1 +720,2 @@ class TopicDocumentEventHandler(TopicListerDocumentEventHandler):
-        with open(filename, 'r') as f:
+        import importlib.resources
+        with importlib.resources.files("awscli").joinpath(filename).open('r') as f:
--- contrib/python/awscli/awscli/topictags.py	(index)
+++ contrib/python/awscli/awscli/topictags.py	(working tree)
@@ -74,3 +74 @@ class TopicTagDB(object):
-    TOPIC_DIR = os.path.join(
-        os.path.dirname(
-            os.path.abspath(__file__)), 'topics')
+    TOPIC_DIR = 'topics'
@@ -122,1 +122,2 @@ class TopicTagDB(object):
-        with open(self.index_file, 'r') as f:
+        import importlib.resources
+        with importlib.resources.files("awscli").joinpath(self.index_file).open('r') as f:
--- contrib/python/awscli/awscli/bcdoc/restdoc.py	(index)
+++ contrib/python/awscli/awscli/bcdoc/restdoc.py	(working tree)
@@ -106 +106,2 @@ class ReSTDocument(object):
-        with open(filename, 'r') as f:
+        import importlib.resources
+        with importlib.resources.files("awscli").joinpath(filename).open('r') as f:
--- contrib/python/awscli/awscli/clidocs.py	(index)
+++ contrib/python/awscli/awscli/clidocs.py	(working tree)
@@ -31,2 +31 @@ LOG = logging.getLogger(__name__)
-EXAMPLES_DIR = os.path.join(os.path.dirname(os.path.abspath(__file__)),
-                            'examples')
+EXAMPLES_DIR = 'examples'
--- contrib/python/awscli/awscli/customizations/commands.py	(index)
+++ contrib/python/awscli/awscli/customizations/commands.py	(working tree)
@@ -344,4 +344,3 @@ class BasicHelp(HelpCommand):
-            doc_path = os.path.join(
-                os.path.abspath(os.path.dirname(root_module.__file__)),
-                'examples', trailing_path)
-            with _open(doc_path) as f:
+            doc_path = os.path.join('examples', trailing_path)
+            import importlib.resources
+            with importlib.resources.files("awscli").joinpath(doc_path).open('r') as f:
