--- contrib/python/zstandard/py3/c-ext/backend_c.c	(index)
+++ contrib/python/zstandard/py3/c-ext/backend_c.c	(working tree)
@@ -152,7 +152,7 @@ void zstd_module_init(PyObject *m) {
     PyObject *features = NULL;
     PyObject *feature = NULL;
     unsigned zstd_ver_no = ZSTD_versionNumber();
-    unsigned our_hardcoded_version = 10506;
+    unsigned our_hardcoded_version = 10507;
     if (ZSTD_VERSION_NUMBER != our_hardcoded_version ||
         zstd_ver_no != our_hardcoded_version) {
         PyErr_Format(
