--- contrib/python/async-generator/async_generator/_tests/test_async_generator.py	(index)
+++ contrib/python/async-generator/async_generator/_tests/test_async_generator.py	(working tree)
@@ -7,7 +7,7 @@ from functools import wraps
 import gc
 
 from .conftest import mock_sleep
-from .. import (
+from async_generator import (
     async_generator,
     yield_,
     yield_from_,
@@ -756,7 +756,7 @@ async def test_ag_attributes():
 # Finicky tests to check that the overly clever ctype stuff has plausible
 # refcounting
 
-from .. import _impl
+from async_generator import _impl
 
 
 @pytest.mark.skipif(not hasattr(sys, "getrefcount"), reason="CPython only")
--- contrib/python/async-generator/async_generator/_tests/test_util.py	(index)
+++ contrib/python/async-generator/async_generator/_tests/test_util.py	(working tree)
@@ -1,6 +1,6 @@
 import pytest
 
-from .. import aclosing, async_generator, yield_, asynccontextmanager
+from async_generator import aclosing, async_generator, yield_, asynccontextmanager
 
 
 @async_generator
