--- contrib/python/regex/py2/regex_2/test_regex.py	(index)
+++ contrib/python/regex/py2/regex_2/test_regex.py	(working tree)
@@ -1,5 +1,5 @@
 from __future__ import with_statement
-from test.test_support import run_unittest
+# from test.test_support import run_unittest
 from weakref import proxy
 import copy
 import pickle
@@ -4261,8 +4261,8 @@ if not hasattr(str, "format"):
     del RegexTests.test_format
     del RegexTests.test_subscripted_captures
 
-def test_main():
-    run_unittest(RegexTests)
-
-if __name__ == "__main__":
-    test_main()
+# def test_main():
+#     run_unittest(RegexTests)
+#
+# if __name__ == "__main__":
+#     test_main()
