--- contrib/python/pylint/pylint/lint/run.py	(419ba6ee001b3f259e0d651b20e8e856795b5be5)
+++ contrib/python/pylint/pylint/lint/run.py	(working tree)
@@ -37,6 +37,9 @@ except ImportError:
     ProcessPoolExecutor = None  # type: ignore[assignment,misc]
 
 
+PYLINT_IGNORE_DEFAULT_CONFIGS = os.getenv('PYLINT_IGNORE_DEFAULT_CONFIGS', 'no') == 'yes'
+
+
 def _query_cpu() -> int | None:
     """Try to determine number of CPUs allotted in a docker container.
 
@@ -139,7 +142,7 @@ group are mutually exclusive.",
             sys.exit(32)
 
         # Determine configuration file
-        if self._rcfile is None:
+        if self._rcfile is None and not PYLINT_IGNORE_DEFAULT_CONFIGS:
             default_file = next(config.find_default_config_files(), None)
             if default_file:
                 self._rcfile = str(default_file)
