--- contrib/restricted/abseil-cpp/absl/debugging/internal/elf_mem_image.h	(index)
+++ contrib/restricted/abseil-cpp/absl/debugging/internal/elf_mem_image.h	(working tree)
@@ -35,7 +35,7 @@
 #if defined(__ELF__) && !defined(__OpenBSD__) && !defined(__QNX__) &&    \
     !defined(__asmjs__) && !defined(__wasm__) && !defined(__HAIKU__) &&  \
     !defined(__sun) && !defined(__VXWORKS__) && !defined(__hexagon__) && \
-    !defined(__XTENSA__)  && !defined(__FREERTOS__)
+    !defined(__XTENSA__)  && !defined(__FREERTOS__) && !defined(__ZEPHYR__)
 #define ABSL_HAVE_ELF_MEM_IMAGE 1
 #endif
 
--- contrib/restricted/abseil-cpp/absl/time/internal/cctz/src/time_zone_libc.cc	(index)
+++ contrib/restricted/abseil-cpp/absl/time/internal/cctz/src/time_zone_libc.cc	(working tree)
@@ -81,7 +81,7 @@ auto tm_zone(const std::tm& tm) -> decltype(tzname[0]) {
   const bool is_dst = tm.tm_isdst > 0;
   return tzname[is_dst];
 }
-#elif defined(__FREERTOS__)
+#elif defined(__FREERTOS__) || defined(__ZEPHYR__)
 long int tm_gmtoff(const std::tm& tm) {
   (void)tm;
   return 0;
