--- contrib/python/pycrypto/py2/ya.make	(index)
+++ contrib/python/pycrypto/py2/ya.make	(working tree)
@@ -36,11 +34,23 @@ SRCS(
     src/XOR.c
     src/_counter.c
     src/strxor.c
-    src/winrand.c
 )
 
+IF (OS_WINDOWS)
+    PY_SRCS(
+        TOP_LEVEL
+        Crypto/Random/OSRNG/nt.py
+        Crypto/Util/winrandom.py
+    )
+    SRCS(
+        src/winrand.c
+    )
+    PY_REGISTER(
+        Crupto.Random.OSRNG.winrandom
+    )
+ENDIF()
+
 PY_REGISTER(
-    Crupto.Random.OSRNG.winrandom
     Crypto.Cipher._AES
     Crypto.Cipher._ARC2
     Crypto.Cipher._ARC4
@@ -104,7 +114,6 @@ PY_SRCS(
     Crypto/Random/Fortuna/__init__.py
     Crypto/Random/OSRNG/__init__.py
     Crypto/Random/OSRNG/fallback.py
-    Crypto/Random/OSRNG/nt.py
     Crypto/Random/OSRNG/posix.py
     Crypto/Random/OSRNG/rng_base.py
     Crypto/Random/_UserFriendlyRNG.py
@@ -121,7 +130,6 @@ PY_SRCS(
     Crypto/Util/number.py
     Crypto/Util/py3compat.py
     Crypto/Util/randpool.py
-    Crypto/Util/winrandom.py
     Crypto/__init__.py
     Crypto/pct_warnings.py
 )
