https://github.com/ajakubek/python-llist/pull/17

--- contrib/python/llist/src/dllist.c	(608df08093eb1de320dfd1f155378b1f787f7951)
+++ contrib/python/llist/src/dllist.c	(01d4f151960d1f3cf1eb81ef8413abab126a1607)
@@ -203,6 +203,7 @@ static int dllistnode_clear_refs(DLListNodeObject* self)
 
 static void dllistnode_dealloc(DLListNodeObject* self)
 {
+    PyObject_GC_UnTrack(self);
     PyObject* obj_self = (PyObject*)self;
 
     dllistnode_clear_refs(self);
@@ -606,6 +607,7 @@ static int dllist_clear_refs(DLListObject* self)
 
 static void dllist_dealloc(DLListObject* self)
 {
+    PyObject_GC_UnTrack(self);
     PyObject* obj_self = (PyObject*)self;
 
     dllist_clear_refs(self);
@@ -1758,6 +1760,7 @@ static int dllistiterator_clear_refs(DLListIteratorObject* self)
 
 static void dllistiterator_dealloc(DLListIteratorObject* self)
 {
+    PyObject_GC_UnTrack(self);
     PyObject* obj_self = (PyObject*)self;
 
     dllistiterator_clear_refs(self);
--- contrib/python/llist/src/sllist.c	(608df08093eb1de320dfd1f155378b1f787f7951)
+++ contrib/python/llist/src/sllist.c	(01d4f151960d1f3cf1eb81ef8413abab126a1607)
@@ -131,6 +131,7 @@ static int sllistnode_clear_refs(SLListNodeObject* self)
 
 static void sllistnode_dealloc(SLListNodeObject* self)
 {
+    PyObject_GC_UnTrack(self);
     PyObject* obj_self = (PyObject*)self;
 
     sllistnode_clear_refs(self);
@@ -373,6 +374,7 @@ static int sllist_clear_refs(SLListObject* self)
 
 static void sllist_dealloc(SLListObject* self)
 {
+    PyObject_GC_UnTrack(self);
     PyObject* obj_self = (PyObject*)self;
 
     sllist_clear_refs(self);
@@ -1722,6 +1724,7 @@ static int sllistiterator_clear_refs(SLListIteratorObject* self)
 
 static void sllistiterator_dealloc(SLListIteratorObject* self)
 {
+    PyObject_GC_UnTrack(self);
     PyObject* obj_self = (PyObject*)self;
 
     sllistiterator_clear_refs(self);
