--- contrib/python/forbiddenfruit/forbiddenfruit/__init__.py	(index)
+++ contrib/python/forbiddenfruit/forbiddenfruit/__init__.py	(working tree)
@@ -446,7 +446,8 @@ def curse(klass, attr, value, hide_from_dir=False):
         except AttributeError:
             pass
 
-    ctypes.pythonapi.PyType_Modified(ctypes.py_object(klass))
+    if not issubclass(klass, str):
+        ctypes.pythonapi.PyType_Modified(ctypes.py_object(klass))
 
     if hide_from_dir:
         __hidden_elements__[klass.__name__].append(attr)
@@ -481,7 +482,8 @@ def reverse(klass, attr):
     dikt = patchable_builtin(klass)
     del dikt[attr]
 
-    ctypes.pythonapi.PyType_Modified(ctypes.py_object(klass))
+    if not issubclass(klass, str):
+        ctypes.pythonapi.PyType_Modified(ctypes.py_object(klass))
 
 
 def curses(klass, name):
