diff --git a/thirdparty/CMakeLists.txt b/thirdparty/CMakeLists.txt
index fc8f51d..49c55b2 100644
--- a/thirdparty/CMakeLists.txt
+++ b/thirdparty/CMakeLists.txt
@@ -70,8 +70,8 @@ endif()
 
 if (NOT DEFINED JINJA2_PRIVATE_LIBS_INT)
     set(JINJA2CPP_PRIVATE_LIBS ${JINJA2CPP_PRIVATE_LIBS}
-        Boost::variant Boost::filesystem Boost::algorithm Boost::lexical_cast Boost::json
-        Boost::regex fmt RapidJson)
+        boost_variant boost_filesystem boost_algorithm boost_json
+        boost_regex fmt RapidJson)
 else ()
     set (JINJA2CPP_PRIVATE_LIBS ${JINJA2_PRIVATE_LIBS_INT})
 endif ()
