--- contrib/python/aiopg/aiopg/connection.py	(index)
+++ contrib/python/aiopg/aiopg/connection.py	(working tree)
@@ -1,11 +1,13 @@
 import abc
 import asyncio
 import contextlib
+import ctypes
 import datetime
 import enum
 import errno
 import platform
 import select
+import socket
 import sys
 import traceback
 import uuid
@@ -43,6 +45,17 @@ TIMEOUT = 60.0
 # to OSError.errno EBADF
 WSAENOTSOCK = 10038
 
+# Connection status from psycopg2 (psycopg2/psycopg/connection.h)
+CONN_STATUS_CONNECTING = 20
+
+# In socket.socket we should know type and family to shutdown socket by fd
+# This function is used for shutdown libpq connection
+# where family and type is unknown
+libc = ctypes.CDLL(None)
+socket_shutdown = libc.shutdown
+socket_shutdown.restypes = ctypes.c_int
+socket_shutdown.argtypes = ctypes.c_int, ctypes.c_int
+
 
 def connect(
     dsn: Optional[str] = None,
@@ -760,6 +773,11 @@ class Connection:
         self._conn = psycopg2.connect(dsn, **kwargs)
 
         self._dsn = self._conn.dsn
+        self._dns_params = self._conn.get_dsn_parameters()
+        self._conn_timeout = self._dns_params.get('connect_timeout')
+        if self._conn_timeout:
+            self._conn_timeout = float(self._conn_timeout)
+
         assert self._conn.isexecuting(), "Is conn an async at all???"
         self._fileno: Optional[int] = self._conn.fileno()
         self._timeout = timeout
@@ -772,11 +790,31 @@ class Connection:
         self._loop.add_reader(
             self._fileno, self._ready, self._weakref  # type: ignore
         )
+        self._conn_timeout_handler = None
+        if self._conn_timeout:
+            self._conn_timeout_handler = self._loop.call_later(
+                self._conn_timeout, self._shutdown, self._weakref)
 
         if self._loop.get_debug():
             self._source_traceback = traceback.extract_stack(sys._getframe(1))
 
     @staticmethod
+    def _shutdown(weak_self):
+        # Make sure that we won't get stuck in a blocking read or write
+        # inside poll, then give libpq a chance to try another host.
+        # If there is an error, we'll get it from poll.
+        self = weak_self()
+
+        # dead already.
+        if self is None:
+            return
+
+        if self._conn.status == CONN_STATUS_CONNECTING:
+            socket_shutdown(ctypes.c_int(self._fileno),
+                            ctypes.c_int(socket.SHUT_RDWR))
+            self._ready(self._weakref)
+
+    @staticmethod
     def _ready(weak_self: "weakref.ref[Any]") -> None:
         self = cast(Connection, weak_self())
         if self is None:
@@ -822,6 +860,28 @@ class Connection:
                     waiter.set_exception(
                         psycopg2.OperationalError("Connection closed")
                     )
+
+            if self._conn.status == CONN_STATUS_CONNECTING \
+                    and state != psycopg2.extensions.POLL_ERROR:
+                # libpq could close and open new connection to the next host
+                old_fileno = self._fileno
+                self._fileno = self._conn.fileno()
+
+                if self._conn_timeout_handler:
+                    self._conn_timeout_handler.cancel()
+                    self._conn_timeout_handler = self._loop.call_later(
+                        self._conn_timeout, self._shutdown, self._weakref)
+
+                with contextlib.suppress(OSError):
+                    # if we are using select selector
+                    self._loop.remove_reader(old_fileno)
+                    if self._writing:
+                        self._loop.remove_writer(old_fileno)
+
+                self._loop.add_reader(self._fileno, self._ready, weak_self)
+                if self._writing:
+                    self._loop.add_writer(self._fileno, self._ready, weak_self)
+
             if state == psycopg2.extensions.POLL_OK:
                 if self._writing:
                     self._loop.remove_writer(self._fileno)  # type: ignore
