https://a.yandex-team.ru/review/1117620/details

--- contrib/python/py-radix/py3/radix/_radix/radix.c        (index)
+++ contrib/python/py-radix/py3/radix/_radix/radix.c        (working tree)
@@ -371,11 +371,14 @@ radix_node_t

         bitlen = prefix->bitlen;

-        RADIX_SEARCH_FOREACH_INCLUSIVE(node, head, prefix) {
-                if (node->prefix && (inclusive || node->bit != bitlen))
-                    stack[cnt++] = node;
+        RADIX_SEARCH_FOREACH(node, head, prefix) {
+                if (node->prefix)
+                        stack[cnt++] = node;
         }

+        if (inclusive && node && node->prefix)
+                stack[cnt++] = node;
+
         if (cnt <= 0)
                 return (NULL);

@@ -411,11 +414,14 @@ radix_node_t

         bitlen = prefix->bitlen;

-        RADIX_SEARCH_FOREACH_INCLUSIVE(node, head, prefix) {
-                if (node->prefix && (inclusive || node->bit != bitlen))
+        RADIX_SEARCH_FOREACH(node, head, prefix) {
+                if (node->prefix)
                     stack[cnt++] = node;
         }

+        if (inclusive && node && node->prefix)
+                stack[cnt++] = node;
+
         if (cnt <= 0)
                 return (NULL);
         
