Index: contrib/python/facebook-business/facebook_business/session.py
===================================================================
--- contrib/python/facebook-business/facebook_business/session.py	(rev 7641126)
+++ contrib/python/facebook-business/facebook_business/session.py	(working copy)
@@ -22,10 +22,11 @@
 The purpose of the session module is to encapsulate authentication classes and
 utilities.
 """
+import facebook_business.certs
+
 import hashlib
 import hmac
 import requests
-import os
 
 
 class FacebookSession(object):
@@ -59,10 +60,7 @@ class FacebookSession(object):
         self.timeout = timeout
         self.debug = debug
         self.requests = requests.Session()
-        self.requests.verify = os.path.join(
-            os.path.dirname(__file__),
-            'fb_ca_chain_bundle.crt',
-        )
+        self.requests.verify = facebook_business.certs.cert_manager.get_cert_file()
         params = {
             'access_token': self.access_token
         }
Index: contrib/python/facebook-business/facebook_business/adobjects/serverside/event_request_async.py
===================================================================
--- contrib/python/facebook-business/facebook_business/adobjects/serverside/event_request_async.py	(rev 7641126)
+++ contrib/python/facebook-business/facebook_business/adobjects/serverside/event_request_async.py	(working copy)
@@ -27,11 +27,13 @@ from facebook_business.adobjects.serverside.event_response import EventResponse
 from facebook_business.api import FacebookAdsApi
 from facebook_business.session import FacebookSession
 
+import facebook_business.certs
+
 import aiohttp
 import json
-import os
 import ssl
 
+
 class EventRequestAsync(EventRequest):
     def __init__(
         self,
@@ -71,12 +73,7 @@ class EventRequestAsync(EventRequest):
         params['access_token'] = access_token
         if facebook_session.app_secret:
             params['appsecret_proof'] = facebook_session._gen_appsecret_proof()
-        cafile = os.path.join(
-            os.path.dirname(__file__),
-            '..',
-            '..',
-            'fb_ca_chain_bundle.crt'
-        )
+        cafile = facebook_business.certs.cert_manager.get_cert_file()
         sslcontext = ssl.create_default_context(cafile=cafile)
 
         async with session.post(
Index: contrib/python/facebook-business/facebook_business/adobjects/serverside/util.py
===================================================================
--- contrib/python/facebook-business/facebook_business/adobjects/serverside/util.py	(rev 7641126)
+++ contrib/python/facebook-business/facebook_business/adobjects/serverside/util.py	(working copy)
@@ -17,12 +17,13 @@
 # LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
 # FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
 # DEALINGS IN THE SOFTWARE.
+import facebook_business.certs
 
 import hmac
 import hashlib
-import os
 import sys
 
+
 class Util:
 
     @staticmethod
@@ -31,12 +32,7 @@ class Util:
 
     @staticmethod
     def ca_bundle_path():
-        return os.path.join(
-            os.path.dirname(__file__),
-            '..',
-            '..',
-            'fb_ca_chain_bundle.crt',
-        )
+        return facebook_business.certs.cert_manager.get_cert_file()
 
     @staticmethod
     def appsecret_proof(appsecret, access_token):
