--- contrib/libs/pocketfft/pocketfft_hdronly.h	(index)
+++ contrib/libs/pocketfft/pocketfft_hdronly.h	(working tree)
@@ -3531,7 +3531,7 @@ template<typename T> void r2c(const shape_t &shape_in,
 
   shape_t shape_out(shape_in);
   shape_out[axes.back()] = shape_in[axes.back()]/2 + 1;
-  auto newaxes = shape_t{axes.begin(), --axes.end()};
+  auto newaxes = shape_t{axes.begin(), std::prev(axes.end())};
   c2c(shape_out, stride_out, stride_out, newaxes, forward, data_out, data_out,
     T(1), nthreads);
   }
@@ -3569,7 +3569,7 @@ template<typename T> void c2r(const shape_t &shape_out,
     stride_inter[size_t(i)] =
       stride_inter[size_t(i+1)]*ptrdiff_t(shape_in[size_t(i+1)]);
   arr<std::complex<T>> tmp(nval);
-  auto newaxes = shape_t{axes.begin(), --axes.end()};
+  auto newaxes = shape_t{axes.begin(), std::prev(axes.end())};
   c2c(shape_in, stride_in, stride_inter, newaxes, forward, data_in, tmp.data(),
     T(1), nthreads);
   c2r(shape_out, stride_inter, stride_out, axes.back(), forward,
