--- contrib/python/odfpy/odf/odf2xhtml.py	(index)
+++ contrib/python/odfpy/odf/odf2xhtml.py	(working tree)
@@ -27,1 +27,1 @@ from xml.sax import handler
-from opendocument import load
+from .opendocument import load
--- contrib/python/odfpy/tests/testload.py	(index)
+++ contrib/python/odfpy/tests/testload.py	(working tree)
@@ -24,1 +24,1 @@ from io import BytesIO
-from elementparser import ElementParser
+from .elementparser import ElementParser
--- contrib/python/odfpy/tests/testmasterstyles.py	(index)
+++ contrib/python/odfpy/tests/testmasterstyles.py	(working tree)
@@ -24,1 +24,1 @@ from odf import style, text
-from elementparser import ElementParser
+from .elementparser import ElementParser
--- contrib/python/odfpy/tests/testpresentations.py	(index)
+++ contrib/python/odfpy/tests/testpresentations.py	(working tree)
@@ -23,1 +23,1 @@ from odf.opendocument import OpenDocumentPresentation, OpenDocumentText
-from elementparser import ElementParser
+from .elementparser import ElementParser
--- contrib/python/odfpy/tests/teststyles.py	(index)
+++ contrib/python/odfpy/tests/teststyles.py	(working tree)
@@ -24,1 +24,1 @@ from odf import style, text
-from elementparser import ElementParser
+from .elementparser import ElementParser
--- contrib/python/odfpy/tests/testsubobjects.py	(index)
+++ contrib/python/odfpy/tests/testsubobjects.py	(working tree)
@@ -22,1 +22,1 @@ import unittest, os, zipfile, re
-from elementparser import ElementParser
+from .elementparser import ElementParser
