--- contrib/python/typeguard/tests/mypy/test_type_annotations.py	(index)
+++ contrib/python/typeguard/tests/mypy/test_type_annotations.py	(working tree)
@@ -75,6 +75,8 @@ def chdir_local() -> None:
     os.chdir(os.path.dirname(__file__))
 
 
+# Этот тест ожидает "mypy" в PATH
+@pytest.mark.skip
 @pytest.mark.usefixtures("chdir_local")
 def test_positive() -> None:
     """
@@ -83,6 +85,8 @@ def test_positive() -> None:
     subprocess.check_call(get_mypy_cmd(POSITIVE_FILE))
 
 
+# Этот тест ожидает "mypy" в PATH
+@pytest.mark.skip
 @pytest.mark.usefixtures("chdir_local")
 def test_negative() -> None:
     """
--- contrib/python/typeguard/tests/test_checkers.py	(index)
+++ contrib/python/typeguard/tests/test_checkers.py	(working tree)
@@ -771,7 +771,7 @@ class TestNamedTuple:
 
     def test_type_mismatch(self):
         pytest.raises(TypeCheckError, check_type, ("bob", 1), Employee).match(
-            r"tuple is not a named tuple of type tests.Employee"
+            r"tuple is not a named tuple of type __tests__.Employee"
         )
 
     def test_wrong_field_type(self):
@@ -913,7 +913,7 @@ class TestTypevar:
         check_type(Child(), TParent)
 
     def test_bound_fail(self):
-        with pytest.raises(TypeCheckError, match="is not an instance of tests.Child"):
+        with pytest.raises(TypeCheckError, match="is not an instance of __tests__.Child"):
             check_type(Parent(), TChild)
 
     @pytest.mark.parametrize(
--- contrib/python/typeguard/tests/test_importhook.py	(index)
+++ contrib/python/typeguard/tests/test_importhook.py	(working tree)
@@ -65,5 +65,6 @@ def test_debug_instrumentation(monkeypatch, capsys):
     import_dummymodule()
     out, err = capsys.readouterr()
     path_str = str(dummy_module_path)
-    assert f"Source code of {path_str!r} after instrumentation:" in err
+    # в ya make "path_str" разрешается в подкаталог ~/.ya/build/build_root/...
+    assert f"{path_str!r} after instrumentation:"[1:] in err
     assert "class DummyClass" in err
--- contrib/python/typeguard/tests/test_typechecked.py	(index)
+++ contrib/python/typeguard/tests/test_typechecked.py	(working tree)
@@ -613,6 +613,8 @@ def test_existing_method_decorator():
         assert value == 7
 
 
+# Этот тест не укладывается в 300s
+@pytest.mark.skip
 @pytest.mark.parametrize(
     "flags, expected_return_code",
     [
--- contrib/python/typeguard/tests/test_utils.py	(index)
+++ contrib/python/typeguard/tests/test_utils.py	(working tree)
@@ -9,7 +9,7 @@ from . import Child
     "inputval, add_class_prefix, expected",
     [
         pytest.param(qualified_name, False, "function", id="func"),
-        pytest.param(Child(), False, "tests.Child", id="instance"),
+        pytest.param(Child(), False, "__tests__.Child", id="instance"),
         pytest.param(int, False, "int", id="builtintype"),
         pytest.param(int, True, "class int", id="builtintype_classprefix"),
     ],
