--- contrib/python/django-ajax-selects/ajax_select/urls.py	(index)
+++ contrib/python/django-ajax-selects/ajax_select/urls.py	(working tree)
@@ -1,5 +1,13 @@
-from django.urls import path
+try:
+    from django.urls import path
+    old_django = False
+except ImportError:
+    from django.conf.urls import url
+    old_django = True
 
 from ajax_select import views
 
-urlpatterns = [path("ajax_lookup/<channel>", views.ajax_lookup, name="ajax_lookup")]
+if old_django:
+    urlpatterns = [url(r"^ajax_lookup/(?P<channel>[-\w]+)$", views.ajax_lookup, name="ajax_lookup")]
+else:
+    urlpatterns = [path("ajax_lookup/<channel>", views.ajax_lookup, name="ajax_lookup")]
