--- contrib/python/django-ajax-selects/ajax_select/__init__.py	(index)
+++ contrib/python/django-ajax-selects/ajax_select/__init__.py	(working tree)
@@ -12,3 +12,7 @@ from ajax_select.lookup_channel import LookupChannel  # noqa
 # and any specified in settings.AJAX_LOOKUP_CHANNELS
 # It will do this after all apps are imported.
 from django.apps import AppConfig  # noqa
+from django import VERSION
+
+if VERSION < (3,):
+    default_app_config = "ajax_select.apps.AjaxSelectConfig"
