--- contrib/python/spacy/spacy/cli/init_config.py	(index)
+++ contrib/python/spacy/spacy/cli/init_config.py	(working tree)
@@ -15,1 +15,2 @@ from ._util import init_cli, Arg, Opt, show_validation_error, COMMAND
-RECOMMENDATIONS = srsly.read_yaml(ROOT / "quickstart_training_recommendations.yml")
+import pkgutil
+RECOMMENDATIONS = srsly.yaml_loads(pkgutil.get_data(__package__, "templates/quickstart_training_recommendations.yml"))
--- contrib/python/spacy/spacy/util.py	(index)
+++ contrib/python/spacy/spacy/util.py	(working tree)
@@ -670,8 +670,15 @@ def load_config(
     interpolate (bool): Whether to interpolate and resolve variables.
     RETURNS (Config): The loaded config.
     """
+    import pkgutil
     config_path = ensure_path(path)
     config = Config(section_order=CONFIG_SECTION_ORDER)
+    try:
+        data = pkgutil.get_data(__package__, os.path.basename(path))
+    except:
+        data = None
+    if data:
+        return config.from_str(data.decode("utf-8"), overrides=overrides, interpolate=interpolate)
     if str(config_path) == "-":  # read from standard input
         return config.from_str(
             sys.stdin.read(), overrides=overrides, interpolate=interpolate
