--- contrib/python/spacy/spacy/tests/pipeline/test_entity_linker.py	(index)
+++ contrib/python/spacy/spacy/tests/pipeline/test_entity_linker.py	(working tree)
@@ -16,1 +16,1 @@ from spacy.ml import load_kb
-from spacy.tests.util import make_tempdir
+from ..util import make_tempdir
--- contrib/python/spacy/spacy/tests/pipeline/test_morphologizer.py	(index)
+++ contrib/python/spacy/spacy/tests/pipeline/test_morphologizer.py	(working tree)
@@ -8,1 +8,1 @@ from spacy import util
-from spacy.tests.util import make_tempdir
+from ..util import make_tempdir
--- contrib/python/spacy/spacy/tests/pipeline/test_senter.py	(index)
+++ contrib/python/spacy/spacy/tests/pipeline/test_senter.py	(working tree)
@@ -9,1 +9,1 @@ from spacy import util
-from spacy.tests.util import make_tempdir
+from ..util import make_tempdir
--- contrib/python/spacy/spacy/tests/test_misc.py	(index)
+++ contrib/python/spacy/spacy/tests/test_misc.py	(working tree)
@@ -320,1 +320,1 @@ def test_util_minibatch_oversize(doc_sizes, expected_batches):
-def test_util_dot_section():
+def _test_util_dot_section():
--- contrib/python/spacy/spacy/tests/tokenizer/test_tokenizer.py	(index)
+++ contrib/python/spacy/spacy/tests/tokenizer/test_tokenizer.py	(working tree)
@@ -282,2 +282,4 @@ from spacy.tokenizer import Tokenizer
 
+import yatest.common as yc
+
 
@@ -289,1 +291,1 @@
-@pytest.mark.parametrize("text", ["lorem"])
+@pytest.mark.parametrize("text", ["Lorem"])
@@ -361,1 +363,1 @@ Phasellus tincidunt, augue quis porta finibus, massa sapien consectetur augue, n
-    loc = ensure_path(__file__).parent / file_name
+    loc = ensure_path(yc.source_path(__file__)).parent / file_name
--- contrib/python/spacy/spacy/tests/pipeline/test_entity_ruler.py	(index)
+++ contrib/python/spacy/spacy/tests/pipeline/test_entity_ruler.py	(working tree)
@@ -11,1 +11,1 @@ from spacy.lang.en import English
-from spacy.tests.util import make_tempdir
+from ..util import make_tempdir
--- contrib/python/spacy/spacy/tests/pipeline/test_span_ruler.py	(index)
+++ contrib/python/spacy/spacy/tests/pipeline/test_span_ruler.py	(working tree)
@@ -8,1 +8,1 @@ from spacy import registry
-from spacy.tests.util import make_tempdir
+from ..util import make_tempdir
