--- contrib/python/flask-swagger-ui/flask_swagger_ui/flask_swagger_ui.py	(index)
+++ contrib/python/flask-swagger-ui/flask_swagger_ui/flask_swagger_ui.py	(working tree)
@@ -1,6 +1,9 @@
 import os
 import json
 from flask import Blueprint, send_from_directory, render_template, request
+from flask import render_template_string, abort, Response
+import mimetypes
+import importlib.resources
 
 
 def get_swaggerui_blueprint(
@@ -49,12 +52,14 @@ def get_swaggerui_blueprint(
                     }
                 )
                 fields["config_json"] = json.dumps(default_config)
-            return render_template("index.template.html", **fields)
+            return render_template_string((importlib.resources.files(__package__) / "templates" / "index.template.html").read_text(), **fields)
         else:
-            return send_from_directory(
-                # A bit of a hack to not pollute the default /static path with our files.
-                os.path.join(swagger_ui.root_path, swagger_ui._static_folder),
-                path,
-            )
+            filename = path.split("/")[-1]
+            try:
+                file_content = (importlib.resources.files(__package__) / "dist" / filename).read_bytes()
+            except:
+                abort(404)
+            else:
+                return Response(file_content, mimetype=mimetypes.guess_type(filename)[0])
 
     return swagger_ui
