--- contrib/python/python-docx/docx/api.py	(index)
+++ contrib/python/python-docx/docx/api.py	(working tree)
@@ -12,6 +12,7 @@ import os
 
 from docx.opc.constants import CONTENT_TYPE as CT
 from docx.package import Package
+from docx.utils import get_file
 
 
 def Document(docx=None):
@@ -33,5 +34,4 @@ def _default_docx_path():
     """
     Return the path to the built-in default .docx package.
     """
-    _thisdir = os.path.split(__file__)[0]
-    return os.path.join(_thisdir, 'templates', 'default.docx')
+    return get_file('templates', 'default.docx')
--- contrib/python/python-docx/docx/parts/hdrftr.py	(index)
+++ contrib/python/python-docx/docx/parts/hdrftr.py	(working tree)
@@ -9,6 +9,7 @@ import os
 from docx.opc.constants import CONTENT_TYPE as CT
 from docx.oxml import parse_xml
 from docx.parts.story import BaseStoryPart
+from docx.utils import get_file
 
 
 class FooterPart(BaseStoryPart):
@@ -25,11 +26,11 @@ class FooterPart(BaseStoryPart):
     @classmethod
     def _default_footer_xml(cls):
         """Return bytes containing XML for a default footer part."""
-        path = os.path.join(
-            os.path.split(__file__)[0], '..', 'templates', 'default-footer.xml'
+        f = get_file(
+            'templates',
+            'default-footer.xml'
         )
-        with open(path, 'rb') as f:
-            xml_bytes = f.read()
+        xml_bytes = f.read()
         return xml_bytes
 
 
@@ -47,9 +48,8 @@ class HeaderPart(BaseStoryPart):
     @classmethod
     def _default_header_xml(cls):
         """Return bytes containing XML for a default header part."""
-        path = os.path.join(
-            os.path.split(__file__)[0], '..', 'templates', 'default-header.xml'
+        f = get_file(
+            'templates', 'default-header.xml'
         )
-        with open(path, 'rb') as f:
-            xml_bytes = f.read()
+        xml_bytes = f.read()
         return xml_bytes
--- contrib/python/python-docx/docx/parts/settings.py	(index)
+++ contrib/python/python-docx/docx/parts/settings.py	(working tree)
@@ -15,6 +15,7 @@ from ..opc.packuri import PackURI
 from ..opc.part import XmlPart
 from ..oxml import parse_xml
 from ..settings import Settings
+from ..utils import get_file
 
 
 class SettingsPart(XmlPart):
@@ -45,10 +46,9 @@ class SettingsPart(XmlPart):
         """
         Return a bytestream containing XML for a default settings part.
         """
-        path = os.path.join(
-            os.path.split(__file__)[0], '..', 'templates',
+        f = get_file(
+            'templates',
             'default-settings.xml'
         )
-        with open(path, 'rb') as f:
-            xml_bytes = f.read()
+        xml_bytes = f.read()
         return xml_bytes
--- contrib/python/python-docx/docx/parts/styles.py	(index)
+++ contrib/python/python-docx/docx/parts/styles.py	(working tree)
@@ -15,6 +15,7 @@ from ..opc.packuri import PackURI
 from ..opc.part import XmlPart
 from ..oxml import parse_xml
 from ..styles.styles import Styles
+from ..utils import get_file
 
 
 class StylesPart(XmlPart):
@@ -46,10 +47,9 @@ class StylesPart(XmlPart):
         """
         Return a bytestream containing XML for a default styles part.
         """
-        path = os.path.join(
-            os.path.split(__file__)[0], '..', 'templates',
+        f = get_file(
+            'templates',
             'default-styles.xml'
         )
-        with open(path, 'rb') as f:
-            xml_bytes = f.read()
+        xml_bytes = f.read()
         return xml_bytes
