--- contrib/python/pyviz-comms/pyviz_comms/__init__.py	(index)
+++ contrib/python/pyviz-comms/pyviz_comms/__init__.py	(working tree)
@@ -1,4 +1,5 @@
 import os
+import pkgutil
 import sys
 import uuid
 import traceback
@@ -13,10 +14,8 @@ import param
 
 from ._version import __version__
 
-HERE = os.path.abspath(os.path.dirname(__file__))
-
-with open(os.path.join(HERE, 'labextension', 'package.json')) as fid:
-    data = json.load(fid)
+data_str = pkgutil.get_data('pyviz_comms', 'labextension/package.json')
+data = json.loads(data_str)
 
 def _jupyter_labextension_paths():
     return [{
@@ -31,9 +30,8 @@ except NameError:
   get_ipython = None
 
 # nb_mime_js is used to enable the necessary mime type support in classic notebook
-comm_path = os.path.dirname(os.path.abspath(__file__))
-with open(os.path.join(comm_path, 'notebook.js')) as f:
-    nb_mime_js = '\n\n' + f.read()
+notebook_js_data = pkgutil.get_data('pyviz_comms', 'notebook.js')
+nb_mime_js = nb_mime_js = '\n\n' + notebook_js_data.decode()
 
 
 class extension(param.ParameterizedFunction):
