--- contrib/python/pycountry/pycountry/__init__.py	(index)
+++ contrib/python/pycountry/pycountry/__init__.py	(working tree)
@@ -35 +35 @@ def get_version(distribution_name: str) -> Optional[str]:
-LOCALES_DIR: str = resource_filename("pycountry", "locales")
+LOCALES_DIR: str = os.path.join(os.path.dirname(__file__), "locales")
--- contrib/python/pycountry/pycountry/db.py	(index)
+++ contrib/python/pycountry/pycountry/db.py	(working tree)
@@ -103 +103,3 @@ class Database:
-        with open(self.filename, encoding="utf-8") as f:
+        import os
+        import importlib.resources
+        with (importlib.resources.files(__package__) / "databases" / os.path.basename(self.filename)).open(encoding="utf-8") as f:
