--- contrib/python/numpy/py3/numpy/_core/src/common/numpyos.c	(index)
+++ contrib/python/numpy/py3/numpy/_core/src/common/numpyos.c	(working tree)
@@ -783,7 +783,7 @@ NumPyOS_strtoull(const char *str, char **endptr, int base)
 
 #include <stdlib.h>
 
-#if _MSC_VER >= 1900
+#if _MSC_VER >= 1900 && 0
 /* npy3k_compat.h uses this function in the _Py_BEGIN/END_SUPPRESS_IPH
  * macros. It does not need to be defined when building using MSVC
  * earlier than 14.0 (_MSC_VER == 1900).
--- contrib/python/numpy/py3/numpy/_core/src/common/npy_config.h	(index)
+++ contrib/python/numpy/py3/numpy/_core/src/common/npy_config.h	(working tree)
@@ -129 +129 @@
-#include <cygwin/version.h>
+#error #include <cygwin/version.h>
