--- contrib/python/numpy/py3/numpy/_core/src/umath/scalarmath.c	(index)
+++ contrib/python/numpy/py3/numpy/_core/src/umath/scalarmath.c	(working tree)
@@ -35582,7 +35582,7 @@ long_true_divide(PyObject *a, PyObject *b)
      */
     PyObject *ret;
     npy_float64 arg1, arg2, other_val;
-    npy_long other_val_conv;
+    npy_long other_val_conv = 0;
 
     int is_forward;
     if (Py_TYPE(a) == &PyLongArrType_Type) {
