--- contrib/python/numpy/py3/numpy/_core/src/umath/scalarmath.c.src	(index)
+++ contrib/python/numpy/py3/numpy/_core/src/umath/scalarmath.c.src	(working tree)
@@ -1411,7 +1411,6 @@ static PyObject *
     npy_bool may_need_deferring;
     conversion_result res = convert_to_@name@(
             other, &other_val_conv, &may_need_deferring);
-    other_val = other_val_conv;  /* Need a float value */
     if (res == CONVERSION_ERROR) {
         return NULL;  /* an error occurred (should never happen) */
     }
@@ -1422,6 +1421,7 @@ static PyObject *
         case DEFER_TO_OTHER_KNOWN_SCALAR:
             Py_RETURN_NOTIMPLEMENTED;
         case CONVERSION_SUCCESS:
+            other_val = other_val_conv;  /* Need a float value */
             break;  /* successfully extracted value we can proceed */
         case OTHER_IS_UNKNOWN_OBJECT:
         case PROMOTION_REQUIRED:
--- contrib/python/numpy/py3/numpy/_core/src/umath/scalarmath.c	(index)
+++ contrib/python/numpy/py3/numpy/_core/src/umath/scalarmath.c	(working tree)
@@ -35430,7 +35430,6 @@ int_true_divide(PyObject *a, PyObject *b)
     npy_bool may_need_deferring;
     conversion_result res = convert_to_int(
             other, &other_val_conv, &may_need_deferring);
-    other_val = other_val_conv;  /* Need a float value */
     if (res == CONVERSION_ERROR) {
         return NULL;  /* an error occurred (should never happen) */
     }
@@ -35441,6 +35440,7 @@ int_true_divide(PyObject *a, PyObject *b)
         case DEFER_TO_OTHER_KNOWN_SCALAR:
             Py_RETURN_NOTIMPLEMENTED;
         case CONVERSION_SUCCESS:
+            other_val = other_val_conv;  /* Need a float value */
             break;  /* successfully extracted value we can proceed */
         case OTHER_IS_UNKNOWN_OBJECT:
         case PROMOTION_REQUIRED:
