--- a/toro/__init__.py	(index)
+++ b/toro/__init__.py	(working tree)
@@ -2,7 +2,10 @@ import contextlib
 import heapq
 import collections
 from functools import partial
-from Queue import Full, Empty
+try:
+    from queue import Full, Empty
+except ImportError:
+    from Queue import Full, Empty
 
 import tornado
 from tornado import ioloop
@@ -919,7 +922,7 @@ class RWLock(object):
             a deadline relative to the current time.
         """
         futures = [self._block.acquire(deadline) for _ in
-                   xrange(self._max_readers)]
+                   range(self._max_readers)]
         try:
             managers = yield tornado_multi_future(futures,
                                                   quiet_exceptions=Timeout)
@@ -952,7 +955,7 @@ class RWLock(object):
         """
         if not self.locked():
             raise RuntimeError('release unlocked lock')
-        for i in xrange(self._max_readers):
+        for i in range(self._max_readers):
             self._block.release()
 
     def locked(self):
