--- contrib/python/pycares/ya.make	(index)
+++ contrib/python/pycares/ya.make	(working tree)
@@ -33,6 +33,13 @@ PY_SRCS(
     pycares/utils.py
 )
 
+RUN_PROGRAM(
+    contrib/python/cffi/py3/gen _cffi_src/build_cares.py:ffi pycares._cares.cffi.py3.c
+    IN _cffi_src/build_cares.py
+    OUT pycares._cares.cffi.py3.c
+    OUTPUT_INCLUDES ares.h ares_nameser.h
+)
+
 RESOURCE_FILES(
     PREFIX contrib/python/pycares/
     .dist-info/METADATA
