--- contrib/python/nbconvert/nbconvert/exporters/html.py	(index)
+++ contrib/python/nbconvert/nbconvert/exporters/html.py	(working tree)
@@ -12,6 +12,7 @@ from traitlets import default, Unicode, Bool
 from traitlets.config import Config
 from jupyter_core.paths import jupyter_path
 import jinja2
+import markupsafe
 
 if tuple(int(x) for x in jinja2.__version__.split(".")[:3]) < (3, 0, 0):
     from jinja2 import contextfilter
@@ -148,12 +150,12 @@ class HTMLExporter(TemplateExporter):
         def resources_include_css(name):
             env = self.environment
             code = """<style type="text/css">\n%s</style>""" % (env.loader.get_source(env, name)[0])
-            return jinja2.Markup(code)
+            return markupsafe.Markup(code)
 
         def resources_include_js(name):
             env = self.environment
             code = """<script>\n%s</script>""" % (env.loader.get_source(env, name)[0])
-            return jinja2.Markup(code)
+            return markupsafe.Markup(code)
 
         def resources_include_url(name):
             env = self.environment
@@ -177,7 +179,7 @@ class HTMLExporter(TemplateExporter):
             data = base64.b64encode(data)
             data = data.replace(b'\n', b'').decode('ascii')
             src = 'data:{mime_type};base64,{data}'.format(mime_type=mime_type, data=data)
-            return jinja2.Markup(src)
+            return markupsafe.Markup(src)
         resources = super()._init_resources(resources)
         resources['theme'] = self.theme
         resources['include_css'] = resources_include_css
--- contrib/python/nbconvert/nbconvert/exporters/templateexporter.py    (index)
+++ contrib/python/nbconvert/nbconvert/exporters/templateexporter.py    (working tree)
@@ -7,6 +7,9 @@ that uses Jinja2 to export notebook files into different formats.
 
 from __future__ import print_function, absolute_import
 
+import atexit
+import functools
 import os
+import shutil
 import uuid
 import json
@@ -624,6 +625,7 @@ class TemplateExporter(Exporter):
         root_dirs = []
         if DEV_MODE:
             root_dirs.append(os.path.abspath(os.path.join(ROOT, '..', '..', 'share', 'jupyter')))
+        root_dirs.append(os.path.join(_get_internal_tmpl_data(), 'resfs/file/contrib/python/nbconvert/nbconvert-6.3.0.data/data/share/jupyter'))
         root_dirs.extend(jupyter_path())
         return root_dirs
 
@@ -631,3 +633,20 @@ class TemplateExporter(Exporter):
         resources = super()._init_resources(resources)
         resources['deprecated'] = deprecated
         return resources
+
+@functools.cache
+def _get_internal_tmpl_data():
+    import tempfile
+    import library.python.resource as __res
+
+    tmp_dir = tempfile.mkdtemp(prefix='nbconvert-temp', dir=tempfile.gettempdir())
+
+    for key in __res.iterkeys(b"resfs/file/contrib/python/nbconvert/nbconvert-6.3.0.data"):
+        filename = f"{tmp_dir}/{str(key, 'ascii')}"
+        os.makedirs(os.path.dirname(filename), exist_ok=True)
+        with open(filename, 'wb') as f:
+            f.write(__res.find(key))
+
+    atexit.register(lambda: shutil.rmtree(tmp_dir))
+
+    return tmp_dir
