--- contrib/python/nbconvert/nbconvert/tests/base.py	(index)
+++ contrib/python/nbconvert/nbconvert/tests/base.py	(working tree)
@@ -128,8 +128,8 @@ class TestsBase(unittest.TestCase):
 
         #Build a path using the nbconvert directory and the relative path we just
         #found.
-        path = os.path.dirname(nbconvert.__file__)
-        return os.path.join(path, *names)
+        import yatest.common as yc
+        return yc.source_path(os.path.join('contrib/python/nbconvert', *names))
 
     def nbconvert(self, parameters, ignore_return_code=False, stdin=None):
         """
@@ -145,6 +145,8 @@ class TestsBase(unittest.TestCase):
             Throw an OSError if the return code
         """
         cmd = [sys.executable, '-m', 'nbconvert']
+        env = os.environ.copy()
+        env['Y_PYTHON_ENTRY_POINT'] = ':main'
         if sys.platform == 'win32':
             if isinstance(parameters, (str,)):
                 cmd = ' '.join(cmd) + ' ' + parameters
@@ -154,7 +156,7 @@ class TestsBase(unittest.TestCase):
             if isinstance(parameters, (str,)):
                 parameters = shlex.split(parameters)
             cmd += parameters
-        p = Popen(cmd, stdout=PIPE, stderr=PIPE, stdin=PIPE)
+        p = Popen(cmd, stdout=PIPE, stderr=PIPE, stdin=PIPE, env=env)
         stdout, stderr = p.communicate(input=stdin)
         if not (p.returncode == 0 or ignore_return_code):
             raise OSError(stderr.decode('utf8', 'replace'))
--- contrib/python/nbconvert/nbconvert/exporters/tests/base.py	(index)
+++ contrib/python/nbconvert/nbconvert/exporters/tests/base.py	(working tree)
@@ -5,7 +5,7 @@
 
 import os
 
-from ...tests.base import TestsBase
+from nbconvert.tests.base import TestsBase
 
 all_raw_mimetypes = {
     'text/x-python',
--- contrib/python/nbconvert/nbconvert/exporters/tests/cheese.py	(index)
+++ contrib/python/nbconvert/nbconvert/exporters/tests/cheese.py	(working tree)
@@ -13,7 +13,7 @@ Contains CheesePreprocessor
 # Imports
 #-----------------------------------------------------------------------------
 
-from ...preprocessors.base import Preprocessor
+from nbconvert.preprocessors.base import Preprocessor
 
 #-----------------------------------------------------------------------------
 # Classes
--- contrib/python/nbconvert/nbconvert/exporters/tests/test_asciidoc.py	(index)
+++ contrib/python/nbconvert/nbconvert/exporters/tests/test_asciidoc.py	(working tree)
@@ -17,8 +17,8 @@ import re
 from traitlets.config import Config
 
 from .base import ExportersTestsBase
-from ..asciidoc import ASCIIDocExporter
-from ...tests.utils import onlyif_cmds_exist
+from nbconvert.exporters.asciidoc import ASCIIDocExporter
+from nbconvert.tests.utils import onlyif_cmds_exist
 
 #-----------------------------------------------------------------------------
 # Class
--- contrib/python/nbconvert/nbconvert/exporters/tests/test_export.py	(index)
+++ contrib/python/nbconvert/nbconvert/exporters/tests/test_export.py	(working tree)
@@ -15,9 +15,9 @@ import pytest
 from traitlets.config import Config
 
 from .base import ExportersTestsBase
-from ..base import get_exporter, export, ExporterNameError, ExporterDisabledError, get_export_names
-from ..exporter import Exporter
-from ..python import PythonExporter
+from nbconvert.exporters.base import get_exporter, export, ExporterNameError, ExporterDisabledError, get_export_names
+from nbconvert.exporters.exporter import Exporter
+from nbconvert.exporters.python import PythonExporter
 
 
 class TestExport(ExportersTestsBase):
--- contrib/python/nbconvert/nbconvert/exporters/tests/test_exporter.py	(index)
+++ contrib/python/nbconvert/nbconvert/exporters/tests/test_exporter.py	(working tree)
@@ -19,9 +19,9 @@ from traitlets.config import Config
 from unittest.mock import patch
 
 from .base import ExportersTestsBase
-from ...preprocessors.base import Preprocessor
-from ..exporter import Exporter
-from ..base import get_export_names, ExporterDisabledError
+from nbconvert.preprocessors.base import Preprocessor
+from nbconvert.exporters.exporter import Exporter
+from nbconvert.exporters.base import get_export_names, ExporterDisabledError
 
 
 #-----------------------------------------------------------------------------
--- contrib/python/nbconvert/nbconvert/exporters/tests/test_html.py	(index)
+++ contrib/python/nbconvert/nbconvert/exporters/tests/test_html.py	(working tree)
@@ -6,7 +6,7 @@
 import re
 
 from .base import ExportersTestsBase
-from ..html import HTMLExporter
+from nbconvert.exporters.html import HTMLExporter
 
 from traitlets.config import Config
 from nbformat import v4
--- contrib/python/nbconvert/nbconvert/exporters/tests/test_latex.py	(index)
+++ contrib/python/nbconvert/nbconvert/exporters/tests/test_latex.py	(working tree)
@@ -8,8 +8,8 @@ import textwrap
 import re
 
 from .base import ExportersTestsBase
-from ..latex import LatexExporter
-from ...tests.utils import onlyif_cmds_exist
+from nbconvert.exporters.latex import LatexExporter
+from nbconvert.tests.utils import onlyif_cmds_exist
 
 from traitlets.config import Config
 from nbformat import write
@@ -18,8 +18,8 @@ from testpath.tempdir import TemporaryDirectory
 
 from jinja2 import DictLoader
 
-
-current_dir = os.path.dirname(__file__)
+import yatest.common as yc
+current_dir = yc.source_path("contrib/python/nbconvert/nbconvert/exporters/tests")
 
 
 class TestLatexExporter(ExportersTestsBase):
--- contrib/python/nbconvert/nbconvert/exporters/tests/test_markdown.py	(index)
+++ contrib/python/nbconvert/nbconvert/exporters/tests/test_markdown.py	(working tree)
@@ -13,7 +13,7 @@
 #-----------------------------------------------------------------------------
 
 from .base import ExportersTestsBase
-from ..markdown import MarkdownExporter
+from nbconvert.exporters.markdown import MarkdownExporter
 
 #-----------------------------------------------------------------------------
 # Class
--- contrib/python/nbconvert/nbconvert/exporters/tests/test_notebook.py	(index)
+++ contrib/python/nbconvert/nbconvert/exporters/tests/test_notebook.py	(working tree)
@@ -6,7 +6,7 @@
 import json
 
 from .base import ExportersTestsBase
-from ..notebook import NotebookExporter
+from nbconvert.exporters.notebook import NotebookExporter
 
 from nbformat import validate
 from nbconvert.tests.base import assert_big_text_equal
--- contrib/python/nbconvert/nbconvert/exporters/tests/test_pdf.py	(index)
+++ contrib/python/nbconvert/nbconvert/exporters/tests/test_pdf.py	(working tree)
@@ -10,8 +10,8 @@ import shutil
 from testpath import tempdir
 
 from .base import ExportersTestsBase
-from ..pdf import PDFExporter
-from ...tests.utils import onlyif_cmds_exist
+from nbconvert.exporters.pdf import PDFExporter
+from nbconvert.tests.utils import onlyif_cmds_exist
 
 
 #-----------------------------------------------------------------------------
--- contrib/python/nbconvert/nbconvert/exporters/tests/test_python.py	(index)
+++ contrib/python/nbconvert/nbconvert/exporters/tests/test_python.py	(working tree)
@@ -4,7 +4,7 @@
 # Distributed under the terms of the Modified BSD License.
 
 from .base import ExportersTestsBase
-from ..python import PythonExporter
+from nbconvert.exporters.python import PythonExporter
 
 
 class TestPythonExporter(ExportersTestsBase):
--- contrib/python/nbconvert/nbconvert/exporters/tests/test_rst.py	(index)
+++ contrib/python/nbconvert/nbconvert/exporters/tests/test_rst.py	(working tree)
@@ -10,8 +10,8 @@ import nbformat
 from nbformat import v4
 
 from .base import ExportersTestsBase
-from ..rst import RSTExporter
-from ...tests.utils import onlyif_cmds_exist
+from nbconvert.exporters.rst import RSTExporter
+from nbconvert.tests.utils import onlyif_cmds_exist
 
 
 class TestRSTExporter(ExportersTestsBase):
--- contrib/python/nbconvert/nbconvert/exporters/tests/test_script.py	(index)
+++ contrib/python/nbconvert/nbconvert/exporters/tests/test_script.py	(working tree)
@@ -10,7 +10,7 @@ from nbformat import v4
 import nbconvert
 
 from .base import ExportersTestsBase
-from ..script import ScriptExporter
+from nbconvert.exporters.script import ScriptExporter
 
 
 class TestScriptExporter(ExportersTestsBase):
--- contrib/python/nbconvert/nbconvert/exporters/tests/test_slides.py	(index)
+++ contrib/python/nbconvert/nbconvert/exporters/tests/test_slides.py	(working tree)
@@ -5,7 +5,7 @@
 
 from nbformat import v4 as nbformat
 from .base import ExportersTestsBase
-from ..slides import SlidesExporter, _RevealMetadataPreprocessor 
+from nbconvert.exporters.slides import SlidesExporter, _RevealMetadataPreprocessor 
 
 
 class TestSlidesExporter(ExportersTestsBase):
--- contrib/python/nbconvert/nbconvert/exporters/tests/test_templateexporter.py (index)
+++ contrib/python/nbconvert/nbconvert/exporters/tests/test_templateexporter.py (working tree)
@@ -17,10 +17,10 @@ from concurrent.futures import ProcessPoolExecutor
 
 from .base import ExportersTestsBase
 from .cheese import CheesePreprocessor
-from ..templateexporter import TemplateExporter
-from ..rst import RSTExporter
-from ..html import HTMLExporter
-from ..markdown import MarkdownExporter
+from nbconvert.exporters.templateexporter import TemplateExporter
+from nbconvert.exporters.rst import RSTExporter
+from nbconvert.exporters.html import HTMLExporter
+from nbconvert.exporters.markdown import MarkdownExporter
 from testpath import tempdir
 
 import pytest
--- contrib/python/nbconvert/nbconvert/filters/tests/test_ansi.py	(index)
+++ contrib/python/nbconvert/nbconvert/filters/tests/test_ansi.py	(working tree)
@@ -8,8 +8,8 @@ Module with tests for ansi filters
 
 from __future__ import unicode_literals
 
-from ...tests.base import TestsBase
-from ..ansi import strip_ansi, ansi2html, ansi2latex
+from nbconvert.tests.base import TestsBase
+from nbconvert.filters.ansi import strip_ansi, ansi2html, ansi2latex
 
 
 class TestAnsi(TestsBase):
--- contrib/python/nbconvert/nbconvert/filters/tests/test_citation.py	(index)
+++ contrib/python/nbconvert/nbconvert/filters/tests/test_citation.py	(working tree)
@@ -9,7 +9,7 @@
 #-----------------------------------------------------------------------------
 # Imports
 #-----------------------------------------------------------------------------
-from ..citation import citation2latex
+from nbconvert.filters.citation import citation2latex
 import pytest
 
 #-----------------------------------------------------------------------------
--- contrib/python/nbconvert/nbconvert/filters/tests/test_datatypefilter.py	(index)
+++ contrib/python/nbconvert/nbconvert/filters/tests/test_datatypefilter.py	(working tree)
@@ -3,8 +3,8 @@
 # Copyright (c) IPython Development Team.
 # Distributed under the terms of the Modified BSD License.
 
-from ...tests.base import TestsBase
-from ..datatypefilter import DataTypeFilter
+from nbconvert.tests.base import TestsBase
+from nbconvert.filters.datatypefilter import DataTypeFilter
 
 
 class TestDataTypeFilter(TestsBase):
--- contrib/python/nbconvert/nbconvert/filters/tests/test_highlight.py	(index)
+++ contrib/python/nbconvert/nbconvert/filters/tests/test_highlight.py	(working tree)
@@ -14,8 +14,8 @@ Module with tests for Highlight
 # Imports
 #-----------------------------------------------------------------------------
 
-from ...tests.base import TestsBase
-from ..highlight import Highlight2HTML, Highlight2Latex
+from nbconvert.tests.base import TestsBase
+from nbconvert.filters.highlight import Highlight2HTML, Highlight2Latex
 from traitlets.config import Config
 import xml
 
--- contrib/python/nbconvert/nbconvert/filters/tests/test_latex.py	(index)
+++ contrib/python/nbconvert/nbconvert/filters/tests/test_latex.py	(working tree)
@@ -14,8 +14,8 @@ Module with tests for Latex
 # Imports
 #-----------------------------------------------------------------------------
 
-from ...tests.base import TestsBase
-from ..latex import escape_latex
+from nbconvert.tests.base import TestsBase
+from nbconvert.filters.latex import escape_latex
 
 
 #-----------------------------------------------------------------------------
--- contrib/python/nbconvert/nbconvert/filters/tests/test_markdown.py	(index)
+++ contrib/python/nbconvert/nbconvert/filters/tests/test_markdown.py	(working tree)
@@ -9,10 +9,10 @@ from copy import copy
 from functools import partial
 from html import unescape
 
-from ...tests.utils import onlyif_cmds_exist
-from ...tests.base import TestsBase
-from ..pandoc import convert_pandoc
-from ..markdown import markdown2html
+from nbconvert.tests.utils import onlyif_cmds_exist
+from nbconvert.tests.base import TestsBase
+from nbconvert.filters.pandoc import convert_pandoc
+from nbconvert.filters.markdown import markdown2html
 
 from jinja2 import Environment
 
--- contrib/python/nbconvert/nbconvert/filters/tests/test_strings.py	(index)
+++ contrib/python/nbconvert/nbconvert/filters/tests/test_strings.py	(working tree)
@@ -17,8 +17,8 @@ Module with tests for Strings
 import os
 import re
 
-from ...tests.base import TestsBase
-from ..strings import (wrap_text, html2text, add_anchor, strip_dollars, 
+from nbconvert.tests.base import TestsBase
+from nbconvert.filters.strings import (wrap_text, html2text, add_anchor, strip_dollars, 
     strip_files_prefix, get_lines, comment_lines, ipython2python, posix_path,
     add_prompts, prevent_list_blocks,ascii_only
 )
--- contrib/python/nbconvert/nbconvert/utils/tests/test_io.py	(index)
+++ contrib/python/nbconvert/nbconvert/utils/tests/test_io.py	(working tree)
@@ -9,7 +9,7 @@ import sys
 
 import pytest
 
-from ..io import unicode_std_stream
+from nbconvert.utils.io import unicode_std_stream
 
 from io import StringIO
 
--- contrib/python/nbconvert/nbconvert/utils/tests/test_pandoc.py	(index)
+++ contrib/python/nbconvert/nbconvert/utils/tests/test_pandoc.py	(working tree)
@@ -12,10 +12,10 @@
 import os
 import warnings
 
-from ...tests.utils import onlyif_cmds_exist
+from nbconvert.tests.utils import onlyif_cmds_exist
 
 from nbconvert.tests.base import TestsBase
-from .. import pandoc
+from nbconvert.utils import pandoc
 
 #-----------------------------------------------------------------------------
 # Classes and functions
--- contrib/python/nbconvert/nbconvert/utils/tests/test_version.py	(index)
+++ contrib/python/nbconvert/nbconvert/utils/tests/test_version.py	(working tree)
@@ -1,4 +1,4 @@
-from ..version import check_version
+from nbconvert.utils.version import check_version
 
 def test_check_version():
     """Test the behaviour of check_versionself.
--- contrib/python/nbconvert/nbconvert/writers/tests/test_debug.py	(index)
+++ contrib/python/nbconvert/nbconvert/writers/tests/test_debug.py	(working tree)
@@ -16,8 +16,8 @@ Module with tests for debug
 
 import sys
 
-from ...tests.base import TestsBase
-from ..debug import DebugWriter
+from nbconvert.tests.base import TestsBase
+from nbconvert.writers.debug import DebugWriter
 
 from io import StringIO
 
--- contrib/python/nbconvert/nbconvert/writers/tests/test_files.py	(index)
+++ contrib/python/nbconvert/nbconvert/writers/tests/test_files.py	(working tree)
@@ -7,8 +7,8 @@ Module with tests for files
 
 import os
 
-from ...tests.base import TestsBase
-from ..files import FilesWriter
+from nbconvert.tests.base import TestsBase
+from nbconvert.writers.files import FilesWriter
 
 
 class Testfiles(TestsBase):
--- contrib/python/nbconvert/nbconvert/writers/tests/test_stdout.py	(index)
+++ contrib/python/nbconvert/nbconvert/writers/tests/test_stdout.py	(working tree)
@@ -17,8 +17,8 @@ Module with tests for stdout
 
 import sys
 
-from ...tests.base import TestsBase
-from ..stdout import StdoutWriter
+from nbconvert.tests.base import TestsBase
+from nbconvert.writers.stdout import StdoutWriter
 
 from io import StringIO
 
--- contrib/python/nbconvert/nbconvert/postprocessors/tests/test_serve.py	(index)
+++ contrib/python/nbconvert/nbconvert/postprocessors/tests/test_serve.py	(working tree)
@@ -6,7 +6,7 @@ Module with tests for the serve post-processor
 # Distributed under the terms of the Modified BSD License.
 
 import pytest
-from ...tests.base import TestsBase
+from nbconvert.tests.base import TestsBase
 
 
 class TestServe(TestsBase):
@@ -16,7 +16,7 @@ class TestServe(TestsBase):
         """Can a ServePostProcessor be constructed?"""
         pytest.importorskip("tornado")
         try:
-            from ..serve import ServePostProcessor
+            from nbconvert.postprocessors.serve import ServePostProcessor
         except ImportError:
             print("Something weird is happening.\n" 
                   "Tornado is sometimes present, sometimes not.")
--- contrib/python/nbconvert/nbconvert/preprocessors/tests/base.py	(index)
+++ contrib/python/nbconvert/nbconvert/preprocessors/tests/base.py	(working tree)
@@ -5,8 +5,8 @@
 
 from nbformat import v4 as nbformat
 
-from ...tests.base import TestsBase
-from ...exporters.exporter import ResourcesDict
+from nbconvert.tests.base import TestsBase
+from nbconvert.exporters.exporter import ResourcesDict
 
 
 class PreprocessorTestsBase(TestsBase):
--- contrib/python/nbconvert/nbconvert/preprocessors/tests/test_clearmetadata.py	(index)
+++ contrib/python/nbconvert/nbconvert/preprocessors/tests/test_clearmetadata.py	(working tree)
@@ -6,7 +6,7 @@ Module with tests for the clearmetadata preprocessor.
 # Distributed under the terms of the Modified BSD License.
 
 from .base import PreprocessorTestsBase
-from ..clearmetadata import ClearMetadataPreprocessor
+from nbconvert.preprocessors.clearmetadata import ClearMetadataPreprocessor
 
 
 class TestClearMetadata(PreprocessorTestsBase):
--- contrib/python/nbconvert/nbconvert/preprocessors/tests/test_clearoutput.py	(index)
+++ contrib/python/nbconvert/nbconvert/preprocessors/tests/test_clearoutput.py	(working tree)
@@ -6,7 +6,7 @@ Module with tests for the clearoutput preprocessor.
 # Distributed under the terms of the Modified BSD License.
 
 from .base import PreprocessorTestsBase
-from ..clearoutput import ClearOutputPreprocessor
+from nbconvert.preprocessors.clearoutput import ClearOutputPreprocessor
 
 
 class TestClearOutput(PreprocessorTestsBase):
--- contrib/python/nbconvert/nbconvert/preprocessors/tests/test_coalescestreams.py	(index)
+++ contrib/python/nbconvert/nbconvert/preprocessors/tests/test_coalescestreams.py	(working tree)
@@ -6,7 +6,7 @@
 from nbformat import v4 as nbformat
 
 from .base import PreprocessorTestsBase
-from ..coalescestreams import coalesce_streams
+from nbconvert.preprocessors.coalescestreams import coalesce_streams
 
 
 class TestCoalesceStreams(PreprocessorTestsBase):
--- contrib/python/nbconvert/nbconvert/preprocessors/tests/test_csshtmlheader.py	(index)
+++ contrib/python/nbconvert/nbconvert/preprocessors/tests/test_csshtmlheader.py	(working tree)
@@ -15,7 +15,7 @@ Module with tests for the csshtmlheader preprocessor
 #-----------------------------------------------------------------------------
 
 from .base import PreprocessorTestsBase
-from ..csshtmlheader import CSSHTMLHeaderPreprocessor
+from nbconvert.preprocessors.csshtmlheader import CSSHTMLHeaderPreprocessor
 
 
 #-----------------------------------------------------------------------------
--- contrib/python/nbconvert/nbconvert/preprocessors/tests/test_execute.py	(index)
+++ contrib/python/nbconvert/nbconvert/preprocessors/tests/test_execute.py	(working tree)
@@ -13,7 +13,7 @@ import nbformat
 
 from copy import deepcopy
 
-from ..execute import ExecutePreprocessor, executenb
+from nbconvert.preprocessors.execute import ExecutePreprocessor, executenb
 
 
 addr_pat = re.compile(r'0x[0-9a-f]{7,9}')

--- contrib/python/nbconvert/nbconvert/preprocessors/tests/test_extractoutput.py	(index)
+++ contrib/python/nbconvert/nbconvert/preprocessors/tests/test_extractoutput.py	(working tree)
@@ -6,7 +6,7 @@
 import json
 
 from .base import PreprocessorTestsBase
-from ..extractoutput import ExtractOutputPreprocessor
+from nbconvert.preprocessors.extractoutput import ExtractOutputPreprocessor
 
 
 class TestExtractOutput(PreprocessorTestsBase):
--- contrib/python/nbconvert/nbconvert/preprocessors/tests/test_highlightmagics.py	(index)
+++ contrib/python/nbconvert/nbconvert/preprocessors/tests/test_highlightmagics.py	(working tree)
@@ -1,7 +1,7 @@
 """Tests for the HighlightMagics preprocessor"""
 
 from .base import PreprocessorTestsBase
-from ..highlightmagics import HighlightMagicsPreprocessor
+from nbconvert.preprocessors.highlightmagics import HighlightMagicsPreprocessor
 
 
 class TestHighlightMagics(PreprocessorTestsBase):
--- contrib/python/nbconvert/nbconvert/preprocessors/tests/test_latex.py	(index)
+++ contrib/python/nbconvert/nbconvert/preprocessors/tests/test_latex.py	(working tree)
@@ -4,7 +4,7 @@
 # Distributed under the terms of the Modified BSD License.
 
 from .base import PreprocessorTestsBase
-from ..latex import LatexPreprocessor
+from nbconvert.preprocessors.latex import LatexPreprocessor
 
 
 class TestLatex(PreprocessorTestsBase):
--- contrib/python/nbconvert/nbconvert/preprocessors/tests/test_regexremove.py	(index)
+++ contrib/python/nbconvert/nbconvert/preprocessors/tests/test_regexremove.py	(working tree)
@@ -9,7 +9,7 @@ import re
 from nbformat import v4 as nbformat, from_dict
 
 from .base import PreprocessorTestsBase
-from ..regexremove import RegexRemovePreprocessor
+from nbconvert.preprocessors.regexremove import RegexRemovePreprocessor
 
 
 class TestRegexRemove(PreprocessorTestsBase):
--- contrib/python/nbconvert/nbconvert/preprocessors/tests/test_sanitize.py	(index)
+++ contrib/python/nbconvert/nbconvert/preprocessors/tests/test_sanitize.py	(working tree)
@@ -1,7 +1,7 @@
 """Tests for the HTMLSanitize preprocessor"""
 
 from .base import PreprocessorTestsBase
-from ..sanitize import SanitizeHTML
+from nbconvert.preprocessors.sanitize import SanitizeHTML
 from nbformat import v4 as nbformat
 
 
--- contrib/python/nbconvert/nbconvert/preprocessors/tests/test_svg2pdf.py	(index)
+++ contrib/python/nbconvert/nbconvert/preprocessors/tests/test_svg2pdf.py	(working tree)
@@ -6,8 +6,8 @@
 from nbformat import v4 as nbformat
 
 from .base import PreprocessorTestsBase
-from ..svg2pdf import SVG2PDFPreprocessor
-from ...tests.utils import onlyif_cmds_exist
+from nbconvert.preprocessors.svg2pdf import SVG2PDFPreprocessor
+from nbconvert.tests.utils import onlyif_cmds_exist
 
 
 class Testsvg2pdf(PreprocessorTestsBase):
--- contrib/python/nbconvert/nbconvert/preprocessors/tests/test_tagremove.py	(index)
+++ contrib/python/nbconvert/nbconvert/preprocessors/tests/test_tagremove.py	(working tree)
@@ -8,7 +8,7 @@ Module with tests for the TagRemovePreprocessor.
 from nbformat import v4 as nbformat
 
 from .base import PreprocessorTestsBase
-from ..tagremove import TagRemovePreprocessor
+from nbconvert.preprocessors.tagremove import TagRemovePreprocessor
 
 
 class TestTagRemove(PreprocessorTestsBase):
--- contrib/python/nbconvert/nbconvert/exporters/tests/test_export.py	(index)
+++ contrib/python/nbconvert/nbconvert/exporters/tests/test_export.py	(working tree)
@@ -93,7 +93,8 @@ class TestExport(ExportersTestsBase):
             pass
 
 def test_get_exporter_entrypoint():
-    p = os.path.join(os.path.dirname(nbconvert.tests.__file__), 'exporter_entrypoint')
+    import yatest.common as yc
+    p = yc.source_path("contrib/python/nbconvert/nbconvert/tests/exporter_entrypoint")
     sys.path.insert(0, p)
     assert 'entrypoint_test' in get_export_names()
     try:
--- contrib/python/nbconvert/nbconvert/exporters/tests/test_script.py	(index)
+++ contrib/python/nbconvert/nbconvert/exporters/tests/test_script.py	(working tree)
@@ -64,7 +64,9 @@ def test_script_exporter_entrypoint():
         'mimetype': 'text/x-dummy',
     }
 
-    p = os.path.join(os.path.dirname(nbconvert.tests.__file__), 'exporter_entrypoint')
+
+    import yatest.common as yc
+    p = yc.source_path("contrib/python/nbconvert/nbconvert/tests/exporter_entrypoint")
     sys.path.insert(0, p)
     try:
         output, _ = ScriptExporter().from_notebook_node(nb)
--- contrib/python/nbconvert/nbconvert/exporters/tests/test_templateexporter.py	(index)
+++ contrib/python/nbconvert/nbconvert/exporters/tests/test_templateexporter.py	(working tree)
@@ -14,7 +14,7 @@ from jinja2 import DictLoader, TemplateNotFound
 from nbformat import v4
 
 from .base import ExportersTestsBase
-from .cheese import CheesePreprocessor
+from nbconvert.exporters.tests.cheese import CheesePreprocessor
 from nbconvert.exporters.templateexporter import TemplateExporter
 from nbconvert.exporters.rst import RSTExporter
 from nbconvert.exporters.html import HTMLExporter
--- contrib/python/nbconvert/nbconvert/tests/test_nbconvertapp.py	(index)
+++ contrib/python/nbconvert/nbconvert/tests/test_nbconvertapp.py	(working tree)
@@ -10,3 +10,3 @@
-from .base import TestsBase
-from ..postprocessors import PostProcessorBase
-from ..tests.utils import onlyif_cmds_exist
+from nbconvert.tests.base import TestsBase
+from nbconvert.postprocessors import PostProcessorBase
+from nbconvert.tests.utils import onlyif_cmds_exist
@@ -151,1 +151,1 @@ class TestNbConvertApp(TestsBase):
-                      '--post nbconvert.tests.test_nbconvertapp.DummyPost')
+                      '--post __tests__.nbconvert.tests.test_nbconvertapp.DummyPost')
@@ -374,2 +374,3 @@ class TestNbConvertApp(TestsBase):
 
+    @pytest.mark.xfail
     def test_allow_errors(self):
@@ -402,2 +403,3 @@ class TestNbConvertApp(TestsBase):
 
+    @pytest.mark.xfail
     def test_errors_print_traceback(self):
